{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import json\n",
    "import pandas as pd\n",
    "import yaml\n",
    "from itertools import chain\n",
    "import os\n",
    "import random"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_file_names(path):\n",
    "    # get list of papers .json from path\n",
    "    file_names = []\n",
    "    for file_name in os.listdir(path):\n",
    "        file_names.append(file_name)\n",
    "    return file_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def generate_sha(used=[]):\n",
    "    hash = ''\n",
    "    while (hash == 'nan') or (hash in used):\n",
    "        hash = \"%032x\" % random.getrandbits(128)\n",
    "    return hash"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Classes\n",
    "\n",
    "- Paper => BiorxivPaper, (and others)\n",
    "- Scraper => BiorxivScraper (and others)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "code_folding": [
     1,
     5,
     15,
     24,
     28,
     32,
     45,
     56
    ]
   },
   "outputs": [],
   "source": [
    "class Paper:\n",
    "    def __init__(self, source):\n",
    "        self.source = source\n",
    "        self.paper = {} # contains paper data\n",
    "    \n",
    "    def import_json(self, path):\n",
    "        ''' If exists, saves JSON article to file'''\n",
    "        if os.path.exists(path):\n",
    "            with open(path) as json_file:\n",
    "                self.paper = json.load(json_file)\n",
    "            self.string = self.to_str()\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "    \n",
    "    def save_json(self, path):\n",
    "        ''' Import paper from json file '''\n",
    "        if self.paper and os.path.exists(path):\n",
    "            with (path, 'w') as outfile:\n",
    "                json.dump(self.paper, outfile)\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "    \n",
    "    def to_html(self):\n",
    "        ''' Convert article to HTML '''\n",
    "        return '<html></html>'\n",
    "    \n",
    "    def save_html(self, path):\n",
    "        ''' If exists, saves HTML article to file '''\n",
    "        return True\n",
    "    \n",
    "    def ref_to_str(self, reference):\n",
    "        ''' Convert reference to string '''\n",
    "        output = ''\n",
    "        if reference['label'] != '':\n",
    "            output = reference['label'] + ' - '\n",
    "        if reference['title'] != '':\n",
    "            output += reference['title'] + ', '\n",
    "        if reference['authors'] != []:\n",
    "            output += ', '.join(reference['authors'])\n",
    "        if reference['date'] != None:\n",
    "            output += reference['date']\n",
    "        return output\n",
    "\n",
    "    def graph_to_str(self, graph, indent=''):\n",
    "        ''' Converts a graphic element: table or figure to string '''\n",
    "        output = '\\n'\n",
    "        output += indent + '-'*100 + '\\n'\n",
    "        output += indent + graph['id'] + ' | ' + graph['label'] + '\\n'\n",
    "        output += indent + graph['img'] + '\\n'\n",
    "        output += indent + graph['description']\n",
    "        output += indent + '-'*100\n",
    "        output += '\\n'\n",
    "        return output\n",
    "    \n",
    "    def section_to_str(self, section, depth=0):\n",
    "        indent = '  '*depth\n",
    "        output = indent + section['title']\n",
    "        for content in section['content']:\n",
    "            if type(content) == dict:\n",
    "                if 'title' in content.keys():\n",
    "                    # is section = explore further \n",
    "                    output += '\\n'\n",
    "                    output += self.section_to_str(content, depth+1)\n",
    "                elif 'id' in content.keys():\n",
    "                    # is fig or table, get string\n",
    "                    if content['type'] == 'table':\n",
    "                        output += '\\n'\n",
    "                        output += self.graph_to_str(self.paper['tables'][content['id']], indent) # content['id']\n",
    "                        # insert table description from paper['tables']\n",
    "                    elif content['type'] == 'figure':\n",
    "                        output += ' \\n'\n",
    "                        output += self.graph_to_str(self.paper['figures'][content['id']], indent) # content['id']\n",
    "                        # insert figure description from paper['figures']\n",
    "                    else:\n",
    "                        print('content not recognized: {}'.format(content))\n",
    "            elif type(content) == str:\n",
    "                output += '\\n' + indent + content\n",
    "        return output\n",
    "    \n",
    "    def to_str(self, appendix=False):\n",
    "        ''' Convert article to string '''\n",
    "        ''' Converts a paper to string formart '''\n",
    "        output = '({} | {} | {})\\n'.format(self.paper['source'], self.paper['id'], self.paper['url'])\n",
    "        output += self.paper['title'] + '\\n'\n",
    "        output += ', '.join(self.paper['authors']) + '\\n'\n",
    "        for section in self.paper['sections']:\n",
    "            if (not appendix) and ('appendix' in section['title'].lower()):\n",
    "                pass\n",
    "            else:\n",
    "                output += self.section_to_str(section)\n",
    "                output += '\\n'\n",
    "        output += '\\n\\n'.join([self.ref_to_str(ref) for ref in self.paper['references']])    \n",
    "        return output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0,
     1,
     6,
     11,
     15,
     30,
     52,
     76,
     82,
     105,
     156
    ]
   },
   "outputs": [],
   "source": [
    "class BiorxivScraper():\n",
    "    def __init__(self, papers):\n",
    "        self.source = 'biorxiv'\n",
    "        self.session = requests.session()\n",
    "        self.base_url = ''\n",
    "    \n",
    "    def get_paper_soup(self, url):\n",
    "        ''' Get paper soup from source '''\n",
    "        html = self.session.get(url)\n",
    "        self.soup = BeautifulSoup(html.content, 'html.parser')\n",
    "    \n",
    "    def paper_from_soup(self):\n",
    "        ''' Extract paper data from soup '''\n",
    "        return \n",
    "    \n",
    "    def get_authors(authors_html):\n",
    "        ''' HTML list of authors '''\n",
    "        authors = []\n",
    "        for author in authors_html:\n",
    "            try:\n",
    "                given_name = author.find('span',attrs={'class':'nlm-given-names'}).text\n",
    "            except:\n",
    "                given_name = ''\n",
    "            try:\n",
    "                surname = author.find('span',attrs={'class':'nlm-surname'}).text\n",
    "            except:\n",
    "                surname = ''\n",
    "            authors.append(given_name + ' ' + surname)\n",
    "        return authors\n",
    "    \n",
    "    def get_tables(html_tables, base_url):\n",
    "        ''' Get tables from HTML '''\n",
    "        tables = {}\n",
    "        for html_table in html_tables:\n",
    "            table = {}\n",
    "\n",
    "            table['id'] = html_table['id']\n",
    "            table_label = html_table.find('span',attrs={'class':'table-label'})\n",
    "            if table_label:\n",
    "                table['label'] = table_label.text\n",
    "\n",
    "            table_caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "            if table_caption:\n",
    "                table['caption'] = table_caption.text\n",
    "\n",
    "            table['description'] = ''\n",
    "            for p in html_table.find_all('p'):\n",
    "                table['description'] += p.text + '\\n'\n",
    "            table['img'] = '{}/{}.large.jpg'.format(base_url,table['id'])\n",
    "            tables[table['id']] = table\n",
    "        return tables\n",
    "    \n",
    "    def get_biorxiv_figures(figures_html):\n",
    "        ''' Get figures from HTML '''\n",
    "        figures = {}\n",
    "        for html_figure in figures_html:\n",
    "            figure = {}\n",
    "            figure['id'] = html_figure['id']\n",
    "            figure_label = html_figure.find('span',attrs={'class':'fig-label'})\n",
    "            if figure_label:\n",
    "                figure['label'] = figure_label.text\n",
    "\n",
    "            figure_title = html_figure.find('span',attrs={'class':'caption-title'})\n",
    "            if figure_title:\n",
    "                figure['title'] = figure_title.text\n",
    "\n",
    "            figure['description'] = ''\n",
    "            figure_ps = html_figure.find_all('p')\n",
    "            for p in figure_ps:\n",
    "                figure['description'] += p.text + '\\n'\n",
    "            figure_img = html_figure.find('a')\n",
    "            if figure_img:\n",
    "                figure['img'] = figure_img['href'].split('?')[0]\n",
    "            figures[figure['id']] = figure\n",
    "        return figures\n",
    "    \n",
    "    def process_paragraph(paragraph):\n",
    "        ''' Removes and modifies html tags in paragraphs '''\n",
    "        for anchor in paragraph.find_all('a'):\n",
    "            anchor.replace_with('[' + anchor['href'].strip('#') + ',\"' + anchor.text + '\"]')\n",
    "        return paragraph.text\n",
    "    \n",
    "    def explore_section(root_tag):\n",
    "        ''' Recursively explore section and all subsections for content '''\n",
    "        content = {}\n",
    "        content['title'] = ''\n",
    "        content['content'] = []\n",
    "        for child in root_tag.findChildren(recursive=False):\n",
    "            if child.name == 'h2' or child.name == 'h3' or child.name == 'h4':\n",
    "                content['title'] = child.text\n",
    "            elif child.name == 'p':\n",
    "                content['content'].append(process_paragraph(child))\n",
    "            if child.name == 'div':\n",
    "                if 'class' in child.attrs:\n",
    "                    if 'subsection' in child['class']:\n",
    "                        # if subsection explore recursively\n",
    "                        content['content'].append(explore_section(child))\n",
    "                if 'id' in child.attrs.keys():\n",
    "                    if 'T' in child['id']:\n",
    "                        content['content'].append({'id':child['id'],'type':'table'})\n",
    "                    elif 'F' in child['id']:\n",
    "                        content['content'].append({'id':child['id'],'type':'figure'})\n",
    "        else:\n",
    "            return content\n",
    "    \n",
    "    def get_references(references_html):\n",
    "        ''' HTML list of references '''\n",
    "        references = []\n",
    "        for ref in references_html:\n",
    "            reference = {}\n",
    "            # label\n",
    "            try:\n",
    "                reference['label'] = ref.find('span',attrs={'class':'ref-label'}).text\n",
    "            except:\n",
    "                reference['label'] = None\n",
    "\n",
    "            # authors\n",
    "            author_names = []\n",
    "            try:\n",
    "                authors = ref.find('cite').find_all('span',attrs={'class':'cit-auth'})\n",
    "                for author in authors:\n",
    "                    surname = author.find('span',attrs={'class':'cit-name-surname'}).text\n",
    "                    given_names = author.find('span',attrs={'class':'cit-name-given-names'}).text\n",
    "                    author_name = surname + ' ' + given_names\n",
    "                    author_names.append(author_name)\n",
    "                reference['authors'] = author_names\n",
    "            except:\n",
    "                reference['authors'] = author_names\n",
    "\n",
    "            # date\n",
    "            try:\n",
    "                reference['date'] = ref.find('cite').find('span',attrs={'class':'cit-pub-date'}).text\n",
    "            except:\n",
    "                reference['date'] = None\n",
    "\n",
    "            # title\n",
    "            try:\n",
    "                reference['title'] = ref.find('cite').find('span',attrs={'class':'cit-article-title'}).text\n",
    "            except:\n",
    "                reference['title'] = ''\n",
    "\n",
    "            # links\n",
    "            ref_links = []\n",
    "            try:\n",
    "                links = ref.find('div',attrs={'class':'cit-extra'}).find_all('a')\n",
    "                for link in links:\n",
    "                    if link.text != 'OpenUrl':\n",
    "                        link_name = link.text\n",
    "                        link_url = link['href']\n",
    "                        ref_links.append((link_name,link_url))\n",
    "                reference['links'] = ref_links\n",
    "            except:\n",
    "                reference['links'] = ref_links\n",
    "            references.append(reference)\n",
    "        return references\n",
    "    \n",
    "    def get_data_from_soup(self):\n",
    "        '''\n",
    "            Extract paper information from html webpage of biorxiv\n",
    "        '''\n",
    "        content_url = self.soup.find('meta',attrs={'name':'citation_pdf_url'})['content'].strip('.full.pdf')\n",
    "\n",
    "        paper = {}\n",
    "        paper['title'] = self.soup.h1.text\n",
    "\n",
    "        authors_html = self.soup.find('span',attrs={'class':'highwire-citation-authors'}) \\\n",
    "                           .find_all('span',attrs={'class':'highwire-citation-author'})\n",
    "        paper['authors'] = get_authors(authors_html)\n",
    "\n",
    "        sections_html = self.soup.find_all('div',attrs={'class':'section'})\n",
    "        paper['sections'] = get_sections(sections_html)\n",
    "\n",
    "        tables_html = self.soup.find_all('div',attrs={'class':'table'})\n",
    "        paper['tables'] = get_tables(tables_html, content_url)\n",
    "\n",
    "        # get figures\n",
    "        figures_html = self.soup.find_all('div',attrs={'class':'fig'})\n",
    "\n",
    "\n",
    "        paper['figures'] = get_figures(figures_html)\n",
    "\n",
    "        # get references\n",
    "        references_html = self.soup.find('div',attrs={'class':'section ref-list'}).find_all('li')\n",
    "        paper['references'] = get_references(references_html)\n",
    "\n",
    "        return paper\n",
    "    \n",
    "    def scrape_paper(self, url):\n",
    "        self.get_paper_soup()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_paper = Paper('biorxiv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "string indices must be integers",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-246-2f0f81fa5546>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mtest_paper\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mimport_json\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'data/biorxiv/01e3b313e78a352593be2ff64927192af66619b5.json'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-243-bed6dabd8f72>\u001b[0m in \u001b[0;36mimport_json\u001b[1;34m(self, path)\u001b[0m\n\u001b[0;32m     13\u001b[0m             \u001b[1;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mjson_file\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpaper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mjson\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mload\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mjson_file\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstring\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mto_str\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-243-bed6dabd8f72>\u001b[0m in \u001b[0;36mto_str\u001b[1;34m(self, appendix)\u001b[0m\n\u001b[0;32m     91\u001b[0m         \u001b[0moutput\u001b[0m \u001b[1;33m+=\u001b[0m \u001b[1;34m', '\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'authors'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m'\\n'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     92\u001b[0m         \u001b[1;32mfor\u001b[0m \u001b[0msection\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'sections'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 93\u001b[1;33m             \u001b[1;32mif\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;32mnot\u001b[0m \u001b[0mappendix\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mand\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;34m'appendix'\u001b[0m \u001b[1;32min\u001b[0m \u001b[0msection\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'title'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     94\u001b[0m                 \u001b[1;32mpass\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     95\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m: string indices must be integers"
     ]
    }
   ],
   "source": [
    "test_paper.import_json('data/biorxiv/01e3b313e78a352593be2ff64927192af66619b5.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'source': 'biorxiv',\n",
       " 'id': '01e3b313e78a352593be2ff64927192af66619b5',\n",
       " 'url': 'https://biorxiv.org/content/10.1101/029397v1.full',\n",
       " 'title': 'Viruses are a dominant driver of protein adaptation in mammals',\n",
       " 'authors': ['David Enard', 'Le Cai', 'Carina Gwenapp', 'Dmitri A. Petrov'],\n",
       " 'sections': {'Summary': 'Viruses interact with hundreds to thousands of proteins in mammals, yet adaptation against viruses has only been studied in a few proteins specialized in antiviral defense. Whether adaptation to viruses typically involves only specialized antiviral proteins or affects a broad array of proteins is unknown. Here, we analyze adaptation in ~1,300 virus-interacting proteins manually curated from a set of 9,900 proteins conserved across mammals. We show that viruses (i) use the more evolutionarily constrained proteins from the cellular functions they hijack and that (ii) despite this high constraint, virus-interacting proteins account for a high proportion of all protein adaptation in humans and other mammals. Adaptation is elevated in virus-interacting proteins across all functional categories, including both immune and non-immune functions. Our results demonstrate that viruses are one of the most dominant drivers of evolutionary change across mammalian and human proteomes.\\n',\n",
       "  'Introduction': 'A number of virus-interacting proteins (VIPs) with a specialized role in antiviral defense have been shown to have exceptionally high rates of adaptation (Cagliani et al., 2012; Cagliani et al., 2011; Elde et al., 2009; Fumagalli et al., 2010; Kerns et al., 2008; Liu et al., 2005; Sawyer et al., 2004, 2007; Sawyer et al., 2005; Sironi et al., 2012; Vasseur et al., 2011). One example is protein kinase R (PKR), which recognizes viral double-stranded RNA upon infection, halts translation, and as a result blocks viral replication (Elde et al., 2009). PKR is one of the fastest adaptively evolving proteins in mammals. Specific amino acid changes in PKR have been shown to be associated with an arms race against viral decoys for the control of translation (Elde et al., 2009).\\nHowever, PKR and other fast-evolving antiviral defense VIPs are not representative of the hundreds to thousands of other VIPs. Viruses enter and hijack the cells of their hosts by physically interacting with hundreds to thousands of host proteins. Most VIPs are not specialized in antiviral defense, and do not even have any known role in immunity. Many VIPs instead have key functions in basic cellular processes subverted by viruses, and viruses tend to interact with proteins that are functionally important hubs in the protein-protein interaction network of the host (Dyer et al., 2008; Halehalli and Nagarajaram, 2015). It is plausible that many VIPs might evolve to limit the impact of the viruses on the host. However, it is unknown whether the war against viruses is fought by a “professional” army of specifically antiviral proteins, or whether it is a global war fought by a broad range of VIPs.\\nOne reason to believe that the war against viruses might not affect evolution of a broad array of VIPs is that, contrary to the pattern observed for specifically antiviral proteins, most VIPs evolve unusually slowly (Davis et al., 2015; Jager et al., 2012). Furthermore, very few cases of adaptation to viruses have been studied outside of fast evolving, specialized antiviral proteins. Transferrin receptor or TFRC is the most notable exception, as a striking example of a non-immune, housekeeping protein hijacked by viruses (Demogines et al., 2013; Kaelber et al., 2012). TFRC is responsible for iron uptake in many different cell types and is used as a cell surface receptor by diverse viruses in rodents and carnivores. TFRC has repeatedly evaded binding by viruses through recurrent adaptive amino acid changes. As such, it is the only known clear case of a host protein with adaptation in response to viruses that is not specialized in the antiviral response. Whether TFRC represents an exception or the rule for VIPs is currently unknown.\\nWhether the slower evolution of VIPs primarily reflects lower rates of adaptation, greater evolutionary constraint, or both is currently unknown. Here we analyze patterns of evolutionary constraint and adaptation in a manually curated, high quality set of ~1,300 VIPs. The vast majority of these VIPs (80%) have no known antiviral or any other more broadly defined immune activity. We confirm that VIPs tend to evolve slowly and show that this is because VIPs experience stronger evolutionary constraint than other proteins within the same functional categories. Most importantly, we show that despite this greater evolutionary constraint, VIPs have experienced much higher rates of adaptation in humans and other mammals compared to other proteins. This excess of adaptation is visible in VIPs involved in most biological functions hijacked by viruses (such as transcription or signal transduction), showing that adaptation in response to viruses extends well beyond the usual, specialized antiviral proteins. We demonstrate that viruses have driven a substantial proportion of all adaptations across the human and mammalian proteomes, establishing that the war against viruses does indeed affect the proteome as a whole.\\nWe finally showcase the power of our global scan for adaptation in VIPs by studying the case of aminopeptidase N, a well-known multifunctional enzyme (Mina-Osorio, 2008) used by coronaviruses as a receptor (Delmas et al., 1992; Yeager et al., 1992). Using our approach we reach an amino-acid level understanding of parallel adaptive evolution in aminopeptidase N in response to coronaviruses in a wide range of mammals.\\n',\n",
       "  'Results': 'Here we analyze patterns of both adaptive evolution and evolutionary constraint/purifying selection in a large set of 1,256 manually curated VIPs from the low-throughput virology literature (Methods and Table S1 available online). We exclude interactions identified by high-throughput experiments as we are concerned about a high rate of false positives (Mellacheruvu et al., 2013). These 1,256 VIPs were annotated from an initial set of 9,861 proteins with orthologs in 24 mammals with high quality genomes (Figure S1, Table S2 and Methods). VIPs in our dataset interact with viral proteins, viral RNA, or viral DNA. Most of them (95%) correspond to an interaction between a human protein and a virus infecting humans (Table S1). Human Immunodeficiency Virus type 1 (HIV-1) is the best-represented virus with 240 VIPs, with nine other viruses having at least 50 VIPs (Table S1).\\nThis dataset represents the largest, most up-to-date set of VIPs backed up by individual low-throughput publications. Nonetheless, given that many VIPs were discovered only recently, with half of all publications reporting VIPs published in the past 7 years (Figure S2), it is likely that additional VIPs remain to be discovered.\\nThese 1,256 VIPs are involved in very diverse cellular processes, as shown by their representation in the Gene Ontology (GO) annotation system of cellular biological processes (2015; Ashburner et al., 2000) (Table S3) such as transcription (354 VIPs), post-translational protein modification (224 VIPs), signal transduction (396 VIPs), apoptosis (185 VIPs), and transport (264 VIPs). The supracellular processes notably include defense response (103 VIPs) and developmental processes (327 VIPs). Many VIPs have no known specifically antiviral immune activity (defined here as any activity restricting viral replication, Methods), and only 241 or 20% of VIPs have any known immune function (Methods and Table S4). These 241 immune VIPs include the VIPs classified as antiviral (Table S4) throughout this manuscript. In total, 162 overlapping GO cellular and supracellular processes have more than 50 VIPs (Table S3). These observations confirm that viruses interact with proteins involved in the majority of basic cellular processes.\\n',\n",
       "  'Patterns of purifying selection in VIPs': 'We confirm the observations of several recent studies suggesting that VIPs tend to evolve slowly (Davis et al., 2015; Jager et al., 2012). The VIPs have ~15% lower mammals-wide dN/dS ratio on average compared to non-VIPs (0.124 versus 0.145, 95% CI [0.136,0.148]; Methods). The difference in dN/dS is highly significant as shown by a permutation test where VIPs are compared with the same number of randomly chosen non-VIPs many times (simple permutation test P=0 after 109 iterations; Table S2).\\nTo disentangle whether the slower evolution of VIPs is due to stronger purifying selection or to a lower rate of adaptation, we use the ratio of non-synonymous polymorphisms to synonymous polymorphisms pN/pS rather than the dN/dS ratio. Unlike dN/dS that is strongly influenced by both the effects of purifying selection and adaptation, pN/pS is primarily determined by the efficiency of purifying selection in removing deleterious non-synonymous mutations.\\nGenome-wide polymorphisms required to measure pN/pS at the scale of the proteome have become available for humans (Abecasis et al., 2012) (1,000 Genomes Project) (Table S5), and chimpanzee, gorilla, and orangutans (Prado-Martinez et al., 2013) (Great Apes Genome Project) (Table S6). The 1,000 Genomes Project and the Great Apes Genome Project are complementary for this analysis. On the one hand, the 1,000 Genomes Project provides high quality variants with frequencies estimated from a large number of individuals. On the other hand the Great Ape Genome project includes fewer individuals, but provides substantial pN and pS counts for more genes than the 1000 genomes data. This is because the great apes populations are more polymorphic than modern human populations, and are less affected by the noise due to genetic drift and strong bottlenecks. Thus, although the 1,000 genomes are well suited for approaches requiring precise frequency estimates (such as the McDonald-Kreitman test implemented below (Messer and Petrov, 2013)), the Great Ape genome data is well suited for the estimation of the pN/pS ratio in as many proteins as possible. Specifically, we measure pN/pS as the average across non-human great apes (or as the average in the 1,000 Genomes African populations; Supplemental Methods) using the data from the largest chimpanzee, gorilla, and orangutan populations in order to further limit the noise in the estimation of the strength of purifying selection due to drift and bottlenecks (Prado-Martinez et al., 2013). We use only great ape pN and pS in the analyses that involve subsamples of proteins in order to retain sufficient statistical power.\\nIn human African populations from the 1,000 Genomes project (Supplemental Methods), the average pN/pS is 21% lower in VIPs compared to non-VIPs (0.759 versus 0.966, 95% CI [0.92,1.01], simple permutation test P=0 after 109 iterations). This suggests that in the human lineage VIPs have been under stronger purifying selection than non-VIPs. In line with this, VIPs also show an excess of low frequency (≤10%) deleterious non-synonymous variants compared to non-VIPs (Figure S3). In great apes, the average pN/pS ratio is 25% lower in VIPs compared to non-VIPs (0.526 versus 0.697, 95% CI [0.66,0.72], simple permutation test P=0 after 109 iterations). The distribution of pN/pS in VIPs is globally skewed towards lower values compared to non-VIPs (Figure 1A). The difference in pN/pS in great apes between VIPs and non-VIPs is robust to a number of potentially confounding factors, including gene length, GC content and recombination (Table S7). These results show that VIPs do experience stronger purifying selection than non-VIPs. Finally, stronger purifying selection acting on the VIPs is widespread and is not limited to VIPs interacting with any one particular virus (Figure 1B).\\nA) Distribution of pN/pS in VIPs (blue) and non-VIPs (pink). The blue curve is the density curve of pN/(pS+1) for 1,256 VIPs. We use pN/(pS+1) instead of pN/pS to account for those coding sequences where pS=0. pN and pS are measured using great ape genomes from the Great Ape Genome Project (Supplemental Methods). The pink area represents the superimposition of the density curves for each of 5,000 sets of randomly sampled non-VIPs. B) Average pN/pS in VIPs (blue dot) versus average pN/pS in non-VIPs (red dot and red 95% confidence interval) within ten viruses with more than 50 VIPs The number between parentheses is the number of VIPs for each virus. KSHV: Kaposi’s Sarcoma Herpesvirus. HIV-1: Human Immunodeficiency Virus type 1. HBV: Hepatitis B Virus. ADV: Adenovirus. HPV: Human Papillomavirus. HSV: Herpes Simplex Virus. EBV: Epstein-Barr Virus. Influenza: Influenza Virus. HTLV: Human T-lymphotropic Virus. HCV: Hepatitis C virus. C) Same as B), but for the 20 most high level GO processes with the highest number of VIPs. The full GO process name for “protein modification” as written in the figure is “post-translational protein modification”. D) Asymptotic MK test (Supplemental Methods) for the proportion of adaptive amino acid substitutions (a) in VIPs (blue dots and curve) and non-VIPs (red dots and curve). Pink area: superposition of fitted logarithmic curves for 5,000 random sets of 1,256 non-VIPs (as many as VIPs) where the estimated a falls within a’s 95% confidence interval. E) Classic MK test (Supplemental Methods) for VIPs (blue dot) and non-VIPs (red dot and 95% confidence interval) for the ten viruses with 50 or more VIPs. F) Same as E) but for the 20 top high level GO processes with the most VIPs below the dotted black line. Above the dotted black line: the classic MK test for all VIPs, for non-immune VIPs and for immune VIPs (Table S4). See also Tables S3, S4, S5, S6, S7, S8 and S9 and Figures S3 and S4\\nThe higher level of purifying selection in VIPs might be due to the fact that VIPs participate in the more constrained host functions, or, alternatively, because within each specific host function, viruses tend to interact with the more constrained proteins. In order to assess these two non-mutually exclusive scenarios we generated 104 control sets of non-VIPs chosen to be in the same 162 Gene Ontology processes as VIPs (GO processes with more than 50 VIPs; Table S3 and Methods).\\nIn great apes, GO-matched non-VIPs still have a much higher pN/pS ratio compared to VIPs, suggesting that VIPs tend to be more conserved than non-VIPs from the same GO category. On average, pN/pS in the GO-matched non-VIPs is 0.647 (95% CI [0.621,0.674]). This is only slightly lower than the average ratio in non-VIPs in general (pN/pS=0.697, P=2×10−3), but much higher than the average ratio in VIPs (0.526, permutation test P=0 after 109 iterations). The stronger purifying selection acting on VIPs is apparent within most functions. Figure 1C shows stronger purifying selection in the 20 high level GO categories with the most VIPs. In all the 20 GO categories pN/pS is lower in VIPs than in non-VIPs, and the difference is significant for 17 of these categories (Table S3). This shows that within a wide range of host functions, viruses tend to target the most conserved proteins.\\nInterestingly, even immune VIPs (Table S4) have a significantly reduced pN/pS ratio compared to immune non-VIPs (Figure 1C), which suggests that immune proteins in direct contact with viruses are more constrained. The reduction in pN/pS in non-immune VIPs (no antiviral or any other immune function, Table S4) is very similar to the reduction observed in the entire set of VIPs (Figure 1C). Table S3 further shows stronger purifying selection in 124 of the 162 GO categories (77%) with more than 50 VIPs. These results suggest that viruses target the most constrained proteins within many diverse host functions, rather than specifically targeting only those host functions that are under strong evolutionary constraint.\\n',\n",
       "  'Frequent adaptation in VIPs in the human lineage': 'Our pN/pS analysis demonstrates that viruses tend to interact with proteins that are under greater evolutionarily constraint. However, this does not necessarily imply that VIPs experience less adaptation than non-VIPs. We estimate the proportion of adaptive non-synonymous substitutions (noted α) in VIPs and non-VIPs in the human lineage by using the McDonald-Kreitman test (MK test) (Methods). We first use the classic MK test (McDonald and Kreitman, 1991) with the 1,000 Genomes Project polymorphism data from African populations (Supplemental Methods and Table S5). Since VIPs are more constrained than non-VIPs and tend to have more non-synonymous deleterious low frequency variants than non-VIPs (Figures 1 and S3), we limit the effect of deleterious variants by excluding all variants with a derived allele frequency lower than 10% (Supplemental Methods) (Charlesworth and Eyre-Walker, 2008; Eyre-Walker and Keightley, 2009; Keightley and Eyre-Walker, 2007; Messer and Petrov, 2013). We find that α is strongly elevated in VIPs compared to non-VIPs (α=0.19 in VIPs versus -0.02 in non-VIPs, permutation test P=2.×10−5). This shows that VIPs have a substantial excess of adaptation compared to non-VIPs.\\nThe classic MK test is known to be biased downward by the presence of slightly deleterious non-synonymous variants and this bias is difficult to eliminate fully even by excluding low frequency variants (Messer and Petrov, 2013). Note that our application of the classic MK test to discover the higher rate of adaptation in VIPs compared to non-VIPs is conservative given that the VIPs have a higher proportion of slightly deleterious non-synonymous variants and thus should show a stronger downward bias in the estimation of α (Figure S3). However, the nominal estimates of α=0.19 and -0.02 in VIPs and non-VIPs are likely underestimates of the true proportions of adaptive amino acid changes.\\nWe therefore apply an asymptotic modification of the MK test known to provide estimates of α without a downward bias in the presence of slightly deleterious variants (Messer and Petrov, 2013). To further validate the asymptotic MK test we carry out extensive population simulations (Messer, 2013) to show that this test is indeed robust to a number of potential biases (Supplemental Methods and Table S8).\\nUsing the asymptotic MK test we estimate that in VIPs, ~27% of the 1,897 amino acid substitutions along the human lineage were adaptive (Figure 1D). This proportion is three times higher than the estimated proportion of ~9% in non-VIPs (Figure 1D). Thus, although VIPs represent only 13% of the orthologs in our dataset, we estimate that in human evolution they account for almost 30% of all adaptive amino-acid changes. Note that both VIPs and non-VIPs in our dataset are limited to the proteins conserved across all mammals.\\nThe high α in VIPs is not explained by higher rates of adaptation in the host GO processes where VIPs are well represented (Figure S4 and Methods). Furthermore, the large difference in α observed between VIPs and non-VIPs is robust to a number of potentially confounding factors such as recombination, GC content or gene length (Table S9 and Supplemental Methods). The lower pN/pS in VIPs does not explain their higher α either (Table S9).\\nWe further use the classic MK test (excluding variants below 10%) to investigate the excess of adaptation for the specific VIPs of ten human viruses and in the 20 high level GO categories with the most VIPs (Figure 1E and F). We do not use the asymptotic MK test because it does not have sufficient precision when using a small number of proteins (Supplemental Methods). Although the small number of proteins interacting with individual viruses precludes precise estimates of α (see the large confidence intervals on Figure 1E), the VIPs show nominally higher values of α for eight out of 10 viruses, with HIV-1 and Hepatitis B Virus (HBV) displaying statistically significant increases in adaptation. Likewise, VIPs in most GO categories show higher rates of adaptation (14 out of 20) with 9 of 14 showing statistically significant increases (Figure 1F).\\nFinally and importantly, the 80% of VIPs with no known antiviral or broader immune function (Table S4) have a strongly increased rate of adaptation according to both the classic MK test (α=0.26 in VIPs versus -0.02 in non-VIPs, permutation test P=3×10−7; Figure 1F) and the asymptotic MK test, with the latter estimating α=38% in non-immune VIPs against only 11% for non-immune non-VIPs. Intriguingly, unlike for non-immune VIPs or all VIPs considered together (top of Figure 1F), immune VIPs, including antiviral VIPs (Table S4), do not show any increase of adaptation compared to immune non-VIPs. We speculate that this pattern might reflect the masking effect of balancing selection within immune genes that would bias the MK tests downward (Cagliani et al., 2012). These results show that viruses have been a major selective pressure in human evolution, and have had an influence on the rate of adaptation in the human proteome that extends far beyond antiviral defense proteins, and on a much greater scale than previously appreciated.\\n',\n",
       "  'Frequent adaptation in VIPs across mammals': 'The increased rate of adaptation in VIPs in the human lineage strongly suggests that VIPs in our dataset, 95% of which interact with modern viruses (Table S1), were also VIPs during past human evolution. It is also plausible that a substantial proportion of the VIPs we study are also VIPs in multiple mammalian lineages. Indeed, viruses infecting humans (including the ten viruses with the most VIPs) are known to have close viral relatives in many other mammals, with the exception of Hepatitis C Virus (HCV) for which only distant relatives are known and primarily in bats (Quan et al., 2013). There is also growing evidence that distantly related viruses tend to interact with overlapping sets of host proteins (Davis et al., 2015; Jager et al., 2012). We thus hypothesize that VIPs, while identified primarily in humans, may have also experienced frequent adaptation in mammals in general, with the possible exception of the VIPs interacting with HCV.\\nTo test this hypothesis we use the Branch-Site Random Effect Likelihood test (BS-REL test) (Kosakovsky Pond et al., 2011) available in the HYPHY package (Pond et al., 2005) to detect episodes of adaptive evolution in each of the 44 branches of the mammalian tree used for the analysis (Methods). For a specific coding sequence, the BS-REL test estimates the proportion of codons where the rate of non-synonymous substitutions is higher than the rate of synonymous substitutions (dN/dS>1), which is a hallmark of adaptive evolution. The BS-REL test then compares two competing models of evolution, one with adaptive substitutions and one without adaptive substitutions, and decides which of the two models is the best fit. For each branch of the tree, the BS-REL test provides a P-value that corresponds to the probability that no adaptation occurred in the branch. The product of P-values across all branches in the tree then gives the probability that no adaptation occurred anywhere along the entire tree (Supplemental Methods).\\nThe product of P-values is a good measure of whether a specific protein experienced adaptation in the history of mammalian evolution. In addition to presence/absence of adaptation, we assess the amount of adaptation experienced by a particular protein by estimating the average proportion of selected codons along all mammalian branches. For example we measure that on average the antiviral protein PKR has had 4.6% of positively selected codons across the 44 branches of the mammalian tree.\\nWe compare the proportion of selected codons detected by the BS-REL test between VIPs and non-VIPs. The statistical power of the BS-REL test has been shown to depend strongly on the amount of constraint in a coding sequence, with higher constraint/purifying selection decreasing the ability to detect adaptation (Kosakovsky Pond et al., 2011). We confirm this in our dataset by observing a strong positive correlation between the pN/pS ratio in great apes and the proportion of selected codons across mammals estimated by the BS-REL test (Spearman’s rank correlation ρ = 0.34, P<2×10−16, n=9,861). We therefore use a purifying selection-wise permutation test that matches VIPs and non-VIPs with similar pN/pS ratios in order to compare VIPs and non-VIPs that experience similar levels of purifying selection and providing us with similar power to detect adaptation (Supplemental Methods and Figures S5 and S6).\\nThe purifying selection-wise permutation test shows that adaptation has been much more common in VIPs than in non-VIPs across mammals (Figure 2). We estimate that VIPs have experienced 77% more adaptation compared to non-VIPs (Figure 2A). In total, this represents ~76,000 more adaptive amino acid changes in VIPs compared to non-VIPs. We further use an increasingly strict level of evidence for the presence of adaptation, by including only proteins with increasingly low products of P-values; that is, increasingly low probability that no adaptation occurred (Figure 2A). Figure 2A shows that VIPs with the strongest evidence of adaptation (product of P-values lower than 10−9) have a ~200% excess of strong signals of adaptation. This excess of adaptation in VIPs across mammals is due to i) more VIPs with signals of adaptation than non-VIPs, ii) more branches of the tree per VIP showing adaptation, and iii) a greater proportion of codons evolving adaptively per branch (Figure S7).\\nA) Thick white curve: average excess of adaptation in all VIPs. Narrow white curves: 95% confidence interval for the excess of adaptation in all VIPs. The grey background represents the density distributions of the excess of adaptation based on 5,000 iterations of the purifying selection-wise permutation test (Supplemental Methods). Thick black curve: excess of adaptation in non-immune VIPs. Narrow black curve: lower bound of the 95% confidence interval for the excess of adaptation in non-immune VIPs. B) Virus-by-virus excess of adaptation in VIPs. Black dot is the average excess and the represented interval is the 95% confidence interval. C) Excess of adaptation within the top 20 high level GO processes with the most VIPs. D) Proportions of selected codons in VIPs (blue dot) and non-VIPs (red dot and 95% confidence interval) in the mammalian clades represented by more than one species in the tree. All: entire tree. Primata: primates. Glires: rodents and rabbit. Cetartyodactyla: sheep, cow, pig. Zooamata: carnivores and horse. See also Tables S4 and S4 and Figures S5, S6, S7, S8 and S9.\\nAs mentioned above, HCV stands out among the ten viruses with the largest number of VIPs in humans in that it has no known close viral relatives despite extensive screening of diverse mammalian species (Quan et al., 2013). If this reflects a true lack of close viral relatives of HCV, then we predict a limited excess of adaptation in HCV VIPs. In line with this prediction, the 109 VIPs of HCV are the only ones where we do not detect any excess of adaptation (Figure 2B) despite being one of the largest groups of VIPs. VIPs that interact with all other viruses all show substantial elevation of adaptation (Figure 2B).\\nGO processes with a strong excess of adaptation include cellular processes such as transcription, signal transduction, apoptosis, or post-translational protein modification, but also supracellular processes related to development (Figure 2C and Table S3). Importantly, VIPs with no known immune function (Table S4) show a 40% excess of adaptation, and a ~130% excess of adaptation in the proteins with the highest evidence of adaptation (Figure 2A). These results show that the arms race with viruses has strongly increased the rate of adaptation in a wide range of VIPs.\\nSince 95% of the VIPs were discovered for viruses infecting humans, it is possible that the observed excess of adaptation in VIPs in mammals is due to higher rates of adaptation exclusively in the primate branches of the mammalian tree (Figure S1). However, all mammalian clades in the tree show a similar excess of adaptation in VIPs (Figure 2D). Primates stand out due to their low overall proportions of positively selected codons compared to the other mammalian clades in the tree (Figure 2D). This is most likely due to a lower statistical power of the BS-REL test in the short primate branches (Kosakovsky Pond et al., 2011). In line with this, VIPs with strong signals of adaptation show such signals in all the mammalian clades represented (Figure 3). This includes well-known antiviral VIPs (Figure 3A), antiviral VIPs where adaptation was previously unknown (Figure 3B), and non-antiviral VIPs with diverse, well-studied functions in the mammalian hosts (Figure 3C). This phylogenetically widespread excess of adaptation implies that many of the VIPs annotated in humans were also VIPs for a substantial evolutionary time in a wide range of mammals.\\nA) Signals of adaptation in eight antiviral proteins with well-known adaptation. Red: BS-REL P≤0.001. Orange: BS-REL P≤0.05. Yellow: BS-REL P≤0.1. B) Top eight antiviral proteins with the highest number of branches under selection, and no previously know adaptive evolution. C) Top eight non-antiviral proteins with well-known functions and the highest number of branches under selection. Proteins are ordered according to the number of branches with signals of adaptation. All proteins in the figure have their product of BS-REL test P-values lower than 10−9.\\n',\n",
       "  'From global patterns of adaptation to understanding specific instances of adaptation to viruses: the case of coronaviruses and aminopeptidase N': 'We have shown that rates of adaptation are globally elevated in VIPs in humans and mammals in general, suggesting the existence of tens of thousands of isolated events of adaptations to a diverse range of viruses. Here, we test if our global approach has enough power to isolate new specific cases of adaptation to viruses by looking for instances where viruses are the plausible cause of adaptation in a VIP with no known antiviral activity. This is particularly relevant because, to our knowledge, the transferrin receptor is the only well understood case of a non-antiviral protein adapting in response to viruses (Demogines et al., 2013).\\nTo identify a new non-antiviral protein we first exclude all VIPs with a well-known antiviral activity (Table S4) and then select all remaining VIPs with strong overall evidence of adaptation (Table S10) and at least 10 branches with signals of adaptation. Because we want to understand how adaptation to viruses proceeded, we then select proteins with i) at least one available tertiary structure, ii) amino acid level resolution of the interaction with one or more viruses, and iii) host tropism.\\nThe most positively selected non-antiviral VIP that fulfills all these requirements is aminopeptidase N, abbreviated ANPEP, APN or CD13 (Mina-Osorio, 2008). The analysis of a phylogenetic tree including 84 mammals (Table S11) confirms pervasive adaptation of ANPEP across mammals, with 76 out of 165 branches in the tree evolving adaptively (Figure 4A). ANPEP is a cell-surface enzyme well known for its surprisingly wide range and diversity of functions (Mina-Osorio, 2008). It is used by group I coronaviruses as a receptor, including the Human Coronavirus 229E (HCoV-229E) (Yeager et al., 1992), Transmissible Gastroenteritis Virus (TGEV) (Delmas et al., 1992), Feline Coronavirus (FCoV) (Tresnan and Holmes, 1998), Canine Coronavirus (CCoV) (Tusell et al., 2007), Porcine Respiratory Coronavirus (PRCV) (Delmas et al., 1993) and Porcine Epidemic Diarrhea Virus (PEDV) (Oh et al., 2003). Reguera et al. (Reguera et al., 2012) have solved the structure of porcine ANPEP bound together with TGEV and PRCV. The authors identified in the extracellular domain of ANPEP 22 amino acids that form a surface of contact with TGEV and PRCV (Figure 4B) (Pettersen et al., 2004). The most important component of this contact surface for host tropism is a N-glycosylation site at position 736 in porcine ANPEP (orthologous position 739 in human ANPEP) that forms hydrogen bonds with TGEV and PRCV (Reguera et al., 2012; Tusell et al., 2007). Deleting this site abolishes the ability of TGEV and PRCV to bind porcine ANPEP (Reguera et al., 2012). Adding the glycosylation site in human ANPEP that natively lacks it transforms it into a receptor for TGEV and PRCV (Reguera et al., 2012).\\nA) BS-REL test results for ANPEP in a tree of 84 mammalian species. Legend is on the figure. B) Contact surface with PRCV and TGEV on ANPEP structure (PDB 4FYQ). The figure includes visualizations of all the six different faces of ANPEP. Legend is in the figure. C) Excess of adaptation in and near the contact interface with PRCV and TGEV. Within the contact interface plus a given number of neighboring amino acids (one, five, ten or 20 in the figure), adaptation excess (y axis) is defined as the number of observed codons with a MEME P-value lower than the P-value threshold on the x axis, divided by the average number of codons under the same P-value threshold obtained after randomizing the location of adaptation signals over the entire ANPEP coding sequence 5,000 times. Dark red curve: adaptation excess within the contact interface with TGEV and PRCV plus one neighboring amino acid. Red curve: plus five neighboring amino acids. Orange: plus ten neighboring amino acids. Light orange: plus 20 neighboring amino acids. Numbers in the figure represent the number of adapting codons, and the stars give the significance of the excess. One star: excess P≤0.05. Two stars: P≤0.01. D) Losses and gains of the N-glycosylation across the mammalian phylogeny. See also Tables S10 and S11.\\nWe use the MEME test from HYPHY (Murrell et al., 2012) to identify codons in ANPEP that were under episodic adaptive evolution in mammals. MEME detects significant adaptation (MEME P≤0.05) in 85 of the 931 aligned codons. Interestingly, several of these adaptively evolving codons are within, or right next to the surface of contact with TGEV and PRCV (Figure 4B,C). The codons in contact with TGEV and PRCV and their neighbors are strongly enriched in adaptation compared to ANPEP codons as a whole (Figure 4C). This enrichment fades very rapidly as one gets further from the surface of contact with TGEV and PRCV, which shows that it is due to the interaction with viruses, and not to a more diffuse, less specific enrichment within a wider segment of ANPEP (Figure 4C).\\nAdaptively evolving codons in the contact surface with TGEV and PRCV most notably include two codons within the consensus motif for the N-glycosylation site responsible for host tropism (Figure 4B). N-glycosylation is governed by a three amino acids consensus, NX[ST], where X can be anything except proline (Bause, 1983). The first and third positions in the consensus evolved adaptively in mammals (MEME P=0.005 for both). The ancestral states of the two positions shows that the mammalian ancestor had a fully functional consensus for N-glycosylation, and that the consensus was lost independently 11 times in mammals, either by modification of the first or by modification of the third position (Figure 4D). The consensus was regained only two times after loss (Figure 4D). This means that the signals of adaptation detected at the first and third positions in the consensus mainly reflect parallel, adaptive losses of the N-glycosylation site in multiple mammalian lineages. Given the crucial role of this N-glycosylation site in the binding of TGEV, PRCV, FCoV and CCoV to ANPEP, it is very likely that these parallel adaptive losses were the result of the selective pressure exerted by ancient coronaviruses.\\n',\n",
       "  'Discussion': 'Here, we have shown that viruses have been a major selective pressure in the evolution of the mammalian proteome. Indeed viruses appear to drive ~30% of all adaptive amino acid changes in the conserved part of the human proteome, as evidenced by the MK test. Furthermore, the footprints of the arms race with viruses are visible in a large number of VIPs, and in a broad range of mammals. Importantly, we find a substantial enrichment in strong signals of adaptation in VIPs with no known antiviral or other immune functions (Figures 1F and 2A). Instead adaptation to viruses is visible in VIPs with a very diverse range of functions including such core functions as transcription or signal transduction (Figure 2C and Table S3). Our results thus draw a broader picture where the war against viruses is a global war that involves not only the specialized soldiers of the antiviral response, but also the entire population of host proteins that come into contact with viruses. The best-known case of a housekeeping protein having adapted in response to viruses, the transferrin receptor, may in this respect represent the rule more than an exception (Demogines et al., 2013).\\nAlthough we already find a strong signal of increased adaptation, the amount of adaptive evolution that can be attributed to viruses is probably underestimated by our analysis. First, there may still be many undiscovered VIPs. Within the past few years, there has been no sign that the pace of discovery of new VIPs is slowing down (Figure S2). A slowdown would happen if we were getting close to the point where only few, hard to discover VIPs remain to be found. This means that a substantial number of proteins classified as non-VIPs in this analysis actually are VIPs. This makes the current non-VIPs a conservative control.\\nSecond, adaptation in response to viruses is most likely not restricted to proteins that physically interact with viruses. For example, adaptation to viruses might happen in proteins that act downstream of VIPs in signaling cascades, or in non-coding sequences that regulate the expression of VIPs.\\nThird, not all of the 1,256 VIPs we use here have been consistently interacting with viruses during evolution. Most VIPs in the dataset (95%) were discovered in humans, and how frequently these VIPs have also been interacting with viruses in other mammals is currently unknown. Some VIPs like PKR have probably been in very frequent contact with diverse viruses. Conversely, other VIPs may have been in contact with viruses for a very limited evolutionary time in mammals, and only in a limited range of lineages. This would apply to VIPs that interact with viruses with a limited host range and few other phylogenetically closely related viruses, as is the case with HCV (Figure 2B).\\nOur analysis greatly extends the prospects for paleovirology by making it possible to detect past infections with great statistical power (Emerman and Malik, 2010; Patel et al., 2011). Indeed, globally increased adaptation in VIPs means that we have far more statistical power to study the adaptive footprints viruses have left in host proteins. This is a very complementary approach to the usual scans for integrated viral genomes in host genomes, especially for viruses that do not infect the germ line or for viruses that integrate in the host genome only rarely (Katzourakis and Gifford, 2010). In the dataset analyzed here, only HCV does not show any excess of adaptation in mammals. Two of the ten viruses listed in Figure 2B, HIV-1 and HBV, have VIPs with highly increased adaptation in humans. In other words, these results demonstrate that viruses related to HIV-1 and HBV have repeatedly infected humans since their divergence with chimpanzees. These results show that it should be possible to detect the footprint of past infections by specific viruses in specific evolutionary lineages with great statistical power.\\nGlobally increased adaptation in VIPs also raises the question of pleiotropy. To what extent have adaptive amino acid changes in VIPs affected their native functions in the host? We show that there has been so much adaptation in VIPs that it is very hard to imagine that none of these adaptive events had any consequences on host phenotypes. Interestingly, VIPs tend to be multifunctional proteins. Indeed they represent 13% of all the orthologs in the analysis, 33% of the orthologs with 60 or more annotated GO processes, and 40% of orthologs with 100 or more GO processes (Figure S8A). Pleiotropy is more likely in proteins with many functions (He and Zhang, 2006), and the subset of VIPs with many annotated GO processes has an excess of adaptation that is very similar to the one observed when using all VIPs (Figure S8B). Adaptation to viruses could thus have affected the evolution of host phenotypes in unexpected ways. In this respect, it is particularly intriguing that VIPs have experienced highly increased rates of adaptation within host functions such as development or neurogenesis (Table S3).\\nFinally, our results might explain a puzzling observation that rates of protein adaptation appear relatively invariant across different biological functions (Bierne and Eyre-Walker, 2004). Because viruses (and likely other pathogens) interact with diverse proteins across most GO functions, they elevate the rate of adaptation across the whole proteome in a way that appears independent of specific functions in the GO analysis. We argue that grouping genes together based on the way they interact with diverse pathogens or other environmental stimuli might be a profitable way for discerning the nature of selective pressures that have molded animal genomes.\\nIn conclusion, our analysis demonstrates that viruses have exerted a very powerful selective pressure across the breadth of the mammalian proteome, and suggests the possibility that pathogens in general are the key driver of protein adaptation in mammals and likely other lineages and might have driven many pleiotropic effects on diverse biological functions.\\n',\n",
       "  'Methods': 'We identified 1,256 VIPs out of a total of 9,861 proteins with orthologs in the genomes of the 24 mammals included in the analysis (Figure S1 and Tables S1 and S2). Annotation of the 1,256 VIPs was performed by querying PUBMED for publications reporting interactions found by low-throughput methods (Gray et al., 2015) (http://www.ncbi.nlm.nih.gov/pubmed). We completed our own annotations of 982 VIPs with 274 additional proteins identified with low throughput methods listed in the VirHostNet (http://virhostnet.prabi.fr/) (Guirimand et al., 2015) and HPIDB (http://www.agbase.msstate.edu/hpi/main.html) (Kumar and Nanduri, 2010) databases. See Supplemental Methods for details.\\nWe aligned 9,861 orthologous Ensembl coding sequences (Flicek et al., 2012) from 24 mammals using a combination of Blat (Kent, 2002) and PRANK (Fletcher and Yang, 2010; Jordan and Goldman, 2012; Loytynoja and Goldman, 2008). See Supplemental Methods for details.\\nIn addition to annotating VIPs, we also annotated which VIPs have known antiviral or broader immune activities. See Supplemental Methods for details.\\nWe use the classic and the asymptotic McDonald-Kreitman tests to compare adaptation in VIPs and non-VIPs in human. See Supplemental Methods for details.\\nWe use the CDS alignments of the 24 mammals to quantify adaptation in their phylogeny. To do so we use the Branch-Site REL test (Kosakovsky Pond et al., 2011) from the HYPHY package (Pond et al., 2005). See Supplemental Methods for details.\\nWe created a permutation test that compares VIPs and non-VIPs with the same amount of purifying selection. This is achieved by using the pN/pS ratio as a proxy for purifying selection. See Supplemental Methods for details.\\nWe analyzed patterns of adaptation in ANPEP in a tree of mammals including 84 species. These species are the ones with annotated, known or predicted mRNAs (Table S11 for their Genbank identifiers). The coding sequences were extracted from the mRNAs and aligned with PRANK.\\nWe created a permutation scheme that compares VIPs with random samples of non-VIPs with the same representation of GO categories. See Supplemental Methods for details.\\n',\n",
       "  'Author Contributions': 'DE and DAP conceived and designed the experiments. DE, LC and CG performed the experiments. DE and DAP interpreted the results. DE and DAP wrote the paper.\\n',\n",
       "  'Acknowledgements': 'We thank Kerry Samerotte, Sandeep Venkataram, Emily Ebel and Pleuni Pennings for comments on the manuscript. This work is funded by NIH grants R01GM089926 and R01GM097415 to DAP.\\n',\n",
       "  'Footnotes': '↵* Contact information: denard{at}stanford.edu\\n'},\n",
       " 'tables': [],\n",
       " 'figures': [{'label': 'Figure 1.',\n",
       "   'title': 'Patterns of purifying selection and human adaptation in VIPs',\n",
       "   'description': 'A) Distribution of pN/pS in VIPs (blue) and non-VIPs (pink). The blue curve is the density curve of pN/(pS+1) for 1,256 VIPs. We use pN/(pS+1) instead of pN/pS to account for those coding sequences where pS=0. pN and pS are measured using great ape genomes from the Great Ape Genome Project (Supplemental Methods). The pink area represents the superimposition of the density curves for each of 5,000 sets of randomly sampled non-VIPs. B) Average pN/pS in VIPs (blue dot) versus average pN/pS in non-VIPs (red dot and red 95% confidence interval) within ten viruses with more than 50 VIPs The number between parentheses is the number of VIPs for each virus. KSHV: Kaposi’s Sarcoma Herpesvirus. HIV-1: Human Immunodeficiency Virus type 1. HBV: Hepatitis B Virus. ADV: Adenovirus. HPV: Human Papillomavirus. HSV: Herpes Simplex Virus. EBV: Epstein-Barr Virus. Influenza: Influenza Virus. HTLV: Human T-lymphotropic Virus. HCV: Hepatitis C virus. C) Same as B), but for the 20 most high level GO processes with the highest number of VIPs. The full GO process name for “protein modification” as written in the figure is “post-translational protein modification”. D) Asymptotic MK test (Supplemental Methods) for the proportion of adaptive amino acid substitutions (a) in VIPs (blue dots and curve) and non-VIPs (red dots and curve). Pink area: superposition of fitted logarithmic curves for 5,000 random sets of 1,256 non-VIPs (as many as VIPs) where the estimated a falls within a’s 95% confidence interval. E) Classic MK test (Supplemental Methods) for VIPs (blue dot) and non-VIPs (red dot and 95% confidence interval) for the ten viruses with 50 or more VIPs. F) Same as E) but for the 20 top high level GO processes with the most VIPs below the dotted black line. Above the dotted black line: the classic MK test for all VIPs, for non-immune VIPs and for immune VIPs (Table S4). See also Tables S3, S4, S5, S6, S7, S8 and S9 and Figures S3 and S4\\n',\n",
       "   'img': 'https://www.biorxiv.org/content/biorxiv/early/2015/10/18/029397/F1.medium.gif'},\n",
       "  {'label': 'Figure 2.',\n",
       "   'title': 'Excess of adaptation across mammals in VIPs',\n",
       "   'description': 'A) Thick white curve: average excess of adaptation in all VIPs. Narrow white curves: 95% confidence interval for the excess of adaptation in all VIPs. The grey background represents the density distributions of the excess of adaptation based on 5,000 iterations of the purifying selection-wise permutation test (Supplemental Methods). Thick black curve: excess of adaptation in non-immune VIPs. Narrow black curve: lower bound of the 95% confidence interval for the excess of adaptation in non-immune VIPs. B) Virus-by-virus excess of adaptation in VIPs. Black dot is the average excess and the represented interval is the 95% confidence interval. C) Excess of adaptation within the top 20 high level GO processes with the most VIPs. D) Proportions of selected codons in VIPs (blue dot) and non-VIPs (red dot and 95% confidence interval) in the mammalian clades represented by more than one species in the tree. All: entire tree. Primata: primates. Glires: rodents and rabbit. Cetartyodactyla: sheep, cow, pig. Zooamata: carnivores and horse. See also Tables S4 and S4 and Figures S5, S6, S7, S8 and S9.\\n',\n",
       "   'img': 'https://www.biorxiv.org/content/biorxiv/early/2015/10/18/029397/F2.medium.gif'},\n",
       "  {'label': 'Figure 3.',\n",
       "   'title': 'Examples of mammalian orthologs with adaptation spread across clades',\n",
       "   'description': 'A) Signals of adaptation in eight antiviral proteins with well-known adaptation. Red: BS-REL P≤0.001. Orange: BS-REL P≤0.05. Yellow: BS-REL P≤0.1. B) Top eight antiviral proteins with the highest number of branches under selection, and no previously know adaptive evolution. C) Top eight non-antiviral proteins with well-known functions and the highest number of branches under selection. Proteins are ordered according to the number of branches with signals of adaptation. All proteins in the figure have their product of BS-REL test P-values lower than 10−9.\\n',\n",
       "   'img': 'https://www.biorxiv.org/content/biorxiv/early/2015/10/18/029397/F3.medium.gif'},\n",
       "  {'label': 'Figure 4.',\n",
       "   'title': 'Patterns of adaptation to coronaviruses in aminopeptidase N',\n",
       "   'description': 'A) BS-REL test results for ANPEP in a tree of 84 mammalian species. Legend is on the figure. B) Contact surface with PRCV and TGEV on ANPEP structure (PDB 4FYQ). The figure includes visualizations of all the six different faces of ANPEP. Legend is in the figure. C) Excess of adaptation in and near the contact interface with PRCV and TGEV. Within the contact interface plus a given number of neighboring amino acids (one, five, ten or 20 in the figure), adaptation excess (y axis) is defined as the number of observed codons with a MEME P-value lower than the P-value threshold on the x axis, divided by the average number of codons under the same P-value threshold obtained after randomizing the location of adaptation signals over the entire ANPEP coding sequence 5,000 times. Dark red curve: adaptation excess within the contact interface with TGEV and PRCV plus one neighboring amino acid. Red curve: plus five neighboring amino acids. Orange: plus ten neighboring amino acids. Light orange: plus 20 neighboring amino acids. Numbers in the figure represent the number of adapting codons, and the stars give the significance of the excess. One star: excess P≤0.05. Two stars: P≤0.01. D) Losses and gains of the N-glycosylation across the mammalian phylogeny. See also Tables S10 and S11.\\n',\n",
       "   'img': 'https://www.biorxiv.org/content/biorxiv/early/2015/10/18/029397/F4.medium.gif'}],\n",
       " 'references': [{'label': '',\n",
       "   'authors': [],\n",
       "   'date': '2015',\n",
       "   'title': 'Gene Ontology Consortium: going forward',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/nar/gku1179&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=25428369&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Abecasis G.R.',\n",
       "    'Auton A.',\n",
       "    'Brooks L.D.',\n",
       "    'DePristo M.A.',\n",
       "    'Durbin R.M.',\n",
       "    'Handsaker R.E.',\n",
       "    'Kang H.M.',\n",
       "    'Marth G.T.',\n",
       "    'McVean G.A.'],\n",
       "   'date': '2012',\n",
       "   'title': 'An integrated map of genetic variation from 1,092 human genomes',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/nature11632&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=23128226&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000310434500030&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Ashburner M.',\n",
       "    'Ball C.A.',\n",
       "    'Blake J.A.',\n",
       "    'Botstein D.',\n",
       "    'Butler H.',\n",
       "    'Cherry J.M.',\n",
       "    'Davis A.P.',\n",
       "    'Dolinski K.',\n",
       "    'Dwight S.S.',\n",
       "    'Eppig J.T.'],\n",
       "   'date': '2000',\n",
       "   'title': 'Gene ontology: tool for the unification of biology. The Gene Ontology Consortium',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/75556&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=10802651&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000086884000011&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Bause E.'],\n",
       "   'date': '1983',\n",
       "   'title': 'Structural requirements of N-glycosylation of proteins. Studies with proline peptides as conformational probes',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6MTA6InBwYmlvY2hlbWoiO3M6NToicmVzaWQiO3M6OToiMjA5LzIvMzMxIjtzOjQ6ImF0b20iO3M6Mzc6Ii9iaW9yeGl2L2Vhcmx5LzIwMTUvMTAvMTgvMDI5Mzk3LmF0b20iO31zOjg6ImZyYWdtZW50IjtzOjA6IiI7fQ==']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Bierne N.', 'Eyre-Walker A.'],\n",
       "   'date': '2004',\n",
       "   'title': 'The genomic rate of adaptive amino acid substitution in Drosophila',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msh134&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=15044594&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000222334300020&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Cagliani R.',\n",
       "    'Guerini F.R.',\n",
       "    'Fumagalli M.',\n",
       "    'Riva S.',\n",
       "    'Agliardi C.',\n",
       "    'Galimberti D.',\n",
       "    'Pozzoli U.',\n",
       "    'Goris A.',\n",
       "    'Dubois B.',\n",
       "    'Fenoglio C.'],\n",
       "   'date': '2012',\n",
       "   'title': 'A trans-specific polymorphism in ZC3HAV1 is maintained by long-standing balancing selection and may confer susceptibility to multiple sclerosis',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/mss002&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22319148&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000304200600011&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Cagliani R.',\n",
       "    'Riva S.',\n",
       "    'Fumagalli M.',\n",
       "    'Biasin M.',\n",
       "    'Caputo S.L.',\n",
       "    'Mazzotta F.',\n",
       "    'Piacentini L.',\n",
       "    'Pozzoli U.',\n",
       "    'Bresolin N.',\n",
       "    'Clerici M.'],\n",
       "   'date': '2011',\n",
       "   'title': 'A positively selected APOBEC3H haplotype is associated with natural resistance to HIV-1 infection',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1111/j.1558-5646.2011.01368.x&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22023594&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Charlesworth J.', 'Eyre-Walker A.'],\n",
       "   'date': '2008',\n",
       "   'title': 'The McDonald-Kreitman test and slightly deleterious mutations',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msn005&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=18195052&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000255758200002&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Davis Z.H.',\n",
       "    'Verschueren E.',\n",
       "    'Jang G.M.',\n",
       "    'Kleffman K.',\n",
       "    'Johnson J.R.',\n",
       "    'Park J.',\n",
       "    'Von Dollen J.',\n",
       "    'Maher M.C.',\n",
       "    'Johnson T.',\n",
       "    'Newton W.'],\n",
       "   'date': '2015',\n",
       "   'title': 'Global mapping of herpesvirus-host protein complexes reveals a transcription strategy for late genes',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1016/j.molcel.2014.11.026&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=25544563&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Delmas B.',\n",
       "    'Gelfi J.',\n",
       "    'L’Haridon R.',\n",
       "    'Vogel L.K.',\n",
       "    'Sjostrom H.',\n",
       "    'Noren O.',\n",
       "    'Laude H.'],\n",
       "   'date': '1992',\n",
       "   'title': 'Aminopeptidase N is a major receptor for the entero-pathogenic coronavirus TGEV',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/357417a0&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=1350661&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Delmas B.', 'Gelfi J.', 'Sjostrom H.', 'Noren O.', 'Laude H.'],\n",
       "   'date': '1993',\n",
       "   'title': 'Further characterization of aminopeptidase-N as a receptor for coronaviruses',\n",
       "   'links': [['PubMed',\n",
       "     '/lookup/external-ref?access_num=7911642&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=A1993BZ77F00043&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Demogines A.',\n",
       "    'Abraham J.',\n",
       "    'Choe H.',\n",
       "    'Farzan M.',\n",
       "    'Sawyer S.L.'],\n",
       "   'date': '2013',\n",
       "   'title': 'Dual host-virus arms races shape an essential housekeeping protein',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.pbio.1001571&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=23723737&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Dyer M.D.', 'Murali T.M.', 'Sobral B.W.'],\n",
       "   'date': '2008',\n",
       "   'title': 'The landscape of human proteins interacting with viruses and other pathogens',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.ppat.0040032&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=18282095&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Elde N.C.', 'Child S.J.', 'Geballe A.P.', 'Malik H.S.'],\n",
       "   'date': '2009',\n",
       "   'title': 'Protein kinase R reveals an evolutionary model for defeating viral mimicry',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/nature07529&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=19043403&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000262519200048&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Emerman M.', 'Malik H.S.'],\n",
       "   'date': '2010',\n",
       "   'title': 'Paleovirology--modern consequences of ancient viruses',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.pbio.1000301&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=20161719&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Eyre-Walker A.', 'Keightley P.D.'],\n",
       "   'date': '2009',\n",
       "   'title': 'Estimating the rate of adaptive molecular evolution in the presence of slightly deleterious mutations and population size change',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msp119&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=19535738&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000269001500015&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Fletcher W.', 'Yang Z.'],\n",
       "   'date': '2010',\n",
       "   'title': 'The effect of insertions, deletions, and alignment errors on the branch-site test of positive selection',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msq115&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=20447933&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000282174600007&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Flicek P.',\n",
       "    'Amode M.R.',\n",
       "    'Barrell D.',\n",
       "    'Beal K.',\n",
       "    'Brent S.',\n",
       "    'Carvalho-Silva D.',\n",
       "    'Clapham P.',\n",
       "    'Coates G.',\n",
       "    'Fairley S.',\n",
       "    'Fitzgerald S.'],\n",
       "   'date': '2012',\n",
       "   'title': 'Ensembl 2012',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/nar/gkr991&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22086963&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000298601300014&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Fumagalli M.',\n",
       "    'Cagliani R.',\n",
       "    'Riva S.',\n",
       "    'Pozzoli U.',\n",
       "    'Biasin M.',\n",
       "    'Piacentini L.',\n",
       "    'Comi G.P.',\n",
       "    'Bresolin N.',\n",
       "    'Clerici M.',\n",
       "    'Sironi M.'],\n",
       "   'date': '2010',\n",
       "   'title': 'Population genetics of IFIH1: ancient population structure, local selection, and implications for susceptibility to type 1 diabetes',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msq141&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=20538742&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000283118000012&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Gray K.A.',\n",
       "    'Yates B.',\n",
       "    'Seal R.L.',\n",
       "    'Wright M.W.',\n",
       "    'Bruford E.A.'],\n",
       "   'date': '2015',\n",
       "   'title': 'Genenames.org: the HGNC resources in 2015',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/nar/gku1071&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=25361968&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Guirimand T.', 'Delmotte S.', 'Navratil V.'],\n",
       "   'date': '2015',\n",
       "   'title': 'VirHostNet 2.0: surfing on the web of virus/host molecular interactions data',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/nar/gku1121&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=25392406&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Halehalli R.R.', 'Nagarajaram H.A.'],\n",
       "   'date': '2015',\n",
       "   'title': 'Molecular principles of human virus protein-protein interactions',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/bioinformatics/btu763&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=25417202&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['He X.', 'Zhang J.'],\n",
       "   'date': '2006',\n",
       "   'title': 'Toward a molecular understanding of pleiotropy',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6ODoiZ2VuZXRpY3MiO3M6NToicmVzaWQiO3M6MTA6IjE3My80LzE4ODUiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNS8xMC8xOC8wMjkzOTcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Jager S.',\n",
       "    'Cimermancic P.',\n",
       "    'Gulbahce N.',\n",
       "    'Johnson J.R.',\n",
       "    'McGovern K.E.',\n",
       "    'Clarke S.C.',\n",
       "    'Shales M.',\n",
       "    'Mercenne G.',\n",
       "    'Pache L.',\n",
       "    'Li K.'],\n",
       "   'date': '2012',\n",
       "   'title': 'Global landscape of HIV-human protein complexes',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/nature10719&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22190034&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000299210600043&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Jordan G.', 'Goldman N.'],\n",
       "   'date': '2012',\n",
       "   'title': 'The effects of alignment error and alignment filtering on the sitewise detection of positive selection',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msr272&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22049066&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000302018100004&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Kaelber J.T.',\n",
       "    'Demogines A.',\n",
       "    'Harbison C.E.',\n",
       "    'Allison A.B.',\n",
       "    'Goodman L.B.',\n",
       "    'Ortega A.N.',\n",
       "    'Sawyer S.L.',\n",
       "    'Parrish C.R.'],\n",
       "   'date': '2012',\n",
       "   'title': 'Evolutionary reconstructions of the transferrin receptor of Caniforms supports canine parvovirus being a re-emerged and not a novel pathogen in dogs',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.ppat.1002666&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22570610&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Katzourakis A.', 'Gifford R.J.'],\n",
       "   'date': '2010',\n",
       "   'title': 'Endogenous viral elements in animal genomes',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.pgen.1001191&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=21124940&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Keightley P.D.', 'Eyre-Walker A.'],\n",
       "   'date': '2007',\n",
       "   'title': 'Joint inference of the distribution of fitness effects of deleterious mutations and population demography based on nucleotide polymorphism frequencies',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6ODoiZ2VuZXRpY3MiO3M6NToicmVzaWQiO3M6MTA6IjE3Ny80LzIyNTEiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNS8xMC8xOC8wMjkzOTcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Kent W.J.'],\n",
       "   'date': '2002',\n",
       "   'title': 'BLAT--the BLAST-like alignment tool',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NjoiZ2Vub21lIjtzOjU6InJlc2lkIjtzOjg6IjEyLzQvNjU2IjtzOjQ6ImF0b20iO3M6Mzc6Ii9iaW9yeGl2L2Vhcmx5LzIwMTUvMTAvMTgvMDI5Mzk3LmF0b20iO31zOjg6ImZyYWdtZW50IjtzOjA6IiI7fQ==']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Kerns J.A.', 'Emerman M.', 'Malik H.S.'],\n",
       "   'date': '2008',\n",
       "   'title': 'Positive selection and increased antiviral activity associated with the PARP-containing isoform of human zinc-finger antiviral protein',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.pgen.0040021&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=18225958&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Kimura M.'],\n",
       "   'date': '1977',\n",
       "   'title': 'Preponderance of synonymous changes as evidence for the neutral theory of molecular evolution',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/267275a0&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=865622&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=A1977DF51700060&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Kosakovsky Pond S.L.',\n",
       "    'Murrell B.',\n",
       "    'Fourment M.',\n",
       "    'Frost S.D.',\n",
       "    'Delport W.',\n",
       "    'Scheffler K.'],\n",
       "   'date': '2011',\n",
       "   'title': 'A random effects branch-site model for detecting episodic diversifying selection',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/molbev/msr125&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=21670087&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000296336600009&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Kumar R.', 'Nanduri B.'],\n",
       "   'date': '2010',\n",
       "   'title': 'HPIDB--a unified resource for host-pathogen interactions',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1186/1471-2105-11-S6-S16&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=20946599&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Liu H.L.',\n",
       "    'Wang Y.Q.',\n",
       "    'Liao C.H.',\n",
       "    'Kuang Y.Q.',\n",
       "    'Zheng Y.T.',\n",
       "    'Su B.'],\n",
       "   'date': '2005',\n",
       "   'title': 'Adaptive evolution of primate TRIM5alpha, a gene restricting HIV-1 infection',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1016/j.gene.2005.06.045&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=16226405&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000233699000012&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Loytynoja A.', 'Goldman N.'],\n",
       "   'date': '2008',\n",
       "   'title': 'A model of evolution and structure for multiple sequence alignment',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1098/rstb.2008.0170&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=18852103&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['McDonald J.H.', 'Kreitman M.'],\n",
       "   'date': '1991',\n",
       "   'title': 'Adaptive protein evolution at the Adh locus in Drosophila',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/351652a0&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=1904993&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=A1991FT11200066&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Mellacheruvu D.',\n",
       "    'Wright Z.',\n",
       "    'Couzens A.L.',\n",
       "    'Lambert J.P.',\n",
       "    'St-Denis N.A.',\n",
       "    'Li T.',\n",
       "    'Miteva Y.V.',\n",
       "    'Hauri S.',\n",
       "    'Sardiu M.E.',\n",
       "    'Low T.Y.'],\n",
       "   'date': '2013',\n",
       "   'title': 'The CRAPome: a contaminant repository for affinity purification-mass spectrometry data',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/nmeth.2557&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=23921808&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000322453600019&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Messer P.W.'],\n",
       "   'date': '2013',\n",
       "   'title': 'SLiM: simulating evolution with selection and linkage',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6ODoiZ2VuZXRpY3MiO3M6NToicmVzaWQiO3M6MTA6IjE5NC80LzEwMzciO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNS8xMC8xOC8wMjkzOTcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Messer P.W.', 'Petrov D.A.'],\n",
       "   'date': '2013',\n",
       "   'title': 'Frequent adaptation and the McDonald-Kreitman test',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NDoicG5hcyI7czo1OiJyZXNpZCI7czoxMToiMTEwLzIxLzg2MTUiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNS8xMC8xOC8wMjkzOTcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Mina-Osorio P.'],\n",
       "   'date': '2008',\n",
       "   'title': 'The moonlighting enzyme CD13: old and new functions to target',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1016/j.molmed.2008.06.003&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=18603472&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000258816700005&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Murrell B.',\n",
       "    'Wertheim J.O.',\n",
       "    'Moola S.',\n",
       "    'Weighill T.',\n",
       "    'Scheffler K.',\n",
       "    'Kosakovsky Pond S.L.'],\n",
       "   'date': '2012',\n",
       "   'title': 'Detecting individual sites subject to episodic diversifying selection',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.pgen.1002764&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22807683&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Oh J.S.', 'Song D.S.', 'Park B.K.'],\n",
       "   'date': '2003',\n",
       "   'title': 'Identification of a putative cellular receptor 150 kDa polypeptide for porcine epidemic diarrhea virus in porcine enterocytes',\n",
       "   'links': [['PubMed',\n",
       "     '/lookup/external-ref?access_num=14685034&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Patel M.R.', 'Emerman M.', 'Malik H.S.'],\n",
       "   'date': '2011',\n",
       "   'title': 'Paleovirology - ghosts and gifts of viruses past',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1016/j.coviro.2011.06.007&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22003379&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Pettersen E.F.',\n",
       "    'Goddard T.D.',\n",
       "    'Huang C.C.',\n",
       "    'Couch G.S.',\n",
       "    'Greenblatt D.M.',\n",
       "    'Meng E.C.',\n",
       "    'Ferrin T.E.'],\n",
       "   'date': '2004',\n",
       "   'title': 'UCSF Chimera--a visualization system for exploratory research and analysis',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1002/jcc.20084&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=15264254&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000223379100005&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Pond S.L.', 'Frost S.D.', 'Muse S.V.'],\n",
       "   'date': '2005',\n",
       "   'title': 'HyPhy: hypothesis testing using phylogenies',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/bioinformatics/bti079&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=15509596&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000227241200017&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Prado-Martinez J.',\n",
       "    'Sudmant P.H.',\n",
       "    'Kidd J.M.',\n",
       "    'Li H.',\n",
       "    'Kelley J.L.',\n",
       "    'Lorente-Galdos B.',\n",
       "    'Veeramah K.R.',\n",
       "    'Woerner A.E.',\n",
       "    'O’Connor T.D.',\n",
       "    'Santpere G.'],\n",
       "   'date': '2013',\n",
       "   'title': 'Great ape genetic diversity and population history',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/nature12228&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=23823723&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000322157900039&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Quan P.L.',\n",
       "    'Firth C.',\n",
       "    'Conte J.M.',\n",
       "    'Williams S.H.',\n",
       "    'Zambrana-Torrelio C.M.',\n",
       "    'Anthony S.J.',\n",
       "    'Ellison J.A.',\n",
       "    'Gilbert A.T.',\n",
       "    'Kuzmin I.V.',\n",
       "    'Niezgoda M.'],\n",
       "   'date': '2013',\n",
       "   'title': 'Bats are a major natural reservoir for hepaciviruses and pegiviruses',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NDoicG5hcyI7czo1OiJyZXNpZCI7czoxMToiMTEwLzIwLzgxOTQiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNS8xMC8xOC8wMjkzOTcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Reguera J.',\n",
       "    'Santiago C.',\n",
       "    'Mudgal G.',\n",
       "    'Ordono D.',\n",
       "    'Enjuanes L.',\n",
       "    'Casasnovas J.M.'],\n",
       "   'date': '2012',\n",
       "   'title': 'Structural bases of coronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.ppat.1002859&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=22876187&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Sawyer S.L.', 'Emerman M.', 'Malik H.S.'],\n",
       "   'date': '2004',\n",
       "   'title': 'Ancient adaptive evolution of the primate antiviral DNA-editing enzyme APOBEC3G',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.pbio.0020275&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=15269786&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Sawyer S.L.', 'Emerman M.', 'Malik H.S.'],\n",
       "   'date': '2007',\n",
       "   'title': 'Discordant evolution of the adjacent antiretroviral genes TRIM22 and TRIM5 in mammals',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1371/journal.ppat.0030197&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=18159944&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Sawyer S.L.', 'Wu L.I.', 'Emerman M.', 'Malik H.S.'],\n",
       "   'date': '2005',\n",
       "   'title': 'Positive selection of primate TRIM5alpha identifies a critical species-specific retroviral restriction domain',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NDoicG5hcyI7czo1OiJyZXNpZCI7czoxMDoiMTAyLzgvMjgzMiI7czo0OiJhdG9tIjtzOjM3OiIvYmlvcnhpdi9lYXJseS8yMDE1LzEwLzE4LzAyOTM5Ny5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Sironi M.',\n",
       "    'Biasin M.',\n",
       "    'Cagliani R.',\n",
       "    'Forni D.',\n",
       "    'De Luca M.',\n",
       "    'Saulle I.',\n",
       "    'Lo Caputo S.',\n",
       "    'Mazzotta F.',\n",
       "    'Macias J.',\n",
       "    'Pineda J.A.'],\n",
       "   'date': '2012',\n",
       "   'title': 'A common polymorphism in TLR3 confers natural resistance to HIV-1 infection',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6ODoiamltbXVub2wiO3M6NToicmVzaWQiO3M6OToiMTg4LzIvODE4IjtzOjQ6ImF0b20iO3M6Mzc6Ii9iaW9yeGl2L2Vhcmx5LzIwMTUvMTAvMTgvMDI5Mzk3LmF0b20iO31zOjg6ImZyYWdtZW50IjtzOjA6IiI7fQ==']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Tresnan D.B.', 'Holmes K.V.'],\n",
       "   'date': '1998',\n",
       "   'title': 'Feline aminopeptidase N is a receptor for all group I coronaviruses',\n",
       "   'links': [['PubMed',\n",
       "     '/lookup/external-ref?access_num=9782266&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Tusell S.M.', 'Schittone S.A.', 'Holmes K.V.'],\n",
       "   'date': '2007',\n",
       "   'title': 'Mutational analysis of aminopeptidase N, a receptor for several group 1 coronaviruses, identifies key determinants of viral host range',\n",
       "   'links': [['Abstract/FREE Full Text',\n",
       "     '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6MzoianZpIjtzOjU6InJlc2lkIjtzOjk6IjgxLzMvMTI2MSI7czo0OiJhdG9tIjtzOjM3OiIvYmlvcnhpdi9lYXJseS8yMDE1LzEwLzE4LzAyOTM5Ny5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Vasseur E.',\n",
       "    'Patin E.',\n",
       "    'Laval G.',\n",
       "    'Pajon S.',\n",
       "    'Fornarino S.',\n",
       "    'Crouau-Roy B.',\n",
       "    'Quintana-Murci L.'],\n",
       "   'date': '2011',\n",
       "   'title': 'The selective footprints of viral pressures at the human RIG-I-like receptor family',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1093/hmg/ddr377&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=21865300&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom'],\n",
       "    ['Web of Science',\n",
       "     '/lookup/external-ref?access_num=000296103800015&link_type=ISI']]},\n",
       "  {'label': '',\n",
       "   'authors': ['Yeager C.L.',\n",
       "    'Ashmun R.A.',\n",
       "    'Williams R.K.',\n",
       "    'Cardellichio C.B.',\n",
       "    'Shapiro L.H.',\n",
       "    'Look A.T.',\n",
       "    'Holmes K.V.'],\n",
       "   'date': '1992',\n",
       "   'title': 'Human aminopeptidase N is a receptor for human coronavirus 229E',\n",
       "   'links': [['CrossRef',\n",
       "     '/lookup/external-ref?access_num=10.1038/357420a0&link_type=DOI'],\n",
       "    ['PubMed',\n",
       "     '/lookup/external-ref?access_num=1350662&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2015%2F10%2F18%2F029397.atom']]}]}"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_paper.paper"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Scraping functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def scrape_biorxiv(paper):\n",
    "    print('Fetching paper data at {}'.format(paper['url']))\n",
    "    html = session.get(paper['url'])\n",
    "    tmp_soup = BeautifulSoup(html.content, 'html.parser')\n",
    "    paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "    paper = dict(chain(paper.items(),paper_webdata.items()))\n",
    "    return paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_authors(authors_html):\n",
    "    ''' HTML list of authors\n",
    "    '''\n",
    "    authors = []\n",
    "    for author in authors_html:\n",
    "        try:\n",
    "            given_name = author.find('span',attrs={'class':'nlm-given-names'}).text\n",
    "        except:\n",
    "            given_name = ''\n",
    "        try:\n",
    "            surname = author.find('span',attrs={'class':'nlm-surname'}).text\n",
    "        except:\n",
    "            surname = ''\n",
    "        authors.append(given_name + ' ' + surname)\n",
    "    return authors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {
    "code_folding": [
     1
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# get reference anchor id\n",
    "def get_biorxiv_references(references_html):\n",
    "    ''' HTML list of references\n",
    "    soup.find('div',attrs={'class':'section ref-list'}).find_all('li')\n",
    "    '''\n",
    "    references = []\n",
    "    for ref in references_html:\n",
    "        reference = {}\n",
    "        # label\n",
    "        try:\n",
    "            reference['label'] = ref.find('span',attrs={'class':'ref-label'}).text\n",
    "        except:\n",
    "            reference['label'] = None\n",
    "\n",
    "        # authors\n",
    "        author_names = []\n",
    "        try:\n",
    "            authors = ref.find('cite').find_all('span',attrs={'class':'cit-auth'})\n",
    "            for author in authors:\n",
    "                surname = author.find('span',attrs={'class':'cit-name-surname'}).text\n",
    "                given_names = author.find('span',attrs={'class':'cit-name-given-names'}).text\n",
    "                author_name = surname + ' ' + given_names\n",
    "                author_names.append(author_name)\n",
    "            reference['authors'] = author_names\n",
    "        except:\n",
    "            reference['authors'] = author_names\n",
    "\n",
    "        # date\n",
    "        try:\n",
    "            reference['date'] = ref.find('cite').find('span',attrs={'class':'cit-pub-date'}).text\n",
    "        except:\n",
    "            reference['date'] = None\n",
    "\n",
    "        # title\n",
    "        try:\n",
    "            reference['title'] = ref.find('cite').find('span',attrs={'class':'cit-article-title'}).text\n",
    "        except:\n",
    "            reference['title'] = ''\n",
    "\n",
    "        # links\n",
    "        ref_links = []\n",
    "        try:\n",
    "            links = ref.find('div',attrs={'class':'cit-extra'}).find_all('a')\n",
    "            for link in links:\n",
    "                if link.text != 'OpenUrl':\n",
    "                    link_name = link.text\n",
    "                    link_url = link['href']\n",
    "                    ref_links.append((link_name,link_url))\n",
    "            reference['links'] = ref_links\n",
    "        except:\n",
    "            reference['links'] = ref_links\n",
    "        references.append(reference)\n",
    "    return references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_sections(sections_html):\n",
    "    '''\n",
    "    '''\n",
    "    sections = []\n",
    "    for section in sections_html:\n",
    "        # ignore references\n",
    "        if 'ref-list' not in section.attrs['class'] :\n",
    "            sections.append(explore_section(section))\n",
    "    return sections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {
    "code_folding": [
     1
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# make tables a dict with table id for easy searching\n",
    "def get_biorxiv_tables(html_tables, base_url):\n",
    "    tables = {}\n",
    "    for html_table in html_tables:\n",
    "        table = {}\n",
    "        \n",
    "        table['id'] = html_table['id']\n",
    "        table_label = html_table.find('span',attrs={'class':'table-label'})\n",
    "        if table_label:\n",
    "            table['label'] = table_label.text\n",
    "\n",
    "        table_caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "        if table_caption:\n",
    "            table['caption'] = table_caption.text\n",
    "\n",
    "        table['description'] = ''\n",
    "        for p in html_table.find_all('p'):\n",
    "            table['description'] += p.text + '\\n'\n",
    "        table['img'] = '{}/{}.large.jpg'.format(base_url,table['id'])\n",
    "        tables[table['id']] = table\n",
    "    return tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {
    "code_folding": [
     1
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# make figures a dict with figure id for easy searching\n",
    "def get_biorxiv_figures(figures_html):\n",
    "    figures = {}\n",
    "    for html_figure in figures_html:\n",
    "        figure = {}\n",
    "        figure['id'] = html_figure['id']\n",
    "        figure_label = html_figure.find('span',attrs={'class':'fig-label'})\n",
    "        if figure_label:\n",
    "            figure['label'] = figure_label.text\n",
    "\n",
    "        figure_title = html_figure.find('span',attrs={'class':'caption-title'})\n",
    "        if figure_title:\n",
    "            figure['title'] = figure_title.text\n",
    "\n",
    "        figure['description'] = ''\n",
    "        figure_ps = html_figure.find_all('p')\n",
    "        for p in figure_ps:\n",
    "            figure['description'] += p.text + '\\n'\n",
    "        figure_img = html_figure.find('a')\n",
    "        if figure_img:\n",
    "            figure['img'] = figure_img['href'].split('?')[0]\n",
    "        figures[figure['id']] = figure\n",
    "    return figures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def process_paragraph(paragraph):\n",
    "    ''' Removes and modifies html tags in paragraphs\n",
    "    '''\n",
    "    for anchor in paragraph.find_all('a'):\n",
    "        anchor.replace_with('[' + anchor['href'].strip('#') + ',\"' + anchor.text + '\"]')\n",
    "    return paragraph.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def explore_section(root_tag):\n",
    "    '''Recursively explore section and all subsections for content\n",
    "    '''\n",
    "    content = {}\n",
    "    content['title'] = ''\n",
    "    content['content'] = []\n",
    "    for child in root_tag.findChildren(recursive=False):\n",
    "        if child.name == 'h2' or child.name == 'h3' or child.name == 'h4':\n",
    "            content['title'] = child.text\n",
    "        elif child.name == 'p':\n",
    "            content['content'].append(process_paragraph(child))\n",
    "        if child.name == 'div':\n",
    "            if 'class' in child.attrs:\n",
    "                if 'subsection' in child['class']:\n",
    "                    # if subsection explore recursively\n",
    "                    content['content'].append(explore_section(child))\n",
    "            if 'id' in child.attrs.keys():\n",
    "                if 'T' in child['id']:\n",
    "                    content['content'].append({'id':child['id'],'type':'table'})\n",
    "                elif 'F' in child['id']:\n",
    "                    content['content'].append({'id':child['id'],'type':'figure'})\n",
    "    else:\n",
    "        return content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def biorxiv_html_paper_data_extractor(soup):\n",
    "    '''\n",
    "        Extract paper information from html webpage of biorxiv\n",
    "    '''\n",
    "    content_url = soup.find('meta',attrs={'name':'citation_pdf_url'})['content'].strip('.full.pdf')\n",
    "    \n",
    "    paper = {}\n",
    "    paper['title'] = soup.h1.text\n",
    "\n",
    "    authors_html = soup.find('span',attrs={'class':'highwire-citation-authors'}) \\\n",
    "                       .find_all('span',attrs={'class':'highwire-citation-author'})\n",
    "    paper['authors'] = get_biorxiv_authors(authors_html)\n",
    "\n",
    "    sections_html = soup.find_all('div',attrs={'class':'section'})\n",
    "    paper['sections'] = get_biorxiv_sections(sections_html)\n",
    "\n",
    "    tables_html = soup.find_all('div',attrs={'class':'table'})\n",
    "    paper['tables'] = get_biorxiv_tables(tables_html, content_url)\n",
    "\n",
    "    # get figures\n",
    "    figures_html = soup.find_all('div',attrs={'class':'fig'})\n",
    "    \n",
    "\n",
    "    paper['figures'] = get_biorxiv_figures(figures_html)\n",
    "\n",
    "    # get references\n",
    "    references_html = soup.find('div',attrs={'class':'section ref-list'}).find_all('li')\n",
    "    paper['references'] = get_biorxiv_references(references_html)\n",
    "    \n",
    "    return paper"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Reading from json function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {
    "code_folding": [
     0,
     13,
     24
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def ref_to_str(reference):\n",
    "    ''' Convert reference to string '''\n",
    "    output = ''\n",
    "    if reference['label'] != '':\n",
    "        output = reference['label'] + ' - '\n",
    "    if reference['title'] != '':\n",
    "        output += reference['title'] + ', '\n",
    "    if reference['authors'] != []:\n",
    "        output += ', '.join(reference['authors'])\n",
    "    if reference['date'] != None:\n",
    "        output += reference['date']\n",
    "    return output\n",
    "\n",
    "def tab_to_str(table, indent=''):\n",
    "    ''' Convert table to string '''\n",
    "    output = '\\n'\n",
    "    output += indent + '-'*100 + '\\n'\n",
    "    output += indent + table['id'] + ' | ' + table['label'] + '\\n'\n",
    "    output += indent + table['img'] + '\\n'\n",
    "    output += indent + table['description']\n",
    "    output += indent + '-'*100\n",
    "    output += '\\n'\n",
    "    return output\n",
    "\n",
    "def fig_to_str(figure, indent=''):\n",
    "    ''' Convert figure to string '''\n",
    "    output = '\\n'\n",
    "    output += indent + '-'*100 + '\\n'\n",
    "    output += indent + figure['id'] + ' | ' + figure['label'] + '\\n'\n",
    "    output += indent + figure['img'] + '\\n'\n",
    "    output += indent + figure['description']\n",
    "    output += indent + '-'*100\n",
    "    output += '\\n'\n",
    "    return output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def paper_to_string(paper, appendix=False):\n",
    "    ''' Converts a paper to string formart '''\n",
    "    output = '({} | {} | {})\\n'.format(paper['source'], paper['id'], paper['url'])\n",
    "    output += paper['title'] + '\\n'\n",
    "    output += ', '.join(paper['authors']) + '\\n'\n",
    "    for section in paper['sections']:\n",
    "        if (not appendix) and ('appendix' in section['title'].lower()):\n",
    "            pass\n",
    "        else:\n",
    "            output += section_to_str(section, paper)\n",
    "            output += '\\n'\n",
    "    output += '\\n\\n'.join([ref_to_str(ref) for ref in paper['references']])    \n",
    "    return output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {
    "code_folding": [
     0
    ],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def section_to_str(section, paper, depth=0):\n",
    "    indent = '  '*depth\n",
    "    output = indent + section['title']\n",
    "    for content in section['content']:\n",
    "        if type(content) == dict:\n",
    "            if 'title' in content.keys():\n",
    "                # is section = explore further \n",
    "                output += '\\n'\n",
    "                output += section_to_str(content, paper, depth+1)\n",
    "            elif 'id' in content.keys():\n",
    "                # is fig or table, get string\n",
    "                if content['type'] == 'table':\n",
    "                    output += '\\n'\n",
    "                    output += tab_to_str(paper['tables'][content['id']], indent) # content['id']\n",
    "                    # insert table description from paper['tables']\n",
    "                elif content['type'] == 'figure':\n",
    "                    output += ' \\n'\n",
    "                    output += fig_to_str(paper['figures'][content['id']], indent) # content['id']\n",
    "                    # insert figure description from paper['figures']\n",
    "                else:\n",
    "                    print('content not recognized: {}'.format(content))\n",
    "        elif type(content) == str:\n",
    "            output += '\\n' + indent + content\n",
    "    return output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test Process\n",
    "\n",
    "Test for a single file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Scraping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['f056da9c64fbf00a4645ae326e8a4339d015d155' '10.1101/001727']\n"
     ]
    }
   ],
   "source": [
    "# init session\n",
    "session = requests.session()\n",
    "\n",
    "# load metadata\n",
    "df_meta = pd.read_csv('metadata.csv')\n",
    "\n",
    "papers_df = df_meta[df_meta['source_x']=='biorxiv'][['sha','doi']]\n",
    "papers_array = papers_df.to_numpy()\n",
    "\n",
    "row = papers_array[0]\n",
    "print(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "paper = {}\n",
    "paper['source'] = 'biorxiv'\n",
    "paper['id'] = row[0]\n",
    "paper['url'] = 'https://biorxiv.org/content/' + row[1].strip('doi.org') + 'v1.full'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/001727v1.full\n"
     ]
    }
   ],
   "source": [
    "print('Fetching paper data at {}'.format(paper['url']))\n",
    "html = session.get(paper['url'])\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "paper = dict(chain(paper.items(),paper_webdata.items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# save to file \n",
    "with open('data/biorxiv/{}.json'.format(paper['id']), 'w') as fp:\n",
    "    json.dump(paper, fp, indent=4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Reading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "f056da9c64fbf00a4645ae326e8a4339d015d155.json\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "existing_papers = get_file_names('data/biorxiv')\n",
    "print(existing_papers[0])\n",
    "\n",
    "test_paper = Paper('biorxiv')\n",
    "test_paper.import_json('data/biorxiv/{}'.format(existing_papers[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(biorxiv | f056da9c64fbf00a4645ae326e8a4339d015d155 | https://biorxiv.org/content/10.1101/001727v1.full)\n",
      "SIANN: Strain Identification by Alignment to Near Neighbors\n",
      "Samuel S. Minot, Stephen D. Turner, Krista L. Ternus, Dana R. Kadavy\n",
      "Abstract\n",
      "Next-generation sequencing is increasingly being used to study samples composed of mixtures of organisms, such as in clinical applications where the presence of a pathogen at very low abundance may be highly important. We present an analytical method (SIANN: Strain Identification by Alignment to Near Neighbors) specifically designed to rapidly detect a set of target organisms in mixed samples that achieves a high degree of species- and strain-specificity by aligning short sequence reads to the genomes of near neighbor organisms, as well as that of the target. Empirical benchmarking alongside the current state-of-the-art methods shows an extremely high Positive Predictive Value, even at very low abundances of the target organism in a mixed sample. SIANN is available as an Illumina BaseSpace app, as well as through Signature Science, LLC. SIANN results are presented in a streamlined report designed to be comprehensible to the non-specialist user, providing a powerful tool for rapid species detection in a mixed sample. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.\n",
      "Introduction\n",
      "There are many different methods that characterize the mixture of organisms present within a metagenomic dataset. Such datasets are generated when a complex environmental sample is processed by a “next-generation” high-throughput genome sequencing protocol, and they consist of large numbers of short nucleotide sequences. Each sequence represents a small fragment of a randomly selected genome from the very large collection of genomes present in the source sample. Those sequences indicate the presence of one organism or another according to their similarity to a set of known reference genomes. While a given sequence may be unique to one species, it also may be found in diverse organisms across the tree of life. Therefore, one analytical challenge (among many) is to take that collection of sequences (likely numbering in the millions) and accurately determine what species are present in the sample. Here we describe a novel method (SIANN: Strain Identification by Alignment to Near Neighbors) that is specifically designed to rapidly detect a set of targeted organisms from a metagenomic dataset by aligning reads to genomic regions that are unique at the strain or species level.\n",
      "The analytical question motivating a particular piece of metagenomic bioinformatic analysis may vary widely by user and sample type ([ref-11,\"Segata, et al., 2013\"]). For example, the function of the human gut microbiome may depend on the relative abundance of hundreds of species of bacteria and the types of metabolic genes they contain ([ref-15,\"Wu, et al., 2011\"]; [ref-10,\"Schloissnig, et al., 2013\"]). In contrast, the clinical treatment of a patient may depend on whether or not a particular virus, or a consortium of co-infecting pathogens, is/are detected in their blood. It is this second class of presence/absence questions that SIANN is designed to address. SIANN is appropriate for situations in which a user wants to know whether a particular organism or set of organisms is present in a sample, but isn’t interested in the functions encoded in their genomes, the relative abundance of each organism, or any other more in-depth analysis.\n",
      "Methods\n",
      "  Approach\n",
      "  Metagenomic classification methods are based on a wide variety of theoretical underpinnings. The basic varieties include alignment of reads to various nucleotide databases or exact matching to nucleotide or protein signature sequences (or kmers). A representative set of recent methods are described in [T1,\"Table 1\"] (also see [ref-2,\"Bazinet & Cummings 2012\"]).\n",
      "\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "  T1 | Table 1.\n",
      "  https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T1.large.jpg\n",
      "  Summary of methods for metagenomic classification.\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "\n",
      "  Overall, these methods are designed to either classify individual reads to, and/or predict the total abundance of, clades (e.g. genus or species) across the entire tree of life. They generally require reference databases that are very large and/or require a large amount of processing to generate. The gap SIANN is designed to fill is when the entire tree of life is irrelevant, and only predefined subsets of organisms need to be detected. For an underlying method we chose read alignment to diagnostic genomic regions because the algorithms for read alignment are highly parallelizable and have been optimized heavily by the community at large (the current implementation of SIANN uses bowtie2 [[ref-7,\"Langmead & Salzberg, 2012\"]] for the alignment function, but can be adapted to any alignment algorithm). This approach is distinct from using clade-specific marker genes ([ref-12,\"Segata, et al., 2012\"]) because unique regions that are larger, smaller, or outside of genes can also be used. Furthermore, this approach supports the rapid construction of custom databases using reference genome sets that require only minimal user-supplied structure.\n",
      "  To understand the principle at work, consider a set of reads that have been aligned to the genomes of several strains belonging to two species. Some regions of those genomes are species-specific, some are strain-specific, and some are shared ([F1,\"Figure 1a\"]). When a set of reads is aligned to those genomes such that each read is placed in as many locations as it has a match (at a reasonably stringent threshold), visual inspection of the distribution of reads yields an intuitive understanding of the true source organism as Species I/Strain B ([F1,\"Figure 1b\"]). If Strain B were not present in the reference database, it would still be clear that the organism was an unknown strain of Species I. \n",
      "\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "  F1 | Figure 1.\n",
      "  https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F1.large.jpg\n",
      "  A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "\n",
      "  The unique identification of a species or strain is quantified by the proportion of the genome that is determined to be species- or strain-specific (defined as reads that are aligned to regions that are species- or strain-specific). Each species and strain is then assigned a numerical measure of the proportion that is covered by these diagnostic reads, and that proportional measure is compared to the ideal case, where sequences from a single organism (generated in silico) are aligned against the database in an identical manner. After that normalization factor is applied, the resulting score indicates whether the source sample contained any of the organisms in the reference database. The analysis is conducted independently on both the species and the strain level, so that if the true strain is not present in the database, the species of origin will still be identified. While many methods consider the complete taxonomic tree and assign reads to the least common ancestor, SIANN considers only two taxonomic levels: species and strain, throwing out anything that is not unique at one of those levels and thus obviating many of the confounding factors introduced by manually curated taxonomies.\n",
      "  The example shown in [F1,\"Figure 1b\"] indicates that species-specific reads are identified as reads that align to one species (Species I, in that case) but not the other. If Species II were not present in the example shown in [F1,\"Figure 1b\"], a much larger number of reads would be assigned as “species-specific,” when in fact those regions are shared with other species. Therefore, the ability of this method to identify strain- and species-specific sequences is a direct function of the inclusion of near neighbors in the reference database. This characteristic is shared among many classification algorithms, but it is of particular note for this method when users have an opportunity to construct their own database. In order to detect a target species with a high degree of specificity (reducing false positives), it is necessary to include other related species in the reference database. Only by parallel alignment to those near neighbors can the redundant sequences be separated from the species-specific ones. For example, in order to detect Bacillus anthracis in a sample, it would be necessary to include other species of Bacilli in the reference database so that the presence of B. cereus or B. thuringiensis in a sample does not lead to a false call for B. anthracis.\n",
      "  The nomenclature of genus, species, and strain is potentially problematic because it does not correspond to a consistent degree of evolutionary distance or genomic distinctiveness. The ability to distinguish two organisms by any method using genomic sequence data is proportional to the amount of each genome that is shared or unique. One might assume that any two organisms of the same species will have a relatively predictable amount of shared genomic identity. However, some pairs of organisms from the same species may have less in common than other pairs of organisms from different species or even genera. This ambiguity impacts SIANN in two ways. If two organisms have very little genomic sequence to distinguish them, the sensitivity of SIANN to detect either one will diminish (the rate of false negatives will increase as the likelihood of sequencing unique regions decreases). Conversely, if an organism is extremely dissimilar to the near neighbors selected for the database, the specificity with which SIANN detects that organism will decline (the rate of false positives will increase as the number of related genomes available in the database decreases). For example, if a database contained only E. coli and B.anthracis, a sample containing B. cereus would be misidentified as contraining B. anthracis. In the intended use case, a database targeting B. anthracis would contain B. cereus and a number of other near neighbors to prevent that kind of misidentification. It would be convenient to say that an ideal database can be made by calculating the ideal genetic distance between all references and then finding an ideal set of organisms to make up that database, but the behavior of any database will be governed by the particular genomes of the organisms it encounters in the wild. Because not all organisms evolve in the same manner (differences in mutation rate, horizontal gene transfer, recombination, etc), the suitability of a database and method to detect a given organism can only be determined by thorough validation and benchmarking, as well as updating the reference database as needed. Users of SIANN may construct their own custom databases to include newly identified genomes or specific subsets of genomes that best suit their research interests.\n",
      "  Steps to construct a custom database:\n",
      "\n",
      "  The files contained within each SIANN database are a compressed genomic index and a list containing the proportion of each reference genome that was found to be strain- or species-specific during database construction.\n",
      "  To run SIANN:\n",
      "\n",
      "  Benchmarking\n",
      "  The performance of SIANN (version 1.6) was tested in comparison to the following state-of-the-art metagenomic classification programs: LMAT (version 1.2), MetaPhlAn (version 1.7.7), and Kraken (version 0.9.1b). All of the programs in [T1,\"Table 1\"] were investigated for this effort, and three were chosen based on their ability to run on our high-performance computing cluster with an execution time and memory requirement that would be suitable to a clinical lab. Each program was run on a set of 600 simulated datasets generated by MetaSim ([ref-9,\"Richter, et al., 2008\"]). Each dataset consisted of 15,000,000 reads (100bp single-ended) with Illumina-simulated error (fourth-degree polynomial) ([ref-6,\"Korbel, et al., 2009\"]). The 600 datasets were broken into 12 sets of 50 replicates. Each of the 12 sets contained organisms at different levels of abundance as shown in [T2,\"Table 2\"]. Organisms were specifically chosen in pairs so that the ability to distinguish these near neighbors could be determined. The abundances were staggered at 4-fold intervals so that a wide range could be evaluated. All known species of near neighbors for each of the 12 target organisms were included in the reference database used by SIANN for this benchmarking (“Target Pathogen Database”) and are shown in [app-1,\"Appendix 1\"].\n",
      "\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "  T2 | Table 2.\n",
      "  https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T2.large.jpg\n",
      "    ----------------------------------------------------------------------------------------------------\n",
      "\n",
      "  Each program outputs a distinct measure. Kraken and LMAT both count the reads assigned to each taxon, MetaPhlAn calculates the abundance, and SIANN outputs a measure of the proportion of diagnostic genomic regions present. To put these measures on an even footing, we empirically calculated the false positive rate for each method over all 600 samples, at each possible measure of output. Because each dataset is made up of known organisms, any result can be classified as true or false. Therefore, for any possible result (say, 513 reads classified by LMAT or 1.6% abundance assigned by MetaPhlAn), one can calculate the proportion of calls with at least the same amount of support that were correct (True Positives/[True Positives + False Positives]), over all of the 600 datasets. That measure is commonly given as Positive Predictive Value (PPV). For each program, the results can be translated from the raw value into a PPV that is based on this empirical measure of error. The key item of interest is the PPV value for the results that we know to be true positives, the defined spike organisms. Another way of describing this approach is to say that the results of each program have been normalized to the false positive error rate that was empirically observed. If another set of samples were generated, the PPV vs. raw value curve ([F2,\"Figure 2\"]) would likely fall differently, but in this case it gives us a means of comparing a diverse set of methods against the same ground truth. If method 1 detects an organism with a higher PPV than method 2 does, it means that method 1 has fewer false positives in the range that it reports true positives, which is the definition of utility in this setting. \n",
      "\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "  F2 | Figure 2.\n",
      "  https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F2.large.jpg\n",
      "  Relationship of reported value for each program (horizontal axis, log scale) to the empirically-determined Positive Predictive Value (PPV), shown on the vertical axis. While the exact values depend on the test data used, the general values at significant cutoff values (0.8, 0.9, 0.95 PPV) remain relatively constant across different datasets (data not shown).\n",
      "  ----------------------------------------------------------------------------------------------------\n",
      "\n",
      "  For each method, PPV was calculated as a function of raw output value. Briefly, this was done by compiling the output for all 600 samples, labeling each result as false or true based on the sample set that it came from, and then calculating (at each possible value of output) what the proportion of TP/[TP + FP] was for results with at least that level of raw output. Some simplification steps were taken, such as focusing on the species-level assignments (for comparison with methods that do not perform strain assignment), and only taking the top hit for each species from each dataset. Custom R and BASH scripts were used for the data compilation and analysis.\n",
      "Results\n",
      "The relationship of raw output value to PPV is shown for each of the four methods in [F2,\"Figure 2\"]. The point at which PPV is very close to 1 (where 95% of results are true positives) is ∼41,000 reads for Kraken, ∼2,800 reads for LMAT, ∼38% abundance for MetaPhlAn, and 0.21 for SIANN. For SIANN this means that having 38% of the species-unique genome covered by reads resulted in the vast majority of calls being accurate.\n",
      "For read-assignment methods (such as LMAT and Kraken), manual inspection of the results may yield a different understanding of confidence than is presented here, or in any automated analysis. For example, while each read that is assigned by LMAT and Kraken fall above a certain cutoff for species-specificity, some individual reads may be much more specific than others. One could identify a read that aligns to a single species of bacteria with 100% accuracy over its 300bp length, with the next closest match being only 90% similar. It is extremely unlikely that a 300bp exact match would arise due to random chance, and so the user could say with confidence that the organism of interest is found within the sequence data (not considering contamination, horizontal gene transfer, etc). However, such an approach is not currently implemented in an automated method, and many of the steps needed to make that assertion are performed manually by a domain expert, including alignment to near neighbors and ensuring that the read does not fall within a transposon, plasmid, etc. Therefore, while one could say that a single read is all that is needed to state with high PPV that an organism is present, the amount of reads assigned in an automated manner needed to achieve that level of PPV will number in the thousands ([F2,\"Fig 2\"]).\n",
      "The next phase of benchmarking was to determine how many raw input reads were needed to achieve the threshold for high PPV. To demonstrate this we plotted the known abundance of each spike organism against the PPV value generated by each method ([F3,\"Figure 3\"]). Each point (an organism at a known level of abundance) is comprised of a maximum of 50 replicates, where the diameter of each point increases with an increasing number of replicates. For demonstration purposes we are showing two pairs of bacteria and three viruses. Recall that for each of the pairs of bacteria (and the two poxviruses) any sample containing one did not contain the other (as shown in [T1,\"Table 1\"]). The empty boxes result from the organisms not being called at any abundance. For MetaPhlAn, that is a result of no viruses being included in the version of the reference database available for this analysis. Kraken assigned no reads to Hanta virus because viral RNA genomes were not included in this version of the reference database (personal communication with D. Wood). This emphasizes the point that a) the ability to create custom databases targeting organisms of interest can be valuable, and b) the performance of any method must be benchmarked against each potential target of interest. \n",
      "\n",
      "----------------------------------------------------------------------------------------------------\n",
      "F3 | Figure 3.\n",
      "https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F3.large.jpg\n",
      "The Positive Predictive Value (PPV, vertical axis) is shown for each organism (boxes on right), at each level of known abundance (horizontal axis, see Table 2), and each program (boxes at top), across a maximum of 50 replicates (indicated by the size of each point). Note that the reference database for MetaPhlAn does not include viruses, and the reference database for Kraken does not include RNA viruses\n",
      "----------------------------------------------------------------------------------------------------\n",
      "\n",
      "All methods were able identify the bulk of organisms in their databases at high abundances (75% and 18%, [F3,\"Figure 3\"]), however performance varied considerably at lower abundances and depended on the particular organism and method used. SIANN detected each organism at high confidence, even at levels as low as 0.3% and 0.07% of the total.\n",
      "Discussion\n",
      "The process of detecting trace amounts of a specific organism in a complex mixture of DNA is challenging enough for an expert, but that pales in comparison to the difficulty of accomplishing the same certainty of detection in an automated manner. The results presented here show that SIANN rapidly detects the presence of a given set of organisms with a high degree of specificity and sensitivity. For example, at the 95% confidence (PPV) cutoff of 0.2, SIANN reliably detects all of the organisms tested here at as low as 0.3% abundance. This strong performance is likely due to the fact that SIANN is able to use a method (read alignment to whole genomes) that would be far too computationally costly if it were applied to the entire collection of known genomes. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.\n",
      "SIANN is available on Illumina’s BaseSpace ([http://www.basespace.illumina.com,\"www.basespace.illumina.com\"]) as a NativeApp, with the database tested here ([app-1,\"Appendix 1\"]), as well as a database made from the NCBI representative set of prokaryotic genomes ([ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/,\"ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/\"]) ([app-2,\"Appendix 2\"]) and the complete set of NCBI viral genomes ([ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/,\"ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/\"]) ([app-3,\"Appendix 3\"]).\n",
      "BaseSpace was chosen as an appropriate release platform because while the entire set of software and dependencies can be deployed by the user from within a graphical user interface, the actual computation takes place in a controlled ‘cloud’ environment. Such a distribution strategy obviates the need to satisfy the multiple software or OS dependencies that often arises with academic computational methods. Results for SIANN are compiled into a report format, showing both the organisms that surpass 95% confidence, as well as the closest strain match for each species. The default view masks the raw data output, so that the results are human-readable and do not present extraneous information. While the code for execution and database-construction on a users system is available from Signature Science, LLC, additional databases on the BaseSpace platform can be made available upon request.\n",
      "There is a neverending list of questions that one could ask of metagenomic sequencing data generated from important samples. Instead of answering them all, we demonstrate a technique with a very narrow focus that is able to report with a high degree of confidence whether a given set of organisms is present in a sample. These results are presented to the user in a comprehensible format, and accessible on a commonly-used web platform. The world of bioinformatics will continue to progress and develop more sophisticated tools for metagenomic analysis, and we hope that the utility of SIANN will convince others to package and benchmark their tools in a way that they can be used with confidence by the larger public, as well as the research community.\n",
      "Scalable metagenomic taxonomy classification using a reference genome database, Ames SK, Hysom DA, Gardner SN, Lloyd GS, Gokhale MB, Allen JE2013\n",
      "\n",
      "A comparative evaluation of sequence classification programs, Bazinet AL, Cummings MP2012\n",
      "\n",
      "Rapid phylogenetic and functional classification of short genomic fragments with signature peptides, Berendzen J, Bruno WJ, Cohn JD, Hengartner NW, Kuske CR, McMahon BH, Wolinsky MA, Xie G2012\n",
      "\n",
      "Phymm and PhymmBL: metagenomic phylogenetic classification with interpolated Markov models, Brady A, Salzberg SL2009\n",
      "\n",
      "Integrative analysis of environmental sequences using MEGAN4, Huson DH, Mitra S, Weber N, Ruscheweyh H, Schuster SC2011\n",
      "\n",
      "PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data, Korbel JO, Abyzov A, Mu XJ, Carriero N, Cayting P, Zhang Z, Snyder M, Gerstein MB2009\n",
      "\n",
      "Fast gapped-read alignment with Bowtie 2, Langmead B, Salzberg S2012\n",
      "\n",
      "Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences, Liu B, Gibbons T, Ghodsi M, Treangen T, Pop M2011\n",
      "\n",
      "MetaSim: a sequencing simulator for genomics and metagenomics, Richter DC, Ott F, Auch AF, Schmid R, Huson DH2008\n",
      "\n",
      "Genomic variation landscape of the human gut microbiome, Schloissnig S, Arumugam M, Sunagawa S, Mitreva M, Tap J, Zhu A, Waller A, Mende DR, Kultima JR, Martin J, Kota K, Sunyaev SR, Weinstock GM, Bork P2013\n",
      "\n",
      "Computational meta’omics for microbial community studies, Segata N, Boernigen D, Tickle TL, Morgan XC, Garrett WS, Huttenhower C2013\n",
      "\n",
      "Metagenomic microbial community profiling using unique clade-specific marker genes, Segata N, Waldron L, Ballarini A, Narasimhan V, Jousson O, Huttenhower C2012\n",
      "\n",
      "Metagenomic species profiling using universal phylogenetic marker genes, Sunagawa S, Mende DR, Zeller G, Izquierdo-Carrasco F, Berger SA, Kultima JR, Coelho LP, Arumugam M, Tap J, Nielsen HB, Rasmussen S, Brunak S, Pedersen O, Guarner F, de Vos WM, Wang J, Li J, Doré J, Ehrlich SD, Stamatakis A, Bork P2013\n",
      "\n",
      "Ultrafast metagenomic sequence classification using exact alignments, Wood DE, Salzberg SL\n",
      "\n",
      "Linking long-term dietary patterns with gut microbial enterotypes, Wu GD, Chen J, Hoffmann C, Bittinger K, Chen YY, Keilbaugh SA, Bewtra M, Knights D, Walters WA, Knight R, Sinha R, Gilroy E, Gupta K, Baldassano R, Nessel L, Li H, Bushman FD, Lewis JD2011\n"
     ]
    }
   ],
   "source": [
    "print(test_paper.to_str())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyzing\n",
    "\n",
    "In a research paper, all sections are not equal. The Abstract should contain the most important words and the Results or Discussion should have the most interesting aspects. \n",
    "Let's analyze a paper for each of it's sections."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import spacy\n",
    "\n",
    "nlp = spacy.load('en_core_web_sm')\n",
    "\n",
    "stopwords = spacy.lang.en.stop_words.STOP_WORDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abstract\n",
      "Introduction\n",
      "Methods\n",
      "Results\n",
      "Discussion\n",
      "Appendix 1: Target Pathogen Database\n",
      "Appendix 2: Viral Database\n",
      "Appendix 3: Bacterial Database\n"
     ]
    }
   ],
   "source": [
    "for section in test_paper.paper['sections']:\n",
    "    print(section['title'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Abaca bunchy top virus DNA-C',\n",
       " 'Abaca bunchy top virus DNA-M',\n",
       " 'Abaca bunchy top virus DNA-N',\n",
       " 'Abaca bunchy top virus DNA-R',\n",
       " 'Abaca bunchy top virus DNA-S',\n",
       " 'Abaca bunchy top virus segment 2',\n",
       " 'Abalone shriveling syndrome-associated virus',\n",
       " 'Abutilon Brazil virus DNA A',\n",
       " 'Abutilon Brazil virus DNA B',\n",
       " 'Abutilon mosaic virus DNA A',\n",
       " 'Abutilon mosaic virus DNA B',\n",
       " 'Acanthocystis turfacea Chlorella virus 1',\n",
       " 'Acheta domesticus densovirus',\n",
       " 'Acholeplasma phage L2',\n",
       " 'Acholeplasma phage MV-L1',\n",
       " 'Acidianus bottle-shaped virus',\n",
       " 'Acidianus filamentous virus 1',\n",
       " 'Acidianus filamentous virus 2',\n",
       " 'Acidianus filamentous virus 3',\n",
       " 'Acidianus filamentous virus 6',\n",
       " 'Acidianus filamentous virus 7',\n",
       " 'Acidianus filamentous virus 8',\n",
       " 'Acidianus filamentous virus 9',\n",
       " 'Acidianus rod-shaped virus 1',\n",
       " 'Acidianus spindle-shaped virus 1',\n",
       " 'Acidianus two-tailed virus',\n",
       " 'Actinomyces phage Av-1',\n",
       " 'Actinoplanes phage phiAsp2',\n",
       " 'Acyrthosiphon pisum bacteriophage APSE-1',\n",
       " 'Adeno-associated virus-1',\n",
       " 'Adeno-associated virus-2',\n",
       " 'Adeno-associated virus-3',\n",
       " 'Adeno-associated virus-4',\n",
       " 'Adeno-associated virus 5',\n",
       " 'Adeno-associated virus-7',\n",
       " 'Adeno-associated virus-8',\n",
       " 'Adoxophyes honmai NPV',\n",
       " 'Adoxophyes orana granulovirus',\n",
       " 'Adoxophyes orana nucleopolyhedrovirus',\n",
       " 'Aedes aegypti densovirus',\n",
       " 'Aedes albopictus densovirus',\n",
       " 'Aedes taeniorhynchus iridescent virus',\n",
       " 'Aeromonas phage 25',\n",
       " 'Aeromonas phage 31',\n",
       " 'Aeromonas phage 44RR2.8t',\n",
       " 'Aeromonas phage Aeh1',\n",
       " 'Aeromonas phage phiO18P african cassava mosaic virus DNA A african cassava mosaic virus DNA B',\n",
       " 'African green monkey polyomavirus',\n",
       " 'African swine fever virus',\n",
       " 'Ageratum enation virus',\n",
       " 'Ageratum leaf Cameroon betasatellite',\n",
       " 'Ageratum leaf curl virus-G52',\n",
       " 'Ageratum yellow vein China virus-associated DNA beta',\n",
       " 'Ageratum yellow vein Chinavirus',\n",
       " 'Ageratum yellow vein Hualian virus-TaiwanHsinchutom2003 DNA A',\n",
       " 'Ageratum yellow vein Sri Lanka virus segment A',\n",
       " 'Ageratum yellow vein Taiwan virus',\n",
       " 'Ageratum yellow vein virusassociated DNA beta',\n",
       " 'Ageratum yellow veinvirus',\n",
       " 'Agrotis ipsilon multiple nucleopolyhedrovirus',\n",
       " 'Agrotis segetum granulovirus',\n",
       " 'Agrotis segetum nucleopolyhedrovirus',\n",
       " 'Alcelaphine herpesvirus 1',\n",
       " 'Aleutian mink disease virus',\n",
       " 'Allamanda leaf curl virus DNA-A',\n",
       " 'Alternanthera yellow vein virus DNA-A',\n",
       " 'Alternanthera yellow vein virus satellite DNA beta',\n",
       " 'Ambystoma tigrinum virus',\n",
       " 'Amsacta moorei entomopoxvirus L',\n",
       " 'Anguillid herpesvirus 1',\n",
       " 'Anopheles gambiae densonucleosis virus',\n",
       " 'Antheraea pernyi nucleopolyhedrovirus',\n",
       " 'Anticarsia gemmatalis nucleopolyhedrovirus',\n",
       " 'Archaeal BJ1 virus',\n",
       " 'Ateline herpesvirus 3',\n",
       " 'Autographa californica nucleopolyhedrovirus',\n",
       " 'Avian adeno-associated virus',\n",
       " 'ATCC VR-865',\n",
       " 'Avian adeno-associated virus strain DA-1',\n",
       " 'Avian endogenous retrovirus EAV-HP',\n",
       " 'Azospirillum phage Cd',\n",
       " 'Bacillus phage 0305phi8-36',\n",
       " 'Bacillus phage AP50',\n",
       " 'Bacillus phage B103',\n",
       " 'Bacillus phage Bam35c',\n",
       " 'Bacillus phage BCJA1c',\n",
       " 'Bacillus phage Cherry',\n",
       " 'Bacillus phage Fah',\n",
       " 'Bacillus phage GA-1',\n",
       " 'Bacillus phage Gamma',\n",
       " 'Bacillus phage GIL16c',\n",
       " 'Bacillus phage IEBH',\n",
       " 'Bacillus phage phi105',\n",
       " 'Bacillus phage phi29',\n",
       " 'Bacillus phage SPBc2',\n",
       " 'Bacillus phage SPO1',\n",
       " 'Bacillus phage SPP1',\n",
       " 'Bacillus phage TP21-L',\n",
       " 'Bacillus phage WBeta',\n",
       " 'Bacillus prophage phBC6A51',\n",
       " 'Bacillus prophage phBC6A52',\n",
       " 'Bacillus virus 1',\n",
       " 'Bacteriophage Aaphi23',\n",
       " 'Bacteriophage APSE-2',\n",
       " 'Bacteriophage PSA',\n",
       " 'Bacteriophage RB32',\n",
       " 'Bacteroides phage B40-8',\n",
       " 'Banana bunchy top virus DNA C',\n",
       " 'Banana bunchy top virus DNA M',\n",
       " 'Banana bunchy top virus DNA N',\n",
       " 'Banana bunchy top virus DNA R',\n",
       " 'Banana bunchy top virus DNA S',\n",
       " 'Banana bunchy top virus DNA U3',\n",
       " 'Banana streak GF virus',\n",
       " 'Banana streak Mysore virus',\n",
       " 'Banana streak OL virus',\n",
       " 'Banana streak virus genome',\n",
       " 'Banana streak virus strain Acuminata Vietnam',\n",
       " 'Bandicoot papillomatosis carcinomatosis virus type 1',\n",
       " 'Bandicoot papillomatosis carcinomatosis virus type 2',\n",
       " 'Bat adeno-associated virus YNM',\n",
       " 'Bdellovibrio phage phiMH2K',\n",
       " 'Beak and feather disease virus',\n",
       " 'Bean calico mosaic virus DNA A',\n",
       " 'Bean calico mosaic virus DNA B',\n",
       " 'Bean dwarf mosaic virus DNA A',\n",
       " 'Bean dwarf mosaic virus DNA B',\n",
       " 'Bean golden mosaic virus DNA A',\n",
       " 'Bean golden mosaic virus DNA B',\n",
       " 'Bean golden yellow mosaic virus DNA A',\n",
       " 'Bean golden yellow mosaic virus DNA B',\n",
       " 'Bean yellow dwarf virus putative genes V1',\n",
       " 'Beet curly top Iran virus-K',\n",
       " 'Beet curly top virus-California Logan',\n",
       " 'Beet mild curly top virus-Worland4',\n",
       " 'Beet severe curly top virus-Cfh',\n",
       " 'Begomovirus-associated DNA II',\n",
       " 'Begomovirus-associated DNA-III',\n",
       " 'Bettongia penicillata papillomavirus 1',\n",
       " 'Bhendi yellow vein',\n",
       " 'Bhubhaneswar virus DNA-A',\n",
       " 'Bhendi yellow vein Delhi virus 2004New Delhi DNA-A',\n",
       " 'Bhendi yellow vein mosaic virus-associated DNA beta',\n",
       " 'Bhendi yellow vein mosaic Virus',\n",
       " 'Bitter gourd leaf curl disease-associated DNA beta',\n",
       " 'BK polyomavirus',\n",
       " 'Blainvillea yellow spot virus DNA-A',\n",
       " 'Blainvillea yellow spot virus DNA-B',\n",
       " 'Blattella germanica densovirus',\n",
       " 'Blueberry red ringspot virus',\n",
       " 'Bocavirus gorillaGBoV12009',\n",
       " 'Bombyx mandarina nucleopolyhedrovirus',\n",
       " 'Bombyx mori densovirus 5',\n",
       " 'Bombyx mori NPV',\n",
       " 'Bordetella phage BIP-1',\n",
       " 'Bordetella phage BMP-1',\n",
       " 'Bordetella phage BPP-1',\n",
       " 'Bougainvillea spectabilis chlorotic vein-banding virus',\n",
       " 'Bovine adeno-associated virus',\n",
       " 'Bovine adenovirus A',\n",
       " 'Bovine adenovirus B',\n",
       " 'Bovine adenovirus D',\n",
       " 'Bovine ephemeral fever virus',\n",
       " 'Bovine foamy virus',\n",
       " 'Bovine herpesvirus 1',\n",
       " 'Bovine herpesvirus 4 long unique region',\n",
       " 'Bovine herpesvirus 5',\n",
       " 'Bovine papillomavirus-1',\n",
       " 'Bovine papillomavirus 3',\n",
       " 'Bovine papular stomatitis virus',\n",
       " 'Bovine parvovirus 2',\n",
       " 'Bovine Parvovirus',\n",
       " 'Bovine polyomavirus',\n",
       " 'Burkholderia ambifaria phage BcepF1',\n",
       " 'Burkholderia phage Bcep 176',\n",
       " 'Burkholderia phage Bcep1',\n",
       " 'Burkholderia phage Bcep22',\n",
       " 'Burkholderia phage Bcep43',\n",
       " 'Burkholderia phage Bcep781',\n",
       " 'Burkholderia phage BcepB1A',\n",
       " 'Burkholderia phage BcepC6B',\n",
       " 'Burkholderia phage BcepGomr',\n",
       " 'Burkholderia phage BcepIL02',\n",
       " 'Burkholderia phage BcepMu',\n",
       " 'Burkholderia phage BcepNazgul',\n",
       " 'Burkholderia phage BcepNY3',\n",
       " 'Burkholderia phage KS10',\n",
       " 'Burkholderia phage KS9',\n",
       " 'Burkholderia phage phi1026b',\n",
       " 'Burkholderia phage phi644-2 chromosome',\n",
       " 'Burkholderia phage phiE12-2 chromosome',\n",
       " 'Burkholderia phage phiE125',\n",
       " 'Burkholderia phage phiE202 chromosome',\n",
       " 'Burkholderia phage phiE255 chromosome',\n",
       " 'Burkholderia prophage phi52237',\n",
       " 'Cabbage leaf curl virus DNA A',\n",
       " 'Cabbage leaf curl virus DNA B',\n",
       " 'Cacao swollen shoot virus',\n",
       " 'California sea lion anellovirus',\n",
       " 'California sea lion polyomavirus 1',\n",
       " 'Callitrichine herpesvirus 3 strain CJ0149',\n",
       " 'Camelpox virus',\n",
       " 'Campoletis sonorensis ichnovirus chromosome segment W',\n",
       " 'Campoletis sonorensis ichnovirus segment B',\n",
       " 'Campoletis sonorensis ichnovirus segment C',\n",
       " 'Campoletis sonorensis ichnovirus segment D',\n",
       " 'Campoletis sonorensis ichnovirus segment E',\n",
       " 'Campoletis sonorensis ichnovirus segment F',\n",
       " 'Campoletis sonorensis ichnovirus segmentG2',\n",
       " 'Campoletis sonorensis ichnovirus segment G',\n",
       " 'Campoletis sonorensis ichnovirus segment H',\n",
       " 'Campoletis sonorensis ichnovirus segmentI2',\n",
       " 'Campoletis sonorensis ichnovirus segment I',\n",
       " 'Campoletis sonorensis ichnovirus segment J',\n",
       " 'Campoletis sonorensis ichnovirus segment L',\n",
       " 'Campoletis sonorensis ichnovirus segment M',\n",
       " 'Campoletis sonorensis ichnovirus segment N',\n",
       " 'Campoletis sonorensis ichnovirus segment O1',\n",
       " 'Campoletis sonorensis ichnovirus segment P',\n",
       " 'Campoletis sonorensis ichnovirus segment Q',\n",
       " 'Campoletis sonorensis ichnovirus segment T',\n",
       " 'Campoletis sonorensis ichnovirus segment U',\n",
       " 'Campoletis sonorensis ichnovirus segment V',\n",
       " 'Campoletis sonorensis ichnovirus segment Z',\n",
       " 'Campoletis sonorensis ichnovirus superhelical segment A',\n",
       " 'Campoletis sonorensis ichnovirus superhelical segment aprime',\n",
       " 'Canary circovirus',\n",
       " 'Canarypox virus',\n",
       " 'Canine adenovirus Canine minute virus',\n",
       " 'Canine oral papillomavirus',\n",
       " 'Canine papillomavirus 2',\n",
       " 'Canine papillomavirus 3',\n",
       " 'Canine papillomavirus 4',\n",
       " 'Canine papillomavirus 6',\n",
       " 'Canine parvovirus',\n",
       " 'Capra hircus papillomavirus type 1',\n",
       " 'Cardiospermum yellow leaf curl virus satellite DNA beta',\n",
       " 'Caretta caretta papillomavirus 1',\n",
       " 'Carnation etched ring virus',\n",
       " 'Casphalia extranea densovirus',\n",
       " 'Cassava vein mosaic virus',\n",
       " 'Cauliflower mosaic virus',\n",
       " 'Cercopithecine herpesvirus 2',\n",
       " 'Cercopithecine herpesvirus 5 strain 2715',\n",
       " 'Cercopithecine herpesvirus 9',\n",
       " 'Cestrum yellow leaf curling virus',\n",
       " 'Chaetoceros salsugineum DNA virus',\n",
       " 'Chayote yellow mosaic virus',\n",
       " 'Chicken anemia virus',\n",
       " 'Chickpea chlorotic dwarf virus',\n",
       " 'Chilli leaf curl disease associated sequence virion',\n",
       " 'Chilli leaf curl Multan alphasatellite',\n",
       " 'Chilli leaf curl virus',\n",
       " 'Chino del tomate virus DNA A',\n",
       " 'Chino del tomate virus DNA B',\n",
       " 'Chlamydia phage 3',\n",
       " 'Chlamydia phage 4',\n",
       " 'Chlamydia phage Chp1',\n",
       " 'Chlamydia phage Chp2',\n",
       " 'Chlamydia phage PhiCPG1',\n",
       " 'Chloris striate mosaic virus',\n",
       " 'Choristoneura fumiferana DEF MNPV',\n",
       " 'Choristoneura fumiferana MNPV',\n",
       " 'Choristoneura occidentalis granulovirus',\n",
       " 'Chrysodeixis chalcites nucleopolyhedrovirus',\n",
       " 'Circovirus-like genome BBC-A',\n",
       " 'Circovirus-like genome CB-A',\n",
       " 'Circovirus-like genome CB-B',\n",
       " 'Circovirus-like genome RW-A',\n",
       " 'Circovirus-like genome RW-B',\n",
       " 'Circovirus-like genome RW-C',\n",
       " 'Circovirus-like genome RW-D',\n",
       " 'Circovirus-like genome RW-E',\n",
       " 'Circovirus-like genome SAR-A',\n",
       " 'Circovirus-like genome SAR-B',\n",
       " 'Citrus psorosis virus RNA1',\n",
       " 'Citrus psorosis virus RNA2',\n",
       " 'Citrus psorosis virus RNA3',\n",
       " 'Citrus yellow mosaic virus',\n",
       " 'Clanis bilineata nucleopolyhedrosis virus',\n",
       " 'Clavibacter phage CMP1',\n",
       " 'Clerodendron yellow mosaic virus',\n",
       " 'Clerodendrum golden mosaic',\n",
       " 'China virus DNA A',\n",
       " 'Clerodendrum golden mosaic',\n",
       " 'China virus DNA B',\n",
       " 'Clerodendrum golden mosaic virus DNA-A',\n",
       " 'Clerodendrum golden mosaic virus DNA-B',\n",
       " 'Clostridium phage 39-O',\n",
       " 'Clostridium phage c-st',\n",
       " 'Clostridium phage phi3626',\n",
       " 'Clostridium phage phiC2',\n",
       " 'Clostridium phage phi CD119',\n",
       " 'Clostridium phage phiCD27',\n",
       " 'Clostridium phage phiCTP1',\n",
       " 'Coconut foliar decay virus',\n",
       " 'Columbid circovirus',\n",
       " 'Commelina yellow mottle virus',\n",
       " 'Corchorus golden mosaic virus DNA-A',\n",
       " 'Corchorus golden mosaic virus DNA-B',\n",
       " 'Corchorus yellow spot virus DNA A',\n",
       " 'Corchorus yellow spot virus DNA B',\n",
       " 'Corchorus yellow vein virus-Hoa Binh DNA A',\n",
       " 'Corchorus yellow vein virus-Hoa Binh DNA B',\n",
       " 'Corynebacterium phage BFK20',\n",
       " 'Corynebacterium phage P1201',\n",
       " 'Cotesia congregata bracovirus segment Circle10',\n",
       " 'Cotesia congregata bracovirus segment Circle11',\n",
       " 'Cotesia congregata bracovirus segment Circle12',\n",
       " 'Cotesia congregata bracovirus segment Circle13',\n",
       " 'Cotesia congregata bracovirus segment Circle14',\n",
       " 'Cotesia congregata bracovirus segment Circle15',\n",
       " 'Cotesia congregata bracovirus segment Circle17',\n",
       " 'Cotesia congregata bracovirus segment Circle19',\n",
       " 'Cotesia congregata bracovirus segment circle1',\n",
       " 'Cotesia congregata bracovirus segment Circle20',\n",
       " 'Cotesia congregata bracovirus segment Circle21',\n",
       " 'Cotesia congregata bracovirus segment Circle22',\n",
       " 'Cotesia congregata bracovirus segment Circle23',\n",
       " 'Cotesia congregata bracovirus segment Circle25',\n",
       " 'Cotesia congregata bracovirus segment Circle26',\n",
       " 'Cotesia congregata bracovirus segment circle2',\n",
       " 'Cotesia congregata bracovirus segment Circle30',\n",
       " 'Cotesia congregata bracovirus segment Circle31',\n",
       " 'Cotesia congregata bracovirus segment Circle32',\n",
       " 'Cotesia congregata bracovirus segment Circle33',\n",
       " 'Cotesia congregata bracovirus segment Circle35',\n",
       " 'Cotesia congregata bracovirus segment Circle36',\n",
       " 'Cotesia congregata bracovirus segment circle3',\n",
       " 'Cotesia congregata bracovirus segment Circle4',\n",
       " 'Cotesia congregata bracovirus segment Circle5',\n",
       " 'Cotesia congregata bracovirus segment Circle9',\n",
       " 'Cotesia congregata virus segment Circle18',\n",
       " 'Cotesia congregata virus segment Circle6',\n",
       " 'Cotesia congregata virus segment Circle7',\n",
       " 'Cotesia congregata virus segment Circle8',\n",
       " 'Cotton leaf crumple geminivirus DNA B',\n",
       " 'Cotton leaf crumple virus DNA A',\n",
       " 'Cotton leaf curl Alabad virus',\n",
       " 'Cotton leaf curl Bangalore virus-associated DNA beta',\n",
       " 'Cotton leaf curl Bangalore virus segment A',\n",
       " 'Cotton leaf curl Burewala alphasatellite',\n",
       " 'Cotton leaf curl Burewala betasatellite',\n",
       " 'Cotton leaf curl Burewala virus-IndiaVehari2004',\n",
       " 'Cotton leaf curl Gezira alphasatellite',\n",
       " 'Cotton leaf curl Gezira beta',\n",
       " 'Cotton leaf curl Gezira Betasatellite extrachromosomal',\n",
       " 'Cotton leaf curl Gezira virus',\n",
       " 'Cotton leaf curl Kokhran virus',\n",
       " 'Cotton leaf curl Multan Virus',\n",
       " 'Cotton leaf curl Multan virus satellite DNA beta',\n",
       " 'Cotton leaf curl Multan virus satellite U36-1',\n",
       " 'Cotton leaf curl Rajasthan virus segment A',\n",
       " 'Cotton leaf curl virus-associated DNA beta',\n",
       " 'Cottontail rabbit papillomavirus',\n",
       " 'Cowpea severe leaf curl-associated DNA beta',\n",
       " 'Cowpox virus Crassocephalum yellow vein virus-Jinghong',\n",
       " 'Croton yellow vein mosaic alphasatellite',\n",
       " 'Croton yellow vein mosaic virus',\n",
       " 'Croton yellow vein mosaic Virus satellite DNA beta',\n",
       " 'Croton yellow vein virus Crow polyomavirus',\n",
       " 'Cryptophlebia leucotreta granulovirus',\n",
       " 'Cucurbita yellow vein virus-associated DNA beta',\n",
       " 'Cucurbit leaf crumple virus DNA A',\n",
       " 'Cucurbit leaf crumple virus DNA B',\n",
       " 'Culex nigripalpus NPV',\n",
       " 'Culex pipiens densovirus Cyanophage PSS2',\n",
       " 'Cyanophage Syn5 Cycad leaf necrosis virus',\n",
       " 'Cydia pomonella granulovirus',\n",
       " 'Cyprinid herpesvirus 3',\n",
       " 'Deer papillomavirus',\n",
       " 'Deerpox virus W-1170-84',\n",
       " 'Deerpox virus W-848-83',\n",
       " 'Deftia phage phiW-14',\n",
       " 'Dendrolimus punctatus densovirus',\n",
       " 'Desmodium leaf distortion virus DNA A',\n",
       " 'Desmodium leaf distortion virus DNA B',\n",
       " 'Diadromus pulchellus ascovirus 4a',\n",
       " 'Diatraea saccharalis densovirus',\n",
       " 'Dicliptera yellow mottle virus DNA A',\n",
       " 'Dicliptera yellow mottle virus DNA B',\n",
       " 'Digitaria streak virus',\n",
       " 'Dioscorea bacilliform virus',\n",
       " 'Dolichos yellow mosaic virus',\n",
       " 'Dracaena mottle virus',\n",
       " 'Duck adenovirus A',\n",
       " 'Duck circovirus',\n",
       " 'Duck hepatitis B virus',\n",
       " 'East African cassava mosaic Cameroon virus DNA A',\n",
       " 'East African cassava mosaic Cameroon virus DNA B',\n",
       " 'East African cassava mosaic Kenya virus DNA A',\n",
       " 'East African cassava mosaic Kenya virus DNA B',\n",
       " 'East African cassava mosaic virus DNA A',\n",
       " 'East African cassava mosaic virus DNA B',\n",
       " 'East African cassava mosaic Zanzibar virus DNA-A',\n",
       " 'East African cassava mosaic Zanzibar virus DNA B',\n",
       " 'Ecotropis obliqua NPV',\n",
       " 'Ectocarpus siliculosus virus 1',\n",
       " 'Ectromelia virus',\n",
       " 'Emiliania huxleyi virus 86',\n",
       " 'Emilia yellow vein virus-associated DNA beta',\n",
       " 'Emilia yellow vein virus-Fz1',\n",
       " 'Enterobacteria phage 13a',\n",
       " 'Enterobacteria phage 933W',\n",
       " 'Enterobacteria phage alpha3',\n",
       " 'Enterobacteria phage BA14',\n",
       " 'Enterobacteria phage BP-4795',\n",
       " 'Enterobacteria phage EcoDS1',\n",
       " 'Enterobacteria phage EPS7',\n",
       " 'Enterobacteria phage epsilon15',\n",
       " 'Enterobacteria phage ES18',\n",
       " 'Enterobacteria phage Felix 01',\n",
       " 'Enterobacteria phage Fels-2',\n",
       " 'Enterobacteria phage G4',\n",
       " 'Enterobacteria phage HK022',\n",
       " 'Enterobacteria phage HK620',\n",
       " 'Enterobacteria phage HK97',\n",
       " 'Enterobacteria phage I2-2',\n",
       " 'Enterobacteria phage ID18',\n",
       " 'Enterobacteria phage ID2 MoscowID2001',\n",
       " 'Enterobacteria phage If1',\n",
       " 'Enterobacteria phage Ike',\n",
       " 'Enterobacteria phage IME08',\n",
       " 'Enterobacteria phage JK06',\n",
       " 'Enterobacteria phage JS10',\n",
       " 'Enterobacteria phage JS98',\n",
       " 'Enterobacteria phage JSE',\n",
       " 'Enterobacteria phage K1-5',\n",
       " 'Enterobacteria phage K1E',\n",
       " 'Enterobacteria phage K1F',\n",
       " 'Enterobacteria phage lambda',\n",
       " 'Enterobacteria phage LKA1',\n",
       " 'Enterobacteria phage M13',\n",
       " 'Enterobacteria phage Min27',\n",
       " 'Enterobacteria phage Mu',\n",
       " 'Enterobacteria phage N15',\n",
       " 'Enterobacteria phage N4',\n",
       " 'Enterobacteria phage P1',\n",
       " 'Enterobacteria phage P22 virus',\n",
       " 'Enterobacteria phage P2 virus',\n",
       " 'Enterobacteria phage P4',\n",
       " 'Enterobacteria phage Phi1',\n",
       " 'Enterobacteria phage phiEco32',\n",
       " 'Enterobacteria phage phiEcoM-GJ1',\n",
       " 'Enterobacteria phage phiP27',\n",
       " 'Enterobacteria phage phiX174 sensu lato',\n",
       " 'Enterobacteria phage PRD1',\n",
       " 'Enterobacteria phage PsP3',\n",
       " 'Enterobacteria phage RB14',\n",
       " 'Enterobacteria phage RB16',\n",
       " 'Enterobacteria phage RB43',\n",
       " 'Enterobacteria phage RB49',\n",
       " 'Enterobacteria phage RB51',\n",
       " 'Enterobacteria phage RB69',\n",
       " 'Enterobacteria phage RTP',\n",
       " 'Enterobacteria phage Sf6',\n",
       " 'Enterobacteria phage SfV',\n",
       " 'Enterobacteria phage SP6',\n",
       " 'Enterobacteria phage SSL-2009a',\n",
       " 'Enterobacteria phage ST104',\n",
       " 'Enterobacteria phage St-1',\n",
       " 'Enterobacteria phage ST64T',\n",
       " 'Enterobacteria phage T1',\n",
       " 'Enterobacteria phage T3',\n",
       " 'Enterobacteria phage T4',\n",
       " 'Enterobacteria phage T5',\n",
       " 'Enterobacteria phage T7',\n",
       " 'Enterobacteria phage TLS',\n",
       " 'Enterobacteria phage VT2-Sakai',\n",
       " 'Enterobacteria phage WA13',\n",
       " 'Enterobacteria phage WV8',\n",
       " 'Enterobacteria phage YYZ-2008',\n",
       " 'Enterococcus faecalis 62 chromosome',\n",
       " 'Enterococcus faecalis 62 plasmid EF62pA',\n",
       " 'Enterococcus faecalis 62 plasmid EF62pB',\n",
       " 'Enterococcus faecalis 62 plasmid EF62pC',\n",
       " 'Enterococcus phage EF62phi',\n",
       " 'Enterococcus phage EFAP-1',\n",
       " 'Enterococcus phage phiEf11',\n",
       " 'Enterococcus phage phiEF24C',\n",
       " 'Enterococcus phage phiFL1A',\n",
       " 'Enterococcus phage phiFL2A',\n",
       " 'Enterococcus phage phiFL3A',\n",
       " 'Enterococcus phage phiFL4A',\n",
       " 'Enzootic nasal tumour virus of goats',\n",
       " 'Epiphyas postvittana NPV',\n",
       " 'Equid herpesvirus 1',\n",
       " 'Equid herpesvirus 2',\n",
       " 'Equid herpesvirus 4',\n",
       " 'Equid herpesvirus 9',\n",
       " 'Equine papillomavirus 2',\n",
       " 'Eragrostis curvula streak virus',\n",
       " 'Eragrostis streak virus',\n",
       " 'Erectites yellow mosaic virus DNA-A',\n",
       " 'Erectites yellow mosaic virus satellite DNA beta',\n",
       " 'Erethizon dorsatum papillomavirus type 1',\n",
       " 'Erinaceus europaeus papillomavirus',\n",
       " 'Erwinia amylovora phage Era103',\n",
       " 'Erwinia phage phiEa21-4',\n",
       " 'Escherichia coli bacteriophage rv5',\n",
       " 'Escherichia phage D108',\n",
       " 'Escherichia phage phiV10',\n",
       " 'Eupatorium vein clearing virus',\n",
       " 'Eupatorium yellow vein associated DNA beta',\n",
       " 'Eupatorium yellow vein virus',\n",
       " 'Euphorbia leaf curl virus DNA A',\n",
       " 'Euphorbia yellow mosaic virus DNA A',\n",
       " 'Euphorbia yellow mosaic virus DNA B',\n",
       " 'Euproctis pseudoconspersa nucleopolyhedrovirus',\n",
       " 'European elk papillomavirus',\n",
       " 'Faba bean necrotic stunt virus DNA C',\n",
       " 'Faba bean necrotic stunt virus DNA M',\n",
       " 'Faba bean necrotic stunt virus DNA N',\n",
       " 'Faba bean necrotic stunt virus DNA R',\n",
       " 'Faba bean necrotic stunt virus DNA S',\n",
       " 'Faba bean necrotic stunt virus DNA U1',\n",
       " 'Faba bean necrotic stunt virus DNA U2',\n",
       " 'Faba bean necrotic stunt virus DNA U4',\n",
       " 'Faba bean necrotic yellows virus DNA 10',\n",
       " 'Faba bean necrotic yellows virus DNA-1',\n",
       " 'Faba bean necrotic yellows virus DNA 2',\n",
       " 'Faba bean necrotic yellows virus DNA 4',\n",
       " 'Faba bean necrotic yellows virus DNA 5',\n",
       " 'Faba bean necrotic yellows virus DNA 7',\n",
       " 'Faba bean necrotic yellows virus DNA 8',\n",
       " 'Faba bean necrotic yellows virus DNA 9',\n",
       " 'Faba bean necrotic yellows Virus',\n",
       " 'Feldmannia species virus Felid herpesvirus 1',\n",
       " 'Felis domesticus papillomavirus type 1',\n",
       " 'Fenneropenaeus chinensis hepatopancreatic densovirus',\n",
       " 'Figwort mosaic virus',\n",
       " 'Finch circovirus Finch polyomavirus',\n",
       " 'Flavobacterium phage 11b',\n",
       " 'Fowl adenovirus A',\n",
       " 'Fowl adenovirus D',\n",
       " 'Fowlpox virus',\n",
       " 'Francolinus leucoscepus papillomavirus 1',\n",
       " 'Fringilla coelebs papillomavirus',\n",
       " 'Frog adenovirus 1',\n",
       " 'Frog virus 3',\n",
       " 'Galleria mellonella densovirus',\n",
       " 'Gallid herpesvirus 1',\n",
       " 'Gallid herpesvirus 2',\n",
       " 'Gallid herpesvirus 3',\n",
       " 'Gammapapillomavirus HPV127',\n",
       " 'Geobacillus phage',\n",
       " 'GBSV1 Geobacillus virus E2',\n",
       " 'Glossina pallidipes salivary gland hypertrophy virus',\n",
       " 'Glypta fumiferanae ichnovirus segmentA10',\n",
       " 'Glypta fumiferanae ichnovirus segment A1',\n",
       " 'Glypta fumiferanae ichnovirus segment A2',\n",
       " 'Glypta fumiferanae ichnovirus segment A3',\n",
       " 'Glypta fumiferanae ichnovirus segment A4',\n",
       " 'Glypta fumiferanae ichnovirus segment A5',\n",
       " 'Glypta fumiferanae ichnovirus segment A6',\n",
       " 'Glypta fumiferanae ichnovirus segment A7',\n",
       " 'Glypta fumiferanae ichnovirus segment A8',\n",
       " 'Glypta fumiferanae ichnovirus segment A9',\n",
       " 'Glypta fumiferanae ichnovirus segment B10',\n",
       " 'Glypta fumiferanae ichnovirus segment B11',\n",
       " 'Glypta fumiferanae ichnovirus segment B12',\n",
       " 'Glypta fumiferanae ichnovirus segment B13',\n",
       " 'Glypta fumiferanae ichnovirus segment B14',\n",
       " 'Glypta fumiferanae ichnovirus segment B15',\n",
       " 'Glypta fumiferanae ichnovirus segment B16',\n",
       " 'Glypta fumiferanae ichnovirus segment B17',\n",
       " 'Glypta fumiferanae ichnovirus segment B18',\n",
       " 'Glypta fumiferanae ichnovirus segment B19',\n",
       " 'Glypta fumiferanae ichnovirus segment-B1',\n",
       " 'Glypta fumiferanae ichnovirus segment B20',\n",
       " 'Glypta fumiferanae ichnovirus segment B21',\n",
       " 'Glypta fumiferanae ichnovirus segment B22',\n",
       " 'Glypta fumiferanae ichnovirus segment B23',\n",
       " 'Glypta fumiferanae ichnovirus segment B24',\n",
       " 'Glypta fumiferanae ichnovirus segment B25',\n",
       " 'Glypta fumiferanae ichnovirus segment B26',\n",
       " 'Glypta fumiferanae ichnovirus segment B27',\n",
       " 'Glypta fumiferanae ichnovirus segment B28',\n",
       " 'Glypta fumiferanae ichnovirus segment B29',\n",
       " 'Glypta fumiferanae ichnovirus segment-B2',\n",
       " 'Glypta fumiferanae ichnovirus segment B30',\n",
       " 'Glypta fumiferanae ichnovirus segment B31',\n",
       " 'Glypta fumiferanae ichnovirus segment B32',\n",
       " 'Glypta fumiferanae ichnovirus segment B33',\n",
       " 'Glypta fumiferanae ichnovirus segment B34',\n",
       " 'Glypta fumiferanae ichnovirus segment B35',\n",
       " 'Glypta fumiferanae ichnovirus segment B36',\n",
       " 'Glypta fumiferanae ichnovirus segment B37',\n",
       " 'Glypta fumiferanae ichnovirus segment B38',\n",
       " 'Glypta fumiferanae ichnovirus segment B39',\n",
       " 'Glypta fumiferanae ichnovirus segment-B3',\n",
       " 'Glypta fumiferanae ichnovirus segment B40',\n",
       " 'Glypta fumiferanae ichnovirus segment B41',\n",
       " 'Glypta fumiferanae ichnovirus segment B42',\n",
       " 'Glypta fumiferanae ichnovirus segment B43',\n",
       " 'Glypta fumiferanae ichnovirus segment B44',\n",
       " 'Glypta fumiferanae ichnovirus segment B45',\n",
       " 'Glypta fumiferanae ichnovirus segment B46',\n",
       " 'Glypta fumiferanae ichnovirus segment B47',\n",
       " 'Glypta fumiferanae ichnovirus segment B48',\n",
       " 'Glypta fumiferanae ichnovirus segment B49',\n",
       " 'Glypta fumiferanae ichnovirus segment-B4',\n",
       " 'Glypta fumiferanae ichnovirus segment B50',\n",
       " 'Glypta fumiferanae ichnovirus segment B51',\n",
       " 'Glypta fumiferanae ichnovirus segment B52',\n",
       " 'Glypta fumiferanae ichnovirus segment B53',\n",
       " 'Glypta fumiferanae ichnovirus segment B54',\n",
       " 'Glypta fumiferanae ichnovirus segment B55',\n",
       " 'Glypta fumiferanae ichnovirus segment B56',\n",
       " 'Glypta fumiferanae ichnovirus segment B57',\n",
       " 'Glypta fumiferanae ichnovirus segment B58',\n",
       " 'Glypta fumiferanae ichnovirus segment B59',\n",
       " 'Glypta fumiferanae ichnovirus segment-B5',\n",
       " 'Glypta fumiferanae ichnovirus segment B60',\n",
       " 'Glypta fumiferanae ichnovirus segment B61',\n",
       " 'Glypta fumiferanae ichnovirus segment B62',\n",
       " 'Glypta fumiferanae ichnovirus segment B63',\n",
       " 'Glypta fumiferanae ichnovirus segment B64',\n",
       " 'Glypta fumiferanae ichnovirus segment B65',\n",
       " 'Glypta fumiferanae ichnovirus segment-B6',\n",
       " 'Glypta fumiferanae ichnovirus segment B7',\n",
       " 'Glypta fumiferanae ichnovirus segment B8',\n",
       " 'Glypta fumiferanae ichnovirus segment B9',\n",
       " 'Glypta fumiferanae ichnovirus segment C10',\n",
       " 'Glypta fumiferanae ichnovirus segment C11',\n",
       " 'Glypta fumiferanae ichnovirus segment C12',\n",
       " 'Glypta fumiferanae ichnovirus segment C13',\n",
       " 'Glypta fumiferanae ichnovirus segment C14',\n",
       " 'Glypta fumiferanae ichnovirus segment C15',\n",
       " 'Glypta fumiferanae ichnovirus segment C16',\n",
       " 'Glypta fumiferanae ichnovirus segment C17',\n",
       " 'Glypta fumiferanae ichnovirus segment C18',\n",
       " 'Glypta fumiferanae ichnovirus segment C19',\n",
       " 'Glypta fumiferanae ichnovirus segment-C1',\n",
       " 'Glypta fumiferanae ichnovirus segment C20',\n",
       " 'Glypta fumiferanae ichnovirus segment C21',\n",
       " 'Glypta fumiferanae ichnovirus segment C22',\n",
       " 'Glypta fumiferanae ichnovirus segment-C2',\n",
       " 'Glypta fumiferanae ichnovirus segment C3',\n",
       " 'Glypta fumiferanae ichnovirus segment C4',\n",
       " 'Glypta fumiferanae ichnovirus segment C5',\n",
       " 'Glypta fumiferanae ichnovirus segment C6',\n",
       " 'Glypta fumiferanae ichnovirus segment C7',\n",
       " 'Glypta fumiferanae ichnovirus segment C8',\n",
       " 'Glypta fumiferanae ichnovirus segment C9',\n",
       " 'Glypta fumiferanae ichnovirus segment D1',\n",
       " 'Glypta fumiferanae ichnovirus segment D2',\n",
       " 'Glypta fumiferanae ichnovirus segment D3',\n",
       " 'Glypta fumiferanae ichnovirus segment D4',\n",
       " 'Glypta fumiferanae ichnovirus segment D5',\n",
       " 'Glypta fumiferanae ichnovirus segment D6',\n",
       " 'Glypta fumiferanae ichnovirus segment D7',\n",
       " 'Glypta fumiferanae ichnovirus segment E1',\n",
       " 'Goatpox virus Pellor Goose circovirus',\n",
       " 'Goose hemorrhagic polyomavirus',\n",
       " 'Goose parvovirus',\n",
       " 'Gossypium darwinii symptomless alphasatellite DNA-alpha',\n",
       " 'Gossypium darwinii symptomless virus DNA-A',\n",
       " 'Gossypium davidsonii symptomless alphasatellite DNA-alpha-B',\n",
       " 'Gossypium mustilinum symptomless alphasatellite DNA-alpha-B',\n",
       " 'Gossypium punctatum mild leaf curl virus DNA A',\n",
       " 'Gossypium punctatum mild leaf curl virus DNA B',\n",
       " 'Ground squirrel hepatitis virus',\n",
       " 'Gryllus bimaculatus nudivirus Gull circovirus',\n",
       " 'Haemophilus phage HP1',\n",
       " 'Haemophilus phage HP2',\n",
       " 'Haloarcula hispanica pleomorphic virus 1',\n",
       " 'Haloarcula phage SH1',\n",
       " 'Halomonas phage phiHAP-1',\n",
       " 'Halorubrum phage HF2',\n",
       " 'Halorubrum pleomorphic virus 1',\n",
       " 'Halovirus HF1',\n",
       " 'Hamster polyomavirus',\n",
       " 'Helicoverpa armigera granulovirus',\n",
       " 'Helicoverpa armigera multiple nucleopolyhedrovirus',\n",
       " 'Helicoverpa armigera-NPV',\n",
       " 'Helicoverpa armigera NPV NNg1',\n",
       " 'Helicoverpa armigera nucleopolyhedrovirus G4',\n",
       " 'Helicoverpa zea SNPV',\n",
       " 'Heliothis virescens ascovirus 3e',\n",
       " 'Hepatitis B virus',\n",
       " 'Heron hepatitis B virus',\n",
       " 'His1 virus',\n",
       " 'His2 virus',\n",
       " 'Hollyhock leaf crumple virus',\n",
       " 'Honeysuckle yellow vein beta-JapanFukui2001',\n",
       " 'Honeysuckle yellow vein mosaic beta-JapanMiyizaki2001',\n",
       " 'Honeysuckle yellow vein mosaic disease associated satellite DNA beta-Ibaraki',\n",
       " 'Honeysuckle yellow vein mosaic Virus',\n",
       " 'Honeysuckle yellow vein mosaic virus-Kagoshima',\n",
       " 'Honeysuckle yellow vein mosaic virus satellite DNA beta',\n",
       " 'Honeysuckle yellow vein virus-UK1',\n",
       " 'Horsegram yellow mosaic Virus DNA B',\n",
       " 'Horsegram yellow mosaic virus',\n",
       " 'Horseradish curly top virus',\n",
       " 'Human adenovirus 54',\n",
       " 'Human adenovirus A',\n",
       " 'Human adenovirus B1',\n",
       " 'Human adenovirus B2',\n",
       " 'Human adenovirus C',\n",
       " 'Human adenovirus D',\n",
       " 'Human adenovirus E',\n",
       " 'Human adenovirus F',\n",
       " 'Human bocavirus 1',\n",
       " 'Human bocavirus 2',\n",
       " 'Human bocavirus 3',\n",
       " 'Human bocavirus 4',\n",
       " 'Human erythrovirus V9',\n",
       " 'Human herpesvirus 1',\n",
       " 'Human herpesvirus 2',\n",
       " 'Human herpesvirus 3',\n",
       " 'Human herpesvirus 4',\n",
       " 'Human herpesvirus 5 strain Merlin',\n",
       " 'Human herpesvirus 7',\n",
       " 'Human herpesvirus 8',\n",
       " 'Human papillomavirus-18',\n",
       " 'Human papillomavirus 1',\n",
       " 'Human papillomavirus-2',\n",
       " 'Human papillomavirus 54',\n",
       " 'Human papillomavirus-5',\n",
       " 'Human papillomavirus type 101',\n",
       " 'Human papillomavirus type 103',\n",
       " 'Human papillomavirus type 108',\n",
       " 'Human papillomavirus type-10',\n",
       " 'Human papillomavirus type 16',\n",
       " 'Human papillomavirus type 26',\n",
       " 'Human papillomavirus type 32',\n",
       " 'Human papillomavirus type 34',\n",
       " 'Human papillomavirus type 41',\n",
       " 'Human papillomavirus type 48',\n",
       " 'Human papillomavirus type 49',\n",
       " 'Human papillomavirus type-4',\n",
       " 'Human papillomavirus type 50',\n",
       " 'Human papillomavirus type 53',\n",
       " 'Human papillomavirus type 60',\n",
       " 'Human papillomavirus type 63',\n",
       " 'Human papillomavirus type 6b',\n",
       " 'Human papillomavirus type 7',\n",
       " 'Human papillomavirus type 88',\n",
       " 'Human papillomavirus type 90',\n",
       " 'Human papillomavirus type 92',\n",
       " 'Human papillomavirus type 96',\n",
       " 'Human papillomavirus type-9',\n",
       " 'Human parvovirus B19',\n",
       " 'Human T-lymphotropic virus 1',\n",
       " 'Human T-lymphotropic virus 4',\n",
       " 'Hyperthermophilic Archaeal Virus 1',\n",
       " 'Hyperthermophilic Archaeal Virus 2',\n",
       " 'Hyphantria cunea nucleopolyhedrovirus',\n",
       " 'Hyposoter fugitivus ichnovirus segment A1',\n",
       " 'Hyposoter fugitivus ichnovirus segment A2',\n",
       " 'Hyposoter fugitivus ichnovirus segment A3',\n",
       " 'Hyposoter fugitivus ichnovirus segment B10',\n",
       " 'Hyposoter fugitivus ichnovirus segment B11',\n",
       " 'Hyposoter fugitivus ichnovirus segment B12',\n",
       " 'Hyposoter fugitivus ichnovirus segment B13',\n",
       " 'Hyposoter fugitivus ichnovirus segment B14',\n",
       " 'Hyposoter fugitivus ichnovirus segment B15',\n",
       " 'Hyposoter fugitivus ichnovirus segment B16',\n",
       " 'Hyposoter fugitivus ichnovirus segment B17',\n",
       " 'Hyposoter fugitivus ichnovirus segment B18',\n",
       " 'Hyposoter fugitivus ichnovirus segment-B1',\n",
       " 'Hyposoter fugitivus ichnovirus segment B2',\n",
       " 'Hyposoter fugitivus ichnovirus segment B3',\n",
       " 'Hyposoter fugitivus ichnovirus segment B4',\n",
       " 'Hyposoter fugitivus ichnovirus segment B5',\n",
       " 'Hyposoter fugitivus ichnovirus segment B6',\n",
       " 'Hyposoter fugitivus ichnovirus segment B7',\n",
       " 'Hyposoter fugitivus ichnovirus segment B8',\n",
       " 'Hyposoter fugitivus ichnovirus segment B9',\n",
       " 'Hyposoter fugitivus ichnovirus segment C10',\n",
       " 'Hyposoter fugitivus ichnovirus segment C11',\n",
       " 'Hyposoter fugitivus ichnovirus segment C12',\n",
       " 'Hyposoter fugitivus ichnovirus segment C13',\n",
       " 'Hyposoter fugitivus ichnovirus segment C14',\n",
       " 'Hyposoter fugitivus ichnovirus segment C15',\n",
       " 'Hyposoter fugitivus ichnovirus segment C16',\n",
       " 'Hyposoter fugitivus ichnovirus segment C17',\n",
       " 'Hyposoter fugitivus ichnovirus segment C18',\n",
       " 'Hyposoter fugitivus ichnovirus segment C19',\n",
       " 'Hyposoter fugitivus ichnovirus segment-C1',\n",
       " 'Hyposoter fugitivus ichnovirus segment C20',\n",
       " 'Hyposoter fugitivus ichnovirus segment-C2',\n",
       " 'Hyposoter fugitivus ichnovirus segment C3',\n",
       " 'Hyposoter fugitivus ichnovirus segment C4',\n",
       " 'Hyposoter fugitivus ichnovirus segment C5',\n",
       " 'Hyposoter fugitivus ichnovirus segment C6',\n",
       " 'Hyposoter fugitivus ichnovirus segment C7',\n",
       " 'Hyposoter fugitivus ichnovirus segment C8',\n",
       " 'Hyposoter fugitivus ichnovirus segment C9',\n",
       " 'Hyposoter fugitivus ichnovirus segment D10',\n",
       " 'Hyposoter fugitivus ichnovirus segment D11',\n",
       " 'Hyposoter fugitivus ichnovirus segment D12',\n",
       " 'Hyposoter fugitivus ichnovirus segment-D1',\n",
       " 'Hyposoter fugitivus ichnovirus segment D2',\n",
       " 'Hyposoter fugitivus ichnovirus segment D3',\n",
       " 'Hyposoter fugitivus ichnovirus segment D4',\n",
       " 'Hyposoter fugitivus ichnovirus segment D5',\n",
       " 'Hyposoter fugitivus ichnovirus segment D6',\n",
       " 'Hyposoter fugitivus ichnovirus segment D7',\n",
       " 'Hyposoter fugitivus ichnovirus segment D8',\n",
       " 'Hyposoter fugitivus ichnovirus segment D9',\n",
       " 'Hyposoter fugitivus ichnovirus segment E1',\n",
       " 'Hyposoter fugitivus ichnovirus segment E2',\n",
       " 'Hyposoter fugitivus ichnovirus segment G1',\n",
       " 'Ictalurid herpesvirus 1 strain Auburn 1',\n",
       " 'Indian cassava mosaic virus DNA A',\n",
       " 'Indian cassava mosaic virus DNA B',\n",
       " 'Infectious hypodermal and hematopoietic necrosis virus',\n",
       " 'Infectious spleen and kidney necrosis virus',\n",
       " 'Invertebrate iridescent virus 6',\n",
       " 'Iodobacteriophage phiPLPE',\n",
       " 'Ipomoea yellow vein virus',\n",
       " 'Jatropha leaf curl virus DNA A',\n",
       " 'Jatropha yellow mosaic India virus DNA-A',\n",
       " 'JC polyomavirus',\n",
       " 'Junonia coenia densovirus',\n",
       " 'Kalanchoe top-spotting virus',\n",
       " 'Kenaf leaf curl virus DNA A',\n",
       " 'KI polyomavirus Stockholm 60',\n",
       " 'Klebsiella phage KP15',\n",
       " 'Klebsiella phage KP32',\n",
       " 'Klebsiella phage KP34',\n",
       " 'Klebsiella phage phiKO2',\n",
       " 'Kluyvera phage Kvp1',\n",
       " 'Kudzu mosaic virus DNA-A',\n",
       " 'Kudzu mosaic virus DNA-B',\n",
       " 'Lactobacillus johnsonii prophage Lj771',\n",
       " 'Lactobacillus phage A2',\n",
       " 'Lactobacillus phage KC5a',\n",
       " 'Lactobacillus phage Lb338-1',\n",
       " 'Lactobacillus phage Lc-Nu',\n",
       " 'Lactobacillus phage LL-H',\n",
       " 'Lactobacillus phage LP65',\n",
       " 'Lactobacillus phage Lrm1',\n",
       " 'Lactobacillus phage Lv-1',\n",
       " 'Lactobacillus phage phiAT3',\n",
       " 'Lactobacillus phage phig1e',\n",
       " 'Lactobacillus phage phiJL-1',\n",
       " 'Lactobacillus prophage Lj928',\n",
       " 'Lactobacillus prophage Lj965',\n",
       " 'Lactobacillus prophage phiadh',\n",
       " 'Lactococcus phage 1706',\n",
       " 'Lactococcus phage 4268',\n",
       " 'Lactococcus phage 712',\n",
       " 'Lactococcus phage asccphi28',\n",
       " 'Lactococcus phage bIBB29',\n",
       " 'Lactococcus phage bIL170',\n",
       " 'Lactococcus phage bIL67',\n",
       " 'Lactococcus phage BK5-T',\n",
       " 'Lactococcus phage c2',\n",
       " 'Lactococcus phage jj50',\n",
       " 'Lactococcus phage KSY1',\n",
       " 'Lactococcus phage P008',\n",
       " 'Lactococcus phage P087',\n",
       " 'Lactococcus phage phiLC3',\n",
       " 'Lactococcus phage Q54',\n",
       " 'Lactococcus phage r1t',\n",
       " 'Lactococcus phage sk1',\n",
       " 'Lactococcus phage TP901-1',\n",
       " 'Lactococcus phage Tuc2009',\n",
       " 'Lactococcus phage ul36',\n",
       " 'Lactococcus prophage bIL285',\n",
       " 'Lactococcus prophage bIL286',\n",
       " 'Lactococcus prophage bIL309',\n",
       " 'Lactococcus prophage bIL310',\n",
       " 'Lactococcus prophage bIL311',\n",
       " 'Lactococcus prophage bIL312',\n",
       " 'Lamium leaf distortion associated virus',\n",
       " 'Leucania separata nuclear polyhedrosis virus',\n",
       " 'Leucas zeylanica yellow vein virus satellite DNA beta',\n",
       " 'Lindernia anagallis yellow vein virus DNA-A',\n",
       " 'Lindernia anagallis yellow vein virus satellite DNA beta',\n",
       " 'Listeria phage A006',\n",
       " 'Listeria phage A118',\n",
       " 'Listeria phage A500',\n",
       " 'Listeria phage A511',\n",
       " 'Listeria phage B025',\n",
       " 'Listeria phage B054',\n",
       " 'Listeria phage P35',\n",
       " 'Listeria phage P40',\n",
       " 'Listonella phage phiHSIC',\n",
       " 'Loofa yellow mosaic virus DNA A',\n",
       " 'Loofa yellow mosaic virus DNA B',\n",
       " 'Lucky bamboo bacilliform virus',\n",
       " 'Ludwigia leaf distortion betasatellite IndiaAmadalavalasaHibiscus2 007',\n",
       " 'Ludwigia yellow vein virus-associated DNA beta',\n",
       " 'Ludwigia yellow vein virus DNA-A',\n",
       " 'Luffa begomovirus associated DNA beta',\n",
       " 'Luffa puckering and leaf distortion-associated DNA beta',\n",
       " 'LuIII virus',\n",
       " 'Lumpy skin disease virus NI-2490',\n",
       " 'Lymantria dispar MNPV',\n",
       " 'Lymantria xylina MNPV',\n",
       " 'Lymphocystis disease virus 1',\n",
       " 'Lymphocystis disease virus-isolate China',\n",
       " 'Macacine herpesvirus 1',\n",
       " 'Macacine herpesvirus 3',\n",
       " 'Macacine herpesvirus 4',\n",
       " 'Macacine herpesvirus 5',\n",
       " 'Macaque simian foamy virus',\n",
       " 'Macroptilium golden mosaic virus-',\n",
       " 'JamaicaWissadulaAugust Town DNA A',\n",
       " 'Macroptilium golden mosaic virus-',\n",
       " 'JamaicaWissadulaAugust Town DNA B',\n",
       " 'Macroptilium mosaic Puerto Rico virus DNA A',\n",
       " 'Macroptilium mosaic Puerto Rico virus DNA B',\n",
       " 'Macroptilium yellow mosaic Florida virus DNA A',\n",
       " 'Macroptilium yellow mosaic Florida virus DNA B',\n",
       " 'Macroptilium yellow mosaic virus DNA A',\n",
       " 'Macroptilium yellow mosaic virus DNA B',\n",
       " 'Maize streak virus-ASouth Africa',\n",
       " 'Malachra yellow vein mosaic virus-associated satellite DNA beta',\n",
       " 'Mal de Rio Cuarto virus segment 9',\n",
       " 'Malvastrum leaf curl Guangdong virus',\n",
       " 'Malvastrum leaf curl virus-associated defective DNA beta',\n",
       " 'Malvastrum leaf curl virus-G87',\n",
       " 'Malvastrum yellow mosaic virus-associated DNA 1',\n",
       " 'Malvastrum yellow mosaic virus DNA-A',\n",
       " 'Malvastrum yellow mosaic virus satellite DNA beta',\n",
       " 'Malvastrum yellow vein Baoshan virus DNA-A',\n",
       " 'Malvastrum yellow vein-virus',\n",
       " 'Malvastrum yellow vein virus satellite DNA beta',\n",
       " 'Malvastrum yellow vein Yunnan-virus',\n",
       " 'Malvastrum yellow vein Yunnan virus satellite DNA beta',\n",
       " 'Mamestra configurata NPV-A',\n",
       " 'Mamestra configurata NPV-B',\n",
       " 'Mannheimia phage phiMHaA1',\n",
       " 'Maruca vitrata MNPV',\n",
       " 'Mastomys coucha papillomavirus 2',\n",
       " 'Mastomys natalensis papillomavirus',\n",
       " 'Melanoplus sanguinipes entomopoxvirus',\n",
       " 'Meleagrid herpesvirus 1',\n",
       " 'Melon chlorotic leaf curl virus DNA A',\n",
       " 'Melon chlorotic mosaic virus-associated alphasatellite',\n",
       " 'Melon chlorotic mosaic virus DNA-A',\n",
       " 'Melon chlorotic mosaic virus DNA-B',\n",
       " 'Merkel cell polyomavirus',\n",
       " 'Merremia mosaic virus DNA A',\n",
       " 'Merremia mosaic virus DNA B',\n",
       " 'Mesta yellow vein mosaic Bahraich virus-',\n",
       " 'IndiaBahraich2007 DNA A',\n",
       " 'Mesta yellow vein mosaic virus-associated DNA beta',\n",
       " 'Mesta yellow vein mosaic virus DNA-A',\n",
       " 'Methanobacterium phage psiM2',\n",
       " 'Methanothermobacter prophage psiM100',\n",
       " 'Microbacterium phage Min1',\n",
       " 'Microcystis phage Ma-LMM01',\n",
       " 'Microplitis demolitor bracovirus segment A',\n",
       " 'Microplitis demolitor bracovirus segment B',\n",
       " 'Microplitis demolitor bracovirus segment C',\n",
       " 'Microplitis demolitor bracovirus segment D',\n",
       " 'Microplitis demolitor bracovirus segment E',\n",
       " 'Microplitis demolitor bracovirus segment F',\n",
       " 'Microplitis demolitor bracovirus segment G',\n",
       " 'Microplitis demolitor bracovirus segment H',\n",
       " 'Microplitis demolitor bracovirus segment I',\n",
       " 'Microplitis demolitor bracovirus segment J',\n",
       " 'Microplitis demolitor bracovirus segment K',\n",
       " 'Microplitis demolitor bracovirus segment L',\n",
       " 'Microplitis demolitor bracovirus segment M',\n",
       " 'Microplitis demolitor bracovirus segment N',\n",
       " 'Microplitis demolitor bracovirus segment O',\n",
       " 'Milk vetch dwarf virus segment 10',\n",
       " 'Milk vetch dwarf virus segment 11',\n",
       " 'Milk vetch dwarf virus segment-1',\n",
       " 'Milk vetch dwarf virus segment 2',\n",
       " 'Milk vetch dwarf virus segment 3',\n",
       " 'Milk vetch dwarf virus segment 4',\n",
       " 'Milk vetch dwarf virus segment 5',\n",
       " 'Milk vetch dwarf virus segment 6',\n",
       " 'Milk vetch dwarf virus segment 7',\n",
       " 'Milk vetch dwarf virus segment 8',\n",
       " 'Milk vetch dwarf virus segment 9',\n",
       " 'Mimosa yellow leaf curl virus-associated DNA 1',\n",
       " 'Mimosa yellow leaf curl virus DNA-A',\n",
       " 'Mimosa yellow leaf curl virus satellite DNA beta',\n",
       " 'Minute virus of mice',\n",
       " 'Mirabilis mosaic virus',\n",
       " 'Miscanthus streak virus-91',\n",
       " 'Molluscum contagiosum virus subtype 1',\n",
       " 'Monkeypox virus Zaire-96-I-16',\n",
       " 'Morganella phage MmP1',\n",
       " 'Mouse mammary tumor virus',\n",
       " 'Mouse parvovirus 1',\n",
       " 'Mouse parvovirus 2',\n",
       " 'Mouse parvovirus 3',\n",
       " 'Mouse parvovirus 4',\n",
       " 'Mouse parvovirus 5',\n",
       " 'Mulard duck circovirus',\n",
       " 'Mungbean yellow mosaic India virus DNA A',\n",
       " 'Mungbean yellow mosaic India virus DNA B',\n",
       " 'Mungbean yellow mosaic virus DNA A',\n",
       " 'Mungbean yellow mosaic virus DNA B',\n",
       " 'Murid herpesvirus 1',\n",
       " 'Murid herpesvirus 2',\n",
       " 'Murid herpesvirus 4',\n",
       " 'Murine adenovirus 3',\n",
       " 'Murine adenovirus A',\n",
       " 'Murine pneumotropic virus',\n",
       " ...]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_paper.paper['sections'][6]['content']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "abstract_text = ' '.join(test_paper.paper['sections'][0]['content'])\n",
    "abstract_doc = nlp(abstract_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "lemmas = [token.lemma_ for token in abstract_doc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "a_lemmas = [lemma.lower() for lemma in lemmas if lemma.isalpha() and lemma not in stopwords]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'generation sequencing increasingly use study sample compose mixture organism clinical application presence pathogen low abundance highly important present analytical method siann strain identification alignment near neighbors specifically design rapidly detect set target organism mixed sample achieve high degree strain specificity align short sequence read genome near neighbor organism target empirical benchmarke alongside current state art method extremely high positive predictive value low abundance target organism mixed sample siann available illumina basespace app signature science llc siann result present streamlined report design comprehensible non specialist user provide powerful tool rapid specie detection mixed sample focus set customizable target organism near neighbor siann operate quickly low computational requirement deliver highly accurate result'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "' '.join(a_lemmas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('organism', 5), ('sample', 4), ('siann', 4), ('target', 4), ('low', 3)]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "word_count = Counter(a_lemmas)\n",
    "word_count.most_common(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('Positive Predictive Value', 'ORG'), ('Illumina BaseSpace', 'ORG'), ('Signature Science', 'WORK_OF_ART')]\n"
     ]
    }
   ],
   "source": [
    "# Named Entity Recognition test\n",
    "ne = [(ent.text, ent.label_) for ent in abstract_doc.ents]\n",
    "print(ne)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Next-generation sequencing is increasingly being used to study samples composed of mixtures of organisms, such as in clinical applications where the presence of a pathogen at very low abundance may be highly important. We present an analytical method (SIANN: Strain Identification by Alignment to Near Neighbors) specifically designed to rapidly detect a set of target organisms in mixed samples that achieves a high degree of species- and strain-specificity by aligning short sequence reads to the genomes of near neighbor organisms, as well as that of the target. Empirical benchmarking alongside the current state-of-the-art methods shows an extremely high Positive Predictive Value, even at very low abundances of the target organism in a mixed sample. SIANN is available as an Illumina BaseSpace app, as well as through Signature Science, LLC. SIANN results are presented in a streamlined report designed to be comprehensible to the non-specialist user, providing a powerful tool for rapid species detection in a mixed sample. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.\n"
     ]
    }
   ],
   "source": [
    "text = ' '.join(test_paper.paper['sections'][0]['content'])\n",
    "print(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_section_text(section):\n",
    "    ''' Returns a string containing only paragraph text of sections '''\n",
    "    text = ''\n",
    "    for content in section['content']:\n",
    "        if type(content) == str:\n",
    "            text += ' ' + content\n",
    "        elif type(content) == dict and 'title' in content.keys():\n",
    "            text += ' ' + get_section_text(content)\n",
    "    return text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abstract\n",
      "[('organism', 5), ('sample', 4), ('siann', 4), ('target', 4), ('low', 3)]\n",
      "NER:\n",
      "[('Positive Predictive Value', 'ORG'), ('Illumina BaseSpace', 'ORG'), ('Signature Science', 'WORK_OF_ART')]\n",
      "------------------------------------------------------------\n",
      "Introduction\n",
      "[('organism', 7), ('sample', 5), ('genome', 5), ('sequence', 5), ('present', 4)]\n",
      "NER:\n",
      "[('one', 'CARDINAL'), ('one', 'CARDINAL'), ('one', 'CARDINAL'), ('millions', 'CARDINAL'), ('2013', 'DATE'), ('hundreds', 'CARDINAL'), ('ref-15,\"Wu', 'PRODUCT'), ('al.', 'GPE'), ('2011', 'DATE'), ('2013', 'DATE'), ('second', 'ORDINAL')]\n",
      "------------------------------------------------------------\n",
      "Methods\n",
      "[('database', 26), ('specie', 24), ('organism', 22), ('read', 17), ('method', 16)]\n",
      "NER:\n",
      "[('2012', 'DATE'), ('SIANN', 'PRODUCT'), ('SIANN', 'PRODUCT'), ('& Salzberg', 'GPE'), ('2012', 'DATE'), ('algorithm', 'ORG'), ('ref-12,\"Segata', 'LAW'), ('al.', 'GPE'), ('2012', 'DATE'), ('two', 'CARDINAL'), ('only two', 'CARDINAL'), ('one', 'CARDINAL'), ('one', 'CARDINAL'), ('Bacilli', 'GPE'), ('two', 'CARDINAL'), ('One', 'CARDINAL'), ('two', 'CARDINAL'), ('two', 'CARDINAL'), ('two', 'CARDINAL'), ('only E. coli', 'QUANTITY'), ('B.anthracis', 'PERSON'), ('B. anthracis', 'PERSON'), ('B. cereus', 'PERSON'), ('SIANN', 'ORG'), ('SIANN', 'PRODUCT'), ('1.6', 'CARDINAL'), ('LMAT', 'PERSON'), ('1.2', 'CARDINAL'), ('1.7.7', 'CARDINAL'), ('Kraken', 'PERSON'), ('T1,\"Table 1', 'PRODUCT'), ('three', 'CARDINAL'), ('600', 'CARDINAL'), ('MetaSim', 'PERSON'), ('2008', 'DATE'), ('15,000,000 reads', 'MONEY'), ('100bp', 'CARDINAL'), ('Illumina', 'GPE'), ('fourth', 'ORDINAL'), ('al.', 'GPE'), ('2009', 'DATE'), ('600', 'CARDINAL'), ('12', 'CARDINAL'), ('50', 'CARDINAL'), ('12', 'CARDINAL'), ('2', 'CARDINAL'), ('12', 'CARDINAL'), ('Target Pathogen Database', 'WORK_OF_ART'), ('Kraken', 'GPE'), ('LMAT', 'PERSON'), ('SIANN', 'PRODUCT'), ('600', 'CARDINAL'), ('513', 'CARDINAL'), ('LMAT', 'ORG'), ('1.6%', 'PERCENT'), ('MetaPhlAn', 'GPE'), ('600', 'CARDINAL'), ('PPV', 'ORG'), ('PPV', 'ORG'), ('1', 'CARDINAL'), ('2', 'CARDINAL'), ('1', 'CARDINAL'), ('PPV', 'ORG'), ('600', 'CARDINAL'), ('BASH', 'ORG')]\n",
      "------------------------------------------------------------\n",
      "Results\n",
      "[('read', 12), ('organism', 9), ('method', 7), ('ppv', 6), ('abundance', 6)]\n",
      "NER:\n",
      "[('PPV', 'ORG'), ('four', 'CARDINAL'), ('PPV', 'ORG'), ('1', 'CARDINAL'), ('95%', 'PERCENT'), ('Kraken', 'GPE'), ('LMAT', 'ORG'), ('∼38%', 'PERCENT'), ('0.21', 'CARDINAL'), ('SIANN', 'ORG'), ('SIANN', 'ORG'), ('38%', 'PERCENT'), ('LMAT', 'ORG'), ('Kraken', 'GPE'), ('LMAT', 'PERSON'), ('Kraken', 'ORG'), ('100%', 'PERCENT'), ('300bp', 'CARDINAL'), ('only 90%', 'PERCENT'), ('300bp', 'ORG'), ('PPV', 'ORG'), ('thousands', 'CARDINAL'), ('PPV', 'ORG'), ('50', 'CARDINAL'), ('two', 'CARDINAL'), ('three', 'CARDINAL'), ('two', 'CARDINAL'), ('MetaPhlAn', 'GPE'), ('Kraken', 'ORG'), ('Hanta', 'GPE'), ('D. Wood', 'PERSON'), ('75%', 'PERCENT'), ('18%', 'PERCENT'), ('3', 'CARDINAL'), ('as low as 0.3% and 0.07%', 'PERCENT')]\n",
      "------------------------------------------------------------\n",
      "Discussion\n",
      "[('siann', 7), ('organism', 6), ('set', 6), ('genome', 6), ('result', 5)]\n",
      "NER:\n",
      "[('95%', 'PERCENT'), ('0.2', 'CARDINAL'), ('as low as 0.3%', 'PERCENT'), ('Illumina', 'GPE'), ('BaseSpace', 'ORG'), ('1', 'CARDINAL'), ('NCBI', 'ORG'), ('2', 'CARDINAL'), ('NCBI', 'ORG'), ('app-3,\"Appendix 3', 'WORK_OF_ART'), ('BaseSpace', 'ORG'), ('OS', 'ORG'), ('95%', 'PERCENT'), ('Signature Science', 'ORG'), ('LLC', 'ORG'), ('BaseSpace', 'ORG'), ('SIANN', 'PRODUCT')]\n",
      "------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "for section in test_paper.paper['sections']:\n",
    "    if 'Appendix' not in section['title']:\n",
    "        print(section['title'])\n",
    "        # process section content\n",
    "        text = get_section_text(section)\n",
    "        doc  = nlp(text)\n",
    "        lemmas = [token.lemma_ for token in doc]\n",
    "        a_lemmas = [lemma.lower() for lemma in lemmas if lemma.isalpha() and lemma not in stopwords]\n",
    "\n",
    "        # get top 5 most frequent words \n",
    "        word_count = Counter(a_lemmas)\n",
    "        print(word_count.most_common(5))\n",
    "\n",
    "        # test NER on content \n",
    "        ne = [(ent.text, ent.label_) for ent in doc.ents]\n",
    "        print('NER:')\n",
    "        print(ne)\n",
    "\n",
    "        print('-'*60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "# build keyterms database from key terms in ARTICLES\n",
    "key_terms = { 'SIANN':{'type':'method',\n",
    "                       'def':'Strain Identification by Alignment to Near Neighbors',\n",
    "                       'from':test_paper.paper['id']}\n",
    "            }\n",
    "\n",
    "for section in test_paper.paper['sections']:\n",
    "    if 'Appendix 1' in section['title']:\n",
    "        print(section['title'])\n",
    "        for content in section['content']:\n",
    "            key_terms[content] = {'type':'target pathogen','from':test_paper.paper['id']}\n",
    "    elif 'Appendix 2' in section['title']:\n",
    "        print(section['title'])\n",
    "        for content in section['content']:\n",
    "            key_terms[content] = {'type':'viral','from':test_paper.paper['id']}\n",
    "    elif 'Appendix 3' in section['title']:\n",
    "        print(section['title'])\n",
    "        for content in section['content']:\n",
    "            key_terms[content] = {'type':'bacterial','from':test_paper.paper['id']}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "key_terms"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### HTML Display\n",
    "\n",
    "Objective: display text with special formating."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.core.display import display, HTML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def text_to_html(text, words=[]):\n",
    "    ''' Replaces words in text with html tags that highlight them '''\n",
    "    html = '<p>{}</p>'.format(text)\n",
    "    if len(words) > 0:\n",
    "        for word in words:\n",
    "            html_word = '<strong>{}</strong>'.format(word)\n",
    "            html = html.replace(word, html_word)\n",
    "    return html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<p>Next-<strong>generation</strong> <strong>sequencing</strong> is <strong>increasingly</strong> being <strong>use</strong>d to <strong>study</strong> <strong><strong><strong><strong>sample</strong></strong></strong></strong>s <strong>compose</strong>d of <strong>mixture</strong>s of <strong><strong><strong><strong><strong>organism</strong></strong></strong></strong></strong>s, such as in <strong>clinical</strong> <strong><strong>app</strong>lication</strong>s where the <strong>presence</strong> of a <strong>pathogen</strong> at very <strong><strong><strong>low</strong></strong></strong> <strong><strong>abundance</strong></strong> may be <strong><strong><strong>high</strong></strong>ly</strong> <strong>important</strong>. We <strong><strong>present</strong></strong> an <strong>analytical</strong> <strong><strong>method</strong></strong> (<strong><strong><strong><strong>SIANN</strong></strong></strong></strong>: Strain Identification by <strong>Alignment</strong> to <strong>Near</strong> <strong>Neighbors</strong>) <strong>specifically</strong> <strong><strong>design</strong></strong>ed to <strong><strong>rapid</strong>ly</strong> <strong>detect</strong> a <strong><strong>set</strong></strong> of <strong><strong><strong><strong>target</strong></strong></strong></strong> <strong><strong><strong><strong><strong>organism</strong></strong></strong></strong></strong>s in <strong><strong><strong>mixed</strong></strong></strong> <strong><strong><strong><strong>sample</strong></strong></strong></strong>s that <strong>achieve</strong>s a <strong><strong>high</strong></strong> <strong>degree</strong> of <strong>specie</strong>s- and <strong><strong>strain</strong></strong>-<strong>specificity</strong> by <strong>align</strong>ing <strong>short</strong> <strong>sequence</strong> <strong>read</strong>s to the <strong>genome</strong>s of <strong><strong>near</strong></strong> <strong><strong>neighbor</strong></strong> <strong><strong><strong><strong><strong>organism</strong></strong></strong></strong></strong>s, as well as that of the <strong><strong><strong><strong>target</strong></strong></strong></strong>. Empirical benchmarking <strong>alongside</strong> the <strong>current</strong> <strong>state</strong>-of-the-<strong>art</strong> <strong><strong>method</strong></strong>s shows an <strong>extremely</strong> <strong><strong>high</strong></strong> <strong>Positive</strong> <strong>Predictive</strong> <strong>Value</strong>, even at very <strong><strong><strong>low</strong></strong></strong> <strong><strong>abundance</strong></strong>s of the <strong><strong><strong><strong>target</strong></strong></strong></strong> <strong><strong><strong><strong><strong>organism</strong></strong></strong></strong></strong> in a <strong><strong><strong>mixed</strong></strong></strong> <strong><strong><strong><strong>sample</strong></strong></strong></strong>. <strong><strong><strong><strong>SIANN</strong></strong></strong></strong> is <strong>available</strong> as an <strong>Illumina</strong> <strong>BaseSpace</strong> <strong>app</strong>, as well as through <strong>Signature</strong> <strong>Science</strong>, <strong>LLC</strong>. <strong><strong><strong><strong>SIANN</strong></strong></strong></strong> <strong><strong>result</strong></strong>s are <strong><strong>present</strong></strong>ed in a <strong>streamlined</strong> <strong>report</strong> <strong><strong>design</strong></strong>ed to be <strong>comprehensible</strong> to the <strong>non</strong>-<strong>specialist</strong> <strong>use</strong>r, providing a <strong>powerful</strong> <strong>tool</strong> for <strong>rapid</strong> <strong>specie</strong>s <strong>detect</strong>ion in a <strong><strong><strong>mixed</strong></strong></strong> <strong><strong><strong><strong>sample</strong></strong></strong></strong>. By <strong>focus</strong>ing on a <strong><strong>set</strong></strong> of (<strong>customizable</strong>) <strong><strong><strong><strong>target</strong></strong></strong></strong> <strong><strong><strong><strong><strong>organism</strong></strong></strong></strong></strong>s and their <strong><strong>near</strong></strong> <strong><strong>neighbor</strong></strong>s, <strong><strong><strong><strong>SIANN</strong></strong></strong></strong> can <strong>operate</strong> <strong>quickly</strong> and with <strong><strong><strong>low</strong></strong></strong> <strong>computational</strong> <strong>requirement</strong>s while <strong>deliver</strong>ing <strong><strong><strong>high</strong></strong>ly</strong> <strong>accurate</strong> <strong><strong>result</strong></strong>s.</p>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "html_text = text_to_html(abstract_text, a_lemmas)\n",
    "display(HTML(html_text))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Visualizing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "code_folding": [
     0,
     12,
     32,
     57
    ]
   },
   "outputs": [],
   "source": [
    "def authors_to_html(authors):\n",
    "    ''' Converts list of authors to html '''\n",
    "    html = ''\n",
    "    html += ', '.join(['<span class=\"author\">{}</span>'.format(author) for author in authors])\n",
    "    return html\n",
    "\n",
    "def graph_to_html(graph):\n",
    "    ''' Converts graphic element to html '''\n",
    "    html = '<table id={} class=\"figure\"><tr><td>'.format(graph['id'])\n",
    "    html += '<tr><td><img src={} alt={}></tr></td>'.format(graph['img'],graph['label'])\n",
    "    html += '<tr><td><b>{}</b></tr></td>'.format(graph['label'])\n",
    "    html += '<tr><td><span class=\"description\">{}</span></td></tr>'.format(graph['description'])\n",
    "    html += '</table>'\n",
    "    return html\n",
    "\n",
    "def section_to_html(section, paper):\n",
    "    ''' Converts a section to html '''\n",
    "    html = '<table>'\n",
    "    html += '<tr><td>{}</td></tr>'.format(section['title'])\n",
    "    for content in section['content']:\n",
    "        html += '<tr>'\n",
    "        if type(content) == str:\n",
    "            html += '<td><p>{}</p></td>'.format(content)\n",
    "        elif type(content) == dict:\n",
    "            if 'title' in content.keys():\n",
    "                # subsection\n",
    "                html += '<td>{}</td>'.format(section_to_html(content, paper))\n",
    "            elif 'id' in content.keys():\n",
    "                # graphic element\n",
    "                graph_html = graph_to_html(paper[content['type'] + 's'][content['id']])\n",
    "                html += '<td>{}</td>'.format(graph_html)\n",
    "        html += '</tr>'\n",
    "    html += '</table>'\n",
    "    return html\n",
    "\n",
    "def ref_to_html(ref):\n",
    "    ''' Converts a reference to html '''\n",
    "    html = '<span class=\"reference\">'\n",
    "    if ref['label'] != '':\n",
    "        html += ref['label'] + ' - '\n",
    "    html += ref['title'] + ', '\n",
    "    html += authors_to_html(ref['authors']) + ', '\n",
    "    if ref['date']:\n",
    "        html += ref['date'] + ', '\n",
    "    if ref['links']:\n",
    "        html += ', '.join(['<a href=\"{}\">{}</a>'.format(link[1],link[0]) for link in ref['links']])\n",
    "    html += '</span>'\n",
    "    return html\n",
    "\n",
    "def refs_to_html(refs):\n",
    "    ''' Converts all references to html '''\n",
    "    refs_html = '<table id=\"references\" class=\"section\">'\n",
    "    refs_html += '<tr><td><h2>References</h2></td></tr>'\n",
    "    for ref in refs:\n",
    "        refs_html += '<tr><td>'\n",
    "        refs_html += ref_to_html(ref)\n",
    "        refs_html += '</td></tr>'\n",
    "    refs_html += '</table>'\n",
    "    return refs_html\n",
    "\n",
    "def paper_to_html(paper):\n",
    "    ''' Converts a json paper to HTML format '''\n",
    "    html = '<table>'\n",
    "    html += '<tr><td style=\"text-align:left;\">{}</td></tr>'.format(paper['title'])\n",
    "    html += '<tr><td>{}</td></tr>'.format(authors_to_html(paper['authors']))\n",
    "    for section in paper['sections']:\n",
    "        if 'Appendix' not in section['title']:\n",
    "            html += '<tr><td>{}</td></tr>'.format(section_to_html(section, paper))\n",
    "    html += '<tr><td>{}</td></tr>'.format(refs_to_html(paper['references']))\n",
    "    html += '</table>'\n",
    "    return html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "css= '''<style type=\"text/css\">\n",
    "    .article {\n",
    "      width: 100%;\n",
    "    }\n",
    "\n",
    "    .section {\n",
    "      border-top: 3px solid black;\n",
    "      width: 100%;\n",
    "    }\n",
    "\n",
    "    #title {\n",
    "      text-align: center;\n",
    "    }\n",
    "\n",
    "    .section td {\n",
    "      text-align: left;\n",
    "    }\n",
    "\n",
    "    .subtitle {\n",
    "      font-weight: bold;\n",
    "    }\n",
    "\n",
    "    img {\n",
    "      width: 50%;\n",
    "      height: 50%;\n",
    "    }\n",
    "\n",
    "    .figure {\n",
    "      border: 1px solid black;\n",
    "      vertical-align: center;\n",
    "    }\n",
    "  </style>'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "    .article {\n",
       "      width: 100%;\n",
       "    }\n",
       "\n",
       "    .section {\n",
       "      border-top: 3px solid black;\n",
       "      width: 100%;\n",
       "    }\n",
       "\n",
       "    #title {\n",
       "      text-align: center;\n",
       "    }\n",
       "\n",
       "    .section td {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    .subtitle {\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    img {\n",
       "      width: 50%;\n",
       "      height: 50%;\n",
       "    }\n",
       "\n",
       "    .figure {\n",
       "      border: 1px solid black;\n",
       "      vertical-align: center;\n",
       "    }\n",
       "  </style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table><tr><td style=\"text-align:left;\">SIANN: Strain Identification by Alignment to Near Neighbors</td></tr><tr><td><span class=\"author\">Samuel S. Minot</span>, <span class=\"author\">Stephen D. Turner</span>, <span class=\"author\">Krista L. Ternus</span>, <span class=\"author\">Dana R. Kadavy</span></td></tr><tr><td><table><tr><td>Abstract</td></tr><tr><td><p>Next-generation sequencing is increasingly being used to study samples composed of mixtures of organisms, such as in clinical applications where the presence of a pathogen at very low abundance may be highly important. We present an analytical method (SIANN: Strain Identification by Alignment to Near Neighbors) specifically designed to rapidly detect a set of target organisms in mixed samples that achieves a high degree of species- and strain-specificity by aligning short sequence reads to the genomes of near neighbor organisms, as well as that of the target. Empirical benchmarking alongside the current state-of-the-art methods shows an extremely high Positive Predictive Value, even at very low abundances of the target organism in a mixed sample. SIANN is available as an Illumina BaseSpace app, as well as through Signature Science, LLC. SIANN results are presented in a streamlined report designed to be comprehensible to the non-specialist user, providing a powerful tool for rapid species detection in a mixed sample. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.</p></td></tr></table></td></tr><tr><td><table><tr><td>Introduction</td></tr><tr><td><p>There are many different methods that characterize the mixture of organisms present within a metagenomic dataset. Such datasets are generated when a complex environmental sample is processed by a “next-generation” high-throughput genome sequencing protocol, and they consist of large numbers of short nucleotide sequences. Each sequence represents a small fragment of a randomly selected genome from the very large collection of genomes present in the source sample. Those sequences indicate the presence of one organism or another according to their similarity to a set of known reference genomes. While a given sequence may be unique to one species, it also may be found in diverse organisms across the tree of life. Therefore, one analytical challenge (among many) is to take that collection of sequences (likely numbering in the millions) and accurately determine what species are present in the sample. Here we describe a novel method (SIANN: Strain Identification by Alignment to Near Neighbors) that is specifically designed to rapidly detect a set of targeted organisms from a metagenomic dataset by aligning reads to genomic regions that are unique at the strain or species level.</p></td></tr><tr><td><p>The analytical question motivating a particular piece of metagenomic bioinformatic analysis may vary widely by user and sample type ([ref-11,\"Segata, et al., 2013\"]). For example, the function of the human gut microbiome may depend on the relative abundance of hundreds of species of bacteria and the types of metabolic genes they contain ([ref-15,\"Wu, et al., 2011\"]; [ref-10,\"Schloissnig, et al., 2013\"]). In contrast, the clinical treatment of a patient may depend on whether or not a particular virus, or a consortium of co-infecting pathogens, is/are detected in their blood. It is this second class of presence/absence questions that SIANN is designed to address. SIANN is appropriate for situations in which a user wants to know whether a particular organism or set of organisms is present in a sample, but isn’t interested in the functions encoded in their genomes, the relative abundance of each organism, or any other more in-depth analysis.</p></td></tr></table></td></tr><tr><td><table><tr><td>Methods</td></tr><tr><td><table><tr><td>Approach</td></tr><tr><td><p>Metagenomic classification methods are based on a wide variety of theoretical underpinnings. The basic varieties include alignment of reads to various nucleotide databases or exact matching to nucleotide or protein signature sequences (or kmers). A representative set of recent methods are described in [T1,\"Table 1\"] (also see [ref-2,\"Bazinet & Cummings 2012\"]).</p></td></tr><tr><td><table id=T1 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T1.large.jpg alt=Table 1.></tr></td><tr><td><h3>Table 1.</h3></tr></td><tr><td><span class=\"description\">Summary of methods for metagenomic classification.\n",
       "</span></td></tr></table></td></tr><tr><td><p>Overall, these methods are designed to either classify individual reads to, and/or predict the total abundance of, clades (e.g. genus or species) across the entire tree of life. They generally require reference databases that are very large and/or require a large amount of processing to generate. The gap SIANN is designed to fill is when the entire tree of life is irrelevant, and only predefined subsets of organisms need to be detected. For an underlying method we chose read alignment to diagnostic genomic regions because the algorithms for read alignment are highly parallelizable and have been optimized heavily by the community at large (the current implementation of SIANN uses bowtie2 [[ref-7,\"Langmead & Salzberg, 2012\"]] for the alignment function, but can be adapted to any alignment algorithm). This approach is distinct from using clade-specific marker genes ([ref-12,\"Segata, et al., 2012\"]) because unique regions that are larger, smaller, or outside of genes can also be used. Furthermore, this approach supports the rapid construction of custom databases using reference genome sets that require only minimal user-supplied structure.</p></td></tr><tr><td><p>To understand the principle at work, consider a set of reads that have been aligned to the genomes of several strains belonging to two species. Some regions of those genomes are species-specific, some are strain-specific, and some are shared ([F1,\"Figure 1a\"]). When a set of reads is aligned to those genomes such that each read is placed in as many locations as it has a match (at a reasonably stringent threshold), visual inspection of the distribution of reads yields an intuitive understanding of the true source organism as Species I/Strain B ([F1,\"Figure 1b\"]). If Strain B were not present in the reference database, it would still be clear that the organism was an unknown strain of Species I.</p></td></tr><tr><td><table id=F1 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F1.large.jpg alt=Figure 1.></tr></td><tr><td><h3>Figure 1.</h3></tr></td><tr><td><span class=\"description\">A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.\n",
       "</span></td></tr></table></td></tr><tr><td><p>The unique identification of a species or strain is quantified by the proportion of the genome that is determined to be species- or strain-specific (defined as reads that are aligned to regions that are species- or strain-specific). Each species and strain is then assigned a numerical measure of the proportion that is covered by these diagnostic reads, and that proportional measure is compared to the ideal case, where sequences from a single organism (generated in silico) are aligned against the database in an identical manner. After that normalization factor is applied, the resulting score indicates whether the source sample contained any of the organisms in the reference database. The analysis is conducted independently on both the species and the strain level, so that if the true strain is not present in the database, the species of origin will still be identified. While many methods consider the complete taxonomic tree and assign reads to the least common ancestor, SIANN considers only two taxonomic levels: species and strain, throwing out anything that is not unique at one of those levels and thus obviating many of the confounding factors introduced by manually curated taxonomies.</p></td></tr><tr><td><p>The example shown in [F1,\"Figure 1b\"] indicates that species-specific reads are identified as reads that align to one species (Species I, in that case) but not the other. If Species II were not present in the example shown in [F1,\"Figure 1b\"], a much larger number of reads would be assigned as “species-specific,” when in fact those regions are shared with other species. Therefore, the ability of this method to identify strain- and species-specific sequences is a direct function of the inclusion of near neighbors in the reference database. This characteristic is shared among many classification algorithms, but it is of particular note for this method when users have an opportunity to construct their own database. In order to detect a target species with a high degree of specificity (reducing false positives), it is necessary to include other related species in the reference database. Only by parallel alignment to those near neighbors can the redundant sequences be separated from the species-specific ones. For example, in order to detect Bacillus anthracis in a sample, it would be necessary to include other species of Bacilli in the reference database so that the presence of B. cereus or B. thuringiensis in a sample does not lead to a false call for B. anthracis.</p></td></tr><tr><td><p>The nomenclature of genus, species, and strain is potentially problematic because it does not correspond to a consistent degree of evolutionary distance or genomic distinctiveness. The ability to distinguish two organisms by any method using genomic sequence data is proportional to the amount of each genome that is shared or unique. One might assume that any two organisms of the same species will have a relatively predictable amount of shared genomic identity. However, some pairs of organisms from the same species may have less in common than other pairs of organisms from different species or even genera. This ambiguity impacts SIANN in two ways. If two organisms have very little genomic sequence to distinguish them, the sensitivity of SIANN to detect either one will diminish (the rate of false negatives will increase as the likelihood of sequencing unique regions decreases). Conversely, if an organism is extremely dissimilar to the near neighbors selected for the database, the specificity with which SIANN detects that organism will decline (the rate of false positives will increase as the number of related genomes available in the database decreases). For example, if a database contained only E. coli and B.anthracis, a sample containing B. cereus would be misidentified as contraining B. anthracis. In the intended use case, a database targeting B. anthracis would contain B. cereus and a number of other near neighbors to prevent that kind of misidentification. It would be convenient to say that an ideal database can be made by calculating the ideal genetic distance between all references and then finding an ideal set of organisms to make up that database, but the behavior of any database will be governed by the particular genomes of the organisms it encounters in the wild. Because not all organisms evolve in the same manner (differences in mutation rate, horizontal gene transfer, recombination, etc), the suitability of a database and method to detect a given organism can only be determined by thorough validation and benchmarking, as well as updating the reference database as needed. Users of SIANN may construct their own custom databases to include newly identified genomes or specific subsets of genomes that best suit their research interests.</p></td></tr><tr><td><p>Steps to construct a custom database:\n",
       "</p></td></tr><tr><td><p>The files contained within each SIANN database are a compressed genomic index and a list containing the proportion of each reference genome that was found to be strain- or species-specific during database construction.</p></td></tr><tr><td><p>To run SIANN:\n",
       "</p></td></tr></table></td></tr><tr><td><table><tr><td>Benchmarking</td></tr><tr><td><p>The performance of SIANN (version 1.6) was tested in comparison to the following state-of-the-art metagenomic classification programs: LMAT (version 1.2), MetaPhlAn (version 1.7.7), and Kraken (version 0.9.1b). All of the programs in [T1,\"Table 1\"] were investigated for this effort, and three were chosen based on their ability to run on our high-performance computing cluster with an execution time and memory requirement that would be suitable to a clinical lab. Each program was run on a set of 600 simulated datasets generated by MetaSim ([ref-9,\"Richter, et al., 2008\"]). Each dataset consisted of 15,000,000 reads (100bp single-ended) with Illumina-simulated error (fourth-degree polynomial) ([ref-6,\"Korbel, et al., 2009\"]). The 600 datasets were broken into 12 sets of 50 replicates. Each of the 12 sets contained organisms at different levels of abundance as shown in [T2,\"Table 2\"]. Organisms were specifically chosen in pairs so that the ability to distinguish these near neighbors could be determined. The abundances were staggered at 4-fold intervals so that a wide range could be evaluated. All known species of near neighbors for each of the 12 target organisms were included in the reference database used by SIANN for this benchmarking (“Target Pathogen Database”) and are shown in [app-1,\"Appendix 1\"].</p></td></tr><tr><td><table id=T2 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T2.large.jpg alt=Table 2.></tr></td><tr><td><h3>Table 2.</h3></tr></td><tr><td><span class=\"description\"></span></td></tr></table></td></tr><tr><td><p>Each program outputs a distinct measure. Kraken and LMAT both count the reads assigned to each taxon, MetaPhlAn calculates the abundance, and SIANN outputs a measure of the proportion of diagnostic genomic regions present. To put these measures on an even footing, we empirically calculated the false positive rate for each method over all 600 samples, at each possible measure of output. Because each dataset is made up of known organisms, any result can be classified as true or false. Therefore, for any possible result (say, 513 reads classified by LMAT or 1.6% abundance assigned by MetaPhlAn), one can calculate the proportion of calls with at least the same amount of support that were correct (True Positives/[True Positives + False Positives]), over all of the 600 datasets. That measure is commonly given as Positive Predictive Value (PPV). For each program, the results can be translated from the raw value into a PPV that is based on this empirical measure of error. The key item of interest is the PPV value for the results that we know to be true positives, the defined spike organisms. Another way of describing this approach is to say that the results of each program have been normalized to the false positive error rate that was empirically observed. If another set of samples were generated, the PPV vs. raw value curve ([F2,\"Figure 2\"]) would likely fall differently, but in this case it gives us a means of comparing a diverse set of methods against the same ground truth. If method 1 detects an organism with a higher PPV than method 2 does, it means that method 1 has fewer false positives in the range that it reports true positives, which is the definition of utility in this setting.</p></td></tr><tr><td><table id=F2 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F2.large.jpg alt=Figure 2.></tr></td><tr><td><h3>Figure 2.</h3></tr></td><tr><td><span class=\"description\">Relationship of reported value for each program (horizontal axis, log scale) to the empirically-determined Positive Predictive Value (PPV), shown on the vertical axis. While the exact values depend on the test data used, the general values at significant cutoff values (0.8, 0.9, 0.95 PPV) remain relatively constant across different datasets (data not shown).\n",
       "</span></td></tr></table></td></tr><tr><td><p>For each method, PPV was calculated as a function of raw output value. Briefly, this was done by compiling the output for all 600 samples, labeling each result as false or true based on the sample set that it came from, and then calculating (at each possible value of output) what the proportion of TP/[TP + FP] was for results with at least that level of raw output. Some simplification steps were taken, such as focusing on the species-level assignments (for comparison with methods that do not perform strain assignment), and only taking the top hit for each species from each dataset. Custom R and BASH scripts were used for the data compilation and analysis.</p></td></tr></table></td></tr></table></td></tr><tr><td><table><tr><td>Results</td></tr><tr><td><p>The relationship of raw output value to PPV is shown for each of the four methods in [F2,\"Figure 2\"]. The point at which PPV is very close to 1 (where 95% of results are true positives) is ∼41,000 reads for Kraken, ∼2,800 reads for LMAT, ∼38% abundance for MetaPhlAn, and 0.21 for SIANN. For SIANN this means that having 38% of the species-unique genome covered by reads resulted in the vast majority of calls being accurate.</p></td></tr><tr><td><p>For read-assignment methods (such as LMAT and Kraken), manual inspection of the results may yield a different understanding of confidence than is presented here, or in any automated analysis. For example, while each read that is assigned by LMAT and Kraken fall above a certain cutoff for species-specificity, some individual reads may be much more specific than others. One could identify a read that aligns to a single species of bacteria with 100% accuracy over its 300bp length, with the next closest match being only 90% similar. It is extremely unlikely that a 300bp exact match would arise due to random chance, and so the user could say with confidence that the organism of interest is found within the sequence data (not considering contamination, horizontal gene transfer, etc). However, such an approach is not currently implemented in an automated method, and many of the steps needed to make that assertion are performed manually by a domain expert, including alignment to near neighbors and ensuring that the read does not fall within a transposon, plasmid, etc. Therefore, while one could say that a single read is all that is needed to state with high PPV that an organism is present, the amount of reads assigned in an automated manner needed to achieve that level of PPV will number in the thousands ([F2,\"Fig 2\"]).</p></td></tr><tr><td><p>The next phase of benchmarking was to determine how many raw input reads were needed to achieve the threshold for high PPV. To demonstrate this we plotted the known abundance of each spike organism against the PPV value generated by each method ([F3,\"Figure 3\"]). Each point (an organism at a known level of abundance) is comprised of a maximum of 50 replicates, where the diameter of each point increases with an increasing number of replicates. For demonstration purposes we are showing two pairs of bacteria and three viruses. Recall that for each of the pairs of bacteria (and the two poxviruses) any sample containing one did not contain the other (as shown in [T1,\"Table 1\"]). The empty boxes result from the organisms not being called at any abundance. For MetaPhlAn, that is a result of no viruses being included in the version of the reference database available for this analysis. Kraken assigned no reads to Hanta virus because viral RNA genomes were not included in this version of the reference database (personal communication with D. Wood). This emphasizes the point that a) the ability to create custom databases targeting organisms of interest can be valuable, and b) the performance of any method must be benchmarked against each potential target of interest.</p></td></tr><tr><td><table id=F3 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F3.large.jpg alt=Figure 3.></tr></td><tr><td><h3>Figure 3.</h3></tr></td><tr><td><span class=\"description\">The Positive Predictive Value (PPV, vertical axis) is shown for each organism (boxes on right), at each level of known abundance (horizontal axis, see Table 2), and each program (boxes at top), across a maximum of 50 replicates (indicated by the size of each point). Note that the reference database for MetaPhlAn does not include viruses, and the reference database for Kraken does not include RNA viruses\n",
       "</span></td></tr></table></td></tr><tr><td><p>All methods were able identify the bulk of organisms in their databases at high abundances (75% and 18%, [F3,\"Figure 3\"]), however performance varied considerably at lower abundances and depended on the particular organism and method used. SIANN detected each organism at high confidence, even at levels as low as 0.3% and 0.07% of the total.</p></td></tr></table></td></tr><tr><td><table><tr><td>Discussion</td></tr><tr><td><p>The process of detecting trace amounts of a specific organism in a complex mixture of DNA is challenging enough for an expert, but that pales in comparison to the difficulty of accomplishing the same certainty of detection in an automated manner. The results presented here show that SIANN rapidly detects the presence of a given set of organisms with a high degree of specificity and sensitivity. For example, at the 95% confidence (PPV) cutoff of 0.2, SIANN reliably detects all of the organisms tested here at as low as 0.3% abundance. This strong performance is likely due to the fact that SIANN is able to use a method (read alignment to whole genomes) that would be far too computationally costly if it were applied to the entire collection of known genomes. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.</p></td></tr><tr><td><p>SIANN is available on Illumina’s BaseSpace ([http://www.basespace.illumina.com,\"www.basespace.illumina.com\"]) as a NativeApp, with the database tested here ([app-1,\"Appendix 1\"]), as well as a database made from the NCBI representative set of prokaryotic genomes ([ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/,\"ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/\"]) ([app-2,\"Appendix 2\"]) and the complete set of NCBI viral genomes ([ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/,\"ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/\"]) ([app-3,\"Appendix 3\"]).</p></td></tr><tr><td><p>BaseSpace was chosen as an appropriate release platform because while the entire set of software and dependencies can be deployed by the user from within a graphical user interface, the actual computation takes place in a controlled ‘cloud’ environment. Such a distribution strategy obviates the need to satisfy the multiple software or OS dependencies that often arises with academic computational methods. Results for SIANN are compiled into a report format, showing both the organisms that surpass 95% confidence, as well as the closest strain match for each species. The default view masks the raw data output, so that the results are human-readable and do not present extraneous information. While the code for execution and database-construction on a users system is available from Signature Science, LLC, additional databases on the BaseSpace platform can be made available upon request.</p></td></tr><tr><td><p>There is a neverending list of questions that one could ask of metagenomic sequencing data generated from important samples. Instead of answering them all, we demonstrate a technique with a very narrow focus that is able to report with a high degree of confidence whether a given set of organisms is present in a sample. These results are presented to the user in a comprehensible format, and accessible on a commonly-used web platform. The world of bioinformatics will continue to progress and develop more sophisticated tools for metagenomic analysis, and we hope that the utility of SIANN will convince others to package and benchmark their tools in a way that they can be used with confidence by the larger public, as well as the research community.</p></td></tr></table></td></tr><tr><td><table id=\"references\" class=\"section\"><tr><td><h2>References</h2></td></tr><tr><td><span class=\"reference\">Scalable metagenomic taxonomy classification using a reference genome database, <span class=\"author\">Ames SK</span>, <span class=\"author\">Hysom DA</span>, <span class=\"author\">Gardner SN</span>, <span class=\"author\">Lloyd GS</span>, <span class=\"author\">Gokhale MB</span>, <span class=\"author\">Allen JE</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1093/bioinformatics/btt389&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=23828782&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">A comparative evaluation of sequence classification programs, <span class=\"author\">Bazinet AL</span>, <span class=\"author\">Cummings MP</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1186/1471-2105-13-92&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22574964&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Rapid phylogenetic and functional classification of short genomic fragments with signature peptides, <span class=\"author\">Berendzen J</span>, <span class=\"author\">Bruno WJ</span>, <span class=\"author\">Cohn JD</span>, <span class=\"author\">Hengartner NW</span>, <span class=\"author\">Kuske CR</span>, <span class=\"author\">McMahon BH</span>, <span class=\"author\">Wolinsky MA</span>, <span class=\"author\">Xie G</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1186/1756-0500-5-460&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22925230&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Phymm and PhymmBL: metagenomic phylogenetic classification with interpolated Markov models, <span class=\"author\">Brady A</span>, <span class=\"author\">Salzberg SL</span>, 2009, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.1358&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=19648916&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000269382900018&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Integrative analysis of environmental sequences using MEGAN4, <span class=\"author\">Huson DH</span>, <span class=\"author\">Mitra S</span>, <span class=\"author\">Weber N</span>, <span class=\"author\">Ruscheweyh H</span>, <span class=\"author\">Schuster SC</span>, 2011, <a href=\"/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NjoiZ2Vub21lIjtzOjU6InJlc2lkIjtzOjk6IjIxLzkvMTU1MiI7czo0OiJhdG9tIjtzOjM3OiIvYmlvcnhpdi9lYXJseS8yMDE0LzAxLzEwLzAwMTcyNy5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=\">Abstract/FREE Full Text</a></span></td></tr><tr><td><span class=\"reference\">PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data, <span class=\"author\">Korbel JO</span>, <span class=\"author\">Abyzov A</span>, <span class=\"author\">Mu XJ</span>, <span class=\"author\">Carriero N</span>, <span class=\"author\">Cayting P</span>, <span class=\"author\">Zhang Z</span>, <span class=\"author\">Snyder M</span>, <span class=\"author\">Gerstein MB</span>, 2009, <a href=\"/lookup/external-ref?access_num=10.1186/gb-2009-10-2-r23&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=19236709&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Fast gapped-read alignment with Bowtie 2, <span class=\"author\">Langmead B</span>, <span class=\"author\">Salzberg S</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.1923&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22388286&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences, <span class=\"author\">Liu B</span>, <span class=\"author\">Gibbons T</span>, <span class=\"author\">Ghodsi M</span>, <span class=\"author\">Treangen T</span>, <span class=\"author\">Pop M</span>, 2011, <a href=\"/lookup/external-ref?access_num=10.1186/1471-2164-12-S2-S4&link_type=DOI\">CrossRef</a></span></td></tr><tr><td><span class=\"reference\">MetaSim: a sequencing simulator for genomics and metagenomics, <span class=\"author\">Richter DC</span>, <span class=\"author\">Ott F</span>, <span class=\"author\">Auch AF</span>, <span class=\"author\">Schmid R</span>, <span class=\"author\">Huson DH</span>, 2008, <a href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0003373&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=18841204&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Genomic variation landscape of the human gut microbiome, <span class=\"author\">Schloissnig S</span>, <span class=\"author\">Arumugam M</span>, <span class=\"author\">Sunagawa S</span>, <span class=\"author\">Mitreva M</span>, <span class=\"author\">Tap J</span>, <span class=\"author\">Zhu A</span>, <span class=\"author\">Waller A</span>, <span class=\"author\">Mende DR</span>, <span class=\"author\">Kultima JR</span>, <span class=\"author\">Martin J</span>, <span class=\"author\">Kota K</span>, <span class=\"author\">Sunyaev SR</span>, <span class=\"author\">Weinstock GM</span>, <span class=\"author\">Bork P</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1038/nature11711&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=23222524&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000312933800027&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Computational meta’omics for microbial community studies, <span class=\"author\">Segata N</span>, <span class=\"author\">Boernigen D</span>, <span class=\"author\">Tickle TL</span>, <span class=\"author\">Morgan XC</span>, <span class=\"author\">Garrett WS</span>, <span class=\"author\">Huttenhower C</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1038/msb.2013.22&link_type=DOI\">CrossRef</a></span></td></tr><tr><td><span class=\"reference\">Metagenomic microbial community profiling using unique clade-specific marker genes, <span class=\"author\">Segata N</span>, <span class=\"author\">Waldron L</span>, <span class=\"author\">Ballarini A</span>, <span class=\"author\">Narasimhan V</span>, <span class=\"author\">Jousson O</span>, <span class=\"author\">Huttenhower C</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.2066&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22688413&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000307015700022&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Metagenomic species profiling using universal phylogenetic marker genes, <span class=\"author\">Sunagawa S</span>, <span class=\"author\">Mende DR</span>, <span class=\"author\">Zeller G</span>, <span class=\"author\">Izquierdo-Carrasco F</span>, <span class=\"author\">Berger SA</span>, <span class=\"author\">Kultima JR</span>, <span class=\"author\">Coelho LP</span>, <span class=\"author\">Arumugam M</span>, <span class=\"author\">Tap J</span>, <span class=\"author\">Nielsen HB</span>, <span class=\"author\">Rasmussen S</span>, <span class=\"author\">Brunak S</span>, <span class=\"author\">Pedersen O</span>, <span class=\"author\">Guarner F</span>, <span class=\"author\">de Vos WM</span>, <span class=\"author\">Wang J</span>, <span class=\"author\">Li J</span>, <span class=\"author\">Doré J</span>, <span class=\"author\">Ehrlich SD</span>, <span class=\"author\">Stamatakis A</span>, <span class=\"author\">Bork P</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.2693&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=24141494&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000327698100019&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Ultrafast metagenomic sequence classification using exact alignments, <span class=\"author\">Wood DE</span>, <span class=\"author\">Salzberg SL</span>, </span></td></tr><tr><td><span class=\"reference\">Linking long-term dietary patterns with gut microbial enterotypes, <span class=\"author\">Wu GD</span>, <span class=\"author\">Chen J</span>, <span class=\"author\">Hoffmann C</span>, <span class=\"author\">Bittinger K</span>, <span class=\"author\">Chen YY</span>, <span class=\"author\">Keilbaugh SA</span>, <span class=\"author\">Bewtra M</span>, <span class=\"author\">Knights D</span>, <span class=\"author\">Walters WA</span>, <span class=\"author\">Knight R</span>, <span class=\"author\">Sinha R</span>, <span class=\"author\">Gilroy E</span>, <span class=\"author\">Gupta K</span>, <span class=\"author\">Baldassano R</span>, <span class=\"author\">Nessel L</span>, <span class=\"author\">Li H</span>, <span class=\"author\">Bushman FD</span>, <span class=\"author\">Lewis JD</span>, 2011, <a href=\"/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEyOiIzMzQvNjA1Mi8xMDUiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNC8wMS8xMC8wMDE3MjcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9\">Abstract/FREE Full Text</a></span></td></tr></table></td></tr></table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "paper_html = paper_to_html(test_paper.paper)\n",
    "display(HTML(css))\n",
    "display(HTML(paper_html))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'<table><tr><td style=\"text-align:left;\">SIANN: Strain Identification by Alignment to Near Neighbors</td></tr><tr><td><span class=\"author\">Samuel S. Minot</span>, <span class=\"author\">Stephen D. Turner</span>, <span class=\"author\">Krista L. Ternus</span>, <span class=\"author\">Dana R. Kadavy</span></td></tr><tr><td><table><tr><td>Abstract</td></tr><tr><td><p>Next-generation sequencing is increasingly being used to study samples composed of mixtures of organisms, such as in clinical applications where the presence of a pathogen at very low abundance may be highly important. We present an analytical method (SIANN: Strain Identification by Alignment to Near Neighbors) specifically designed to rapidly detect a set of target organisms in mixed samples that achieves a high degree of species- and strain-specificity by aligning short sequence reads to the genomes of near neighbor organisms, as well as that of the target. Empirical benchmarking alongside the current state-of-the-art methods shows an extremely high Positive Predictive Value, even at very low abundances of the target organism in a mixed sample. SIANN is available as an Illumina BaseSpace app, as well as through Signature Science, LLC. SIANN results are presented in a streamlined report designed to be comprehensible to the non-specialist user, providing a powerful tool for rapid species detection in a mixed sample. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.</p></td></tr></table></td></tr><tr><td><table><tr><td>Introduction</td></tr><tr><td><p>There are many different methods that characterize the mixture of organisms present within a metagenomic dataset. Such datasets are generated when a complex environmental sample is processed by a “next-generation” high-throughput genome sequencing protocol, and they consist of large numbers of short nucleotide sequences. Each sequence represents a small fragment of a randomly selected genome from the very large collection of genomes present in the source sample. Those sequences indicate the presence of one organism or another according to their similarity to a set of known reference genomes. While a given sequence may be unique to one species, it also may be found in diverse organisms across the tree of life. Therefore, one analytical challenge (among many) is to take that collection of sequences (likely numbering in the millions) and accurately determine what species are present in the sample. Here we describe a novel method (SIANN: Strain Identification by Alignment to Near Neighbors) that is specifically designed to rapidly detect a set of targeted organisms from a metagenomic dataset by aligning reads to genomic regions that are unique at the strain or species level.</p></td></tr><tr><td><p>The analytical question motivating a particular piece of metagenomic bioinformatic analysis may vary widely by user and sample type ([ref-11,\"Segata, et al., 2013\"]). For example, the function of the human gut microbiome may depend on the relative abundance of hundreds of species of bacteria and the types of metabolic genes they contain ([ref-15,\"Wu, et al., 2011\"]; [ref-10,\"Schloissnig, et al., 2013\"]). In contrast, the clinical treatment of a patient may depend on whether or not a particular virus, or a consortium of co-infecting pathogens, is/are detected in their blood. It is this second class of presence/absence questions that SIANN is designed to address. SIANN is appropriate for situations in which a user wants to know whether a particular organism or set of organisms is present in a sample, but isn’t interested in the functions encoded in their genomes, the relative abundance of each organism, or any other more in-depth analysis.</p></td></tr></table></td></tr><tr><td><table><tr><td>Methods</td></tr><tr><td><table><tr><td>Approach</td></tr><tr><td><p>Metagenomic classification methods are based on a wide variety of theoretical underpinnings. The basic varieties include alignment of reads to various nucleotide databases or exact matching to nucleotide or protein signature sequences (or kmers). A representative set of recent methods are described in [T1,\"Table 1\"] (also see [ref-2,\"Bazinet & Cummings 2012\"]).</p></td></tr><tr><td><table id=T1 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T1.large.jpg alt=Table 1.></tr></td><tr><td><h3>Table 1.</h3></tr></td><tr><td><span class=\"description\">Summary of methods for metagenomic classification.\\n</span></td></tr></table></td></tr><tr><td><p>Overall, these methods are designed to either classify individual reads to, and/or predict the total abundance of, clades (e.g. genus or species) across the entire tree of life. They generally require reference databases that are very large and/or require a large amount of processing to generate. The gap SIANN is designed to fill is when the entire tree of life is irrelevant, and only predefined subsets of organisms need to be detected. For an underlying method we chose read alignment to diagnostic genomic regions because the algorithms for read alignment are highly parallelizable and have been optimized heavily by the community at large (the current implementation of SIANN uses bowtie2 [[ref-7,\"Langmead & Salzberg, 2012\"]] for the alignment function, but can be adapted to any alignment algorithm). This approach is distinct from using clade-specific marker genes ([ref-12,\"Segata, et al., 2012\"]) because unique regions that are larger, smaller, or outside of genes can also be used. Furthermore, this approach supports the rapid construction of custom databases using reference genome sets that require only minimal user-supplied structure.</p></td></tr><tr><td><p>To understand the principle at work, consider a set of reads that have been aligned to the genomes of several strains belonging to two species. Some regions of those genomes are species-specific, some are strain-specific, and some are shared ([F1,\"Figure 1a\"]). When a set of reads is aligned to those genomes such that each read is placed in as many locations as it has a match (at a reasonably stringent threshold), visual inspection of the distribution of reads yields an intuitive understanding of the true source organism as Species I/Strain B ([F1,\"Figure 1b\"]). If Strain B were not present in the reference database, it would still be clear that the organism was an unknown strain of Species I.</p></td></tr><tr><td><table id=F1 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F1.large.jpg alt=Figure 1.></tr></td><tr><td><h3>Figure 1.</h3></tr></td><tr><td><span class=\"description\">A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.\\n</span></td></tr></table></td></tr><tr><td><p>The unique identification of a species or strain is quantified by the proportion of the genome that is determined to be species- or strain-specific (defined as reads that are aligned to regions that are species- or strain-specific). Each species and strain is then assigned a numerical measure of the proportion that is covered by these diagnostic reads, and that proportional measure is compared to the ideal case, where sequences from a single organism (generated in silico) are aligned against the database in an identical manner. After that normalization factor is applied, the resulting score indicates whether the source sample contained any of the organisms in the reference database. The analysis is conducted independently on both the species and the strain level, so that if the true strain is not present in the database, the species of origin will still be identified. While many methods consider the complete taxonomic tree and assign reads to the least common ancestor, SIANN considers only two taxonomic levels: species and strain, throwing out anything that is not unique at one of those levels and thus obviating many of the confounding factors introduced by manually curated taxonomies.</p></td></tr><tr><td><p>The example shown in [F1,\"Figure 1b\"] indicates that species-specific reads are identified as reads that align to one species (Species I, in that case) but not the other. If Species II were not present in the example shown in [F1,\"Figure 1b\"], a much larger number of reads would be assigned as “species-specific,” when in fact those regions are shared with other species. Therefore, the ability of this method to identify strain- and species-specific sequences is a direct function of the inclusion of near neighbors in the reference database. This characteristic is shared among many classification algorithms, but it is of particular note for this method when users have an opportunity to construct their own database. In order to detect a target species with a high degree of specificity (reducing false positives), it is necessary to include other related species in the reference database. Only by parallel alignment to those near neighbors can the redundant sequences be separated from the species-specific ones. For example, in order to detect Bacillus anthracis in a sample, it would be necessary to include other species of Bacilli in the reference database so that the presence of B. cereus or B. thuringiensis in a sample does not lead to a false call for B. anthracis.</p></td></tr><tr><td><p>The nomenclature of genus, species, and strain is potentially problematic because it does not correspond to a consistent degree of evolutionary distance or genomic distinctiveness. The ability to distinguish two organisms by any method using genomic sequence data is proportional to the amount of each genome that is shared or unique. One might assume that any two organisms of the same species will have a relatively predictable amount of shared genomic identity. However, some pairs of organisms from the same species may have less in common than other pairs of organisms from different species or even genera. This ambiguity impacts SIANN in two ways. If two organisms have very little genomic sequence to distinguish them, the sensitivity of SIANN to detect either one will diminish (the rate of false negatives will increase as the likelihood of sequencing unique regions decreases). Conversely, if an organism is extremely dissimilar to the near neighbors selected for the database, the specificity with which SIANN detects that organism will decline (the rate of false positives will increase as the number of related genomes available in the database decreases). For example, if a database contained only E. coli and B.anthracis, a sample containing B. cereus would be misidentified as contraining B. anthracis. In the intended use case, a database targeting B. anthracis would contain B. cereus and a number of other near neighbors to prevent that kind of misidentification. It would be convenient to say that an ideal database can be made by calculating the ideal genetic distance between all references and then finding an ideal set of organisms to make up that database, but the behavior of any database will be governed by the particular genomes of the organisms it encounters in the wild. Because not all organisms evolve in the same manner (differences in mutation rate, horizontal gene transfer, recombination, etc), the suitability of a database and method to detect a given organism can only be determined by thorough validation and benchmarking, as well as updating the reference database as needed. Users of SIANN may construct their own custom databases to include newly identified genomes or specific subsets of genomes that best suit their research interests.</p></td></tr><tr><td><p>Steps to construct a custom database:\\n</p></td></tr><tr><td><p>The files contained within each SIANN database are a compressed genomic index and a list containing the proportion of each reference genome that was found to be strain- or species-specific during database construction.</p></td></tr><tr><td><p>To run SIANN:\\n</p></td></tr></table></td></tr><tr><td><table><tr><td>Benchmarking</td></tr><tr><td><p>The performance of SIANN (version 1.6) was tested in comparison to the following state-of-the-art metagenomic classification programs: LMAT (version 1.2), MetaPhlAn (version 1.7.7), and Kraken (version 0.9.1b). All of the programs in [T1,\"Table 1\"] were investigated for this effort, and three were chosen based on their ability to run on our high-performance computing cluster with an execution time and memory requirement that would be suitable to a clinical lab. Each program was run on a set of 600 simulated datasets generated by MetaSim ([ref-9,\"Richter, et al., 2008\"]). Each dataset consisted of 15,000,000 reads (100bp single-ended) with Illumina-simulated error (fourth-degree polynomial) ([ref-6,\"Korbel, et al., 2009\"]). The 600 datasets were broken into 12 sets of 50 replicates. Each of the 12 sets contained organisms at different levels of abundance as shown in [T2,\"Table 2\"]. Organisms were specifically chosen in pairs so that the ability to distinguish these near neighbors could be determined. The abundances were staggered at 4-fold intervals so that a wide range could be evaluated. All known species of near neighbors for each of the 12 target organisms were included in the reference database used by SIANN for this benchmarking (“Target Pathogen Database”) and are shown in [app-1,\"Appendix 1\"].</p></td></tr><tr><td><table id=T2 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T2.large.jpg alt=Table 2.></tr></td><tr><td><h3>Table 2.</h3></tr></td><tr><td><span class=\"description\"></span></td></tr></table></td></tr><tr><td><p>Each program outputs a distinct measure. Kraken and LMAT both count the reads assigned to each taxon, MetaPhlAn calculates the abundance, and SIANN outputs a measure of the proportion of diagnostic genomic regions present. To put these measures on an even footing, we empirically calculated the false positive rate for each method over all 600 samples, at each possible measure of output. Because each dataset is made up of known organisms, any result can be classified as true or false. Therefore, for any possible result (say, 513 reads classified by LMAT or 1.6% abundance assigned by MetaPhlAn), one can calculate the proportion of calls with at least the same amount of support that were correct (True Positives/[True Positives + False Positives]), over all of the 600 datasets. That measure is commonly given as Positive Predictive Value (PPV). For each program, the results can be translated from the raw value into a PPV that is based on this empirical measure of error. The key item of interest is the PPV value for the results that we know to be true positives, the defined spike organisms. Another way of describing this approach is to say that the results of each program have been normalized to the false positive error rate that was empirically observed. If another set of samples were generated, the PPV vs. raw value curve ([F2,\"Figure 2\"]) would likely fall differently, but in this case it gives us a means of comparing a diverse set of methods against the same ground truth. If method 1 detects an organism with a higher PPV than method 2 does, it means that method 1 has fewer false positives in the range that it reports true positives, which is the definition of utility in this setting.</p></td></tr><tr><td><table id=F2 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F2.large.jpg alt=Figure 2.></tr></td><tr><td><h3>Figure 2.</h3></tr></td><tr><td><span class=\"description\">Relationship of reported value for each program (horizontal axis, log scale) to the empirically-determined Positive Predictive Value (PPV), shown on the vertical axis. While the exact values depend on the test data used, the general values at significant cutoff values (0.8, 0.9, 0.95 PPV) remain relatively constant across different datasets (data not shown).\\n</span></td></tr></table></td></tr><tr><td><p>For each method, PPV was calculated as a function of raw output value. Briefly, this was done by compiling the output for all 600 samples, labeling each result as false or true based on the sample set that it came from, and then calculating (at each possible value of output) what the proportion of TP/[TP + FP] was for results with at least that level of raw output. Some simplification steps were taken, such as focusing on the species-level assignments (for comparison with methods that do not perform strain assignment), and only taking the top hit for each species from each dataset. Custom R and BASH scripts were used for the data compilation and analysis.</p></td></tr></table></td></tr></table></td></tr><tr><td><table><tr><td>Results</td></tr><tr><td><p>The relationship of raw output value to PPV is shown for each of the four methods in [F2,\"Figure 2\"]. The point at which PPV is very close to 1 (where 95% of results are true positives) is ∼41,000 reads for Kraken, ∼2,800 reads for LMAT, ∼38% abundance for MetaPhlAn, and 0.21 for SIANN. For SIANN this means that having 38% of the species-unique genome covered by reads resulted in the vast majority of calls being accurate.</p></td></tr><tr><td><p>For read-assignment methods (such as LMAT and Kraken), manual inspection of the results may yield a different understanding of confidence than is presented here, or in any automated analysis. For example, while each read that is assigned by LMAT and Kraken fall above a certain cutoff for species-specificity, some individual reads may be much more specific than others. One could identify a read that aligns to a single species of bacteria with 100% accuracy over its 300bp length, with the next closest match being only 90% similar. It is extremely unlikely that a 300bp exact match would arise due to random chance, and so the user could say with confidence that the organism of interest is found within the sequence data (not considering contamination, horizontal gene transfer, etc). However, such an approach is not currently implemented in an automated method, and many of the steps needed to make that assertion are performed manually by a domain expert, including alignment to near neighbors and ensuring that the read does not fall within a transposon, plasmid, etc. Therefore, while one could say that a single read is all that is needed to state with high PPV that an organism is present, the amount of reads assigned in an automated manner needed to achieve that level of PPV will number in the thousands ([F2,\"Fig 2\"]).</p></td></tr><tr><td><p>The next phase of benchmarking was to determine how many raw input reads were needed to achieve the threshold for high PPV. To demonstrate this we plotted the known abundance of each spike organism against the PPV value generated by each method ([F3,\"Figure 3\"]). Each point (an organism at a known level of abundance) is comprised of a maximum of 50 replicates, where the diameter of each point increases with an increasing number of replicates. For demonstration purposes we are showing two pairs of bacteria and three viruses. Recall that for each of the pairs of bacteria (and the two poxviruses) any sample containing one did not contain the other (as shown in [T1,\"Table 1\"]). The empty boxes result from the organisms not being called at any abundance. For MetaPhlAn, that is a result of no viruses being included in the version of the reference database available for this analysis. Kraken assigned no reads to Hanta virus because viral RNA genomes were not included in this version of the reference database (personal communication with D. Wood). This emphasizes the point that a) the ability to create custom databases targeting organisms of interest can be valuable, and b) the performance of any method must be benchmarked against each potential target of interest.</p></td></tr><tr><td><table id=F3 class=\"figure\"><tr><td><tr><td><img src=https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F3.large.jpg alt=Figure 3.></tr></td><tr><td><h3>Figure 3.</h3></tr></td><tr><td><span class=\"description\">The Positive Predictive Value (PPV, vertical axis) is shown for each organism (boxes on right), at each level of known abundance (horizontal axis, see Table 2), and each program (boxes at top), across a maximum of 50 replicates (indicated by the size of each point). Note that the reference database for MetaPhlAn does not include viruses, and the reference database for Kraken does not include RNA viruses\\n</span></td></tr></table></td></tr><tr><td><p>All methods were able identify the bulk of organisms in their databases at high abundances (75% and 18%, [F3,\"Figure 3\"]), however performance varied considerably at lower abundances and depended on the particular organism and method used. SIANN detected each organism at high confidence, even at levels as low as 0.3% and 0.07% of the total.</p></td></tr></table></td></tr><tr><td><table><tr><td>Discussion</td></tr><tr><td><p>The process of detecting trace amounts of a specific organism in a complex mixture of DNA is challenging enough for an expert, but that pales in comparison to the difficulty of accomplishing the same certainty of detection in an automated manner. The results presented here show that SIANN rapidly detects the presence of a given set of organisms with a high degree of specificity and sensitivity. For example, at the 95% confidence (PPV) cutoff of 0.2, SIANN reliably detects all of the organisms tested here at as low as 0.3% abundance. This strong performance is likely due to the fact that SIANN is able to use a method (read alignment to whole genomes) that would be far too computationally costly if it were applied to the entire collection of known genomes. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.</p></td></tr><tr><td><p>SIANN is available on Illumina’s BaseSpace ([http://www.basespace.illumina.com,\"www.basespace.illumina.com\"]) as a NativeApp, with the database tested here ([app-1,\"Appendix 1\"]), as well as a database made from the NCBI representative set of prokaryotic genomes ([ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/,\"ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/\"]) ([app-2,\"Appendix 2\"]) and the complete set of NCBI viral genomes ([ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/,\"ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/\"]) ([app-3,\"Appendix 3\"]).</p></td></tr><tr><td><p>BaseSpace was chosen as an appropriate release platform because while the entire set of software and dependencies can be deployed by the user from within a graphical user interface, the actual computation takes place in a controlled ‘cloud’ environment. Such a distribution strategy obviates the need to satisfy the multiple software or OS dependencies that often arises with academic computational methods. Results for SIANN are compiled into a report format, showing both the organisms that surpass 95% confidence, as well as the closest strain match for each species. The default view masks the raw data output, so that the results are human-readable and do not present extraneous information. While the code for execution and database-construction on a users system is available from Signature Science, LLC, additional databases on the BaseSpace platform can be made available upon request.</p></td></tr><tr><td><p>There is a neverending list of questions that one could ask of metagenomic sequencing data generated from important samples. Instead of answering them all, we demonstrate a technique with a very narrow focus that is able to report with a high degree of confidence whether a given set of organisms is present in a sample. These results are presented to the user in a comprehensible format, and accessible on a commonly-used web platform. The world of bioinformatics will continue to progress and develop more sophisticated tools for metagenomic analysis, and we hope that the utility of SIANN will convince others to package and benchmark their tools in a way that they can be used with confidence by the larger public, as well as the research community.</p></td></tr></table></td></tr><tr><td><table id=\"references\" class=\"section\"><tr><td><h2>References</h2></td></tr><tr><td><span class=\"reference\">Scalable metagenomic taxonomy classification using a reference genome database, <span class=\"author\">Ames SK</span>, <span class=\"author\">Hysom DA</span>, <span class=\"author\">Gardner SN</span>, <span class=\"author\">Lloyd GS</span>, <span class=\"author\">Gokhale MB</span>, <span class=\"author\">Allen JE</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1093/bioinformatics/btt389&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=23828782&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">A comparative evaluation of sequence classification programs, <span class=\"author\">Bazinet AL</span>, <span class=\"author\">Cummings MP</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1186/1471-2105-13-92&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22574964&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Rapid phylogenetic and functional classification of short genomic fragments with signature peptides, <span class=\"author\">Berendzen J</span>, <span class=\"author\">Bruno WJ</span>, <span class=\"author\">Cohn JD</span>, <span class=\"author\">Hengartner NW</span>, <span class=\"author\">Kuske CR</span>, <span class=\"author\">McMahon BH</span>, <span class=\"author\">Wolinsky MA</span>, <span class=\"author\">Xie G</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1186/1756-0500-5-460&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22925230&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Phymm and PhymmBL: metagenomic phylogenetic classification with interpolated Markov models, <span class=\"author\">Brady A</span>, <span class=\"author\">Salzberg SL</span>, 2009, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.1358&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=19648916&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000269382900018&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Integrative analysis of environmental sequences using MEGAN4, <span class=\"author\">Huson DH</span>, <span class=\"author\">Mitra S</span>, <span class=\"author\">Weber N</span>, <span class=\"author\">Ruscheweyh H</span>, <span class=\"author\">Schuster SC</span>, 2011, <a href=\"/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NjoiZ2Vub21lIjtzOjU6InJlc2lkIjtzOjk6IjIxLzkvMTU1MiI7czo0OiJhdG9tIjtzOjM3OiIvYmlvcnhpdi9lYXJseS8yMDE0LzAxLzEwLzAwMTcyNy5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=\">Abstract/FREE Full Text</a></span></td></tr><tr><td><span class=\"reference\">PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data, <span class=\"author\">Korbel JO</span>, <span class=\"author\">Abyzov A</span>, <span class=\"author\">Mu XJ</span>, <span class=\"author\">Carriero N</span>, <span class=\"author\">Cayting P</span>, <span class=\"author\">Zhang Z</span>, <span class=\"author\">Snyder M</span>, <span class=\"author\">Gerstein MB</span>, 2009, <a href=\"/lookup/external-ref?access_num=10.1186/gb-2009-10-2-r23&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=19236709&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Fast gapped-read alignment with Bowtie 2, <span class=\"author\">Langmead B</span>, <span class=\"author\">Salzberg S</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.1923&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22388286&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences, <span class=\"author\">Liu B</span>, <span class=\"author\">Gibbons T</span>, <span class=\"author\">Ghodsi M</span>, <span class=\"author\">Treangen T</span>, <span class=\"author\">Pop M</span>, 2011, <a href=\"/lookup/external-ref?access_num=10.1186/1471-2164-12-S2-S4&link_type=DOI\">CrossRef</a></span></td></tr><tr><td><span class=\"reference\">MetaSim: a sequencing simulator for genomics and metagenomics, <span class=\"author\">Richter DC</span>, <span class=\"author\">Ott F</span>, <span class=\"author\">Auch AF</span>, <span class=\"author\">Schmid R</span>, <span class=\"author\">Huson DH</span>, 2008, <a href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0003373&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=18841204&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a></span></td></tr><tr><td><span class=\"reference\">Genomic variation landscape of the human gut microbiome, <span class=\"author\">Schloissnig S</span>, <span class=\"author\">Arumugam M</span>, <span class=\"author\">Sunagawa S</span>, <span class=\"author\">Mitreva M</span>, <span class=\"author\">Tap J</span>, <span class=\"author\">Zhu A</span>, <span class=\"author\">Waller A</span>, <span class=\"author\">Mende DR</span>, <span class=\"author\">Kultima JR</span>, <span class=\"author\">Martin J</span>, <span class=\"author\">Kota K</span>, <span class=\"author\">Sunyaev SR</span>, <span class=\"author\">Weinstock GM</span>, <span class=\"author\">Bork P</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1038/nature11711&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=23222524&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000312933800027&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Computational meta’omics for microbial community studies, <span class=\"author\">Segata N</span>, <span class=\"author\">Boernigen D</span>, <span class=\"author\">Tickle TL</span>, <span class=\"author\">Morgan XC</span>, <span class=\"author\">Garrett WS</span>, <span class=\"author\">Huttenhower C</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1038/msb.2013.22&link_type=DOI\">CrossRef</a></span></td></tr><tr><td><span class=\"reference\">Metagenomic microbial community profiling using unique clade-specific marker genes, <span class=\"author\">Segata N</span>, <span class=\"author\">Waldron L</span>, <span class=\"author\">Ballarini A</span>, <span class=\"author\">Narasimhan V</span>, <span class=\"author\">Jousson O</span>, <span class=\"author\">Huttenhower C</span>, 2012, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.2066&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=22688413&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000307015700022&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Metagenomic species profiling using universal phylogenetic marker genes, <span class=\"author\">Sunagawa S</span>, <span class=\"author\">Mende DR</span>, <span class=\"author\">Zeller G</span>, <span class=\"author\">Izquierdo-Carrasco F</span>, <span class=\"author\">Berger SA</span>, <span class=\"author\">Kultima JR</span>, <span class=\"author\">Coelho LP</span>, <span class=\"author\">Arumugam M</span>, <span class=\"author\">Tap J</span>, <span class=\"author\">Nielsen HB</span>, <span class=\"author\">Rasmussen S</span>, <span class=\"author\">Brunak S</span>, <span class=\"author\">Pedersen O</span>, <span class=\"author\">Guarner F</span>, <span class=\"author\">de Vos WM</span>, <span class=\"author\">Wang J</span>, <span class=\"author\">Li J</span>, <span class=\"author\">Doré J</span>, <span class=\"author\">Ehrlich SD</span>, <span class=\"author\">Stamatakis A</span>, <span class=\"author\">Bork P</span>, 2013, <a href=\"/lookup/external-ref?access_num=10.1038/nmeth.2693&link_type=DOI\">CrossRef</a>, <a href=\"/lookup/external-ref?access_num=24141494&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom\">PubMed</a>, <a href=\"/lookup/external-ref?access_num=000327698100019&link_type=ISI\">Web of Science</a></span></td></tr><tr><td><span class=\"reference\">Ultrafast metagenomic sequence classification using exact alignments, <span class=\"author\">Wood DE</span>, <span class=\"author\">Salzberg SL</span>, </span></td></tr><tr><td><span class=\"reference\">Linking long-term dietary patterns with gut microbial enterotypes, <span class=\"author\">Wu GD</span>, <span class=\"author\">Chen J</span>, <span class=\"author\">Hoffmann C</span>, <span class=\"author\">Bittinger K</span>, <span class=\"author\">Chen YY</span>, <span class=\"author\">Keilbaugh SA</span>, <span class=\"author\">Bewtra M</span>, <span class=\"author\">Knights D</span>, <span class=\"author\">Walters WA</span>, <span class=\"author\">Knight R</span>, <span class=\"author\">Sinha R</span>, <span class=\"author\">Gilroy E</span>, <span class=\"author\">Gupta K</span>, <span class=\"author\">Baldassano R</span>, <span class=\"author\">Nessel L</span>, <span class=\"author\">Li H</span>, <span class=\"author\">Bushman FD</span>, <span class=\"author\">Lewis JD</span>, 2011, <a href=\"/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEyOiIzMzQvNjA1Mi8xMDUiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNC8wMS8xMC8wMDE3MjcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9\">Abstract/FREE Full Text</a></span></td></tr></table></td></tr></table>'"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper_html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get All Biorxiv Papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "source = 'biorxiv'\n",
    "print(\"\\nThis script only scrapes articles from {}, if not already downloaded.\".format(source))\n",
    "\n",
    "session = requests.session()\n",
    "\n",
    "# import list of articles in data folder\n",
    "existing_shas = [name.strip('.json') for name in get_file_names('data/{}'.format(source))]\n",
    "\n",
    "# import metadata dataframe\n",
    "df_meta = pd.read_csv('metadata.csv')\n",
    "\n",
    "'''\n",
    "sources = df_meta['source_x'].unique()\n",
    "for source in sources:\n",
    "    papers_df = df_meta[df_meta['source_x']==source][['sha','doi']]\n",
    "    papers_array = papers_df.to_numpy()\n",
    "    print('There are {} papers ids from {}.'.format(len(papers_array), source))\n",
    "'''\n",
    "\n",
    "missing_articles = df_meta[(df_meta['source_x']=='biorxiv')&(~df_meta['sha'].isin(existing_shas))]\n",
    "print(\"{} articles to scrape\".format(missing_articles.shape[0]))\n",
    "\n",
    "papers_df = missing_articles[['sha','doi']]\n",
    "papers_array = papers_df.to_numpy()\n",
    "indexes = papers_df.index.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/010389v1.full\n",
      "Paper ID: eccef80cfbe078235df22398f195d5db462d8000 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/012070v1.full\n",
      "Paper ID: c41fdb2efd6d61384a92a84cbba3f8233629a41b saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/018481v1.full\n",
      "Paper ID: 33565294e6bc67fb7ee14dcae6cfdb08148f4ea5 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/027722v1.full\n",
      "Paper ID: 1f9d3f9a1a0e8db6a086e0a2b5ba50cf9f235dae saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/029397v1.full\n",
      "Paper ID: 01e3b313e78a352593be2ff64927192af66619b5 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/030742v1.full\n",
      "Paper ID: nan saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/020834v1.full\n",
      "Paper ID: 3461d71f6890f7e5ba53bf168be3945cdb16d901 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/017178v1.full\n",
      "Paper ID: 1dd898b5ca1ae70ec0e3cad89fc87a165002a99e saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/041558v1.full\n",
      "Paper ID: 8cfb8dfc2f8b2c6e1d60ff48da02de6670694869 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/044594v1.full\n",
      "Paper ID: 6499f5c865c3db3bdfc1e969ea940a80d547a5bb saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/046532v1.full\n",
      "Paper ID: a59906b732bf4a489e282c3e4f499d4166c622e7 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/053967v1.full\n",
      "Paper ID: 1c0b25c8e3c89243fd8797e2e964e12689bc8556 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/054072v1.full\n",
      "Paper ID: d1b26388670660f66249b55ece2beccee4c18d1a saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/048801v1.full\n",
      "Paper ID: 99dab788e984601ee4efcd1138926caf10c00dcf saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/060434v1.full\n",
      "Paper ID: nan saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/061788v1.full\n",
      "Paper ID: 53442eacc3f233078507fa37b78267399e8c1e3b saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/041509v1.full\n",
      "Paper ID: 3b73c8abb7528c958cec0bae1890828af007b599 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/053421v1.full\n",
      "Paper ID: 97339c9d365437d54ea33be1a1b634d35f460208 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/065250v1.full\n",
      "Paper ID: c04e10858012bdd666d50c0d69c3d1e7224ccbea saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/069013v1.full\n",
      "Paper ID: ffbf8ea9948d73572fd052a74afa01b19e6758a3 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/069799v1.full\n",
      "Paper ID: fd3fc2c49f5cc27e4262261d0c9045911d65cb6e saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/070656v1.full\n",
      "Paper ID: 44a440cc1c135c938d2216ea672f8ef4f9c01296 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/073098v1.full\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'text'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-53-07640c99b4ca>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     13\u001b[0m         \u001b[0mhtml\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m         \u001b[0mtmp_soup\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhtml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcontent\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'html.parser'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m         \u001b[0mpaper_webdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbiorxiv_html_paper_data_extractor\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtmp_soup\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m         \u001b[0mpaper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mchain\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mpaper_webdata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m         \u001b[1;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'data/biorxiv/{}.json'\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'id'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'w'\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mfp\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-52-beba7bd08c13>\u001b[0m in \u001b[0;36mbiorxiv_html_paper_data_extractor\u001b[1;34m(soup)\u001b[0m\n\u001b[0;32m      9\u001b[0m     \u001b[0mauthors\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mauthor\u001b[0m \u001b[1;32min\u001b[0m \u001b[0msoup\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'highwire-citation-authors'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind_all\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'highwire-citation-author'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 11\u001b[1;33m         \u001b[0mauthor\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mauthor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'nlm-given-names'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m' '\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mauthor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'nlm-surname'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     12\u001b[0m         \u001b[0mauthors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mauthor\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     13\u001b[0m     \u001b[1;31m#print(authors)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'text'"
     ]
    }
   ],
   "source": [
    "# get papers from data/biorxiv\n",
    "existing_paper_ids = set([name.strip('.json') for name in get_file_names('data/biorxiv')])\n",
    "\n",
    "# go through all papers\n",
    "# new_paper_cnt = 0\n",
    "for row in papers_array:\n",
    "    if row[0] not in existing_paper_ids:\n",
    "        paper = {}\n",
    "        paper['source'] = 'biorxiv'\n",
    "        paper['id'] = row[0]\n",
    "        paper['url'] = 'https://biorxiv.org/content/' + row[1].strip('doi.org') + 'v1.full'\n",
    "        print('Fetching paper data at {}'.format(paper['url']))\n",
    "        html = session.get(paper['url'])\n",
    "        tmp_soup = BeautifulSoup(html.content,'html.parser')\n",
    "        paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "        paper = dict(chain(paper.items(),paper_webdata.items()))    \n",
    "        with open('data/biorxiv/{}.json'.format(paper['id']), 'w') as fp:\n",
    "            json.dump(paper, fp, indent=4)\n",
    "        print('Paper ID: {} saved to data/biorxiv.'.format(paper['id']))\n",
    "        new_paper_cnt += 1\n",
    "        print('-'*60)\n",
    "print('Added {} papers to data/biorxiv.'.format(new_paper_cnt))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://biorxiv.org/content/10.1101/010389v1.full'\n",
    "html = session.get(url)\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Abstract': 'Background Developing methods to reconstruct transmission histories for viral outbreaks could provide critical information to support locating sources of disease transmission. Phylogenetic methods used to measure the degree of relatedness among sequenced viral samples have proven useful in identifying potential outbreak sources. The complex nature of infectious disease, however, makes it difficult to assign a rigorously defined quantitative confidence value assessing the likelihood of a true direct transmission event using genetic data alone.\\nResults A new method is presented to calculate a confidence value assessing the likelihood of a transmission event using both phylogenetic inference and limited knowledge of incubation and infectious duration times. The method is applied to simulations of a foot and mouth disease (FMD) outbreak to demonstrate how the combination of both phylogenetic and epidemiology data can be used to strengthen the assessment of the likelihood of direct transmission over methods using just phylogenetic data or infection timing data alone. The method is applied to a previous FMD outbreak to identify areas where over confidence in previously inferred direct transmission may exist.\\nConclusion Combining knowledge from viral evolution and epidemiology within a single integrated transmission inference framework is an important approach to assess the potential likelihood of transmission events and makes clear how specific features of a virus’ spread through the course of an outbreak will directly determine the potential for confidence in inferred host transmission links.\\n', 'Background': 'Developing methods to protect against RNA virus outbreaks in the face of constantly changing disease dynamics remains one of the great public health challenges of our time. One small piece of this problem involves determining the host-pathogen transmission history to identify an outbreak’s origin. In recent cases such as the 2003 SARS outbreak [1, 2], H5N1 influenza [3] infection in humans and Marburgvirus [4], phylogenetic inference has been used to help identify potential outbreak sources. Typically, a collection of sequences from candidate related sources are compared to infer transmission links between the most closely related sequences. These methods have been used beyond the public health domain to provide evidence in HIV [5, 6] and other forensic investigations [7] to help infer when a suspect is involved in infecting a victim.\\nSchematic of transmission links overlaid on phylogenetic trees T0 (left tree) and T1 (right tree). Transmission links are shown as directed edges connecting two infected hosts X0 to X1 and Y0 to Y1 respectively.\\nWhile phylogenetic data helps establish the evolutionary relationships among the sequenced viral samples, there are no well established tools to translate these results to explicitly infer host to host transmission links with confidence values that accurately capture the uncertainty of host transmission inference. The result is a sampling and transmission inference process that could lead to false confidence when using genetic data to establish host to host transmission [8, 9].\\nComplementary to the viral genome sequencing advances is contact tracing work to find host to host transmission patterns independent of molecular genetic sequence data [10–13]. Previously a transmission inference framework was described that estimates the probability of viral transmission between two hosts using both viral sequence samples and contact tracing data [14, 15]. The method uses labeled training sets from past outbreaks to correlate models of viral evolution associated with direct transmission and indirect transmission to assign a posterior probability that gives a confidence value for inferring a transmission relationship between two infected hosts. One of the limitations of the approach is the need for a robust training set, which can limit application to novel outbreaks. Even when training data is available, a training set over-represented with local outbreak features could skew confidence values. Thus, there is a need to develop additional tools to mitigate limits introduced from training. One feature that can be difficult to accurately account for with limited exemplars is time series data. The objective of this article is to present an approach to incorporate time series data with molecular genetic sequence data to measure the uncertainty of host to host transmission events in the face of limited training data and variability in host infection times.\\nFigure 1 shows a simple schematic to illustrate the variable host infection time transmission inference problem. The figure shows two simple trees, T0 and T1 (left and right tree respectively) where the infected hosts are overlaid as circles on the trees to show the transmission event from host X0 to X1 in T0 and Y0 to Y1 in T1. The larger circle for Y1 represents a longer infection time and the two trees represent the evolutionary relationship between the sequences. Assuming comparable homogeneous nucleotide substitution rates, the longer branch length in T1 could be explained by the longer infection time in host Y1 prior to sampling. Including the direct transmission branch lengths from T0 and T1 in a single genetic based transmission inference model leads to a distribution of genetic distances associated with direct transmission for a wider range of values. Thus, estimating accurate confidence values means ensuring that a representative sampling from the true distribution of infection times can be obtained. To measure the impact of this problem, a new method is introduced to incorporate variation in infection times when assigning a final posterior probability confidence value. The proposed method forgoes the use of training data by relying on prior knowledge of viral infection times to act as a constraint in conjunction with phylogenetic data. The results show that with knowledge of infection times confidence in transmission inference can be high and even with limited knowledge of timing patterns, high confidence values can be obtained that exceed the training based methods. Thus, this new transmission inference approach allows for the use of clinical observations of infection to be combined with inferred viral evolution when training data is not available to measure uncertainty associated with specific transmission hypotheses.\\nEfforts to integrate models of epidemiology and genetic evolution have been ongoing for some time [16–19], yet less work has focused on the specific problem of measuring the certainty of inferring individual host transmission events. One related effort is a method called SeqTrack, which was used to examine the origins of the 2009 H1N1 outbreak [20]. This work is similar to our previous transmission inference [14] approach with a few key differences. Rather than inferring a host transmission graph, a genealogy tree is inferred where nodes represent sequence samples and edges between nodes link a sequence to putative descendants. Sample collection times exclude some candidate edges and edge weights represent genetic distances between two sequences. The genealogy tree is inferred by finding the directed graph analog of the minimum spanning tree. Given the elapsed time between two sequences and a per day mutation rate, a confidence value is assigned for direct genealogy linkages. The framework presented here is a more general framework where variable within host evolution times are assumed to play a role. A second closely related effort is the work to recover transmission links between farms applied to the 2001 FMD outbreak [21].\\nFMD samples were collected and sequenced and an inferred phylogenetic tree was used in conjunction with farm infection times to evaluate the most likely transmission tree. Data from that study is used here and a comparison with this work will be covered in detail later in the paper. While the Cottam et al. approach evaluates a putative outbreak tree overlaid onto a single phylogenetic tree, our main contribution is to provide a framework that models features of transmission inference not previously considered. In particular, uncertainty in transmission inference is newly captured by explicitly considering unobserved nodes in the outbreak tree and uncertainty of phylogeny through the use of multiple potentially competing phylogenetic inferences. The result is a more realistic measure of uncertainty when assigning a transmission link. The framework’s added use of competing transmission hypotheses does, however, come with an increased computational cost. The remainder of the paper details the modifications to our original framework, shows how new information is used in simulations and real data, details important differences with the Cottam et al. approach and introduces a way for dealing with limitations introduced by the framework’s increased computational cost.\\n', 'Results': 'The previous framework is briefly reviewed to motivate the notation used to incorporate outbreak timing data. For any two infected hosts A and B, distance δ = δ(A, B) defines a genetic distance between samples taken from the two hosts. In the simplest form one sequenced sample is available for each host and the genetic relationship is described by a Jukes-Cantor like distance measure (δ can take on more complex forms as will be described later). The variable M represents the number of edges (ignoring direction) linking two nodes in an outbreak tree where nodes represent infected entities and edges represent directed transmission events from a source to a recipient. For the direct transmission hypothesis M=1 the posterior probability is defined to be:\\n\\n\\nwhere  for some maximum transmission chain length n. P (δ|M) is estimated from labeled training data using density estimators or fitting the observed data to a distribution such as Poisson, Negative Binomial or Gamma.\\nThe modified framework explicitly combines knowledge of viral infection times within the genetic based transmission inference framework. This is accomplished by adding a parameter σ and computing the value P (δ, σ|M) rather than P (δ|M). This reflects the desire to estimate the probability of observing the genetic relation δ in conjunction with infection related timing values for the transmission hypothesis M. The σ parameter values are motivated by a SEIR type infection concept [22], which describes a host in one of four states 1) Susceptible (S) to infection 2) Exposed (E) to an infection (but not infectious) 3) Infectious (I) - able to transmit virus to other hosts and 4) Recovered state (R) - where the host is no longer able to transmit the virus. The SEIR model describes outbreak dynamics in a homogeneous mixing population and since the focus of the framework is to predict individual host to host transmission relationships a pre-determined outbreak tree is used, which excludes hosts in the susceptible state. Figure 2 shows the graphical representation of two infected hosts A and B linked by a direct transmission event. Each circle represents a discrete time point marking the transition between states or time intervals. Transmission time points marked red in Figure 2 denote the time of transmission from Host A to Host B. Sample time points marked in purple represent times when viral samples are collected for sequencing. The representation is analogous to a phylogenetic tree where branches (or tree edges) represent time intervals and nodes represent fixed time points. Here edges or transmission events between infected nodes represent instantaneous time events and the host nodes represent time intervals defined by a linear chain of discrete events. The motivation for this representation is to employ a sufficiently realistic epidemiology model to describe the transmission relationships of interest and use existing methods for modeling viral evolution by mapping them to the epidemiology model.\\nHost infection time model. The small circle nodes denote discrete time events and edges denote time intervals. The large blue circles denote a single infected host.\\nThe modified framework reports the probability of jointly observing both the inferred genetic relationship between sequences from host A and B and the timing events associated with the viral infections of the two hosts. A key phylogenetic inferred parameter becomes the estimated “wall clock” time of the inferred most recent common ancestor between the time stamped sequence samples taken from the two hosts and potentially using samples from the larger outbreak. Here the Bayesian estimation software BEAST [23] is used to estimate a series of candidate phylogenetic relationships and inferred times, weighted by a likelihood value for each estimate. Transmission hypothesis testing determines whether the constraints implied by the hypothesis are consistent with the inferred most recent common ancestor times. Next, a description for estimating the probability of direct transmission is given, which is then extended to evaluate any transmission relationship between two nodes in an outbreak tree.\\nProbability distributions describe the expected exposure time and the expected infectious time in the host. An additional key requirement is the assignment of a fixed time point to each observed host that is relative to some global time clock. The point in time context can always be satisfied by assuming that the time is noted when a sample is collected from each infected individual. An alternative is to assume a fixed recovery time. For notational simplicity, a fixed recovery time is used although a comparable procedure is available for working from sample times as the reference. With host X recovery time RX, infectious time duration infX occurring with probability P (infX) and exposure time expX occurring with probability P (expX), the direct transmission hypothesis that host A transmitted the virus to host B is calculated by evaluating all pairwise combinations of expx and infx for each host. Candidate transmission events are marked by cases where the infectious time interval for host A overlaps with a potential initial infection time for host B and the inferred most recent common ancestor time for the sequence samples occurs after the initial infection time of A and prior to the initial infection time of B.\\nFigure 3 shows the notation and the constraints for the most recent common ancestor time (MRCAT). The initial infection time EB of host B is EB = RB − infB − expB and similarly the initial infection time EA for host A is EA = RA − infA − expA, which leads to the constraints defined by the specific transmission hypothesis. The transmission constraint C1(EB, RA, IA) returns 1 if IA ≤ EB < RA and 0 otherwise. The most recent common ancestor constraint C2(M RCAT, EB, EA) returns 1 if EA ≤ M RCAT < EB and 0 otherwise. The probability of EB and EA is defined to be P (EB|expB, infB, RB) = P (expB) × P (infB) and P (EA|expA, infA, RA) = P (expA) × P (infA) respectively. Now the genetic relation parameter δ is a set of estimates M RCATi for 1 ≤ i ≤ q where q is a finite integer determined during an independent phylogenetic parameter estimation process. (For example, using Bayesian genetic inference software BEAST, a default value could be q = 10, 000 estimates.) The direct transmission hypothesis is tested by finding the infection time intervals that are consistent with the inferred most recent common ancestor times and is described as follows:\\n\\n\\nand\\n\\n\\nwhere Z is a normalizing constant that reflects the maximum probability value of the individual timing interval events independent of constraints.\\nExample mapping of a two leaf phylogenetic tree onto the host node n (A or B). En is the initial infection time, In is initial infectious time, TA is transmission time from host A to host B. Sn,1 is the sample collection time (for the first of potentially multiple samples) and Rn is the initial recovery time. G – SA, 1 and G − SB,1 are genetic sequences for the sampled isolates. MRCAT is an inferred time that occurs within the time interval: EA ≤ M RCAT < TA.\\nTo maintain consistency with previous descriptions of the original framework [14], the transmission hypothesis M up to now was described by an integer value, which reports the number of edges in a tree separating the two nodes in question. This is sufficient for genetic only transmission inference when genetic distance is computed without regard to transmission direction. Thus, different sub tree configurations containing the same number of edges are treated identically without regard to transmission direction.\\nFigure 4 shows how the time series process of outbreak events distinguishes transmission direction. For a sub tree connecting two nodes with k edges, there are k distinct sub trees with each sub tree representing a distinct potential source node for the infection. For example, the transmission sub tree with two edges separating host A and B has two possible sub trees to describe the transmission relationship between the two hosts, one where the source node is A and one sub tree where A and B receive the infection through direct transmission from a common source. In general for a transmission hypothesis linking two nodes by k edges, there are k distinct sub trees that must be evaluated.\\nTransmission chains. The nodes labeled “s” represent the putative infection source. Blue nodes represent the two hosts (A and B) with sequenced samples. Green nodes reflect putative infected (hidden) intermediate hosts. Examples of transmission chains from length 1 (the direct transmission case) to 4 are shown.\\nThe color coding of nodes in Figure 4 highlights the two distinct node types in the hypothesis testing framework. The hypothesis test considers an outbreak tree with “observed” nodes and “hidden” nodes. The hidden nodes are marked green in the figure and represent infected individuals where no direct observable data is available. The hidden nodes explicitly model the probability of the alternative transmission hypothesis linking two observed nodes A and B. To evaluate the M > 1 case an evaluation procedure for each sub tree tM ∊ TM (A, B) is used. Thus, P (σ, δ|M > 1) becomes the value of the most probable alternative hypothesis:\\n\\n\\nand the calculation of P (σ, δ|tM) invokes a recursive tree traversal procedure to check the possible transmission timing values that are consistent with the transmission hypothesis tM . Hidden nodes cannot be tied to a global reference time but rather occur within a time range that is relative to the reference time points specified by the observed nodes and satisfy the transmission and phylogenetic constraints. A tree traversal process sets the range of possible infection times using the reference time points for the observed nodes. One of two traversal procedures is invoked depending on the sub tree structure: directed linear chain, where an observed node A is the source and the last node in the transmission chain is the other observed node B (see for example the bottom panel in Figure 4). The alternative traversal process is invoked when the two observed nodes are linked by transmission from a common source as shown in Figure 5. Figure 5 shows an example of the tree traversal process for one specific instance of tM where M = 3. The common source sub tree traversal algorithm starts with each observed node (leaf) independently and moves up the tree to the root. Each node n is assigned an earliest infection time Emin,n and a latest infection time Emax,n. The parent of node n is assigned an earliest and latest infection time for each minimum and maximum future initial infection time at node n. The total number of possible initial infection time ranges for the source node is 2d/2 where d is the length of the chain from the source to the leaf. The set of initial timing event ranges derived from one observable node are compared with the set of infection time ranges derived from the other observable node to determine whether there is a candidate infection time interval that supports the tree’s transmission hypothesis linking both observed nodes.\\nTree traversal procedure (right panel) for the transmission hypothesis shown in the left panel. Left panel shows a transmission hypothesis linking two observed infected nodes A and B (marked in blue), connected by a common unobserved labeled source node marked 0 and two intermediate source nodes marked 1 and 2. The right panel shows the post order tree traversal, where the observed time stamp for each observed node serves as the starting point and the possible infection times for each source node are computed by moving up the tree.\\nThe consistency function g(tM, INC, INF, MRCAT) checks the sub tree tM assuming each node n is assigned a fixed incubation time interval incn ∊ INC and a fixed infectious time interval infn ∊ INF and returns 1 if the constraints are met and 0 otherwise. The final probability calculation is the set of timing intervals that maximize the product of the individual timing probabilities over all nodes:\\nA similar procedure is applied for the linear transmission chain case. Here the algorithm starts with the observed source node and propagates candidate infection timing intervals forward to the next recipient node (rather than backwards to the parent node) until the final observable node is reached and the set of possible infection timing intervals is compared with terminal node timing to determine if a candidate timing interval supports the proposed transmission chain hypothesis.\\nThe key drawback to the method is computational cost. There are 2d+1 timing assignments that must be considered for each combination of timing intervals. If the number of possible distinct values for incn and infn is x the number of timing interval combinations becomes xM+1. Thus, there is a practical limit to the maximal value for M and x. Measuring the potential linkages for distantly related host pairs can be determined by viewing the estimated probability values of linkage up to the maximum M and when all probability values are low, M can be inferred to be too small to allow transmission inference.\\nTo evaluate the potential strength of inference from combining sequence and timing data, a simulation was used to generate genetic sequences from pre-defined outbreak conditions. The simulation uses a generative stochastic process for P (σ, δ|tM=1) and takes the following parameters as input:\\nA tree of infected hosts and their respective transmission links.\\nTwo probability density functions Pinc and Pinf determining the length of time intervals incn and infn respectively for each node n in a outbreak tree.\\nA transmission probability density function Ptrans determining the probability of transmitting the virus from host A to host B at time t within time interval infA.\\nA sample probability density function Psample determining when a sequence sample is collected within time interval infA for host A.\\nA per day mutation rate and an initial source index sequence.\\nA single run simulates an outbreak by propagating sequences across the pre-specified outbreak tree. The index case node A’s initial infection time is set to 0 (e.g. EA = 0) and the initial infectious time IA and recovery time RA are determined by values drawn from the Pinc and Pinf distributions. Secondary infected host B’s initial infection time EB is determined by drawing from the Ptrans distribution to fix the time within the infectious interval from IA to RA that the actual transmission event takes place. The index sequence evolves daily according to the per day mutation rate with distinct sequence stored for each host n for each day from En to Rn. When host A transmits the virus to host B at time t, the sequence in host A at time t becomes the seed sequence for host B. A single sequence is used to represent each host node, the chosen sampled sequence is determined by Psample, which gives a discrete time point within the infectious time interval from In to Rn in host node n. Sampled sequences are assumed to be taken following presentation of symptoms in the infected individual. This simulation approach is similar to a related effort to use an epidemic simulation model to construct large scale seasonal outbreaks of measles with viral genome sequence data [24]. Simulations were used to evaluate different viral sampling strategies for their proficiency in reconstructing viral population histories. A key difference with the work presented here is the focus on recovering individual host transmission chains rather than homogeneous host population infection models [25].\\nUsing the simulated sampled sequences from host nodes, phylogenetic trees were generated with BEAST running 1,000,000 iterations with 100,000 initial runs discarded and evaluating 10,000 sampled trees. The HKY nucleotide substitution model was used with fixed variation, and a relaxed molecular clock [26] with exponential distribution and a fixed population size coalescent model. The genetic distance between two sequences was taken to be the sum of the absolute time difference between two leaves in the tree and the most recent common ancestor. The final distance value is the weighted average over all available inferred trees. The same set of most recent common ancestor times was used as input to the method for integrating time with genetic data. For the test results reported below, the probability distributions for the incubation time and infectious duration time were taken to be uniform across a pre-determined range described below. The simulator reflects reported features from the 2001 UK FMD outbreak described in [21]. Infected hosts here do not reflect individual animals but rather infected farms. An average incubation time of 5 days is assumed with a variation between 2 and 14 days determined by a gamma distribution. The infectious time interval varied from 1 to 10 days drawn from a distribution model on previously described time intervals given by [21]. The objective was to evaluate the impact of combining the genetic data with timing information and apply the evaluation methods to the original 2001 UK FMD outbreak data to assess the confidence of the associated transmission links predicted using similar evidence but using the new inference approach. A per day mutation rate of 2.226 × 10−5 is used following a previously published report [10].\\nFigure 6 shows the tree used for the simulation. The tree is based on a subset of the 2001 outbreak given in [21]. The nodes labeled with two character codes were added to the published tree to connect nodes thought to be linked indirectly such that the total time duration distribution generated from repeated simulations covered the same time frame of 110 days reported for the outbreak. Results are based on 50 independent outbreak simulations. In the genetic distance only based framework labeled training data is used to estimate the likelihood of observing a genetic distance under the different possible transmission hypotheses. In the modified framework, a prior belief of how the virus spreads replaces the evidence taken from a training set. To evaluate the two approaches, 5 fold cross-validation takes the 50 randomly generated outbreaks and partitions the outbreaks into 5 non overlapping sets of 10 distinct outbreaks to evaluate predicted transmission link accuracy using the remaining 40 outbreaks for training. Evaluations are repeated 5 times to report performance for all 50 outbreaks. When training data is not needed, statistics are simply collected on all 50 outbreaks in one step. Three basic prediction methods are compared - using genetic data only, using the epidemiology timing data only and using both lines of evidence together to inform the final prediction.\\nTree used to generate simulated sequences and reflect known transmission links. Two character codes reflect hypothetical farms included in the tree and edges represent transmission relationships between infected farms.\\nPerformance with knowledge of timing. ROC curves showing the ability to correctly infer direct transmission cases (Sensitivity) versus falsely labeling indirect transmission pairs as direct transmission cases (False Positive Rate) using different confidence cut off values. Three methods are shown, using exact knowledge of timing data alone (Timing-Only), using only genetic data (MRCAT-Only) or combining the two and assuming exact knowledge of infection times.\\nFigure 7 shows a ROC curve for the rate of true direct transmission links identified (y-axis) versus the rate of host nodes falsely predicted to be linked (x-axis) using different prediction cutoff values. First consider prediction accuracy with near perfect knowledge of the timing events for the observed host nodes shown by the Timing-Only plot in Figure 7. The linear plot reflects the fact that the method generates a single prediction value so that above a certain cutoff either all of the true direct transmission links are predicted at the expense of falsely predicting transmission links for 3% of the other non-linked host node pairs or using a lower cutoff yields no predicted links. Recall that the calculation is meant to represent the most probable collection of incubation and infectious times consistent with the transmission hypothesis relative to the most probable collection of incubation times independent of the hypothesis. Here just a single set of incubation and infectious duration times are considered (the true time) and thus every pair of nodes with a timing pattern consistent with the predicted transmission event occurs with probability 1. Thus, the false positive rate reflects the fact that roughly 3% of node pairs that are not linked have a timing pattern consistent with the potential for direct transmission. These pairs likely refer to “siblings” or other nodes that are close by in the tree. This provides a baseline measure for quantifying the difficulty of the inference problem and shows that when exact timing knowledge is available the possible transmission inferences become highly constrained. The MRCAT-Only line shows the other end of the spectrum where there are no explicit timing constraints to prevent inference of direct transmission. The plot shows that a low confidence threshold is required to correctly identify all true direct transmission links, which shows that there is a wide overlap between the genetic distance range for both direct and indirectly linked hosts. Combining the two approaches shows that the already low false positive rate from the highly constrained timing data can be reduced still further by adding the constraints introduced by the timing data inferred from the time stamped genetic sequence data.\\nLimited timing knowledge. ROC curves showing the ability to correctly infer direct transmission cases (Sensitivity) versus falsely labeling indirect transmission pairs as direct transmission cases (False Positive Rate) using different confidence cut off values. Four methods are shown, using only timing data (Timing-Only), using only genetic data (MRCAT-Only) or combining the two assuming either a 2-14 day incubation period (Combined:2-14) or a 2-9 day incubation period (Combined:2-9).\\nFigure 7 illustrates how the algorithm works but assumes unrealistically precise knowledge of incubation and infectious duration times. Figure 8 shows the same comparison, but reducing the knowledge of incubation and infectious times. There are two distinct parameter settings shown in the Figure 8. One setting where every observed node is assumed to have an incubation time that could range from 2 to 9 days with equal probability and the estimated infectious duration for each node is based on the known window but assigned and uncertainty of plus or minus one day (Combined:2-9 in Figure 8). The second setting is identical except the incubation time is assumed to extend from 2 to 14 days (Combined:2-14 in Figure 8). The Timing-Only plot uses the 2 to 9 day incubation window and shows that (without using the genetic data) the range of possible timing scenarios more than triples the number of node pairs that are falsely linked by direct transmission from 3% to 10%. Combining the data sources shows improved inference with consistently higher sensitivity levels and less false positives. The results also show there are clear performance trade offs illustrating the underlying difficulty of the problem. On the one hand capping the maximal incubation time at 9 days greatly constrains the number of plausible transmission hypotheses, thus limiting the number of false positives. The trade off is that for the transmission cases where the incubation time actually exceeds 9 days the true positives can be missed, thus there is a hard upper bound on the sensitivity rate that is less than 1. Further relaxing restrictions to allow for the longer incubation times of up to 14 days allows detection of all true positives, but the number of false positives increases too. The key here seems to be the uncertainty associated with assigning a time to the most recent common ancestor. Longer incubation times open more plausible phylogenetic scenarios where two hosts are falsely linked by direct transmission but are actually linked by a shared infectious source. One potential way to strengthen inference could be to collect more sequences per farm spread over multiple time points, which in some situations may reduce the range of most likely most recent common ancestor times.\\nA second experiment was conducted to re-examine the transmission relationships reported by Cottom et al. on samples taken from a subset of farms infected during the 2001 United Kingdom outbreak. Although the true outbreak tree cannot be empirically determined, re-examination of this dataset proved useful in determining whether our new transmission inference method using similar information as input would lead to similar conclusions. Moreover, differences in output should highlight important differences in the two approaches and help assess the strengths and weaknesses of the different approaches.\\nFigure 9 shows the higher confidence inferred transmission links for the published 2001 outbreak data with whole viral genome sequence data available for each farm. An incubation window of 2 to 14 days and an estimated infectious time interval with a 1 day uncertainty was used to find the most likely transmission chain. Cottom et al. similarly apply a transmission inference algorithm to the data using a closely related approach that takes both estimated infection times and genetic sequence data as input. Cottom et al. use a statistical parsimony tree relating the collection of sequences as the basis for building putative transmission trees. The algorithm starts at the leaves of the parsimony tree and assigns the inferred common ancestor nodes to one of the farms (with a descendant leaf) as the putative source. All possible tree labellings are evaluated using an infection time likelihood score, which similar to our approach estimates likely infection times based on estimated incubation and infectious duration times. Several important differences between the two approaches emerge after examining the differences in output. One difference is that our method incorporates putative unobserved farms into transmission hypothesis tests, which help prevent over confidence in certain transmission inferences. As will be discussed in the next section, this can also form the basis for supporting a more flexible framework for evaluating more sparsely sampled outbreaks. A second difference is that multiple phylogenetic tree topologies are used in the form of multiple most recent common ancestor time estimates, which better reflect the uncertainty of phylogenetic inference. Finally, a key difference is in the information representation format. The Cottom et al. approach assigns outbreak tree nodes to phylogenetic tree nodes whereas our method assigns phylogenetic tree nodes to outbreak tree nodes, which represent variable length time. The significance of these differences is that the new framework can test many additional plausible transmission hypotheses, which would otherwise be ignored.\\nA most likely inferred transmission tree with links comparing the tree to previous reports. Red lines show novel links and dashed lines show previously reported alternative links. Each node represents an infected farm.\\nLikelihood for different transmission hypothesis for C to P and O to P, with different transmission chain lengths M.\\nDespite the differences, there was largely strong consensus for predictions between the two methods with all of the farms linked with high confidence in [21] predicted in the new method except the K to B link (results not shown). The contiguous edges marked in black show transmission inferences that match [21]. Linkage between K and B required increasing the infectious window beyond the initial reported estimate. To make a complete comparison the discrepancy is noted and the infectious window was increased to 4 days to accommodate the possibility of a transmission link from K to B. The two dashed edges show links made by the new inference tool not present in the most probable transmission tree reported in [21]. The two red edges show the direct transmission inferences in the most likely Cottam outbreak tree that are not present in the most probable tree reported here. Next the key differences in the two approaches are discussed with respect to differences in observed transmission inference.\\nTransmission hypotheses include the potential for intermediate unobserved farm involvement to estimate confidence values.\\nFor farm P, the new method only slightly favors an alternative source - C over O (0.57 versus 0.59) making it impossible to preclude either farm as the likely source. The favoring of C over O can be explained in part by differences in the alternative hypothesis likelihood scores. Figure 10 shows the likelihood scores for each of the individual transmission hypothesis linking C to P and O to P. The x-axis is labeled by the number of intermediate host nodes (+ 1). Adjacent unlabeled values have the same M value but a different sub tree structure. The figure shows that the direct hypothesis scores are nearly identical but the alternative hypothesis for M=2 shows that an indirect link for O to P is more plausible. Thus, the explicit scoring of alternative transmission hypotheses involving hypothetical infected farms highlight potential transmission inference ambiguity. In the case of the Cottam tree, the potential for an O to P link is not discussed, however, it appears that this possibility may be precluded due to the structure of the parsimony tree.\\nInferred most recent common ancestor times. Range of inferred most recent common ancestor times (MRCAT) between the B,O sequence pairs (red) and the K,O sequence pairs. Shaded area under the two curves show the time intervals consistent with the respective transmission hypotheses.\\nUse of multiple most recent common ancestor time estimates in addition to transmission time is included in the transmission likelihood calculation.\\nFor farm O, the current method more clearly favors B as the source over K. Figure 11 shows weighted probability density estimates for the most recent common ancestor times for B and O (red) and K and O (blue). The red and blue shaded area under the curves show the range of predicted initial infection times consistent with the MRCAT and the direct transmission hypothesis for B to O and K to O respectively.\\nPortion of the maximum credibility tree. Phylogenetic tree leaves represent sequence taken from four farms: O, C, D, M and P. The putative transmission chain O to C to P to M is overlaid onto the phylogenetic tree, with each farm assigned to an ancestral node.\\nThe red area exceeds the blue area indicating that the inferred MRCA times between B and O are more frequently consistent with the B to O direct transmission hypothesis than the K to O alternative. Similar to the previous case, it appears that the inferred tree used in Cottam may preclude the possibility of a K to B to O transmission path.\\nFarm transmission is modeled to occur instantaneously (with respect to time) and the variable length evolution occurs within the host.\\nThe maximum credibility tree shown in Figure 12 indicates the limitation of using the inferred ancestral nodes as the basis for reconstruction of the outbreak tree. The transmission chain O to C to P to M includes a 0 length branch from C to P that suggests a transmission from farm C to farm P with no genetic variation in the virus, which would be rejected as unlikely given minimum incubation times and the expected mutation rate. In the new method, however, since farm infections are represented by time intervals, the initial infection time of farm C can be shown to occur higher up the tree between two phylogenetic tree nodes, which would imply some mutation to occur on farm C prior to transmission to farm P. Thus, mapping the inferred phylogeny onto the outbreak tree rather than assigning outbreak nodes to phylogenetic tree nodes allows for the consideration of transmission hypotheses that would otherwise be excluded.\\nA strength of the Cottam et al. approach is that the transmission tree is built using all the sequences collectively by iteratively labeling the internal tree nodes rather than treating pairs of sequences independently, thus some transmission hypotheses conceivably could be more readily rejected by looking at multiple farms simultaneously. The new method shows the use of multiple tree topologies in a way that is similar to other recent applications [27] and demonstrates a natural way to incorporate multiple tree topologies in a single sequence comparison procedure. One potential improvement to the Cottam et al. approach would be to include additional candidate tree topologies for evaluation and report likelihood values that are averaged across multiple trees.\\nWhile it may be possible to densely sample from an outbreak in select circumstances, in many cases only sparse sampling is possible and this is likely to be another limitation of the published Cottam et al. approach, which used dense farm sampling. Integrating phylogenetic data with infection timing data to infer more distant transmission linkages, however, is expected to be considerably more difficult. Intuitively, distinguishing between a transmission chain of length 20 from a chain of length 25 is not likely given the range of possible infection times. This is not a limitation of the method but a limitation of exploitable transmission inference knowledge. Nevertheless, a potential contribution of the method presented here is that if it is possible to relax the requirements of an exhaustive search of infection times, the framework could be used to more precisely quantify the uncertainty associated with making inherently more difficult transmission inferences and thus shed light on the conditions where more distant transmission links can be made.\\nTo investigate this problem we introduce two random sampling steps to the algorithm, which when not implemented would otherwise preclude consideration of longer transmission chains. The first sampling step is to draw k different timing samples from the underlying incubation and infectious time duration distributions Pinc and Pinf respectively. For the results described here k was set to 50. The second random sampling step is in the tree traversal procedure illustrated in Figure 5, which is designed to determine whether there is a sequence of timing events for a particular transmission chain and a given collection of infection time durations that is consistent with the inferred most recent common ancestor time. Rather than an exhaustive search, a fixed number of random trials are taken. This approach is thus biased against inferring links requiring rarely occurring timing intervals that would be required to support the most recent common ancestor time. We note that additional sampling strategies could be implemented to make this approach more robust, however, the current modifications should be sufficient for evaluating the potential to apply this approach to larger sparsely sampled outbreaks.\\nFor evaluation a variation on the previously described FMD outbreak simulation is used. Rather than using the smaller fixed tree topology, a tree topology was generated at random starting with an index node and randomly generating child nodes using a branching factor of 1.5, a maximal outbreak duration of 110 days and a maximal node size of 2,000. 100 trees were generated and the 40 trials that generated trees with 2,000 nodes were selected for further analysis. For each tree a pair of nodes was chosen at random to represent each path length present in the tree starting from 1 up to the maximal observed path length. To limit computational costs, only odd path lengths were evaluated.\\nAn example maximum credibility tree for a sparsely sampled simulated FMD outbreak. Tree leaves are labeled with numbers corresponding to nodes taken from a 2,000 node simulated outbreak tree. Select node pairs with known transmission relationships were evaluated. The selected node pairs are labeled by their transmission links using blue edges to link pairs and the path length marked in blue. The two sub trees are connected by a common root (not shown for image compactness).\\nFigure 13 shows an example consensus tree for the selected pairs taken from an outbreak chosen at random. The directed edges labeled in blue show the path lengths connecting the selected nodes with the path length labels from 1 to the maximum observed (23) for this particular outbreak tree. The tree infers evolutionary relationships that are generally consistent with what would be expected given the different transmission chain lengths observed. For example, a path length of 23 separates node 628 from 1740 and indeed the two nodes end up in two different subclades of the tree. The key question becomes can additional information be obtained by incorporating knowledge of infection times and alternative transmission hypothesis testing with the phylogenetic data to better capture the range of possible relationships among the infected farms. To answer this, we first show the likelihood evaluation for the node pairs taken from the tree in Figure 13 and then show the same likelihood evaluations averaged across all 40 randomized outbreak trials. To limit computational costs, evaluated path lengths start with 1 and 5 and continue in intervals of 5 up to 35. The likelihood value P (δ|M) is reported as the maximal likelihood value for all alternative sub trees of path length M. Figure 14 shows the value of P (δ|M) for each pair of nodes with node pairs labeled by their true transmission path length. For the sampled outbreak shown in Figure 13, with the exception of M=11, node pairs for path lengths up to 15 show P (δ|M) values that favor a specific hypothesis or a small subset of hypotheses. As the true path length increases, the average likelihood value drops and the distinction between different path lengths becomes less clear. For the node pair separated by path length of 15 for example, there are clearly higher values for inferring a path length from between 5 and 20, but there is limited distinction within this range. For the most part the framework largely reinforces what is visualized using the consensus phylogenetic tree. An interesting case, however, is the node pair (1255, 1888) linked by a transmission chain of 9, compared with node pair (713, 1119) separated by chain length 13. In the framework, both pairs of nodes have likelihood values that peak at M=5, however the alternative transmission hypotheses for linking (713, 1119) with M=10 or M=15 are much higher than they are for the (1255, 1881) case. This is an example where the relationships inferred by the tree may be misconstrued without the explicit use of a transmission inference framework. It is important to note that the transmission framework does not resolve all conflicts. For the M=11 case for node pair (138, 184), the nodes show up in two different subclades suggesting a more distant transmission relationship, than might be expected by the arrangement of other pairs in the same subclade and separated by longer transmission chains. In this case, the transmission inference framework also suggests a more distant transmission link, given the relatively uniform distribution of low P (δ|M) values.\\nLikelihood values are given for the 12 selected transmission pairs shown in Figure 13. Each node pair is evaluated for different values of M from 1 to 35 and the true path length for each node pair is shown in the legend.\\nLikelihood values for all node pairs for a select path length averaged across all 40 trials. Plot shows path lengths where the average likelihood value peaks at M=5 or less.\\nFigure 15 and Figure 16 show P (δ|M) values averaged over the 40 random trials. For clarity the two figures are used to separate the distributions into two classes, those with peaks at M ≤ 5 and those that peak at M > 5. (Note that the maximal path length observed was 37, however, due to the 2,000 tree node limit, longer path lengths occur less frequently in the 40 trial experiment and thus path lengths with less than 40 observations are not shown.) The results show that the framework assigns a likelihood score that roughly reflects the true underlying path length. In particular, all path lengths less than 10 clearly peak at M=1 or M=5 in the figure. The results do suggest room for improvement, however, since path lengths for M=11-17 still show their maximum likelihood value at M=5. A possible reason for this is that for longer transmission chains the space of inconsistent infection times may expand at a rate that is not proportional to the increase in the number consistent infection times. Since the method uses a fixed random sample size this would favor shorter more constrained transmission hypotheses. Nevertheless, even where the likelihood value peaks at M=5, for longer transmission chains the likelihood values for M > 5 come increasingly closer to the peak until there is finally a peak shift at path length 19 from M=5 to M=10. It is interesting to note that as the path length increases, the peak likelihood value decreases quantifying the increasing ambiguity associated with making long transmission chain inferences. Despite this difficulty, the results indicate that in many cases it may be possible to assign a transmission relationship using a more inclusive class of transmission chain lengths. For example, nodes linked by a transmission path length less than 10 appear to be fairly distinct from those separated by path lengths greater than 20. These distinctions could prove to be of critical importance, particularly when evaluated in the context of an underlying contact network.\\nLikelihood values for node pairs connected by longer path lengths. Plot is similar to Figure 15 but shows path lengths where the average likelihood value peaks at M > 5.\\n', 'Discussion': 'It is important to note that while the presented method is designed to use prior knowledge of viral infection timing patterns, uniform probability distributions were used since an accurate estimate of the distribution is not likely in the general case. More generally since the method requires both viral evolution and epidemiology knowledge there are significant outstanding challenges not addressed in this report. The method makes the simplifying assumption of a neutral evolution model without recombination. This assumption may be appropriate for certain acute viral infection types where host immunity does not play a significant role, however, chronic infections with wider persistent circulation of multiple subtypes increase the chances to misinterpret the transmission process. Additional work is need to ensure that any violation of assumptions of evolution are detected and (ideally) corrected. The model currently lacks inter-species evolution modeling and as sequencing technology advances there will be opportunities to better measure within host viral populations [28] and the affects on the genetic features of the recipient host’s initial seeding population. A consequence of this limited knowledge is reliance on a potentially simplifying feature of viral population structure. When testing direct transmission hypotheses the most recent common ancestor between sequences sampled from the source and recipient are assumed to reside in the source host. However, with a sufficiently large and genetically diverse host population it is at least conceivable that the most recent common ancestor resides in an earlier predecessor node.\\nWith respect to epidemiology modeling, the paper assumes there are accurate values for P (M), which describe the prior knowledge about the relative frequency of the different types of transmission chains. There has been considerable progress in this area in recent years linking contact networks with the more traditional stochastic compartmentalized transmission models to yield estimates [11, 29] that can be inferred using knowledge gleaned from previous outbreaks. Moreover, there are other potentially useful epidemiological models that could be substituted for the SEIR approach presented here [30]. Another important area of development is the use of sequence data and a coalescent model to estimate epidemiological parameters. Recent work demonstrates the potential to estimate parameters for a SIR model from genetic sequence data [31], which could provide an important basis for specifying the epidemiological parameters for transmission inference, when limited non-genome sequence derived observation is available. A strength of the approach is to provide a framework that can readily make use of increasing amounts of knowledge for a particular outbreak. For example, awareness of infections may increase during the course of an outbreak and thus favor using two different priors for the infectious duration times, one at an early outbreak stage and another at a later stage of the outbreak. Other clinical features of infection can be taken into account either by changing prior likelihoods for alternatives in the underlying outbreak tree structure or by making adjustment to priors in the timing distributions. In some cases prior knowledge could be used to narrow the window of possible infection times and greatly strengthen the confidence of transmission inference. In other cases prior knowledge may end up increasing the window of possible infection times and thus highlight the inherent complexity of reconstructing transmission events.\\n', 'Conclusion': 'A new method is presented that takes as input a collection of time-stamped viral genomes and basic knowledge about the timing features of the infection, the underlying transmission network and returns confidence values that assess the probability of correctly inferring host transmission links. Simulations of an FMD outbreak show how the combined use of the epidemiology based timing constraints with the phylogenetic inferred most recent common ancestor times improve confidence in transmission inference over methods that rely on just one individual feature. The simulations show that labeled training data measuring genetic distance between direct and indirect transmission cases can be used without the additional knowledge of the features of the viral transmission, however, appropriate training data in some cases may not be available. The approach presented is expected to provide a basis for building flexible tools to more readily map inferred viral evolutionary relationships onto epidemiological models to more accurately quantify host transmission relationships.\\n', 'Footnotes': 'Email: Jonathan E Allen*- allen99{at}llnl.gov; Stephan P Velsko - velsko2{at}llnl.gov;\\n', 'Reference': ''}\n"
     ]
    }
   ],
   "source": [
    "sections = {}\n",
    "for section in tmp_soup.find_all('div',attrs={'class':'section'}):\n",
    "    section_header = section.find('h2')\n",
    "    if section_header:\n",
    "        section_name = section_header.text\n",
    "    \n",
    "    if section_header and section_name != 'References' and section_name != 'Acknowledgments':\n",
    "        #print(section_name)\n",
    "        sections[section_name] = ''\n",
    "        for p in section.find_all('p'):\n",
    "            sections[section_name] += p.text + '\\n'\n",
    "        #print(sections[section_name])\n",
    "        #print()\n",
    "print(sections)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'T1', 'label': 'Table 1.', 'description': 'Summary of methods for metagenomic classification.\\n', 'image': None}\n",
      "{'id': 'T2', 'label': 'Table 2.', 'caption': 'The abundance of each target organism in each set of simulated datasets. Each set is indicated by the number in the top row, and was generated with 50 replicates.', 'description': '', 'image': None}\n"
     ]
    }
   ],
   "source": [
    "# get tables\n",
    "tables = []\n",
    "html_tables = tmp_soup.find_all('div',attrs={'class':'table'})\n",
    "for html_table in html_tables:\n",
    "    table = {}\n",
    "    table['id'] = html_table['id']\n",
    "    label = html_table.find('span',attrs={'class':'table-label'})\n",
    "    if label:\n",
    "        table['label'] = label.text\n",
    "    caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "    if caption:\n",
    "        table['caption'] = caption.text\n",
    "    table['description'] = ''\n",
    "    for p in html_table.find_all('p'):\n",
    "        table['description'] += p.text + '\\n'\n",
    "    table['image'] = None\n",
    "    tables.append(table)\n",
    "\n",
    "for table in tables:\n",
    "    print(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<p class=\"first-child\" id=\"p-9\">A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.</p>]"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "html_figures[0].find_all('p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "collapsed": true,
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/030742v1.full\n",
      "What’s in my pot? Real-time species identification on the MinION™ \n",
      "Paper ID:  saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/060434v1.full\n",
      "The ATP synthase subunit β (ATP5B) is an entry factor for the hepatitis E virus 52a98150f5d9e362964cbaaaab7921df\n",
      "Paper ID: 52a98150f5d9e362964cbaaaab7921df saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/073098v1.full\n",
      "Zika virus infection as a cause of congenital brain abnormalities and Guillain-Barré syndrome: systematic review 46920aca40b20654ab2ca3372206bc2f\n",
      "Paper ID: 46920aca40b20654ab2ca3372206bc2f saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/074559v1.full\n",
      "Identification of quercetin from fruits to immediately fight zika 8f4b98da50277e6c9516aef0227f5cca9af6b5e8\n",
      "Paper ID: 8f4b98da50277e6c9516aef0227f5cca9af6b5e8 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/078113v1.full\n",
      "Unique properties of Zika NS2B-NS3pro complexes as decoded by experiments and MD simulations f9b108c052ded463773a9838a8157ac7d5ba6d1a\n",
      "Paper ID: f9b108c052ded463773a9838a8157ac7d5ba6d1a saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/078584v1.full\n",
      "Comparative analysis of variation and selection in the HCV genome f147fd5b711abecc5989e5dda047657f\n",
      "Paper ID: f147fd5b711abecc5989e5dda047657f saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/078543v1.full\n",
      "Identification of a Zika NS2B-NS3pro pocket susceptible to allosteric inhibition by small molecules including qucertin rich in edible plants 5b27a16123f46b5aaeda3171d22e075dcbd36ee6\n",
      "Paper ID: 5b27a16123f46b5aaeda3171d22e075dcbd36ee6 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/080333v1.full\n",
      "Genetic, transcriptome, proteomic and epidemiological evidence for blood brain barrier disruption and polymicrobial brain invasion as determinant factors in Alzheimer’s disease 28b107243576723248ad4053261000311a22f134\n",
      "Paper ID: 28b107243576723248ad4053261000311a22f134 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/065789v1.full\n",
      "The genome of the crustacean Parhyale hawaiensis: a model for animal development, regeneration, immunity and lignocellulose digestion 5570cb9ff2905fda60cfd6e94f3c32774b765265\n",
      "Paper ID: 5570cb9ff2905fda60cfd6e94f3c32774b765265 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/084129v1.full\n",
      "Accurate prediction of human essential genes using only nucleotide composition and association information e6fe63419fe399f9c77513c1c11a043db006a2f3\n",
      "Paper ID: e6fe63419fe399f9c77513c1c11a043db006a2f3 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/089128v1.full\n",
      "Mycoplasma Faucium and Breast Cancer dff716159738acd6bd53b4be12af45358a58752e\n",
      "Paper ID: dff716159738acd6bd53b4be12af45358a58752e saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/090936v1.full\n",
      "Lung epithelial cells have virus-specific and shared gene expression responses to infection by diverse respiratory viruses 8c900409a679dc1f16ef13c73500cbb53605b683\n",
      "Paper ID: 8c900409a679dc1f16ef13c73500cbb53605b683 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/094623v1.full\n",
      "Vitrification after multiple rounds of sample application and blotting improves particle density on cryo-electron microscopy grids 6aa2716c00d89f4b760a9167b15d6203303ae6b9\n",
      "Paper ID: 6aa2716c00d89f4b760a9167b15d6203303ae6b9 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/072652v1.full\n",
      "Containing Emerging Epidemics: a Quantitative Comparison of Quarantine and Symptom Monitoring 9b5f5119bbfbded3245acc37859cefde967458e7\n",
      "Paper ID: 9b5f5119bbfbded3245acc37859cefde967458e7 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/099069v1.full\n",
      "Epidemiological and ecological modelling reveal diversity in upper respiratory tract microbial population structures from a cross-sectional community swabbing study 8eb99586e9103599ab2f58cc33ca94cd0b60ae53\n",
      "Paper ID: 8eb99586e9103599ab2f58cc33ca94cd0b60ae53 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/101774v1.full\n",
      "Spike-based phylogenetically defined clades within the Alphacoronavirus 1 species e0fd364797eb02a280e25e2649dfe6d6c0d33e05\n",
      "Paper ID: e0fd364797eb02a280e25e2649dfe6d6c0d33e05 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/104026v1.full\n",
      "The immunomodulatory CEA cell adhesion molecule 6 (CEACAM6/CD66c) is a candidate receptor for the influenza A virus c2add3e0b0de0b9c5a3c18ea9bdebfbb\n",
      "Paper ID: c2add3e0b0de0b9c5a3c18ea9bdebfbb saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/111567v1.full\n",
      "PGC-1α coordinates with Bcl-2 to control cell cycle in U251 cells through reducing ROS 95754ec84bdee0637765a5ee34983a5f6cc86eed\n",
      "Paper ID: 95754ec84bdee0637765a5ee34983a5f6cc86eed saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/112904v1.full\n",
      "The translational landscape of Zika virus during infection of mammalian and insect cells b4fc44eabb1f640124fff98990c85dc5c9984aee\n",
      "Paper ID: b4fc44eabb1f640124fff98990c85dc5c9984aee saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/114819v1.full\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-231-353969646f51>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     16\u001b[0m     \u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m'https://biorxiv.org/content/'\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mrow\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'doi.org'\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m'v1.full'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0msource\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'biorxiv'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 18\u001b[1;33m         \u001b[0mpaper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mscrape_biorxiv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     19\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     20\u001b[0m         \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Error source not biorxiv. ({})'\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msource\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-7-a77319cc75df>\u001b[0m in \u001b[0;36mscrape_biorxiv\u001b[1;34m(paper)\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0mscrape_biorxiv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Fetching paper data at {}'\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m     \u001b[0mhtml\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m     \u001b[0mtmp_soup\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhtml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcontent\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'html.parser'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[0mpaper_webdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbiorxiv_html_paper_data_extractor\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtmp_soup\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\requests\\sessions.py\u001b[0m in \u001b[0;36mget\u001b[1;34m(self, url, **kwargs)\u001b[0m\n\u001b[0;32m    544\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    545\u001b[0m         \u001b[0mkwargs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'allow_redirects'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 546\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'GET'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    547\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    548\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0moptions\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\requests\\sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    531\u001b[0m         }\n\u001b[0;32m    532\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 533\u001b[1;33m         \u001b[0mresp\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    534\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    535\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\requests\\sessions.py\u001b[0m in \u001b[0;36msend\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    644\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    645\u001b[0m         \u001b[1;31m# Send the request\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 646\u001b[1;33m         \u001b[0mr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    647\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    648\u001b[0m         \u001b[1;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\requests\\adapters.py\u001b[0m in \u001b[0;36msend\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    447\u001b[0m                     \u001b[0mdecode_content\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    448\u001b[0m                     \u001b[0mretries\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmax_retries\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 449\u001b[1;33m                     \u001b[0mtimeout\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    450\u001b[0m                 )\n\u001b[0;32m    451\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    598\u001b[0m                                                   \u001b[0mtimeout\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mtimeout_obj\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    599\u001b[0m                                                   \u001b[0mbody\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mbody\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 600\u001b[1;33m                                                   chunked=chunked)\n\u001b[0m\u001b[0;32m    601\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    602\u001b[0m             \u001b[1;31m# If we're going to release the connection in ``finally:``, then\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    341\u001b[0m         \u001b[1;31m# Trigger any extra validation we need to do.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    342\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 343\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_validate_conn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    344\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    345\u001b[0m             \u001b[1;31m# Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36m_validate_conn\u001b[1;34m(self, conn)\u001b[0m\n\u001b[0;32m    837\u001b[0m         \u001b[1;31m# Force connect early to allow us to validate the connection.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    838\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'sock'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m  \u001b[1;31m# AppEngine might not have  `.sock`\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 839\u001b[1;33m             \u001b[0mconn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    840\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    841\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mconn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_verified\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\urllib3\\connection.py\u001b[0m in \u001b[0;36mconnect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    299\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mconnect\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    300\u001b[0m         \u001b[1;31m# Add certificate verification\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 301\u001b[1;33m         \u001b[0mconn\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_new_conn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    302\u001b[0m         \u001b[0mhostname\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mhost\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    303\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\urllib3\\connection.py\u001b[0m in \u001b[0;36m_new_conn\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    157\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    158\u001b[0m             conn = connection.create_connection(\n\u001b[1;32m--> 159\u001b[1;33m                 (self._dns_host, self.port), self.timeout, **extra_kw)\n\u001b[0m\u001b[0;32m    160\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    161\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mSocketTimeout\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\urllib3\\util\\connection.py\u001b[0m in \u001b[0;36mcreate_connection\u001b[1;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[0;32m     68\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0msource_address\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     69\u001b[0m                 \u001b[0msock\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msource_address\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 70\u001b[1;33m             \u001b[0msock\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msa\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     71\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0msock\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     72\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# keep track of index\n",
    "i = 0\n",
    "modified = False\n",
    "for row in papers_array:\n",
    "    paper = {}\n",
    "    paper['source'] = source\n",
    "    # if no sha => generate one that doesn't exist.\n",
    "    if str(row[0]) == 'nan':\n",
    "        paper['id'] = generate_sha(existing_shas)\n",
    "        # save new hash and add to existing_shas\n",
    "        existing_shas.append(paper['id'])\n",
    "        df_meta.loc[indexes[i], 'sha'] = paper['id']\n",
    "        modified = True\n",
    "    else:\n",
    "        paper['id'] = row[0]\n",
    "    paper['url'] = 'https://biorxiv.org/content/' + row[1].strip('doi.org') + 'v1.full'\n",
    "    if source == 'biorxiv':\n",
    "        paper = scrape_biorxiv(paper)\n",
    "    else:\n",
    "        print('Error source not biorxiv. ({})'.format(source))\n",
    "    print(paper['title'],paper['id'])\n",
    "    '''\n",
    "    with open('data/{}/{}.json'.format(source,paper['id'])):\n",
    "        json.dump(paper, fp, indent=4)\n",
    "    '''\n",
    "    print('Paper ID: {} saved to data/{}'.format(paper['id'], source))\n",
    "    print('-'*60)\n",
    "    i += 1\n",
    "\n",
    "# if dataframe was changed, save it to csv\n",
    "if modified:\n",
    "    df_meta.to_csv('new_metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "273.594px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
