{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CORD19 Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import nltk\n",
    "# nltk.download('stopwords')\n",
    "# nltk.download('punkt')\n",
    "# nltk.download('averaged_perceptron_tagger')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import yaml\n",
    "import os\n",
    "import nltk\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import pandas as pd\n",
    "from nltk.corpus import stopwords \n",
    "#import plotly.graph_objects as go\n",
    "import networkx as nx"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\david\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:3: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "# import configurations\n",
    "with open('config.yaml','r') as ymlfile:\n",
    "    cfg = yaml.load(ymlfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## General Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "code_folding": [
     0,
     7,
     20,
     33
    ]
   },
   "outputs": [],
   "source": [
    "def get_papers(path):\n",
    "    # get list of papers .json from path\n",
    "    papers = []\n",
    "    for file_name in os.listdir(path):\n",
    "        papers.append(file_name)\n",
    "    return papers\n",
    "\n",
    "def extract_authors(authors_list):\n",
    "    '''\n",
    "        Function to extract authors metadata from list of authors \n",
    "    '''\n",
    "    authors = []\n",
    "    for curr_author in authors_list:\n",
    "        author = {}\n",
    "        author['first']  = curr_author['first']\n",
    "        author['middle'] = curr_author['middle']\n",
    "        author['last']   = curr_author['last']\n",
    "        authors.append(author)\n",
    "    return authors\n",
    "\n",
    "def extract_abstract(abstract):\n",
    "    # use regex to remove 'word count: 194', get word count ?\n",
    "    # use regex to remove Text word count: 5168', get text word count ?\n",
    "    # remove 1,2 digit numbers that don't have text attached ? \n",
    "    stop_sentences = ['All rights reserved.','No reuse allowed without permission.','Abstract','author/funder']\n",
    "    abstract_text = ''\n",
    "    for section in abstract:\n",
    "        abstract_text = abstract_text + ' ' + section['text']\n",
    "    abstract_text = abstract_text.strip(\" \")\n",
    "    for s in stop_sentences:\n",
    "        abstract_text = abstract_text.replace(s,\"\")\n",
    "    return abstract_text\n",
    "\n",
    "def extract_references(bib_entries):\n",
    "    refs = []\n",
    "    for r in bib_entries:\n",
    "        ref = {}\n",
    "        ref['id'] = bib_entries[r]['ref_id']\n",
    "        ref['title'] = bib_entries[r]['title']\n",
    "        ref['authors'] = bib_entries[r]['authors']\n",
    "        ref['year'] = bib_entries[r]['year']\n",
    "        refs.append(ref)\n",
    "    return refs\n",
    "\n",
    "def extract_paper_metadata(paper):\n",
    "    paper_metadata = {}\n",
    "    paper_metadata['id']       = paper['paper_id']\n",
    "    paper_metadata['title']    = paper['metadata']['title']\n",
    "    paper_metadata['authors']  = extract_authors(paper['authors'])\n",
    "    paper_metadata['abstract'] = extract_abstract(paper['abstract'])\n",
    "    paper_metadata['refs']     = extract_references(paper['bib_entries'])\n",
    "    return paper_metadata\n",
    "\n",
    "def get_paper_data(path, paper_id):\n",
    "    file_path = os.path.join(path, paper_id)\n",
    "    with open(file_path, 'r') as f:\n",
    "        paper_info = json.load(f)\n",
    "    return paper_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Objects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "class Author:\n",
    "    def __init__(self, firstname, middlename, lastname):\n",
    "        self.firstName = firstname\n",
    "        self.middleName = middlename\n",
    "        self.lastName = lastname\n",
    "    \n",
    "    def __str__(self):\n",
    "        return '{} {} {}'.format(self.firstName, self.middleName, self.lastName)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "class Paper:\n",
    "    def __init__(self, sha, title='', authors=None, date=None):\n",
    "        self.id = sha\n",
    "        self.title = title\n",
    "        self.authors = authors\n",
    "        self.date = date\n",
    "        self.url = ''\n",
    "    \n",
    "    def __str__(self):\n",
    "        s = ''\n",
    "        s += 'Paper ID: {}\\n'.format(self.id)\n",
    "        s += 'Title: {}\\n'.format(self.title)\n",
    "        if self.authors:\n",
    "            s += '# Authors: {}\\n'.format(len(self.authors))\n",
    "        else:\n",
    "            s += '# Authors: 0\\n'\n",
    "        s += 'Date: {}\\n'.format(self.date)\n",
    "        s += 'URL: {}'.format(self.url)\n",
    "        return s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\david\\OneDrive\\Bureau\\CORD-19-research-challenge\\\\2020-03-13\\biorxiv_medrxiv\\biorxiv_medrxiv\n"
     ]
    }
   ],
   "source": [
    "path = cfg['data-path'] + biorxiv\n",
    "print(path)\n",
    "#paper = json.loads(path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c630ebcdf30652f0422c3ec12a00b50241dc9bd9</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) as a SA...</td>\n",
       "      <td>10.1007/s00134-020-05985-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125455.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>2.002765e+09</td>\n",
       "      <td>#3252</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>53eccda7977a31e3d0f565c884da036b1e85438e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comparative genetic analysis of the novel coro...</td>\n",
       "      <td>10.1038/s41421-020-0147-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...</td>\n",
       "      <td>Cell Discovery</td>\n",
       "      <td>3.003431e+09</td>\n",
       "      <td>#1861</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>210a892deb1c61577f6fba58505fd65356ce6636</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Incubation Period and Other Epidemiological Ch...</td>\n",
       "      <td>10.3390/jcm9020538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The geographic spread of 2019 novel coronaviru...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...</td>\n",
       "      <td>Journal of Clinical Medicine</td>\n",
       "      <td>3.006065e+09</td>\n",
       "      <td>#1043</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>e3b40cc8e0e137c416b4a2273a4dca94ae8178cc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Characteristics of and Public Health Responses...</td>\n",
       "      <td>10.3390/jcm9020575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32093211.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, cases of unidentified pneumo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Deng, Sheng-Qun; Peng, Hong-Juan</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>1.776631e+08</td>\n",
       "      <td>#1999</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92c2c9839304b4f2bc1276d41b1aa885d8b364fd</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Imaging changes in severe COVID-19 pneumonia</td>\n",
       "      <td>10.1007/s00134-020-05976-w</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125453.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Wei</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>3.006643e+09</td>\n",
       "      <td>#3242</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        sha source_x  \\\n",
       "0  c630ebcdf30652f0422c3ec12a00b50241dc9bd9      CZI   \n",
       "1  53eccda7977a31e3d0f565c884da036b1e85438e      CZI   \n",
       "2  210a892deb1c61577f6fba58505fd65356ce6636      CZI   \n",
       "3  e3b40cc8e0e137c416b4a2273a4dca94ae8178cc      CZI   \n",
       "4  92c2c9839304b4f2bc1276d41b1aa885d8b364fd      CZI   \n",
       "\n",
       "                                               title  \\\n",
       "0  Angiotensin-converting enzyme 2 (ACE2) as a SA...   \n",
       "1  Comparative genetic analysis of the novel coro...   \n",
       "2  Incubation Period and Other Epidemiological Ch...   \n",
       "3  Characteristics of and Public Health Responses...   \n",
       "4       Imaging changes in severe COVID-19 pneumonia   \n",
       "\n",
       "                          doi pmcid   pubmed_id   license  \\\n",
       "0  10.1007/s00134-020-05985-9   NaN  32125455.0  cc-by-nc   \n",
       "1   10.1038/s41421-020-0147-1   NaN         NaN     cc-by   \n",
       "2          10.3390/jcm9020538   NaN         NaN     cc-by   \n",
       "3          10.3390/jcm9020575   NaN  32093211.0     cc-by   \n",
       "4  10.1007/s00134-020-05976-w   NaN  32125453.0  cc-by-nc   \n",
       "\n",
       "                                            abstract publish_time  \\\n",
       "0                                                NaN         2020   \n",
       "1                                                NaN         2020   \n",
       "2  The geographic spread of 2019 novel coronaviru...         2020   \n",
       "3  In December 2019, cases of unidentified pneumo...         2020   \n",
       "4                                                NaN         2020   \n",
       "\n",
       "                                             authors  \\\n",
       "0  Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...   \n",
       "1  Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...   \n",
       "2  Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...   \n",
       "3                   Deng, Sheng-Qun; Peng, Hong-Juan   \n",
       "4                                         Zhang, Wei   \n",
       "\n",
       "                        journal  Microsoft Academic Paper ID WHO #Covidence  \\\n",
       "0            Intensive Care Med                 2.002765e+09          #3252   \n",
       "1                Cell Discovery                 3.003431e+09          #1861   \n",
       "2  Journal of Clinical Medicine                 3.006065e+09          #1043   \n",
       "3                    J Clin Med                 1.776631e+08          #1999   \n",
       "4            Intensive Care Med                 3.006643e+09          #3242   \n",
       "\n",
       "  has_full_text  \n",
       "0          True  \n",
       "1          True  \n",
       "2          True  \n",
       "3          True  \n",
       "4         False  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = '2020-03-13/all_sources_metadata_2020-03-13.csv'\n",
    "df_meta = pd.read_csv(cfg['data-path'] + meta)\n",
    "df_meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c630ebcdf30652f0422c3ec12a00b50241dc9bd9</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) as a SA...</td>\n",
       "      <td>10.1007/s00134-020-05985-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125455.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>2.002765e+09</td>\n",
       "      <td>#3252</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>53eccda7977a31e3d0f565c884da036b1e85438e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comparative genetic analysis of the novel coro...</td>\n",
       "      <td>10.1038/s41421-020-0147-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...</td>\n",
       "      <td>Cell Discovery</td>\n",
       "      <td>3.003431e+09</td>\n",
       "      <td>#1861</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>210a892deb1c61577f6fba58505fd65356ce6636</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Incubation Period and Other Epidemiological Ch...</td>\n",
       "      <td>10.3390/jcm9020538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The geographic spread of 2019 novel coronaviru...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...</td>\n",
       "      <td>Journal of Clinical Medicine</td>\n",
       "      <td>3.006065e+09</td>\n",
       "      <td>#1043</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>e3b40cc8e0e137c416b4a2273a4dca94ae8178cc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Characteristics of and Public Health Responses...</td>\n",
       "      <td>10.3390/jcm9020575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32093211.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, cases of unidentified pneumo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Deng, Sheng-Qun; Peng, Hong-Juan</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>1.776631e+08</td>\n",
       "      <td>#1999</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0df0d5270a9399cf4e23c0cdd877a80616a9725e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>An updated estimation of the risk of transmiss...</td>\n",
       "      <td>10.1016/j.idm.2020.02.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The basic reproduction number of an infectious...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#729</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>f24242580be243d5fc3f432915d86af6854bb8b7</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Real-time forecasts of the 2019-nCoV epidemic ...</td>\n",
       "      <td>10.1016/j.idm.2020.02.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The initial cluster of severe pneumonia cases ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#865</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19 outbreak on the Diamond Princess crui...</td>\n",
       "      <td>10.1093/jtm/taaa030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Cruise ships carry a large number of people in...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.</td>\n",
       "      <td>Journal of Travel Medicine</td>\n",
       "      <td>3.006304e+09</td>\n",
       "      <td>#2926</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>e9239100c5493ea914dc23c3d7a262f4326022ac</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Distinct Roles for Sialoside and Protein Recep...</td>\n",
       "      <td>10.1128/mBio.02764-19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Coronaviruses (CoVs) are common human and anim...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Qing, Enya; Hantak, Michael; Perlman, Stanley;...</td>\n",
       "      <td>mBio</td>\n",
       "      <td>3.005811e+09</td>\n",
       "      <td>#2427</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>469ed0f00c09e2637351c9735c306f27acf3aace</td>\n",
       "      <td>CZI</td>\n",
       "      <td>First two months of the 2019 Coronavirus Disea...</td>\n",
       "      <td>10.1186/s41256-020-00137-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Similar to outbreaks of many other infectious ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Xinguang; Yu, Bin</td>\n",
       "      <td>Global Health Research and Policy</td>\n",
       "      <td>3.006646e+09</td>\n",
       "      <td>#5595</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4e550e034ccca6fa2a91e481ddba24db67bc9ae5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness of airport screening at detectin...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.5.2000080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>We simulated 100 2019-nCoV infected travellers...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Quilty, Billy J; Clifford, Sam; group2, CMMID ...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005152e+09</td>\n",
       "      <td>#682</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4bbb0c59babc718f67953fae032dad6ae0d7aeb1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Genome Detective Coronavirus Typing Tool for r...</td>\n",
       "      <td>10.1093/bioinformatics/btaa145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32108862.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>SUMMARY: Genome Detective is a web-based, user...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>3.003549e+09</td>\n",
       "      <td>#2776</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>c821803c55c6aad89b6d0c1d3ba252051e464017</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Case of the Index Patient Who Caused Tertiary ...</td>\n",
       "      <td>10.3346/jkms.2020.35.e79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Since mid-December of 2019, coronavirus diseas...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...</td>\n",
       "      <td>Journal of Korean Medical Science</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#3770</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>c3bee2a4caca614b34f92c17b643b854dcdab28d</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Emergence of Novel Coronavirus 2019-nCoV: Need...</td>\n",
       "      <td>10.3390/pathogens9020148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Novel Coronavirus (2019-nCoV) is an emerging p...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...</td>\n",
       "      <td>Pathogens</td>\n",
       "      <td>3.003886e+09</td>\n",
       "      <td>#1502</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>715842fa536064980818ad7e31ce511272a4b6bc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The coronavirus 2019-nCoV epidemic: Is hindsig...</td>\n",
       "      <td>10.1016/j.eclinm.2020.100289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...</td>\n",
       "      <td>EClinicalMedicine</td>\n",
       "      <td>3.004913e+09</td>\n",
       "      <td>#3333</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Nonstructural proteins NS7b and NS8 are likely...</td>\n",
       "      <td>10.1016/j.meegid.2020.104272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The seventh novel human infecting Betacoronavi...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro</td>\n",
       "      <td>Infection, Genetics and Evolution</td>\n",
       "      <td>2.286137e+09</td>\n",
       "      <td>#4581</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>06c89f69aa7b5f9648d2c1543b8246fe9c3610cf</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Pathogenicity and Transmissibility of 2019-nCo...</td>\n",
       "      <td>10.1016/j.micinf.2020.01.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>A zoonotic coronavirus, labeled as 2019-nCoV b...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Jieliang</td>\n",
       "      <td>Microbes and Infection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#236</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>acb678bdd7634055de18d0b89bb6a4890e6a0306</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Remdesivir and chloroquine effectively inhibit...</td>\n",
       "      <td>10.1038/s41422-020-0282-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...</td>\n",
       "      <td>Cell Research</td>\n",
       "      <td>3.005213e+09</td>\n",
       "      <td>#205</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>47772f3e98d8c61bb9782531a0338ba85f27bf3f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential for global spread of a novel coronav...</td>\n",
       "      <td>10.1093/jtm/taaa011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...</td>\n",
       "      <td>Journal of travel medicine</td>\n",
       "      <td>3.004048e+09</td>\n",
       "      <td>#102</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Severe acute respiratory symptoms and suspecte...</td>\n",
       "      <td>10.12809/hkmj208375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Hon, K. L.; Leung, K. K. Y.</td>\n",
       "      <td>Hong Kong Medical Journal</td>\n",
       "      <td>2.029408e+09</td>\n",
       "      <td>#3395</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4ff89a71126d2932544a8337ba28787fde5f02a8</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Statistics-Based Predictions of Coronavirus Ep...</td>\n",
       "      <td>10.20535/ibb.2020.4.1.195074</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Information about the open-access article 'Sta...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Nesteruk, Igor</td>\n",
       "      <td>Innovative Biosystems and Bioengineering</td>\n",
       "      <td>3.006305e+09</td>\n",
       "      <td>#1218</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ac37ba61c91bb6939a507dbf4efe2119ddafeb9c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Early transmission patterns of coronavirus dis...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.8.2000097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#2582</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0</td>\n",
       "      <td>CZI</td>\n",
       "      <td>HRCT Imaging Features in Representative Import...</td>\n",
       "      <td>10.1093/pcmedi/pbaa004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>With the spread of novel coronavirus (2019-nCo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...</td>\n",
       "      <td>Precision Clinical Medicine</td>\n",
       "      <td>3.006468e+09</td>\n",
       "      <td>#781</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>c9fee561c2a3834645dbb61dc4ae6448051da492</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comprehensive Genomic Characterization Analysi...</td>\n",
       "      <td>10.3389/fmicb.2019.03036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Porcine delta coronavirus (PDCoV) is a novel e...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...</td>\n",
       "      <td>Frontiers in Microbiology</td>\n",
       "      <td>3.003968e+09</td>\n",
       "      <td>#5462</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>655537fc8cc52bccf43cf7189ab060d3097caa7a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential Factors Influencing Repeated SARS Ou...</td>\n",
       "      <td>10.3390/ijerph17051633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Within last 17 years two widespread epidemics ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....</td>\n",
       "      <td>International Journal of Environmental Researc...</td>\n",
       "      <td>2.615949e+09</td>\n",
       "      <td>#3296</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>bb7cd48a85b72f8d73e57d655c6d8a034fce3778</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Puzzle of highly pathogenic human coronaviruse...</td>\n",
       "      <td>10.1007/s13238-020-00693-y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32088858.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Li, Jing; Liu, Wenjun</td>\n",
       "      <td>Protein Cell</td>\n",
       "      <td>2.766462e+09</td>\n",
       "      <td>#1815</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>6a93283b499ae5bc6aaf29f14e701dc8f25138ea</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Critical care response to a hospital outbreak ...</td>\n",
       "      <td>10.1186/s13054-020-2786-x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liu, Yong; Li, Jinxiu; Feng, Yongwen</td>\n",
       "      <td>Critical Care</td>\n",
       "      <td>2.537201e+09</td>\n",
       "      <td>#1225</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>1d7f8850c5244fdc9b387038e7eeae9bcbbde6d2</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Optimization Method for Forecasting Confirmed ...</td>\n",
       "      <td>10.3390/jcm9030674</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32131537.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, a novel coronavirus, called ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Al-Qaness, Mohammed A. A.; Ewees, Ahmed A.; Fa...</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>3.006672e+09</td>\n",
       "      <td>#4638</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>f294f0df7468a8ac9e27776cc15fa20297a9f040</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Systematic Comparison of Two Animal-to-Human T...</td>\n",
       "      <td>10.3390/v12020244</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>After the outbreak of the severe acute respira...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Xu, Jiabao; Zhao, Shizhe; Teng, Tieshan; Abdal...</td>\n",
       "      <td>Viruses</td>\n",
       "      <td>2.163319e+09</td>\n",
       "      <td>#1449</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>60e51d3d13b027492b9f3b8693c58dd8b5385ada</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus infection during the 2019–20...</td>\n",
       "      <td>10.1007/s00134-020-05954-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Novel coronavirus infection special intensive ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liao, Xuelian; Wang, Bo; Kang, Yan</td>\n",
       "      <td>Intensive Care Medicine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#305</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>253cfd411f93ef88f702accd0fa195f24d1d2925</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Return of the Coronavirus: 2019-nCoV</td>\n",
       "      <td>10.3390/v12020135</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The emergence of a novel coronavirus (2019-nCo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Gralinski, E. Lisa; Menachery, D. Vineet</td>\n",
       "      <td>Viruses</td>\n",
       "      <td>3.002812e+09</td>\n",
       "      <td>#61</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29465</th>\n",
       "      <td>a0ecc9b46addbc1faab491434b5331702f7883b8</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20031849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29466</th>\n",
       "      <td>b4edd81e0d2770b0d6e0e3c93c7fffc43e3caa80</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20031872</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29467</th>\n",
       "      <td>a42ff48d50aa3a0ebfe840d46ed2204c49955442</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20031898</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29468</th>\n",
       "      <td>5129bb31949c66ebd544658c531004046d4102da</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20031906</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29469</th>\n",
       "      <td>e05c39ce913edc81f93c9247f9714d6cddd30073</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20031971</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29471</th>\n",
       "      <td>93d8f1552a81587e2f88e38580cbe4753b9934f1</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20032011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29472</th>\n",
       "      <td>dd2cd004a197872cfce1c31a9685d6972f838efa</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20029793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29473</th>\n",
       "      <td>e53f5a6c4cbac340f0d97a4ba3289c30b7222dd5</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29474</th>\n",
       "      <td>fd847a97e6fce134345b25ae9240c4ebb680cca5</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29475</th>\n",
       "      <td>610549c879c62efc42f2ad32f63c65b25d8931c7</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29477</th>\n",
       "      <td>fd96fc1859201e13e35526c188e022fa6e335327</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29478</th>\n",
       "      <td>26e75d3c815aae7fd9b094c3e5c74d3f7132ca13</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29480</th>\n",
       "      <td>ea17aad7692bb5040f3b2ba1d01b8249e4969469</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032334</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29481</th>\n",
       "      <td>6d3b3f4ab80a61c45f82c61c6c756cfc6ddf4bf2</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29483</th>\n",
       "      <td>c1a08ce8f2c5d5311b19822c3df5aa9d25c4d93c</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20031393</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29484</th>\n",
       "      <td>c6012a6141f5ad05ac983aa9f1a7c57938363b16</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20031575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29485</th>\n",
       "      <td>74bee8bf3229f4c28991a7231c02d911d24770e9</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032326</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29486</th>\n",
       "      <td>ec90b0b861d31f910433d388370344024ca7c35d</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032524</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29487</th>\n",
       "      <td>970a28c4a3c1fbcb5322772956955b8c4a3bb257</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29488</th>\n",
       "      <td>abe590b95aa51e7309844156acdae4085870ea33</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29489</th>\n",
       "      <td>9a077c5ff1dcbb9104feeaa832e73ccdcf5fdcc9</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032672</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29490</th>\n",
       "      <td>6cf87a546884756094da0d300e85a061c2cc43ea</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20029710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29491</th>\n",
       "      <td>43f760b5cb9c3b430c6d0b31fa90d42e71a6429c</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20030643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29492</th>\n",
       "      <td>901572f1595cb6ea733858d3fc15b8ea46054061</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20031229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29493</th>\n",
       "      <td>4c84d1153daa335f5ff6d61db665024b32ac8195</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20031658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29494</th>\n",
       "      <td>e5f19b6daf956e815c779228cc0cad1293d65bbb</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29495</th>\n",
       "      <td>c42a617a00afe6a36bde0a8e3638e0f55bfee4f7</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032847</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29496</th>\n",
       "      <td>c4ce14ce42fa4360dfe3515ec9d1584847381c27</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032854</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29498</th>\n",
       "      <td>ca88735399ff43d0e673876200655099f06f5567</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20033183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29499</th>\n",
       "      <td>0b48e1bfcdff9a42c88cc80a98661feb2703390d</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20033217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13219 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sha source_x  \\\n",
       "0      c630ebcdf30652f0422c3ec12a00b50241dc9bd9      CZI   \n",
       "1      53eccda7977a31e3d0f565c884da036b1e85438e      CZI   \n",
       "2      210a892deb1c61577f6fba58505fd65356ce6636      CZI   \n",
       "3      e3b40cc8e0e137c416b4a2273a4dca94ae8178cc      CZI   \n",
       "5      0df0d5270a9399cf4e23c0cdd877a80616a9725e      CZI   \n",
       "6      f24242580be243d5fc3f432915d86af6854bb8b7      CZI   \n",
       "8      e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1      CZI   \n",
       "9      e9239100c5493ea914dc23c3d7a262f4326022ac      CZI   \n",
       "10     469ed0f00c09e2637351c9735c306f27acf3aace      CZI   \n",
       "11     4e550e034ccca6fa2a91e481ddba24db67bc9ae5      CZI   \n",
       "12     4bbb0c59babc718f67953fae032dad6ae0d7aeb1      CZI   \n",
       "13     c821803c55c6aad89b6d0c1d3ba252051e464017      CZI   \n",
       "14     c3bee2a4caca614b34f92c17b643b854dcdab28d      CZI   \n",
       "15     715842fa536064980818ad7e31ce511272a4b6bc      CZI   \n",
       "16     601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f      CZI   \n",
       "17     06c89f69aa7b5f9648d2c1543b8246fe9c3610cf      CZI   \n",
       "18     acb678bdd7634055de18d0b89bb6a4890e6a0306      CZI   \n",
       "19     47772f3e98d8c61bb9782531a0338ba85f27bf3f      CZI   \n",
       "20     544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5      CZI   \n",
       "21     4ff89a71126d2932544a8337ba28787fde5f02a8      CZI   \n",
       "22     ac37ba61c91bb6939a507dbf4efe2119ddafeb9c      CZI   \n",
       "23     6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0      CZI   \n",
       "24     c9fee561c2a3834645dbb61dc4ae6448051da492      CZI   \n",
       "25     655537fc8cc52bccf43cf7189ab060d3097caa7a      CZI   \n",
       "26     bb7cd48a85b72f8d73e57d655c6d8a034fce3778      CZI   \n",
       "28     6a93283b499ae5bc6aaf29f14e701dc8f25138ea      CZI   \n",
       "30     1d7f8850c5244fdc9b387038e7eeae9bcbbde6d2      CZI   \n",
       "31     f294f0df7468a8ac9e27776cc15fa20297a9f040      CZI   \n",
       "32     60e51d3d13b027492b9f3b8693c58dd8b5385ada      CZI   \n",
       "33     253cfd411f93ef88f702accd0fa195f24d1d2925      CZI   \n",
       "...                                         ...      ...   \n",
       "29465  a0ecc9b46addbc1faab491434b5331702f7883b8  medrxiv   \n",
       "29466  b4edd81e0d2770b0d6e0e3c93c7fffc43e3caa80  medrxiv   \n",
       "29467  a42ff48d50aa3a0ebfe840d46ed2204c49955442  medrxiv   \n",
       "29468  5129bb31949c66ebd544658c531004046d4102da  medrxiv   \n",
       "29469  e05c39ce913edc81f93c9247f9714d6cddd30073  medrxiv   \n",
       "29471  93d8f1552a81587e2f88e38580cbe4753b9934f1  medrxiv   \n",
       "29472  dd2cd004a197872cfce1c31a9685d6972f838efa  medrxiv   \n",
       "29473  e53f5a6c4cbac340f0d97a4ba3289c30b7222dd5  medrxiv   \n",
       "29474  fd847a97e6fce134345b25ae9240c4ebb680cca5  medrxiv   \n",
       "29475  610549c879c62efc42f2ad32f63c65b25d8931c7  medrxiv   \n",
       "29477  fd96fc1859201e13e35526c188e022fa6e335327  medrxiv   \n",
       "29478  26e75d3c815aae7fd9b094c3e5c74d3f7132ca13  medrxiv   \n",
       "29480  ea17aad7692bb5040f3b2ba1d01b8249e4969469  medrxiv   \n",
       "29481  6d3b3f4ab80a61c45f82c61c6c756cfc6ddf4bf2  medrxiv   \n",
       "29483  c1a08ce8f2c5d5311b19822c3df5aa9d25c4d93c  medrxiv   \n",
       "29484  c6012a6141f5ad05ac983aa9f1a7c57938363b16  medrxiv   \n",
       "29485  74bee8bf3229f4c28991a7231c02d911d24770e9  medrxiv   \n",
       "29486  ec90b0b861d31f910433d388370344024ca7c35d  medrxiv   \n",
       "29487  970a28c4a3c1fbcb5322772956955b8c4a3bb257  medrxiv   \n",
       "29488  abe590b95aa51e7309844156acdae4085870ea33  medrxiv   \n",
       "29489  9a077c5ff1dcbb9104feeaa832e73ccdcf5fdcc9  medrxiv   \n",
       "29490  6cf87a546884756094da0d300e85a061c2cc43ea  medrxiv   \n",
       "29491  43f760b5cb9c3b430c6d0b31fa90d42e71a6429c  medrxiv   \n",
       "29492  901572f1595cb6ea733858d3fc15b8ea46054061  medrxiv   \n",
       "29493  4c84d1153daa335f5ff6d61db665024b32ac8195  medrxiv   \n",
       "29494  e5f19b6daf956e815c779228cc0cad1293d65bbb  medrxiv   \n",
       "29495  c42a617a00afe6a36bde0a8e3638e0f55bfee4f7  medrxiv   \n",
       "29496  c4ce14ce42fa4360dfe3515ec9d1584847381c27  medrxiv   \n",
       "29498  ca88735399ff43d0e673876200655099f06f5567  medrxiv   \n",
       "29499  0b48e1bfcdff9a42c88cc80a98661feb2703390d  medrxiv   \n",
       "\n",
       "                                                   title  \\\n",
       "0      Angiotensin-converting enzyme 2 (ACE2) as a SA...   \n",
       "1      Comparative genetic analysis of the novel coro...   \n",
       "2      Incubation Period and Other Epidemiological Ch...   \n",
       "3      Characteristics of and Public Health Responses...   \n",
       "5      An updated estimation of the risk of transmiss...   \n",
       "6      Real-time forecasts of the 2019-nCoV epidemic ...   \n",
       "8      COVID-19 outbreak on the Diamond Princess crui...   \n",
       "9      Distinct Roles for Sialoside and Protein Recep...   \n",
       "10     First two months of the 2019 Coronavirus Disea...   \n",
       "11     Effectiveness of airport screening at detectin...   \n",
       "12     Genome Detective Coronavirus Typing Tool for r...   \n",
       "13     Case of the Index Patient Who Caused Tertiary ...   \n",
       "14     Emergence of Novel Coronavirus 2019-nCoV: Need...   \n",
       "15     The coronavirus 2019-nCoV epidemic: Is hindsig...   \n",
       "16     Nonstructural proteins NS7b and NS8 are likely...   \n",
       "17     Pathogenicity and Transmissibility of 2019-nCo...   \n",
       "18     Remdesivir and chloroquine effectively inhibit...   \n",
       "19     Potential for global spread of a novel coronav...   \n",
       "20     Severe acute respiratory symptoms and suspecte...   \n",
       "21     Statistics-Based Predictions of Coronavirus Ep...   \n",
       "22     Early transmission patterns of coronavirus dis...   \n",
       "23     HRCT Imaging Features in Representative Import...   \n",
       "24     Comprehensive Genomic Characterization Analysi...   \n",
       "25     Potential Factors Influencing Repeated SARS Ou...   \n",
       "26     Puzzle of highly pathogenic human coronaviruse...   \n",
       "28     Critical care response to a hospital outbreak ...   \n",
       "30     Optimization Method for Forecasting Confirmed ...   \n",
       "31     Systematic Comparison of Two Animal-to-Human T...   \n",
       "32     Novel coronavirus infection during the 2019–20...   \n",
       "33                  Return of the Coronavirus: 2019-nCoV   \n",
       "...                                                  ...   \n",
       "29465                                                NaN   \n",
       "29466                                                NaN   \n",
       "29467                                                NaN   \n",
       "29468                                                NaN   \n",
       "29469                                                NaN   \n",
       "29471                                                NaN   \n",
       "29472                                                NaN   \n",
       "29473                                                NaN   \n",
       "29474                                                NaN   \n",
       "29475                                                NaN   \n",
       "29477                                                NaN   \n",
       "29478                                                NaN   \n",
       "29480                                                NaN   \n",
       "29481                                                NaN   \n",
       "29483                                                NaN   \n",
       "29484                                                NaN   \n",
       "29485                                                NaN   \n",
       "29486                                                NaN   \n",
       "29487                                                NaN   \n",
       "29488                                                NaN   \n",
       "29489                                                NaN   \n",
       "29490                                                NaN   \n",
       "29491                                                NaN   \n",
       "29492                                                NaN   \n",
       "29493                                                NaN   \n",
       "29494                                                NaN   \n",
       "29495                                                NaN   \n",
       "29496                                                NaN   \n",
       "29498                                                NaN   \n",
       "29499                                                NaN   \n",
       "\n",
       "                                          doi pmcid   pubmed_id  \\\n",
       "0                  10.1007/s00134-020-05985-9   NaN  32125455.0   \n",
       "1                   10.1038/s41421-020-0147-1   NaN         NaN   \n",
       "2                          10.3390/jcm9020538   NaN         NaN   \n",
       "3                          10.3390/jcm9020575   NaN  32093211.0   \n",
       "5                   10.1016/j.idm.2020.02.001   NaN         NaN   \n",
       "6                   10.1016/j.idm.2020.02.002   NaN         NaN   \n",
       "8                         10.1093/jtm/taaa030   NaN         NaN   \n",
       "9                       10.1128/mBio.02764-19   NaN         NaN   \n",
       "10                 10.1186/s41256-020-00137-4   NaN         NaN   \n",
       "11     10.2807/1560-7917.ES.2020.25.5.2000080   NaN         NaN   \n",
       "12             10.1093/bioinformatics/btaa145   NaN  32108862.0   \n",
       "13                   10.3346/jkms.2020.35.e79   NaN         NaN   \n",
       "14                   10.3390/pathogens9020148   NaN         NaN   \n",
       "15               10.1016/j.eclinm.2020.100289   NaN         NaN   \n",
       "16               10.1016/j.meegid.2020.104272   NaN         NaN   \n",
       "17               10.1016/j.micinf.2020.01.004   NaN         NaN   \n",
       "18                  10.1038/s41422-020-0282-0   NaN         NaN   \n",
       "19                        10.1093/jtm/taaa011   NaN         NaN   \n",
       "20                        10.12809/hkmj208375   NaN         NaN   \n",
       "21               10.20535/ibb.2020.4.1.195074   NaN         NaN   \n",
       "22     10.2807/1560-7917.ES.2020.25.8.2000097   NaN         NaN   \n",
       "23                     10.1093/pcmedi/pbaa004   NaN         NaN   \n",
       "24                   10.3389/fmicb.2019.03036   NaN         NaN   \n",
       "25                     10.3390/ijerph17051633   NaN         NaN   \n",
       "26                 10.1007/s13238-020-00693-y   NaN  32088858.0   \n",
       "28                  10.1186/s13054-020-2786-x   NaN         NaN   \n",
       "30                         10.3390/jcm9030674   NaN  32131537.0   \n",
       "31                          10.3390/v12020244   NaN         NaN   \n",
       "32                 10.1007/s00134-020-05954-2   NaN         NaN   \n",
       "33                          10.3390/v12020135   NaN         NaN   \n",
       "...                                       ...   ...         ...   \n",
       "29465     doi.org/10.1101/2020.03.05.20031849   NaN         NaN   \n",
       "29466     doi.org/10.1101/2020.03.05.20031872   NaN         NaN   \n",
       "29467     doi.org/10.1101/2020.03.05.20031898   NaN         NaN   \n",
       "29468     doi.org/10.1101/2020.03.05.20031906   NaN         NaN   \n",
       "29469     doi.org/10.1101/2020.03.05.20031971   NaN         NaN   \n",
       "29471     doi.org/10.1101/2020.03.05.20032011   NaN         NaN   \n",
       "29472     doi.org/10.1101/2020.03.06.20029793   NaN         NaN   \n",
       "29473     doi.org/10.1101/2020.03.06.20031278   NaN         NaN   \n",
       "29474     doi.org/10.1101/2020.03.06.20031377   NaN         NaN   \n",
       "29475     doi.org/10.1101/2020.03.06.20031856   NaN         NaN   \n",
       "29477     doi.org/10.1101/2020.03.06.20031955   NaN         NaN   \n",
       "29478     doi.org/10.1101/2020.03.06.20032144   NaN         NaN   \n",
       "29480     doi.org/10.1101/2020.03.06.20032334   NaN         NaN   \n",
       "29481     doi.org/10.1101/2020.03.06.20032417   NaN         NaN   \n",
       "29483     doi.org/10.1101/2020.03.07.20031393   NaN         NaN   \n",
       "29484     doi.org/10.1101/2020.03.07.20031575   NaN         NaN   \n",
       "29485     doi.org/10.1101/2020.03.07.20032326   NaN         NaN   \n",
       "29486     doi.org/10.1101/2020.03.07.20032524   NaN         NaN   \n",
       "29487     doi.org/10.1101/2020.03.07.20032573   NaN         NaN   \n",
       "29488     doi.org/10.1101/2020.03.07.20032599   NaN         NaN   \n",
       "29489     doi.org/10.1101/2020.03.07.20032672   NaN         NaN   \n",
       "29490     doi.org/10.1101/2020.03.08.20029710   NaN         NaN   \n",
       "29491     doi.org/10.1101/2020.03.08.20030643   NaN         NaN   \n",
       "29492     doi.org/10.1101/2020.03.08.20031229   NaN         NaN   \n",
       "29493     doi.org/10.1101/2020.03.08.20031658   NaN         NaN   \n",
       "29494     doi.org/10.1101/2020.03.08.20032821   NaN         NaN   \n",
       "29495     doi.org/10.1101/2020.03.08.20032847   NaN         NaN   \n",
       "29496     doi.org/10.1101/2020.03.08.20032854   NaN         NaN   \n",
       "29498     doi.org/10.1101/2020.03.09.20033183   NaN         NaN   \n",
       "29499     doi.org/10.1101/2020.03.09.20033217   NaN         NaN   \n",
       "\n",
       "                                               license  \\\n",
       "0                                             cc-by-nc   \n",
       "1                                                cc-by   \n",
       "2                                                cc-by   \n",
       "3                                                cc-by   \n",
       "5                                          cc-by-nc-nd   \n",
       "6                                          cc-by-nc-nd   \n",
       "8                                             cc-by-nc   \n",
       "9                                                cc-by   \n",
       "10                                               cc-by   \n",
       "11                                               cc-by   \n",
       "12                                            cc-by-nc   \n",
       "13                                            cc-by-nc   \n",
       "14                                               cc-by   \n",
       "15                                         cc-by-nc-nd   \n",
       "16                                         cc-by-nc-nd   \n",
       "17                                         cc-by-nc-nd   \n",
       "18                                               cc-by   \n",
       "19                                            cc-by-nc   \n",
       "20                                         cc-by-nc-nd   \n",
       "21                                               cc-by   \n",
       "22                                               cc-by   \n",
       "23                                            cc-by-nc   \n",
       "24                                               cc-by   \n",
       "25                                               cc-by   \n",
       "26                                               cc-by   \n",
       "28                                               cc-by   \n",
       "30                                               cc-by   \n",
       "31                                               cc-by   \n",
       "32                                            cc-by-nc   \n",
       "33                                               cc-by   \n",
       "...                                                ...   \n",
       "29465  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29466  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29467  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29468  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29469  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29471  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29472  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29473  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29474  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29475  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29477  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29478  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29480  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29481  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29483  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29484  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29485  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29486  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29487  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29488  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29489  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29490  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29491  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29492  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29493  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29494  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29495  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29496  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29498  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "29499  See https://www.medrxiv.org/submit-a-manuscript   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "0                                                    NaN         2020   \n",
       "1                                                    NaN         2020   \n",
       "2      The geographic spread of 2019 novel coronaviru...         2020   \n",
       "3      In December 2019, cases of unidentified pneumo...         2020   \n",
       "5      The basic reproduction number of an infectious...         2020   \n",
       "6      The initial cluster of severe pneumonia cases ...         2020   \n",
       "8      Cruise ships carry a large number of people in...         2020   \n",
       "9      Coronaviruses (CoVs) are common human and anim...         2020   \n",
       "10     Similar to outbreaks of many other infectious ...         2020   \n",
       "11     We simulated 100 2019-nCoV infected travellers...         2020   \n",
       "12     SUMMARY: Genome Detective is a web-based, user...         2020   \n",
       "13     Since mid-December of 2019, coronavirus diseas...         2020   \n",
       "14     Novel Coronavirus (2019-nCoV) is an emerging p...         2020   \n",
       "15                                                   NaN         2020   \n",
       "16     The seventh novel human infecting Betacoronavi...         2020   \n",
       "17     A zoonotic coronavirus, labeled as 2019-nCoV b...         2020   \n",
       "18                                                   NaN         2020   \n",
       "19                                                   NaN         2020   \n",
       "20                                                   NaN         2020   \n",
       "21     Information about the open-access article 'Sta...         2020   \n",
       "22                                                   NaN         2020   \n",
       "23     With the spread of novel coronavirus (2019-nCo...         2020   \n",
       "24     Porcine delta coronavirus (PDCoV) is a novel e...         2020   \n",
       "25     Within last 17 years two widespread epidemics ...         2020   \n",
       "26                                                   NaN         2020   \n",
       "28                                                   NaN         2020   \n",
       "30     In December 2019, a novel coronavirus, called ...         2020   \n",
       "31     After the outbreak of the severe acute respira...         2020   \n",
       "32     Novel coronavirus infection special intensive ...         2020   \n",
       "33     The emergence of a novel coronavirus (2019-nCo...         2020   \n",
       "...                                                  ...          ...   \n",
       "29465                                                NaN          NaN   \n",
       "29466                                                NaN          NaN   \n",
       "29467                                                NaN          NaN   \n",
       "29468                                                NaN          NaN   \n",
       "29469                                                NaN          NaN   \n",
       "29471                                                NaN          NaN   \n",
       "29472                                                NaN          NaN   \n",
       "29473                                                NaN          NaN   \n",
       "29474                                                NaN          NaN   \n",
       "29475                                                NaN          NaN   \n",
       "29477                                                NaN          NaN   \n",
       "29478                                                NaN          NaN   \n",
       "29480                                                NaN          NaN   \n",
       "29481                                                NaN          NaN   \n",
       "29483                                                NaN          NaN   \n",
       "29484                                                NaN          NaN   \n",
       "29485                                                NaN          NaN   \n",
       "29486                                                NaN          NaN   \n",
       "29487                                                NaN          NaN   \n",
       "29488                                                NaN          NaN   \n",
       "29489                                                NaN          NaN   \n",
       "29490                                                NaN          NaN   \n",
       "29491                                                NaN          NaN   \n",
       "29492                                                NaN          NaN   \n",
       "29493                                                NaN          NaN   \n",
       "29494                                                NaN          NaN   \n",
       "29495                                                NaN          NaN   \n",
       "29496                                                NaN          NaN   \n",
       "29498                                                NaN          NaN   \n",
       "29499                                                NaN          NaN   \n",
       "\n",
       "                                                 authors  \\\n",
       "0      Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...   \n",
       "1      Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...   \n",
       "2      Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...   \n",
       "3                       Deng, Sheng-Qun; Peng, Hong-Juan   \n",
       "5      Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...   \n",
       "6      Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...   \n",
       "8              Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.   \n",
       "9      Qing, Enya; Hantak, Michael; Perlman, Stanley;...   \n",
       "10                               Chen, Xinguang; Yu, Bin   \n",
       "11     Quilty, Billy J; Clifford, Sam; group2, CMMID ...   \n",
       "12     Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...   \n",
       "13     Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...   \n",
       "14     Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...   \n",
       "15     Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...   \n",
       "16       Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro   \n",
       "17                                        Chen, Jieliang   \n",
       "18     Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...   \n",
       "19     Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...   \n",
       "20                           Hon, K. L.; Leung, K. K. Y.   \n",
       "21                                        Nesteruk, Igor   \n",
       "22     Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...   \n",
       "23     Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...   \n",
       "24     Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...   \n",
       "25     Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....   \n",
       "26                                 Li, Jing; Liu, Wenjun   \n",
       "28                  Liu, Yong; Li, Jinxiu; Feng, Yongwen   \n",
       "30     Al-Qaness, Mohammed A. A.; Ewees, Ahmed A.; Fa...   \n",
       "31     Xu, Jiabao; Zhao, Shizhe; Teng, Tieshan; Abdal...   \n",
       "32                    Liao, Xuelian; Wang, Bo; Kang, Yan   \n",
       "33              Gralinski, E. Lisa; Menachery, D. Vineet   \n",
       "...                                                  ...   \n",
       "29465                                                NaN   \n",
       "29466                                                NaN   \n",
       "29467                                                NaN   \n",
       "29468                                                NaN   \n",
       "29469                                                NaN   \n",
       "29471                                                NaN   \n",
       "29472                                                NaN   \n",
       "29473                                                NaN   \n",
       "29474                                                NaN   \n",
       "29475                                                NaN   \n",
       "29477                                                NaN   \n",
       "29478                                                NaN   \n",
       "29480                                                NaN   \n",
       "29481                                                NaN   \n",
       "29483                                                NaN   \n",
       "29484                                                NaN   \n",
       "29485                                                NaN   \n",
       "29486                                                NaN   \n",
       "29487                                                NaN   \n",
       "29488                                                NaN   \n",
       "29489                                                NaN   \n",
       "29490                                                NaN   \n",
       "29491                                                NaN   \n",
       "29492                                                NaN   \n",
       "29493                                                NaN   \n",
       "29494                                                NaN   \n",
       "29495                                                NaN   \n",
       "29496                                                NaN   \n",
       "29498                                                NaN   \n",
       "29499                                                NaN   \n",
       "\n",
       "                                                 journal  \\\n",
       "0                                     Intensive Care Med   \n",
       "1                                         Cell Discovery   \n",
       "2                           Journal of Clinical Medicine   \n",
       "3                                             J Clin Med   \n",
       "5                           Infectious Disease Modelling   \n",
       "6                           Infectious Disease Modelling   \n",
       "8                             Journal of Travel Medicine   \n",
       "9                                                   mBio   \n",
       "10                     Global Health Research and Policy   \n",
       "11                                      Eurosurveillance   \n",
       "12                      Bioinformatics (Oxford, England)   \n",
       "13                     Journal of Korean Medical Science   \n",
       "14                                             Pathogens   \n",
       "15                                     EClinicalMedicine   \n",
       "16                     Infection, Genetics and Evolution   \n",
       "17                                Microbes and Infection   \n",
       "18                                         Cell Research   \n",
       "19                            Journal of travel medicine   \n",
       "20                             Hong Kong Medical Journal   \n",
       "21              Innovative Biosystems and Bioengineering   \n",
       "22                                      Eurosurveillance   \n",
       "23                           Precision Clinical Medicine   \n",
       "24                             Frontiers in Microbiology   \n",
       "25     International Journal of Environmental Researc...   \n",
       "26                                          Protein Cell   \n",
       "28                                         Critical Care   \n",
       "30                                            J Clin Med   \n",
       "31                                               Viruses   \n",
       "32                               Intensive Care Medicine   \n",
       "33                                               Viruses   \n",
       "...                                                  ...   \n",
       "29465                                                NaN   \n",
       "29466                                                NaN   \n",
       "29467                                                NaN   \n",
       "29468                                                NaN   \n",
       "29469                                                NaN   \n",
       "29471                                                NaN   \n",
       "29472                                                NaN   \n",
       "29473                                                NaN   \n",
       "29474                                                NaN   \n",
       "29475                                                NaN   \n",
       "29477                                                NaN   \n",
       "29478                                                NaN   \n",
       "29480                                                NaN   \n",
       "29481                                                NaN   \n",
       "29483                                                NaN   \n",
       "29484                                                NaN   \n",
       "29485                                                NaN   \n",
       "29486                                                NaN   \n",
       "29487                                                NaN   \n",
       "29488                                                NaN   \n",
       "29489                                                NaN   \n",
       "29490                                                NaN   \n",
       "29491                                                NaN   \n",
       "29492                                                NaN   \n",
       "29493                                                NaN   \n",
       "29494                                                NaN   \n",
       "29495                                                NaN   \n",
       "29496                                                NaN   \n",
       "29498                                                NaN   \n",
       "29499                                                NaN   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "0                     2.002765e+09          #3252          True  \n",
       "1                     3.003431e+09          #1861          True  \n",
       "2                     3.006065e+09          #1043          True  \n",
       "3                     1.776631e+08          #1999          True  \n",
       "5                     3.006029e+09           #729          True  \n",
       "6                     3.006029e+09           #865          True  \n",
       "8                     3.006304e+09          #2926          True  \n",
       "9                     3.005811e+09          #2427          True  \n",
       "10                    3.006646e+09          #5595          True  \n",
       "11                    3.005152e+09           #682          True  \n",
       "12                    3.003549e+09          #2776          True  \n",
       "13                    3.005657e+09          #3770          True  \n",
       "14                    3.003886e+09          #1502          True  \n",
       "15                    3.004913e+09          #3333          True  \n",
       "16                    2.286137e+09          #4581          True  \n",
       "17                             NaN           #236          True  \n",
       "18                    3.005213e+09           #205          True  \n",
       "19                    3.004048e+09           #102          True  \n",
       "20                    2.029408e+09          #3395          True  \n",
       "21                    3.006305e+09          #1218          True  \n",
       "22                    3.005657e+09          #2582          True  \n",
       "23                    3.006468e+09           #781          True  \n",
       "24                    3.003968e+09          #5462          True  \n",
       "25                    2.615949e+09          #3296          True  \n",
       "26                    2.766462e+09          #1815          True  \n",
       "28                    2.537201e+09          #1225          True  \n",
       "30                    3.006672e+09          #4638          True  \n",
       "31                    2.163319e+09          #1449          True  \n",
       "32                             NaN           #305          True  \n",
       "33                    3.002812e+09            #61          True  \n",
       "...                            ...            ...           ...  \n",
       "29465                          NaN            NaN          True  \n",
       "29466                          NaN            NaN          True  \n",
       "29467                          NaN            NaN          True  \n",
       "29468                          NaN            NaN          True  \n",
       "29469                          NaN            NaN          True  \n",
       "29471                          NaN            NaN          True  \n",
       "29472                          NaN            NaN          True  \n",
       "29473                          NaN            NaN          True  \n",
       "29474                          NaN            NaN          True  \n",
       "29475                          NaN            NaN          True  \n",
       "29477                          NaN            NaN          True  \n",
       "29478                          NaN            NaN          True  \n",
       "29480                          NaN            NaN          True  \n",
       "29481                          NaN            NaN          True  \n",
       "29483                          NaN            NaN          True  \n",
       "29484                          NaN            NaN          True  \n",
       "29485                          NaN            NaN          True  \n",
       "29486                          NaN            NaN          True  \n",
       "29487                          NaN            NaN          True  \n",
       "29488                          NaN            NaN          True  \n",
       "29489                          NaN            NaN          True  \n",
       "29490                          NaN            NaN          True  \n",
       "29491                          NaN            NaN          True  \n",
       "29492                          NaN            NaN          True  \n",
       "29493                          NaN            NaN          True  \n",
       "29494                          NaN            NaN          True  \n",
       "29495                          NaN            NaN          True  \n",
       "29496                          NaN            NaN          True  \n",
       "29498                          NaN            NaN          True  \n",
       "29499                          NaN            NaN          True  \n",
       "\n",
       "[13219 rows x 14 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['has_full_text']==True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 29500 entries, 0 to 29499\n",
      "Data columns (total 14 columns):\n",
      "sha                            17420 non-null object\n",
      "source_x                       29500 non-null object\n",
      "title                          29130 non-null object\n",
      "doi                            26357 non-null object\n",
      "pmcid                          27337 non-null object\n",
      "pubmed_id                      16730 non-null float64\n",
      "license                        17692 non-null object\n",
      "abstract                       26553 non-null object\n",
      "publish_time                   18248 non-null object\n",
      "authors                        28554 non-null object\n",
      "journal                        17791 non-null object\n",
      "Microsoft Academic Paper ID    1134 non-null float64\n",
      "WHO #Covidence                 1236 non-null object\n",
      "has_full_text                  17420 non-null object\n",
      "dtypes: float64(2), object(12)\n",
      "memory usage: 3.2+ MB\n"
     ]
    }
   ],
   "source": [
    "df_meta.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CZI', 'PMC', 'biorxiv', 'medrxiv'], dtype=object)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta['source_x'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'db459b8a4b49c8b1ad883d0824c6bffccefb525e',\n",
       " '80a789fb3ed3279911f241585e315718b54c9a91',\n",
       " '3d6462be232a3f4b4aab41386a74bb8e6a118b44',\n",
       " '7d07ea5145ff28047a49998617c8e015c5d19ece',\n",
       " '4e6709b68cb7312fda8a1c5a2b2505d21a38a1e9',\n",
       " 'd90a211ef31ec56091a8c30bb8ee39e9f07ca755',\n",
       " '044e273c24dbb1ec49d20e760568a100efffab20',\n",
       " 'd6303899bb27b192349eb8f8a8f5498cdc963ccd',\n",
       " 'e1d8c7f82bea0b322ce13645fd051d992a2cec84',\n",
       " '8b12db91c31cd4a723270f3cbb56f7919581bb02',\n",
       " '306ae95aab18fe16957a74a8681ed32130c052f4',\n",
       " 'c46022b8376c55d77bf1e504feaa1e6c5e3ed626',\n",
       " 'd4f1ccfcf8b71a2828f39991e67142dd3a57fd1a',\n",
       " 'cb7bbade2083d71f793469f51b908363e9f25cbd',\n",
       " 'e5c06914d0daee6e881665b23adca340d0a6271a',\n",
       " '816fc25c7f0997c17db295c0a77b18b1a2375338',\n",
       " '93507716d80b9834a1ee584b038c1be54c0790a8',\n",
       " '2183478af3f9343c0f566eee8b5d95a1abea6754',\n",
       " '2238250b340cce482ba8dab2be98d10fcee965d4',\n",
       " '0d9a4673c7ff4a10e5e84667278b2fa3f8ac729f',\n",
       " 'c6bf372c094f035a514975c35a7f9c094abbe493',\n",
       " 'f293ae4ed7e4a778e7f9cc024d4e4cb33d69cc6e',\n",
       " '45edbd85cb93950fecbc8da86f684a9e45c1d8f1',\n",
       " '4156b347411e4071b10550a6acc2725c5e82df99',\n",
       " 'e1f2c0859f44eaf3e7a2110a5ca98ac0104815bc',\n",
       " '1477e7acddb857281705b48912ade3fd5b3f2a6e',\n",
       " '3677789674e47d9e9b35f87c659430e5295370e9',\n",
       " '645a3eaa114cc4a4827d02c0ef31b8a19c45b802',\n",
       " 'f0dd1cca2c2cc4ff4f1a37a74ca694382ba9eed4',\n",
       " '93aab5344693c6df1630f5dcd54b16ad82d7966a',\n",
       " 'ba0f23393bc8673f2e26b02053f6135e5a2f6a44',\n",
       " '74bee8bf3229f4c28991a7231c02d911d24770e9',\n",
       " 'a0aed635308ebac3b0e430d3c7722756638b1510',\n",
       " '87c6c45d9946889a9a3551b31adff4c32c6c237e',\n",
       " 'ff5a79ed22ea416e6d89caad1cf0d83dbc741a4b',\n",
       " '808f6be54af8e7c20446dc1ebdeeb68af2c28fba',\n",
       " 'f84decc775e81850bbfcbf71972377db10e4863c',\n",
       " '387ef042d74eed570e829ad4b9d7b721cfc9ea27',\n",
       " '118373b29085c6212547a34993be16eb98008dd0',\n",
       " '0a64b2c55fc2091358da4f83ac6fdc7701ddb46e',\n",
       " '29b0c94901ca75a77502042d007037709ecfe199',\n",
       " 'b4a9b6b4be03252d0b1e4c237603c3d2d13dc1f5',\n",
       " 'e18f75d74176ea6eff4a4e8ef6ad0a0836a92a5e',\n",
       " '5bc13185f8fc98b99cd1fb22078afeda41eb1e7e',\n",
       " '0a003aa69f43cee4357f1e943df79a8b87c0a88e',\n",
       " '1d4793842e3eeae37f34abb1541b0cee61ff976d',\n",
       " 'f7735a5236aeba514810bd4538357cb4f4e09fce',\n",
       " '6f3dddc4c061491fb43651e9bb61039ac7594a78',\n",
       " 'e0531f9c84005ac77b1bdb3075c486e6ac116760',\n",
       " 'f0cbe63a26a901e5873aefd4c01faaa078c89c81',\n",
       " '6050d0c1b7c6990972e6cdfca89f79745b13cc14',\n",
       " '515da9e83a07430922c9fe2a979c0698799aa56c',\n",
       " 'cbd966ad23d27df95cf1959ae5d944fb363de366',\n",
       " '1919c783b2fb4127114d196eb2b30fcd7a67d15d',\n",
       " 'fd421b0d1a42db7b23a87c7c7fb668d54437d70a',\n",
       " '67282d82fcba34faf696a510ca4d54b8804289cd',\n",
       " '53f05daf7342372ab5fbeb2f4f5710254fcad893',\n",
       " '7a5478dfe79fbf67551ac9261f190d899ba719b3',\n",
       " '70f2b5c592dd6da732100912bcd2ca928a882842',\n",
       " '7e7c6670854897412ce9930eefc95f348ae432a2',\n",
       " 'af8d060acff4a631e09af5425cdab023103e9d8a',\n",
       " 'b313d134237c1d12ea832bcad736937d908a6c25',\n",
       " '190505ad74408b21fea8434a44a2f43b1ef198b7',\n",
       " '38e37fa66e4acf2931dbc37ac44d66cce82bae55',\n",
       " '424d6c622643d36a7f8d9a1d2714f4cef49387ed',\n",
       " '54e77bff1912076bccf12a6b099e48a59f634d5f',\n",
       " '6b881730788724e674806854f0c20fec32ee9d96',\n",
       " '5a34dd045c63483689fd09292fc0256f32b137a9',\n",
       " 'b1b2eb98641f5fbf2b51cb26566196ae59a7ad96',\n",
       " '5b7753b6a659e6afa0a46fe171a372e682b07010',\n",
       " 'b1ef060481c45d71c380d4ac3ec6d8a4f0f26945',\n",
       " '543819a7a29de2c78dd6e147beed7be72466c610',\n",
       " '2f314a77ea26734587af46d6ed32345b0160760e',\n",
       " '340036a465efeb78d0b0160bf6dddd2322f293ef',\n",
       " '7d1b56a1da849ab1a3e0ee57807511ebac91de24',\n",
       " '7b0abf2414de3794e54ed37050884f636333972f',\n",
       " 'f80a73f77c3831a921a232122099f8aa5798dfd2',\n",
       " '92f24acca725725852dd170c673f8adac4250faa',\n",
       " 'cac8f96521f21bbd0a84861a98a5715d86b293f6',\n",
       " '434ab543a48b7fb395d181ffb947814bd1b506c4',\n",
       " '0a01f5cf1c5cdc2711bcef74315dc54a6e143df0',\n",
       " 'a279441c0873b6bfdfbd63cee1220cf9596acaee',\n",
       " 'ad147a4279222c5e480a5636fb879540eded760e',\n",
       " '02610c838845176ed2d97c20f415843e1b5edcc1',\n",
       " '9170d2682a483c1480ce002486cff53e99789484',\n",
       " 'f0f56c1e82c717ced3a4f2ca872dd303581ea750',\n",
       " '090b6c8b3df30bc248221869f673a2d970caa1b9',\n",
       " '5f3fc1bf0b828ea47a282433d84726eb5b25447c',\n",
       " '7bf3533ae81b36dc5c59339e5d62eb57065f27cd',\n",
       " '2ef7ce66fc8445aa7bfabc121193e2ae0091fecb',\n",
       " '7e136fdfc76cd52435cf224414e8a59fa3eaf227',\n",
       " '6efe01046ce81279412ea440a4b246f942f29124',\n",
       " '3d2962558f0a2ed4ddc00989168efe60856bd792',\n",
       " 'f5ad2323eb387f6e271e2842bb2cc4a33504fde3',\n",
       " '772e7c8efc8fed6e7c27602dc255fd97f2f55a7e',\n",
       " '409e8fa9c8b69d7982ac0ecc28437a2e8a246d2a',\n",
       " 'cb763a61411514366d16bb2571461b6df1361517',\n",
       " 'e6e00bfca850f42e1b2baa9ce190d37554f4a42e',\n",
       " 'c61f5c4971f07156d3c1d9dd76de4b158855ecc1',\n",
       " 'e5a309be751ae9e2575c3e35edd7f37fa0e99b3d',\n",
       " 'ef000d8cdab3895e2321286f16cce2b8aea458d1',\n",
       " 'dc22742fe9f296e13c6d9b375fc35407d5c3609d',\n",
       " 'b79bd179bc542b31f7c76f7115c5a338dfa3c614',\n",
       " '26c0767fd866d344dc410abe37641bf0a82704ee',\n",
       " '73560247ea59c1db0dd80dc9022c7d6795f7d671',\n",
       " '11f3022ad1def6dfb1b29712112b96739ff3c498',\n",
       " 'a3718ef8f5fc2e8b32992c568bc66646a870656c',\n",
       " '94b25ecf44f04f66625ca9975f07118df3a0033f',\n",
       " 'c031da739085e21f3acac50ca0fa90c443466894',\n",
       " '737c765488563c2864d576b696f16198bc83344c',\n",
       " 'acdc7132a3cc967450623c8ddcb38ec4214f6850',\n",
       " '3f65b990d1f24de2bf3cd52d8559e8ccba9961af',\n",
       " 'bf0bd7e3007cf87a4907ecf1bbf819d28e5c3c63',\n",
       " 'e73fb635f2109543e96f53739fdbc9db8576c17f',\n",
       " 'b09fe466052873919115ea1da6510a01704c6c3a',\n",
       " 'ab8c1e32b66b02cd703799df3d2ee37a1cb369b7',\n",
       " '073d9b195e4d3e325a8ec7cd30b9ec23bb0a00a7',\n",
       " '70c7260fc0702bd4a554421ac68972c991313e56',\n",
       " 'bf20dda99538a594eafc258553634fd9195104cb',\n",
       " '9d77f5c67963a68f5ede5bf53e95fdb03270fd2b',\n",
       " '9ba0bcbbdf07e2bc0f584257747428a133ce438e',\n",
       " 'aeb2fd035cedb1adb0f4498c351310f5f2fea5a8',\n",
       " '978fcb9694711bd974f21203205a86d359868b3d',\n",
       " '01af2562df4acf3113843d039c3b82bb801b1427',\n",
       " 'f1541ead6bf6988b4d08b12873806a89a4b7ff1b',\n",
       " 'e505d4ad17c8007a18529518f34fc491594e70e4',\n",
       " '1edab5890fbff22ad353739d3d1e80a86d482820',\n",
       " '554f6f5ca9031e8c6dd85eb32f3ad3e6c222c18f',\n",
       " 'de652a9f987a68568c425859c662ea8d2325631c',\n",
       " 'dcfbf04c5bf819de0d261d15a1936288f578d416',\n",
       " '25707e09aa87bb548138776e263ab037f4e6cda0',\n",
       " '2ab8cb240c2f488f5a7bbcf117239eed7e3b9633',\n",
       " '7a16fe9d867203bf83dfe50ef6dd93b8e1179db7',\n",
       " 'f3598c3dfc758286cc4af164c6cfb66f5bd4826d',\n",
       " '4c8110bfcde6d110c4a455f7bb8ea1a7801e8434',\n",
       " '041827953afeda8e10e25ae78437e37f8a453578',\n",
       " '558eec44990e2c8c43145a48bcae68219ae2ba52',\n",
       " '188a2e1351ceae0aa8347e1a1d7af1e3b7b66a02',\n",
       " 'bee2faef06dd2aefa2a126de2e15cbe403562aae',\n",
       " '50627f92db9c17c742fb2cc95641bad921bec3ee',\n",
       " 'f4761a305755e1f614a75cc8ade098b09d005728',\n",
       " '90f2e2a33f7f6bae27a703b4c30a99f87d404474',\n",
       " '1579fbff7af9b156c6f49fee0526e48f852ea460',\n",
       " '135f5ee5fba1a4280521d45e4fa882354f63e691',\n",
       " '0a789bc7f1713d549288d3effb72ff34cd3b0ab0',\n",
       " 'd12f82f17862869b5118dc55c63929adc6597dd4',\n",
       " '231ba154ea0d9a668c5c5e7e5732c4f223608f6d',\n",
       " 'c32d0c0e6f09f6da1b8d9075733574b26bfec04a',\n",
       " 'a87c83a672852a6d046357d68715f3c3c0cdf5b1',\n",
       " 'c9df58c92109a1057c10b24867cb3566ffb46408',\n",
       " '30abdf7a423427b001aee5d319abecfb5ad1671b',\n",
       " 'f14d0b6eecd99d0bc6afefbb3db486a895ce2efc',\n",
       " '7fd0972f4d5eb0e4c7ec68efad48fc01a1f547c2',\n",
       " 'b9eaf7c97ff51c43bbcf3ed0605c2a040fe8c588',\n",
       " '8654df96fb9cd8ae9dc157fd4bdbf918760de8a1',\n",
       " 'cc4677c77d61de96ba2873ecfc8faa4a2f393def',\n",
       " '3606dd5863dea1e8d6b7d7480ffb9baa06305382',\n",
       " '94305928dcc0731a946af0b72fa89c5e79a86546',\n",
       " '2a17719f2c1d211a651060441bda8bc1c052e2aa',\n",
       " '1f7f414d82475a8c5b2272163e6f31b474904486',\n",
       " '0fe7a393c5db38f3913c46f25a3ed75d14f76900',\n",
       " '2f10cd7b6249a3e9e55501f270f39a62c982981d',\n",
       " 'bc7b16d381d2acf67eb97db19f1686c90d72ecf1',\n",
       " 'e4007988dba3dcfa65314d50324bdff1ba7f55c2',\n",
       " '691bcdc9ee856e121db71833a0c56c43c3c25555',\n",
       " '2c2f0107d55f1c770896c982d59e0bfa035fd4d7',\n",
       " '261df5b4f76de3cb7aa2e8dbeac321221f065df1',\n",
       " 'c4e99c8b861f4457b4a0518860bc9e52f5348e87',\n",
       " 'a599eb2c6ca51522fa234d8059f989569f4fe298',\n",
       " '8792b598db3c64a2da937a9cefc989a7fd0a67aa',\n",
       " 'e197a8c2fc80727b35e38ea7aca28d8e6edfb27a',\n",
       " 'a3a2b5d607691f30dd1b1baeca0e41817e2fde86',\n",
       " '49471df378964515ebc4e4f35f1248d84ac7d626',\n",
       " 'fe66306680e5be9820e2cc3742b663a782eb1b31',\n",
       " 'f243e05a2d51029218e0c313b20fa57c65104a62',\n",
       " 'aebf3e6b69b0563ab38d9d973db9a7795f75f47e',\n",
       " 'cd57034187ff47256ee511723a6a05b18a923026',\n",
       " '3de90c8c18dbed90408c4a5cc5cb94f856c9f637',\n",
       " 'ffba0e28de0ad62150f452c30641548e56503bb1',\n",
       " '9ae5509495e79b5d077ce582f092950b38982dfa',\n",
       " '7e55726d345571690d0e3046664cfac5003c2a89',\n",
       " 'f4f804bac7b32c84ad6572776df684df2a2e5fda',\n",
       " '871bb1d475d0dbb910cce25e16d066adf9706a4c',\n",
       " '02c9b77cb2898b1597278dbf32a0c61be93d1983',\n",
       " '3725a278a669559c9fd096a9a7dd5f35c6928c6c',\n",
       " '3f94161603e0a339aa90475fe4c7001bb54d1ba7',\n",
       " '4abc50cfc81265083d4cb2aed0a493bb704d0b5b',\n",
       " '2e237dc7ffdacf42c76c0671bea72fae2a3ec9e1',\n",
       " 'f5df82125769765004c7dd11f066688e6dc3cc76',\n",
       " '24c4787033b7069da07ce6b0b8f85f6fd5ad84b2',\n",
       " '285852264b2c14f53cff87f8ae9ef7738892aa3f',\n",
       " 'b5215808485ffba5aa16d95577bd6344ede9d45c',\n",
       " 'dbd121bc5733e77f68826080a177fb8b0e8d3106',\n",
       " '4dd0f09aa22d567e9f5f9a0bebd8686be35da853',\n",
       " '71edbd57cdd9af956a12054932e0cbdb87ce1fea',\n",
       " '24e17488d399c436305c819953beae2961214771',\n",
       " '1b050ba5ac020c50042f89a5f4936404ab2f507f',\n",
       " 'a43dc2a14f1a0ea6e5203a062f130693f5fd2a4d',\n",
       " '40b1e8bb31863fa9488fcbfc590edf4972fe4c0c',\n",
       " 'a8557b122155d371bbf43c23abf059b526da6496',\n",
       " '896cfa254a80c192a7aa872d955e750d6280e071',\n",
       " '4d140ee8fb10afd86c19fc5ad4749771dc37d146',\n",
       " '3455e9fd57dea013498de20c236c15b7bf19b424',\n",
       " 'fdf021cfe745daed338cce7eaa5e548581477ff4',\n",
       " '29e75f8938836f57cbdfa2e73f05939ee5f561a1',\n",
       " '7aa5c87d322c83d77e3a4d1a3ae55b0bace68625',\n",
       " 'a48482a6bd5e78f140777702151cffe852ea5f1e',\n",
       " '6da3f66c681b4b60ce1f6f246340517b79b67137',\n",
       " '0ca8648d40bee3056b3f9840de6d34b57ed121d0',\n",
       " 'e523533cb5df819c3354420da60b7c82f9f92485',\n",
       " '662d865c6858d8e9f3b7f583719579ee0aeabdfb',\n",
       " 'afcc1a28b96721fb9f9544943254ab73d656b635',\n",
       " 'd54725b7c8072e09a2498c8f2ce1e7ee7839eaf6',\n",
       " 'a7c6ab514af0b9e5edc73e08e496ceb65111cfa9',\n",
       " '65cc7496f21429d81b3ae129d9c39764e2a1f568',\n",
       " 'b07bcd8179099f8a5e49b8a7a2530590da99403b',\n",
       " 'e3af2ca43010f59c3d1bb731abd011e3dd0fc51c',\n",
       " 'b44258fb9be12d1ddecb77617f2cf7914313d35c',\n",
       " '67c9acbf8204919fe540d58d7db296b9210f1e90',\n",
       " '95aec306660c1b7519e9f019d92cc3a8206481e4',\n",
       " '3e8f1cce602f86dd1f7227dbfe0c21ac995ff470',\n",
       " '1029b20dd40ca982d05b80adad4c125beb10217b',\n",
       " 'a60d8041651d988fe83f9d4def187dcec19e1b66',\n",
       " '215fb42f8f5f2ede1ed7e7b1561e51f494593578',\n",
       " '5390aefc8bb1a06635a321c0cd82589e2b91c9ee',\n",
       " '7d25d7686c679f9f1e1f04a605f6bbaa385435ea',\n",
       " '5dc0b8b662824323881c3a1ae3a1bae2a821484d',\n",
       " 'a093f4ef60a47e5419181f6826ad4a45af198056',\n",
       " 'ce43f8f05a5db366babc9a6bfa086219534de208',\n",
       " 'fd8ee38a505cf79732f5ba487e15f11619b84150',\n",
       " 'fcc20c80be8954f1682993217bbd2a78659a16f1',\n",
       " 'd107ccddb5e20fc5a311869da6760711f2f1f8e1',\n",
       " '7893ef5afeb48e571b8c0e088e1167898d522a0b',\n",
       " 'aeae3e593ea4d3d0b59c83839837be75d2ea47d5',\n",
       " 'a3de313c4235416a3a155a4bf154221a068ba051',\n",
       " '1244246ac28247b5674d6f4a8587e064bb565d7f',\n",
       " 'c8f9c5d7e30e601b4ea6e3c42696177c57aa1c47',\n",
       " '7b22c0d8cb7675bcc5aa283fe3bfef6c72052519',\n",
       " 'cee73a9e8367f60e1149b8db0b8a00a780c84d9d',\n",
       " '47266ea82145a11ad6e82db70a4d0fbd86a27cb2',\n",
       " 'e2b3cf2cab4bdb3f5d357c987852c829cb4f6995',\n",
       " '6620e8406c8ad908234205ff102d7bd2cc0dce72',\n",
       " '09e25e413faba97b87efc701d1ab8d2a18386efb',\n",
       " '2afff5fe89a3e3ec270a41c2b929dfeec68d7d50',\n",
       " 'cdfa3f7c065622bfb2c811f71fc0dbae5b1a4d3d',\n",
       " '49241d35c68b9ffeb0057758abdf00fc18cce2f0',\n",
       " '2d81bf103ad8ee0f0f3caf92fe2c523f0024691d',\n",
       " '26b49bdf0c3d50386ae603783e9674b7349852df',\n",
       " '9af0b8e13e1e5ac714135dca8ba6d5216966066b',\n",
       " 'e68ff81b1cbbfd6bb525f60a9efdfacf899dabd5',\n",
       " '850d93fde366090620b90ed908b64de128048119',\n",
       " '85a6db45e29636f124b7b1adc5230dfca7b37f9f',\n",
       " '8fefbc58375c244a64a82e3f88c4084076838b88',\n",
       " 'dbf76534e0b330bb0f155296e251c1e7031c90c0',\n",
       " 'c856a5583dad88feeb274cf1ba74ce99632d3d6c',\n",
       " 'deae71f6c2acd88de567196bd0db6a426b759ff7',\n",
       " '5dc2b500e260938d6db9120669c23b90942bb07c',\n",
       " '1d3cf8875813bb56f4d5df8c11d0785ddeaf1a99',\n",
       " 'bb8a9e29bc65471177f470285c32d879c2cb7263',\n",
       " '46c58d9482fd8d00108402c45033131561eb8921',\n",
       " 'ebc8949731f5cd59084c4733eb4211d83547e4e8',\n",
       " 'e633cc8cbce07eacd478bb25aa824872ad55c59d',\n",
       " '1f5c1597a84ed1d4f84c488cd19098a091a3d513',\n",
       " 'ba24b8b4f656dd857e9b151c22c0619b88c63f65',\n",
       " '27c85eb0542256f6599ba43e90986abfdb1ddc5b',\n",
       " '2f2921851e825705103320bf3d61f9d76e4c6b45',\n",
       " '8d355f050721e1e9d9288c7824cc38546b6ef2a6',\n",
       " '2b6ea0603deab65ea1e0ddf4367985f6061bac9b',\n",
       " 'b80d9de7750d627a0189fcd2dfd82c7efcb1c079',\n",
       " '58654073410779249fcd789d116f5537c87c5357',\n",
       " '898b2d11e25f2f6ccc7dab4b572f6c6f0628639f',\n",
       " '7146a5eb303bef6ef7c4bc2588a0305fc7b58cd3',\n",
       " '916bd646b1fe224dfdda73b6b039922829b1f448',\n",
       " 'cb0ba244ebf7524d9303edd3c52946f08560f597',\n",
       " 'f03bbdf1292de07aefe78b58be581fabb2871325',\n",
       " '306840cf9baaaf3a2e493a39f3ebd67529b8cb2a',\n",
       " 'fa3d312d96b3eb8ed53b8e1ebf3d6f1025863c76',\n",
       " '0a1f43c04e0e22fb6efbd94611920bc9680d7ae3',\n",
       " '8e34a3230f0fc157b5d5ed998fa8792f0ecf3d0e',\n",
       " 'a14b5655cb13ed64cb8cff7c806a7b58c858b8b7',\n",
       " '643f2dd329f342e9e85753ab315722bc1120d0ef',\n",
       " 'cbd8eed5192247d945644651f0a9de6166992567',\n",
       " '8b71af2afcb6a76e746d1a9d77d018d28efd4969',\n",
       " 'ed3dac12ca72f06c51d7f68ae5fe6aaaa6f3e740',\n",
       " 'f82bb9d2d37888e87a1f8f42c7f898a809c3bc94',\n",
       " '9adb9a55bc7756ff69f23f66796941b73f40f4a1',\n",
       " 'ee1b5a9618dcc4080ed100486cedd0969e80fa4d',\n",
       " '74eda2d826f50948657d096912bb32ccfbb17d0f',\n",
       " '0255ea4b2f26a51a3bfa3bd8f3e1978c82c976d5',\n",
       " '1adab26eb1167a39014dc503174f7d34f9087664',\n",
       " 'f67740f11517f204388333d646050566bba9b65b',\n",
       " '36c09af15d4a614f954aae5a5f8a4199f1ff8b2f',\n",
       " '343aeda9b3a81a9db2f40386f3991ef6c48338ce',\n",
       " '3846cb057daf0b69fec4d889ff8657d252d94509',\n",
       " '0f38c2681e9e286de25759866e018b1c4a2a88c6',\n",
       " '3e339ee119ba0065cad4c40897cd1bc9dd0f118a',\n",
       " 'ff15158f190f330aaae175b1dda34fea9377abf5',\n",
       " '2b554bfe12a457f1092ae7b12537e48cb3392f67',\n",
       " '12834756cf70a24e0bc14906b131ca587f86dc02',\n",
       " 'ba4c40b65eb342713e70c96a8d6336b22cca54fc',\n",
       " '38e796bbec2f90b1b802c14dc922102e96f6361e',\n",
       " '2e4ddad3f80f7e1cdad82f054190c0656b1e22a4',\n",
       " 'fbf232152712b5e379797c6b0008bb7926001bb7',\n",
       " 'c63e2bbb0915e7d4d40813d58286d51e80453507',\n",
       " 'd6043608d84c9697099d1e554487278e735942d2',\n",
       " '1a465d982030d8f361dc914ff2defa359fdbe5f9',\n",
       " '1460431a11ae554624fded17f039eacbda52effb',\n",
       " 'bf573e1259148bb11f71b1dce3ad11c0a091ce36',\n",
       " '0a19aacc124c9c42d66d481a9d0c837a76a252e6',\n",
       " 'f633ec24c26ab3db499227405678f21cc6ab2af5',\n",
       " '4c6112b657933e7254342266bc7afdeb42af507d',\n",
       " '210912792a1547a437a4640fadc339c08982bb14',\n",
       " 'bab4754e6414e57b1085e839c602053e49510939',\n",
       " 'c6e1ce0bd3692673434c291e76fd1f545a4115b7',\n",
       " '04bc03c90437934a75fc6fdc228817234ef84c3a',\n",
       " '8d581519c2e082ec871aeb8e6486e3195111b6b7',\n",
       " '82bf46aebd99e6e67368b9c789d01e561ed147af',\n",
       " '510df094b943c60422ef861ed91abade4a4ade4d',\n",
       " '680c8cd646f4f78d97eddff21292668beba44d09',\n",
       " 'e6fcd7b5c6235d17f46c1721bd8695775f1df236',\n",
       " 'cfb3f090f09ba13429edc88b5f1d8a595df81fc1',\n",
       " '91dc59536a7c1113cc9bef3fbb8972ea8555b0f6',\n",
       " 'f0b95482306158df395281317e4f3f7b37c5a52c',\n",
       " '420944cb316179f6529d1d7a99d92910df8e49a5',\n",
       " 'be9cb78bf2453f68b260e9ed4f75d9ceccb77754',\n",
       " '596e771e860af036be53a0f54e56d815348c329c',\n",
       " 'ea538974071c2e340580098fe6ea6998654c1fcb',\n",
       " 'b95e82e0343a470f5b39c97eae9f8541e218b76d',\n",
       " '5148f1055dda3d084d867a119c10c397fb647c55',\n",
       " '35173a96f21f1e674e63d473864215d025f76d0d',\n",
       " '57df78198df2774be87635b81f850b6eeb2dd6f8',\n",
       " 'b921ec0b4974533423b7af989620ff4ccc5e2f79',\n",
       " 'dcd1759241f327b5235520cebe1f7432aa936c5d',\n",
       " '27ee6e389b8f87a3545232db8b8fe385f73e5003',\n",
       " '1ee231435b5c7636ca5cca30617617256bdadaa7',\n",
       " '5c525cc04075b7c82e52b1e80d96bb394b1a0907',\n",
       " '0e2ba405f636e06821999876cd82c00e26404b43',\n",
       " '73aec7f72de1b225ab650b78a7633aab4ebd97bc',\n",
       " '4f8d24c531d2c334969e09e4b5aed66dcc925c4b',\n",
       " '6f14312d95463d446d7f58d981a725f9efe2b5b4',\n",
       " 'dfdeaf42e832860f70d1583ef54f9b319b5dd2a9',\n",
       " 'b60e2996ec59f48cb2376f9c791d7253d0f220cd',\n",
       " 'f0cee74f65b4b34666a289e300164727a3da3ff4',\n",
       " 'ef6dbb94ddc618e006e993f32893043b9fa87463',\n",
       " 'f6f37ba5a2dd88775b18db2b63ea9d1b5dabe8ec',\n",
       " '2c6d1fc20495ba6b8b43ffe06cc0d9bb80244d96',\n",
       " 'b9b56a7b507acbddb2122ecc330a8d331c5ff33c',\n",
       " 'a9a03b8e5d632fb01f973df42763da7fb0179629',\n",
       " '2d78b0600b84cfd57e11ba249ee58c159fe517ae',\n",
       " '568917a765516d1397d8cb461644e755cd703c90',\n",
       " 'fcbb20fc325656ad86d5ed3e810cd0f4639ada77',\n",
       " '7c31dec4fa9159b171d8e08d603ad0535a379d39',\n",
       " '8d7d39313cd9791fca858c229b4af66a8e0927ad',\n",
       " 'f4d811f1ad17486fd10dad9a4a5cbbdc59b4d9ad',\n",
       " 'a820465dbd9b10702c009445daa65b9bd48e0c72',\n",
       " 'a92d2df5aae260091456d3a65800c01cca79d599',\n",
       " 'cae0b002b145d21b55c8fdde52e28b269aa405f7',\n",
       " 'c92d4b764634f984f72efccaa63b365bebc10616',\n",
       " 'b05030dae8169323463be3fd847e447f48d4f902',\n",
       " 'c6c0a0d35bdedb534df909619bd7f240f9d52ae6',\n",
       " 'bd788a8d8076040ff66825818479d21cfae5929e',\n",
       " 'f8710043513b5790e20921e8a36d1cc94b402701',\n",
       " 'd4f4ca21b787bb6a090b859a21aade4e7a0fb633',\n",
       " '743c5db7611d4af74204901e0ed3b6317a0bb3c4',\n",
       " '8d7027f9c0c85c6aad1c9459d1c16f2b3b918280',\n",
       " '29bc34aad802ab6b35074f23c631540a84b16d0d',\n",
       " 'e2b5298da8d9d86daf286304637a644139dc6dc1',\n",
       " 'e447d139f1d046120a293f1219944e82c77bc829',\n",
       " 'b4c0e3c28f1d56b15dc109d4a59731161e07fb41',\n",
       " '7fd909b06fcb46cf45b5d099741e4a86138ac6f4',\n",
       " '9d343eb083c8716c946575e15df1688e58d0b288',\n",
       " '5ec99e67f23c0c2795fad1ad551ab1abe4406841',\n",
       " '8946e4b778ae5312b4db84cd75b68848c39a3a93',\n",
       " '15f149f2f2ca015d5b8b5c6ef59595c907f4c01f',\n",
       " 'fca0580d965f2ac6c66e345eab40c436d6468c66',\n",
       " '618cd102ec5051a05ca5131e21961c964152f15c',\n",
       " '7c64dd916107c4834e94b209c7aadc61be6d7c3f',\n",
       " 'e7b9c386495606981ed7a87be8a325c87c35fef9',\n",
       " '4e550e034ccca6fa2a91e481ddba24db67bc9ae5',\n",
       " 'c0cf1c6377c3115aa80ff787332120d4ba1cd386',\n",
       " '7ef029ee55ef6b0a0e6443f8ca750e0569024789',\n",
       " '1a8fca50e01d853310149bc3dddcd14cfbeb32e8',\n",
       " '58d91841094c91e8269b22337c66dfa16e04b965',\n",
       " 'daf955dfd5f5ca4e6967e9c1a03904f441d4c319',\n",
       " 'a234c5ac982ffc63b7be9d7c1b3473f92849192e',\n",
       " '4ba13c94e10bfebdd334f377f78e40bbf7e25d72',\n",
       " '73c97b301049429f21da128565e117ee1d368046',\n",
       " 'c865d4f4df0bee73efdd9413b23217e1b05b0f5e',\n",
       " 'cfb5101deb971a2922f94946edee44ebe7a7ac97',\n",
       " 'f5e8e27a130f3239cafe592f4174b290849b0f2f',\n",
       " '7a8f5602e6965231fd63c55c95b2d1523c3f8091',\n",
       " 'e25a627093e802ba8b7b53987ebae5366e98ea63',\n",
       " '4c3ff586d4238ccd7324950db31a02037463f973',\n",
       " 'fb4f03ec469640093e45b1bf924456d1cf05c07b',\n",
       " '359d79257b07cb81eddbcbcc2882bbb43362cfe4',\n",
       " '1b41bb775aaf552e944aceaa31f52814e264c78d',\n",
       " '310c97d16197e1ab6ff3227b706bed41f079b8ae',\n",
       " '18aa7bfcf8b4dfcad03cf4836aaaa5345d6cfdc7',\n",
       " '9ba8b44bb13969b76e138f9593a787943c0ad9e9',\n",
       " 'ab58c54193c5257b96d0d8e098fb18d7e2d8d081',\n",
       " '56b2d409fd554bd2694eb7d938fd44fc282fd484',\n",
       " '5d62acbb74aea7559783bcf556ce7352b55ca03a',\n",
       " '4790e3eb0374a7f159014ef77ec42a6b9de91c29',\n",
       " '35fa07c7c7d1803f22859f6647d5d743e7ad7652',\n",
       " 'a1f353aafa8d3dfc2e76c6d3d75df0342c8a9e6e',\n",
       " '8a244626a996d30b62a52bb8705b889e74f3a5cf',\n",
       " '423e1f15afb86012057acacc26d0766aa4bc582a',\n",
       " 'f619b2eb37e9970f65936ac361feb22743395e65',\n",
       " 'ebcbb4241c95ad27eced8844b2efa4e450535931',\n",
       " '34483f6b88d4999066d631b65cdf0f256d305b6f',\n",
       " 'e3f168287dc3669b48862c0e70e000fc49edf00a',\n",
       " '2c092e37343c79101dfbf1f9f66a87f569432a29',\n",
       " '4572a8588d9d97cf1aa46241f2cf9e4a4d52dab8',\n",
       " 'f01ad3545245b4f884b48aa2b69c9deb942c3e77',\n",
       " 'e87dc948912d7ec9174234037dec7c4369c5360c',\n",
       " 'e63edd41c1aca8fe732d5285a64fcbd45357cdfb',\n",
       " '950476a6a916a066fb215b34bdf68f4ff79f2928',\n",
       " 'b4662fd83b035e7eb06014924a4a03c854ad797f',\n",
       " '6d04e85d2c994d73c9a7e2692af91470f407ddeb',\n",
       " '37a9917cf6525a9daa87e8f3d14393d0fe0a0147',\n",
       " '36eedeee8fd40eda2f7f1d0c0cb0e2a4971439a6',\n",
       " '94cad8eb2f7ef7b18edd43115de077da98425871',\n",
       " 'e47d1f361bc4007b3fd859fd80fda5b40cc4a684',\n",
       " 'd86146bbcb9ae18278150f4cb241ca1ce31fbf28',\n",
       " '264e7bc2caec36fca2b89d1f21198b8abb4a2d3e',\n",
       " '423a10412afb0aef150b143711b96b0dae789ba4',\n",
       " '87cb1c3bdd1b8701c07a17f66a17e5d7a55797e5',\n",
       " 'a636cda3a52177d0c79927f54079b476b04b85b3',\n",
       " 'ace47aac143455a20fa2323df2cccbc6dc1b4166',\n",
       " 'bca258b97648522c7513be8ef70fdfd2ba5e3240',\n",
       " 'a6f4882f1c3862d065c63f0f352541404aed8143',\n",
       " 'b475438a09acce47100210534c623e75f3b5bcc2',\n",
       " '9671b5d479956e73b3f749873f672dd421019b80',\n",
       " 'cccb1a743e85a745309415410b3101c2ab1b59bb',\n",
       " 'b1d31bf64148c3dabdd5b8a288b78c0c6d3a7cea',\n",
       " 'd2d95788c3cd662cef7efad97646e0a9d2b3e47a',\n",
       " '239ca8536007960c7aed8369da414b93702217f2',\n",
       " 'd1d6a474989c30c3c93ecc00f886125a384efb35',\n",
       " '21da52646864c1ad3303daf10f0b3e4c9d12fa5c',\n",
       " 'e68b0f3da8a02a8f0b86869e533eaf549f70ff9a',\n",
       " '415ccc587d634fc429ac080cf06694f78621ef73',\n",
       " 'fd80a87a02d130989492acfc82db606ee470ab53',\n",
       " '97a632fdbc3ebd9ace9c84c90229bfa08e166ecd',\n",
       " 'd03888523e81c821f95247b544bdfac0c94a8b0c',\n",
       " '47f27a1405f69dd7569f6b5d2f0f73b72fcf4c38',\n",
       " '8fa2be5d5abb0473dda3892cf6f42a318f87b39b',\n",
       " '789679b6d2b9b81d978d080a3470cbff850ac835',\n",
       " '17cfe7cc0c68b1002bb31a2b2afa988ab9cdc890',\n",
       " 'a67212b1e5d4cac7cd2deb4a3b065e7aac6d8aea',\n",
       " '741878140cd73151d5bcb9d8a951b286434f148e',\n",
       " 'cbabbd687725e949c99a0a6f53f033128ea1aaca',\n",
       " '43d4ca7da86537799185698ae3463b736bc56fc1',\n",
       " '7d286a7255cf820dba6ea89a13764045161ddbea',\n",
       " 'fae8f01ff5ef7b1737b60dfba0e183bdf482aaba',\n",
       " '4eb8b7fd0032816e4a29d65b06939266d6446624',\n",
       " '03dce266594748330a10d2d11dbe611bf42aa42d',\n",
       " '235a2dcaae4126e0c1afeb28607fc683d4a6bfbb',\n",
       " '02aabee05e6968897e1031a767f6658ef0fc7b19',\n",
       " '0ed0bedd5005e893440208e3a176694c21e29db3',\n",
       " 'd5b981ab9223ede1243af2a7030ced8570c82edc',\n",
       " 'da9beea6cc4a217fb31b54caf455ca229bf8a27b',\n",
       " '9572c5aae9bef12492279374fa1f562a03eafd08',\n",
       " 'cdfc8be07677f50c91106e5b56379acb0d0eb289',\n",
       " '5159ab148faaafed700258c0a6278cb1fb5476e7',\n",
       " '6cfcbd2a102a3666d0d168f9bdaba0b2107fc8bd',\n",
       " '492ae848b7e2b20047ac406af1c7712ea6255461',\n",
       " '63a48c7eba8cc3c450016ba75dd52baf62ab0196',\n",
       " 'b31e3e25fa04d242866652434296925ac784d358',\n",
       " '190ffaedada9b387fb9c9f7923121d0bc894e317',\n",
       " '16dd2eb417bbac88ac2235efe705761a57807ae6',\n",
       " 'aa13831b8a93267215d369c250da28f78e323203',\n",
       " '1ff2f23a621ca476cd59ade285f8b9198ca178fd',\n",
       " '5208ab18e0b8b851eabda0c5c83c1d31156da8e0',\n",
       " 'd5c3ab7d56a6d04ce4feb57a5dd9d0059abca0a7',\n",
       " '4ba9bd26567dc15bc9a0abf88342604d62e336ab',\n",
       " '36e757c81ded2a278e917ab9030d1f682e4c4b9f',\n",
       " '7e757d19351d6680088ee7343c19873de2e1bcfa',\n",
       " 'c9b696a38ac25d3289283045ac72c8af370dda7a',\n",
       " '8e0481979dcd13b668686cbf3e2c8b51e5b8cf64',\n",
       " '3d758d41ef57c3d1e9ffe346ece0dd6b59e68cd8',\n",
       " 'f3b46e7e8f58799207cc44515f859c1daf5e4dfc',\n",
       " '359417eec03774de55b638845646f0f377f35de8',\n",
       " '54fc333fdf388bbf38305116d4971247a3681ca1',\n",
       " '6dc7f0acaa45433001b3b0586094e88c0d400911',\n",
       " '5b252be4b0ef235379d048786133fbe4a2af3976',\n",
       " '532f2c636fca1caae1f23885b9dc0e3302a0afd5',\n",
       " 'b8a55c284d17093b3d9c033f5c74633f47aaabe3',\n",
       " '74eccf041d34a1cebad39e386cba617d1bd996ea',\n",
       " '5f4cca82a9cbf23b9e0088a81966c3c9ae2c3f20',\n",
       " '83338ce69242b1a646ff6cd0234ae9d4197c374a',\n",
       " '5a717164844c095a4fb195350b44048059245b21',\n",
       " 'efd6ea1e3d5223f220c6c09084ac5fb043618b03',\n",
       " 'b470eec84fd5bb0a9a7079506df05c3cdf398832',\n",
       " 'dc1a1a0fec049e850789175fe06b1453c9114c62',\n",
       " 'fc9f4ba0a657fb6fe7c562097f43162406151a7e',\n",
       " '0080d3bd9fb92e022c27715c2d1249042aa998b8',\n",
       " '1fe6e8c22714c1484434a2b5b334bedf7f216739',\n",
       " '5c3f5dde0537e3072d046990501a4e71471d9ddc',\n",
       " '6624feb71eb178c0bf6ad26181e82432e634f991',\n",
       " 'd3630872afc160701dea448dea6409e50b98642f',\n",
       " '190a3e95092a129b94052ac128ce9b690350d9af',\n",
       " 'ceba28d8304f3df8001164f8edeef545154289f6',\n",
       " '3b5290a5bd5c9ac633320eed77f1a7929d72b749',\n",
       " '479b8abcecf4072618ebe266fb3d47b4b72ea91f',\n",
       " 'b7089fc75836770d44da4bc7d6200e81426b2d4e',\n",
       " '50fa3228f8067d61d3debffbe06c53deaf9dc863',\n",
       " 'db36b989b9a9b5e309143650fca156e2613c034d',\n",
       " '0c27c0ddaa4761f6155838df81a88d24619720f8',\n",
       " 'c8f255c7dc52b6b1fe70e16f28080de37702a94c',\n",
       " 'c6ea09cae9a6185a84ab731e86257d6507f5a102',\n",
       " '6a93af8b84900413c8f93f69f5e6d7c8cad9249a',\n",
       " 'f126b7f9c8ba02a0f872b51cf95d08086e0bd4ac',\n",
       " 'e3f5b91907bbd807d31fa063a9d629e574b4dbde',\n",
       " '3c7a2bc73e390cf72261519f241cd12b363c7444',\n",
       " 'a0f886e37b6c1b0dbce48aa05a6a7638368110d4',\n",
       " '1ef187d95b5c98c9ef187f3b5cf23e276bf1b018',\n",
       " 'c1edf39ded74b76134cc304dd347a4eb20bae012',\n",
       " '234e0ff5a56d4d4900f17fab95a1dbbab8c3b034',\n",
       " 'b6eed04e43ad653c06247b10ab05d03bdccf04bc',\n",
       " '357d5838127aad0439fe56f9a619d11919e51d01',\n",
       " 'f5fb59693963a0b96d112341f8f2a9f2b1d7a34c',\n",
       " 'dcbb50e1f581ab72a3070c7639c8c1a23e031c27',\n",
       " '94a10870a3585f2d71edeb6de9c2235d59f8bae4',\n",
       " 'c16fecd2784f334dbf3265085fa2fc4d20ea19ae',\n",
       " 'efcfbaace6df1f5383c167fe24ecfd918b77f8b4',\n",
       " '6380762dca586738fd20d4b1c4bbdc3722a313aa',\n",
       " 'b18e1ab81f9263fe8f2ec5bec5c54c31371f4a4c',\n",
       " 'b7b122d6d4cc409f7a22bec63936640b4cd0f334',\n",
       " '9effda40334841573b2aa2600f56566a9faf71ff',\n",
       " '148a0ab3b9bb851a56bbe926abed11c0d71fc61c',\n",
       " '2c18d80cfc792faa3d6e35c72d4b6b58ee93fdad',\n",
       " '34a95c9d94919eaef9a5bb6a863cf75a41d8e10a',\n",
       " '9835beda27f1dd6f88639b7047079c9e509afcdd',\n",
       " '9543d56a69a743e9472e1955f27bfa935ae43942',\n",
       " 'eff26d8739498efca2d32fe2e66cdbebf0569c50',\n",
       " 'cb56f2e5caa4992c01aedc47943146ee81cac007',\n",
       " '4ec1787f65505b0751777ff1f2cb40d13c0d2b96',\n",
       " 'b036ad57f875ee14fd3bef635efad5811fab932c',\n",
       " 'da9c2a65da6499124bab5ad5754dddf47fe0baaf',\n",
       " '131e6a035170ed5bd2dba6bec02649e9508a0c55',\n",
       " 'e3dd1ad4bb9542f0f8fc13f5faab78ee531ae89b',\n",
       " 'e0f39f2d61d66f276287f181454ae85cdafb7ffa',\n",
       " '725984c84bc703a34bc290f897e9f411ba3a67ec',\n",
       " '1582cbf653a14f9033c7a33567909f54e736ba6e',\n",
       " '95f0f21246953b8b6258cdf4afd61011c32a686f',\n",
       " '3cf75eb0fab0b107484f83f5d2bde3a7100f9b9e',\n",
       " '526876b78f8e3e24f74eb7a1096efa2fb92c07c4',\n",
       " 'd3d998de1e3cb950c425decbadc73af2c26102fb',\n",
       " '0bbb3db3fcea884e703a8ab09176b0b6305a6620',\n",
       " '9c1c6037784f97a987e79a92077dad0d252b7236',\n",
       " '283604cd89c223271768c0a2fb065cba3a015b13',\n",
       " '2d1c0935153f0c1139529c1c05373504028abd4a',\n",
       " 'c055818c61327ee072c403cb911793c9c49ca7bf',\n",
       " '136ee862ad53b606c8f2bea917b9705f5079e88e',\n",
       " '5644db6f53b8425eedbdb33ae2c8924e02cdf447',\n",
       " 'edea2cda674c33bb251b64e5416f5380f13d5489',\n",
       " 'a5b61a0c00daea67b98e23a9210b108dab2070c3',\n",
       " '3daf73ca2c2bb003b6f6201b1924660cc2fd75a2',\n",
       " 'ecd247bf710a751792bb6ca40d0b8bcacf2d56ee',\n",
       " '9e4f96618ed7ba4ba54c5c479ed595f1eb12fe9a',\n",
       " 'ab2dc5037a08c6bab002949e27839054d0ff87bf',\n",
       " '97f1fe02cd9244df3c903856dcb615d9fb6ec02a',\n",
       " 'ec09394ffef7660b3a7013dda8e10786bb51e3d6',\n",
       " 'ff986fb85af07d524411d6eb829e38be349df839',\n",
       " '8b3f1411a204bf1cd829342b87e5fa88bb9f12ed',\n",
       " '0043d044273b8eb1585d3a66061e9b4e03edc062',\n",
       " '45e91042ac20319b849f63a7dd00dda7d11de638',\n",
       " 'b0746b57a6c8f3cd9e52a27b0d708d069134b502',\n",
       " '41baca2bc5af0758c336c9b56d09a5f95c7afcae',\n",
       " 'b25291591ed2b529efff2b281c7be556950a256b',\n",
       " '7cf06eaa3245081fa84a74a5171810e05baf2d7a',\n",
       " '3c93efd2cee508cb7d0eb6f77bdd42c959c75da1',\n",
       " '045a46ba847ef55dc47efe6862aae106f5091bac',\n",
       " 'bc3575cce4fb95a77596f85eec9ad9e75137e06e',\n",
       " '668bff16980e8d6d020f89939588b3eddaa16dbb',\n",
       " '7364053f8be4467970465443c10cc3b39e573137',\n",
       " '290c2714886f1aaa00c374d3315381b0f0aa8033',\n",
       " '21469fd00f33c1dcdde68abf4b8da0508b7fe2d8',\n",
       " '648043eabf0ac87a2bd28909cd4a739f8b2c4a35',\n",
       " '1b08674379f9805e1ab55ce13c056b716433b8a2',\n",
       " 'b3873c777b754d108ce93d5feca0c3dff4a74564',\n",
       " 'b0db6a5fa1ab9115d590c33a7934a6fbab65558e',\n",
       " '3d690de86ca6e200f4f9cac61b004e47ad753aae',\n",
       " 'd617306cda56236d02117ae7a5fc5e7fcd015554',\n",
       " '4e6e76a1940fbfadea1d88bbcbfe4f1554f7719b',\n",
       " 'cd22fe835feb95cc3cc02e08dd38f962d4f69185',\n",
       " '254d6eaca1e97c257285000fbb96e9544dbee648',\n",
       " 'd4fa60c6535f776f6350942c3636069ed04b2771',\n",
       " 'f660fc62642dc8cdac853e68b3d2427fc967a361',\n",
       " '472602b70bca7504518686f34f5b1b15dbe21725',\n",
       " '327d5fb31ab8eb04ed90082b0ba448675c90208d',\n",
       " '7c8670a19ee742ba73283dedb53245f0489b74b7',\n",
       " '078025b470595541bbb492b1aaed90862cdca287',\n",
       " 'b0e519e2224e5ee8dd2303ed3f347e049cb599a2',\n",
       " 'f0cd43222c9a51d619357babc4d0ac984e8e45a0',\n",
       " '585e27107bd360c6d5a6519bbafe7006a1ce7cce',\n",
       " '94827ed2231500b8b66fde9844d2aebe81a4aba9',\n",
       " '72d70ffa18b4fd87526024500bf99d609b0e04cf',\n",
       " 'affa4df43665fb042bffde141c288050aa99ed47',\n",
       " '9ceee77cb6dcbc0a9568b770103698bb7603c97d',\n",
       " 'e98b7f593cd7bdd1bff632f4770505561768b9c8',\n",
       " 'fe2000f280297c40bc53ce95d703a9ca6aac19fd',\n",
       " '3855de3900dee3cecb63982884763c5dac76b8cb',\n",
       " '381bb3c674e68a97e38efe5d6f9952c7a64522d3',\n",
       " '2264a7f6a968665b70a4a06c73e096e0710fc3f0',\n",
       " 'b38c12ccd389c27706fcf2a50c2274352539fbc3',\n",
       " '08e78c0222e84457195ccefb2e371f1847670436',\n",
       " 'a510ff06475118256003b81fbc9abdaa33242da1',\n",
       " '2f09116d82b287cf3731d3baacc75895e7999203',\n",
       " '09525027db30dc3a6945da255f80b2099e239c48',\n",
       " 'bc0e1d6e96b004296ea87bde20bf39334df6d9b6',\n",
       " 'a04f2512a74dc49c895acb697ee84d9a22101dc1',\n",
       " '98f15c53c23507e24cf28eeaf7e097a099bdce2b',\n",
       " '78b714501be0196bf41e08915784eef9a6dbfbf3',\n",
       " '4a8a6f65da6c9eca4117de59c6377a0ff02a0d09',\n",
       " '232e85eda8060808405e79f0dbe13309eccddf58',\n",
       " '46795b2f533aa0d3e20b1df752cf6c18455ad862',\n",
       " '8884a33d015cfe35ba1d0df44c0cc33c8e5a23c1',\n",
       " 'eb869313e793af520e10df92b34b3e0e6a277d9d',\n",
       " 'd0c4236f95ba2b3d5cdc71fde524b26262f05dd3',\n",
       " '7b9fb5e842666cdb5418bacb81aefb0226088c06',\n",
       " '0094b25e2500306fadbdfb41d520f2970bb086d3',\n",
       " 'b8cd20475a12eda90f29fcf38b449c086e703144',\n",
       " '6a5d030ebc7695f82f5dd6e738f9aee0d24633b3',\n",
       " '713fce03f461b2c103732cff207a5de11bb69125',\n",
       " 'fbb2b307ad6be9561286c3b8a9114d283fb2e224',\n",
       " '72e0e9c2d7ebfae9d3b3cfb3fbae4981744e7a2e',\n",
       " 'c229a350536a9e087003fb14e9e52aab0245d2ec',\n",
       " '9d9ad4fc953115d139ae6bad7522c4e685022065',\n",
       " '8fc871d3cd3d6bb6981bebb026942df214442035',\n",
       " '61fabfc758361c0d84c86c3d735de857d9537b94',\n",
       " '3f01702c71fa172be560638b826441239c956227',\n",
       " '20154fa12ca525582cd1b42019927e8f7816fb51',\n",
       " '4d754b08444d19e6df8559afad357bb13be89bd8',\n",
       " 'ae098292360fde9d65ff12d2beef927dff6c6077',\n",
       " 'aeae7709de0f5cc24b6e055115c3105f44da9c61',\n",
       " '03cff1790e31683b4367f41b09f7efbbf2dc53e6',\n",
       " '625afb5a588fd265e15b652a437ef9f521682aee',\n",
       " 'e6dcc692a39166e78808eeb7486b0fa926174c07',\n",
       " 'b54c80297872e9ea4036868879e47e9e6cd7e79c',\n",
       " '27052d6e2693fe9dfae4a746408e16180ed57243',\n",
       " 'd90b2c4decfc17d7349325f437a9ab6082742c24',\n",
       " '7bdceb3b947ec934199d0c97fb9ec5a0e2bac776',\n",
       " '1b4ab28a9391d516e1753874c73d2d244104f825',\n",
       " 'e6f54fcf3460165fb558903e975a526bc98e4eba',\n",
       " '3aeb264d986c987ced699cb2c711eb022b697d5a',\n",
       " '2bf3f075175c307923fc400707b30616a676419c',\n",
       " '07567f87e6aace24f9146427f637ffba089ae28e',\n",
       " '6c18f2bf81bf9b00f2a628d540151912b2dfe980',\n",
       " 'bd246a6d199cb961f7c4c3421d22324aface4ff4',\n",
       " '5d38ec30f0a4ea9b504808bc6bb2c8793db72e7a',\n",
       " '58bc0f2712ea1ced1f886c9fdd17b8cf014e66cb',\n",
       " '7037e1fbb04e2f1299b12d0f5c7ee9a6d6c8973f',\n",
       " 'cf665ea806b5a1b271c377e3826bf9d8bd0ebf9f',\n",
       " '3c988a00ed94308e36a8814b391ead01480efb60',\n",
       " 'dfa3dbdc7b28b9075361efcb919d3a1b19636edd',\n",
       " '1c202f2c3924d86f516deecda1d47f55d08337da',\n",
       " '4ada63940f8817b1305316a9c44cfbb5bc39b762',\n",
       " '86fca5af635ee9425e3375140fb48cbe6d429411',\n",
       " '228c21cccdcae8d5b8e512fb10fade61422aa4e1',\n",
       " 'ed437ca7e40c57b5dd02dea1b7f11fa221590b60',\n",
       " 'e0fd364797eb02a280e25e2649dfe6d6c0d33e05',\n",
       " '7cdd50888f639629380fa9906d73f97010158127',\n",
       " 'bafffaf17be4302a4a6e79b3f65b9238ab00c258',\n",
       " '084df43c5fc9b52ac507f090c6d892b69e5b0999',\n",
       " '9d63191505e7cdb2dc1436228923fc97515e2ea1',\n",
       " '7248a954a71c38b84dcc81be1105e231728286fc',\n",
       " 'e59cf9611fc13cb4b124e5677ba3290a3c6bf7d7',\n",
       " '8da224b5eba13cee28c3a95cb3c1e3c4d26e5a3a',\n",
       " '24ffda519bebb75d54ef12eea8759ca4be138262',\n",
       " '9f08226b10da36589025f4b568fa4734be47f0c1',\n",
       " 'ecdc56575a3711cd6b7f3207098bae3b82fd79a4',\n",
       " 'a50f2af4833bd314b5664e5f30a6d39792a73f5c',\n",
       " 'c4e5c3ffe90100931729abab4aa1155d8c145816',\n",
       " '7c3ebae910203c1f867fad1b2c14c555fc3e263c',\n",
       " '165d5a3cf0dc10f8229b0c6fcf149cd244cbd73a',\n",
       " '7321a41ee2054c07a18ede4e04a420621cb33087',\n",
       " '362a6ff84fb689abbd3ca367fb077715c3dab5e1',\n",
       " 'a09daa4101801d9b65bcc053116505b0022e0640',\n",
       " 'cf410c7fd76487d1244517521fbe521465e75310',\n",
       " '3eb3c6212432888ecf0f2d3d2dc4a2fb6b4f17c0',\n",
       " 'f489bc214786cc253c014ae8e42d1c7e0c590138',\n",
       " '13f67fd2f646199f7deed8d013bce39056332123',\n",
       " 'b56db50e2116c867bbc86aefcdfec72c74e971ed',\n",
       " '0498793e595e90458c63f2f26b7bee68050cfdbc',\n",
       " 'b4fc44eabb1f640124fff98990c85dc5c9984aee',\n",
       " '5e442899227a0032456c65aca20dfa2f6fa6e7fd',\n",
       " 'f50562c5a5e9f206dba3c559e6b4f5b65aadf7f4',\n",
       " '6e6e7f0218c6b5affa20bb27bb263517a113f7b1',\n",
       " 'f481624491ae43193cf1229d54d99e8ddcef9479',\n",
       " '47772f3e98d8c61bb9782531a0338ba85f27bf3f',\n",
       " '6049f1c151fa95bc40deaf050225dd7a00880171',\n",
       " '3b2a2a17eb789fee6c46ff6cdc7b6e794f7d545c',\n",
       " '15f3676a843fc2b08f47c1e0c49d78f63519caeb',\n",
       " 'a63f9b0555e4ec19bb5a38d4e4f282ed9bbb8612',\n",
       " 'a765ae8d7eae55875cb832428c9082a6e4e4ae82',\n",
       " '4dcde213ebf72aff0935b2714faf7fd691736c5e',\n",
       " '5ab1eceaa469bc34270595cb9b5ea00e6ed3fe71',\n",
       " '039ca136c69c998e9a2677259d9de2941a13304a',\n",
       " '9ea1a900f1243ce3264700e9eec29b402892674c',\n",
       " '72e1f47796b2283c03f3d7aa4f6044592f8c198d',\n",
       " 'bada3cc0577a226694617d77f95ac443c6431969',\n",
       " 'cc2e7594049d91279f4e3d69d27b440ddd8de1f9',\n",
       " '60e51d3d13b027492b9f3b8693c58dd8b5385ada',\n",
       " '7266d5a2b588f4bd08e301590247156d59a7f227',\n",
       " '5912189e6ca5053f8e7a2de341c90341f737aa7e',\n",
       " '801bdee04f820908695b58d1d23a04d3ecbf103f',\n",
       " '7293d0d1c84570d510be54e720daecfcf7a77990',\n",
       " 'dd84d0eafb5b8a5eaaa188b96cee26d26cafce9b',\n",
       " 'caf95fa590e7cbb34e10314bfd93beb44e35caab',\n",
       " 'd25529180f950874e0b4f62a976959d683a2ceaf',\n",
       " '9a5a8345c449ca529d68c57e0b52b8f90b1297bd',\n",
       " '2cf45a5a4224dbf38d537eace1f6d49327e3487d',\n",
       " '44d0aa144ed562f3811296688dc32d5add62678b',\n",
       " '14d04f36cb13550aa7769b61a079fa54031a21eb',\n",
       " 'bf0e3ed6e8fa4adbfacbf76e9fa2f5aa1ccdef19',\n",
       " 'af6cc29d9bc7f9d84ded3d7ad30509a945e425d3',\n",
       " '568b4324519f51485f598d1819341d084dee43a8',\n",
       " 'd9a66cbdd96e0bce18bc370c96ef5e71c5a7a27d',\n",
       " '1363643127b0fd76d3c6b848e07cfc2ed3f75328',\n",
       " '16f1872aff5293a40e4e8047a141b18b3a35a28e',\n",
       " '77b8d67e208f5ba81ff129e0d4ff7e1b73c45482',\n",
       " '519b82201745afd08726b2c3e87cad65f8409da5',\n",
       " 'd31e34a1b5d8447f077c8edd36938c68470db017',\n",
       " '80cdffc26a496e89e2d94c7478749adbabafe180',\n",
       " '5caf6fa187000ae494d50b2ed2ab15911b7da3b6',\n",
       " '359aef6ff466266be0ea21eb665fb887cec16713',\n",
       " '06c4f9daabdf5ee16fb4b581a954e3909d9f17a0',\n",
       " '5535ed240ab00c780ce2aa42f1fbe13ef24b9120',\n",
       " 'fff6f848d352c486049ec8046bc44e900fefb2e3',\n",
       " '0dc510092156faececbbdf95cb57ba32ea03da4d',\n",
       " 'a2a6e262098539eb875a26800d9f6d3d0d5d1875',\n",
       " 'bbe2e49fac5f007f3313b35f9b264bb4219aa67d',\n",
       " '8977825a7fbaa1e4d672a361e7122b48785f93dd',\n",
       " '965d0f74c85dad6e4119bee49d3026fbb91c80b7',\n",
       " '91098f6fe46a21565bf0cb06fe960cdb2c3f5e38',\n",
       " '417cbe38a2f8a93d0827cd67f4c42076921d6050',\n",
       " '5fb9518f07541bb5ee6f8f9ef9d374416ffeecc7',\n",
       " 'e0d4c25b7e80c41abb0a816770a803b29746b2ee',\n",
       " '76d3c799a37ac490dde80c4ba90fca49e1cc206d',\n",
       " '0276133204889a33c9fcc887c7034fa8b78c9ce9',\n",
       " '7151947d17c83f7764b115d4e21a982738bb12f6',\n",
       " '82742db34be3eee4be2483c14909b1e62e3912ae',\n",
       " '427b6f00b5d8c62a878fc84e4870cce9a7f4cde9',\n",
       " '26f94b094ce4d39665edddef1942f04dd041f363',\n",
       " 'b68d33bbec8af949490bf52a5b8dd215c2b37590',\n",
       " '2a92147a3684058e383a4b403c25cacbda28022b',\n",
       " '37bc284af4f2f731792f44339253abc79c6d2782',\n",
       " 'a6b05fa393f9404c906099e179ca476294068394',\n",
       " 'bad0e9f737316570c33138d5cc95cc233cd937ab',\n",
       " '313d6762ff0c7e18ed7af39482b04fbd2d280bc7',\n",
       " '997ebf63461efa3e7cb1fc3014fc957659705202',\n",
       " '02974b9466bf593193d05ada669d7ff0567bb657',\n",
       " 'c5ad260f605645f5fe9da46ff2c2738984ef9ff8',\n",
       " 'a0e91efcafa0a743be0342254943e8d31cd36558',\n",
       " '8dad44cafa563c90b07152ef2038e530b19edb31',\n",
       " '6d1a2055895ffdcdf57aabcaf436f176b1650c50',\n",
       " 'bb11206963e831f1652775d26e3e5e48634a4545',\n",
       " 'c54a28ed68b1d30d44a3bce4033fa3e74103a33b',\n",
       " '0d9c772336146ec59aac5659f00313e1a63f8fe6',\n",
       " 'd67640da9a65493bdc6c39501c483b723c722fcb',\n",
       " 'be459dd25c7c6092a050c722c69ddd3d34ce0f94',\n",
       " '5080b4cb5df826f641dc6602e27dcca510ccada8',\n",
       " '97d0e0033e74b01cefd3c71a23aa1fec45a72296',\n",
       " 'f32841f22efe8c2bccd0d04a0c0b47e6ecf86b61',\n",
       " '4b4c47bbc2473d07c8ffd4d940a099e7546b9f25',\n",
       " '51a24b884e01df122d889f1133e7727e2478be24',\n",
       " '632bcacf9702ee34c5424827ab7e5449dd21ddcf',\n",
       " '5813a786b66940f45bd1083081448cc53edcadd4',\n",
       " 'f2f78c95ab378a31bd35dc1de84e0ec75eb7ce1b',\n",
       " 'a014fffc3ab64385d716b0da1cd66fcd75969083',\n",
       " '6ff336c5f099acbe84cf7cfe5f68bd5ba672d6b8',\n",
       " 'ae8ccebf81f9909e96af7c707cca66e5ed716b1f',\n",
       " '95b1f681cfe2bb33c6433b07cd38f0ed3ce3667d',\n",
       " '29ee29dff88547e7bce5bca49ed91282811fda0c',\n",
       " '90d553a545579b2e27d0899c558e004ff33cd0f4',\n",
       " 'dc45028785e8308de18df3a42fe11fe571da2cdc',\n",
       " 'c7fbc3c2ef9549d8a6531237b2e9d09cf09a6567',\n",
       " '06cdbcb7d40c21ce5e10c5a2bc1a1583e66f49d7',\n",
       " '9786f61a3f8175ac0619b3c0d9910e1ec06fa182',\n",
       " 'ccde33bdca9b5c7f879f105a4596e5b1ad3f9408',\n",
       " '5d2c1aaad487ae9b76099721947c8948a57ebba3',\n",
       " 'fb4c3d10a9ea514f135f1fdbc5b535bd7955d888',\n",
       " '386d7389837dbaa87d3b0119af4d965b197651f6',\n",
       " '621c8306765f418717c3de4497c2dace5f8a9c73',\n",
       " 'ff6bd29c8ecd325c4edddb2b050d5f4919db6112',\n",
       " '2a423f15be9162d939e0402df6f74c31ee2e9486',\n",
       " 'a4b7fac91178fae8acd884c584fbbb525b883c65',\n",
       " '8a0731d1284b815e6139b1a40a4cc1f3b68dd4f3',\n",
       " 'a1bff76ce360e8990b0a4ee2a5228a6e6e63d9c1',\n",
       " 'f084dcc7e442ab282deb97670e1843e347cf1fd5',\n",
       " 'aef8022f38dee0b0eadc00eb8a955384b87f3626',\n",
       " 'c96149ec9735f620b4acd1c4da5c537ed2175619',\n",
       " '42f17b849628fb1db7c74cc15899bf9b9c59de93',\n",
       " 'f9a61ae749c3d53492b38119c9fbe5f0e448b52a',\n",
       " '428d1091cf63872ea81cb3c1632d76c4813748a1',\n",
       " '6660c114ae3ea21b9a1551e9253d767d5b7ae9f7',\n",
       " '578def3a338542ea59f0251c8f5fa32acc37a83c',\n",
       " '111968f7fb102882c193488965e41d5d0f0103dc',\n",
       " '345180770287ab9188e2dcfa2fd2ba0cd58214bb',\n",
       " '5a2360e93ec038502133c97cf78a114c012e5df4',\n",
       " '137b25ee1545c4b01332709ab1e24dcf935d6f96',\n",
       " '1552cf77b6385950e197c67c2ed55ed6e1879ced',\n",
       " 'aaaaf1da07387bece9bfca99582b31ebe33b1fbd',\n",
       " 'ee92720c10a0cb395a8c1fd940dc0d12c44577f1',\n",
       " 'e0ec066b46e51073a3ccfbc05fe626d0c93f8b29',\n",
       " '173ce92cbc6f228d5bfbdbd7ba7137523ebe36bd',\n",
       " 'e489f92a7fda5ea493642945b61c85b6bde8a76e',\n",
       " 'c46b171cf1c9b3e37b52549cb2c4356b5058f0c2',\n",
       " '2064f80647340c485e6143344cfe4223e8b08e2d',\n",
       " 'e400bf79de752ad6ef0811e6dcce76a963ed09d2',\n",
       " '5678f2e8dc1710c96b5f4e5ae8d6ddae9c5e6580',\n",
       " '43fa93dd99dc6f9a8204dc22c79fb80ddb44dfba',\n",
       " '60e49bb744eb678ae1ac1fb9bb920a5b01954041',\n",
       " 'c92206983f9523a96c8ea02210f70ee5640edd14',\n",
       " 'e05de9e9983b9e4ee063afa20cbe46a38243b9d1',\n",
       " 'bb2c8c79dafcc25bcd68644bf23a4a368bcbd01a',\n",
       " 'b6b5b7639a890decf17cb745a6551bb3cf8f7b1e',\n",
       " '859410b0052a50da3868ab631c6e1a58ff3d4c49',\n",
       " 'ecbb9acde61473bf4f5942d6bbef461eee034a9a',\n",
       " '2e84c01cad3e1f37940539a14a1f043fadf5b36d',\n",
       " 'f1548a1bcf0d81c278b46d24408dc7e63d9ce09d',\n",
       " 'f5ed30bc906fca5572d04c5258cfec3d91093058',\n",
       " 'a3ee4c7c3f02fd1d582f46a2e8965b2ea5da5ea8',\n",
       " 'c1f6fc9f26417df40a63350abb544265516ccad1',\n",
       " '2911b1cc77f7c2bf1f8743673d3d2df7d7993aee',\n",
       " 'a7a5f2d3cff5287f3c65c24cb77de1caf5872687',\n",
       " '29da5f5d00af5c47f9b3a0a49d561f8caffb0cf3',\n",
       " 'ff9d9713206da30022af4f9095058368eeb1f3f8',\n",
       " '5a55cfceb7365142481967c11c4f70b359a1e4b1',\n",
       " 'f33e644d058bc8373378c115878be06346298687',\n",
       " '3f962f74bcf72c06a4f0941b8da40742024dae3f',\n",
       " '3282216c38dff96f0fa23f520317ad7572fa1da2',\n",
       " 'a3e8cfc509af101c522d4193c8917d6e8f68b37b',\n",
       " 'a452e108006dbd4db3375ce517ab1b7b35425197',\n",
       " '8a2b2c72af5a57c21d912a3f1e543bcd75345883',\n",
       " '516c3a32adbf893d696dec21d087c848b85b3940',\n",
       " '4a82252b3b4f4f90c0c0bc98ab7d817798d48c98',\n",
       " '423276fb8fbadb4e79c066d4fe8d121c308bc465',\n",
       " 'd31b3d462fe4af4bac27a166ddfdf39b03c8e911',\n",
       " '49f04c6a31256ab8fb38bdcc329a6744248287fe',\n",
       " 'fd59786161c6f852a77338bc0cfe577bde28458c',\n",
       " '4a79e0975e99b65456d69c3310e842402202e64e',\n",
       " '7e216d0c059fbbc7cace4a9d12a48482ad781366',\n",
       " '59f43728701d0747e4f79309a62d2416b3564f2b',\n",
       " 'c590a12be43ec57a2dde2a473802c5e47e354434',\n",
       " 'e73a6a49cecb243ce24b57182cd354db7922d7e0',\n",
       " 'fdd06eeb020a963754fee36852a5918a93f7856d',\n",
       " 'a15c6c90166e58a98c9d77cbc91709e106287822',\n",
       " '2c7260dadc28671d3ba4e14120b63a5232765805',\n",
       " 'c5e6b31b9d1b7d0b62564566e8deaf48a03a8f32',\n",
       " '442483b51febb0ef0cedea0d4283da1cb8c7ec01',\n",
       " 'bc7219efb063d1c0aa002e536f9503f999fbcb42',\n",
       " '898880683d9a962428033fd7500f74647c4a36f0',\n",
       " 'bb6be6d67dbc92ee528f4c09ca1643eeda4316bd',\n",
       " 'd7a470f0dbe60172ed2a21ce65b9903b4825e3a3',\n",
       " '8ad769c159358842b5ccacbd383987335844ad86',\n",
       " '1cc437f58a3eda9fc4e73c1cabb18964d2a7e705',\n",
       " 'b41f3634a764bf08cbc4588f37d7d422a02fa541',\n",
       " '621832d547930a8c23859fa443c2c9c38ff07e42',\n",
       " '721b3e6ae8af24a4f625540bbb6e2c02fb77b491',\n",
       " '953d78fb78ffe2bd8db79fac2390d45bd28a4952',\n",
       " '09561ed721369a7a9e077f7bea9022499c12b13d',\n",
       " '95d1135211f245b882e54f2842ab328b89066452',\n",
       " '78f939545e7217684295ab63900119ab1ebdb173',\n",
       " '4185d64883b5f5a49c9032005bd17bebe47b7f90',\n",
       " '2aa0a28e4246f367ffe104837e755b28d2021f45',\n",
       " '6aa7ff7bbaba12fef375a84c5ba297d9f848541f',\n",
       " '5e77269850b7b4bf57514e226ea66cd48ac31db5',\n",
       " '7d92312dac3822da87cb86d3de9fe818e0ee2874',\n",
       " 'f582baa8aa4a5f3af8356e86df3d5c29ff426c85',\n",
       " '70501576998f6532d594f5f7198b7473bc55a32a',\n",
       " 'c838ca724d9916867c83ad89c823ca61027bd046',\n",
       " '3ce427cdc17933b5d2a4834fe316196eec3c7113',\n",
       " 'b31b7e11a6cf05cf0132d92bc09ee5911e7672f0',\n",
       " '9f234fa9b0d6d7c0809106111104ad8843f3a931',\n",
       " '24fc42faaa7cf3f1b483806e77056a1d253715c5',\n",
       " 'de01b5c28a44e4d80b38e389342275d0353673e7',\n",
       " '74b1c6107f2da5209abaa2521b5a93d885e74cf5',\n",
       " '96fc4a051f2ecc366f49ca34902324d059577798',\n",
       " 'f580401828d99aa0874fc10a731aa34a8cfb69b3',\n",
       " 'c7cb43c8bc34a2bf0c4a4fcc68525880e194a781',\n",
       " 'e7e0c619c9f078101beffb8b3822c65e95995856',\n",
       " '53abe67758ae950feb6f7ee22501a345f597e98b',\n",
       " 'b23a403b1daf18cd8a06665ad3111e92453bd61b',\n",
       " '5ac39b49492fe7a6f53ad50d840e4ce8ed579e77',\n",
       " '18d7f4c8e204349b39a037c19874cff1b34d0f97',\n",
       " '9da90a751f76421e3069d40bd2eae1891f778965',\n",
       " 'a70d334286efa96684bd35e383175063c285ab8b',\n",
       " 'dd94afcee3d0c4dbc80b40435ff4fb65487af3b6',\n",
       " 'a50e01474c7fd48a938089de57f4ad1c7eaa3c1b',\n",
       " 'b603913495d3cbce15eff09e645007cdf42a6e1b',\n",
       " '8275311de465e4fdbb25ebddf76302cd4603c36c',\n",
       " '60bb4454a34fb7642d6805ff6261abc91809a135',\n",
       " '2c22051b7eac78f942e796dd838481e126eebeba',\n",
       " '3472bd9fd14feea62d7520f8ad10a999921a9b1a',\n",
       " '46768e7c839a7f4de0d5a73d25897f948b4eac30',\n",
       " 'cec3e151f7ac8b3a8766ac9fbc08c9429eaed31c',\n",
       " '4a25649bbc72d2ef0e57a19221b21c486271bcfd',\n",
       " 'f2e0d9212887c0032b1887f34ee3b3042510d7f9',\n",
       " '0e7db64290a2be55897c1cedf4e15f2b5a86c06e',\n",
       " '6abb30ae61aa5e41f16a28b9437940d5d76d745b',\n",
       " 'e2bfc8942cb7075af5dea8abbb281e0a7ec146a8',\n",
       " '3053729be9dadd6180b0b30ed2048dec12bd0401',\n",
       " 'c761570812ca07ce01397ebabc220556f51e786e',\n",
       " '9599f421f59c01ea41789ba70c79330cdea8e1d2',\n",
       " 'e560eb28bb3f35099d2632f80adaa14436516474',\n",
       " '24f4f393fb59daab089844e7af64e913d2951034',\n",
       " '57a67fd5273fb4aac8505d1ae7b32c2582d12df3',\n",
       " '9200568d5a5c88f0f1b203240e94e90f96546431',\n",
       " 'cd06185d94b38abd3c8209a45897a589da964bf2',\n",
       " '71ff6abe3023a4c48fd6e6c54265bce395b88397',\n",
       " 'cda93f801384df590343a55a0c01a418a67684f4',\n",
       " '4485f96496a208b7e77c9f53a8feb468483776db',\n",
       " 'd05db9cdccfa196d13b8eb58b09208937911d70e',\n",
       " '3ab8bf523ee26f6e2f7d3cf11d757f8b449d9dcd',\n",
       " '6b8791fc696da0f017d4c3b92ed3aabc4a470e68',\n",
       " '83a9e757fc42183eef4f36f50cf84044f7b780fb',\n",
       " '1b58422e266ab9339c919119923229d080f27360',\n",
       " 'b18ff7b65ed56a87c0c0de1ee312b8a0ddb51e4c',\n",
       " '8e549a3356d7bd16111c4f682ae12a21ad4c5633',\n",
       " '3c6b88ab8a6694c82e04480184ae76dbe302c95d',\n",
       " '1c956bc94ce590a7ca6c2ace2c0766bbbbfaa0e3',\n",
       " '740a68304e37d9c1e1ab4bf433594a5a47179c61',\n",
       " 'bef31fcf85c2a93a2eeae36e5574372c3aa1c8f1',\n",
       " 'a6f36e3233319626ec737895c1d52dedd7aac0bf',\n",
       " '57935bbfea5e9eb4e25ef185f059ebb4362714d8',\n",
       " '2a0dccbe06cd6c48fdafcae920f220b3fdccab41',\n",
       " '3222436ea52738f6e055fcc74fdf7ed9da92feef',\n",
       " '59c932b49a6498b87e3d43f42038cef789094b4d',\n",
       " 'e8fdad25eff3afb451c3d0d95f11cc4225b84895',\n",
       " '8171299fc5073dfe98c4c56d5760ccb5f802f8f0',\n",
       " 'd729eef3818cc854484ab0550aa245e5eaa2b72e',\n",
       " '4a80fc8045f1a85c35e038160bee95c2be0c2fd2',\n",
       " '33fbcefc12a3f06bf2e20b88ac1b69d481746250',\n",
       " '71e8fcabcecb3f5ce3a699f95a618bd21a68bd76',\n",
       " 'b70352cde76b15b3f8eb60a492ad27bd0a8c4a20',\n",
       " 'dc37b885f85c99c777d87658f2194b71e17c237e',\n",
       " 'bdffa5902b25099211f840d9e3b8cc7e16f54632',\n",
       " '3b1c5eb21d8e235d526814f8f6b5d3a1a3831a4a',\n",
       " 'bda64532f4a16e2885868385170f8bc89ab8f789',\n",
       " '77ee6f11d8a0a54182e837f5c19ffedcf3d07e01',\n",
       " '419a452d263cb907fb5c6b3e47c96aebabb78d64',\n",
       " '7178530e60694f264910bd591cc70a707d498bfa',\n",
       " '50f9ea2210efc3c98630da060b4254036f83c516',\n",
       " 'b09362a1f23af2b6603e7b92b58003b7f9719840',\n",
       " 'eee528b0235a1312f9d280b603d91e241a38bb40',\n",
       " '6a80b22e84d2692545c6f11d7cb4c96602a25c39',\n",
       " '87b2f2205b9dea38eeaeddd6e3ddbb6e45f542ae',\n",
       " 'ac50b9faf4e811ff32603511ecdeee8e8d1eb897',\n",
       " 'ae6773868c2850e5c8b8177735b19d01341a53b6',\n",
       " '4a1b2f689df4d8dd7521873bf16eed8523cf968d',\n",
       " 'b2ff10bcb290b39b39f3f8a2206555baad677a5d',\n",
       " 'ffb9c96d6bf428d73c5bcbea0524288661f0f78a',\n",
       " '7d9e398e0de06f3fc877f3d67489e798b7c31c20',\n",
       " '39c58dbaa3f69e91698b2efe6d3b12504aa3904d',\n",
       " '6a3fa8ed278df0d05c5e009521de11c72308f60b',\n",
       " '2aff265697682af71816737a2fd85f1d9bc9a0c4',\n",
       " 'cfd27fc9582d326ef3b09be615d649e3b6e85c61',\n",
       " '1447cbaffd1c03718da0534b7bfa97435cd3c78d',\n",
       " '6ed77e5ebba3c0f0b10c6e0af3e8df28b57335e6',\n",
       " '3aeb4a0f6893b933dc061157d196e5618380c63b',\n",
       " 'ed745dd99691994096cdc638c5398cd65a2c9539',\n",
       " '610cdf0a90ee7272f51c5531375a7355768f865a',\n",
       " 'e98c156b8632c84d14cfb03a1614b36563479bed',\n",
       " '7608309e163c6a51f211cc02d6e34659feb64418',\n",
       " '6799e37b5fda2d18292b86b6a6c7ba1eb185d018',\n",
       " '71d134916c750fb6f875d3936a957b27a5df72bc',\n",
       " '68b905ee32b8aad54ae9006fc7aab007c63e9895',\n",
       " '6a83b934090801454dc8e81a559ea4c20c0ef3c7',\n",
       " '7b1a62bfe713b9fe81bb84360d31590a6083b499',\n",
       " '5a616c081f34c8ead17742a91a121ae167d77e17',\n",
       " '9033d46664d7ded6d2cd79ddb75c2c36beab4802',\n",
       " 'fe5d9afd46890e2338d8d0febf8eaa8bd2166d60',\n",
       " 'fa4e07fc8a5200929c2299116a7784db01fafed0',\n",
       " 'e8679d8c149ea6d25afb92e8a4f4b622dadcd872',\n",
       " 'cf533451f25297fe0d1a6006fc562568358cefb6',\n",
       " 'dcde88584805e783b69565364975fc31053b6b4e',\n",
       " '2e8fb12daa448c956d8b7b348df5e46d29c5f51a',\n",
       " 'd47377d0da60ea7aab2522c204ad65753b0545fe',\n",
       " '4bc632592ca471bbff7effd71f3a9f930913f3c6',\n",
       " '22b6982820cdeb330dc2086ea69713f6e109a671',\n",
       " 'f6b7e95bba8fc989701322484c6318774eb1ed6b',\n",
       " '4b2a69ec1289a04f3a94db719f855631f93f2d54',\n",
       " 'dd0fdbeb9f3ba21911ba2c667ff85f2e43c696a6',\n",
       " '4cd3f43501d49e775838473a3edafb55e57d08df',\n",
       " 'a0f13b751004e29186fa070fdffa2e876815e41a',\n",
       " '7d4b0efca3cb904640b8fd489fdfc413a8fae264',\n",
       " '4edda548e89522b4df20e723e29c63f309857b10',\n",
       " '4564dfbcb341f91caefc5a880a127a629416c1c6',\n",
       " 'c3d69c9288a27fca5044723a7c9152d1f58b2795',\n",
       " '79a0f19a4e1f786f96d4882f85a185630004aab0',\n",
       " '1f16c5e424d5b436085329d94649118ac319e748',\n",
       " 'e9d32ac6db6f898b035aab6d4357342c3ae2442b',\n",
       " '805edd3fe7b8a903e0c308d2d70a941e5d8e00fd',\n",
       " 'a2fc621a71cddf750d46902684501f5f18eba3a0',\n",
       " '10f5615e0997e9d30c6a42a2d91e8cc9e896b8f4',\n",
       " '36e5a1e939633cc6a8b4fa2dc7d70f9dbe5f0ee8',\n",
       " '881739c1040a5d2fa7024857e21aec68e4d25230',\n",
       " 'aca7ebdfaf454b97c94d94e2d4267f347b1d718d',\n",
       " ...}"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper_ids = set(df_meta.iloc[:,0])\n",
    "paper_ids.pop()\n",
    "paper_ids"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# biorxiv_medrxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 803/803 papers present in the metadataset.\n",
      "Examples:\n",
      "0015023cc06b5362d332b3baf348d11567ca2fbb.json\n",
      "004f0f8bb66cf446678dc13cf2701feec4f36d76.json\n",
      "00d16927588fb04d4be0e6b269fc02f0d3c2aa7b.json\n",
      "013d9d1cba8a54d5d3718c229b812d7cf91b6c89.json\n",
      "01d162d7fae6aaba8e6e60e563ef4c2fca7b0e18.json\n"
     ]
    }
   ],
   "source": [
    "biorxiv = '\\\\2020-03-13\\\\biorxiv_medrxiv\\\\biorxiv_medrxiv'\n",
    "path = cfg['data-path'] + biorxiv\n",
    "papers = get_papers(path)\n",
    "cnt = 0\n",
    "# check if paper are in metadata dataframe\n",
    "for paper in papers:\n",
    "    if paper[:-5] not in paper_ids:\n",
    "        print(paper)\n",
    "    else:\n",
    "        cnt += 1\n",
    "print('There are {}/{} papers present in the metadataset.'.format(cnt, len(papers)))\n",
    "print('Examples:')\n",
    "for paper in papers[:5]:\n",
    "    print(paper)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "paper_info = get_paper_data(cfg['data-path'] + biorxiv, papers[10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paper_id': '03ea3a614b56409d3f099c9ad764864293132540',\n",
       " 'metadata': {'title': 'Live-cell single RNA imaging reveals bursts of translational frameshifting',\n",
       "  'authors': [{'first': 'Kenneth',\n",
       "    'middle': [],\n",
       "    'last': 'Lyon',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Luis',\n",
       "    'middle': ['U'],\n",
       "    'last': 'Aguilera',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Tatsuya',\n",
       "    'middle': [],\n",
       "    'last': 'Morisaki',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Brian',\n",
       "    'middle': [],\n",
       "    'last': 'Munsky',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Timothy',\n",
       "    'middle': ['J'],\n",
       "    'last': 'Stasevich',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': 'tim.stasevich@colostate.edu'}]},\n",
       " 'abstract': [{'text': 'Ribosomal frameshifting during the translation of RNA is implicated in both human disease and viral infection. While previous work has uncovered many mechanistic details about single RNA frameshifting kinetics in vitro, very little is known about how single RNA frameshift in living systems. To confront this problem, we have developed technology to quantify live-cell single RNA translation dynamics in frameshifted open reading frames. Applying this technology to RNA encoding the HIV-1 frameshift sequence reveals a small subset (~8%) of the translating pool robustly frameshift in living cells. Frameshifting RNA are preferentially in multi-RNA \"translation factories,\" are translated at about the same All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint 2 rate as non-frameshifting RNA (~2 aa/sec), and can continuously frameshift for more than four rounds of translation. Fits to a bursty model of frameshifting constrain frameshifting kinetic rates and demonstrate how ribosomal traffic jams contribute to the persistence of the frameshifting state. These data provide novel insight into retroviral frameshifting and could lead to new strategies to perturb the process in living cells.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'}],\n",
       " 'body_text': [{'text': 'Frameshifting is a fundamental biological process in which a ribosome translating an RNA slips by +/-1 nucleotides, resulting in the translation of an entirely different peptide sequence from that point forward. While frameshifting is generally detrimental to protein fidelity (Belew et al., 2014; Choi et al., 2015) , the process effectively creates two distinct proteins from a single RNA (Clark et al., 2007; Meydan et al., 2017; Yordanova et al., 2015) . Viruses exploit this aspect of frameshifting to minimize their genomes and to successfully replicate in host cells (Brierley and Dos Ramos, 2006; Brierley et al., 1989; Caliskan et al., 2015; Cardno et al., 2015; Mouzakis et al., 2013) . A prototypical example is HIV, which utilizes frameshifting to translate the gag-pol proteins from a single viral RNA (Guerrero et al., 2015) .',\n",
       "   'cite_spans': [{'start': 277,\n",
       "     'end': 297,\n",
       "     'text': '(Belew et al., 2014;',\n",
       "     'ref_id': 'BIBREF4'},\n",
       "    {'start': 298,\n",
       "     'end': 316,\n",
       "     'text': 'Choi et al., 2015)',\n",
       "     'ref_id': 'BIBREF13'},\n",
       "    {'start': 387,\n",
       "     'end': 411,\n",
       "     'text': 'RNA (Clark et al., 2007;',\n",
       "     'ref_id': None},\n",
       "    {'start': 412, 'end': 432, 'text': 'Meydan et al., 2017;', 'ref_id': None},\n",
       "    {'start': 433,\n",
       "     'end': 456,\n",
       "     'text': 'Yordanova et al., 2015)',\n",
       "     'ref_id': None},\n",
       "    {'start': 574,\n",
       "     'end': 604,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 605,\n",
       "     'end': 627,\n",
       "     'text': 'Brierley et al., 1989;',\n",
       "     'ref_id': 'BIBREF6'},\n",
       "    {'start': 628,\n",
       "     'end': 650,\n",
       "     'text': 'Caliskan et al., 2015;',\n",
       "     'ref_id': 'BIBREF9'},\n",
       "    {'start': 651,\n",
       "     'end': 671,\n",
       "     'text': 'Cardno et al., 2015;',\n",
       "     'ref_id': 'BIBREF10'},\n",
       "    {'start': 672,\n",
       "     'end': 694,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None},\n",
       "    {'start': 811,\n",
       "     'end': 838,\n",
       "     'text': 'RNA (Guerrero et al., 2015)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'INTRODUCTION'},\n",
       "  {'text': 'Although frameshifting has been extensively studied in vitro and in bulk assays (Caliskan et al., 2014; Charbonneau et al., 2012; Chen et al., 2014; Lopinski et al., 2000; Mouzakis et al., 2013; Ritchie et al., 2017) , the process has never been observed at the single molecule level in living cells. This leaves many basic questions about frameshifting unresolved. In particular, it is not clear how heterogenous frameshifting is from one RNA to another, nor is it clear if single RNA continuously frameshift in a constitutive fashion or if instead they frameshift in definable bursts, as has been observed for transcription (Lionnet and Singer, 2012) and 0-frame translation (Wu et al., 2016) . Finally, the localization of frameshifting has never been investigated, so it is not clear if frameshifting occurs all throughout the cell or is instead preferentially localized to specific sub-cellular regions. In the case of HIV-1 gag-pol, for example, previous assays have shown that 5-10% of translated protein product is frameshifted (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . This is thought to occur when ribosomes translate through a specialized frameshift sequence (FSS) containing a stem loop structure that slows down incoming ribosomes and causes them to slip back one nucleotide on a slippery sequence preceding the stem loop. Whether or not this occurs constitutively and with equal probability on all HIV-1 RNA or if instead it occurs on a specialized subset that are in the right place, at the right time, and with the right factors remains to be determined.',\n",
       "   'cite_spans': [{'start': 80,\n",
       "     'end': 103,\n",
       "     'text': '(Caliskan et al., 2014;',\n",
       "     'ref_id': None},\n",
       "    {'start': 104,\n",
       "     'end': 129,\n",
       "     'text': 'Charbonneau et al., 2012;',\n",
       "     'ref_id': 'BIBREF11'},\n",
       "    {'start': 130,\n",
       "     'end': 148,\n",
       "     'text': 'Chen et al., 2014;',\n",
       "     'ref_id': 'BIBREF12'},\n",
       "    {'start': 149,\n",
       "     'end': 171,\n",
       "     'text': 'Lopinski et al., 2000;',\n",
       "     'ref_id': None},\n",
       "    {'start': 172,\n",
       "     'end': 194,\n",
       "     'text': 'Mouzakis et al., 2013;',\n",
       "     'ref_id': None},\n",
       "    {'start': 195,\n",
       "     'end': 216,\n",
       "     'text': 'Ritchie et al., 2017)',\n",
       "     'ref_id': None},\n",
       "    {'start': 626,\n",
       "     'end': 652,\n",
       "     'text': '(Lionnet and Singer, 2012)',\n",
       "     'ref_id': None},\n",
       "    {'start': 677, 'end': 694, 'text': '(Wu et al., 2016)', 'ref_id': None},\n",
       "    {'start': 1036,\n",
       "     'end': 1066,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 1067,\n",
       "     'end': 1087,\n",
       "     'text': 'Dulude et al., 2002;',\n",
       "     'ref_id': None},\n",
       "    {'start': 1088,\n",
       "     'end': 1112,\n",
       "     'text': 'Grentzmann et al., 1998;',\n",
       "     'ref_id': None},\n",
       "    {'start': 1113,\n",
       "     'end': 1135,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'INTRODUCTION'},\n",
       "  {'text': 'To directly address these sorts of questions, we have developed technology to visualize and quantify single RNA frameshifting dynamics in living cells. Using multi-frame repeat epitopes, complementary high-affinity fluorescent probes that selectively bind the epitopes, and multicolor single-molecule microscopy, we are able to simultaneously monitor the translation of single RNA into two unique nascent polypeptide chains encoded in shifted open reading frames. Application to the HIV-1 FSS uncovers unexpected heterogeneity in the production of frameshifted product and implicates a novel bursty frameshifting mechanism. Besides frameshifting, our technology can now be used to examine other translational regulatory dynamics, including upstream open reading frame selection, non-canonical initiation, and ribosomal shunting. We anticipate multi-frame nascent chain tracking will be a powerful new tool to dissect complex translational regulatory dynamics in living cells and organisms.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'INTRODUCTION'},\n",
       "  {'text': 'A multi-frame tag to monitor single RNA translation in two reading frames simultaneously We created a multi-frame (MF) tag to monitor, in living cells, the translation of single RNAs with overlapping open reading frames (ORFs). The tag builds off earlier technology to visualize translation using repeat FLAG or SunTag epitopes labeled by fluorescent Fab or scFv, respectively (Lyon and Stasevich, 2017; Morisaki and Stasevich, 2018) . In the MF tag, FLAG epitopes in the 0 frame are separated from one another by SunTag epitopes in the -1 frame. With this arrangement, single RNAs with ribosomes translating the 0 frame will produce FLAG epitopes labeled by Fab, while those with ribosomes translating the -1 frame will produce SunTag epitopes labeled by scFv. Thus, depending on the chosen frame(s), polysomes will appear all green (all ribosomes translating the 0 frame), all blue (all ribosomes translating the -1 frame), or some combination of the two (Fig. 1A) .',\n",
       "   'cite_spans': [{'start': 377,\n",
       "     'end': 403,\n",
       "     'text': '(Lyon and Stasevich, 2017;',\n",
       "     'ref_id': None},\n",
       "    {'start': 404,\n",
       "     'end': 433,\n",
       "     'text': 'Morisaki and Stasevich, 2018)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 957,\n",
       "     'end': 966,\n",
       "     'text': '(Fig. 1A)',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'RESULTS'},\n",
       "  {'text': 'To ensure both frames of the MF tag encode functional proteins and have similar ribosomal occupancy, we inserted the first exon of the human GNAS1 locus downstream of the FLAG and SunTag epitopes. The GNAS1 locus contains two overlapping ORFs of roughly equivalent lengths that encode peripheral membrane proteins in the 0 and -1 frames: XXLb1 and AlexX (Abramowitz et al., 2004; Aydin et al., 2009) (Fig. 1A ). In combination with the epitopes, this arrangement has several advantages: 1) epitope densities are maximal since -1 frame epitopes act as linkers for the 0 frame and vice versa; 2) signals are digital, so frameshifted and non-frameshifted species are marked by two distinct probes/colors, and 3) epitopes are placed in nearly equivalent positions, so signals appear at roughly the same time and with similar amplification when translated with similar kinetics.',\n",
       "   'cite_spans': [{'start': 354,\n",
       "     'end': 379,\n",
       "     'text': '(Abramowitz et al., 2004;',\n",
       "     'ref_id': None},\n",
       "    {'start': 380,\n",
       "     'end': 399,\n",
       "     'text': 'Aydin et al., 2009)',\n",
       "     'ref_id': 'BIBREF2'}],\n",
       "   'ref_spans': [{'start': 400,\n",
       "     'end': 408,\n",
       "     'text': '(Fig. 1A',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'RESULTS'},\n",
       "  {'text': 'As a first application of the MF tag, we focused on -1 programmed ribosomal frameshifting caused by the HIV-1 frameshift sequence (FSS). We inserted the FSS upstream of our MF tag and transiently transfected the resulting construct into U-2 OS cells. The FSS contains a slippery poly-U stretch nine nucleotides upstream of a stem loop. Two to ten hours post transfection, we observed cells with tens or hundreds of individual RNA diffusing throughout the nucleus and cytoplasm (Fig. 1B) . Nascent Chain To confirm these RNA were active translation sites, or polysomes, we performed two experiments. First, we re-imaged cells 12-24 hours after transfection. At these later time points, Fab and scFv began to accumulate in the cell membrane ( Fig. S1 and Movie this frame encoding a functional protein, as demonstrated by shifting the sequence by one nucleotide into the 0 frame ( Fig. S1 and Movie S2, right panels). Second, we treated cells with the translational inhibitor puromycin. Just minutes after treatment, we observed a dramatic decrease in the number of Fab-and/or scFv-labeled RNA, consistent with the premature release of nascent chains ( Fig. 1D and Movie S3). Together, these data provide strong evidence that we are able to detect single RNA frameshifting dynamics with the MF tag.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 477,\n",
       "     'end': 486,\n",
       "     'text': '(Fig. 1B)',\n",
       "     'ref_id': 'FIGREF9'},\n",
       "    {'start': 741, 'end': 748, 'text': 'Fig. S1', 'ref_id': 'FIGREF9'},\n",
       "    {'start': 879, 'end': 886, 'text': 'Fig. S1', 'ref_id': 'FIGREF9'},\n",
       "    {'start': 1151, 'end': 1158, 'text': 'Fig. 1D', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'RESULTS'},\n",
       "  {'text': 'The HIV-1 FSS structure has been previously shown to produce frameshifted protein with an efficiency of 5-10% based on the dual luciferase assay and similar bulk assays (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . However, it remains unclear how this percentage is established. One possibility is that all RNA behave more or less the same and their ribosomes frameshift with 5-10% probability. On the opposite end of the spectrum, it is possible that RNA display a high degree of heterogeneity, so that just 5-10% of RNA have ribosomes that frameshift with nearly 100% probability. A third possibility is that frameshifting is common on all RNA, but frameshifted proteins are less stable and degraded faster than nonframeshifted proteins.',\n",
       "   'cite_spans': [{'start': 169,\n",
       "     'end': 199,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 200, 'end': 220, 'text': 'Dulude et al., 2002;', 'ref_id': None},\n",
       "    {'start': 221,\n",
       "     'end': 245,\n",
       "     'text': 'Grentzmann et al., 1998;',\n",
       "     'ref_id': None},\n",
       "    {'start': 246,\n",
       "     'end': 268,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'As a first step to quantify frameshifting dynamics at the single-molecule level, we tracked thousands of individual RNA approximately two to ten hours post transfection with and without the FSS sequence present (±FSS; Fig. 2A -C). In the +FSS cells, we found 92 ± 1.3% of translation sites were translating the canonical 0 frame alone, while 6.2 ± 1.1% were translating both the 0 and -1 frames. Only rarely did we observe translation All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 218,\n",
       "     'end': 225,\n",
       "     'text': 'Fig. 2A',\n",
       "     'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint sites translating just the -1 frame (1.6 ± 0.5%, Fig. S2 and Movie S4). To ensure these results were not influenced by the multi-frame tag, we reversed the FLAG and SunTag epitopes in the tag and repeated experiments. This gave nearly the same percentages, confirming the tag order and/or epitope positioning did not bias measurements ( Fig. S3 and Movie S5). We then repeated experiments in cells transfected with the control -FSS construct. In this case, we observed virtually no frameshifting sites (0.9 ± 0.7%) ( Fig. 2C and Movie S6). Taken together, these data suggest the FSS alone causes ~8% of translation sites to frameshift.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 195,\n",
       "     'end': 202,\n",
       "     'text': 'Fig. S2',\n",
       "     'ref_id': 'FIGREF10'},\n",
       "    {'start': 483, 'end': 490, 'text': 'Fig. S3', 'ref_id': 'FIGREF12'},\n",
       "    {'start': 663, 'end': 670, 'text': 'Fig. 2C', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'To further characterize the efficiency of frameshifting, we quantified the precise number of frameshifted versus non-frameshifted ribosomes per translation site. To do this, we re-imaged the MF tag at high laser powers so we could simultaneously visualize single mature XXLb1 or AlexX proteins in comparison to their nascent chains within translation sites (Fig. S4 ). After renormalizing the fluorescence, we found frameshifting sites with translation in both the 0 and -1 frames had 7.8 ± 0.7 ribosomes total, 1.9 ± 0.3 of which were frameshifted, while 0-frame only translation sites had 3.2 ± 0.1 ribosomes and -1-frame only translation sites had 3.4 ± 1.4 frameshifted ribosomes (Fig. 2B,D) .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 357,\n",
       "     'end': 365,\n",
       "     'text': '(Fig. S4',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 684, 'end': 695, 'text': '(Fig. 2B,D)', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'Thus, although just ~8% of translation sites contained frameshifted ribosomes, within these subset of sites, anywhere between 30 to 100% of ribosomes were frameshifted.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'These data support a heterogenous RNA model in which frameshifting occurs on a small subset of RNA with high probability.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'Given the heterogeneity of observed frameshifting, we hypothesized that the frameshifting state could be stimulated by a specific sub-cellular environment. To test this All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint hypothesis, we performed a statistical analysis of all tracks to see if any biophysical parameters correlated with frameshifting. This revealed frameshifting sites diffuse more slowly than other translation sites or RNA (Fig. 2E ) and were also slightly more likely to be near the nuclear periphery (Fig. 2F) . These results are consistent with a slight preference for frameshifting in the perinuclear endoplasmic reticulum, where RNA are less mobile and more efficiently translated (Voigt et al., 2017; Wang et al., 2016) .',\n",
       "   'cite_spans': [{'start': 629,\n",
       "     'end': 649,\n",
       "     'text': '(Voigt et al., 2017;',\n",
       "     'ref_id': None},\n",
       "    {'start': 650, 'end': 668, 'text': 'Wang et al., 2016)', 'ref_id': None}],\n",
       "   'ref_spans': [{'start': 366,\n",
       "     'end': 374,\n",
       "     'text': '(Fig. 2E',\n",
       "     'ref_id': 'FIGREF10'},\n",
       "    {'start': 445, 'end': 454, 'text': '(Fig. 2F)', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'Of the parameters we quantified, one of the strongest correlates of frameshifting was RNA intensity. Specifically, sites translating the -1 frame had an average RNA intensity that was nearly 30% brighter than RNA only spots ( Fig. 2B and S5A ). In addition, these brighter frameshifting sites were translation-dependent, dissociating upon puromycin treatment (Movie S7). We wondered if this might be an artifact due to the aggregation of probes. Self-aggregation of Fab was unlikely as this would cause 0-frame translation sites to also be brighter, which was not the case (Fig. 2B) . Similarly, it was unlikely to be due to self-aggregation of scFv, as this would cause 0-frame translation sites to be brighter in the reverse MF tag, which was also not the case (Fig. S5B ). This left the possibility that Fab aggregate with scFv. To rule this out, we reimaged the MF tag without Fab. As frameshifting translation sites remained brighter than other sites ( Fig.   S5C ), we conclude the brighter RNA signal is not a tagging artifact, but instead represents a propensity for frameshifting sites to associate with other translating RNA in higher-order complexes reminiscent of \"translation factories\" (Morisaki et al., 2016; Pichon et al., 2016) .',\n",
       "   'cite_spans': [{'start': 1200,\n",
       "     'end': 1223,\n",
       "     'text': '(Morisaki et al., 2016;',\n",
       "     'ref_id': None},\n",
       "    {'start': 1224,\n",
       "     'end': 1244,\n",
       "     'text': 'Pichon et al., 2016)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 226,\n",
       "     'end': 241,\n",
       "     'text': 'Fig. 2B and S5A',\n",
       "     'ref_id': 'FIGREF10'},\n",
       "    {'start': 573, 'end': 582, 'text': '(Fig. 2B)', 'ref_id': 'FIGREF10'},\n",
       "    {'start': 763, 'end': 772, 'text': '(Fig. S5B', 'ref_id': 'FIGREF14'},\n",
       "    {'start': 958, 'end': 968, 'text': 'Fig.   S5C', 'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'We did not observe factories in cells transfected with the control -FSS construct ( Fig. S5D ). We therefore reasoned the FSS itself might facilitate frameshifting in factories. All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 84,\n",
       "     'end': 92,\n",
       "     'text': 'Fig. S5D',\n",
       "     'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https: //doi.org/10.1101/478040 doi: bioRxiv preprint To test this model, we co-transfected a short oligo RNA containing the FSS alone into cells expressing the +FSS MF tag. This led to a significant increase in the fraction of frameshifting sites translating just the -1 frame, from 1.6 to 5.6% when 1 \\uf06dg oligo was added, and up to 9.5% when 4 \\uf06dg oligo was added (Fig. 3A,B) . The significant increase in the -1 only sites suggests RNA can exist in a state in which all or nearly all ribosomes are frameshifted. Moreover, frameshifting continued to occur preferentially within factories, having RNA signals from 30 to 50% brighter than RNA-only spots (Fig. 3C) , even though the overall fraction of RNA within factories remained unchanged (Fig. S6 ).',\n",
       "   'cite_spans': [{'start': 100,\n",
       "     'end': 146,\n",
       "     'text': '//doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 457,\n",
       "     'end': 468,\n",
       "     'text': '(Fig. 3A,B)',\n",
       "     'ref_id': 'FIGREF12'},\n",
       "    {'start': 745, 'end': 754, 'text': '(Fig. 3C)', 'ref_id': 'FIGREF12'},\n",
       "    {'start': 833, 'end': 841, 'text': '(Fig. S6', 'ref_id': 'FIGREF16'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'Importantly, in control cells expressing the -FSS construct, we did not see an increase in frameshifting upon oligo RNA co-transfection. Collectively, these data provide further support for a model in which the HIV-1 FSS itself facilitates frameshifting in factories.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'The ~8% of frameshifted translation sites we observed is consistent with previous measurements of 5-10% frameshifted protein product (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . All else equal, this implies that frameshifting alone can explain the steady-state levels of frameshifted protein, without the need for other regulatory mechanisms, such as protein degradation.',\n",
       "   'cite_spans': [{'start': 133,\n",
       "     'end': 163,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 164, 'end': 184, 'text': 'Dulude et al., 2002;', 'ref_id': None},\n",
       "    {'start': 185,\n",
       "     'end': 209,\n",
       "     'text': 'Grentzmann et al., 1998;',\n",
       "     'ref_id': None},\n",
       "    {'start': 210,\n",
       "     'end': 232,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translational output of frameshifted ribosomes'},\n",
       "  {'text': 'To test this hypothesis, we measured the elongation rate of frameshifted ribosomes compared to non-frameshifted ribosomes. Fits to the linear portion of run-off curves yielded similar post-tag run-off times of ~386 s and ~475 s, corresponding to average elongation rates of 2.7 ± 0.3 aa/sec and 2.2 ± 0.6 aa/sec, respectively ( Fig. 4A and Fig.   S7 ). Fluorescence recovery after photobleaching experiments further confirmed these measurements without the use of drugs ( Fig. S8 and Movie S8). Based on these similar All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 328,\n",
       "     'end': 349,\n",
       "     'text': 'Fig. 4A and Fig.   S7',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 472, 'end': 479, 'text': 'Fig. S8', 'ref_id': 'FIGREF24'}],\n",
       "   'section': 'Translational output of frameshifted ribosomes'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint elongation rates for both non-frameshifting and frameshifting sites, a single round of translation would take ~9 minutes. Accounting for the number of ribosomes per translation site and their relative fractions, we calculate a cell with 100 RNA would produce ~115 frameshifted protein per hour compared to ~2,220 canonical proteins. In other words, frameshifted proteins would account for ~5% of the total, in agreement with earlier measurements. Thus, the dynamics of the FSS sequence alone can be sufficient to account for the steady-state levels of frameshifted protein in living cells, without the need for additional regulatory mechanisms.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translational output of frameshifted ribosomes'},\n",
       "  {'text': 'Despite the similar elongation rates, we noticed a slight (~60 s) delay in the run-off response at frameshifting translation sites (Sun sites Fig 4A) . Given earlier work showing the potential of the FSS to pause ribosomes (Dulude et al., 2002) , we envisioned this delay could be due to a queue or traffic jam of ribosomes upstream of the FSS within frameshifting sites. As the backed-up ribosomes clear the traffic jam, they replenish the loss of ribosomes running-off. Only after the traffic jam is fully cleared does the number of ribosomes beyond the FSS (with labeled nascent chains) begin to decay.',\n",
       "   'cite_spans': [{'start': 223,\n",
       "     'end': 244,\n",
       "     'text': '(Dulude et al., 2002)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 142,\n",
       "     'end': 149,\n",
       "     'text': 'Fig 4A)',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Evidence for ribosomal traffic jams at HIV-1 frameshifting translation sites'},\n",
       "  {'text': 'To test this possibility, we added a 10x HA epitope repeat upstream of the FSS (Fig. 4B ). This served two purposes: first, it allowed us to monitor both ribosomes upstream of the FSS (translating HA epitopes) and ribosomes downstream (translating either 0 frame FLAG or -1 frame SunTag epitopes); second, the arrangement more closely mimicked the natural placement of the FSS between the gag-pol polyproteins. In particular, the additional sequence space upstream of the FSS could accommodate longer ribosomal traffic jams, should they occur. We hypothesized that longer ribosomal All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 79,\n",
       "     'end': 87,\n",
       "     'text': '(Fig. 4B',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Evidence for ribosomal traffic jams at HIV-1 frameshifting translation sites'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint traffic jams would lead to longer run-off delays. Consistent with this, ribosomes within frameshifting sites took much longer to run-off (Fig. 4B) , with frameshifted ribosome levels fluctuating but remaining overall steady for upwards of 3000 s, despite an overall ribosome loss (Figs. S9 and S10). We observed this trend even at the single-molecule level, where the total number of ribosomes (marked by HA) dropped, but the number of Such a release can be seen in the single-molecule track at the ~1000 s time point, when the frameshift signal gets significantly brighter. Interestingly, this increase in brightness coincided with the association of an additional non-translating +FSS RNA (gray arrows in Fig. 4C ). This is consistent with the hypothesis that multi-RNA factories stimulate frameshifting. Although difficult to capture, we observed this type of stimulated frameshifting burst in another single frameshifting RNA track as well (Fig. S11 ).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 283,\n",
       "     'end': 292,\n",
       "     'text': '(Fig. 4B)',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 853, 'end': 860, 'text': 'Fig. 4C', 'ref_id': 'FIGREF13'},\n",
       "    {'start': 1090, 'end': 1099, 'text': '(Fig. S11', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Evidence for ribosomal traffic jams at HIV-1 frameshifting translation sites'},\n",
       "  {'text': 'To quantify the kinetics of frameshifting, we developed two candidate models and attempted to fit each to our three main observations: (i) the percentages of frameshifting versus non-frameshifting translation sites (Fig. 4C ), (ii) ribosomal occupancies (Fig 4D) , and (iii) run-off kinetics (Figs. 4E,F). Both models include initiation of ribosomes, codondependent elongation of proteins along the RNA template, and ribosomal exclusion to block ribosomes from passing one another or occupying the same place on the RNA.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 215,\n",
       "     'end': 223,\n",
       "     'text': '(Fig. 4C',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 254, 'end': 262, 'text': '(Fig 4D)', 'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The only difference in the two models is the treatment for how ribosomes shift from the 0 to the -1 frame.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The first model assumes constitutive frameshifting, in which all ribosomes frameshift at the FSS with equal probability. This model could capture either observation (i) or (ii), but not both simultaneously; frameshifting either led to excessively large fractions of frameshifting sites or excessively small ribosomal loading, in disagreement with our observations that a relatively small fraction of RNA frameshift with relatively high ribosomal occupancies. Even with addition of distinct pauses in elongation at the FSS in both frames, the constitutive model was unable to fit our data (Fig. S12 ).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 588,\n",
       "     'end': 597,\n",
       "     'text': '(Fig. S12',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': \"The second model is inspired by two-state gene models that are commonly used to describe heterogeneous transcription (Munsky et al., 2012) . In this `bursty' model, RNA stochastically switch between non-frameshifting and frameshifting states in which either 0% or 100% of ribosomes produce frameshifted proteins (Fig. 5A ). In the bursty model, the RNA frameshift state is assumed to switch ON and OFF at rates and , respectively, and the steady state fraction of RNA in the ON state is given by = /( + ). We estimated f from the observed fraction of frameshifting translation sites (Fig. 2C ).\",\n",
       "   'cite_spans': [{'start': 117,\n",
       "     'end': 138,\n",
       "     'text': '(Munsky et al., 2012)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 312,\n",
       "     'end': 320,\n",
       "     'text': '(Fig. 5A',\n",
       "     'ref_id': 'FIGREF14'},\n",
       "    {'start': 583, 'end': 591, 'text': '(Fig. 2C', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'To estimate the timescale of and , we tracked translation sites for longer periods of time. To achieve this tracking, we doubled the number of epitopes in the multiframe reporter (creating the +FSS 2x multi-frame tag) and changed our imaging strategy to sample the RNA signal at all time points and the 0 and -1 translation signals once every fifth time point. This arrangement substantially reduced photobleaching and allowed us to continuously track and monitor the translational status of single translation sites in 3D',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'for nearly an hour. Fig. 5B shows the frameshifting state survival times for the seven translation sites we tracked in this manner, including one site that frameshifted for longer All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 20,\n",
       "     'end': 27,\n",
       "     'text': 'Fig. 5B',\n",
       "     'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint than 40 minutes ( Fig. S13 and Movie S10), representing at least four rounds of translation at our measured elongation rate of ~2.4 aa/sec. Remarkably, this frameshifting translation site associated with another for a large part of the 40-minute imaging window, which supports our hypothesis that the FSS facilitates frameshifting in multi-RNA sites. From the frameshift state survival times, we fit the rate of to be ~0.0013 sec -1 (Fig. 5B S12 ). To account for the different run-off delays seen at frameshifting and non-frameshifting sites (seen in Fig. 4A ,B), the model required elongation pauses of 1/ ~ 48 s at non-frameshifting sites and 1/ * ~ 90 s at frameshifting sites. We explored if codon usage could also explain the differences in run-off times. However, according to the codon adaptation index (Gorgoni et al., 2016; Sharp and Li, 1987) , which is related to the speed at which each codon is translated in the simulation, there is no notable difference between the 0 and -1 frames ( Fig. S14) . Moreover, the distinct pauses predicted by the model are comparable to those previously measured using in vitro and in vivo bulk assays (Lopinski et al., 2000) .',\n",
       "   'cite_spans': [{'start': 957,\n",
       "     'end': 979,\n",
       "     'text': '(Gorgoni et al., 2016;',\n",
       "     'ref_id': None},\n",
       "    {'start': 980, 'end': 999, 'text': 'Sharp and Li, 1987)', 'ref_id': None},\n",
       "    {'start': 1294,\n",
       "     'end': 1317,\n",
       "     'text': '(Lopinski et al., 2000)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 164,\n",
       "     'end': 172,\n",
       "     'text': 'Fig. S13',\n",
       "     'ref_id': 'FIGREF9'},\n",
       "    {'start': 579, 'end': 587, 'text': '(Fig. 5B', 'ref_id': 'FIGREF14'},\n",
       "    {'start': 588, 'end': 591, 'text': 'S12', 'ref_id': 'FIGREF9'},\n",
       "    {'start': 698, 'end': 705, 'text': 'Fig. 4A', 'ref_id': 'FIGREF13'},\n",
       "    {'start': 1146, 'end': 1155, 'text': 'Fig. S14)', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Because the estimated average pause time, 1/ * , is greater than the average initiation time, 1/ , ribosomes could initiate faster than they clear the FSS with an excess rate of − * = 0.009 per second and create upstream traffic jams in frameshifting sites. These traffic jams would continue to build for as long as the RNA remains in the frameshifted state, or approximately 1/ = 770 s on average. This would allow for the accumulation of approximately ( − * ) / ~ 7 ribosomes (on the HA multi-frame tag, Table 1 ). Occasionally, traffic jams can extend all the way back to the start codon, as seen in a sample simulation (Movie S11). However, because these traffic jams occur upstream of the FSS within the HA epitope repeats, the nascent chain intensity at translation sites is only partially correlated with ribosomal occupancy. As a result, the model predicts frameshifting sites are slightly brighter than non-frameshifting sites ( Fig.   5G ), which we also observe experimentally (Fig. S15 ). Moreover, due to the long time it can take to clear traffic jams, frameshifting can persist for hours following the global shut down of translation initiation (with harringtonine, for example), as can be seen in simulations of the best-fit model (Movie S12) and consistent with what we observed in Figs. 4C and S11. Thus, the final bursty model suggests a mechanism by which frameshifting can persist for long periods in the absence of translation initiation.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 652,\n",
       "     'end': 659,\n",
       "     'text': 'Table 1',\n",
       "     'ref_id': 'TABREF1'},\n",
       "    {'start': 1084, 'end': 1093, 'text': 'Fig.   5G', 'ref_id': 'FIGREF14'},\n",
       "    {'start': 1134, 'end': 1143, 'text': '(Fig. S15', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'Frameshifting is a common tactic used by viruses to minimize their genomes for faster, more efficient replication in host cells, effectively getting two viral proteins for the price of one viral RNA. While the general architecture of frameshift sequences is well characterized and the dynamics of frameshifting have been measured with single-All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint nucleotide precision in vitro, until now frameshifting had not been directly observed in a living system. To achieve this, we created a multi-frame repeat epitope tag that can light up single RNA translation sites in different colors depending on which open reading frame is being translated. Together with sensitive single-molecule microscopy and computational modeling, we have demonstrated five novel aspects of HIV-1 frameshifting:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': '(i) frameshifted proteins originate from a small subset of RNA that frameshift with high efficiency; (ii) frameshifting occurs preferentially in multi-RNA translation factories that are facilitated in part by the frameshift sequence; (iii) frameshifting occurs in bursts on single RNA that can last for several rounds of translation; (iv) ribosomes that frameshift are paused for longer at the frameshifting sequence than ribosomes that do not frameshift;',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'and (v) pauses at the frameshift sequence induce ribosomal traffic jams that can maintain the production of frameshifted protein despite global inhibition of translation. Fig. 6 summarizes our findings.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 171,\n",
       "     'end': 177,\n",
       "     'text': 'Fig. 6',\n",
       "     'ref_id': 'FIGREF16'}],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'In contrast to constitutive frameshifting on any RNA, our data indicate that frameshifting occurs in bursts on a subset of RNA. Bursty expression has been demonstrated by others, both at single transcription sites as well as at translation sites in bacteria (Lionnet and Singer, 2012) and eukaryotes (Wu et al., 2016) . The origin of frameshifting bursts is difficult to pinpoint. It is tempting to speculate that there is a unique structure the RNA takes that enhances frameshifting, particularly given the distinct pause times in our best-fit model. According to our observations, multi-RNA translation factories appear to be more susceptible to frameshift bursts. Such factories have been previously observed (Morisaki et al., 2016; Pichon et al., 2016) . Local concentration of translation factors is higher within factories, so translation kinetics are enhanced (Pichon et al., All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 300,\n",
       "     'end': 317,\n",
       "     'text': '(Wu et al., 2016)',\n",
       "     'ref_id': None},\n",
       "    {'start': 712,\n",
       "     'end': 735,\n",
       "     'text': '(Morisaki et al., 2016;',\n",
       "     'ref_id': None},\n",
       "    {'start': 736,\n",
       "     'end': 756,\n",
       "     'text': 'Pichon et al., 2016)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint 2016). However, the crowding of machinery at these factory sites may cause them to be more prone to errors and even more so with increased interactions with other FSSencoding RNA, the FSS being a proverbial wrench within translational factory machinery.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'Such low-fidelity factories may benefit viral robustness, where mutations are desirable.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'Consistent with this idea, HIV-1 RNAs prefer a dimeric state which may promote translation factory formation (Barajas et al., 2018) .',\n",
       "   'cite_spans': [{'start': 109,\n",
       "     'end': 131,\n",
       "     'text': '(Barajas et al., 2018)',\n",
       "     'ref_id': 'BIBREF3'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'In this study, we observed frameshifting in the context of factories in three different circumstances. First, in puromycin experiments, we observed frameshifting sites separate into two frameshifting sites upon translation shut-down (Movie S7). Second, we observed on two occasions non-translating RNA interact with a frameshifting site, after which the frameshifting signal at the site markedly increased (Figs. 4E and S11). Third, when we co-transfected FSS oligo RNA into cells (Fig. 3) , frameshifting increased in proportion to the amount we co-transfected. Collectively, these data suggest a causal ordering, whereby interactions between FSS-encoding sequences lead to enhanced frameshifting.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 481,\n",
       "     'end': 489,\n",
       "     'text': '(Fig. 3)',\n",
       "     'ref_id': 'FIGREF12'}],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'According to our best-fit model, pauses always occur at the frameshift sequence, with longer pauses associated with frameshifting RNA compared to non-frameshifting RNA. Pausing in and of itself is therefore only a weak predictor of frameshifting, as others have shown (Kontos et al., 2001; Tu et al., 1992) . Nevertheless, the longer pauses associated with frameshifting suggest the frameshift sequence can adopt more than one The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint an elevated ribosomal occupancy as far back as ~400 codons from the FSS, a length that coincides almost perfectly with the length of the FSS-upstream sequence in HIV-1 gagpol. This is not unprecedented, as ribosome profiling experiments have also found evidence for relatively high ribosomal densities as far back as the start codon of the gag protein (Napthine et al., 2017) . These data suggest the strength of the HIV-1 pause may have evolved to on occasion produce the longest possible traffic jams. As clearance of these jams takes time, frameshifting can persist on RNA for hours despite a global shutdown of translation initiation. In effect, the jam acts like a battery that continually fuels the Beyond the imaging of HIV-1 translational frameshifting, the multi-frame tag can now be used in a variety of other contexts. For example, it can immediately be used to investigate frameshifting dynamics along other viral RNA sequences, as well as frameshifting thought to occur along endogenous human mRNA (Cardno et al., 2015) .',\n",
       "   'cite_spans': [{'start': 268,\n",
       "     'end': 289,\n",
       "     'text': '(Kontos et al., 2001;',\n",
       "     'ref_id': None},\n",
       "    {'start': 290, 'end': 306, 'text': 'Tu et al., 1992)', 'ref_id': None},\n",
       "    {'start': 925,\n",
       "     'end': 948,\n",
       "     'text': '(Napthine et al., 2017)',\n",
       "     'ref_id': None},\n",
       "    {'start': 1584,\n",
       "     'end': 1605,\n",
       "     'text': '(Cardno et al., 2015)',\n",
       "     'ref_id': 'BIBREF10'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'Likewise, the multi-frame tag can be used to examine other non-canonical translation processes involving more than one open reading frame, including start-codon selection, leaky scanning, ribosomal shunting, and general translation fidelity. In fact, in an accompanying paper, a similar multi-frame tag was used to examine upstream and downstream open reading frame selection (Boersma et al.) . Like us, the authors also saw a high degree of heterogeneity between translating RNAs, with bursts of translation All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 376,\n",
       "     'end': 392,\n",
       "     'text': '(Boersma et al.)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint initiation in multiple open reading frames similar to the bursts of frameshifting we observed at the HIV-1 frameshift sequence. Translational heterogeneity may therefore be far more common than originally appreciated, particularly when it comes to non-canonical translation. We therefore believe multi-color single molecule imaging of both canonical and non-canonical translation will become a powerful new tool for dissecting complex RNA regulatory dynamics in a variety of important contexts. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Dulude, D., Baril, M., and Brakier-Gingras, L. (2002) . Characterization of the frameshift stimulatory signal controlling a programmed -1 ribosomal frameshift in the human immunodeficiency virus type 1. Nucleic Acids Res. 30, 5094-5102. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 890,\n",
       "     'end': 931,\n",
       "     'text': 'Baril, M., and Brakier-Gingras, L. (2002)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. is the translation initiation rate, is the translation elongation rate, is the rate at which RNA switch to the frameshifting state, is the rate at which RNA switch to the non-frameshifting state, is the pause rate at the FSS in the non-frameshifting state, * is the pause rate at the FSS in the frameshifting All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint longer ribosomal traffic jams. These allow frameshifting RNA to continue to be translated for up to an hour longer than non-frameshifting RNA upon global translation initiation shutdown.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint RNA (0, 1, and 4 mg) encoding the frameshift sequence were co-transfected into cells with the +FSS multi-frame reporter or the control -FSS reporter. The bar graph above shows the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). (B)',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The distributions of RNA signals from the four experiments normalized to their mean value. In all +FSS experiments, the fraction of bright RNA (defined here as having an intensity greater than or equal to 1.6 times the mean) remained constant. The -FSS experiment had fewer bright RNA, suggesting the FSS sequence facilitates incorporation into multi-RNA sites. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'For completeness, an additional experiment was performed to generate the ribosome run-off curve from FLAG epitopes in the HA multi-frame tag. Because we can only image two epitopes at the same time (since RNA is imaged in a third color), we examined FLAG and Sun epitopes (Sun epitopes were required to distinguish frameshifting and nonframeshifting sites). The FLAG run-off from this experiment is on the right (green triangles). This curve is compared to the run-off curve of HA in HA only sites (orange squares) from Figure 4B . Both experiments represent non-frameshifting sites (no Sun signal detected). Error bars represent S.E.M.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 520,\n",
       "     'end': 529,\n",
       "     'text': 'Figure 4B',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Figure S10. Ribosome run-off from FLAG epitopes in HA multi-frame tag'},\n",
       "  {'text': 'A single frameshifting translation site encoding the HA multi-frame tag was tracked after harringtonine addition. On the top, a montage of image trims shows the detected RNA-, HA-, and Sun shown signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint ribosomes) is plotted through time. At about the fourth timepoint, another non-translating RNA interacts with the frameshifting RNA, leading to a burst of frameshifting signal (Sun).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions. and frameshifting sites (gray), plotted with data in Fig. 4A (black triangles and gray circles). On the right, simulated run-off (solid lines) from the start site (AUG) of all ribosomes in non-frameshifting (black) and frameshifting sites (gray), plotted with data in The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint channels separated to highlight two frameshifted RNA that split from a single translation site. On the right, the position of the spot is plotted through time. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 139,\n",
       "     'end': 146,\n",
       "     'text': 'Fig. 4A',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'Intensity distribution of non-frameshifted HA signals (black) versus frameshifted HA signals (gray) produced from the HA multi-frame tag.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'},\n",
       "  {'text': 'Movie S1. Dynamics of cell presented in Figure 1B Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 4 seconds for a total of 76 seconds. Circled regions of interest correspond to species identified in Figure 1B and C. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 40,\n",
       "     'end': 49,\n",
       "     'text': 'Figure 1B',\n",
       "     'ref_id': 'FIGREF9'},\n",
       "    {'start': 410, 'end': 419, 'text': 'Figure 1B', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'SUPPLEMENTARY MOVIE LEGENDS'},\n",
       "  {'text': 'Single plane movies showing example U-2 OS cells expressing either +FSS multi-frame tag (left), -FSS control multi-frame tag (middle), or -FSS (+1 nt) tag with XXLb1 shifted into the 0 frame (right). All cells were bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, AlexX), and scFv-sfGFP (Ch3 blue, XXLb1). Images were acquired every 7 seconds for a total of 42 seconds. Scale bar, 10 µm. Figure 1D Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 112 seconds prior to puromycin and 280 seconds following puromycin. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 420,\n",
       "     'end': 429,\n",
       "     'text': 'Figure 1D',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Movie S2. Dynamics of cells presented in Figure S1'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S4. Dynamics of cell presented in Figure S2'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 63 seconds. Circled regions of interest correspond to species identified in Fig. S2 . Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 506,\n",
       "     'end': 513,\n",
       "     'text': 'Fig. S2',\n",
       "     'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Movie S4. Dynamics of cell presented in Figure S2'},\n",
       "  {'text': \"Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS reverse multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, -1 ORF), and scFv-sfGFP (Ch3 blue, 0 ORF). Images were acquired every 7 seconds for a total of 63 seconds. Circled regions of interest correspond to RNA with no translation (numbered '1'), 0 and -1 frame translation site (numbered '2'), and 0 frame only translation site (numbered '3'). Scale bar, 10 µm.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S5. Representative cell transfected with the +FSS reverse multi-frame tag'},\n",
       "  {'text': \"Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the -FSS control tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 63 seconds. Circled regions of interest correspond to RNA with no translation (numbered '1'), and 0 frame only translation site (numbered '2'-'8').\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S6. Representative cell transfected with the -FSS control tag'},\n",
       "  {'text': 'Scale bar, 10 µm. Figure   1D All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 18,\n",
       "     'end': 29,\n",
       "     'text': 'Figure   1D',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Movie S6. Representative cell transfected with the -FSS control tag'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the + FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 217 (of 280) seconds total post puromycin addition. This movie shows the same cell from Movie S3, but now zoomed in to accentuate a frameshifting site splitting (red arrows). Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S7. Zoom-in showing frameshifting site presented as a montage in'},\n",
       "  {'text': 'Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the + FSS 2x multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 10 seconds for a total of 100 seconds pre-FRAP and 790 seconds post-FRAP. Control spot (far left red arrow) and two frameshifting sites (center arrows) were tracked through time to measure their fluorescence (0 frame translation) after photobleaching. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S8. Dynamics of Fluorescence Recovery After Photobleaching (FRAP) experiment shown in Figure S8'},\n",
       "  {'text': 'Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the HA multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), HA Fab Cy3 (Ch2 green, all ORFs), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired post harringtonine every 120 seconds for a total of 3480 seconds. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'Movie S10. Dynamics of tracked frameshifting sites presented in Figure S13 All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 64,\n",
       "     'end': 74,\n",
       "     'text': 'Figure S13',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. Movie S12. Sample simulation of best-fit bursty model of frameshifted and nonframeshifted translation sites during harringtonine run-off All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': \"The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint A best-fit bursty model simulation of multiple non-frameshifting (top four labeled 'nonFS Spots') and frameshifting (bottom four labeled 'FS Spots') translation sites encoding the HA multi-frame tag. In the simulation, harringtonine is added after 1000 s and the simulation continues for an additional 2000 s. Individual ribosomes (gray dots) are shown moving along the reporters (gray lines), with green, blue, and cyan squares marking the positions of FLAG, SunTag, and HA epitopes, respectively. As the ribosomes move along the repeat epitopes, the translated nascent chains appear as green (FLAG-tagged nascent chains), blue (Sun-tagged nascent chains), and cyan (HA-tagged nascent chains) dots that increase in size as more epitopes are translated. When the RNA is in the non-frameshifting state, the FSS is labeled 'FSS(OF)' in green; when the RNA is in the frameshifting state, the FSS is labeled 'FSS(-1F)' in blue. The simulated time is shown to the right of each reporter.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'Requests for further information and resources should be directed to and will be fulfilled by the corresponding authors, Brian Munsky (brian.munsky@colostate.edu) and Tim Stasevich (tim.stasevich@colostate.edu). Key plasmids will be deposited on Addgene.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'CONTACT FOR REAGENTS AND RESOURCE SHARING'},\n",
       "  {'text': \"The HIV-1 frameshift sequence (FSS) followed by either the multi-frame (MF) tag or the reverse multi-frame (revMF) tag were synthesized by GeneArt® gene synthesis service (Thermo Fisher Scientific). The gene fragments were flanked by NotI and NheI, and fused to the upstream of the beta-actin zipcode and 24x MS2 stem loops in the 3' UTR of plasmid pUB_smFLAG_ActB_MS2 (Plasmid #81083, addgene) to obtain the FSS-MF and FSS-revMF, respectively. To double the MF tag, the MF tag region was digested out from FSS-MF with XbaI and AgeI, and then ligated into FSS-MF flanked with NheI and\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': \"AgeI. The open reading frame encoding human XXLb1/AlexX (Abramowitz et al., 2004; Aydin et al., 2009 ) was amplified from U-2 OS cells cDNA with the primers: 5'-GTT GTC ATA TGG GCG TGC GCA ACT -3'; 5'-GAT GTA GCT AGC CTA GAA GCA GCA GGC GGT G -3'. The amplified XXLb1/AlexX was flanked with NsiI and NheI, and then inserted into the C-terminal region of the FSS-MF, FSS-2xMF and FSS-revMF to obtain FSS-MF-AlexX (i.e. the +FSS multi-frame tag), FSS-2xMF-AlexX (i.e. the +FSS 2x multi-frame tag), and FSS-revMF-AlexX (i.e. the +FSS reverse multi-frame tag), respectively. To All rights reserved. No reuse allowed without permission.\",\n",
       "   'cite_spans': [{'start': 56,\n",
       "     'end': 81,\n",
       "     'text': '(Abramowitz et al., 2004;',\n",
       "     'ref_id': None},\n",
       "    {'start': 82,\n",
       "     'end': 100,\n",
       "     'text': 'Aydin et al., 2009',\n",
       "     'ref_id': 'BIBREF2'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': \"The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Fab generation and dye-conjugation Fab generation was done using the Pierce mouse IgG1 preparation kit (Thermo Fisher Scientific) per the manufacturer's instructions. Briefly, beads conjugated with ficin were incubated in 25 mM cysteine to digest FLAG (Wako, 012-22384 Anti DYKDDDDK mouse IgG2b monoclonal) or HA (Sigma-Aldrich, H3663 HA-7 IgG1 mouse monoclonal) antibodies to generate Fab. Fab were separated from the digested Fc region using a NAb Protein A column (Thermo Scientific, product # 1860592). Fab were concentrated to ~1 mg/ml and conjugated to either Cy3 or Alexa Fluor 488 (A488). Cy3 Nhydroxysuccinimide ester (Invitrogen) or A488 tetrafluorophenyl ester (Invitrogen) was suspended in DMSO and stored at -20°C. 100 µg of Fab were mixed with 10 µL of 1M NaHCO3, to a final volume of 100 µL. 2.66 µl of Cy3 (or 5 \\uf06dl of A488) was added to this 100 µL mixture and incubated for 2 hours at room temperature with end-over-end rotation.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'The dye conjugated Fab were eluted from a PBS equilibrated PD-mini G-25 desalting column (GE Healthcare) to remove unconjugated dye. Dye conjugated Fabs then were concentrated in an Ultrafree 0.5 filter (10k-cut off; Millipore) to 1 mg/ml. This conjugation and concentration process was repeated on occasion to ensure a degree of labeling close to one. The ratio of Fab:dye, , was determined using the absorbance at 280 and 550 nm or 495 nm, the extinction coefficient of IgG at 280 nm, Ɛ , the extinction coefficient of the dye, Ɛ , provided by the manufacturer, and the dye correction factor at 280 nm, , provided by the manufacturer. The degree of labeling, , was calculated with the following formula:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint sfGFP and 33 µg/ml of purified MCP-HaloTag protein were mixed with PBS to a final volume of 4 µl. Second, in cell culture hood, DMEM was aspirated from the MatTek chamber and the 4 µl mix was pipetted on top of cells. Third, ~106 µm glass beads (Sigma Aldrich) were evenly sprinkled over the cells. Fourth, the chamber was tapped carefully ~10 times on the cell culture hood bench top. Fifth, DMEM was immediately added back to the cells. Sixth, cells were returned to the incubator for at least an hour to recover from the loading procedure. In most experiments, we also bead-loaded DNA, which we added to the initial 4 µl mix (so that DNA had a final concentration close to 1 mg/ml). On occasion, DNA was transiently transfected ~2 hours before bead-loading. Around one hour before experiments began, bead-loaded cells were washed with phenol-red-free complete DMEM to remove glass beads, and 200 nM of JF646-HaloTag ligand (a cell permeable fluorogenic ligand (Grimm et al., 2015) ) was added to label MCP-HaloTag protein. After 30 mins of incubation, cells were washed three times using phenol-red-free complete DMEM to remove any unconjugated fluorophores. Cells were then immediately imaged for experiments.',\n",
       "   'cite_spans': [{'start': 1200,\n",
       "     'end': 1220,\n",
       "     'text': '(Grimm et al., 2015)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'To track single molecule translation sites, a custom-built widefield fluorescence microscope based on a highly inclined illumination scheme (Tokunaga et al., 2008 ) was used (Morisaki et al., 2016 . Briefly, the excitation beams, 488, 561 and 637 nm solidstate lasers (Vortran), were coupled and focused at the back focal plane of the objective lens (60X, NA 1.49 oil immersion objective, Olympus). The emission signals were split by an imaging grade, ultra-flat dichroic mirror (T660lpxr, Chroma) and detected using two All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 140,\n",
       "     'end': 162,\n",
       "     'text': '(Tokunaga et al., 2008',\n",
       "     'ref_id': None},\n",
       "    {'start': 163,\n",
       "     'end': 196,\n",
       "     'text': ') was used (Morisaki et al., 2016',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Single molecule tracking microscopy'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. Readout time for the cameras from the combination of our imaging size, readout mode, and the vertical shift speed was 23.36 msec, resulting in an imaging rate of 13 Hz (70 msec per image). The excitation laser lines were digitally synched to ensure they only illuminated cells when the camera was exposing in order to avoid excessive photobleaching. To capture the entire volume of the cytoplasm of U-2 OS cells, 13 zstacks with step size of 500 nm (6 µm in total) were imaged using the piezoelectric stage such that the z-position changed every 2 images (one image for Cy3 and one for All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Single molecule tracking microscopy'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint sfGFP/A488 + JF646). The position of the filter wheel was changed during the camera readout time. This resulted in a total cellular imaging rate of 0.5 Hz (2 sec per volume for 3-colors). Note that all colors described in the text and that are shown in the figures are based on the color of the excitation laser: RNA in red (JF646) and protein in green (Cy3) or blue (sfGFP/A488).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Single molecule tracking microscopy'},\n",
       "  {'text': 'Images were first pre-processed using either Fiji (Schindelin et al., 2012) or a customwritten batch processing Mathematica code (Wolfram Research) to make 2D maximum intensity projections from 3D images. Pre-processed images were then analyzed with a custom-written Mathematica code to detect and track particles. Specifically, particles were emphasized with a band-pass filter so the positions could be detected using the built-in Mathmatica routine ComponentMeasurements \"IntensityCentroid\". Detected particles were linked through time by allowing a maximum displacement of 5 pixels between consecutive frames. Particle tracks lasting at least 5-10 frames were selected and their precise coordinates were determined by fitting (using the built-in Mathematica routine NonlinearModelFit) the original 2D maximum intenstiy projected images to a 2D Gaussians of the following form:',\n",
       "   'cite_spans': [{'start': 45,\n",
       "     'end': 75,\n",
       "     'text': 'Fiji (Schindelin et al., 2012)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'where is the background intensity, the particle peak intensity, ( , ) the spread of the particle, and ( 0 , 0 ) the particle location. The offset between the two cameras was All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint registered using the built-in Mathematica routine FindGeometricTransform to find the transform function that best aligned the fitted positions of 100 nm diameter Tetraspeck beads evenly spread out across the image field-of-view. We did not register the images, but only the fitted positions in order to avoid introducing any distortion into images. This is why a slight offset can be observed between the red and the green/blue particles even though they are within a diffraction limited spot, according to our registration.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'For visuallization and some quantification, average intensity image trims were created by averaging images of all detected particles of a given species (each centered by their intensity centroid). To compute the average number of ribosomes in a translation site (see Fig. S4 ), the average image trims of translation sites and single mature proteins were fitted to Eq. (2). The average intensities were calculated by integrating the fitted Gaussian. The ratio of the average intensity of translation sites to that of single mature proteins was then calculated to give the average number of ribosomes in a translation site. The average intensity of single RNAs were also calculated using the same procedure ( Fig. 3C , S5, and S6). Fits were susceptible to noise, so we also used an alternative strategy to determine the average intensity of translation sites and RNA that was robust to noise (in all Figs aside from Fig. 3C , S4, S5, and S6). Specifically, the intensity of centered images of RNA or translation sites was calculated to be the average intensity within a centered radial spot of four pixels in diameter minus the average background intensity from a centered ring with an outer diameter of twelve pixels and an inner diameter of eight pixels. Mathematica source code is available upon request.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 267,\n",
       "     'end': 274,\n",
       "     'text': 'Fig. S4',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 708, 'end': 715, 'text': 'Fig. 3C', 'ref_id': 'FIGREF12'},\n",
       "    {'start': 916, 'end': 923, 'text': 'Fig. 3C', 'ref_id': 'FIGREF12'}],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translation Site Species Identification'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https: //doi.org/10.1101/478040 doi: bioRxiv preprint After RNA particles were identified and tracked using the custom Mathematica code described above, an average centered image of the first five frames from each track was created for RNA (JF646), 0 ORF (FLAG-Cy3 in the +FSS multi-frame tag), and -1 ORF (scFv-sfGFP in the +FSS multi-frame tag). The trims were then hand checked to remove any trims with artifacts, e.g., smears or non-diffraction limited spots. Next, a custom Mathematica code was used to detect particles in the 0 ORF or -1 ORF trim channels, sorting the spots into RNA only, 0 frame translation sites (0 TS only), 0 and -1 TS, and -1 only TS. For all cases, RNA always had to be present. Finally, frameshifting translation sites (the 0 and -1 TS or the -1 only TS) were validated by eye, to further remove artifacts.',\n",
       "   'cite_spans': [{'start': 100,\n",
       "     'end': 146,\n",
       "     'text': '//doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translation Site Species Identification'},\n",
       "  {'text': 'For example, RNA that briefly colocalized with a mature protein puncta were removed at this stage. After all sites were validated, the total count of each type of species was used to determine the percentage of non-translating RNA (no TS), 0 frame translation sites (0 TS), 0 and -1 TS, and -1 only TS.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translation Site Species Identification'},\n",
       "  {'text': 'To confirm active translation elongation, puromycin (Sigma Aldrich) was used to release nascent chains from elongating ribosomes, leading to a rapid loss of nascent chain signal at translation sites. Bead-loaded cells with visible translation sites were imaged at a rate of one volume every seven seconds. After acquiring 16 pre-images, cells were treated with a final concentration of 100 µg/ml puromycin and continuously imaged for an additional 100 time points. As a control, the same imaging conditions were performed except that the cells were treated with vehicle (H2O). In this case, nascent chain signals All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Puromycin treatment'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint did not disappear (data not shown). Both frameshifting and non-frameshifting translation sites were monitored through time using the tracking code described above.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Puromycin treatment'},\n",
       "  {'text': \"Oligo FSS RNA co-transfection An uncapped RNA oligo containing the FSS from HIV-1 was synthesized from IDTDNA with the following RNA sequence: 5'-CUG GCC UUC CCU UGU GGG AAG GCC AGA UCU UCC CUA AAA AA -3'. Co-transfection of FSS-MF-AlexX and the RNA oligo was carried out via bead-loading, as describe above. Briefly, 1 µg plasmid FSS-MF-AlexX construct DNA, 100 µg/ml of fluorescently labeled Fab, 250 µg/ml of purified (GCN4) scFv-sfGFP or 1 µg of (GCN4) scFv-sfGFP plasmid, 1 or 4 µg of RNA oligo, and 33 µg/ml of purified MCP-HaloTag protein were mixed with PBS to a final volume of ~4 µl of PBS.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Puromycin treatment'},\n",
       "  {'text': 'To measure the average elongation rate, harringtonine (Cayman Chemical) was used to block translation initiation and iduce the run-off of all actively elongating ribosomes, leading to a gradual loss of nascent chain signal at translation sites. Beadloaded cells with visible translation sites were imaged as described above except that cell volumes were acquired every 60 or 120 seconds. After acquiring 5 pre-images, cells were treated with a final concentration of 3 µg/ml Harringtonine and continuously imaged for 30 more time points. As a control, the same imaging conditions were performed except that cells were treated with vehicle (DMSO). In this case, nascent chain signals did not decay (data not shown). The intensities of translation sites measured, as described above. The All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint intensity of all translation sites in each frame (and in all cells) were then totaled to produce the run-off curve. Run-off curves were normalized to the mean of the total intensity of the first four time points after treatment of harringtonine began. These curves were then fit to a linear regression to estimate run-off times (see Fig. S7 ). The linear portion of the runoff decay begins when the normalized run-off intensity reaches a fraction 0 , where',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 479,\n",
       "     'end': 486,\n",
       "     'text': 'Fig. S7',\n",
       "     'ref_id': 'FIGREF23'}],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'where 1 is the length of untagged portion of the open reading frame and 2 is the length of the tagged portion (i.e. the length of the repeated epitopes), as described (Morisaki and Stasevich, 2018). The linear portion of the decay was then interpolated to background levels to estimate the run-off time . The elongation rate was calculated as follows:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'Fluorescence recovery after photobleaching (FRAP)',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'To confirm elongation rates of frameshifting translation sites, FRAP experiments were performed. Bead-loaded cells with visible translation sites were imaged once every 10 seconds, with an intentional photobleach at frame 10. The photobleach was performed with a 405 nm laser focused to a spot roughly 5 \\uf06dm in diameter and operating at a minimal power such that the nascent chain signal did not completely vanish. This allowed us to track the translation site continously throughout the experiment. Following the intentional photobleach, the fluorescence recovery of translation sites within the bleach zone were monitored for an additional 80 time points. These translation sites were tracked and their intensities quantified, as described above. To correct for unintentional photobleaching, All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint the intensity of a control translation site was also tracked and quantified. The loss of signal from the control translation site was fit to a single exponential decay and this decay was divided out from the FRAP recovery curves of the intentionally photobleached translation sites. The FRAP recovery curve can be thought of as the inverse of the harringtonine runoff curve (Morisaki and Stasevich, 2018) . In this way, the FRAP recovery curve was fit to determine the average translation elongation rate (Fig. S8 ).',\n",
       "   'cite_spans': [{'start': 520,\n",
       "     'end': 550,\n",
       "     'text': '(Morisaki and Stasevich, 2018)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 651,\n",
       "     'end': 659,\n",
       "     'text': '(Fig. S8',\n",
       "     'ref_id': 'FIGREF24'}],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'The frameshift models describe the stochastic dynamics of nascent translation with single-codon resolution. Both the constitutive and bursting models are formulated to track an arbitrary number of individual ribosomes that can perform three possible reactions: (i)',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': \"A new ribosome can initiate translation at the start codon at rate kini. (ii) An existing ribosome can elongate at rate ke to incorporate one amino acid into the nascent peptide chain. The rate ke is sequence-specific with each codon's rate scaled by the genomic copy of the corresponding codon (Nakamura et al., 1999) . When the ribosome completes the final amino acid, translation is terminated and that ribosome is eliminated. (iii) A ribosome at the frameshift sequence can shift from the 0 to the -1 frame. In addition to these three reactions, ribosomes at the frameshift sequence can pause for an average time of 1/ * or 1/ for the shifted and non-shifted states. For the bursting model, each RNA is assumed to switch back and forth between non-frameshifting and frameshifting states with rates and . With these mechanisms and parameters, the models can be analyzed using either simplified approximations or detailed simulations.\",\n",
       "   'cite_spans': [{'start': 295,\n",
       "     'end': 318,\n",
       "     'text': '(Nakamura et al., 1999)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Several approximate features can be derived directly from the bursting model parameters (see Table 1 ). The average burst time or time spent in a frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 239,\n",
       "     'end': 246,\n",
       "     'text': 'Table 1',\n",
       "     'ref_id': 'TABREF1'}],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The average time spent in a non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The fraction of RNA in a frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The number of ribosomes that initiate during the frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The number of ribosomes that initiate during the non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The time for a ribosome to clear the FSS in the frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'where the second term accounts for ribosome pileup in the 9-codon ribosomal exclusion (Ingolia et al., 2009) , , region upstream from the FSS. Similarly, the time for a ribosome to clear the FSS in the non-frameshifting state is:',\n",
       "   'cite_spans': [{'start': 86,\n",
       "     'end': 108,\n",
       "     'text': '(Ingolia et al., 2009)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint One can approximate = 1 or 0 if initiation is faster ( > ) or slower ( > )',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'than the FSS pause. The number of ribosomes to pass the FSS in the frameshifting state',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The number of ribosomes to pass the FSS in the non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The average number of ribosomes to accumulate upstream of the FSS in the frameshifting state is: * = * − *',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'and the number to clear in a non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'In equations (13) and (14), positive values of correspond to accumulation of ribosomes in the traffic jam, and negative values correspond to clearance of ribosomes. The average elongation time for a ribosome in the frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'and the number to clear in a non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'where is the gene length in codons.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': \"Simulations were started at t = -10,000 seconds to approximate steady state at t = 0 using the Gillespie algorithm (Gillespie, 1976) . Ribosome densities were found by collecting position statistics for multiple simulations. Simulated ribosome numbers and positions and multi-frame tag probe locations were combined to estimate translation site intensities. Harringtonine assays were simulated by preventing the initiation reaction at the time of treatment. Parameter estimation was performed using genetic algorithms and a multiple-objective cost function that considers the frameshifting efficiency, the number of ribosomes per RNA and the Harringtonine assays. A detailed description of the computational methods and codes is given in the 'Computational Details' section below.\",\n",
       "   'cite_spans': [{'start': 115,\n",
       "     'end': 132,\n",
       "     'text': '(Gillespie, 1976)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'Both the constitutive model and the bursting model consist of three general ribosomal reaction types: initiation ( 0 ) , elongation ( or ̂) and frameshifting ( or ̂) as depicted in Eq. 17,',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'Ribosome Initiation. Ribosomes are assumed to bind to the RNA with rate . To account for the fact that ribosomes are large biomolecules that occupy around 20 to 30 All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint nuclear bases on the RNA (Ingolia et. al, 2009) , initiation is blocked by any downstream ribosome within = 9 codons (in either frame). Therefore, initiation rate is set at:',\n",
       "   'cite_spans': [{'start': 171,\n",
       "     'end': 193,\n",
       "     'text': '(Ingolia et. al, 2009)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': \"Ribosome Elongation. Each ribosome moves along the RNA codon by codon in the 5' to 3' direction. The elongation rate for each ℎ codon, ̂ is assumed to be:\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'where denotes the codon usage frequency in the human genome obtained from (Nakamura et. al, 2000) , and ̅ represents the average codon usage frequency in the human genome. The fit parameter specifies the average elongation rate. Ribosomal termination is assumed to be equivalent to elongation of the final codon. ',\n",
       "   'cite_spans': [{'start': 74,\n",
       "     'end': 97,\n",
       "     'text': '(Nakamura et. al, 2000)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'or the ribosome can continue in the -1 frame at rate: All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'For the bursting model, the decision to continue in the 0 or -1 frame depends upon the frameshifting state with the rate given by: ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'Relating model dynamics to experimental fluorescence intensity. To relate ribosome occupancy to experimental translation spots, we proscribe an assumed fluorescence to each ribosome based upon its position. This intensity is proportional to the number of peptide probes upstream from the ribosome location and in the appropriate frame or frames. Ribosomes in the 0 frame include all upstream probes in the 0 frame. Ribosomes in the -1 frame include probes in the 0 frame between the start codon and the FSS (if any) and probes in the -1 frame between the FSS and the current ribosome position. The total intensity vector for the ℎ color is given by',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'where ( ) denotes the position (i.e., frame and codon location) of the ℎ ribosome, and ( ) is the corresponding intensity of the ℎ color.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Fraction of translation spots. Experimental data was measured between 2 to 6 hours post bead-loading for a period of 120 seconds. To reproduce these experimental data, the model was solved using 100 trajectories of 120 seconds starting at steady state. Spots were classified as 0 frame only ( 0 ) , -1 frame only ( −1 ) and both frames ( ) .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Following the experimental observation that frameshifting spots commonly consist of more than one RNA, were equally added to the 0 and −1 . The percentage of nonframeshifting translation spots was calculated as follows:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'where 0 is the sum of 0 and is the total number of translating spots.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The percentage of frameshifting translation spots was calculated as follows: * = ( −1 + ( 1 2 ⁄ ) ) • 100,',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'where −1 is the sum of −1 .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Number of ribosomes per RNA. Experimental observations allowed to quantify the number of ribosomes in frameshifting and non-frameshifting translation spots ( Figure 2 in the main text). To relate model simulations and data, the number of ribosomes per RNA was estimated by directly by adding the number of ribosomes in each RNA at steady state.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 158,\n",
       "     'end': 166,\n",
       "     'text': 'Figure 2',\n",
       "     'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Harringtonine assays. Harringtonine inhibits translation by binding to the ribosomal 60S sub-unit, which blocks new initiation events. Experimental data showed that Harringtonine causes the intensity in translating spots to drop to a basal intensity value after a run-off time (Figure 4 in main text). To mimic the effects of Harringtonine in our model we modified the initiation rate as:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 423,\n",
       "     'end': 432,\n",
       "     'text': '(Figure 4',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': '< and codons {1, … , } unoccupied,',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': \"where is the time of application of Harringtonine. After Harringtonine application, spots simulated from the original construct were classified as 0 frame only ('0F') or both 0 frame and -1 frames ('BF'). After classification, average spot intensities were quantified as:\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': \"where (.) is experimental background expression obtained at the end of the experimental time of the run-off assays. Similarly, spots simulated for the extended construct with upstream HA tags and downstream -1 tags were classified as HA in non-shifted spots ('HA') or HA in shifted spots ('HA*').\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint ( ) = 〈 ( )〉 + , (29) * ( ) = 〈 * ( )〉 + * .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Parameter estimation. For each model, we sought to find a single parameter set that reproduces all experimental data. Given the different sources of experimental data we estimated the parameter values using a multi-objective optimization strategy to simultaneously compare the Harringtonine run-off data, the fractions of shifted and nonshifted spots and the ribosome occupancy, as follows:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'where experimental data are denoted as ̃ and simulations results as ̂. Weights ( , and ) were used to balance constraints by the different experiments and time points.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'represents the number of experimental points in the Harringtonine assays ( = 30), is the number of studied frames ( = 2), is the number of different gene constructs ( = 2), and is the number of different data types ( = 3). The weight used on the Harringtonine data set was defined as = 1 ( • • ) ⁄ . The weight in the fraction of spots per frame was defined as = 1 ( + * ) ⁄ . And the weight for the number of ribosomes was defined as: = 1 ( + * ) ⁄ .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Parameter searches. Parameter searches consisted of optimization routines based on genetic algorithms (GA) using a population of 100 individuals and running the search for 20 generations.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The constitutive model has a total of four parameters ( , , and * ), whereas the bursting model has two additional parameters ( , ). The optimization strategy consisted in to fit four free parameters ( , , and * ) in the models, parameters ( , ) were independently determined from data in Fig. 5B ,C). The parameter set that best reproduces the data was selected as:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 289,\n",
       "     'end': 296,\n",
       "     'text': 'Fig. 5B',\n",
       "     'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': '= argmax − log 10 ( ) .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Optimized parameter values are given in full detail in All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'}],\n",
       " 'bib_entries': {'BIBREF1': {'ref_id': 'b1',\n",
       "   'title': 'XLas, the extra-long form of the -subunit of the Gs G protein, is significantly longer than suspected, and so is its companion Alex',\n",
       "   'authors': [],\n",
       "   'year': None,\n",
       "   'venue': 'Proc. Natl. Acad. Sci',\n",
       "   'volume': '101',\n",
       "   'issn': '',\n",
       "   'pages': '8366--8371',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF2': {'ref_id': 'b2',\n",
       "   'title': 'Extralarge XLαs (XXLαs), a Variant of Stimulatory G Protein α-Subunit (Gsα), Is a Distinct',\n",
       "   'authors': [{'first': 'C', 'middle': [], 'last': 'Aydin', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Aytan', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['J'], 'last': 'Mahon', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['A W'], 'last': 'Tawfeek', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['W'], 'last': 'Kowall', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Dedeoglu', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Bastepe', 'suffix': ''}],\n",
       "   'year': 2009,\n",
       "   'venue': 'Membrane-Anchored GNAS Product that Can Mimic Gsα. Endocrinology',\n",
       "   'volume': '150',\n",
       "   'issn': '',\n",
       "   'pages': '3567--3575',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF3': {'ref_id': 'b3',\n",
       "   'title': 'Identifying the assembly intermediate in which Gag first associates with unspliced HIV-1 RNA suggests a novel model for HIV-1 RNA packaging',\n",
       "   'authors': [{'first': 'B',\n",
       "     'middle': ['C'],\n",
       "     'last': 'Barajas',\n",
       "     'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Tanaka', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['A'], 'last': 'Robinson', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['J'], 'last': 'Phuong', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Chutiraka', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['C'], 'last': 'Reed', 'suffix': ''},\n",
       "    {'first': 'Lingappa', 'middle': [], 'last': '', 'suffix': ''}],\n",
       "   'year': 2018,\n",
       "   'venue': 'J.R',\n",
       "   'volume': '',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF4': {'ref_id': 'b4',\n",
       "   'title': 'Ribosomal frameshifting in the CCR5 mRNA is regulated by miRNAs and the NMD pathway',\n",
       "   'authors': [{'first': 'A', 'middle': ['T'], 'last': 'Belew', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Meskauskas', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Musalgaonkar', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '512',\n",
       "   'issn': '',\n",
       "   'pages': '265--269',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF5': {'ref_id': 'b5',\n",
       "   'title': 'Programmed ribosomal frameshifting in HIV-1 and the SARS-CoV',\n",
       "   'authors': [{'first': 'I', 'middle': [], 'last': 'Brierley', 'suffix': ''},\n",
       "    {'first': 'Dos', 'middle': [], 'last': 'Ramos', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': ['J'], 'last': '', 'suffix': ''}],\n",
       "   'year': 2006,\n",
       "   'venue': 'Virus Res',\n",
       "   'volume': '119',\n",
       "   'issn': '',\n",
       "   'pages': '29--42',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF6': {'ref_id': 'b6',\n",
       "   'title': 'Characterization of an Efficient Coronavirus Ribosomal Frameshifting Signal : Requirement for an RNA Pseudoknot',\n",
       "   'authors': [{'first': 'I', 'middle': [], 'last': 'Brierley', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Digard', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Stephen', 'suffix': ''}],\n",
       "   'year': 1989,\n",
       "   'venue': '',\n",
       "   'volume': '57',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF8': {'ref_id': 'b8',\n",
       "   'title': 'Programmed -1 frameshifting by kinetic partitioning during impeded translocation',\n",
       "   'authors': [],\n",
       "   'year': None,\n",
       "   'venue': 'Cell',\n",
       "   'volume': '157',\n",
       "   'issn': '',\n",
       "   'pages': '1619--1631',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF9': {'ref_id': 'b9',\n",
       "   'title': 'Changed in translation: mRNA recoding by −1 programmed ribosomal frameshifting',\n",
       "   'authors': [{'first': 'N', 'middle': [], 'last': 'Caliskan', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Peske', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['V'], 'last': 'Rodnina', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'Trends Biochem. Sci',\n",
       "   'volume': '40',\n",
       "   'issn': '',\n",
       "   'pages': '265--274',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF10': {'ref_id': 'b10',\n",
       "   'title': 'HIV-1 and Human PEG10 Frameshift Elements Are Functionally Distinct and Distinguished by Novel Small Molecule Modulators',\n",
       "   'authors': [{'first': 'T', 'middle': ['S'], 'last': 'Cardno', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Shimaki', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['E'], 'last': 'Sleebs', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Lackovic', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['P'], 'last': 'Parisot', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['M'], 'last': 'Moss', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Crowe-Mcauliffe', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['F'], 'last': 'Mathew', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['D'], 'last': 'Edgar', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Kleffmann', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'PLoS One',\n",
       "   'volume': '10',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF11': {'ref_id': 'b11',\n",
       "   'title': \"The 5' UTR of HIV-1 full-length mRNA and the Tat viral protein modulate the programmed -1 ribosomal frameshift that generates HIV-1 enzymes\",\n",
       "   'authors': [{'first': 'J',\n",
       "     'middle': [],\n",
       "     'last': 'Charbonneau',\n",
       "     'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Gendron', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Ferbeyre', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Brakier-Gingras', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'RNA',\n",
       "   'volume': '18',\n",
       "   'issn': '',\n",
       "   'pages': '519--529',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF12': {'ref_id': 'b12',\n",
       "   'title': 'Dynamic pathways of −1 translational frameshifting',\n",
       "   'authors': [{'first': 'J', 'middle': [], 'last': 'Chen', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Petrov', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Johansson', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Tsai', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['E'], 'last': 'O&apos;leary', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['D'], 'last': 'Puglisi', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '512',\n",
       "   'issn': '',\n",
       "   'pages': '328--332',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF13': {'ref_id': 'b13',\n",
       "   'title': 'Frameshift mutations in mammalian target of rapamycin pathway genes and their regional heterogeneity in sporadic colorectal cancers',\n",
       "   'authors': [{'first': 'M', 'middle': ['R'], 'last': 'Choi', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['J'], 'last': 'Yoo', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['H'], 'last': 'An', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['H'], 'last': 'Lee', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'Hum. Pathol',\n",
       "   'volume': '46',\n",
       "   'issn': '',\n",
       "   'pages': '753--760',\n",
       "   'other_ids': {}}},\n",
       " 'ref_entries': {'FIGREF0': {'text': 'Tracking (NCT) (Morisaki et al., 2016) of the RNA revealed a high degree of RNA-to-RNA heterogeneity, with a subset of RNA labeled by Fab onlyindicative of FLAG epitopes from translation in the 0 frameand a smaller subset labeled by both Fab and scFvindicative of both FLAG and SunTag epitopes from canonical and frameshifted translation in the 0 and -1 frames, respectively (Fig. 1B,C and Movie S1).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF1': {'text': 'frameshifted ribosomes (marked by the SunTag) fluctuated wildly up and down (Fig. 4C, and Movie S9). These fluctuations appear to arise from the stochastic release of stalled ribosomes within the traffic jam and their subsequent translation of frameshifted epitopes.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF2': {'text': 'state or structure, similar to what has been shown in vitro with sequences encoding pseudoknots (Houck-Loomis et al., 2011). Longer pauses result in longer ribosomal traffic jams. Our model predicts these jams can extend all the way back to the initiation site, involving up to ~40 ribosomes, as shown in the lower panel of Fig. 5F. Our model predicts All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF3': {'text': 'production of downstream frameshifted protein. This unique mechanism would allow viral proteins to continue to increase in numbers during cellular stress. An open question is how these long traffic jams manage to evade protein quality control (Joazeiro, 2017; Juszkiewicz and Hegde, 2017), which recently was shown to target the interface of jammed ribosomes (Juszkiewicz et al., 2018).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF4': {'text': 'We thank Dr. Hataichanok (Mam) Scherman of the Colorado State University Protein Expression/Purification Facility and Amanda Koch for cloning and purification of the SunTag scFv-sfGFP and MCP-Halo. We thank all members of the Stasevich and Munsky labs for helpful discussion and comments. This work is supported by the W.M. Keck Foundation. AUTHOR CONTRIBUTIONS Conceptualization, TJS, KL, and BM; Methodology: KL, LA, TM, BM, and TJS; Software, KL, LA, TM, BM, and TJS; Formal Analysis, KL and LA; Experimentation, KL, TM, and TJS; Computational modeling, LA and BM; Resources, BM and TJS; Writing, Review, and Editing, KL, LA, TM, BM, and TJS; Supervision, BM and TJS; Funding Acquisition, BM and TJS.All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF5': {'text': ', M.B., Jänicke, M., Gottesbühren, U., Kleffmann, T., Legge, M., Poole, E.S., and Tate, W.P. (2007). Mammalian gene PEG10 expresses two reading frames by high efficiency -1 Frameshifting in embryonic-associated tissues. J. Biol. Chem. 282, 37359-37369.All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF6': {'text': \"Edelstein, A.D.,Tsuchida, M.A., Amodaj, N., Pinkard, H., Vale, R.D., and Stuurman, N.  (2014). Advanced methods of microscope control using μManager software. J. Biol. Methods 1, 10.Gillespie, D.T. (1976). A general method for numerically simulating the stochastic time evolution of coupled chemical reactions. J. Comput. Phys.Gorgoni, B., Ciandrini, L., McFarland, M.R., Romano, M.C., and Stansfield, I. (2016). Identification of the mRNA targets of tRNA-specific regulation using genome-wide simulation of translation. Nucleic Acids Res.Grentzmann, G., Ingram, J.A., Kelly, P.J., Gesteland, R.F., and Atkins, J.F. (1998). A dual-luciferase reporter system for studying recoding signals. RNA 4, 479-486. Grimm, J.B., English, B.P., Chen, J., Slaughter, J.P., Zhang, Z., Revyakin, A., Patel, R., Macklin, J.J., Normanno, D., Singer, R.H., et al. (2015). A general method to improve fluorophores for live-cell and single-molecule microscopy. Nat. Methods. Guerrero, S., Batisse, J., Libre, C., Bernacchi, S., Marquet, R., and Paillart, J.-C. (2015). HIV-1 replication and the cellular eukaryotic translation apparatus. Viruses 7, 199-218. Hayashi-Takanaka, Y., Yamagata, K., Wakayama, T., Stasevich, T.J., Kainuma, T., Tsurimoto, T., Tachibana, M., Shinkai, Y., Kurumizaka, H., Nozaki, N., et al. (2011). Tracking epigenetic histone modifications in single cells using Fab-based live endogenous modification labeling. Nucleic Acids Res. 39, 6475-6488. Houck-Loomis, B., Durney, M.A., Salguero, C., Shankar, N., Nagle, J.M., Goff, S.P., and D'Souza, V.M. (2011). An equilibrium-dependent retroviral mRNA switch regulates translational recoding. Nature 480, 561-564. Ingolia, N.T., Ghaemmaghami, S., Newman, J.R.S., and Weissman, J.S. (2009). Genome-wide analysis in vivo of translation with nucleotide resolution using ribosome profiling. Science (80-. ). Joazeiro, C.A.P. (2017). Ribosomal Stalling During Translation : Providing Substrates for Ribosome-Associated Protein Quality Control. 1-26. Juszkiewicz, S., and Hegde, R.S. (2017). Initiation of Quality Control during Poly(A) Translation Requires Site-Specific Ribosome Ubiquitination. Mol. Cell 65, 743-750.e4. Juszkiewicz, S., Chandrasekaran, V., Lin, Z., Kraatz, S., Ramakrishnan, V., and Hegde, R.S. (2018). ZNF598 Is a Quality Control Sensor of Collided Ribosomes. Mol. Cell 1-13.Kontos, H., Napthine, S.,and Brierley, I. (2001). Ribosomal pausing at a frameshifter RNA pseudoknot is sensitive to reading phase but shows little correlation with frameshift efficiency.Mol. Cell. Biol. 21, 8657-8670.   Lionnet, T., and Singer, R.H. (2012). Transcription goes digital. EMBO Rep. 13, 313-321. All rights reserved. No reuse allowed without permission.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF7': {'text': 'S., Klepacki, D., Karthikeyan, S., Margus, T., Thomas, P., Jones, J.E., Khan, Y., Briggs, J., Dinman, J.D.,Vázquez-Laslop, N., et al. (2017). Programmed Ribosomal Frameshifting Generates a Copper Transporter and a Copper Chaperone from the Same Gene. Mol. Cell 65, 207-219. Morisaki, T., and Stasevich, T.J. (2018). Quantifying Single mRNA Translation Kinetics in Living Cells. Cold Spring Harb. Perspect. Biol. 10, a032078. Morisaki, T., Lyon, K., DeLuca, K.F., DeLuca, J.G., English, B.P., Zhang, Z., Lavis, L.D., Grimm, J.B., Viswanathan, S., Looger, L.L., et al. (2016). Real-time quantification of single RNA translation dynamics in living cells. Science 352, 1425-1429. Mouzakis, K.D., Lang, A.L., Vander Meulen, K. a., Easterday, P.D., and Butcher, S.E. (2013). HIV-1 frameshift efficiency is primarily determined by the stability of base pairs positioned at the mRNA entrance channel of the ribosome. Nucleic Acids Res. 41, 1901-1913. Munsky, B., Neuert, G., and Van Oudenaarden, A. (2012). Using gene expression noise to understand gene regulation. Science (80-. ). Nakamura, Y., Gojobori, T., and Ikemura, T. (1999). Codon usage tabulated from the international DNA sequence databases; its status 1999. Nucleic Acids Res. Napthine, S., Ling, R., Finch, L.K., Jones, J.D., Bell, S., Brierley, I., and Firth, A.E. (2017). Protein-directed ribosomal frameshifting temporally regulates gene expression. Nat. Commun. 8, 1-11. Pichon, X., Bastide, A., Safieddine, A., Chouaib, R., Samacoits, A., Basyuk, E., Peter, M., Mueller, F., and Bertrand, E. (2016). Visualization of single endogenous polysomes reveals the dynamics of translation in live human cells. J. Cell Biol. 214, 769-781. Ritchie, D.B., Tittle, C., Rezajooei, N., Rouleau, L., Cappellano, T.R., Sikkema, W., and Woodside, M.T. (2017). Folding Heterogeneity in HIV-1 Frameshifting Hairpin. Biophys. J. 112, 368a.Schindelin, J.,Arganda-Carreras, I., Frise, E., Kaynig, V., Longair, M., Pietzsch, T.,  Preibisch, S., Rueden, C., Saalfeld, S., Schmid, B., et al. (2012). Fiji: An open-source platform for biological-image analysis. Nat. Methods 9, 676-682.Sharp, P.M., and Li, W.H. (1987). The codon adaptation index-a measure of directional synonymous codon usage bias, and its potential applications. Nucleic Acids Res.Tokunaga, M., Imamoto, N., and Sakata-Sogawa, K. (2008). Highly inclined thin illumination enables clear single-molecule imaging in cells. Nat. Methods 5, 159-161. All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF8': {'text': 'for frameshifting. Proc. Natl. Acad. Sci. U. S.A. 89, 8636-8640.   Viswanathan, S., Williams, M.E., Bloss, E.B., Stasevich, T.J., Speer, C.M., Nern, A.,  Pfeiffer, B.D., Hooks, B.M., Li, W.-P., English, B.P., et al. (2015). High-performance probes for light and electron microscopy. Nat. Methods 12, 568-576.Voigt, F., Zhang, H., Cui, X.A., Triebold, D., Liu, A.X., Eglinger, J., Lee, E.S., Chao, J.A., and Palazzo, A.F. (2017). Single-Molecule Quantification of Translation-Dependent Association of mRNAs with the Endoplasmic Reticulum. Cell Rep. 21, 3740-3753. Wang, C., Han, B., Zhou, R., and Zhuang, X. (2016). Real-Time Imaging of Translation on Single mRNA Transcripts in Live Cells. Cell 165, 990-1001. Wu, B., Eliscovich, C., Yoon, Y.J., and Singer, R.H. (2016). Translation dynamics of single mRNAs in live cells and neurons. Science 352, 1430-1435. Yordanova, M.M., Wu, C., Andreev, D.E., Sachs, M.S., and Atkins, J.F. (2015). A nascent peptide signal responsive to endogenous levels of polyamines acts to stimulate regulatory frameshifting on antizyme mRNA. J. Biol. Chem. 290, 17863-17878.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF9': {'text': \"A multi-frame tag to image single RNA frameshifting dynamics in living cells (A) The multi-frame (MF) tag contains 12 repeated FLAG epitopes in the 0 frame interspaced between 12 repeated SunTag v4 epitopes in the -1 frame. Depending on which frame is translated, nascent epitopes are labeled by fluorescent anti-FLAG antibody fragments (Cy3-Fab, green 'Y') or anti-SunTag single chain variable fragments (scFv-GFP, blue 'Y'). Following the repeat epitopes is exon 1 of the GNAS locus, in which the peripheral membrane proteins AlexX (689 aa) and XXLb1 (690 aa) were placed in the 0 and -1 frames, respectively. Preceding the multi-frame tag is a multi-frame element (MFE). In this study, the HIV-1 frameshift sequence was used as the MFE. To facilitate single-RNA tracking, a 24X MS2 stem-loop tag was also placed in the 3' UTR. This tag is labeled by MCP-HaloTag (with JF646-HaloTag ligand, red). (B) A representative cell ~10 hours after transient transfection with the multi-frame tag depicted in A. The red circle (labeled '1') highlights a non-translating RNA, the yellow circle (labeled '2') highlights a 0frame translation site (TS), and the white circle (labeled '3') highlights a 0 & -1 TS. Scale bar, 10 µm. (C) Montages showing the temporal evolution of the RNA spots circled in B. (D) A representative montage showing the loss of signal from the 0 and -1 open reading frames upon addition of the translational inhibitor puromycin (100 µg/mL).\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF10': {'text': 'Quantification of HIV-1 stimulated frameshifting (A) A schematic of the multi-frame (+FSS) and control (-FSS) reporters. (B) Average image trims of all non-translating RNA sites (no TS), 0 frame translation sites (TS), 0 & -1 TS, and -1 TS, with their respective merges. (C) The number of RNA detected per cell All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF11': {'text': 'RNA). The pie charts show the percentage of translating species per transfected cell. The number of detected RNA for each species is shown above each bar. Error bars represent S.E.M. among cells. (D) The number of detected ribosomes per translation site for the +FSS and -FSS reporters. Error bars represent S.E.M. among RNA. (E) The mean squared displacement (MSD) of tracked RNA species as a function of time. Error bars represent S.E.M. among RNA. (F) The average distance (µm) of detected translation sites from the nucleus. An outline of a representative cell on the right shows all detected RNA within the cell and their measured distance from the nuclear border (inner curve).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF12': {'text': 'The frameshift sequence stimulates frameshifting in multi-RNA translation factories (A) Cells were co-transfected with increasing concentrations of short oligo RNA encoding just the frameshift sequence, together with either the multi-frame reporter (+FSS) or control reporter (-FSS). (B) The percentage of translation sites translating just the 0 frame (green), the 0 and -1 frame (cyan), or just the -1 frame (dark blue) at all oligo concentrations for the +FSS and -FSS reporters. (C) The ratio of the intensity of RNA signals within frameshifting sites to non-frameshifting RNA at all oligo concentrations for the +FSS and -FSS reporters. In all bar graphs, the total number of each type of RNA species is shown above each bar and the error represents S.E.M. between cells.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF13': {'text': 'Ribosomal run-off at frameshifting and non-frameshifting translation sites showing harringtonine-induced ribosomal run-off from the +FSS 2x multiframe RNA reporter with FLAG (green) and SunTag epitopes (blue) in the 0 and -1 frames. The intensity of nascent chain signals within non-frameshifting translation sites(marked by just anti-FLAG Fab, green triangles, 217 translation sites initially in 19 cells) and frameshifting sites (marked by anti-SunTag scFv, cyan circles, 32 translation sites initially) drops with time as ribosomes complete translation, i.e. run-off. (B) Same as A, but with a modified multi-frame reporter with extra HA epitopes (orange, HA multi-frame tag) upstream of the frameshift sequence. The intensity of nascent chain signals from non-frameshifting translation sites (marked by just anti-HA Fab, orange triangles, 175 translation sites initially in 27 cells) and frameshifting translation sites (marked by anti-SunTag scFv, orange circles, 42 translation sites initially) drops with time as ribosomes run-off. (C) A sample single translation site encoding the modified HA multi-frame reporter (shown in B) after addition of harringtonine. A montage of image trims shows the detected RNA-, HA-, and Sun-signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting ribosomes) is plotted through time. Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF14': {'text': 'A model for bursty frameshifting (A) A schematic of the model:',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF15': {'text': 'sites through time (gray dots) is fit with a single exponential decay (black line). (C-F) Simultaneous fit of all data. (C) A bar graph comparing the measured (black) and best-fit model predicted (gray) percentage of non-frameshifting (0F) and frameshifting (-1F) translation sites. Error bars represent S.E.M. among cells. (D) A bar graph comparing the measured (black) and best-fit model predicted (gray) number of ribosomes within non-frameshifting (0F) and frameshifting (-1F) translation sites. Error bars represent S.E.M. among RNA. (E) Best-fit model (solid lines) of the data from Error bars represent S.E.M. among RNA. Below, the predicted ribosomal occupancy along the original multi-frame reporter is shown. The positions of the FSS (red) and FLAG (green) and SunTag (blue) epitopes are shown in color. (F) Best-fit model prediction of the data from Figure 4B. Error bars represent S.E.M. among RNA. Below, the predicted ribosomal occupancy along the modified HA-tagged multi-frame reporter is shown. The positions of the FSS (red) and HA (orange), FLAG (green), and SunTag (blue) epitopes are shown in color. (G) Best-fit model prediction of the intensity distribution of HA signals for frameshifting (gray) and non-frameshifting sites (black) encoding the HA multi-frame tag.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF16': {'text': 'The lifecycle of RNA encoding the HIV-1 frameshift sequence RNA (red) encoding the HIV-1 frameshift sequence (stem loop) are activated for canonical 0-frame translation. Occasionally, they switch to a frameshifting state (~8%) that can last from tens of minutes to an hour or so. Transitions to this state are stimulated through interactions with other FSS-encoding RNA, either by chance or within multi-RNA translation factories. When in the frameshifting state, longer pauses at the FSS induce All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF17': {'text': 'Accumulation of mature AlexX and XXLb1 proteins from multi-frame tags Mature AlexX (upper) and XXLb1 (lower) membrane proteins accumulate in distinct punctae that coalesce to form elongated structures that undulate dynamically. These structures are enriched at the top and bottom of cells and around the edge of the cytoplasm (arrows), consistent with membrane localization. Movie S2 shows the dynamics of these structures. In the +FSS multi-frame reporter, mature AlexX comes from 0 frame canonical translation, while mature XXLb1 comes from -1 frameshifting translation. In the -FSS control reporter, only mature AlexX accumulates, indicating little to no frameshifting. The -FSS (+1 nt) column shows accumulation of XXLb1 when an extra nucleotide is inserted in the -FSS control reporter following the start codon. This extra nucleotide pushes XXLb1 into the 0 frame, demonstrating it can be expressed.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF18': {'text': 'A cell with all possible types of translation sites A sample cell exhibiting all four possible translation sites (TS). (1) untranslated TS; (2) 0 frame only TS; (3) 0 and -1 frame translated TS; and (4) -1 frame only TS. Scale bar = 10 µm. Movie S4 showcases the dynamics of these spots.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF19': {'text': 'Epitope order in tags has little impact on percentage of frameshifting RNA The percentage of RNA translating the -1 frame only (dark blue), the 0 and -1 frames (cyan), and the 0 frame only (green) for the +FSS multi-frame tag and the +FSS reverse tag (in which FLAG and SunTag epitopes are reversed). Pie charts show the percentage of translating species per cell. Error bars represent S.E.M. among cells. All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF20': {'text': 'Calibrating translation site intensity to the number of ribosomes(A) Example cell where single mature membrane proteins were detected (green circles), along with translation sites (co-localized RNA and protein; purple circles). (B) Average images of detected spots (33 mature proteins and 35 translation sites) were fit to a Gaussian to determine their intensity ratio, shown in the bar graph on the right. This ratio provides a calibration factor to convert translation site intensites to the number of riboeoms. The error bar represents the propagated 90% confidence interval from the fit.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF21': {'text': 'Frameshifting sites have brighter RNA signals (A-E) Average RNA signals for non-translating RNA (no TS), 0 only translation sites (0 TS), and -1 frameshifting translation sites (-1 TS). The number of RNA used to generate each average image is shown. The bar graphs below show the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). Experiments were done using (A) the +FSS multi-frame tag, (B) the +FSS reverse tag, (C) the +FSS multi-frame tag imaged without anti-FLAG Fab, and (D) the -FSS control tag. Error bars represent the fitted 90%confidence interval. Aside from the -FSS control, frameshifting sites (-1 TS) have significantly brighter RNA signals compared to canonical translation sites (0 TS). Note the frameshifting observed in the -FSS control represents less than 1% of translating RNA and can be considered background frameshifting.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF22': {'text': 'Upon oligo RNA co-transfection, stimulated frameshifting still occurs at bright RNA sites, even though the fraction of bright RNA sites remains unchanged (A) Average RNA signals from non-translating RNA (no TS), 0-frame only translation sites (0 TS), and frameshifting translation sites (-1 TS) when different concentrations of oligo All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF23': {'text': 'Fits to linear portion of ribosomal run-offs provide elongation rate estimates The total intensity of detected non-frameshifting and frameshifting translation sites decays with time after addition (at t = 0 sec) of the translational initiation inhibitor harringtonine. Data is taken from Fig. 4A. If L2 is the length of the tagged portion of the open reading frame and L1 is the length of the non-tagged portion, then the linear (post-tag) portion of the decay begins from approximately L1/(L1+L2/2). This portion of the curve provides an estimate of the elongation rate during the part of the run-off where no new epitopes arebeing translated. Therefore, the run-off fit is independent of the frameshifting kinetics that occur upstream of the epitopes. Curves are normalized to their initial values.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF24': {'text': 'Fluroescence recovery after photobleach of frameshifting sites (A) Fluorescence recovery after photobleaching (FRAP) experiments were performed at frameshifting translation sites (yellow circle marks the photobleach spot). The fluorescence recovery of the 0-frame signal within these sites was quantified as a function , with sample pre, post, and recovery frames shown above. (B) The average FRAP recovery time can be fit to estimate the elongation rate. This rate is similar to what was measured with harringtonine in Fig. S7.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF25': {'text': 'Ribosome run-off from the HA multi-frame tag Ribosome run-off curve showing Sun signal in HA&Sun sites from the experiment performed in Figure 4B. The Sun signal comes from frameshifted ribosomes that have run past the frameshift sequence (FSS). Error bars represent S.E.M.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF26': {'text': 'Comparison between the bursting and constitutive models Model dynamics for the bursting (A,C,E) and constitutive model (B,D,F). Simulations were performed using the best parameter values obtained from the optimization process. (A,B) Simulated time courses representing single molecule fluctuation dynamics from 100 translation sites (a sample trace is shown in bold). Green and blue lines represent the translation of FLAG (0 frame) and Sun (-1 frame) epitopes in the +FSS multi-frame tag, respectively. (C,D) On the left, bar graphs showing the experimental (black) and simulated (gray) percentages of 0 and -1 frame translation. On the right, the number of ribosomes translating either the 0 or -1 frame. (E,F) On the left, simulated run-off (solid lines) from the frameshift sequence (FSS) of all ribosomes in non-frameshifting (black)',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF27': {'text': 'black triangles and gray circles). Error bars represent the standard error of the mean (S.E.M.). Details can be found in the Supplementary Methods.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF28': {'text': 'Single frameshifting RNA track persisting for >35 minutesA sample frameshifting translation site encoding the +FSS 2x multi-frame tag was tracked for >35 minutes. Left shows the cell at the 1200 second time point. The zoom shows the All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF29': {'text': 'Codon usage plays a minor role in traffic jam formation (A) The codon adaptation index (CAI) for the 0 frame (top) and -1 frame (bottom) of the +FSS multi-frame tag. The y-axis shows the CAI calculated using the codon frequency in the human genome. The x-axis shows the length of the genes, in codons. In the plots, rare codons have low CAIs and common codons have high CAIs. The vertical red line represents the location of the frameshift sequence (FSS). Similar CAIs are obtained for the sequences in the 0 frame (CAI0F = 0.76) and the -1 frame (CAI-1F = 0.74).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF30': {'text': \"for 2275 seconds total. Left is the merge, right is Ch3 only. Scale bar, 10 µm. Movie S11. Sample simulation of best-fit bursty model with large ribosomal traffic jam A best-fit bursty model simulation of a translation site encoding the HA multi-frame tag. On the top, the simulated intensity of the site is tracked through time, with green showing signal from FLAG epitopes downstream of the FSS, blue showing signal from SunTag epitopes downstream of the FSS, and cyan showing signal from HA epitopes upstream of the FSS. On the bottom, individual ribosomes (gray dots) are shown moving along the reporter (gray line), with green, blue, and cyan squares marking the positions of FLAG,SunTag, and HA epitopes, respectively. As the ribosomes move along the repeat epitopes, the translated nascent chains appear as green (FLAG-tagged nascent chains), blue (Sun-tagged nascent chains), and cyan (HA-tagged nascent chains) dots that increase in size as more epitopes are translated. When the RNA is in the nonframeshifting state, the FSS is labeled 'FSS(OF)' in green; when the RNA is in the frameshifting state, the FSS is labeled 'FSS(-1F)' in blue. The simulated time is shown to the right of the reporter. A large ribosomal traffic jam is created when the RNA switches into the frameshifting state.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF31': {'text': \"-FSS-MF-AlexX (i.e. the HA multi-frame tag), the spaghetti monster HA (smHA)(Viswanathan et al., 2015)  was flanked with NotI and PstI, and then inserted into the N-terminal region of FSS-MF-AlexX. For the control constructs, FSS was removed using KpnI and XbaI. To keep the same frame for MF and AlexX, the following sequencewas ligated between KpnI and XbaI to obtain MF-AlexX (i.e. the -FSS control tag): 5'-GGT ACC GGG AAT TTT CTT CAG AGC AGA CCA GAG CCA ACA GCC GCA CCG TTT CTA GA -3'. To shift the -1 frame into 0 frame for MF and AlexX, the following sequence was ligated between KpnI and XbaI to obtain MF-AlexX (the -FSS(+1nt) tag): 5'-CGG GAA TTT TCT TCA GAG CAG ACC AGA GCC AAC AGC CGC ACC GTT CT -3'. scFv-sfGFP was amplified from pHR-scFv-GCN4-sfGFP-GB1-dWPRE (Plasmid #60907, addgene) using primers: 5'-GCG CGC ATA TGA TGG GCC CCG ACA TC -3'; 5'-GCC GGA ATT CGC CGC CTT CGG TTA CCG TGA AGG T -3'. The amplified scFv-sfGFP was flanked with NdeI and EcoRI, and then inserted into a pET21 vector backbone for expression and purification from E.coli. For HA multi-frame tag experiments, the HA epitope encoded in the scFv-sfGFP plasmid (Plasmid #60907) from Addgene was removed by site-directed mutagenesis with QuikChange Lightning (Agilent Technologies) per manufacturer's instruction using primers: 5'-CCT CCG CCT CCA CCA GCG TAA TCT GAA CTA GCG GTT CTG CCG CTG CTC ACG GTC ACC AGG GTG CCC -3'; 5'-GGG CAC CCT GGT GAC CGT GAG CAG CGG CAG AAC CGC TAG TTC AGA TTA CGC TGG TGG AGG CGG AGG -3'.All rights reserved. No reuse allowed without permission.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF32': {'text': \"tagged MCP or scFv-sfGFP was purified over a Ni-NTA-agarose (Qiagen) per the manufacturer's instructions with minor modifications. Briefly, bacteria were lysed in a PBS-based buffer with a complete set of protease inhibitors (Roche). Binding to the Ni-NTA resin was done in the presence of 10 mM imidazole. The resin was washed with 20 and 50 mM imidazole in PBS. The protein was then eluted in 300 mM imidazole in PBS. The eluted his-tagged MCP was dialyzed in a HEPES-based buffer (10% glycerol, 25 mM HEPES pH 7.9, 12.5 mM MgCl2, 100 mM KCl, 0.1 mM EDTA, 0.01 % NP-40 detergent, and 1 mM DTT), snap-frozen in liquid nitrogen, and stored at -80C. Cell culture, transfection, and bead-loading U-2 OS cells were grown using DMEM (Thermo Scientific) supplemented with: 10% (v/v) FBS, 1 mM L-glutamine and 1% (v/v) Penicillin-streptomycin. Before experiments, cells were plated on a 35 mm MatTek chamber (MatTek) and DNA was either transiently transfected with Lipofectamine LTX (Thermo Scientific) per the manufacturer's instructions or transiently transfected via bead-loading. As described previously (Hayashi-Takanaka et al., 2011; Morisaki et al., 2016) , bead-loading involved the following six steps: First, 100 µg/ml of fluorescently labeled Fab, 250 µg/ml of purified (GCN4) scFv-All rights reserved. No reuse allowed without permission.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF33': {'text': '100X images with 130 nm/pixel). With this setting, one camera detected farred signals and the other detected either red or green signals. Far red signals were detected with the 637 nm laser and the 731/137 nm emission filter (FF01-731/137/25, Semrock). Red and green signals were separated by the combination of the excitation lasers and the emission filters installed in a filter wheel (HS-625 HSFW TTL, Finger Lakes Instrumentation); namely, the 561 nm laser and 593/46 nm emission filter (FF01-593/46-25, Semrock) were used for Cy3 imaging, and the 488 nm laser and 510/42 nm emission filter (FF01-510/42-25, Semrock) were used for sfGFP or A488 imaging. Live cells were placed into a stage top incubator set to a temperature of 37°C and supplemented with 5% CO2 (Okolab) on a piezoelectric stage (PZU-2150, Applied Scientific Instrumentation). The focus was maintained using the CRISP Autofocus System (CRISP-890, Applied Scientific Instrumentation). The lasers, the cameras, the filter wheel, and the piezoelectric stage were synchronized via an Arduino Mega board (Arduino). Image acquisition was done with open source Micro-Manager software (Edelstein et al., 2014). Imaging size was set to 512 x 512 pixels 2 (66.6 x 66.6 µm 2 ), and exposure time was set to 53.64 msec.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF34': {'text': 'When the ribosome reaches the frameshift site, , it pauses, and may shift from the 0 frame to the -1 frame. For the constitutive model, the ribosome can continue in the 0 frame with rate',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'TABREF1': {'text': 'Bursty Frameshifting Model Parameters',\n",
       "   'latex': None,\n",
       "   'type': 'table'},\n",
       "  'TABREF4': {'text': \"table 1 for the bursting model. Simulation details. To simulated the model's stochastic dynamics, we used the direct method from Gillespie's algorithm (Gillespie 1976) coded in Matlab. Genetic Algorithms were coded in Matlab. Simulations were performed on the W. M. Keck High Performance Computing Cluster at Colorado State University.\",\n",
       "   'latex': None,\n",
       "   'type': 'table'}},\n",
       " 'back_matter': [{'text': \" Figure 1 . A multi-frame tag to image single RNA frameshifting dynamics in living cells (A) The multi-frame (MF) tag contains 12 repeated FLAG epitopes in the 0 frame interspaced between 12 repeated SunTag v4 epitopes in the -1 frame. Depending on which frame is translated, nascent epitopes are labeled by fluorescent anti-FLAG antibody fragments (Cy3-Fab, green 'Y') or anti-SunTag single chain variable fragments (scFv-GFP, blue 'Y'). Following the repeat epitopes is exon 1 of the GNAS locus, in which the peripheral membrane proteins AlexX (689 aa) and XXLb1 (690 aa) were placed in the 0 and -1 frames, respectively. Preceding the multi-frame tag is a multiframe element (MFE) . In this study, the HIV-1 frameshift sequence was used as the MFE. To facilitate single-RNA tracking, a 24X MS2 stem-loop tag was also placed in the 3' UTR. This tag is labeled by MCP-HaloTag (with JF646-HaloTag ligand, red). (B) A representative cell~10 hours after transient transfection with the multi-frame tag depicted in A. The red circle (labeled '1') highlights a non-translating RNA, the yellow circle (labeled '2') highlights a 0-frame translation site . The frameshift sequence stimulates frameshifting in multi-RNA translation factories (A) Cells were co-transfected with increasing concentrations of short oligo RNA encoding just the frameshift sequence, together with either the multi-frame reporter (+FSS) or control reporter (-FSS). (B) The percentage of translation sites translating just the 0 frame (green), the 0 and -1 frame (cyan), or just the -1 frame (dark blue) at all oligo concentrations for the +FSS and -FSS reporters. (C) The ratio of the intensity of RNA signals within frameshifting sites to non-frameshifting RNA at all oligo concentrations for the +FSS and -FSS reporters. In all bar graphs, the total number of each type of RNA species is shown above each bar and the error represents S.E.M. between cells. harringtonine-induced ribosomal run-off from the +FSS 2x multi-frame RNA reporter with FLAG (green) and SunTag epitopes (blue) in the 0 and -1 frames. The intensity of nascent chain signals within non-frameshifting translation sites (marked by just anti-FLAG Fab, green triangles, 217 translation sites initially in 19 cells) and frameshifting sites (marked by anti-SunTag scFv, cyan circles, 32 translation sites initially) drops with time as ribosomes complete translation, i.e. run-off. (B) Same as A, but with a modified multi-frame reporter with extra HA epitopes (orange, HA multi-frame tag) upstream of the frameshift sequence. The intensity of nascent chain signals from non-frameshifting translation sites (marked by just anti-HA Fab, orange triangles, 175 translation sites initially in 27 cells) and frameshifting translation sites (marked by anti-SunTag scFv, orange circles, 42 translation sites initially) drops with time as ribosomes run-off. (C) A sample single translation site encoding the modified HA multi-frame reporter (shown in B) after addition of harringtonine. A montage of image trims shows the detected RNA-, HA-, and Sun-signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting ribosomes) is plotted through time. Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions.\",\n",
       "   'cite_spans': [{'start': 678, 'end': 683, 'text': '(MFE)', 'ref_id': None}],\n",
       "   'ref_spans': [{'start': 1, 'end': 9, 'text': 'Figure 1', 'ref_id': None}],\n",
       "   'section': 'annex'},\n",
       "  {'text': '(A) A schematic of the model: k ini is the translation initiation rate, k e is the translation elongation rate, k on is the rate at which RNA switch to the frameshifting state, k off is the rate at which RNA switch to the non-frameshifting state, k FSS is the pause rate at the FSS in the non-frameshifting state, k* FSS is the pause rate at the FSS in the frameshifting state, and k t is the termination rate (assumed equal to k el ). Occasionally, they switch to a frameshifting state (~8%) that can last from tens of minutes to an hour or so. Transitions to this state are stimulated through interactions with other FSS-encoding RNA, either by chance or within multi-RNA translation factories. When in the frameshifting state, longer pauses at the FSS induce longer ribosomal traffic jams. These allow frameshifting RNA to continue to be translated for up to an hour longer than non-frameshifting RNA upon global translation initiation shut-down.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure 5. A model for bursty frameshifting'},\n",
       "  {'text': 'Mature AlexX (upper) and XXLb1 (lower) membrane proteins accumulate in distinct punctae that coalesce to form elongated structures that undulate dynamically. These structures are enriched at the top and bottom of cells and around the edge of the cytoplasm (arrows), consistent with membrane localization. Movie S2 shows the dynamics of these structures. In the +FSS multi-frame reporter, mature AlexX comes from 0 frame canonical translation, while mature XXLb1 comes from -1 frameshifting translation. In the -FSS control reporter, only mature AlexX accumulates, indicating little to no frameshifting. The -FSS (+1 nt) column shows accumulation of XXLb1 when an extra nucleotide is inserted in the -FSS control reporter following the start codon. This extra nucleotide pushes XXLb1 into the 0 frame, demonstrating it can be expressed. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Fig. S1. Accumulation of mature AlexX and XXLb1 proteins from multi-frame tags'},\n",
       "  {'text': 'The percentage of RNA translating the -1 frame only (dark blue), the 0 and -1 frames (cyan), and the 0 frame only (green) for the +FSS multi-frame tag and the +FSS reverse tag (in which FLAG and SunTag epitopes are reversed). Pie charts show the percentage of translating species per cell. Error bars represent S.E.M. among cells.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S3. Epitope order in tags has little impact on percentage of frameshifting RNA'},\n",
       "  {'text': '(A) Example cell where single mature membrane proteins were detected (green circles), along with translation sites (co-localized RNA and protein; purple circles). (B) Average images of detected spots (33 mature proteins and 35 translation sites) were fit to a Gaussian to determine their intensity ratio, shown in the bar graph on the right. This ratio provides a calibration factor to convert translation site intensites to the number of riboeoms. The error bar represents the propagated 90% confidence interval from the fit. Error bars represent the fitted 90% confidence interval. Aside from the -FSS control, frameshifting sites (-1 TS) have significantly brighter RNA signals compared to canonical translation sites (0 TS). Note the frameshifting observed in the -FSS control represents less than 1% of translating RNA and can be considered background frameshifting. Figure S6 . Upon oligo RNA co-transfection, stimulated frameshifting still occurs at bright RNA sites, even though the fraction of bright RNA sites remains unchanged (A) Average RNA signals from non-translating RNA (no TS), 0-frame only translation sites (0 TS), and frameshifting translation sites (-1 TS) when different concentrations of oligo RNA (0, 1, and 4 mg) encoding the frameshift sequence were co-transfected into cells with the +FSS multi-frame reporter or the control -FSS reporter. The bar graph above shows the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). (B) The distributions of RNA signals from the four experiments normalized to their mean value. In all +FSS experiments, the fraction of bright RNA (defined here as having an intensity greater than or equal to 1.6 times the mean) remained constant. The -FSS experiment had fewer bright RNA, suggesting the FSS sequence facilitates incorporation into multi-RNA sites.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 872,\n",
       "     'end': 881,\n",
       "     'text': 'Figure S6',\n",
       "     'ref_id': None}],\n",
       "   'section': 'Figure S4. Calibrating translation site intensity to the number of ribosomes'},\n",
       "  {'text': 'The total intensity of detected non-frameshifting and frameshifting translation sites decays with time after addition (at t = 0 sec) of the translational initiation inhibitor harringtonine. Data is taken from Fig. 4A . If L2 is the length of the tagged portion of the open reading frame and L1 is the length of the non-tagged portion, then the linear (post-tag) portion of the decay begins from approximately L1/(L1+L2/2). This portion of the curve provides an estimate of the elongation rate during the part of the run-off where no new epitopes are being translated. Therefore, the run-off fit is independent of the frameshifting kinetics that occur upstream of the epitopes. Curves are normalized to their initial values. Figure 4B . The Sun signal comes from frameshifted ribosomes that have run past the frameshift sequence (FSS). Error bars represent S.E.M. Figure S10 . Ribosome run-off from FLAG epitopes in HA multi-frame tag For completeness, an additional experiment was performed to generate the ribosome run-off curve from FLAG epitopes in the HA multi-frame tag. Because we can only image two epitopes at the same time (since RNA is imaged in a third color), we examined FLAG and Sun epitopes (Sun epitopes were required to distinguish frameshifting and non-frameshifting sites). The FLAG run-off from this experiment is on the right (green triangles). This curve is compared to the run-off curve of HA in HA only sites (orange squares) from Figure 4B . Both experiments represent nonframeshifting sites (no Sun signal detected). Error bars represent S.E.M.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 209, 'end': 216, 'text': 'Fig. 4A', 'ref_id': None},\n",
       "    {'start': 724, 'end': 733, 'text': 'Figure 4B', 'ref_id': None},\n",
       "    {'start': 863, 'end': 873, 'text': 'Figure S10', 'ref_id': None},\n",
       "    {'start': 1455, 'end': 1464, 'text': 'Figure 4B', 'ref_id': None}],\n",
       "   'section': 'Figure S7. Fits to linear portion of ribosomal run-offs provide elongation rate estimates'},\n",
       "  {'text': 'A single frameshifting translation site encoding the HA multi-frame tag was tracked after harringtonine addition. On the top, a montage of image trims shows the detected RNA-, HA-, and Sun shown signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting ribosomes) is plotted through time. At about the fourth timepoint, another non-translating RNA interacts with the frameshifting RNA, leading to a burst of frameshifting signal (Sun). Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'Model dynamics for the bursting (A,C,E) and constitutive model (B,D,F). Simulations were performed using the best parameter values obtained from the optimization process. (A,B) Simulated time courses representing single molecule fluctuation dynamics from 100 translation sites (a sample trace is shown in bold). Green and blue lines represent the translation of FLAG (0 frame) and Sun (-1 frame) epitopes in the +FSS multi-frame tag, respectively. (C,D) On the left, bar graphs showing the experimental (black) and simulated (gray) percentages of 0 and -1 frame translation. On the right, the number of ribosomes translating either the 0 or -1 frame. (E,F) On the left, simulated run-off (solid lines) from the frameshift sequence (FSS) of all ribosomes in nonframeshifting (black) and frameshifting sites (gray), plotted with data in Fig. 4A (black triangles and gray circles). On the right, simulated run-off (solid lines) from the start site (AUG) of all ribosomes in non-frameshifting (black) and frameshifting sites (gray), plotted with data in Fig. 4B (black triangles and gray circles). Error bars represent the standard error of the mean (S.E.M.). Details can be found in the Supplementary Methods. Figure S13 . Single frameshifting RNA track persisting for >35 minutes A sample frameshifting translation site encoding the +FSS 2x multi-frame tag was tracked for >35 minutes. Left shows the cell at the 1200 second time point. The zoom shows the channels separated to highlight two frameshifted RNA that split from a single translation site. On the right, the position of the spot is plotted through time.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 835, 'end': 842, 'text': 'Fig. 4A', 'ref_id': None},\n",
       "    {'start': 1050, 'end': 1057, 'text': 'Fig. 4B', 'ref_id': None},\n",
       "    {'start': 1207, 'end': 1217, 'text': 'Figure S13', 'ref_id': None}],\n",
       "   'section': 'Figure S12. Comparison between the bursting and constitutive models'},\n",
       "  {'text': 'The codon adaptation index (CAI) for the 0 frame (top) and -1 frame (bottom) of the +FSS multiframe tag. The y-axis shows the CAI calculated using the codon frequency in the human genome. The x-axis shows the length of the genes, in codons. In the plots, rare codons have low CAIs and common codons have high CAIs. The vertical red line represents the location of the frameshift sequence (FSS). Similar CAIs are obtained for the sequences in the 0 frame (CAI 0F = 0.76) and the -1 frame (CAI -1F = 0.74).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S14. Codon usage plays a minor role in traffic jam formation'},\n",
       "  {'text': 'Intensity distribution of non-frameshifted HA signals (black) versus frameshifted HA signals (gray) produced from the HA multi-frame tag.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'}]}"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "string indices must be integers",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-101-edb163a9ff16>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mextract_paper_metadata\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper_info\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-100-999090a7b134>\u001b[0m in \u001b[0;36mextract_paper_metadata\u001b[1;34m(paper)\u001b[0m\n\u001b[0;32m     48\u001b[0m     \u001b[0mpaper_metadata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'title'\u001b[0m\u001b[1;33m]\u001b[0m    \u001b[1;33m=\u001b[0m \u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'title'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     49\u001b[0m     \u001b[0mpaper_metadata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'authors'\u001b[0m\u001b[1;33m]\u001b[0m  \u001b[1;33m=\u001b[0m \u001b[0mextract_authors\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'authors'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 50\u001b[1;33m     \u001b[0mpaper_metadata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'abstract'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mextract_abstract\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'abstract'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     51\u001b[0m     \u001b[0mpaper_metadata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'refs'\u001b[0m\u001b[1;33m]\u001b[0m     \u001b[1;33m=\u001b[0m \u001b[0mextract_references\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'bib_entries'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     52\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mpaper_metadata\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-100-999090a7b134>\u001b[0m in \u001b[0;36mextract_abstract\u001b[1;34m(abstract)\u001b[0m\n\u001b[0;32m     26\u001b[0m     \u001b[0mabstract_text\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     27\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0msection\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mabstract\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 28\u001b[1;33m         \u001b[0mabstract_text\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mabstract_text\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m' '\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0msection\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'text'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     29\u001b[0m     \u001b[0mabstract_text\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mabstract_text\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\" \"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     30\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0ms\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mstop_sentences\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m: string indices must be integers"
     ]
    }
   ],
   "source": [
    "extract_paper_metadata(paper_info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28769</th>\n",
       "      <td>03ea3a614b56409d3f099c9ad764864293132540</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Live-cell single RNA imaging reveals bursts of...</td>\n",
       "      <td>doi.org/10.1101/478040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.biorxiv.org/about-biorxiv</td>\n",
       "      <td>Ribosomal frameshifting during the translation...</td>\n",
       "      <td>2018-11-24</td>\n",
       "      <td>Lyon, K. R.; Aguilera, L. U.; Morisaki, T.; Mu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sha source_x  \\\n",
       "28769  03ea3a614b56409d3f099c9ad764864293132540  biorxiv   \n",
       "\n",
       "                                                   title  \\\n",
       "28769  Live-cell single RNA imaging reveals bursts of...   \n",
       "\n",
       "                          doi pmcid  pubmed_id  \\\n",
       "28769  doi.org/10.1101/478040   NaN        NaN   \n",
       "\n",
       "                                         license  \\\n",
       "28769  See https://www.biorxiv.org/about-biorxiv   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "28769  Ribosomal frameshifting during the translation...   2018-11-24   \n",
       "\n",
       "                                                 authors journal  \\\n",
       "28769  Lyon, K. R.; Aguilera, L. U.; Morisaki, T.; Mu...     NaN   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "28769                          NaN            NaN          True  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['sha']==paper_info['id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# pmc_custom_license"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['002f09dfc9a1323a15bf72e349d8b733ac97a2aa.json',\n",
       " '0036e8891c93ae63611bde179ada1e03e8577dea.json',\n",
       " '00573277e6be50669016f770bc28ec2da0639a8f.json',\n",
       " '00683d59d56123ae85e080d00ef1b3edd3f7405d.json',\n",
       " '0104f6ceccf92ae8567a0102f89cbb976969a774.json']"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmc = '2020-03-13\\pmc_custom_license\\pmc_custom_license'\n",
    "path = cfg['data-path'] + pmc\n",
    "pmc_papers = get_papers(path)\n",
    "pmc_papers[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1426/1426 papers present in the metadataset.\n"
     ]
    }
   ],
   "source": [
    "cnt = 0\n",
    "# check if paper are in metadata dataframe\n",
    "for paper in pmc_papers:\n",
    "    if paper[:-5] not in paper_ids:\n",
    "        print(paper)\n",
    "    else:\n",
    "        cnt += 1\n",
    "print('There are {}/{} papers present in the metadataset.'.format(cnt, len(pmc_papers)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "paper_info = get_paper_data(path, pmc_papers[10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': '022c31717b2ad8e7c25aab342d27c203d7e32527',\n",
       " 'title': 'Virulence Potential of Fusogenic Orthoreoviruses',\n",
       " 'authors': [{'first': 'Ann', 'middle': ['H'], 'last': 'Wong'},\n",
       "  {'first': 'Peter', 'middle': ['K C'], 'last': 'Cheng'},\n",
       "  {'first': 'Mary', 'middle': ['Y Y'], 'last': 'Lai'},\n",
       "  {'first': 'Peter', 'middle': ['C K'], 'last': 'Leung'},\n",
       "  {'first': 'Kitty', 'middle': ['K Y'], 'last': 'Wong'},\n",
       "  {'first': 'W', 'middle': ['Y'], 'last': 'Lee'},\n",
       "  {'first': 'Wilina', 'middle': ['W L'], 'last': 'Lim'},\n",
       "  {'first': 'Nam', 'middle': ['Cheong'], 'last': 'St'},\n",
       "  {'first': 'Kip', 'middle': [], 'last': 'Shek'},\n",
       "  {'first': '', 'middle': [], 'last': 'Mei'}],\n",
       " 'abstract': '',\n",
       " 'refs': [{'id': 'b0',\n",
       "   'title': 'The diversity of the orthoreoviruses: molecular taxonomy and phylogenetic divides',\n",
       "   'authors': [{'first': 'J', 'middle': ['M'], 'last': 'Day', 'suffix': ''}],\n",
       "   'year': 2009},\n",
       "  {'id': 'b1',\n",
       "   'title': 'Orthoreoviruses and their replication. Fields virology',\n",
       "   'authors': [],\n",
       "   'year': 2007},\n",
       "  {'id': 'b2',\n",
       "   'title': 'A previously unknown reovirus of bat origin is associated with an acute respiratory disease in humans',\n",
       "   'authors': [{'first': 'K', 'middle': ['B'], 'last': 'Chua', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Crameri', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Hyatt', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Yu', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['R'], 'last': 'Tompang', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Roseli', 'suffix': ''}],\n",
       "   'year': 2007},\n",
       "  {'id': 'b3',\n",
       "   'title': 'Identifi cation and characterization of a new orthoreovirus from a patient with acute respiratory infections',\n",
       "   'authors': [{'first': 'K', 'middle': ['B'], 'last': 'Chua', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Voon', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Crameri', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['S'], 'last': 'Tan', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Roseli', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['A'], 'last': 'Mceachern', 'suffix': ''}],\n",
       "   'year': 2008},\n",
       "  {'id': 'b4',\n",
       "   'title': 'Xi River virus: a new bat reovirus isolated in southern China',\n",
       "   'authors': [{'first': 'L', 'middle': [], 'last': 'Du', 'suffix': ''},\n",
       "    {'first': 'Z', 'middle': [], 'last': 'Lu', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Fan', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Meng', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Jiang', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Zhu', 'suffix': ''}],\n",
       "   'year': 2010},\n",
       "  {'id': 'b5',\n",
       "   'title': 'Nipah virus: a recently emergent deadly paramyxovirus',\n",
       "   'authors': [{'first': 'K', 'middle': ['B'], 'last': 'Chua', 'suffix': ''},\n",
       "    {'first': 'W', 'middle': ['J'], 'last': 'Bellini', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['A'], 'last': 'Rota', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['H'], 'last': 'Harcourt', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Tamin', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['K'], 'last': 'Lam', 'suffix': ''}],\n",
       "   'year': 2000},\n",
       "  {'id': 'b6',\n",
       "   'title': 'Breaking the chain in Bangladesh',\n",
       "   'authors': [{'first': 'R', 'middle': [], 'last': 'Stone', 'suffix': ''}],\n",
       "   'year': 2011},\n",
       "  {'id': 'b7',\n",
       "   'title': 'Isolation of Hendra virus from pteropid bats: a natural reservoir of Hendra virus',\n",
       "   'authors': [{'first': 'K', 'middle': [], 'last': 'Halpin', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['L'], 'last': 'Young', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['E'], 'last': 'Field', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['S'], 'last': 'Mackenzie', 'suffix': ''}],\n",
       "   'year': 2000},\n",
       "  {'id': 'b8',\n",
       "   'title': 'Ecoepidemiology and complete genome comparison of different strains of severe acute respiratory syndrome-related Rhinolophus bat coronavirus in China reveal bats as a reservoir for acute, self-limiting infection that allows recombination events',\n",
       "   'authors': [{'first': 'S', 'middle': ['K'], 'last': 'Lau', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': ['S'], 'last': 'Li', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Huang', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['T'], 'last': 'Shek', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': [], 'last': 'Tse', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Wang', 'suffix': ''}],\n",
       "   'year': 2010},\n",
       "  {'id': 'b9',\n",
       "   'title': 'Bats as a continuing source of emerging infections in humans',\n",
       "   'authors': [{'first': 'S', 'middle': [], 'last': 'Wong', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Lau', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Woo', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': ['Y'], 'last': 'Yuen', 'suffix': ''}],\n",
       "   'year': 2007},\n",
       "  {'id': 'b10',\n",
       "   'title': 'A novel reovirus isolated from a patient with acute respiratory disease',\n",
       "   'authors': [{'first': 'P', 'middle': [], 'last': 'Cheng', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['C'], 'last': 'Lau', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Lai', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': [], 'last': 'Ho', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Leung', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Chan', 'suffix': ''}],\n",
       "   'year': 2009},\n",
       "  {'id': 'b11',\n",
       "   'title': 'Diagnostic procedures for viral, rickettsial and chlamydial infection',\n",
       "   'authors': [{'first': 'E',\n",
       "     'middle': ['H'],\n",
       "     'last': 'Lennette',\n",
       "     'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['A'], 'last': 'Lennette', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': ['T'], 'last': 'Lennette', 'suffix': ''}],\n",
       "   'year': 1995},\n",
       "  {'id': 'b12',\n",
       "   'title': 'Crossing the species barrier-one small step to man, one giant leap to mankind',\n",
       "   'authors': [{'first': 'M',\n",
       "     'middle': ['S'],\n",
       "     'last': 'Klempner',\n",
       "     'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['S'], 'last': 'Shapiro', 'suffix': ''}],\n",
       "   'year': 2004},\n",
       "  {'id': 'b13',\n",
       "   'title': 'Crossing the species barrier: the threat of an avian infl uenza pandemic',\n",
       "   'authors': [{'first': 'S', 'middle': [], 'last': 'Riedel', 'suffix': ''}],\n",
       "   'year': 2006},\n",
       "  {'id': 'b14',\n",
       "   'title': 'Severe acute respiratory syndrome',\n",
       "   'authors': [{'first': 'Jsm', 'middle': [], 'last': 'Peiris', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Guan', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': ['Y'], 'last': 'Yuen', 'suffix': ''}],\n",
       "   'year': 2004}]}"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17504</th>\n",
       "      <td>0036e8891c93ae63611bde179ada1e03e8577dea</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Stable Occupancy of the Crimean-Congo Hemorrha...</td>\n",
       "      <td>http://dx.doi.org/10.1128/mBio.01065-19</td>\n",
       "      <td>PMC6650548</td>\n",
       "      <td>31337717.0</td>\n",
       "      <td>NO-CC CODE</td>\n",
       "      <td>Crimean-Congo hemorrhagic fever virus (CCHFV) ...</td>\n",
       "      <td>2019 Jul 23</td>\n",
       "      <td>['Scholte, Florine E. M.', 'Hua, Brian L.', 'S...</td>\n",
       "      <td>mBio</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sha source_x  \\\n",
       "17504  0036e8891c93ae63611bde179ada1e03e8577dea      PMC   \n",
       "\n",
       "                                                   title  \\\n",
       "17504  Stable Occupancy of the Crimean-Congo Hemorrha...   \n",
       "\n",
       "                                           doi       pmcid   pubmed_id  \\\n",
       "17504  http://dx.doi.org/10.1128/mBio.01065-19  PMC6650548  31337717.0   \n",
       "\n",
       "          license                                           abstract  \\\n",
       "17504  NO-CC CODE  Crimean-Congo hemorrhagic fever virus (CCHFV) ...   \n",
       "\n",
       "      publish_time                                            authors journal  \\\n",
       "17504  2019 Jul 23  ['Scholte, Florine E. M.', 'Hua, Brian L.', 'S...    mBio   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "17504                          NaN            NaN          True  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['sha']=='0036e8891c93ae63611bde179ada1e03e8577dea']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# comm_use_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# noncomm_use_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "code_folding": [
     0,
     13,
     26,
     37
    ]
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# extract data from all papers\n",
    "all_papers_data = []\n",
    "for paper_name in papers:\n",
    "    file_path = os.path.join(path,paper_name)\n",
    "    with open(file_path, 'r') as f:\n",
    "        paper_info = extract_paper_metadata(json.load(f))\n",
    "    all_papers_data.append(paper_info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3\n",
      "- Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China\n",
      "- Real-time, MinION-based, amplicon sequencing for lineage typing of infectious bronchitis virus from upper respiratory samples\n",
      "- Assessing spread risk of Wuhan novel coronavirus within and beyond China, January-April 2020: a travel network-based modelling study\n",
      "- TWIRLS, an automated topic-wise inference method based on massive literature, suggests a possible mechanism via ACE2 for the pathological changes in the human host after coronavirus infection\n",
      "- Title: Viruses are a dominant driver of protein adaptation in mammals\n",
      "- The impact of regular school closure on seasonal influenza epidemics: a data-driven spatial transmission model for Belgium\n",
      "- Carbon Nanocarriers Deliver siRNA to Intact Plant Cells for Efficient Gene\n",
      "- Protective Population Behavior Change in Outbreaks of Emerging Infectious Disease 1 2\n",
      "- A hidden gene in astroviruses encodes a cell-permeabilizing protein involved in virus release\n"
     ]
    }
   ],
   "source": [
    "for i in range(10):\n",
    "    print('- {}'.format(all_papers_data[i]['title']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0015023cc06b5362d332b3baf348d11567ca2fbb\n"
     ]
    }
   ],
   "source": [
    "# get json data of current paper\n",
    "file_path = os.path.join(path,papers[0])\n",
    "with open(file_path, 'r') as f:\n",
    "    paper = extract_paper_metadata(json.load(f))\n",
    "print(paper['id'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Authors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def are_equal(author1, author2):\n",
    "    if (author1['first'][0] == author2['first'][0]) and (author1['mid'] == author2['mid']) and (author1['last'] == author2['last']):\n",
    "        return True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Author:\n",
    "    def __init__(self, firstname, middlename, lastname):\n",
    "        self.firstName = firstname\n",
    "        self.middleName = middlename\n",
    "        self.lastName = lastname\n",
    "        self.papers = []\n",
    "    \n",
    "    def __str__(self):\n",
    "        return '{} {} {}'.format(self.firstName, self.middleName, self.lastName)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "authors = []"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Co-Authors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from itertools import combinations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "co_authors_net = nx.Graph()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for each paper\n",
    "for i in range(len(all_papers_data)):\n",
    "    # get list of authors\n",
    "    co_authors = []\n",
    "    for author in all_papers_data[i]['authors']:\n",
    "        author_full_name = ''\n",
    "        \n",
    "        # only keep authors with first and last names\n",
    "        if author['first'] and author['last']:\n",
    "            author_full_name += author['first']\n",
    "            for initial in author['middle']:\n",
    "                author_full_name += ' ' + initial\n",
    "            author_full_name += ' ' + author['last']\n",
    "            author_full_name.strip(' ')\n",
    "            co_authors.append(author_full_name)\n",
    "    #print(co_authors)\n",
    "    for combo in combinations(co_authors,2):\n",
    "        co_authors_net.add_edge(combo[0],combo[1])\n",
    "    #print('-'*60)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1, 2)\n",
      "(1, 3)\n",
      "(2, 3)\n"
     ]
    }
   ],
   "source": [
    "for i in combinations([1,2,3],2):\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\david\\Anaconda3\\lib\\site-packages\\networkx\\drawing\\nx_pylab.py:611: MatplotlibDeprecationWarning: isinstance(..., numbers.Number)\n",
      "  if cb.is_numlike(alpha):\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeQAAAE/CAYAAACXV7AVAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJztvX9wHNd15/vtmcFgAII/BBISaPARcsGmwTwq5RVUzhpMZcuyC1ZAeyhZA9my7KeXfbHot+BW2bFLtmaqnLUSUGtvnpJ4xUq0cVW2UvGTI5KipmOrapGsdpMNGTsrR6lIeVI5lle2ybIkipQo8ReAwdz3x+Fld9++3dPd0z3dM3M+VVPAzPR03+npvuee34YQQoBhGIZhmFTJpT0AhmEYhmFYIDMMwzBMJmCBzDAMwzAZgAUywzAMw2QAFsgMwzAMkwFYIDMMwzBMBmCBzDAMwzAZgAUywzAMw2QAFsgMwzAMkwFYIDMMwzBMBmCBzDAMwzAZgAUywzAMw2QAFsgMwzAMkwFYIDMMwzBMBmCBzDAMwzAZgAUywzAMw2QAFsgMwzAMkwFYIDMMwzBMBmCBzDBdgGkCBw/SX4ZhehNDCCHSHgTDMN6YJrCwILC6amBwEHj8caBcTntU/YNpAsvLwNwcn3cmWVhDZpiM8+ijwOqqAQBYWaHnTGeo1QQ+9rEmDh8G7r6bLRRMsrBAZhiG0WCawL//9wLr6zRNXrpEmjLDJAULZIbJOAcOAMUi/V8s0vM4Yf+0nuVloNm0pshCgczWDJMU7ENmGB+y4j9MahzV6joeesgAkEOhABw7xn5SiWmSmfrSJSCfB770JWBpKe1RMb0MC2SG8aAbJuR2BLVpArffLiCEce21mRngmWdiHmQXE/b8mqbl4z9wgBc3TDhYIDOMBwcPAocPW88LBeB97wNefBG49VbgyJH0xgYACwvA0aPW2MJqt+r3A4DJSeDll2MbYl9hmkCl0sTaGpm5OSKeCQv7kBnGg7k50owljQZw8iRw7hwJwr170xtbrQYcPWqtpRsN4MEHw+1jbo4EuZ177olhcH3K8jKuCWOAIuI5CIwJAwtkhvGgXCYztSq0JCdPOgOhOhkcRccwHK/99Kfh9lEuk1Y9M0OacaUCnD8ffPwcDOZkbs4KvgNIQ+YgMCYMbLJmmBZIP+Kzz5IQtjM/D3z3u7TNnXeSptqJ4KhaDTh0yPlaO/5f0wQ+8QmBy5cNDA8Djz3mP/5ajVKCms32i5VkJXAuDtiHzLSFYBgmMOPjQgDWY36eXp+Zcb4+M6P/fL0uxOIi/W2XSsU6XqHQ3j4XF53jX1z03rZeFyKXa2rPQ1jqdSHy+fVI36FaFWLPHvrLML0Am6wZJgSPPkqmSID+ypzg1193bqczH1PUtoit6tORI0C9Diwutq+Rb94MAEJ5rofycw3vDULw4IO4VnjDzw+umsfJQiDw/PNkKajVYhkOw6QKC2Sm72jH91kuk3l2cdFpplWDod54w73/5WXg0iUSZHFVfSqXgUcead80ev48YPdJ03M9qq+0UIherERduHgtZO66iyLC77qLnqs+dPZjM70AC2Smr5C+0sOHgY99TATWrOxCXCcEl5bIhytpNNwCV9VCX3018te4Rq0G3HRTaw2x1XZzc8DwMP0/PExj9Vq0lMuknc/P06Md7XznTv/nAPDv/t06Vlbof1nLWz0e+2qZniBtmznDdBLVV2oYrX2Q9boQxaL1mdlZ7+2Gh2mb4WG3P1Q9NtCe/5N8yM2W+6pWg20n/dv2/eZy8fi7vajXyXfs5UOu14UAGlp/NfuQmV6DBTLTV1AQkVMo5vP+Qmd+3i1IKxXv/duDtuzPoxzb73vYhSwgxNiYftudO53H3Lkz3H6npsKPLwx+gW7quTcMvdC2f54FNdOtsEBm+o5qlSb2oFHFOoE8Otr6OPW6EEND6w6NWY3GbnVsL3Ta9tCQftugEeBe+y2Vwo8vLtRzr46dznHz2jkOajVgmCzCPmSm71haAh54wKrCNTzsX8DhwAEgp9wpt97a+jjLy8Dly1brvkcfBb7yFWehkSDFI/buBUolZ2Uw3WdGRvSftx+zUKDnXszNAYYSQD0x4T++uLH7uw8ccEa1q2Onc2wFytXrgD3Y64/+qDNjZphYSHtFwDBpESYnuF4n0+3IiLe5WveZgYH1a9pasWiZrufn6dHq2LOzTo3P7r+25yG30gbDfFfyOYuO+JD1x3ZquH5jv+66xrXtBwfd5vZt2zo3doZpF67UxWSeOCs5dboq1L59Ak89ZWlsi4sUoR2UUgnXIowB0hKvXLGe12pW5HecnaiCnifTpNzh11+n1K92xzA8DFy+bD0fGiLNV8fGjcCFCwJSIx4cdJ4rAKhWs9ehi2E8SXtFwDB+2COX8/lmWz5BtcJUkv7FSoX8zLOz/pHXrfDTkNPGHiEd1znN5Zz7y+X026matHzIc20YwS0ZDJMV2IfMZBoqpkH/r68beOih6FWZfuM3nBWmDh1KpqAEtUUUOHeOal+/972kGX/uc/R9dMc0TWD7dmBgwOkr/tKXAJm7nMvJ59lgeZnyre2E0f517NoF2HO16bkbXXONiQmqwb24CDz5ZPrtMRkmLCyQmUyjtkAUAvja16IJUl0hDtkIQKVWA8bGgOuvD78A+M53ALuw+J//k77H7/wOVZtaWHB3ibr9duCVV6wWj1IoU3ERuk2bzWy189MFlq2utrfPD34QsJ87eu5GNaOPjACnTvlXLgtaRIVh0oIFMpNpZAtEe+Tv+no0wXTbbcG2k3WSX38dOHMmfK1ke1lJAFhbIz+rFFarq86FwPIyLTTsfP/79FetoNXpdn61GnDjjcAtt7gXQeUyMDvrfO0jH2nveEG/79IS+Yf37KG/b7/tv1/ZHUvWvl5YaG+cDJMIadvMGSYI1apVVCOKL1ZCPlln1LOKWkjDr+iG11h1n/cqtkHFOLxzitvtEDU7SxHIYf3Pqp9WV5RDCMtfHpfPNs6OWJLJSfc57mT0OMMEgQUy0zXENVG32o+ueEcYgSyEMyVpeNi9TzWdKEwKUxjaCQrTCTG/oiIqSQjWqMRVkIVhkoTTnhhGQfp07XdGlPQZe+oQANxxB/mBJWoKVBIpTLmc83sYhnMMftxyC/CDHzhfm5wEXn659WdNE7jzToFGw0Ch0H57yHYxTef5LxYp6IubUjBZggUykyk6nSfsN44o+bU7dgCnT1PE76lTzvdqNeDrX6fAreFhighO+jsWCuRztzM7C5w40fqz7SxMVGE+MwM880ywMSeFaVq++wMHWBgz2YMFcgbIihBKG9ME7r6b0pzyeQrm6qaiDiSMrUIVo6PA2bPObTr9W1MKlnxmjS2MUI6yMNm8GXjrLev5pk3+PZYZNzwv9CFp2suZ1i37+okorRGzhOqjzErgkOqfBqigR5Js2+Y8nlrCslqlwLVcLlvFTuKiXf85zwv9Cac9pYy98MWlS0CtJhIpVtEN6HKOv/71ZIp3JIG9aYQkC3nDR44AW7Y4X/NqRBEX993n/VymlV2+TD5de951L1Crkb9al3MeBNMEFhfXHfNCFq4jJnlYIKeMPe8SEHj+eQN3352MEDJNYN8+emRRyFk5x5YXpdEINxktLFCN43e9q/Pf8f77nc8Lhc7nDXuh5gurz+NGzRO2m7p1VbbU4LG0qdXI7D40FC5n2TSBhx4S14LH1JzzIJ+/6y7g1ClrZRqkIxjTI6StojNkjtqzJ9mUjHpdiGKx6TBZZsUMpjaUr1atGsnDw0JMTwfLo1V74QL+ubHVKuUcz8zoz0WURvfVKqULee0zLej398+/7hS6PO24zdZRunNJwl5HdnS9s+fngx9b1486zOeZ7oYFckZI2meU1RvdWXyieW3ikz646Wnn5Dg66r2v0VG9H1cnUNWiF+oCRRUaXkI5jNCOIuBbUa8LsWEDjXFiovW2WckLTtKHXK9T/EEUgar7bKvrzo4qkL2Kqfgdn9pIZmPxxHQWFsgZIskJkzodZU8gq5YBdQJTuwn5Ta664CWAjtHquKpVQq2sZa+cJVGFtp9gCSrgw0AVvtwdj/q9y5FOSy2Vgn1Wt3ANK9ClJSKXi/Y7h+mXHWV7JruwQO4jqlVLKGdl5a0zX9oF4/h4OG2lUhFiYCCohuw96aoCWbefINtINm1ybrdpk/95CbI48xIe/S6UdVW5Wp1vid1SBdC1FPZcdtISQS0wrUXZ4GA8x82SNaWfYIHcZ3TqRgtjnq1ULDOhaq7X1XlWJ0hdreYgx1cFql2T1glsVdPWCWQ1vUeiLhLyee9xqd/Z67fSnZuwGmEvotOQw2iq3SKM6nUhtm5t+C5oo+53aKipvR+ZZGGBzMSO6p8NomH4TYJ+ATpqreZWflT3OL0nbXsjCt37Xk0kdExNObfz8y2qroVcrtV3cJutg2qEvYjdbJy0tSAN4V2vkxXAMNy/ez7f3ljqdSFuvPF8rAKeCQ4LZCZ2dP7ZsJHKQbVrnY85zmO1er+V0JbotFmviU6n8bb6Du2cg14krKCM0hHLHrPQKReQutiI83enfa9d3RdryGnAApmJHZ2ACJpmpaactNJudD7myclYvkZggi4g1HQur/NRKoU3P0utaXKys8JYavN+WnzWmZhoiLAdsXSWiU4ESXrFDeTz7f/u6r737GFh3GlYIGeYbvFl6dBFPLcyfXmlnPh9f53mGaZFYKcJ+ptKoZxlXzAJY0sodaNQ1gnWwcHWn9PFD3RCIKsasmHEl/PO5TrThwVyRrHfHIOD3ZnSEFQjlEJKF4wTRJBXq5Ygz1LBk27Hbo7XRbaHNa1nEZ17ZXa2dSqRTiB36rpLMs2pm5WAXqALb6H+QGea6sZVa6sb3F5BzDDcAU1BfXM8kcSL6hvXCeUwwWdZRXWvTEy4C8zorkH1c3EFjiVRPIbpHrj9YkaxtyK0oza1zyJ791Jt4pmZ1i3+9u0DnnrK+drUFDVDuOEG7lubFqUSsLLifl2dLfJ5ahCRy7n7LncLtRrdb+Uy8I1vABcuCKi1tnX3nf1zcbQJ3b0bePFF69hBe08zvQM3l8go5TI1sJ+fB4pFem14OPtF5vfuBU6eFFhZoS4+IyPhmzy8/DLwla8A3/0uC+O0mJlxvzY66n5tfZ2EdLcKY4CE3nPP0f86YVws6u87+bk4hObCglMYA9lsAMMkCwvkDFMuk1A6coRW6I89ln0BRV17rEnl4kXgzju9J5cDB9xtC9fXO9tubu9e0gh7qQVgu5w44ewINToKnD2b3ng6ga4LVS5H91+r+840gVtuAW68kTTnsDz9tPvYWb/XmQRI22bO9BY632OrwCyZsiN9kp30lauFRQBqaMH0H7p0vaDxC/bylVHygdWshKSvQfZVZxMWyEwoghRQmJiIloOcRmCWvbMOC2UmSvtMXQCmrqFJKyoVCijzChCL4/6QVe/si1AWytmBBTITGFWb9BNaWekLXK9T8ZBCQb+I0GnI3RoxzKQDaciiLQ251f7n5y0tPGpVsHpdiIGB9VgWD0wysA+ZAUD+04EBYPt2b3/v974H2P1cL77o7S9bWqLgrGeeSc8XZprA/v3AK68AjQYFmal+4hMngFzOcH12164ODZLpespl4NgxCoSbnIw3OlpmWzz1FNBo0HW6ugo8+mj4fS0vA2tr7imffdXZgQUycy0yutEg4bV/P0V9qgwOul/7+tezGw2qm7T+9m/dr335y87no6PACy8kMyamNymXafH58svxpiotL7tTH6MyN2dlbACAYQCVCqdWZQkWyIwrMhoAjh51a7+f/7z7s42Gf0R01iKYhXAvIJaWSKvZs4f+9no0MdM+Mqr6+uvpb1KLUq80xwMHwu+rXKaI8fl5ejz5JD23Y5rAwYPZXWT3PGnbzJn08YqM1vmWqlUqGxgkItrpn22K8fHOt6lTq0kFKcXJJIfd3zoykvZowiP9ubmc0xebZMlW9f4M05EqKPJ7yap53VgVsBdggcwIIegmVxs7+AWmBIn41EUwd/pGt6JKu7f8aK+gawTSTULZXl8+bGpfu7SKwG6Hel2IoSF3sBcvXDsPm6wD0A9mnBMnqASi3XTr51sql6mUoF9AiK7a06VLnS36US4DP/oRUK93T3GVXkX3u1+40PlxRKWVPzdMFb2FBWDrVn2sho4jR8iVopqY42B5Gbh82SkKuqEqYC/CtaxbYK8pPTQk8O1vk691eZkuWJ7c9SwsyOpDwFtvkbBvNulGZ6HYnywsAEePOstDjowAb7+d3pjCYJpUda7R0L8fdCbdsQM4fdo6D5VKMoI2KPY5bnAQ+OAHuYZ8WrBAbsHBg8Dhw9bznTufw2uv/e+4ciXHwsUDdeKdmgI+/nHg/HlexPQzN90EPP+89dwwaJHWTdRqwEMPuYXv9HSwyHzdoqRUAi5fjnecYTFNbyWjVgP+9E+BsTGqMc/3b3KwyboFc3Ok1QHA0FATxWIRV67Qaeu0+bVbUOvyvvQSpUdt3kw3c5CcZ8aiV1wm6kT+wAPpjKMdlpYoOnlmxqrBHlQYA/qa1fZUpLTwckFVKus4dEjgpz+lbAy/uvRMDKTrwu4O7AFM9sCObg8QkhHTY2PxVhZS6/LKRz5P1b2i1AvuV3rpehOCayjr7o2sngsKwuNgr07CAjkCadRcbgfdJKgrpB9nOkWlQhG0urKU6mPLlviO20vU60Ls3Ok8V/PzaY+KaZdKRYhSSYhNm7IrjIXQ1+hOMr2LYYGcedpNd1AFrxS6e/bohWPcE0S1Spqx1PCKRf1x+SZ3Uq/rzxVPiIydapUEe6kUf0qU3TpjGJQ+yNdesrAPOcPIAJBz56hy1u7d4ffxrW85n588KXDLLeT30hG3f2hpCXjiCSvl6Itf1G/Hvngny8tUs1ilVWU0pncwTeBd7wI2bnSnR5kmMDUlcOiQwFtvAVeu0BwRNI0qCOUy3bOLi+Q3/9GPOKArcdJeETDejI6qGlIzdPekmRm9Rjo8TCvqoaHO+7OqVaf21wu+0bjx0pDZmtAfkK/Z6e6RGrDftTE6mu64mfZgDTnD3Hqr+oqBH/yAcgaDarJf+QqQ0/zKly4BN9xAf4MWA1HZu5ciTYeHvbs+6VhaAlZWuFiHH7/2a1JDdubXTEykc66KRUpTMgwqaMEkR63mTo0CrLx+L+sJoJszuoNeySRom7RXBIw/uqjksJGO9Tppytu2WXWE29VKdX2E0+593CuQZUQfDJdGhOvAgHs8rIklh1d8RysNOYka152AfNVcQ1sI1pAzzwsvkOY6OWnlPYYtaydbw505Q31b49BKdR2iwmrvjJ5z5wD13Eo2b+7oUAAAa2uAOh4aI5MEuvtydtaq5mXv2jQzQ3/rdSp/202YJvCZz1zBAw/8L1y6RNdXv9d24EpdCWOaVl/edsvR+VXT6TTUQ1n/3uIiFRlgorF1q7fAS+PcFotSKFuMjma/TaX9Gu2mEp1A71fHMk3grrsaWFkpIJdbQy6XQ6OR7/vqhyyQE4QuOvKXAqThzs31Tp3YvXuB738fyOepBGKjwbWq46JQANbX3a8dO5aeD1kK5e4Rxk4/bLcJ5VbIevG33ppuLewoqCWJ5+eBd74zG8pGmrDJOkGWly1hDJDAeuqp7jLr1mpUg1gXtHXiBH2nlZX4TOEM0WiQmTKfJ2E4MxNNGO/eTfuIkjJnZ3XV8lYmIYyjBgh6oXOpZLWz1I4dFCy3Y0frbU0T2LcP2LZt1ZES2W66k1+KVRLYSxIPD5OS0qp7XF+Qrgu7t6nX9T2Bu6X8nBq4leWqQowbCgi0fr/p6bRHpEcXINjutUb7dD6y2Ht5ZKTh+O4TE97bUjCXPtgvapBdvU7V3wxDn2KVJN1W8bATsEBOGHnBz8xYwjkLkYT1OlXeKZX0FXh0pTVLpfTHzQRHN3FnEVnJzf4oFtvfr10oZ1EY0z3mFrBe6EpZtiNASWFY43zmDME+5A6SlaAs0wTuuEOg2bRMerkccPy4NS61VZ6EfcR6hoaoWlI+T5XJsnB+BgacvXsLBXdwVhYYHta3H+z1mUl3j/n9RqZJ5mSZg5zL0bm77bZoPmTVj2sn7R7NrZBz6ebNPdbWNe0VAaMnSXOO10rbbkbXacjdZG7vJKWSW9PJgiXBMJy/m2GkPSI9umstq+b1ONF971amemlxm59v/xqz16ouFoUYHydLQifM1e1gH7e877JgdYwDFsgZxHnBxV/UvV4XIpdzTgS5nN5sPTYmxMaN8RUUEUJ2gSIB0Qt+ad2iZWYm7VF1j8laCGc51X4QxpKkWqAGJc6Ff6d8wkEUim4lw7do/6K74Ayj6XnDTk+TcNuyJfjN0MqHrNs+jptN15Ix6yvyVugmh5070x5VdwlkJhwTE9aiNsn7p1oVYnKydRW+el2IUmm9I9qqqrCwhswkCjUGdz90rffUSFogHnNWUui+l2Fkd7xBaGV6VN/v1HelBgW9s/Dpd+SieNOm1Y4satWgM/U+rdeFuO++FXHw4LK44YbHO6qtynNRrfZWpDYL5IyiTqZeF7pqes76ilGaq3vN3CRNj9u2+QvjNIRyO/20mWyg85vaH0lERW/b5u2KsY8nn78s7rrrh5HqUVcqZKUrFt33Tj/ChUEyypEjVMN6bMzq1qSrYb1rFwClIxCQ3Zqwb78NlErO1wYHw9XmziJLS8Brr1G9cHvHLK8CMLKcatIcOUKFPLIcMRuUuIqcZAm/wjt2lpfpniYMqPd8El2eDE059ddfd49nfb2EsbF347HHjFDFgWS/9ytXKHL89deBQ4fiKQzTtaS9Iugm0jKTtPLfSn9SN2jIkjijRbOMl4Y8P5/2yLoLtetZViPGw6CahP2C2ewa6fCw1Ss9SR+ynytGHU+Ue9jd750ee/bE+z26CRbIAaEk+obDZBTlQkwqEjHOxYK9oML4eG8LzE6guh90sQCMP7qJO4vFPsKwceO66zv5Cdc0Klv5RYG3Ox4vt1w/m625MEhAvJLow3TfMU3g7rsFLl0yUCw2sLi4itXV4UwlteuK8hsG8OST2RljNxJn16+sY5r0HV9/HXjf++JpC6gznwI0hXcjtRpw6JDzPgPInaMrktKrLCwA3/kONafZtAm47z6ny6ffYIEcEBKmdj9O+KpVbqG+BmCgo9WvWrV1K5WcDTEk8/PAd7+b/PgYi1zOEjiGQZNWlpDjs4/NNIH9+53bzc62L5S3bwdeecX5Wjd3b/KqhLdpE1WeYvoTDuoKSLlMQnNxkYKtonQ2snc4KRQAYABA5wKw5Kr8pz+lbji33+4OoJiZSX4cOjZupIndMEhL73fswhig/3MZulvt45NCGQA++Un3tn/7t+0f79FHnVpyqdS9whjwnjcOHvT/XK1G5SKHhqJ3ZVpYoJ7bnejqxIQkXYt5/2H39bYbFBGWnTvd/hpdhS61U06xmOz4dMVCZmeTO143oPOtZelu1Y1N1yRCvh4HnfShysI5+Tw94q6WJwTNAZs20f43bWrtO9V1xQob0KX6bf26SzGdJ0O3eP/R6SCNmZlwOcCdGp9uTIODyR4z66h1qOOKLJZCoFRqLzpXNz45xjCBSlmESsu6c311i9dO4dUZKmz+sa4OQNZ+n3qd5qqxsdYVwnoNFsg9QJjydmohkaS13yDoJol+15CFcAq3uIRxnBWedJpwL2hg8/PeFoq0Ctjs2aMfT9jfb2rKvY8stVqs14UoFJzXaD9lJWTIK8VEgfzCwE9+Qn7h/fu9E+vLZWqxOD9PvuL5eSoYkXbE79tvU4COJI4goDgxTfJZGgb53jpFs2lNS1ECunbvpjEPDtI1QUVKnFG9Tz8dfXxCUKEOgP42GnQ9VSrA6Cj9PXUq+v6zSFoFbHT36Oxs+IIvDz/sfk1XVGTr1mSud9MkP7lXwZzlZaDRcF6jjUY2ixwlQtorAqY9dCvnbq8NnSWorngzsxqFF7oa57rXsmaubJfpabICtdMxSvebA2SFSpOglrBWSP+42mqRivU0BeCslR3H9S6PKffrNUeRhuw87/2kIbNAziBhbjyvSlDdXhs6Ltqt5ezV6i3r6GucN6/9HRjoTWFsFyRRg8m8BHKpFO94s0S9LsTAgLtQSRzXu84MDVDRIa/t+9WHzCbrjCFTk6QJ+o47vM07ACXRz846XysUur82dBxQrVzg3Dn6GyXNg86j0O47y+hrnBvX/t53X2/Ut7bz4ouA3SS/vg7s2BF+P2QedVciuXIl6siyz/IysLamFwejo+3vWzVDA1ZdbJVyGXjmGaoN/8wz6bvUOgkL5IzxR38E2CeDZhN48EH/z5w4QbnRk5PkGz52rL8uYi/qdfW5W7C2olwGJifdk8nRo9kugv/Zz/4vAHrHs2H0z4Lt9Onwn/E6N+0KpiwzNwcUi/ZXrHvlj/+4/X1T3QUn73tfe/vtRVggdwFnzrTeZmkJePnl/ltRhmFtbQX33nvMN6hExz33ADotOaudab761Wfxuc+9DSCvff/OO3vzGpmYCPZaK8plWswNDlqvjY5S16xepVwGvvhFe/EVaxHabkBVuUxKwswMCf183jtwUwZ91WrAvn30CHOvdj1p28wZJ62a3TPBcabgNAXwPwRwweFP9Qr+UXOwC4U3tH7FoaHkv0e1SsF7ra6DZrMpfu3Xjru+Yz9dS/bOZ4OD/eV/bBeveInx8eBd2WTMxsRE+NgNr57P/fQ7skDOIH4dVphw2IO6vCac6Wk6z87oTjkhNK7+7xZunQjwUnOHva6HS5cuiX379olc7rBHIFd/XEv1uhD5/JprMpet/uKMkE+j+1KSuCOcndd8q5oFtABuuj4bVCh73Z/9FKTKArlH6LXJIQkoejbeR1xlIb1Q09oGB52CtV4X4lOfekMMDX1cABDAR0U+f1kApG1UKsG0616gXhdi48bXXb+RYVwRcabxyCjgVik83Yj8bmrqURDB6NXfOOj5dmrI/WnpYIHcA6jpHr2WzhInugmjWIwukJM+104XhqV9bNv2h2J4+BPCMk9fEO9732+LN954oy8XZ1SFrqH5jda01o12jmPXwO1m3V5C5zoLpiG3d4/Ya/3Pzwc3lfcKLJC7DHuO3vi0b3X5AAAgAElEQVS4FCb95ytsB/t5kubqLAhjL4Fx770/E4CaI3pFAN8IrL30Ot7lLtdFPn8hFg35P//ncwJ4Q3ucQiHe75M2cp4ZGgre/EKI9nzIDAvkSKSlgeiq2Hg99uzp7Lg6ZcJNCq/Vvb3AxsREMr+7df7s/reGWFp6Ttxww+MC+KhGIDfFt799seMdw7KKV+MUuVBp14dMv5FbM7Yv7LZsIfN1O1XCskC9LsTQkNsXHLRill3LnZmhIkesIASDBXJI7H6OQoEutHYENJmbSdNtddH6BT2kpSHrqhp1q1Cm0oFU4m/nzs6cw3pdiGJRN9E3hd0cXSi86TjPsmlDpxeHQSO+O41fQ4g4zo33vde8KqidC6ZuFsrtBFd5RUqz1S4YLJBD4r5Y10QutxJJS3HXFm5eiwYOWudVNZt1OjK7W0tLZgX9+Wu6JvjFRSulJ+4OSgMDtN+BAf/tgkZ8pwEtbNznMq5yl17lNHWCR1pWuhWv4CqArAADA94mbD9h7mW1C6tRy4XzzEzv+Zh56gxJK6EYxo+nrzfc8BXuqg85bXOQLnK5WzXkNPCa/OyvJ2mOJmFsCRQ/obx9+5lAE2xayIm9VKLxjYzEu38vt0ax2BCG4W7k0c3Ic1mpCLFtm3f/a3Xu8RPmOn9yWI26Xqeoa/t2vRSFzQI5AtUqCR15MciVeTQN2XulbRfulQodM5fLXq/grPuQ5XnO5bIZZOJldu6EObqVdUOO4d57jwngtzOrIXcK2fhFXTzV673jQ1YJq/VKt8bsrHAsVHRR2mH37bW9lyJk1767IfOABXJE7JNlXD7kSkWvFTkT7umRNaGcVdyLHpo4s35jdgovgew+bxfEfff9eWZ9yJ2mn1LLvNwBrTXk1ZaC00+jbldDpn07g9OyHvzIAjlj6G50XcJ9r6VZqIyMxLP40LsFku+x6qVxZg01jmF6Wt832U8LYXof6bcdHw/nQ87l/DVkue8kfMhhteksYAghRNr1tHsZ0wQefRR49VXghhuAAwfCF/aXbQTtjI8DP/95fOPMEhs3AhcuOF/zKkbfit27ZVs+N4uLwCOPhN9nK6hAv4C9QH9W7zLTBPbvX4ezEYX6nKjXe7MpBRMfpgncfTdw6RI9HxoC3vEO4D3viTb3xTkWABgeBh57LLvXMQvkBDFNYGFBYHXVmpiLRepDG/aC2LsXOHmS/s/lgOPHs3tRtYvh7nYIw2jiySdzkb7zjh3uNnyFQnJtKnXjz+pd5h6rALVtzMG+oBgZAd5+u3PjYroP0wT+y39p4i//8ix++MNtsF8/1Sp1pEtjTMvLwObNwPnz1Aoy0/Nmugp6bxO3yaQf/Fb6etPt+3/UoLkkg8904+/076YGGXmlTOkDappi+3arIlU7tZ8rFYp6zuWyGZDIxIPTF6wWscleRH5W4X7ICeJu+k3P5+bIDL1xI7B9O3DLLcH6fpbLZGLN9AqvTfS9V2mlfelStN6s+TygmpDX1yMMLtTxJKRxHj5M5jO/37hYJI1VPnbvjnZ80wRuvx14802aDl980bIQnD5NFgN5PGuMTv7wD7dcm06j9gGuVNZx9KjAlStAs0mPkyfJ2sP0FsvLdtOwW6yUy3Rd3nILcP31wHXX0fy3sNDRYWaftFcEvY4uAEEXNe0XLdhP+HVkiqoh6/bVqghGu8i0uKDWEdrefU2ETaGp190dory+v7NUp3fKXVRyuXOex087Stv++/T7Pdcu1SrVvLbfp5WKs4WsX/2GrKUippkqxQI5BbzalMU1EcriIZ0q/xg31aqzEMH0dHs3hi7SulO4Cx80Xcf3W4SEqfjkl0JiF75kvlaFfzNQIZJ6nSJtCwV/8/Nv/uZvCuAn2kVGO0J5dpYWru1H3jvHxUI5PLJve5CUTL984zh7VLeLrlBJJ1OlWCCngFfFnzg0ZGpDly1tJApx+8vlOUmjpCEJ3IaikdJ79vQu3aOVhlypOL+bU5ivCmBNTE5euCqE18XAwM8FNatwC8pW57xeF66KVLrJ9/d+78eC/IhuoW9/HtavODvbnvVAfod2UrrSWNRlEb8OaToB2y0actqpUn1+WaVHpUKT8fg4abPj4/S8UmlPGOm63oyNxT9+JhiyTrROYyVhrBNazWuarB86Tc++qj9+fF0AHxUbNvzxVUH0UQH8uQAuuwRkq5rPzWZTvOMdx7SLyPPnz4u//uu/Fl/72tfEzMxXtQKPHk4hHXahqAsWCnuPeE24QfajW1T0K35uEVXAquV+x8Yo4FDOd1lCZ2ViDbnPcPqUm8IwnPWspYCenW3dY1Qt68cCOT3cwtjZXlHfx9rq8uQ3EVC9Zr3gKxbPX7turI5RV2yC2P3wol4XYt++HwvD2H9VoDeU7/OTq69/w/bXe7KemQlX6ateF+LDH/6h8NLq5+fD/SY690BQq0mY89brqBpysUjXpDo3VatNl1Ul6xY79iH3OX4+5fl5ffF1L6GsMyVl/QboVfwEE3DmqrC0C+mLYs8ed5encPtuiqEhUywu+rck1Gl78iF99v/2354X9haQboEsBPBjZZvf8tyv33Wrw7mguCCAN137DiuQrf3SI0z6m9d3MozsaXpxIf3Eui5yrcqoevWQnpxMfNhdCwvkBKhWqbTcwAD11m21wlJ9ytIXODzsPan6dbLxu4n6BRlFm2azC39B+LxN2DQFaa/251SH119D9tq3FJoN4TRPN65eX+4aw2obUOvz9m1Urb4phoaeU7b5c83nnJpUEL73ve8J4D86Pru46PS5p5GV4Leg6TWh3O7i3ss9MDMTbTz2fuXbttF+ei0YjwVyzKg9Y6WAtV84sljC0BD5VHbuJK1EmqPVxhW6YIipqfS+Y9ZRJ8u0hLKzZZ0zoEv9TQ3jH7WC1Q9LKHtrpO5gsoa45ZYrPtv7CXp9IJi04AwMrATQyv0jpaen1wVpVc8L4KOiWCQNy26+T7tAjlet7yxFC8eBzk88NBT88zp/bNQa8l4NLpKuSd9pWCDHjFewgzQ9ekVY+60+q1VnBK1hJH8R1utCbNhgHbNbWsrpA53SG49eKMuFgnx+UZCp94KynTtFSoefRqoX1qoPWy9s/YW7JVzrdSE++cmzYtOmT4kvfvGKZl/Ofsv253ahPDnpNOG/612rqQtfHV4NS/pBQw67uPXzx8pWllIp8dO+/dKmstwsIiwskGNGdxHbNWQ/f7FfGog01/h1N4kLr7zYtIWyDJLyK+qhG3dWejQ728FdEPY83VJpRZBmGG4x4ZXG4ydEvQW1//uGYV9guIVqpVIR733v/1A+ty7cvnLr/cFBIS5evCh2794tVH9jmBQ1eV91QkvVtfTs1ZKgsuRq3N9TZ0kE/P3RrCEzkfDzIUfRkDuN12o0jRxeiapZeQnlNIuABEFqDMA/KROSl5baFBMTTc/9vfOd3/EVqoXCE4LMv/aUIz9hHERTdgtVIYT4+7//ewH8trD64Hodz3o+Pv7PAsDVx//neM9rATg9Tb/z6KgQu3evCbIwWJ/rlFA2jP7orR21GItaT92OlyUxiFLCPmQmVuw+5C1brP6ilUo2zHNZ1JB14/EizSIgQVGbrPsL5XXxO7/zO67YglLpovIZ3effFADExMSE+MAHXhOjo03flKlgD+dx5ETtjIrW7z+Xs/tgKZgtl8uJP/uzPxNCWMLW61p797vtVgTvBQaTHjLvWP1d7L+pV2GRrCglacGXbopQ5SO7ma7zpdr8xpYlH3IYgdwNqFWn6Ld3F76wcn3tZSgbmm29hFNDAH8uDh/+mRBCmnaDaMjBBLY92n9xsfVnNmxYF5QTbR3Hbu2oVGiMExO077GxN8WWLf+HMAxDkBbtTqNRx9trwVWdQKYwyQVR1Hu+XvdabLoXyGF8yPYxytrYduUli7EGUejyaa276ZdAhThQmzVkxS/cDq01ZBl89XxgAen1GB5uBhDG4R/SXC2EEAMDeh+4e3Gh12id47NvsyaAj4qtW7eKbdte0+5bRpwnLYzlxC99q4ahL4jRTTi1VW+tNgh+c5rcVxRh6vQ5N68FRJZK6+LXf/0VUSo5iyl1KyyQU8QrpanbL6qkyEJucZzoBZYaMR00Alo/oQZ/z/m+7NhDGomX9m6Zq/VlQP0WHLqqZd6fl8FkhkHCUPqQw1T9ahdd4wH7o1uFsl8ZzLBuHy8N2S6M5TkcGmqKz3/+ohgaam0Z9Bvj4OCLjufdrMxwP+QUKZeBY8eAmRlgchKoVIDFReCxx3q753FUGg265RqNePe7dy9QKnW+T+/oqPqKASCvec0AND2LnQhMTQnkchdt24oAn7M+DzTx1a/+w7Vr8MgR4C/+4hUcOtSAvZe0pFQCfvADOm8XLkC7jf67yP/pmNQ/1/6eyjqEoO8hBPV0Pn6c+jQ/9xywtBTwK7aJs+eve6xPP92ZccSN31yza1fw/Zgm8OCDQC5Hj4EBmtOEAF54gbaxn8PLlw38/u//GJcvt+53ro6xUKC/w8PAF77wHgwPW8/n5oKPOXOkvSJgvOkVv0iWUX25nU5fsdLgwuYEuzVrIYT4xCf+XwGcD7mf5lVNY01s2/avxb333itef/118YUv/JWgiGnnfkZGhKaFYxRN/mKAz+j3u2VLZ38nIVq1t+xeDVmI9n3I1GVOnw9vt2DYz+HwML0XpO2nfYy97EM2hBBBl9BMBOSq8ac/BfJ5YHAQuOee1qv6Wg146CG6pA0DeOCBzmkC/YShUco6fUe4xyAH0Erj1H2mCdKyRcDPO7czjMMoFP4b1tb+TwAfAlBybb9r1z/ihz/8RZ/9Bz12dAwDaDYTPYQW0yQt7sknSVM3DLqnP/IRsij0KwcPAocP69/bs4csGRJ5DufmSPNVn/czLJATxDSBO+/Um1irVW8Ba5rA/v3uSa1e5ws2brIgkDduBC5ckL+3wIYNTVy8CFim3KRZBwnxKwAGfI4rrm57BcBIxGN5CWuvRYh+++lpywzKpE/UuY5xwj7kGDBNWiHWavTXNOn15WVvf6fcRgf5UdyTkJd/JUm2biWhlc/T92OiUasBN93kPIcbN9K5bTSAXG4dAPmAvYVxEisFuc/vABgECWavRYD0+0YVxnIfXmNQ31sB8DcAVpXtgJdeamMITOzY42E2bQKGhoCxMRbGoUnXYt79UIEG6TtxRgt6RVGrfhXdPtWKU2qDik6gS5PpxcR9+/dLAndaSUPk87pykheEvtSljHL2KoPZ7sM7B3pq6pwoFGSXKL+a2W6ftD6qOviYfuEXviy2bdsm8vk7tJ/vNWQq1cRE2iMJj8wfn57ubOR7r9GDl3V42mlI7ZV3J0PvZdWasTFKgJ+cDHax1utUIq5UCtbCMQl038uvtJ0de4nQkZHeCLiIirPBRKsAKJ0wbog/+ZM/iSBkw7znPq4s2GG/P6ygpqQWB/RQA4pkHXP58Ktn3o2oQXKdFMqzs6Q4jI9Hu0fpXncvvMIIZXtQVqs+y+2S5QCwvhfI9bq4lgcXpVKWLvKyV/KIdXWhg9wkzhs02nntJYIKxXz+itC1SwQgfvmXfzmCQI4qMJueAo/qCb8kgI8K/y5TQRYL+vF5CaMgzUW6Fd156ASUZWD9DlE6yXk1zAm6eLfPwWSFiSbUgx5LNnjJ55uZ0+T73oe8vIxreXDSh+WXD6dSLlPO5uIi+UtkDuf3vw/ceCPwrnfR48Yb/X2wpknbDQ3RXz8fc6f48pedz2dng/mDKB/T7g8Mf16TwjSB7dspj3H79mTPs8xv9o6itp7ffPM6nnhiEJcurYIipcXVv5Rw+Td/cx22bVvTfNZrv155vfZt1m3P7X8NCKE/NxQVOwngONwhKOq+1P/tY5N/V1yfO31af6+srtI0vbrqfo+JxsmTgP06ESL8PXrrrfrXgwag2ufgZtMZwxD3/Ul50LT/9XUDX/96Nubaa6S9IkibqBquvfOIaob2KpzuteKr1+29ccXVlWI2tMko5iOvjlZpa8i6lm9J9ZZ25jermqC+6pWdZrMplpeXxa/+6q9e1UZl0wa/qlZuq0RrTfWjAviG0JmgZ2b0381tAbH2Vyyui2JxJdC45DW+a5f72EG1q16BKp1ZD3uN8CTRXRdRzdbSh7xtGzXOGRsL1jDHPgerMTdJaMhqGd4sVfbqe4EsRHgfsld5OHnx+JV50000rfzQ3UjWfMheHaySOs/u60Oa5C6JX/mVvxb6Epl6/9YnPnHGR8AFeXiZiJuiVCpdbdygHkMI4MdiaGhIfPrTnxbvfnejxf6awjB+Jh5++GGxbdu/FqUSLToGBlYE8G0xMuIWunahoy5i5b0ka1QDvdtzWCKFclBhXK1aAmzDhmj3l1oYp90mMu5FbzB3VSd9yPbzlraSoMICOQJeAlQK2ygaclwr1X7F3khdN6l4/Wad0ZCt60Mey1vDvXBVkK2KD3zgd8Xu3btFLne7sHoMBxXAltDM5dZELieDy6zXR0aa4sEHHxQ33nijuO2288I9kf6WACD8m1vIY70lPvSh3xe7du0SP/nJT65NsP/qX/0/wjAMIYQQlYpTKNsrW1Wr9uC3FUGau7tZRa8L5aDorD1R54yo/Y51+CkjrRa+UhDPzpLVMcl+x1kN7GKBHIFWGrIQVmuxqSm9WVsl7hKOsufypk29n4LgLuPoFsppLHqo7697Ja7vuuRuLLF585+I0dHRq0JRDfbSacCqMLb2+5d/+ZdCCHsTiAviF37hv4qhoY+LfD5/dSFg3/+q2LBhg/i93/uxjzCWD2mCvyCmp+8X3/zmN8XKyooQQohdu74oCoU/ENWqEL/0S2cElctcFxMT1gRM1hTVYnBKe1x7d6l+xkvwtRJ6cQpfFasPsvsRpCym7vouFJyfi1N7lm7H+fnsCGYWyBHx8yFHRfph2q2Jq/Px9bJQ1k0Aui419oVUp3rm6lbiei2zIUgj1XVBajfFiPbxT//0T6LZbIqJif9bWD7pC4K0UZ2Qt/uug6Zp/cerCwgE+LzXAqMhqGey+3i9pCHL1MaREboewwhKLyucn2BRI6rjPJf2SGnDWBdyYTg4SN+v1ZwWRLNWrQLtzGk0F1j7KhazIZRZIHeAapUCHIaGSGNN0hQjhDtoAejeIJl6nRoJGIa3f2vDBvf3bdcXliRqmsjoqBBXrlwRMzN/G1Bg+QlFL+HdFFu3bhW/+Iu/KCiIy3rvwx/+oRgYWHUcb2BgVUxOmj7HcWv0QFN86EPfE+9/f0MUCmuiUHi1xbi9H2qQUy8KY2daYXhBGdaHrDvPceHXBzmIAPVaYNhdSqrQbmdO0403CzE7LJATxsvXUyjQjZfP00U3OxufX8NdiKI7NWR1Re8laOt153fuhkpHUijbNXV7jmTwKGn7/+viuuseEZb5WdW2pdBdu/qg90olOr4MoJLPW/X/1Ud7n1aOu+bz+WDfTY6nl/ATYEmZ5ZMUyK06YQURoGp2hmH4Z6+0ryFb+2INuYuRvhJpqvYTpH6mGC+No93IP9Uf3Q0CSsUrdcqrYXpWgzTCYGm3fmUyWwk2MkHfffdjwtJgG8IZIOUWpF6RvfW6d+EH95gua8bX1B4v7PfqxgWlH7ryuPKRlCXAHvgY57xQrZLlL58nwdbOb9jKr8s+ZMYB1adW0z3WPAWpX8S136Nd80mSwRudwEsIBDVF202e3XAO1Ahor9QiKTh1Jl1Ls3hEqGbpIBqqF15WHr3gDSpomyKX+5HI5dYDja1bXS5+tONDjkrc9bK9ro1Nm+gRplwwwwI5NK18JTpBqvMh69Ji5Io5a7lxaaDTkINOIlYkcfcI5TALtTNnzojDhw+LXO5t4dZAL4jjx9fF4cM/EzKgamhoXXgVI7GEuHtM9boQn/rUG+K66+4VFHDmp7UHFca0bbH4j6JYrGjHNT7uHh9P6NkkarQ3o4cFckh0HZyiJplXKm4fslwxc9cUOj8jI+Gba+gmiKynyzj9/l4ackPMzHxVbN68WTgjsGVE8zcE8Fvi059+U1CUMr1fqQjxznfq6k5bx9i163HxB3/wB+Jzn/ucuP766wVQFq0jsf2EcSttWlwd+2uOz8tFl+oWYrKJzgKYlSqD3QgL5AiE8SGHIY6uKUx3Ruhapj81EOsnigD7ttDlJAshhHeaUVMMD/+5sIqLrNkE9qqwCoB8VAA/FMD5q3+dGg9F77erIauPi8Ierd2N8Q79jvQhDwyk15muVzCEECLNWtqMxdatwLlz7tf37AGee67z4+lmNm4ELlyg/2dngRMn0h2PH7UacOiQgLsZhADQBJBXXnM3jRACuPnmE3j22b0eR7kAYADA6wBGAGy6uh9x9bX/C8ATkM0sAIFczkCzCQwPU8OU5WXg8GHXka9+/rqr4zSU96Adr982ExPAqVP6rWs1agZQLgNf+xqwvg7k80Cj4XMIJnFME3j0UWpW8eab9FrW77tMkvaKgLHwiiyenSX/cz5PpuxujyZmCEvj1Pl3m55RuLrHP//zPysasrd5Wqe56mMjmuJf/IvGtWuN0ssuavZzQXzhC3+llL8MqiXr39fhruZkfTafT/a3CpIP3694VS7sBstU1mANWcE0SROYmwvePixOFhaofeH111OLwE2bgJMn7VoR/S+1ljTGyLRPoQCsr8vf1a0l5nKk/TlbN+o1zokJ4PTpJoBTAA4in59HLvdZrK1Z+2o2vUYicN11Tdx55+v45jevd+3b0tJzV/+ugzRt53aLi8Ajj9D1e/So/bOG7S9ps5s3kyVodJS2Uq1COg35woUL2LjRALDB64sgqZnMNIH9+52WCT8tvt84eFBnOSEGB4ErVzo7nm6m7/shmyZdULUasG8fcNddAocPA3ff3fk+maYJ3HAD8NnPAh/8IPUefustIKu9hbNMFvtL21lfB5x9gQ3s2UM9tYWQ79uFjCUQcjlrOxLGAnQr/28A6lhfvw9Hj9J7AAlBOPoSNwCcv/baG28A3/zmCPSmZePqvuVftzAGBA4fFjAMoF4HKhX7Z62/AwNkWj57lsZ+9iw9pGAG3IKuVgNGR3+GjRvPARh2HLNT3HsvoH7n06ezd02lxdwcCV4dMzOdHUvXk7aKniatKhF1MnTfXgtWjmV4OLu9hbOMrpFE1iI/1fKmfiZXP5NuMHOwrsylX7OKsA/n5wcG3PXU8/nw59+r4YDukaTJWlf5rtPzQ9aRRTa2bGFzdTv0tYa8vEyaJuFcAQ8P08qvk2O5fNmpUVy6RBpztUqaXj4PTE+TeZDN1d6QRuOk2cyWRaHRkJpr66Akuwape+6N/XrKIXf1bi8U1mFpvfJ9P41T2P7qtnPeO2trwJEjQKn05rXX1tfDn//jxxtQtWzJ7Gzw89cu73mP+zXD6Nz8sLBAQYpZsfQsLJCVxjCAHTvotXIZ+O53ydoiRfKJE5YFcu/e9r6DaZIF813vInfeLbdk41zETtorgjTR1V/N5ZJv/qBLm5qfd5edYy04Gl4aTTefS13tayH8K3apj5ER0urm54MHV1FvYrUgSGuNmqwUl689HxwMf/6twifWvZlW9Tl7K80tW5K/luScsHlzw3UO0ryOdZ3kvFLV6nUh8vnLrmsmrLXEK2hMbc3YC/S1QBbCyiGenRXCMCxTcZKNsQsFe1StzAWli25+noR0pyKpZ2fpwh4f7+xxk8Q+edqFUa+iE8pq3WLAaoG3fXvwdo7/5t+cc21L0d+6mtX0d3ZWiA9/eM3x/vx8uO/0zW++ph1jP6C2BlQfYc9lnHgtAHX4VTUMY+6Paz/dQF+brAEytczNAd//PiBE8gFTy8tAo2E/7QYoUAZYWQHe+U4K5nrkkeRN0nv3UgR3owG88grlwqYV0BYnL7xApn3J6Cjw9tvpjSdp3n7bPVWdOmWdg1yOAq2OHAF27wZ+/nNprhaYmKDtZQCYHSEAw7gOqrn4X/5LoFhcBRymbMu0fPr08/iLvxDXth8cBA4cCP59TBP49V8fhmpK142xF1leBlZW/HK30+OGG9yvef0uc3Pk+lPJ58OZ+72CxgqFzroVO0HfC2SAbgAZ1Qok+0PPzdH+7cjnOr+13XcyNmb5TuTr+/ZRJOrBg+GF6Pe+B7QTwb17N032AwPkH1pYCHf8uDEM6/G1r1nC6ezZdMeVFi+8QN9/fZ2EMQD88IeAXXj+/Of036lT1sQqhTTgnlSnp8k3+Pd//yMAa8DVAiJym4GBNZw6dQrN5sC1zwwM0II3KHTtyfQmK0I7zTSjWo3StYrF5H25reaeMIubuHn4Yct3D9Bi1+t3KZcp1mVxkXz+IyPA1BTwxBPhlI1yGXj8cWB+nj4/NkbR28eO9WAcTdoqehaw+5Lz+eRLVQYtvamrmw2onYGs/8Oa2tVIX/kIsh8yC7vNatIs2ml0vs1uN70ngWrOD1LkQnd9Tk87uzRNTFjbHD++Lsj37PxNgt5XXn2h04ra1XU0Moxkry+vOIgsFCXphVanWYUF8lWyeJHpg2/8H2F8KmpKVZgqYF5VpNSgo07RD/6luJiept+vncld/f3tfaq9WvKFaaFIQWHZ8CF7dTSamUnumLpucHxN9z5ssr5KuRyP31aG+afhgw2bqnXkCPkWR0fp7wsvBD8Hu3bpX7/11uDHTxr7udi7l8ye+TyZ2XfvTm9cafPCC2TGfuGF6PtQf3/789/9XUBXZCSsmdK/Bnbn8Br3mTPJHfPECTLzFgpWtbZOp2IyKZD2iqCb8DMtLy6SZiBNbXFEatvD/Q1DiI0bqZuKfC2fp1V6kOjoapVM5HGmdKmRvGkXAvAyV3tpG1kw/3UzXpq2lyskLGpEb5yR8vLaNYxgbpZSyf19OtWFLQvWuzjGkMQc1GuwQA5ItSpEPu8Wts5cZmeqRxzmJd2NEPbmUHMH48xlDDuWNCYXr8L3djMrEx+6BVDQxY96fUihHL8wdprDgwhlmQuey3VnS1R7848w59M+xw0OrtPbSekAACAASURBVIkHHmiI+XlyqQW9j9W+yblcM9Tn+wUWyAGgBPemVtiqOXKFQvK5zGEgX5y7m1CncxmtILDOnx/WkDuP/Zz7nWcpgI8eXRX/6T+9IgYH165dH7OzJATjDhTUXQtJd4tKG51fP6hQducBW4pHsRjsPvbyw2dlnswK7EMOAKVFWf4sex6dPS1keBi4/34jU6UtKYUk3Z95927gxRcFZBMFoLPNMez+OFnyb3q6PR8q48+JE9a063WeTROoVFZx+DD9ve++UaysUA7gpUuUI3/uHHWPijOlTpc3690NqzegmBanT172C2+FfY6jFE0rb3N1lfogt4qb8ZoLL12iz+vGK/dZqwE33UR/e560VwTdgN1kUyi4zVVZ8PF4Ua+7S3J2uuScV0R2Fs+XndnZ9Eo19gNuzcu7PGfcGuzAQLbiH5JGNRlHMVvLOBm7C8iegtnKlC99yOPjbteR3XztbLSzKlqlztnHltV5OCgskAPSaaEb5/FkXdyZmXB+n7jQlbL0qn+bFWjCTj8Htpchn6xXiUh33m/c9NuCK6oPWUXOJ1NT+riMVvOLVylMab72K5Wpps7pctaj1E3PCmyyDkhcaVF+LCxQJaBcjhqix1XGslwG3vteKs05PEym4k6mZamlLKens93cfccOYG1NmtgJqmrGxMm5c4BXalMu53z9/e+P//gnTgBXrtDffmBpiUzzQrRXSlZ2dhoacr8XpKuaVylM6cZym8idx7ZDHfucHcFWVvRm8G6ABXJGWFggX9naGt0wcfpaFxaoTvXzzwNHj6ZTr1qWcfTzKWaF06cBVVB4NWCPQj7vLPNpZCPdtuN41UAeGaE86dlZOu+zs/0jNLsJnXJSLLbOlbaXwpyZse4tmWdtL7l57Bi1n92zh/4uLTn35VUvu1sxhKDpn0mXrVulxuDEMIAnn4yumZsmsH+//r3FRdL6GSc7dkihbKGbDKKQzwPNplP7lvTjnSjP9cREtq0mjJ5aDfjTP6WiO+95D9XZDjtXmaalGUeZ50yTNGJq3EOLgiNHshFUGxYWyBlBasgqsktPVA4eBA4fdr+ez4cv8t5PSEFRKAD33x+PMAb8tWG+ExkmOu0K9izAAjlDLCwA9TqZ60ZGSJhKQWCawG/8BgmJLVuAm28Otho1Tdrv6qr1Wj4PfOlLbiEjL+jNm4Hz57v7ws4qLJCZrFKr0RxQLse3AGVCkmZEGRMMKkzijjgMGk3YKsq6WhViaMge2Rpf8Y5KJZniDt3KzTdf0kYWtxPpm3Q0MpNdgmRjBNnGWTikKcbGwlUjkx3sxsa8S2NmOT00K7BA7gL80gDU8pxhL3pdfqLf/sOgluzcsqW/b0bn+YgnpapTKUJM9nCW7dWnE1IdgrVr10k+L8SmTW5h61VJK4hQ1rWJVdtT2nOLBwcpZWrnzu4sQZokHGXdBczNOZuCSwYH6b1aDbj+eoGRkTXcfnsjVBS1fhuynbbbXeb4ccAevPTmm8Cdd6bTCSsLPP00YJ0PA6OjNH21H0HstIOz6bs/oJQf52unT1P8g32b1VWZO2RgfR146y3g0CFn5Ssv11SQe1UGU9kRAnjwQec2ly9baUkvvQT89KfucfQ7LJC7gHKZArCmpoBSCRgfp5SBxx8Hvv99Smk6c8bAxYsDEMIqPRgkXUp3I05Px1P+c33d/Vqj0bmSmVlDbU2ZVKvKXk6jWligjIQ4S2l2K3Nz+t/aniHgtZgHnMJ2aYkyCcbGnNsEuf/n5tz5wgDw+uvObbzSk/p1ga4lbRWdaY84irZXq2TGKpXi9fWSX7r7SmYKkZy/KwmfetLm6qx0OSKTv9NMH2cXqG70cTrPid5sba/QFcQcXa3SvBLWh6yWxNSVGJ6fd5fSZbO1BQvkLkfnA7YHVchgi23bOt+HVDe2rAd3ycWJvT5vN03QcaMrb5nWBCoXBuojDqGs88dm/VqVVKuWD9erJG29Tn7bgQG9DznOsbQS5nJOYh+yGxbIPUC1StGN27Y5L3BqgeecTKM0llCDNsLUoa5WaVxhozb9xpKUFqNrUQfQRNav6ARgWn2kddqgfLTCfg3rBLhX4GS3COUgxH3vSI2X+xrHBwvkHkWNcI4aOU39lFubxTqBbFYfdWHRCi/zf6kU73G6CS+tNMo+4jB563pbt9KQddew+hmv6zxOk3iaOJsw0KOd6H6K3rb6rBsGnateWsCkAQd19RC1GnDjjcB118mqX/ronjCR014BWGppyTAsLFi9iffuDfaZrVuBCxes8OFGwxnFGQcUwOIOUfaqudwPnD3rfm10NPjnqSQsndNms/2o2hMnqHiODCIaGWndKOGuu9yvqb2Ay2X997rhhmjjzBrOJgzEyZPB7z/d/lZXLfEhBJ3TuHtX9xsskHuEWo2irX/yE0ov8qJSCRc57SW8owopKhEqsL5ON3HQSUHXGcgexRkHS0vA6Kh7EfPww/Eep9sQwhJWo6N6Ie2F7neLo3uZbMISpGvRyor7NV1U8NmzTqGcy/XOb0/3sXux+YMfRN+fV8MVSu9josACuUegSU6vEU9PW91SwtbFLpedGgnQXiOA73zHPc6/+7vWn9NpL/fcE20Mfpw9S+fLMKhEab3O5UMBOi9ChBPGXnT6fOqunWPH9NuePUu/+eIi5dH3ym9fLgOzs+75YWYm+D5qNUqLuv56SreUHZvGx53bJZXO1xekbTNn4sGr4lbWfDoUwex8jI8H+2xW0m+Y4Kg+3zRiD4Swrp1ubl4fB5WKlXY0PR38c7qAR/s9yCVy44GbS/QQtRrwrW9RY4hGA7jtNn+NWLYtA6K1TYs6xkOHnK+xFppNTBP45CeBixfbs4rs3Uum0ZkZ7mucNqYJfOxjV7C+XgIgMDVl4OGHW99/N90EPP+887XJSeDll5MaaX/CJuseYmmJbpA33iDfmpcwXlignqH79ws89RTw1FP0Wicq5siKQJOTNEGzMM4mso/2xYv0/PRpYGgo2r7e8Q5gwwb6y6TL8jKuCmMAMPDSS8HK2eoCHrdtS2KE/Q1ryH2GDKrS+ZsXF4FHHun8mJjscd11uuBAum7CzBjq9TY7y1pympgmcPvtcP2GQe59e8/2QoH88LyYjhfWkPsMZ4MDi2KxvUYSTG9x/rzuVbpuwtTKfuIJ63MARdVz7eL0KJdJI7ZTKAS793ftoqCusTHg/vtZGCcBC+Q+QxcBOTVF5m2+wRjJe94Tz36aTfdr/dpcJElqNfLzBsnxPnLE6TYKounKtMozZ4AzZ4CHHuIuTUnAJuuME1fgldzPq6+SP/DsWfIJHjxIft1uoFaj71Eud8+Yuxm92ZoIOmvs3g28+KLzNY4biBcpLMkSIZDPG9iwId57WxfUxWbrBEgzxJvxploVYnJSiFzOSjUoFqOlbOi6vUTdV1qUSg1H2gWnPSXPwIC+/GpYpqfp+tuypbuuuSSR3ZEKhfZKWArhXfJVvU9mZ+l44+PhfwevtMrFxWjdoRg9LJAziFeTg7B1qIWgGy+f976ZugFdx6GdO9MeVe/Tbg1rRg/VzY6vrrSXsARIUArhbjRjGNGE8tiYlcc8POxu+NHpjnK9BvuQM4hX1a0ogVeU5uB+PZejfZkmmbbSDrTxG4eu/KLaSJ2Jn1LJ/zkTDarBHr5anUS9V2Qq4caN7m2lOfnkSecxhQjvy19aAl57jSqYLS4Cjz3mdkf84AfA3XenP590LWmvCBg36op3aip6i7N6naoTqSvngQEhJid/Jgzj0rXVblorW+rn2vQch67jEK/C3STRmrJU4o5XcTM5Gb1andq1Sa2MVanQb6X2PNZpz3FcJ36mbCY8LJATpJ1+oXH6ZeQ4pqbUG6eZ+k00MRFsHK1KH3qdryT7J2cJddHS69+3m9EJsaC/l7tvczOQuVstYRqmbGYrqlVyIcl+02ku7rsdFsgJUK8L8YEPXBSGsXbtBshCDV2vJuxp3UTT025hHNa3Zfmw3AFfpE30/iShW9Rs2JD2qBg/ZNBmWJ+rV9/mIAv32Vmah9oNIvMbWz8sfpOE055ixjSBj3+8iStX3O75tCthUR1bp095fBy4+eboKVWmSb6ouTnn571et5PPu/NUSyXg8uVgx/aqOrZnD/m79u93vpf2+U8Kr0IdfGdHI8i1myb2ilmSPXuA555LZzxMfHBQV8wsL0MrjAcH06+EVS5T5SSrXZrAK68A//2/h9+XaQL79gELCwKHD1uBHKYJbN9OwtD+uo5du9yvfeQjwcfgVXVsehrYv3/d9d7mzcH33U3oelNH7Vfd71ANb7p29+/PZnDSkSNUgtSObuGQlYBNJgRpq+i9ht1MWiySSSpqQFaQY4U1EdXrztzmKP5jMhO79zE/7853brVvMltTKkXY1m1qyoWXb7wfAk3IbE2PtFocZpUw94l6/RpG8uOLil+ciX0eKhT021QqFNw5MMBtE7MCC+QE6IQv5fjxdTE0tB7aPzo/317EpVeO9PCwLmgs+QAjKdCDPNi31X/Y/etB4hN01003ogv+UvsXq/dxK6Esg7c41zg5uvRy60/qdSFuu60hbrzxeZHLPRVJ+9MJ5DCrY11VoMlJGtvOne73kgogsSM1BVmwQH3kcr0zgXBVpODogga3bPH/DFUnsx4DA50ZayuqVUplKpWC3a/1uhX1LB+FgnUf6FIJR0f9j28/l7kcX4NJwAK5S1DziQ1jTRSL3rm7us/Pz9Pq1n6jhjVVqSkbdmGnvhdnakUQ1NSOoNGn3YJabamXvlsS6BZoQUzQUihnSRhHqeylS6+SC3edu8dvLtAtxPP53lnoZgUWyF2CLmVpfj6YabxadfqNCwV/v3YrLcwvZSNtDW52liaKoaHeEli6SXnTprRHlW107oxOLxLjwKtWdRBhSEV3LLeS/TNhfMhpFABJey5JAxbIXYKqIQdtDkG1rNcD30ishWUT3aScFQ0uy9iFctLCOKnYkXaFYVzjkrWsZeBbkrn96gK0X+YhFshdRJTKXzrN2q/alaqFyeL0TLroJuWpqbRHxUjIZ2sFj9mL00St1menWiWrj/zt0yx004mgVXUB2i/zEAvkHkZOBsWi5fPxi5DUaWH9sjLtBux+v14KVOsFZmb0946896LGbKj0SzUsdQHaL/MQV+rqQfbuBf7u7wQajSaAPAYHgQ9+sHU1Lmp0bj2fnQVOnEh8uEwIsl5FKgm64Ttffz1w5ozztaEhfdW5apUqyTH+1Gr025fL/XO+WCD3GHv3AidPCqhVqoKWjQxyE/TjjcKkg2lSqcjVVWo/euRINoXyLbdQ60E7uZy7NCwATE4CL7/ckWExXQYL5B6jVAJWVpyvDQ9T79I4JrIdO4DTpy2B385qvxs0n16AfjP6f2ICOHUq2n5ME3j0Ufo/au3zsOzbBzz1lPW8UADW1pI/blhME7j9djKwSmZnqZe32jN4ZgZ45pnOjo/pEtK0lzPxo+bijo/H529SI7ABKgYSlnqdfG4yFasXOzFlxden6wQVpbSmPWgpTNpNu+iqv/kVsEiTep3ut0LBmSdsj/S2F+eIegyv6yor1xwTHRbIPUhSbdbsaVfyMTMTbh+6iT3OfMbpaQp4SjPflOqFZyP4SpcuE2UZrqvwNj4e/3hVJifjGX/chBV+cQhLqk+tX8TW60IUi2tXr7kml7fsUjJwaTPdgqp9h+1dLIR+Yo+r4o9aKjEtoTw4uN5xweWFvelEOxqy7nfrROMFXbpX2hqyvSxluxpvGNQUxoGB5rVSmrr0xqC1CpjswO0XmcCcOEF+sUKBWjg++WT7fkTDAL70Jfd+ZHvHffuCt48jX52hPO8sCwvAyoozoO7VVzs/DsmpU85WjFF9yAcOuF8rlaKPKyhLSxSnkLs6U42OAmfPJn9cPx58EGg06P9GQ39ukmBuDhgelk5qgbU1A1euUG/kZ5+l/uJ2VlcpRoPpItJeETD9hb3imFeBervJ16+QiUoWTJu6ov1DQ50fRxKo9Y/7JTdURWdGj+tc1OvkNx8Z0ecs1+vejSGqVbI2sYbcvbCGzHSUchl4/HFKwzp+XB+h/eCDznSRlZVgK321abv6XFKrATfdRH/j5tZb3a99/vPxHycNjhwhbXXPnv7Opb3nHvdrQa04fpgmcMcdwEsvARcukOa7Y4dzm3IZ+Oxn3Z+99Vb6PZ54Apifp0dWU8QYH9JeETDpEldpP5V2CsPrNJCgY2sV0NaJWt2VCrXJ27Spf7XIXkeNp4jjd9b5gb2qe1WrQmzcGLwdI9MdsEDuY6pVIQzDEk5BTcOtGBlpT+ipgTxxTTi6Wt0bN8azb6a3qFTIDOx37cXdjUh11WQliI3pHFwYpE+hQgZNqF6LoBW9vFALhwBk4nzuuXD7WVgAnn6aTHFHjrifR+Gmm4Dnn3e+Virpyxsy3UESxWUWFoCjR61reHoaeOGFePbdCtME7rrLWdynUol+zTPdBfuQ+5TlZbiEcT5PE5uKaQIHDwbzk1FFKGeUcdiJ0jSBJ564gnPnqErT7t00QZ47R361hYVw+/Mbx0c+Em1fTPqYJnD33QKHDwN33hlfTMB3vgOo0fpJxBsEwTCAT386nWMzKZC2is6kg9pf2d4yTt1ueFhoixHoUPNeR0aCj6laFWJysilyuTOK2a4ZmwmvWiXfLvve0iUOc6/O5xrHb7ppk3u/nXJt6L5TXEVzmOzDArmP8QvokpWFPvShK6EnBymUwxSgcBeAaF4rvJDEpBsHcfsQe4EgldLiihGo151pPnEFWOmKkXTqulMXyjJ1icti9gcskBkXdq0YuCJyudXAGnJUdFqJFHbWWKzoaVkPe3IyGYHYagJULQFZFMr2frydqFoWtFKaru921OtK14c4jmb2aVYIUxfK9vzifD6b1xoTDyyQGReq2Wx+PtnVeb3uNkvbJ2lVOOoaJtgnqXa1CSqA0bxmylf3o2uysWlTtGMlxfvf33CNMWmhrNMqdeii3QuFaMfURSYHFVit0tPUQihpWGZ0td87Wa6T6SwskBkXYf3G7aKrkzw1pd9W1cJUraheF2JoaN019iBpLELohYVai1rXZKNUau8cxInuO8jFRZJIbdxubvUiqPAOQhRriX3R5SfIg143SeGVm8x+5d6EBTKjpZWWGaf/VCeQvY6ry9O0T6Y339xwTVzvfveaY/L1m1x15lRVkKlFIVrts9P+P535HxBiy5Zkj6uaef2uDcpVtx5hgv/iIIkSp0n8zk73UecWyUw6sEBmQqNqYO0KZXsgSysfmb23rDTfye3VcRUKek3Izxeo8x0ODbn9erOztDjI51sL42KxeU1j9JpI45zMSyX9oqITk3iYhZoUykkLY925Vc3RAP2e7RxDCk77Nel1/Cjjr1Y5sKvXYYHMhIIE1mWtudiPVpNSmEnLK5JX1W4nJ/Uabyvzo6oBVyrUYzaKD29qyrmv+Xn9d4+znZ8qbKam+ncSty/21Ep0qpbcTv9wnWm5WnUuyDggi2kFC2QmME4NNLiGbA9M8cp3jm98zglRfS1oYJNd09NF8noJVvuiQufL1fnG1f3PzEQ/B5JO+j7jCKJLaqyqO0SNB2hV+zwo9kWVfeGm/radslQw3QkLZOYafhPrxz7WEMCaY3IZHAwmXKemmh2blHQm06j+bnk+dD5ZVSCTJrTu0MTGxtyfm5x0H0c3kXcL7QYAqi6FuIWybjGV1CJF5+7YuTPYYo5hhGCBzFzl/vtXhWE0tBOrc9IM5zv2SmmKGiXaqQApSqe55PrO0teoHl9nmtYJZN05U7dBF92Vqqk27O+qBne1E60uTdD2GAFdwGCS+cTVqrXAGh6WDVxYIDPB4FrWDEwT+A//IQch8gCAS5ec/YeffhqwavsayOWC98Ol/ThrWxcK+prZQcZ5993A4cP0N44etDoWFoA77lhFszl09RVr/GNj1MfZXhe7VgNeekm49vOZzzifz87qz9n0tP/zLLN5MwAI5XlwbrjB+XxlJdrvunUrcO4cjePcOXoOAAcOADllltP1rI6LpSXg2DFq0vLYY/T8gQeoTjwADA7SmBhGS9orAiZ9VC1HDSxqp0CCmrYRNcCoXhdi27b1yNpYUM1abw1o+ppk/SpPBTWXByk5mSb1Ov12IyPO379d/zdZUJyPPXvCXyN+VgavsYcZY7tWmXb2kXYuNNM5WCAzDqHpFQnazqQQR9qHzs8adH9kfrYiXf0+585PtczVXoE/cdRmDnt+Ox20ZY8yt3/HyUnndx8bC79/u5lXPsL6o9XfLS6zdJB7I0lUH3tWF2xMPLBAZoQQ2S5er0sp2bkz+OfHx52f9aoCJoQ+PzXIJO+lCQfRkMMGNiUdCKWiO/8yd1hdjBQKzZbXkO6c1OtuS4Pf76RD50P2Iuj13sp6lDS6AiacOtW7sEBmMo9OQw46KemCyloVopDap9pAQhV8rSZ1tcyn15gHBpzHGRjwH5/a4Sif99++XXRmZbuwVM3WfoLUr6iMLk0sicWGU+tt+l5LusjpTpat1BW2iaN5RhT8usMx8cACmekKwtQrtgvKdnvmepmGW6X76CZSL3NuWIGsRu0mXaNaCGe0sGr21zV48DrHauS5/ZzofqskIqKDXhNZKVupVqdLQ0O2F1gB/KvOMdFhgcz0FHfcsSYAq7mEV/vGMOg04fe+15mTrWpNOlPjtm3ufVerQmzcGG7BoFYSa7eohRfq9/azCKjpS16CVP2uGzc6j6dq/0lpyEH6KKuCe3Q0PXNx2r23dYsYbnARPyyQmWtk2Y8cBJ1WateUo3yvatUKaJL+Q13NbL2G7D/hq/splYILoDAVpqKULSU3wVpgbUgXia/bb6uo7HYjooOiM0WrpmCnhuwfad/rsIbcGVggM1d9Q81rE3A3Tjo6X3G7FcF0+5yZcVdf0mm+Qnj325WCSjXf6qp4tQtpg1c8f1eqMOb+3dWCGkGKWdjN+141pOOu290O6iJCp33qgs36VTNkH3LysEDuc0gLiK+SVlq06yvWoavyNDnZXu6tLsVHPtQ6y3HQqpKW1/tq5bGwEc/qubN/3ktjT8NCE8QU3G55UIYJClfq6nOWl4FLl5yVtIaHo1XSSpO5ORq3pFIBjhyJ/zj33AN85StUbQygv1/5SrDPmibwta8JNBr69wcH4xmjnWefBWCrpEXPLV591fk+PQfW1pzbqc/D8tJLVNEMoCpnjzzirHbWqSpsKktLwHPP+VedK5ep6pasvmUfN8PECQvkPscuyAYHgfn57px07JNmvR6PMD5wwCkkKxWauMtlqzzisWPBz9XyMrC+bni+f889rfexdy8wMABs3x5MaL34ImAv/UnPLf7qr5zv03PgU59ybqc+b4WuPKTfeGlhSP+rpVuzgG4RwTCxk7aKzqRPJ0yF3RowFkeVMfl58ue6fdJBo2cputr6vK7JhUqrsqeqH3tgwPIRthvZG8RHK7H7soFmT5WJTDtCmukeWCAziVOtWoKon3xwFMlLKVgyd1ftBhTmXKi5vkF9/X5lNp3Rxv6R41EII4wqlXURtptY1vErhMIwKoYQQrTWo5lexTTJPDg3l4w5zjSBj31MOEy1i4tk/utlTBPYv1/Abg6emgJ+9KPo57xQANbXna/V6+3/brUa8I1vABcuOF+fnwe++9329h2Gm24Cnn/eer5nD/l3uxn1O42OAn/8x2z6ZvSwD7mPqdWAO+8UiQbSqH7TqK0Xuw1d20kZMGX3R9ZqNGnLgCc7pgkcPGj9Lnfc4Xx/dtY9sfvtz4ulJeBXfiX49kmhfpdeEFrO7yBw7lxng9aYLiNtFZ1JB50/M4lUp7S75bRLVB9ykHKSreo668z8rc3P0cyjar3wtHKEe9HfWq26K7ep91q3xlgw8cICuU/pZBebbp1sSKiuRT4/rapOedV1jrpY2rRJv78w4+XCD8ngl8vsVUSF6T/Yh9ynyLzPS5eAfB740pf8czH7keuuA95803o+MwM880x8+7/+euDMGev52Bjw2mtkpj582Ho9nweeeMLfhEs+a+drcn9MNvCKHdi3D3jqKet53NcZ0z2wD7lPseftPvEEC2OVhQXgzTeda1W78IyDz3xG/9yeG14o0GKplT9Vl7er7p9Jl6C5zP/wD+xj7ldYQ2YYDVu3AufOOV+rVuNfuNRqNPmWy859h43EXlgAjh61orqnp4EXXoi2L6azmCYF7DWb1mv9kInAuGGBzDAaSMBZz+0CLot4pQzVasBDDwkIYaBQcFcWq9WAb30L2LaNSoCGEdgs6OOjVgO+/nWg0SDrSDdWy2PahwUyw3iwsAA8/TRw663J1MWOk1oNOHTIel6tAr/0S8Dtt5Mwltj9k/QZS6vWCWwvTBNYWBBYXTUwOAg8/jgLkHbhBQ7DAplhegTV/K0GhwHOQC9VqwaCm0o5EIlh4oeDuhimR1A7F83NAYbSy2LnTuv/6WnA3ukpl4tetOXZZzkQiWHahQUyw3QYtQJXUpTLwAMPkKAF3K0ib7gBsFcTu+224KbSAwes/QIUkJS1Dk0M022wQGaYDkL538mWK7WztAQcP65vFWlPrxoe1rdM9KJcBr78ZasvdDf20GaYrME+ZIbpIKpfN+30lnYDiTgQiWHio5D2ABimn9i8GSC/rWF7nh7lcnuCtN3PMwxjwSZrhukg588Ddr8tPWcYhmGBzDAdRfXbst+VYRgJ+5AZpsOw35VhGB0skBmG6Wl4AcR0CyyQGYbpWUwTuOsuYGUFXOKTyTzsQ2YYpmd59FESxgD9ffTRdMfDMH6wQGYYhmGYDMACmWGYnuXAAaBYpP+LxXDVyBim07APmWGYnoaDuphugQUywzAMw2QANlkzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwXBx6OgAAANFJREFUDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGYAFMsMwDMNkABbIDMMwDJMBWCAzDMMwTAZggcwwDMMwGeD/B1cxd6sBVy3qAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nx.draw(co_authors_net, node_color='blue',node_size=10)\n",
    "plt.savefig(\"graph.png\", dpi=1000)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference Authors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'first': 'Joseph', 'middle': ['C'], 'last': 'Ward'}\n",
      "{'first': 'Lidia', 'middle': [], 'last': 'Lasecka-Dykes'}\n",
      "{'first': 'Chris', 'middle': [], 'last': 'Neil'}\n",
      "{'first': 'Oluwapelumi', 'middle': [], 'last': 'Adeyemi'}\n",
      "{'first': 'Sarah', 'middle': [], 'last': ''}\n",
      "{'first': '', 'middle': [], 'last': 'Gold'}\n",
      "{'first': 'Niall', 'middle': [], 'last': 'Mclean'}\n",
      "{'first': 'Caroline', 'middle': [], 'last': 'Wright'}\n",
      "{'first': 'Morgan', 'middle': ['R'], 'last': 'Herod'}\n",
      "{'first': 'David', 'middle': [], 'last': 'Kealy'}\n",
      "{'first': 'Emma', 'middle': [], 'last': ''}\n",
      "{'first': 'Warner', 'middle': [], 'last': ''}\n",
      "{'first': 'Donald', 'middle': ['P'], 'last': 'King'}\n",
      "{'first': 'Tobias', 'middle': ['J'], 'last': 'Tuthill'}\n",
      "{'first': 'David', 'middle': ['J'], 'last': 'Rowlands'}\n",
      "{'first': 'Nicola', 'middle': ['J'], 'last': ''}\n",
      "{'first': 'Stonehouse', 'middle': [], 'last': 'A#'}\n",
      "{'first': 'T', 'middle': [], 'last': 'Jackson', 'suffix': ''}\n",
      "{'first': 'T', 'middle': ['J'], 'last': 'Tuthill', 'suffix': ''}\n",
      "{'first': 'D', 'middle': ['J'], 'last': 'Rowlands', 'suffix': ''}\n",
      "{'first': 'N', 'middle': ['J'], 'last': 'Stonehouse', 'suffix': ''}\n",
      "{'first': 'N', 'middle': ['D'], 'last': 'Sanderson', 'suffix': ''}\n",
      "{'first': 'N', 'middle': ['J'], 'last': 'Knowles', 'suffix': ''}\n",
      "{'first': 'D', 'middle': ['P'], 'last': 'King', 'suffix': ''}\n",
      "{'first': 'E', 'middle': ['M'], 'last': 'Cottam', 'suffix': ''}\n",
      "{'first': 'A', 'middle': [], 'last': 'Acevedo', 'suffix': ''}\n",
      "{'first': 'R', 'middle': [], 'last': 'Andino', 'suffix': ''}\n",
      "{'first': 'Y', 'middle': [], 'last': 'Peng', 'suffix': ''}\n",
      "{'first': 'Hcm', 'middle': [], 'last': 'Leung', 'suffix': ''}\n",
      "{'first': 'S', 'middle': ['M'], 'last': 'Yiu', 'suffix': ''}\n",
      "{'first': 'Fyl', 'middle': [], 'last': 'Chin', 'suffix': ''}\n",
      "{'first': 'S', 'middle': ['F'], 'last': 'Altschul', 'suffix': ''}\n",
      "{'first': 'W', 'middle': [], 'last': 'Gish', 'suffix': ''}\n",
      "{'first': 'W', 'middle': [], 'last': 'Miller', 'suffix': ''}\n",
      "{'first': 'E', 'middle': ['W'], 'last': 'Myers', 'suffix': ''}\n",
      "{'first': 'D', 'middle': ['J'], 'last': 'Lipman', 'suffix': ''}\n",
      "{'first': 'E', 'middle': [], 'last': 'Rieder', 'suffix': ''}\n",
      "{'first': 'T', 'middle': [], 'last': 'Bunch', 'suffix': ''}\n",
      "{'first': 'F', 'middle': [], 'last': 'Brown', 'suffix': ''}\n",
      "{'first': 'P', 'middle': ['W'], 'last': 'Mason', 'suffix': ''}\n",
      "{'first': 'N', 'middle': ['J'], 'last': 'Stonehouse', 'suffix': ''}\n",
      "{'first': 'L', 'middle': [], 'last': 'Martin', 'suffix': ''}\n",
      "{'first': 'G', 'middle': [], 'last': 'Duke', 'suffix': ''}\n",
      "{'first': 'J', 'middle': [], 'last': 'Osorio', 'suffix': ''}\n",
      "{'first': 'D', 'middle': [], 'last': 'Hall', 'suffix': ''}\n",
      "{'first': 'A', 'middle': [], 'last': 'Palmenberg', 'suffix': ''}\n",
      "{'first': '3d', 'middle': [], 'last': 'Wt', 'suffix': ''}\n",
      "------------------------------------------------------------\n",
      "{'first': 'Hanchu', 'middle': [], 'last': 'Zhou'}\n",
      "{'first': 'Jiannan', 'middle': [], 'last': 'Yang'}\n",
      "{'first': 'Kaicheng', 'middle': [], 'last': 'Tang'}\n",
      "{'first': '†', 'middle': [], 'last': ''}\n",
      "{'first': 'Qingpeng', 'middle': [], 'last': 'Zhang'}\n",
      "{'first': 'Zhidong', 'middle': [], 'last': 'Cao'}\n",
      "{'first': 'Dirk', 'middle': [], 'last': 'Pfeiffer'}\n",
      "{'first': 'Daniel', 'middle': ['Dajun'], 'last': 'Zeng'}\n",
      "{'first': 'C', 'middle': [], 'last': 'Wang', 'suffix': ''}\n",
      "{'first': 'P', 'middle': ['W'], 'last': 'Horby', 'suffix': ''}\n",
      "{'first': 'F', 'middle': ['G'], 'last': 'Hayden', 'suffix': ''}\n",
      "{'first': 'G', 'middle': ['F'], 'last': 'Gao', 'suffix': ''}\n",
      "------------------------------------------------------------\n",
      "{'first': 'Salman', 'middle': ['L'], 'last': 'Butt'}\n",
      "{'first': 'Eric', 'middle': ['C'], 'last': 'Erwood'}\n",
      "{'first': 'Jian', 'middle': [], 'last': 'Zhang'}\n",
      "{'first': 'Holly', 'middle': ['S'], 'last': 'Sellers'}\n",
      "{'first': 'Kelsey', 'middle': [], 'last': 'Young'}\n",
      "{'first': 'Kevin', 'middle': ['K'], 'last': 'Lahmers'}\n",
      "{'first': 'James', 'middle': ['B'], 'last': 'Stanton'}\n",
      "{'first': 'S', 'middle': ['H'], 'last': 'Abro', 'suffix': ''}\n",
      "{'first': 'Y', 'middle': ['A'], 'last': 'Bochkov', 'suffix': ''}\n",
      "{'first': 'S', 'middle': ['L'], 'last': 'Butt', 'suffix': ''}\n",
      "{'first': 'S', 'middle': ['A'], 'last': 'Callison', 'suffix': ''}\n",
      "{'first': 'P', 'middle': [], 'last': 'De Herdt', 'suffix': ''}\n",
      "{'first': 'S', 'middle': [], 'last': 'Escutenaire', 'suffix': ''}\n",
      "{'first': '', 'middle': [], 'last': 'Sybr', 'suffix': ''}\n",
      "{'first': 'H', 'middle': [], 'last': 'Ferreira', 'suffix': ''}\n",
      "{'first': 'T', 'middle': [], 'last': 'Hodgson', 'suffix': ''}\n",
      "{'first': 'M', 'middle': ['W'], 'last': 'Jackwood', 'suffix': ''}\n",
      "{'first': 'M', 'middle': ['W'], 'last': 'Jackwood', 'suffix': ''}\n",
      "{'first': 'M', 'middle': ['W'], 'last': 'Jackwood', 'suffix': ''}\n",
      "{'first': 'M', 'middle': [], 'last': 'Jain', 'suffix': ''}\n",
      "{'first': 'M', 'middle': ['A'], 'last': 'Johnson', 'suffix': ''}\n",
      "{'first': 'N', 'middle': ['M'], 'last': 'Kamble', 'suffix': ''}\n",
      "{'first': 'C', 'middle': ['L'], 'last': 'Keeler', 'suffix': ''}\n",
      "{'first': 'Jr', 'middle': [], 'last': '', 'suffix': ''}\n",
      "{'first': 'D', 'middle': [], 'last': 'Kim', 'suffix': ''}\n",
      "{'first': 'B', 'middle': [], 'last': 'Kingham', 'suffix': ''}\n",
      "{'first': 'E', 'middle': ['T'], 'last': 'Mckinley', 'suffix': ''}\n",
      "{'first': 'M', 'middle': ['M'], 'last': 'Naguib', 'suffix': ''}\n",
      "{'first': 'C', 'middle': ['H'], 'last': 'Okino', 'suffix': ''}\n",
      "{'first': 'T', 'middle': [], 'last': 'Pohuang', 'suffix': ''}\n",
      "{'first': 'J', 'middle': [], 'last': 'Quick', 'suffix': ''}\n",
      "{'first': 'J', 'middle': [], 'last': 'Quick', 'suffix': ''}\n",
      "{'first': 'H-J', 'middle': [], 'last': 'Roh', 'suffix': ''}\n",
      "{'first': 'P', 'middle': ['D'], 'last': 'Schloss', 'suffix': ''}\n",
      "{'first': 'W', 'middle': [], 'last': 'Spaan', 'suffix': ''}\n",
      "{'first': 'S', 'middle': ['J'], 'last': 'Spatz', 'suffix': ''}\n",
      "{'first': 'T', 'middle': [], 'last': 'Stenzel', 'suffix': ''}\n",
      "{'first': 'S', 'middle': [], 'last': 'Sutou', 'suffix': ''}\n",
      "{'first': 'K', 'middle': [], 'last': 'Tamura', 'suffix': ''}\n",
      "{'first': 'Z', 'middle': [], 'last': 'Tarnagda', 'suffix': ''}\n",
      "{'first': 'V', 'middle': [], 'last': 'Valastro', 'suffix': ''}\n",
      "{'first': 'C-H', 'middle': [], 'last': 'Wang', 'suffix': ''}\n",
      "{'first': 'J', 'middle': [], 'last': 'Wang', 'suffix': ''}\n",
      "{'first': 'S', 'middle': [], 'last': 'Wei', 'suffix': ''}\n",
      "{'first': 'I', 'middle': ['A'], 'last': 'Wickramasinghe', 'suffix': ''}\n",
      "{'first': 'A', 'middle': ['K'], 'last': 'Williams', 'suffix': ''}\n",
      "{'first': 'D', 'middle': ['E'], 'last': 'Wood', 'suffix': ''}\n",
      "{'first': 'J', 'middle': [], 'last': 'Ye', 'suffix': ''}\n",
      "------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "for i in range(3):\n",
    "    for author in all_papers_data[i]['authors']:\n",
    "        print(author)\n",
    "    # referenced authors\n",
    "    for ref in all_papers_data[i]['refs']:\n",
    "        for author in ref['authors']:\n",
    "            print(author)\n",
    "    print('-'*60)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extracting Key Words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [{'start': 351,\n",
       "    'end': 360,\n",
       "    'text': 'figure 1A',\n",
       "    'ref_id': 'FIGREF50'}],\n",
       "  'section': ''},\n",
       " {'text': 'The FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).',\n",
       "  'cite_spans': [{'start': 469, 'end': 471, 'text': '87', 'ref_id': None},\n",
       "   {'start': 662, 'end': 664, 'text': '89', 'ref_id': None},\n",
       "   {'start': 857, 'end': 861, 'text': '(19,', 'ref_id': None},\n",
       "   {'start': 862, 'end': 865, 'text': '20)', 'ref_id': None},\n",
       "   {'start': 1117, 'end': 1120, 'text': '104', 'ref_id': None},\n",
       "   {'start': 1637, 'end': 1640, 'text': '117', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': '70'},\n",
       " {'text': 'To introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '120'},\n",
       " {'text': '132 133 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '120'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '120'},\n",
       " {'text': 'Mutations were then introduced into the replicon plasmid as described above.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '135'},\n",
       " {'text': 'To assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).',\n",
       "  'cite_spans': [{'start': 208, 'end': 211, 'text': '143', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': '136'},\n",
       " {'text': 'Transcription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [{'start': 368, 'end': 371, 'text': '225', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': '144'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '144'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '144'},\n",
       " {'text': 'NMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint',\n",
       "  'cite_spans': [{'start': 187, 'end': 190, 'text': '303', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': '301'},\n",
       " {'text': 'orientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.',\n",
       "  'cite_spans': [{'start': 13, 'end': 16, 'text': '351', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': 'Function of the PKs in replication is dependent on downstream interactions and 350'},\n",
       " {'text': 'Changing the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '368'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.',\n",
       "  'cite_spans': [{'start': 920, 'end': 923, 'text': '456', 'ref_id': None},\n",
       "   {'start': 1022, 'end': 1025, 'text': '457', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': '368'},\n",
       " {'text': 'Moreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '468'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).',\n",
       "  'cite_spans': [{'start': 443, 'end': 446, 'text': '475', 'ref_id': None}],\n",
       "  'ref_spans': [],\n",
       "  'section': '468'},\n",
       " {'text': 'The presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '479'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 < 0.05, **** P < 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '479'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '479'},\n",
       " {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint ',\n",
       "  'cite_spans': [],\n",
       "  'ref_spans': [],\n",
       "  'section': '479'}]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper_json['body_text']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "word count: 194 22 Text word count: 5168 23 24 25 .    27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38 for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 .    VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ). The FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 .   can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 .   Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10). To introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF. 132 133 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector. Mutations were then introduced into the replicon plasmid as described above. To assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28). Transcription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The NMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4.  . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint orientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences. Changing the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more.  . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication. Moreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39). The presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation.  . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint  . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 < 0.05, **** P < 0.0001. 674 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 .    . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \n"
     ]
    }
   ],
   "source": [
    "stop_sentences = ['All rights reserved.','No reuse allowed without permission.','Abstract','author/funder','The copyright holder for this preprint (which was not peer-reviewed) is the']\n",
    "abstract_text = extract_abstract(paper_json['abstract'])\n",
    "body_text = ''\n",
    "for t in paper_json['body_text']:\n",
    "    body_text += ' ' + t['text']\n",
    "total_text = abstract_text + ' ' + body_text\n",
    "\n",
    "for s in stop_sentences:\n",
    "    total_text = total_text.replace(s,\"\")\n",
    "\n",
    "print(total_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "stop_words = set(stopwords.words('english'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "punctuation = [',','.',';',':','(',')','′','~']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "positive\n",
      "stranded\n",
      "RNA\n",
      "genomes\n",
      "picornaviruses\n",
      "comprise\n",
      "single\n",
      "large\n",
      "open\n",
      "reading\n",
      "frame\n",
      "flanked\n",
      "untranslated\n",
      "regions\n",
      "UTRs\n",
      "Foot-and-mouth\n",
      "disease\n",
      "virus\n",
      "FMDV\n",
      "unusually\n",
      "large\n",
      "UTR\n",
      "kb\n",
      "containing\n",
      "five\n",
      "structural\n",
      "domains\n",
      "include\n",
      "internal\n",
      "ribosome\n",
      "entry\n",
      "site\n",
      "IRES\n",
      "facilitates\n",
      "initiation\n",
      "translation\n",
      "cis-acting\n",
      "replication\n",
      "element\n",
      "cre\n",
      "Less\n",
      "well\n",
      "characterised\n",
      "structures\n",
      "terminal\n",
      "nucleotide\n",
      "stem-loop\n",
      "variable\n",
      "length\n",
      "poly-C-tract\n",
      "approximately\n",
      "100-200\n",
      "nucleotides\n",
      "series\n",
      "four\n",
      "tandemly\n",
      "repeated\n",
      "pseudoknots\n",
      "PKs\n",
      "investigated\n",
      "structures\n",
      "PKs\n",
      "selective\n",
      "hydroxyl\n",
      "acetylation\n",
      "analysed\n",
      "primer\n",
      "extension\n",
      "SHAPE\n",
      "analysis\n",
      "determined\n",
      "contribution\n",
      "genome\n",
      "replication\n",
      "mutation\n",
      "deletion\n",
      "experiments\n",
      "SHAPE\n",
      "mutation\n",
      "experiments\n",
      "confirmed\n",
      "importance\n",
      "previously\n",
      "predicted\n",
      "PK\n",
      "structures\n",
      "function\n",
      "Deletion\n",
      "experiments\n",
      "showed\n",
      "although\n",
      "PKs\n",
      "essential\n",
      "replication\n",
      "provide\n",
      "genomes\n",
      "competitive\n",
      "advantage\n",
      "However\n",
      "although\n",
      "replicons\n",
      "full-length\n",
      "genomes\n",
      "lacking\n",
      "PKs\n",
      "replication\n",
      "competent\n",
      "infectious\n",
      "virus\n",
      "rescued\n",
      "genomes\n",
      "containing\n",
      "less\n",
      "PK\n",
      "copy\n",
      "consistent\n",
      "earlier\n",
      "report\n",
      "describing\n",
      "presence\n",
      "putative\n",
      "packaging\n",
      "signals\n",
      "PK\n",
      "region\n",
      "VP3\n",
      "VP0\n",
      "processed\n",
      "VP2\n",
      "VP4\n",
      "virus\n",
      "assembly\n",
      "P2\n",
      "P3\n",
      "regions\n",
      "encode\n",
      "non-structural\n",
      "proteins\n",
      "2B\n",
      "2C\n",
      "3A\n",
      "3B\n",
      "1-3\n",
      "VPg\n",
      "3C\n",
      "pro\n",
      "structural\n",
      "protein-coding\n",
      "region\n",
      "replaced\n",
      "reporter\n",
      "genes\n",
      "allow\n",
      "study\n",
      "genome\n",
      "replication\n",
      "without\n",
      "requirement\n",
      "high\n",
      "containment\n",
      "figure\n",
      "1A\n",
      "FMDV\n",
      "UTR\n",
      "largest\n",
      "known\n",
      "picornavirus\n",
      "UTR\n",
      "comprising\n",
      "approximately\n",
      "nucleotides\n",
      "containing\n",
      "several\n",
      "highly\n",
      "structured\n",
      "regions\n",
      "first\n",
      "nucleotides\n",
      "end\n",
      "predicted\n",
      "fold\n",
      "single\n",
      "large\n",
      "stem\n",
      "loop\n",
      "termed\n",
      "S-fragment\n",
      "followed\n",
      "PKs\n",
      "originally\n",
      "predicted\n",
      "consist\n",
      "four\n",
      "tandem\n",
      "repeats\n",
      "nucleotide\n",
      "region\n",
      "containing\n",
      "small\n",
      "stem\n",
      "loop\n",
      "downstream\n",
      "interaction\n",
      "site\n",
      "figure\n",
      "1B\n",
      "Due\n",
      "sequence\n",
      "similarity\n",
      "PKs\n",
      "figure\n",
      "1C\n",
      "speculated\n",
      "formed\n",
      "duplication\n",
      "events\n",
      "viral\n",
      "replication\n",
      "probably\n",
      "involving\n",
      "recombination\n",
      "four\n",
      "PKs\n",
      "present\n",
      "different\n",
      "virus\n",
      "isolates\n",
      "strain\n",
      "identified\n",
      "less\n",
      "PKs\n",
      "emphasising\n",
      "potential\n",
      "importance\n",
      "viral\n",
      "life\n",
      "cycle\n",
      "presence\n",
      "PKs\n",
      "reported\n",
      "UTR\n",
      "picornaviruses\n",
      "occur\n",
      "absence\n",
      "PKs\n",
      "least\n",
      "required\n",
      "wild-type\n",
      "wt\n",
      "replication\n",
      "competition\n",
      "experiments\n",
      "showed\n",
      "extra\n",
      "copies\n",
      "PKs\n",
      "conferred\n",
      "replicative\n",
      "advantage\n",
      "genomes\n",
      "Although\n",
      "replicons\n",
      "full-length\n",
      "genomes\n",
      "lacking\n",
      "PKs\n",
      "replication-competent\n",
      "infectious\n",
      "virus\n",
      "rescued\n",
      "genomes\n",
      "containing\n",
      "less\n",
      "PK\n",
      "copy\n",
      "consistent\n",
      "earlier\n",
      "report\n",
      "describing\n",
      "presence\n",
      "putative\n",
      "packaging\n",
      "signals\n",
      "PK\n",
      "region\n",
      "Plasmid\n",
      "construction\n",
      "FMDV\n",
      "replicon\n",
      "plasmids\n",
      "pRep-ptGFP\n",
      "replication-defective\n",
      "polymerase\n",
      "mutant\n",
      "control\n",
      "3D-GNN\n",
      "already\n",
      "described\n",
      "introduce\n",
      "mutations\n",
      "PK\n",
      "region\n",
      "pRep-ptGFP\n",
      "replicon\n",
      "plasmid\n",
      "digested\n",
      "SpeI\n",
      "KpnI\n",
      "resulting\n",
      "fragment\n",
      "inserted\n",
      "sub-cloning\n",
      "vector\n",
      "pBluescript\n",
      "create\n",
      "pBluescript\n",
      "PK\n",
      "PKs\n",
      "removed\n",
      "digestion\n",
      "HindIII\n",
      "AatII\n",
      "insertion\n",
      "synthetic\n",
      "DNA\n",
      "sequence\n",
      "PK\n",
      "deleted\n",
      "PKs\n",
      "deleted\n",
      "PCR\n",
      "amplification\n",
      "using\n",
      "ΔPK\n",
      "Forward\n",
      "primer\n",
      "FMDV\n",
      "1331-1311\n",
      "reverse\n",
      "primer\n",
      "resultant\n",
      "product\n",
      "digested\n",
      "HindIII\n",
      "AatII\n",
      "ligated\n",
      "pBluescript\n",
      "PK\n",
      "vector\n",
      "Complete\n",
      "PK\n",
      "deletion\n",
      "achieved\n",
      "introduction\n",
      "AflII\n",
      "site\n",
      "end\n",
      "poly-C\n",
      "tract\n",
      "PCR\n",
      "mutagenesis\n",
      "create\n",
      "sub-cloning\n",
      "vector\n",
      "pBluescript\n",
      "C11\n",
      "used\n",
      "remove\n",
      "PKs\n",
      "PCR\n",
      "mutagenesis\n",
      "using\n",
      "ΔPK\n",
      "forward\n",
      "primer\n",
      "FMDV\n",
      "1331-1311\n",
      "reverse\n",
      "primer\n",
      "modified\n",
      "PK\n",
      "sequences\n",
      "removed\n",
      "sub-cloning\n",
      "vectors\n",
      "inserted\n",
      "pRep-ptGFP\n",
      "plasmid\n",
      "using\n",
      "NheI-HF\n",
      "KpnI-131\n",
      "HF\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "Mutations\n",
      "disrupt\n",
      "reform\n",
      "PK\n",
      "structure\n",
      "introduced\n",
      "using\n",
      "synthetic\n",
      "DNA\n",
      "digestion\n",
      "AflII\n",
      "AatII\n",
      "ligation\n",
      "similarly\n",
      "digested\n",
      "pBluescript\n",
      "PK\n",
      "vector\n",
      "Mutations\n",
      "introduced\n",
      "replicon\n",
      "plasmid\n",
      "described\n",
      "assess\n",
      "effects\n",
      "truncation\n",
      "poly-C-tract\n",
      "replication\n",
      "entire\n",
      "sequence\n",
      "removed\n",
      "performed\n",
      "PCR\n",
      "mutagenesis\n",
      "using\n",
      "primers\n",
      "C0\n",
      "SpeI\n",
      "FMDV\n",
      "1331-\n",
      "vitro\n",
      "transcription\n",
      "vitro\n",
      "transcription\n",
      "reactions\n",
      "replicon\n",
      "assays\n",
      "performed\n",
      "described\n",
      "previously\n",
      "Transcription\n",
      "reactions\n",
      "produce\n",
      "large\n",
      "amounts\n",
      "RNA\n",
      "SHAPE\n",
      "analysis\n",
      "performed\n",
      "purified\n",
      "linear\n",
      "DNA\n",
      "described\n",
      "μg\n",
      "linearised\n",
      "DNA\n",
      "used\n",
      "HiScribe\n",
      "T7\n",
      "synthesis\n",
      "kit\n",
      "NEB\n",
      "DNase\n",
      "treatment\n",
      "purification\n",
      "using\n",
      "PureLink\n",
      "FastQ\n",
      "files\n",
      "quality\n",
      "checked\n",
      "using\n",
      "FastQC\n",
      "poor\n",
      "quality\n",
      "reads\n",
      "filtered\n",
      "using\n",
      "Sickle\n",
      "algorithm\n",
      "Host\n",
      "cell\n",
      "reads\n",
      "removed\n",
      "using\n",
      "FastQ\n",
      "Screen\n",
      "algorithm\n",
      "FMDV\n",
      "reads\n",
      "assembled\n",
      "de\n",
      "novo\n",
      "contigs\n",
      "using\n",
      "IDBA-UD\n",
      "Contigs\n",
      "matched\n",
      "FMDV\n",
      "library\n",
      "identified\n",
      "using\n",
      "Basic\n",
      "Local\n",
      "ALighnment\n",
      "Search\n",
      "Tool\n",
      "BLAST\n",
      "assembled\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "consensus\n",
      "sequences\n",
      "using\n",
      "SeqMan\n",
      "Pro\n",
      "software\n",
      "DNA\n",
      "STAR\n",
      "Lasergene\n",
      "package\n",
      "DNA\n",
      "STAR\n",
      "SHAPE\n",
      "data\n",
      "largely\n",
      "agreed\n",
      "predicted\n",
      "structures\n",
      "stems\n",
      "PK\n",
      "interacting\n",
      "nucleotides\n",
      "showed\n",
      "little\n",
      "reactivity\n",
      "suggesting\n",
      "NMIA\n",
      "could\n",
      "interact\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "nucleotides\n",
      "either\n",
      "due\n",
      "predicted\n",
      "base\n",
      "pairing\n",
      "steric\n",
      "hindrance\n",
      "figure\n",
      "2B\n",
      "NMIA\n",
      "reactivity\n",
      "interacting\n",
      "nucleotides\n",
      "stem-loops\n",
      "downstream\n",
      "residues\n",
      "PK\n",
      "largely\n",
      "agreed\n",
      "predicted\n",
      "structure\n",
      "although\n",
      "SHAPE\n",
      "data\n",
      "suggests\n",
      "might\n",
      "fewer\n",
      "interactions\n",
      "previously\n",
      "predicted\n",
      "However\n",
      "differences\n",
      "could\n",
      "due\n",
      "heterogeneity\n",
      "formation\n",
      "PKs\n",
      "experiment\n",
      "evidence\n",
      "loop-downstream\n",
      "interaction\n",
      "weaker\n",
      "PK4\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "orientation\n",
      "Since\n",
      "removal\n",
      "four\n",
      "PKs\n",
      "resulted\n",
      "significant\n",
      "decrease\n",
      "replication\n",
      "minimal\n",
      "requirements\n",
      "maintain\n",
      "wt\n",
      "levels\n",
      "replication\n",
      "investigated\n",
      "near\n",
      "wt\n",
      "level\n",
      "replication\n",
      "observed\n",
      "PK\n",
      "present\n",
      "mutagenesis\n",
      "performed\n",
      "C11\n",
      "replicon\n",
      "plasmid\n",
      "containing\n",
      "PK\n",
      "addition\n",
      "orientation\n",
      "PK\n",
      "reversed\n",
      "\"flipping\"\n",
      "nucleotide\n",
      "sequence\n",
      "potentially\n",
      "facilitate\n",
      "hybridisation\n",
      "loop\n",
      "upstream\n",
      "rather\n",
      "downstream\n",
      "sequences\n",
      "Changing\n",
      "orientation\n",
      "PK\n",
      "reduced\n",
      "replicon\n",
      "replication\n",
      "similar\n",
      "level\n",
      "seen\n",
      "replication\n",
      "decreased\n",
      "passage\n",
      "fold\n",
      "reduction\n",
      "compared\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "passage\n",
      "figure\n",
      "5B\n",
      "Therefore\n",
      "appears\n",
      "replicons\n",
      "single\n",
      "PK\n",
      "competitive\n",
      "disadvantage\n",
      "compared\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "infectious\n",
      "virus\n",
      "despite\n",
      "able\n",
      "replicate\n",
      "transfection\n",
      "cells\n",
      "consistent\n",
      "requirement\n",
      "RNA\n",
      "structure\n",
      "within\n",
      "PK\n",
      "region\n",
      "required\n",
      "virus\n",
      "assembly\n",
      "UTR\n",
      "FMDV\n",
      "unique\n",
      "amongst\n",
      "picornaviruses\n",
      "due\n",
      "large\n",
      "size\n",
      "presence\n",
      "multiple\n",
      "RNA\n",
      "elements\n",
      "still\n",
      "unknown\n",
      "function\n",
      "features\n",
      "series\n",
      "repeated\n",
      "PKs\n",
      "varying\n",
      "number\n",
      "2-4\n",
      "depending\n",
      "virus\n",
      "strain\n",
      "study\n",
      "sequentially\n",
      "deleted\n",
      "mutated\n",
      "PKs\n",
      "help\n",
      "understand\n",
      "role\n",
      "viral\n",
      "life\n",
      "cycle\n",
      "also\n",
      "confirmed\n",
      "predicted\n",
      "PK\n",
      "structures\n",
      "SHAPE\n",
      "mapping\n",
      "although\n",
      "may\n",
      "Although\n",
      "viruses\n",
      "isolated\n",
      "date\n",
      "contain\n",
      "least\n",
      "PKs\n",
      "replicons\n",
      "viruses\n",
      "containing\n",
      "single\n",
      "PK\n",
      "still\n",
      "replication\n",
      "competent\n",
      "However\n",
      "replicons\n",
      "single\n",
      "PK\n",
      "found\n",
      "competitive\n",
      "advantage\n",
      "replicons\n",
      "single\n",
      "PK\n",
      "sequentially\n",
      "passaged\n",
      "Replicons\n",
      "lacking\n",
      "PKs\n",
      "displayed\n",
      "poor\n",
      "passaging\n",
      "potential\n",
      "even\n",
      "co-467\n",
      "transfected\n",
      "yeast\n",
      "tRNA\n",
      "reinforcing\n",
      "observation\n",
      "significant\n",
      "impact\n",
      "replication\n",
      "Moreover\n",
      "viruses\n",
      "recovered\n",
      "genomes\n",
      "reduced\n",
      "numbers\n",
      "PKs\n",
      "slower\n",
      "growing\n",
      "produced\n",
      "smaller\n",
      "plaques\n",
      "addition\n",
      "differences\n",
      "pronounced\n",
      "PKs\n",
      "functionally\n",
      "competent\n",
      "differences\n",
      "seen\n",
      "replicons\n",
      "congaing\n",
      "single\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "copy\n",
      "PK1\n",
      "PK4\n",
      "observation\n",
      "consistent\n",
      "previous\n",
      "report\n",
      "deletion\n",
      "PK1\n",
      "along\n",
      "Poly-C-tract\n",
      "adverse\n",
      "effect\n",
      "viral\n",
      "replication\n",
      "also\n",
      "supports\n",
      "findings\n",
      "truncation\n",
      "Poly-C-tract\n",
      "create\n",
      "C11\n",
      "construct\n",
      "effect\n",
      "replicon\n",
      "replication\n",
      "cell\n",
      "lines\n",
      "tested\n",
      "described\n",
      "Mengo\n",
      "virus\n",
      "possible\n",
      "role\n",
      "poly-C-tract\n",
      "essential\n",
      "aspects\n",
      "viral\n",
      "lifecycle\n",
      "cannot\n",
      "recapitulated\n",
      "standard\n",
      "tissue\n",
      "culture\n",
      "system\n",
      "presence\n",
      "least\n",
      "PKs\n",
      "viral\n",
      "isolates\n",
      "sequenced\n",
      "far\n",
      "suggests\n",
      "multiple\n",
      "PKs\n",
      "confer\n",
      "competitive\n",
      "advantage\n",
      "replication\n",
      "showed\n",
      "sequential\n",
      "passage\n",
      "replicons\n",
      "containing\n",
      "least\n",
      "PKs\n",
      "maintained\n",
      "level\n",
      "similar\n",
      "wt\n",
      "replicons\n",
      "containing\n",
      "PK\n",
      "showed\n",
      "persistent\n",
      "decline\n",
      "unclear\n",
      "viral\n",
      "isolates\n",
      "contain\n",
      "four\n",
      "PKs\n",
      "still\n",
      "unknown\n",
      "may\n",
      "stochastic\n",
      "variation\n",
      "may\n",
      "reflect\n",
      "subtle\n",
      "effects\n",
      "host\n",
      "range\n",
      "geographical\n",
      "localisation\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "Significance\n",
      "shown\n",
      "comparing\n",
      "replication\n",
      "C11\n",
      "PK\n",
      "disrupt\n",
      "C11\n",
      "PK\n",
      "restore\n",
      "Aii\n",
      "Significance\n",
      "shown\n",
      "compared\n",
      "wt\n",
      "replicon\n",
      "Error\n",
      "bars\n",
      "calculated\n",
      "SEM\n",
      "n\n",
      "=\n",
      "*\n",
      "P\n",
      "<\n",
      "****\n",
      "P\n",
      "<\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n",
      "doi\n",
      "bioRxiv\n",
      "preprint\n"
     ]
    }
   ],
   "source": [
    "#only keep nouns ...\n",
    "other_words = set(['two','one','three','bioRxiv','furthermore','word','count','text'])\n",
    "all_words = []\n",
    "for word in total_text.split(\" \"):\n",
    "    for p in punctuation:\n",
    "        word = word.replace(p,\"\")\n",
    "    word = word.strip(\" \")\n",
    "    try:\n",
    "        int(word)\n",
    "    except:\n",
    "        if (not word.lower() in stop_words) and (word) and (word[:4] != 'http') and (not word.lower() in other_words):\n",
    "            print(word)\n",
    "            all_words.append(word)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'also'.lower() in stop_words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(int('5'))\n",
    "except:\n",
    "    print('5′ is text')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "freq = nltk.FreqDist(all_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAHFCAYAAAD/p/5dAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3XecXGXZ//HPtZu66QVCIA1CLwGSUAKoREA6jyCCqDRBQBARlEfxwZ/9UQER1AcQRUQEBESF0BFJqAE2CYSEEEqABBIIIb236/fHfSaZbGZ3z7Q9M2e+79drXtmZ2fuca7Oz5zp3N3dHRERqV13SAYiISLKUCEREapwSgYhIjVMiEBGpcUoEIiI1TolARKTGKRGIiNQ4JQIRkRqnRCAiUuPaJR1AHH379vUhQ4YUVHbFihV07ty54HOrvMqrvMoXI8kYJkyYMM/dt2j1G9294h8jRozwQjU2NhZcVuVVXuVVvlhJxgA0eoxrrJqGRERqnBKBiEiNUyIQEalxSgQiIjVOiUBEpMYpEYiI1LhUJ4IPFq1k+Zr1SYchIlLRUp0IvvfPV7jgoXn8dfy7rF2nhCAikktqE8GyVWtZsnINi1et5/J/TeHIa5/iielzce3RLCKyidQmgi4d23HXuaP49qieDOrdwBtzl3LmzS9y2p9e4LUPFicdnohIxUhtIgAwM0YN6MRjl3ySy4/ehW6d2vHUG/M46tqnuOwfk5m7ZGXSIYqIJC7ViSCjY7t6zv7Edoy7dDRnHDCEOjPueGEWo68cy+/+8wYr16xLOkQRkcTURCLI6N2lAz88bjceufiTHLpLP5atXsdVj77O6KvG8s9J77F+vfoPRKT21FQiyBi6RVf+ePpIbv/qfuzavztzFq3k4jtf5rPXPcMLb89POjwRkTZVk4kg44ChfRlz4UFceeIw+nXvyOT3FnHS75/jvFsn8M68ZUmHJyLSJqpiY5pyqq8zPj9yIEcP68+NT87g9+Nm8PDUD3j8tQ85bdQQPtlH8w9EJN1qukaQraFDO7556I6MvfRgPj9iAGvXOzc9/TaXPDqPRcvXJB2eiEjZKBE00a97J678/J7cf+FBbNe3Cx+vWM8zb81LOiwRkbJRImjGblv34Ng9twZg4rsLEo5GRKR8lAhaMHxwLwAmzlQiEJH0UiJowV4DewIw5f3FrFqrSWcikk5KBC3o0bk9A7q3Y/W69UydrfWJRCSdlAhasVOf9oD6CUQkvZQIWrFjlAgmzVyYcCQiIuWhRNCKnfp0ANRhLCLppUTQim261dO9UzvmLFrJ7IUrkg5HRKTklAhaUWfGXoM0jFRE0kuJIIbhg8Iw0onvqp9ARNJHiSCG4aoRiEiKKRHEsNegnpjB1NmLtJuZiKSOEkEM3Tu1Z4ctu7JmnTN19qKkwxERKSklgpg2NA+pn0BEUqZsicDMBprZE2Y2zcymmtlF0es/NLP3zeyl6HFUuWIoJfUTiEhalXOHsrXAt9x9opl1AyaY2WPRe79296vKeO6SGz44Gjk0cwHujpklHJGISGmUrUbg7nPcfWL09RJgGrBNuc5Xbtv17Ur3Tu34cPEqZi9amXQ4IiIlY+5e/pOYDQGeBHYHLgHOABYDjYRaw2btLWZ2DnAOQP/+/UeMGTOmoHMvX76choaGgso2Lf/Tp+Yz6YPVXLJ/Dw4c2LnNz6/yKq/y1VU+6RhGjhw5wd1HtvqN7l7WB9AVmACcED3vB9QTaiM/A/7U2jFGjBjhhWpsbCy4bNPy1zz2ug/+zv3+w/umJHJ+lVd5la+u8knHADR6jOt0WUcNmVl74B7gNnf/R5R4PnT3de6+HvgDsG85Yyiljf0EGjkkIulRzlFDBtwETHP3q7Ne75/1bccDU8oVQ6ntNTBMLHtVE8tEJEXKWSM4EDgV+HSToaJXmNkrZjYZGA1cXMYYSqpbp/bs1K8ba9Y5U97XxDIRSYeyDR9196eBXGMsHyzXOdvC3oN68doHS5g4cwEjh/ROOhwRkaJpZnGetBKpiKSNEkGehg/eOMPY22DorYhIuSkR5Gm7vl3o2dCeuUtW8b52LBORFFAiyJOZsfdADSMVkfRQIijAxpVItQCdiFQ/JYICZPcTiIhUOyWCAuw5sCd1Bq/OXqyJZSJS9ZQICtC1Yzt27NeNteudye9pYpmIVDclggKpeUhE0kKJoEDqMBaRtFAiKNCGGcYzF2pimYhUNSWCAm3btwu9Gtozb+kq3lugiWUiUr2UCApkZuytDe1FJAWUCIqwcQE6JQIRqV5KBEXY0GGspSZEpIopERQhM7Fs2pzFrFitiWUiUp2UCIrQpWM7dt6qezSxTLUCEalOSgRF0ob2IlLtlAiKNFwjh0SkyikRFCmTCCZpxzIRqVJKBEUa3KeB3l06MG/pambN18QyEak+SgRFMrOs5SbUPCQi1UeJoAQ0w1hEqpkSQQmow1hEqpkSQQnsObAH9XXGtDlLWL56bdLhiIjkRYmgBBo6tGPnrbqxbr3z8iztWCYi1UWJoETUPCQi1UqJoEQyM4wnKRGISJVRIiiR7JVINbFMRKqJEkGJDOrdQJ8uHZi/bDXvfrw86XBERGJTIigR7VgmItVKiaCENq5EqkQgItVDiaCENvQTvKslqUWkeigRlNCwAWFi2WsfLGbZKk0sE5HqoERQQg0d2rFL/26sd3hZO5aJSJUoWyIws4Fm9oSZTTOzqWZ2UfR6bzN7zMzeiP7tVa4YkjBiw/4ESgQiUh3KWSNYC3zL3XcB9gcuMLNdge8Cj7v7DsDj0fPUGD4400+gDmMRqQ5lSwTuPsfdJ0ZfLwGmAdsA/wXcEn3bLcBnyxVDEjbsWDZLE8tEpDq0SR+BmQ0B9gaeB/q5+xwIyQLYsi1iaCsDenWmb9eOzF+2mnc0sUxEqoCV+67VzLoC44Cfufs/zGyhu/fMen+Bu2/WT2Bm5wDnAPTv33/EmDFjCjr/8uXLaWhoKCz4Asv/8pkFvDB7FRfu04N9t/Q2P7/Kq7zKV0b5pGMYOXLkBHcf2eo3unvZHkB74BHgkqzXpgP9o6/7A9NbO86IESO8UI2NjQWXLbT89WPf9MHfud+/94/JiZxf5VVe5SujfNIxAI0e41pdzlFDBtwETHP3q7Peug84Pfr6dODecsWQlOwF6EREKl27Mh77QOBU4BUzeyl67XvAL4C7zOwsYCbw+TLGkIhhA3rQrs6Y/sFiVqzplHQ4IiItKlsicPenAWvm7UPKdd5K0Kl9Pbtu3Z3J7y3izQVrOCjpgEREWqCZxWWSaR6a/vGahCMREWmZEkGZ7D0oDIyaPk+JQEQqmxJBmWRqBK/PX62JZSJS0ZQIymRAr85s0a0jS1c7b320LOlwRESapURQJmbGvkN6A/D82x8nHI2ISPOUCMpo/6F9AHjuLSUCEalcSgRlNGq7kAjGz5ivfgIRqVh5JwIz62Vmw8oRTNoM3aILPTvVMW/pKt6cuzTpcEREcoqVCMxsrJl1N7PewMvAzWZ2dWvlap2ZsfsWHQB4Vs1DIlKh4tYIerj7YuAE4GZ3HwEcWr6w0mOPLUMiUD+BiFSquImgnZn1B04C7i9jPKmze5QIxr/9MevXq59ARCpP3ETwI8Jy0m+6+4tmth3wRvnCSo9+XerZukcnFi5fw2sfLEk6HBGRzcRNBHPcfZi7nw/g7jMA9RHEYGYbh5HOUPOQiFSeuIngtzFfkxwyw0ife2tewpGIiGyuxWWozWwUcACwhZldkvVWd6C+nIGlyaioRvD82/NZt96pr2tudW4RkbbXWo2gA9CVkDC6ZT0WAyeWN7T0GNCrgUG9G1iyci1TZy9KOhwRkU20WCNw93HAODP7s7u/20YxpdKo7fowc/5ynnvrY4YN6Jl0OCIiG8TtI+hoZjea2aNm9p/Mo6yRpcwodRiLSIWKu1Xl3cANwB+BdeULJ70yieCFt+ezZt162tdrmScRqQxxE8Fad7++rJGkXL/undhuiy7M+GgZk99bxIjBvZIOSUQEiN80NMbMzjez/mbWO/Moa2QptHE1UjUPiUjliJsITgcuBZ4FJkSPxnIFlVajtD+BiFSgWE1D7r5tuQOpBftHNYLGd+ezau06OrbTVAwRSV6sRGBmp+V63d3/Utpw0q1v147s2K8rr3+4lJdmLmS/KDGIiCQpbtPQPlmPTwA/BI4rU0ypdsDQvoCGkYpI5YiVCNz9wqzHV4G9CbOOJU/7b6d+AhGpLIUOZl8O7FDKQGrF/tv1xgwmzVzIyjWakiEiyYvbRzAGyOyqUg/sAtxVrqDSrGdDB3bZqjuvzlnMxHcXcMD2fZMOSURqXNwJZVdlfb0WeNfd3ytDPDVh1NA+vDpnMc++9bESgYgkLm4fwTjgNcLKo72A1eUMKu0O0LpDIlJBYiUCMzsJeAH4PGHf4ufNTMtQF2ifbXtTZ/DyrIUsW7U26XBEpMbF7Sz+H2Afdz/d3U8D9gW+X76w0q17p/bssU0P1q53Gt9dkHQ4IlLj4iaCOnefm/X84zzKSg6ZfYyf1faVIpKwuBfzh83sETM7w8zOAB4AHixfWOmXmVg2XvMJRCRhre1ZvD3Qz90vNbMTgIMAA54DbmuD+FJr5OBetKszXnl/EYtXrqF7p/ZJhyQiNaq1GsE1wBIAd/+Hu1/i7hcTagPXlDu4NOvSsR17DuzJeocX356fdDgiUsNaSwRD3H1y0xfdvREY0lJBM/uTmc01sylZr/3QzN43s5eix1EFRZ0So7TchIhUgNYSQacW3uvcStk/A0fkeP3X7r5X9KjpfoZRGzqMlQhEJDmtJYIXzeyrTV80s7MIm9M0y92fBNTm0YIRg3vRob6OaR8sZuFyzdETkWS0lgi+CZxpZmPN7FfRYxxwNnBRgef8uplNjpqOanrj3k7t69l7UE/cYfwM5UwRSYa5e+vfZDYa2D16OtXd/xPr4GZDgPvdfffoeT9gHmEBu58A/d39K82UPQc4B6B///4jxowZE+eUm1m+fDkNDQ0FlW2L8ndNXcqdry7lqO0bOGvv7m1+fpVXeZUvX/mkYxg5cuQEdx/Z6je6e9kehA7lKfm+1/QxYsQIL1RjY2PBZdui/Pi35vng79zvn7l6XCLnV3mVV/nylU86BqDRY1xj23R2sJn1z3p6PDClue+tFXsN6knHdnVM/3AJ85auSjocEalBZUsEZnYHYeLZTmb2XtTBfIWZvWJmk4HRwMXlOn+16Niunn2G9AZgvFYjFZEExN2PIG/ufkqOl28q1/mq2aihfXj6zXk899bHHDNs66TDEZEao4XjKsCGfYxVIxCRBCgRVIBhA3rQ0KGeGR8t48PFK5MOR0RqjBJBBWhfX7ehn0DLTYhIW1MiqBAbtq9UIhCRNqZEUCFGaR9jEUmIEkGF2G3rHnTr1I6Z85fz/sIVSYcjIjVEiaBC1NcZ+22rfgIRaXtKBBUkM4xU+xiLSFtSIqgg2fsYe4zFAEVESkGJoILsvFU3ejW0Z/ailcycvzzpcESkRigRVJC6OmO/bTWMVETalhJBhdEwUhFpa0oEFSZ7H2P1E4hIW1AiqDA7bNmVvl078NGSVbz10bKkwxGRGqBEUGHMTKuRikibUiKoQJnmofHqMBaRNqBEUIFGZdUI1qufQETKTImgAm3btwv9undk/rLVzFq8NulwRCTllAgqkJltmGU8Ze7qhKMRkbRTIqhQmeYhJQIRKbeybV4vxcl0GL/60Wqee+tjzAo7zusfrWZNEaOPki4/Z9EaRhRcWkTiUCKoUAN7N7BNz868v3AFp/xhfHEHG1vd5btu9SGH7dqvuBhEpFlKBBXssqN25vrHptKla9eCj7F0yRK6dutWleWXrVrL1NmL+dGYqXxih750al9fcBwi0jwlggp2zLCt6b9mDiNGFN44MmHChKotv3bdeg694jHeWbCC68e+xcWH7VhwHCLSPHUWS8VqV1/H2cO7A3D9uLeY+bGW5hYpByUCqWi79O3A8Xtvw+q16/nJA68mHY5IKikRSMW77Mid6dKhnsde/ZAnps9NOhyR1FEikIq3ZfdOfPPQ0D/w4zGvsmrtuoQjEkkXJQKpCmccOITtt+zK2/OWcdPTbycdjkiqKBFIVWhfX8ePjtsNgN8+/iazF65IOCKR9FAikKpx4PZ9OXqP/qxYs46fPTgt6XBEUkOJQKrK947ehc7t63lg8hyefXNe0uGIpIISgVSVbXp25uuf3h6AH9w3lTXr1icckUj1UyKQqnP2J7ZlSJ8G3pi7lFuefSfpcESqnhKBVJ2O7er5wbGh4/iaf7/B3CUrE45IpLopEUhVGr3zlhy6y5YsXbWWXzz4WtLhiFS1siUCM/uTmc01sylZr/U2s8fM7I3o317lOr+k3/eP2ZUO7er4x6T3efGd+UmHI1K1ylkj+DNwRJPXvgs87u47AI9Hz0UKMrhPF8775HYA/L97p7JuvScckUh1KlsicPcngaa3af8F3BJ9fQvw2XKdX2rD1w7enm16dmbanMXc/vy7SYcjUpXauo+gn7vPAYj+3bKNzy8p07lDPd8/ZlcArnxkOh8vXZVwRCLVx9zLV502syHA/e6+e/R8obv3zHp/gbvn7Ccws3OAcwD69+8/YsyYMQXFsHz5choaGgoqq/LVUd7d+clTC3j5w9Ucum1nvjayR5ueX+VVvlJjGDly5AR3H9nqN7p72R7AEGBK1vPpQP/o6/7A9DjHGTFihBeqsbGx4LIqXz3l35y7xLf/3gM+5Lv3+0szF7T5+VVe5SsxBqDRY1xj27pp6D7g9Ojr04F72/j8klJDt+jKVw7aFnf4f/dOYb06jkViK+fw0TuA54CdzOw9MzsL+AVwmJm9ARwWPRcpiQs/vQP9unfk5fcWcfeEWUmHI1I1yjlq6BR37+/u7d19gLvf5O4fu/sh7r5D9K8Gf0vJdO3Yju8dtQsAv3x4OouWr0k4IpHqoJnFkirH7bk1+27bm/nLVnP1Y9OTDkekKigRSKqYGT86bjfq64xbx7/LOwtVKxBpTbukAxAptV36d+fU/Qfz52ff4caJi+m9TeEb3r8xZxWLX6ve8rM+Ws1e6536Oiv4GJJ+SgSSShcftiP3T57N9I9Xc+afXyzuYE9Xd/m/TX+ay4/ehQO271tcHJJaSgSSSj06t+fXJ+/FNQ++TLfu3Qs+zqJFi+jRo0fr31ih5V+Z+TGvzlnMF//4PIfusiXfPXIXtt+ya8HHk3RSIpDU+sQOW9BwUC9GjBhR8DEmTJhQ1eWffaGRSct6ct0Tb/LvaXN5YvpHfHm/QVx06I707tKh4ONKuqizWCTFOtYbF4zenrGXjuaUfQfh7tzy3Lt86sonuPHJt1i1dl3SIUoFUCIQqQFbdOvIz0/Yg4cu+iSf3HELlqxcy/8++BqHXj2OBybPySwBIzVKiUCkhuy0VTf+8pV9+fOZ+7Bjv67Mmr+CC26fyIk3PMekmQuSDk8SokQgUoMO3mlLHvzGJ/jf4/egb9cOTHh3Acdf9yzfuGMS7y1YnnR40saUCERqVLv6Or643yCe+PbBnH/wUDq0q+O+l2fz6V+N45cPv8aSlZqMVys0akikxnXr1J7/PmJnvrjfIK58ZDr3vjSb68e+xV0vzuK47Tsyv/OHBR/7rdkriyo/68NVDFu3nvb1umctJyUCEQFgQK8Grv3C3px54Lb89P5XaXx3ATe/vJqbX24s7sDPFFf+r68+yWVH7cKhu2yJmWZIl4MSgYhsYq+BPbn7vFE8POUDbhk7la7dC5/QtnDhInr2LLz8lJkfM2PeMr76l0b23643lx+9K7tvU/jxJDclAhHZjJlx5B792XL17EQnxD3/YiPTVvfhmsffYPyM+Rz7u6c5Ye8BXHr4TmzVo1PBx5VNqeFNRCpWuzrjjAO3Zdy3R3P2QdvSrs64Z+J7jL5qLL9+7HWWr16bdIipoEQgIhWvR0N7Lj9mV/59yac4cvetWLFmHdc+/gYHXzmWuxpnsU5bkxZFiUBEqsbgPl24/ssjuOvcUQwb0IO5S1bx33+fzLG/fZpn35yXdHhVS4lARKrOvtv25l/nH8g1J+/F1j06bVhh9exbXuTNuUuTDq/qKBGISFWqqzM+u/c2/OfbB3Pp4TvRpUM9/542l8OveZIf3DuF+ctWJx1i1dCoIRGpap3a13PB6O35/MgB/PqxN7jzxZnc8ty7/GPS+xy3fSc+6jin4GPPnLOK3daso1P7+hJGXHmUCEQkFbbs1omfn7AHpx8wmJ89MI2n3pjHbVOWctuUiUUd95Yp4/jOkTtz7LD+qZ3QpkQgIqmy81bd+ctX9mXs6x/xx3+/UtSEuKkz5/HewhV8445J3PzM21x+9K6MGNyrhNFWBiUCEUkdM2P0TlvSfWnPoia0vdDYyIz1W3LVo68zaeZCPnf9sxw9rD/fPWJnBvZuKGHEyVJnsYhIM+rN+MK+gxh76cF8ffT2dGxXxwOT53DIr8bx8wensTglK7QqEYiItKJrx3Z8+/CdeOLbB3P83tuwet16fv/kDA6+ciy3PvcOa9etTzrEoigRiIjEtHXPzvz65L247+sHsu+Q3sxftprv3zuVI659iv+89mHVbvmpRCAikqdhA3py57n7c8OXhzO4TwNvzl3KV/7cyKk3vcCrsxcnHV7elAhERApgZhyxe38eu/hTXH70LnTv1I6n35zH0b99iu/8fTJzF69MOsTYNGpIRKQIHdrVcfYntuNzwwfwm/+8wa3PvcudjbMYM3k2Z39iO9ovW8kH7Quf1LZy4RoKH/cUjxKBiEgJ9OrSgR8cuxun7j+YXzz0Go+++iG/efyN8Ob4wie1fXanLnyuRDE2R4lARKSEttuiKzeeNpLn3vqYuxpn8cFH8+jVq/BJaIM7ryhhdLkpEYiIlMGooX0YNbRP0bu0TZgwoYRR5abOYhGRGqdEICJS45QIRERqXCJ9BGb2DrAEWAesdfeRScQhIiLJdhaPdndtMioikjA1DYmI1LikEoEDj5rZBDM7J6EYREQEsCRWyzOzrd19tpltCTwGXOjuTzb5nnOAcwD69+8/YsyYMQWda/ny5TQ0FL6BhMqrvMqrfDGSjGHkyJET4vTBJpIINgnA7IfAUne/qoXv+Qh4t8BT9AWK6YtQeZVXeZUvRpIxDHb3LVr9Lndv0wfQBeiW9fWzwBFlPF+jyqu8yqt8EuUrJYbWHkmMGuoH/NPMIIxaut3dH04gDhERIYHho+4+A9izrc8rIiK51cLw0RtVXuVVXuUTKl8pMbQo8c5iERFJVi3UCEREpAVKBCIiNU6JQESkxqVqhzIzG+Du7zXz3rHunvf0ZDPrBQx098l5lqsnDJXd8H/s7jNjlj0hx8uLgFfcfW4+cWQd8zB3f6yV7/m0u/+nmfPj7v+Iea693X1SIXFG5W9191Nbe63UzOwB4HbgX+6+rJznihHLnsAnoqdPufvLbXjuLsAKd19vZjsCOwMPufuaNoxhG2Awm/79PNl8ic3KDwZ2cPd/m1lnoJ27Lyl9pK3GEfv6YWY7u/tr0dcd3X1V1nv7u/v4ssWZps5iM5sOHO7u7zR5/SvA/7j70JjHGQscR/gQvgR8BIxz90tilr8Q+AHwIbA+etndfVjM8g8Ao4AnopcOBsYDOwI/dvdb4xynyTFnuvugVr7nR+7+AzO7Ocfb7u5fiXmuJ4D+wN3A39x9ap6xTnT34VnP6wlJcNeY5a8AfgqsAB4mDFf+prv/tZVy/wV8ATiE8H9/B/Cgu6/OM/5XCOtpZVsENAI/dfePWyl/EfBVIJN4jwdudPffxjz/RcDNhKXe/wjsDXzX3R+NWX4CIQn1InzuGoHl7v6lVsqNYfOfewN3Py7m+X8JnAy8SliqPioeu/xXCcvT9Hb3oWa2A3CDux/SSrklzcRv0fm7xzz/WAq4fmR/7nP8DWzyvOTKPWOtLR/AUcAbhDuBzGuXAa8AA/I4zqTo37OBH0VfT86j/JtAnyJ+jjFAv6zn/QgXhd7AlBbK3dfMYwywrI1/F1sB3wCeif7/L49R5jLCxWstsDh6LAE+Bn6ex7lfiv49Hrgl+n97OY/ynQkXon8CHwB/Ag7Lo/wVwM+BPaLHz6LHd4AxMcpPBrpkPe+S5+fv5ejfw6Pf/57AxDzKT4z+vRD47+jrSTHKfSp6XAvcCRwbPW4H/jeP808HOhbx2XsJ6JAdM+FGoq0++wVdP5rEO6m598oSc1v957ThL+GQ6EK8O3BNdCHqlecxXiHc0T4K7BP3F5lV/glCVbTQn+GVJs8tkwBa+kAAC4Cjs/4gM4+DgQ/zOH8P4GrCnWAj8CugR4E/yx7ArcDqPMrEvug3U35q9O8fiJYvyScRNDnWMGASsC6PMs8091qcC1L0+euU9bxTPheyzGc1uiAf39rnJkf5SYQa6Xhgt7hxZ5V/Ms5rLZR/COhaxO//+eyfmXBnHvvvN+s4WwKDMo88yhV0/SArWdMkcTd9XupHqvoIANz9cTM7AxhLWMfoEHdfmedhfgw8Ajzt7i+a2XaEmkZcM4CxURPPhnY+d786ZvmnzOx+QtMKwOeAJ6O224UtlBtPqMKPa/pG1GwW15+AKcBJ0fNTCU0NOfsOcpxrF8Id9YmEu/m/Ad+Ke3J3v6zINuIxZvYaoWnofDPbAoj9GTCzfoSf/QtsbOI6M255oKuZ7efuz0fH2xfoGr23Nkb5m4Hnzeyf0fPPAjflcf4JZvYosC1wmZl1Y2MTZRzfJNTO/unuU6PP/xOtlMm2hZlt52EVAcxsW6D1hc82Wg68ZGaPs+nfzzdilh9nZt8DOpvZYcD5hFpxLGZ2HOHmZ2tgLuFzOA3YLeYhMtePZ/K8fgwws98QbvwyXxM93yZu/IVIWx9Bpo3PgI7AGkIbY6w2vlJ1NpvZD3K97u4/ilneCBf/AwmxPw3c4230yzKzl9x9r9Zea6H8eEL7+t3uPruA8/+CcBEuqI04OkYvYLG7rzOzBqC7u3/QSpmvAqcAOxGahe5w92cKiH8fQjLtSvj9LSY0E0wFjnb3u2IcYzhwUFT+Sc+j893M6oC9gBnuvtDM+gDbeP4DHrp4AZ3mZnYEYTbsjOilIcC57v5IzPKq7xqHAAAgAElEQVSn53rd3W+JWb4OOAv4DOH/7xHgj3H/fszsZeDTwL/dfW8zGw2c4u5l3TuluZ87I+7PX9C505QIilWqzuasct0IF7ClpYsy1nn7Ee4gHJjt7h/mWf454FJ3fzp6fiBwlbuPKnmwuc8/HRjmWaMm8ix/Wq7X3f0vrZS7mZDA/u3u+dxBN3e8HoS/sZZqcbnK/Rh4Cni2kAtxdIyCa1RmNopQA+nq7oOiEUznuvv5eZy/I2G0EcBr+f4uzawDYXAEwHRv2xFLje4+MkoIe3sYPfWCu+8bs/yOwPWEfr7dzWwYcJy7/zRG2S0Iv7c38/3cFCNVTUNm1gk4D9ie0OH2J3ePUxXPuBh4zMyOcvc3omNeBnyR0NYeN47dCe3ivaPn84DTPObomWj45i8JbZRG/BrN3oQPYA/g/ejlAWa2EDjf3SfG/BHOA/4SXcgg9D20eLcSnf8udz8px6iZTPyxRk0R7iTbk9UskKd9sr7uROg3mgi0mAjc/UwzawccbWaZi9g04OF8PkfRRfBzhDvhdqGCB+7+45iHeIdQM/lNVMt9ilAruDfm+XOOugHiNq1dw8aOZtz9ZTP7ZMyyRDWwSwhr4X/VzHYws53c/f6Y5Q8mdPK/Q/jsDDSz01tLZM2M1togj8/fQjPrSvj/us3M5hKvSS/jD8ClwO+j8042s9sJI9maZWZnA/8LvAVsa2bnuPt9eZy3YKmqEZjZnYTmoKeAI4F33f2iPI9xCOEX+FlCdX4f4Bh3X5DHMZ4l1CCeiJ4fTBg1cUDM8m8Cx7r7tDxjf4lw5/Z8k9f3B37v7i2u+mpm2cPbjDBaBWAZ4ULeYh+HmfV39zkWxnBvxt1jbS5kZvcQRroU2kbc9Hg9gFtba1oys60JbeFzCB2mRhh6uRUwOm4zl5k9TBguOoGNF2Lc/Vd5xr0Voa/i24QBD91iliu2RvW8u+9nZpPcfe/otZdb+/xklb+T8LOfFt0Rdwaey6NpcQLwRXefHj3fkdBMN6KVcjk/dxl5fP66EPqX6oAvEW6sbvNWhv1mlX/R3fdp8v/XatOqmU0hfM4+ivoVbmurWniqagTAru6+B4CZ3QS8kO8BStTZ3CWTBKJjjo0+XHF9mG8SyDrv801fdPfxMc+fudDsREiA9xIuhl8mxt2ku8/JiuPV7PeiZBh3l7nMsNdSWQ7sEOP7/he43t2vyX7RzL5BGA7aaq0oMsDdj8gvxE3O90dgV8I8lKcIne5xa3NQfI1qlpkdAHjURPMNQs0orqHufrKZnQLg7issUy2Kp30mCUTlXzez9q0Vinuhb4mFOSv3uvuhhA72Qtrl55nZUKLaiZmdSLi5aM1qd/8IwnL9Uc2yTaQtEWxoR3T3tfl99nJ2Nh8CzI0+xK02zWSZYWbfJzQPQbiQvp1HKI3RXdW/2PSOuLWZvQ9ZGKn0F2BW9NpA4DTCxKoWZTqzLYw4Ge7RTEwL24ne3ULRpu4ys1sJ4+k7Rf+OJAxJbFWxnWK26cSmOsJFtdUOWmB/dz8jRzy/sfxGXT1rZnu4+yt5lMnWB6gnjBCbD8zLs4mz2FE35xGGnm4DvEcYBnlBHudfHdUCMhfCoeSXlBqjG7nM38+XCDWMWCz3xLDMhL5veTSaKZdocMFyM+vh7ovyiDnbBYTO8p3N7H3C3/6XY5TLHim02fNCa8RxpK1paB2hGQPCxbwz4Y8i3wt5sXH0An5E1qgP4Idxm5esiJm9ZnYUYVbjNtG53wPuc/cHY4aPhaGXe2aaFqI7k5fdfeeWS24o34XQxzGCUMu4Dfhlax2wLfQxAPHbeM0suz9nLaGJMOdosCblNlTl83kvx/e+SuineptwAcy3jyRznF0IbfUXA/XuPiBmuaJG3RTLzD4D/A8hAT9KGP12ZnYtuZXyHQkX0+y/n+viNnWZ2Y+A2YSJbEYYgbYVYaLa19z94FbK3wXsDzzGxutJ3hfi6O+gzmMubdHc7y3r/Bo11BZK0NmcCmb2P4S26X8SLsjHA3e6+89jlu9AmEl7GGEI5eXu/rcY5YruY4iq9o9EVfu8mNkMQnv8Zm8BV3j8JUqK7SM5hrDEwycJyzw8R1hv6E9xykfHKHjUjYVx/xcSdXZnXm+tj6XJMfoQLqYGjHf32JuvRxfQle6+LnpeT5hpvDxm+efdfb8mr4139/3j9HUUmkib9LHlKh93HlHT43Yi9BnmUyvPS9qahop1Cxs7m48iTCCJ3dlsZte4+zetmTVXYnRW/re7X2Fmv22mfIt3JBbGr19BGDF0GWEs+z6EySzneMyx6O7+MzN7iI2Lnp0Zt2zkRUL/wj6EZo7fm9mJ7n5iK+edE/3R31TIhTw6RjFV+3GEJRFyabWPxMy6u3tmWYxinEAY+35tpoPawkigWKzAUTdZ/kUYPjqG/CaiZc7/uId1fR7I8VocjwOHAplh150JNYtYgy2A9WZ2EvD36Hn2567VO98i7ryb9rFl+rmOJf6ILWBD8vsMYfTY4YRrkhJBGym2sznTpnlVgefPdMg1Flj+/wiL3fUkdHRf7O6HWRgJdR0x2+gBPAw1zaeDMttZ7p75GT4A/svMYq0cWqI22pXAK2aWV9Xe3fOZPZzL7cAxhPbsTF/ThsMD28U8zl45mgGPJKxVFMevgM80HXVDaKqLY6W7/6b1b9tUdOfaAPSNmkczP393wizduDp51twbd19qYUhqXF8i9HFcR/h/Hw98Oeq3+HoL8RfVNFmKPjYLw3S/SFgq5gVCs9q2cWtDhVIi2FRRnc3ununQ2svdr81+z8KKkJst/dCk/Jjo3w13JBZmSXaN7jRb097dH4rK/dLd/x4d73EzKzQ55S0rCRDF0pOwXktcBV3IszxA1t1oXMVW7d39mOjfbfM9d3T+rxGWQ9jOzLJnAXcjrJkVV0GjbrJca2F2/KNs2tnc2o3BuYTlKbYmJMPMH9Biwk1KXMvMbHjmfGY2gjCcM5aoM7i5mt3TLRTN1P5vJlyEZ7XwvS0ZBGSvWLua0MzWIjN7D5hJmAt0qbsvMbO3y50EQImgqT3NLHPBNcJaJYvJv7P5dMIdSbYzcryWk4XJJ+cRxqBPAHqY2dXufmUrRVdGHXU9CEP/Puvu/4o6T9e1UrZoZjYQ+D7hQvAvwh3yTwijlm7P41AFXcgziqjaX0VYufIhNnbyFsQKm9l7e3TunwPfzXp9ibvPz+P0RY26ISwUeCphmYUNy6hHz5sV3fxca2YXeswls5vxTeBuM8vM2+hPmCAXi4XZuV9l8z6OFgdb+Mbhz90Ic4nmE9bJ+rvnNzv/VuAFC2tFZfrYWpzMGLmHMH/pZGCdmd1LjKasUlBncQlZGDf9RcJoh6ey3upGWL0yVru3RZNPzOxLhOr8d4AJrVVNLSwFcAXhj/di4GuEpPQ+oY8g73Vz8mFhH4JxhM7NIwjDb6cSmqhaXOcnx7E6EJYocEJnZ6t7AhRbtTezvQgjTI4gXDjvAB73PP9IrMj19ItVglE3rxEmpOW1D0NW+QsIk6EWRs97EdbquS6PY7QntLUbYYmKfDq7nyX8/TWd0HdP3GNExxlG+D1+Dngvn34rC2tFZfrYYq8VZaEZYjShb+AoQrPaWYR9Mcq2VI0SQQlFo0W2JccdHWEZ2lgjkMxsKmHRsNuB37n7ODObnO/ww7bWdESGmX1IWL4333VmjiLckb1FuBBsS5gx/VAr5TKjji4hR9U+7qid6FgHEP4YDwW+43lM9bciZ/YWqwSjbu4ELvTCd8PLtWhhq8NvrXQ75MVeILGV42wFfJ5wc9Atxo1Ed3dfbGa9c72fZ60ukwyPjM7/GXfvm0/5fKhpqISiC8275NEp24wbCGPQJxOWnx5MmBBTMDM7091zzU8oqSadhB8ADdGFKZ8/hKsJU+3fjI45lNBU1GIiKFXVPmpa2JvQRPIeYSnifBQ7s7dYxY666Qe8ZmYvsmkfQdwaTZ2ZWaYmFSWiDjHKfQr4D7nb952NO7a15n4L64XFnjuTLeqrOZmwdPbfga96k5nyzWg6WGDDIclvsAAAUS3oPuC+qKO7bFQjKAMLa/v8FtiF8AdQT9ghLO5Wd9nLWDthdmy9u3+/iJha3aqyWGb2DqFZKlfburt7rD8EM3vS3T+Z9dwIW/3FXvgsKpdX1d7Mzoy+vxPhAnBXIXfFVuK1kgo4f7HLiOdcYNFz7HPRTPkrCe3zNxA+v+cBs9w99p4UxbAws7gL4f9+DXn28VlYBv1v7v5S+aLMed5SLZqXN9UIyuN3hOrc3YSlFU4jTFKLK7stsBOhetjqWi9NRpps8hbhLq+s3H1IiQ411cweJCwL4YTq+YuZJoO4TQSEO/kPCJvjbBnj+28i7C41kzB2+zPZI8fyuCMu9VpJ+Sp21M04C0uZZ1ZxfSHPhPgdwgiirxE+e48S9k5uUbGjtrK+L9bifC2U/27r39U8yzFnItdrORyT+XZCDfioYuLIhxJBmbj7m2ZWH7XT3hx1YMUtu8kqldHQzzgXln6EC1jTpSyMMK+gzVjY5SlzBz/WYy5BHOlEWHAtc2f6EWFJ72OJ0URQRNV+dB4xNquIUUulUuyom5OAKwkLLxrwWzO7NDMcuTUelhK5Pnrko6gLuJnt7O6vRR21ueIqdF5M3PMXNY8iuw/LzFbl06dVLCWC8lgejXp5ycyuIKw8mM/qo001EK998X7CnIPNqrRmNraI8+clqlrvQ1hjCOAiMzvQ3S+LU96Ln9g1GPhmAVX7t919ZqEnLXbUUql42B5xZwocdUNYJ2ifTC0g6jP5Nxtn6uZU7M/vMXfwa8ElwDmECXUbDpv1dYvDX0sgex5FdtLJdx5Fm1MfQRlEnbsfEvoHLiaM678u0/kZo3z2H1I94c72x+7+uzKEW3JRE9Ve0Z1hprNwUtwLoRWxw1MxzGyiuw+Pvr7H3T+XZ/mS7MdQqBKOunnFoxn20fM6wqKDe7RQrGQ/v4W1+K8lrFXkhOHIF3sLq4Y2KX8SYTOhxRZWAR4O/KTcNYKs8xc0j6JJTeY2wvyPDcoZvxJBGUSjZFY0uRDmM3wv+w9pLWF/grhDTw3Yl6ytKgltvG32i44SwcGZUULRcLqxeSSCcUQ7PPnGjT2muPvu5Yo5Okf2RiKxVxutFGb2I3f/gRWxem10nCuBYYR5FBCalSa7e9wlLopiYc/r/8s6/xcIw1n3a77UJuUnu/swMzuIsMfEr4DvxS1frKg14DyymkYJn+UWa2UW5uFkL02yyd+su5etRqOmofIoavheoXeOFmYVX0dYZG7DVpXA9mZ2vrs/WshxC/BzYFL0wTbCH0SsZqFIg7u/YJsu8dEWq8B6M1/npdhRY4Vy9x9E/xbVtObul5rZ5wjr3Bhwo7v/s7VylnsfgOzjxv35zd1vzXr+VzNrdo2gHDKTyI4GbnD3ey2s99NWriMMH85MoDuVUMM9u5Vy3yGMrpoDYGEV1M8RFg/8YTkCzVAiKI9iF80q1LXAoe7+TvaLFpYVfpBwYSo7d78j6pPYh3Ah+Y7nN7O40B2eipVZYiR7eRHIc/ghxY8aK0ipRt1E33sPYcmD2DKjdczsx4TRWrcS/u++RH4dwU+Y2XcJc0CcUCN5IDNRy1ufj/K+mf2ecDP2Swszrevy+VmKtI9vutT1f8zs5RjlbiDEjIXF535OWA58L8JGNy2u3lsMJYLyKGr4XhHaESZANfU+4Q6lrHKM2sjEsrWZbZ1HG2euHZ6+1HKR4rl7fQmPVfCosSIUNeomI+pj+CVhyK2RfyI8vEkzzPVm9jxh+ZM4MiOczm3y+leINzHrJMIyIVe5+0Iz609oamwr68xsqLu/BRv6POKs9VWfleROJtTE7gHusbAfedkoEZRHUcP3ivAnwnj7v7HpVpVfIIyRL7fsURu5Zla22sYZdUyOdPdDLc8dnipIqUeNxVKCUTcZVxA2Qilk32wIF8IvsfGO/hTyWPTQC1y9Nav8crKGGEdNLW1Ro8y4lFCrmUH47A8G4jTX1ZtZu6g/8BDC31JGWa/V6iwuEyti0awiz7srubeqjDOOvlQxdCYsp3wQ4ULwFGFT+JUxy28ys7jaNDNq7P8yd4htcP5iR9084+4HFnH+IdH5D4zO/wxhOO87Mcs3EG4qBrn7OWa2A7BTnnNREhU1R2X//be63IiFnQGPAuYRlrIe7u5uZtsDtxTzO2n13EoEpVOq4XvVzsKer4vZOI/gFKCnu58Us/z3CU1pd7LpfgR5LdqVFDO7yHPsR9H0tTKev9hRN9cS9vj9F5sukdEmn18Li95NAE6Lhg93Bp7zEiwk1xayEtlgd/9qPoksGmjQH3jU3ZdFr+1ImB+k4aPVoFTD94o4f1fgvwkjDQYQNsR4izBy4s/lPHeTODbbFzbXay2Uf5vcE5LyWrQrKdnzEbJea7PhqNbCnr0xyxc7/LSoeSBm1ujuI5sM5439+UlaNSYy9RGUUKmG7xXhNsKG84cTOsy6ENppLzezHd39e20UxyQz29/dxwOY2X7kt8PWrmzetHRDyaMsMdu4H8W2Zpa9JEg3wnpHbaWoUTcl+Pz+gWgeSHS8yRY2W4o7IXB1dPHMjBobSnIruRZiqLufHH0ecPcVZnlud9jGlAhKqJTD9wo0JOvO/2oze9Hdf2JhVc1XgbImAts4I7o9cJqZzYyeD47OH9cthKalzL65p0SvxWpaStCzhE7Jvmy6zMESwpLibaWoUTfF3tFTxDyQ6IJ5A/AwMNDMbiP0NZwR89yVoOoSmRJBaZVk+F4RlpnZQe7+tJkdS1iPH3df30Z3JMe0/i2x7NSkGeCJmOOwE+Wl24+i2DiKGnVD8Xf0Bc8DiTpHLwI+Q+jsNuAid5+X34+QqB9QZYlMiaCESjh8r1DnAX80s50IyymfBRsWDSv7oleFzojOodimpUSVYBx+secvdtRNsTO7i50HMh7Yzt0L3rc6Se7+mJlNpIoSmTqLy6DY4Xu1zsymEYbeZVYCHUTYj2E94YJa6Vt2vklx4/CLPX9RnZVm9hDwdeBudx8e3dGf5e5HxihbB5zo7ncVOg/EzF4FdiTUrpaxMZFW+u895/LXGeUc9VMsJYIyKHb4XpHnPhz4LJsuOnevuz9c7nOXijWzemVGCWseZVHsOPwSnL+oUTfRjcyNhLWxFhDd0cf9fy92Hkhzv/8q+L0/0cLb7mVcNK5YSgRlUOzwvSLOew3hTuovbFzeYQBhrZs33P2icp5fggoYh/8sYWbqM9Ed/VDgDnffN2b5zKCHzoQ1epYR9sye4DH2eKj2eSC1SImgDCxszLKQTYfvdSRqpy/XH4SZve7uO+Z43YDX3X2HcpxXNpXUPJLo3EZY7fIswjDcR4k6K919bMxj3E5YLO8+QrPM0cCLwM6E5qIW1wyq9nkgxTKzC4Db3H1h9LwXcIq7X9dyyeQoEZRB9IfQHC/XH4SFfQDOdvcXmry+L3CTt7KxiKSDmU1g01E34/PprDSzR4DPebSCbjRR8e/A8YRawa6tlM+1xMgN7t4WCy8mzsxeatof05YTCguhUUNlUILhe4U6g7DSYzc2Ng0NJIzJPyOhmGqOmQ0g7EeQWWvnacLIkVwrw5ZDsaNuBhFmpWesISyXsMLM4oyHr9Z5IKVSZ2bm0V22hY2pOiQcU4uUCMqgBMP3ChKNStjPzLYia9E5z28vACnezcDtwOej51+OXjusjc4/GjjXzAoddXM7MN7M7o2eHwvcEY0CijMxsCrngZTQI8BdZnYD4UbgPMK8goqlpqEyqMS1RizaKyCp89eSZpoGNnutjOcvetSNhT00DiIkkafdvTGPsn8mNAVlzwM53d3Pj3uMahYNoT2X0GFvhH6aP3rYm6IiKRGUQSUummVmM919UFLnryVm9m/gz2wcPnwKcKa7H5JYUG2o2ueB1CI1DZVHImuNmNlvmnsL6Fnu88sGXyFsV/lrwmfgWeJtTJIWRyQdQJKqcdSUEkGJJbxo1pnAt8iddE5pg/NL8BNCU8gCgGjVz6sICSL1Kn3iVxsYmfV1J0JfUe+EYolFTUNlUOzwvSLO+x/gcnffbH9cM3s7wdFMNSXXUMFKHz4o5WVmT7v7QUnH0RzVCMojqUWzTgRybgepJNCm6sysV5Magf7WakSTNYfqCDWEpFcmbpE+nOVR7PC9gmgKf8X4FfCsmf2d0FZ8EvCzZEOSNvQrNvYRrAXeYeNQ4oqkpqEyqMRFs8zsoTirR0ppmNmuwKcJNwGPu3s+G/NIFTOzToTtYoew8Wbb3f3HiQXVCtUIyiCpC34Ly+AaULH7paZRdOHXxb82/Yuw1thEmmmqrTSqEaSIma0DxhEu/E3t7+6d2zgkkZpjZlPcffek48iHagTpMg04193faPqGmc1KIB6RWvSsme3h7q8kHUhcSgTp8kPCKIVcLmzDOERq2UHAGdHEslVUwQ5rahoSESmhShws0holgpRJw1aVItK2lAhSRFtVikghlAhSRFtVikghmutYlOq0MtqWsql9qJLxzCLS9jRqKF3OQFtVikie1DSUQtqqUkTyoRpByphZD+BTZI0aMrNH3H1hspGJSKVSH0GKmNlphPVNDgYagC6ElVAnRO+JiGxGTUMpYmbTgf2a3v2bWS/g+VwjikREVCNIFyPHXqmETcNzLUQnIqI+gpT5GTDRzB4FMovMDQIOI+yjKyKyGTUNpUzUDHQ4WaOGgEcy2yaKiDSlRCAiUuPUR5AiZjbQzP5mZk+Z2ffMrH3We/9KMjYRqVxKBOnyJ2AsYe+B/sA4M+sTvZdzaVwREXUWp8sW7n5D9PWFZvZl4EkzO47co4lERJQIUqa9mXVy95UA7v5XM/sAeIQwuUxEZDNqGkqXPwL7Zb/g7v8GPg9MSSQiEal4GjUkIlLj1DSUImbWDjgLOB7YmqytKoGb3H1NguGJSIVSjSBFzOwOYCFwC5tuVXk60NvdT04qNhGpXEoEKWJm0919p2bey7mNpYiIOovTZYGZfd7MNvxezazOzE4GtMSEiOSkRJAuXwBOBD40s9fN7HXgA+CE6D0Rkc2oaSilohnF5u7zko5FRCqbagQpY2bdzWyou3+cnQTMbFiScYlI5VIiSBEzOwl4DbjHzKaa2T5Zb/85mahEpNIpEaTL94AR7r4XcCZwq5mdEL2nHcpEJCdNKEuXenefA+DuL5jZaOB+MxuAFp0TkWaoRpAuS8xsaOZJlBQOBv4L2C2poESksqlGkC5fo0kTkLsvMbMjgJOSCUlEKp2Gj6aImT0CPAw85O6vJR2PiFQHJYIUMbOtgCOix47A84TE8Li7L00yNhGpXEoEKRUtM7EfcCRwCLACeNTdr0g0MBGpOEoENcLM+gKHu/ttScciIpVFo4Zqx/lKAiKSi2oENcLMZrr7oKTjEJHKo+GjKWJmi5t7C+jclrGISPVQIkiXhcA+7v5h0zfMbFYC8YhIFVAfQbr8BRjczHu3t2UgIlI91EcgIlLj1DSUMmbWgzChbBvCQnOzgUfcfWGigYlIxVLTUIqY2WnARMJCcw1AF2A0MCF6T0RkM2oaShEzmw7s1/Tu38x6Ac+7+47JRCYilUw1gnQxcu87sB5tTCMizVAfQbr8DJhoZo8CmeGig4DDgJ8kFpWIVDQ1DaVM1Ax0OKGz2ID3CJ3FCxINTEQqlhKBiEiNUx9BjTCzV5KOQUQqk/oIUsTMTmjuLWCrtoxFRKqHEkG63AncRu6RQ53aOBYRqRLqI0gRM5sAnO7uU3K8N8vdByYQlohUOPURpMs3geaWoj6+LQMRkeqhGoGISI1TjaBGmNn/SzoGEalMqhHUCG1VKSLN0aihFNFWlSJSCCWCdNFWlSKSN/URpIu2qhSRvKmPQESkxqlpKGW0VaWI5EtNQymirSpFpBBqGkoRbVUpIoVQjSBdtFWliORNfQTpoq0qRSRvahpKGW1VKSL5UiIQEalx6iOoEdqqUkSaoz6CFNFWlSJSCCWCdNFWlSKSN/URpIi2qhSRQqiPIF20VaWI5E2JIF0GAMtyveHujW0ci4hUCfURpMtg4G4zaw88DjwEvOBq/xORFqiPIIXMrBtwKGEV0n2BacDDhIllm21aIyK1TYmgBpjZrsCRwGfc/fCk4xGRyqJEkEJmNjzHy4uAd919bVvHIyKVTYkghcxsPDAcmEyYTLZ79HUf4Dx3fzTB8ESkwmjUUDq9A+zt7iPdfQSwNzCF0G9wRZKBiUjlUSJIp53dfWrmibu/SkgMMxKMSUQqlIaPptPrZnY98Lfo+cnRax2BNcmFJSKVSH0EKWRmnYHzgYMIfQRPA9cBK4EGd1+aYHgiUmGUCFLGzOqBW9z9y0nHIiLVQX0EKePu64AtzKxD0rGISHVQH0E6vQM8Y2b3kbX2kLtfnVhEIlKxlAjSaXb0qAO6JRyLiFQ49RGkWLTmkKtzWERaoj6CFDKz3c1sEmES2VQzm2BmuyUdl4hUJiWCdLoRuMTdB7v7YOBbwB8SjklEKpQSQTp1cfcnMk/cfSzQJblwRKSSqbM4nWaY2feBW6PnXwbeTjAeEalgqhGk01eALYB/AP+Mvj4z0YhEpGJp1JCISI1T01CKmNk17v5NMxsDbJbh3f24BMISkQqnRJAumT6BqxKNQkSqipqGUipaa2jH6Ol0d9fy0yKSkxJBCpnZwcAthDWHDBgInO7uTyYYlohUKCWCFDKzCcAX3X169HxH4I5o20oRkU1o+Gg6tc8kAQB3fx1on2A8IlLB1FmcTo1mdhMbO4+/BExIMB4RqWBqGkqhaG/iC9i4VeWTwHXuvirRwESkIikRiIjUODUNpYiZvUKOiWQZ7j6sDcMRkSqhRJAux0T/XhD9m91HsLztwxGRaqCmoRQys2fc/cDWXhMRAQ0fTasuZnZQ5omZHYD2IxCRZqhpKJ3OAv5kZj0IfQaLCEtTi4hsRk1DKWJmo4DxHv1SzWUG0h0AAATQSURBVKw74Xe8KNnIRKSSqWkoXU4HJpjZ38zsDKBBSUBEWqMaQQqZ2c7AkcDhQA/gCeBh4Bl3X5dkbCJSeZQIUs7MOgOjCYlhlLuPTDgkEakwSgQpYmaNwDPAQ8BYd1+ZcEgiUgWUCFLEzNoR1hc6glAL+Bh4BHgoWoFURGQzSgQpZmb9CU1CRwA7AM+5+/nJRiUilUaJIOXMrA7oCiwl9BE8k3BIIlJhNHw0hczsdjPrbmZdgFeB6cC3lAREJBclgnTa1d0XA58FHgQGAacmG5KIVColgnRqb2btCYngXndfQwvLU4tIbVMiSKffA+8QFpp70swGA4sTjUhEKpY6i2uEmbVz97VJxyEilUc1ghQysz5m9hszm2hmE8zsWsJSEyIim1EiSKe/AR8BnwNOjL6+M9GIRKRiqWkohcxsgruPaPJao9YZEpFcVCNIpyfM7AtmVhc9TgIeSDooEalMqhGkkJktIYwYyiw5XQ8si752d++eSGAiUpGUCEREapyahlLIzB6P85qICGjz+lQxs05AA9DXzHoBFr3VHdg6scBEpKIpEaTLucA3CRf9CWxMBIuB/0sqKBGpbOojSCEz+4a7/6bJax3dfVVSMYlI5VIfQTqdkeO159o6CBGpDmoaShEz2wrYBuhsZsOz3upO6DsQEdmMEkG6HE6oDQwArsp6fQlwWRIBiUjlUx9BCpnZlwn7DwxhY7J3d/9xYkGJSMVSjSCdTgUWABOBlQnHIiIVTjWCFDKzKe6+e9JxiEh10KihdHrWzPZIOggRqQ6qEaSImb1C6BtoB+wAzABWESaWubsPSzA8EalQSgQpEu1N3Cx3f7etYhGR6qFEICJS49RHICJS45QIRERqnBKB1Bwz+x8zm2pmk83sJTPbr4znGmtm2itaKpomlElNMbNRwDHAcHdfZWZ9gQ4JhyWSKNUIpNb0B+ZlluR293nuPtvM/p+ZvWhmU8zsRjMz2HBH/2sze9LMppnZPmb2DzN7w8x+Gn3PEDN7zcxuiWoZfzezzRb5M7PPmNlzZjbRzO42s67R678ws1ejslc1LSdSbkoEUmseBQaa2etmdp2ZfSp6/Xfuvk80I7szodaQsdrdPwncANwLXADsDpxhZn2i79kJuDGaq7EYOD/7pFHN43LgUHcfDjQCl5hZb+B4YLeo7E/L8DOLtEiJQGqKuy8FRgDnAB8Bd5rZGcBoM3s+mpT3aWC3rGL3Rf++Akx19zlRjWIGMDB6b5a7PxN9/VfgoCan3h/YFXjGzF4CTgcGE5LGSuCPZnYCsLxkP6xITOojkJrj7uuAscDY6MJ/LjAMGOnus8zsh0CnrCKZnd3WZ32deb5hddemp2ny3IDH3P2UpvGY2b7AIcAXgK8TEpFIm1GNQGqKme1kZjtkvbQXMD36el7Ubn9iAYceFHVEA5wCPN3k/fHAgWa2fRRHg5ntGJ2vh7s/SNhveq8Czi1SFNUIpNZ0BX5rZj2BtcCbhGaihYSmn3eAFws47jTgdDP7PfAGcH32m+7+UdQEdYeZdYxevpywadC9ZtaJUGu4uIBzixRFS0yIFMnMhgD3a+lvqVZqGhIRqXGqEYiI1DjVCEREapwSgYhIjVMiEBGpcUoEIiI1TolARKTGKRGIiNS4/w+D/E38cS+CGAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "freq.plot(20, cumulative=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "http\n"
     ]
    }
   ],
   "source": [
    "test_word = 'https//'\n",
    "print(test_word[:4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['word', 'count', 'Text', 'word', 'count', 'RNA', 'genomes', 'picornaviruses', 'frame', 'regions', 'UTRs', 'disease', 'virus', 'FMDV', 'UTR', 'kb', 'domains', 'ribosome', 'entry', 'site', 'IRES', 'initiation', 'translation', 'replication', 'element', 'cre', 'Less', 'structures', 'stem-loop', 'length', 'poly-C-tract', 'nucleotides', 'series', 'tandemly', 'pseudoknots', 'PKs', 'structures', 'PKs', 'acetylation', 'primer', 'extension', 'SHAPE', 'analysis', 'contribution', 'replication', 'mutation', 'deletion', 'experiments', 'SHAPE', 'mutation', 'experiments', 'importance', 'PK', 'structures', 'function', 'Deletion', 'experiments', 'PKs', 'replication', 'genomes', 'advantage', 'replicons', 'genomes', 'PKs', 'competent', 'virus', 'genomes', 'PK', 'copy', 'report', 'presence', 'packaging', 'signals', 'PK', 'region', 'VP3', 'VP0', 'VP2', 'VP4', 'virus', 'assembly', 'P2', 'P3', 'regions', 'proteins', 'VPg', 'pro', 'region', 'reporter', 'genes', 'study', 'replication', 'requirement', 'containment', 'FMDV', 'UTR', 'picornavirus', 'UTR', 'nucleotides', 'regions', 'nucleotides', 'end', 'stem', 'loop', 'S-fragment', 'PKs', 'tandem', 'repeats', 'region', 'stem', 'loop', 'downstream', 'interaction', 'site', 'sequence', 'similarity', 'PKs', 'figure', 'duplication', 'events', 'replication', 'recombination', 'PKs', 'virus', 'isolates', 'strain', 'PKs', 'importance', 'life', 'cycle', 'presence', 'PKs', 'UTR', 'picornaviruses', 'absence', 'PKs', 'wild-type', 'wt', 'replication', 'Furthermore', 'competition', 'experiments', 'copies', 'PKs', 'advantage', 'genomes', 'replicons', 'genomes', 'PKs', 'virus', 'genomes', 'PK', 'copy', 'report', 'presence', 'packaging', 'signals', 'PK', 'region', 'Plasmid', 'construction', 'FMDV', 'replicon', 'plasmids', 'pRep-ptGFP', 'polymerase', 'control', 'mutations', 'PK', 'region', 'replicon', 'plasmid', 'SpeI', 'KpnI', 'fragment', 'vector', 'pBluescript', 'pBluescript', 'PK', 'PKs', 'digestion', 'HindIII', 'AatII', 'insertion', 'DNA', 'sequence', 'PK', 'PKs', 'PCR', 'amplification', 'Forward', 'primer', 'FMDV', 'reverse', 'primer', 'product', 'HindIII', 'AatII', 'pBluescript', 'PK', 'vector', 'PK', 'deletion', 'introduction', 'AflII', 'site', 'end', 'tract', 'PCR', 'mutagenesis', 'vector', 'pBluescript', 'C11', 'PKs', 'PCR', 'mutagenesis', 'primer', 'FMDV', 'reverse', 'primer', 'PK', 'sequences', 'vectors', 'plasmid', 'NheI-HF', 'KpnI-131', 'HF', 'https', 'doi', 'bioRxiv', 'preprint', 'Mutations', 'PK', 'structure', 'DNA', 'digestion', 'AflII', 'AatII', 'ligation', 'pBluescript', 'PK', 'vector', 'Mutations', 'replicon', 'plasmid', 'effects', 'truncation', 'poly-C-tract', 'replication', 'sequence', 'PCR', 'mutagenesis', 'primers', 'C0', 'SpeI', 'FMDV', 'transcription', 'transcription', 'reactions', 'replicon', 'assays', 'Transcription', 'reactions', 'amounts', 'RNA', 'SHAPE', 'analysis', 'DNA', 'μg', 'DNA', 'HiScribe', 'T7', 'synthesis', 'kit', 'NEB', 'DNase', 'treatment', 'purification', 'PureLink', 'FastQ', 'files', 'quality', 'FastQC', 'quality', 'reads', 'Sickle', 'algorithm', 'Host', 'cell', 'reads', 'FastQ', 'Screen', 'algorithm', 'FMDV', 'reads', 'contigs', 'IDBA-UD', 'Contigs', 'FMDV', 'library', 'Basic', 'Local', 'ALighnment', 'Search', 'Tool', 'BLAST', 'https', 'doi', 'bioRxiv', 'preprint', 'consensus', 'sequences', 'SeqMan', 'Pro', 'software', 'DNA', 'STAR', 'Lasergene', 'package', 'DNA', 'STAR', 'SHAPE', 'data', 'structures', 'stems', 'PK', 'nucleotides', 'reactivity', 'NMIA', 'https', 'doi', 'bioRxiv', 'preprint', 'nucleotides', 'base', 'pairing', 'hindrance', 'NMIA', 'reactivity', 'interacting', 'nucleotides', 'downstream', 'residues', 'PK', 'structure', 'SHAPE', 'suggests', 'interactions', 'differences', 'heterogeneity', 'formation', 'PKs', 'experiment', 'evidence', 'interaction', 'PK4', 'https', 'doi', 'bioRxiv', 'preprint', 'orientation', 'removal', 'PKs', 'decrease', 'replication', 'requirements', 'levels', 'replication', 'level', 'replication', 'PK', 'mutagenesis', 'C11', 'replicon', 'plasmid', 'PK', 'addition', 'orientation', 'PK', 'sequence', 'hybridisation', 'loop', 'upstream', 'downstream', 'sequences', 'orientation', 'PK', 'replication', 'level', 'replication', 'passage', 'reduction', 'https', 'doi', 'bioRxiv', 'preprint', 'passage', 'figure', 'replicons', 'PK', 'disadvantage', 'https', 'doi', 'bioRxiv', 'preprint', 'virus', 'transfection', 'cells', 'requirement', 'RNA', 'structure', 'PK', 'region', 'virus', 'assembly', 'UTR', 'FMDV', 'picornaviruses', 'size', 'presence', 'RNA', 'elements', 'function', 'features', 'series', 'PKs', 'varying', 'number', 'virus', 'strain', 'study', 'PKs', 'role', 'life', 'cycle', 'PK', 'structures', 'SHAPE', 'mapping', 'viruses', 'date', 'contain', 'PKs', 'replicons', 'viruses', 'PK', 'competent', 'replicons', 'PK', 'advantage', 'replicons', 'PK', 'Replicons', 'PKs', 'co-467', 'yeast', 'tRNA', 'observation', 'impact', 'replication', 'viruses', 'genomes', 'numbers', 'PKs', 'plaques', 'addition', 'differences', 'PKs', 'differences', 'replicons', 'https', 'doi', 'bioRxiv', 'preprint', 'copy', 'PK1', 'PK4', 'observation', 'report', 'deletion', 'PK1', 'Poly-C-tract', 'effect', 'replication', 'findings', 'truncation', 'Poly-C-tract', 'C11', 'construct', 'effect', 'replicon', 'replication', 'cell', 'lines', 'Mengo', 'virus', 'role', 'poly-C-tract', 'aspects', 'lifecycle', 'tissue', 'culture', 'system', 'presence', 'PKs', 'isolates', 'PKs', 'advantage', 'replication', 'passage', 'replicons', 'PKs', 'level', 'replicons', 'PK', 'decline', 'isolates', 'contain', 'PKs', 'variation', 'effects', 'host', 'range', 'localisation', 'https', 'doi', 'bioRxiv', 'preprint', 'https', 'doi', 'bioRxiv', 'preprint', 'https', 'doi', 'bioRxiv', 'preprint', 'Significance', 'replication', 'C11', 'PK', 'disrupt', 'C11', 'PK', 'restore', 'Aii', 'Significance', 'replicon', 'Error', 'bars', 'SEM', '=', '*', 'P', '<', '****', 'P', 'https', 'doi', 'bioRxiv', 'preprint', 'https', 'doi', 'bioRxiv', 'preprint']\n"
     ]
    }
   ],
   "source": [
    "lines = 'lines is some string of words'\n",
    "# function to test if something is a noun\n",
    "is_noun = lambda pos: pos[:2] == 'NN'\n",
    "# do the nlp stuff\n",
    "tokenized = nltk.word_tokenize(total_text)\n",
    "nouns = [word for (word, pos) in nltk.pos_tag(tokenized) if is_noun(pos)] \n",
    "\n",
    "print(nouns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "remove = set(['=','<','*','http','https','doi','biorxiv','preprint','word','count','text'])\n",
    "words = [noun.replace('PKs','pseudoknot').replace('PK','pseudoknot') for noun in nouns if not noun.lower() in remove]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAE+CAYAAACA8heHAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJztnXmYXFW1t99fd2fqzBPQTEkgYZJB6ARkUJnEWdGLIBcRFEG9iiDqh14HhKvX4SqiqAgXRa4iMihCUGYZJAJJhwQCBExIIAyBkHnoTJ2s7499qlNdqe4651RXV3ed9T5PPVXn1Fl7r6o6tdfea6+9tswMx3EcJ7vUVVsBx3Ecp7q4IXAcx8k4bggcx3EyjhsCx3GcjOOGwHEcJ+O4IXAcx8k4bggcx3EyjhsCx3GcjOOGwHEcJ+M0VFuBOIwZM8bGjx+fSnb9+vUMGjQodd0u7/Iu7/J9VX7mzJlLzWxsyQvNrNc/mpubLS0tLS2pZV3e5V3e5fuyPNBiMdpYdw05juNkHDcEjuM4GccNgeM4TsZxQ+A4jpNx3BA4juNkHDcEjuM4GadmDUHblq38bc5i7pi/rtqqOI7j9Gpq1hDUSZz/x9lcPWsN6zdtqbY6juM4vZbaNQR1YsfhAwB4bfWGKmvjOI7Te6lZQwCw07CBACxetb7KmjiO4/ReatsQDA85Ol5b5SMCx3GczqhpQ9A0PIwI3DXkOI7TOTVtCHKuIR8ROI7jdE5NG4LciGCxGwLHcZxOqWlDsNNwHxE4juOUIhOGwEcEjuM4nVPThmDskAHUAUvXbmRT29Zqq+M4jtMrqWlD0FBfx4hB4SMuWeOjAsdxnGLUtCEAGD2oHvB5AsdxnM7IgCEIH9HnCRzHcYpT+4ag0UcEjuM4XVHzhmBU5BryEYHjOE5xat4Q5FxDr632xHOO4zjFyIAhcNeQ4zhOVzRUsnBJLwBrgC1Am5lNljQKuAEYD7wAnGxmKyqlw+jGaETghsBxHKcoPTEiOMbM3mxmk6PjrwL3mdkk4L7ouGKMGhhGBK+v2ciWrVbJqhzHcfok1XANfRC4Nnp9LXBiJSvrVy/GDOnPlq3G0rUbK1mV4zhOn0RmleslS1oIrAAMuNLMrpK00sxG5F2zwsxGFpE9BzgHoKmpqXnq1KmpdGhtbeWiaa0sWNnG948bxaRR/RPLNzY2pqrb5V3e5V2+mvKTJ0+emeeN6Rwzq9gD2Dl63gF4AngbsLLgmhWlymlubra0tLS02Fm/nW7jLrzd7pjzair5cnB5l3d5l6+WPNBiMdrqirqGzOzV6HkJcAtwKPC6pCaA6HlJJXUAT0ftOI7TFRUzBJIGSxqaew2cADwF3AacEV12BnBrpXTI0RTtXbzYt6x0HMfZjkqGj+4I3CIpV88fzOxOSTOAGyWdBSwCPlJBHQDfstJxHKcrKmYIzGwBcFCR88uA4ypVbzF8y0rHcZzOqfmVxeBzBI7jOF2ROUNgFQyXdRzH6YtkwhA09m9g2MAGNm3ZyvJ1m6qtjuM4Tq8iE4YAtkUOveaRQ47jOB3IjCHweQLHcZziZMYQeOSQ4zhOcTJjCHxE4DiOU5zsGIJhPiJwHMcpRnYMQW5E4FtWOo7jdCAzhqA9ashHBI7jOB3IjCHYKW+y2BeVOY7jbCMzhmDYwAYa+9fTumkLaza2VVsdx3GcXkNmDIEkjxxyHMcpQmYMAXjkkOM4TjGyZQjaRwQeOeQ4jpMjU4agqd0QbKyyJo7jOL2HTBmCndoTz/mIwHEcJ0emDEGTzxE4juNsR6YMgUcNOY7jbE+mDIFnIHUcx9meTBmCUYP707++jlXrN7N+05Zqq+M4jtMryJQhkMSOwwcAvlOZ4zhOjkwZAoCmYSFyaLGvJXAcxwEyaAh8wthxHKcjmTMEPmHsOI7TkcwZAh8ROI7jdCR7hsAXlTmO43Qge4bAt6x0HMfpQOYMwbYtKz3xnOM4DmTQEIwdOoD6OrF07UY2tW2ttjqO4zhVp+KGQFK9pFmSbo+OJ0h6TNI8STdI6l9pHfKprxM7DA2Lyl73RWWO4zg9MiI4D5ibd/wD4CdmNglYAZzVAzp0YNs8gRsCx3GcihoCSbsC7wWujo4FHAvcHF1yLXBiJXUohkcOOY7jbENmVrnCpZuB7wFDgS8DZwKPmtnE6P3dgDvMbP8isucA5wA0NTU1T506NZUOra2tNDY2djj3m9mr+eu8Vj5+4FA+uPfgxPLl1u/yLu/yLt8T8pMnT55pZpNLXmhmFXkA7wN+Gb0+GrgdGAvMz7tmN2BOqbKam5stLS0tLdudu/LB+Tbuwtvt4tueTiVfbv0u7/Iu7/I9IQ+0WIz2uiG1qSnNkcAHJL0HGAgMAy4DRkhqMLM2YFfg1QrqUBTfstJxHGcbFZsjMLOvmdmuZjYe+CjwdzM7DbgfOCm67Azg1krp0Bmeb8hxHGcb1VhHcCFwgaT5wGjg1z2tQG6y2PMNOY7jUFHXUDtm9gDwQPR6AXBoT9TbGTtGhmDJmo20bdlKQ33m1tU5juO0k8kWsH9DHWOG9GfLVmPp2k3VVsdxHKeqZNIQgC8qcxzHyZFdQzAsl3zOI4ccx8k2mTUEHjnkOI4TyKwh8J3KHMdxApk1BD4icBzHCWTWEPhaAsdxnEB2DYFHDTmO4wBuCHht1YZcAjzHcZxMkllD0Ni/geGD+rFpy1aWr/NFZY7jZJfMGgLwCWPHcRzIuCHwEFLHcZysG4LclpU+Yew4TobJtiFoHxF4mgnHcbJLpg1BU7sh2FhlTRzHcapHpg2Bb1npOI6TcUPgUUOO4zgZNwS+qMxxHCfjhmDogAYa+9fTumkLqze0VVsdx3GcqpBpQyDJ1xI4jpN5EhsCSSMlHVgJZapBkyefcxwn48QyBJIekDRM0ijgCeAaSZdWVrWewbesdBwn68QdEQw3s9XAh4FrzKwZOL5yavUcHjnkOE7WiWsIGiQ1AScDt1dQnx7H5wgcx8k6cQ3BxcBdwHwzmyFpD2Be5dTqOdrzDbkhcBwnozTEvG6xmbVPEJvZgpqZI/ARgeM4GSfuiODymOf6HB415DhO1ulyRCDpcOAIYKykC/LeGgbUV1KxnmLU4P70r69j1frNtG5qo7F/3EGS4zhObVBqRNAfGEIwGEPzHquBkyqrWs/gi8ocx8k6XXZ/zexB4EFJvzWzF5MULGkg8BAwIKrnZjO7SNIE4I/AKOBx4HQzq+qmwTsNH8ii5a28tmoDe4wdUk1VHMdxepy4fpABkq4CxufLmNmxXchsBI41s7WS+gEPS7oDuAD4iZn9UdKvgLOAK1Jp3034WgLHcbJMXENwE/Ar4GpgSxwBC+k810aH/aKHAccC/x6dvxb4NlU2BLkQUp8wdhwni8Q1BG1mlrixllQPzAQmAr8AngdWmlku1efLwC5Jy+1udmofEXiaCcdxsofi5OGX9G1gCXALweUDgJktj1WJNCKS/RYhRcXE6PxuwN/M7IAiMucA5wA0NTU1T506NU5V29Ha2kpjY2OX1zz68gb+55GVTNl5AF89cmRi+XLrd3mXd3mXr4T85MmTZ5rZ5JIXmlnJB7CwyGNBHNm8Mi4CvgIsBRqic4cDd5WSbW5utrS0tLSUvGbWohU27sLb7b0/eyiVfLn1u7zLu7zLV0IeaLEY7XMs15CZTUhqiSSNBTab2UpJgwhJ6n4A3E8IPf0jcAZwa9Kyu5smDx91HCfDxDIEkj5e7LyZ/V8XYk3AtdE8QR1wo5ndLukZ4I+SvgPMAn6dUOduZ8yQAdTXiaVrN7GxbQsDGmpirZzjOE4s4k4WT8l7PRA4jrAGoFNDYGZPAgcXOb8AODSBjhWnvk7sMHQAi1dtYMnqjew2Kr1PznEcp68R1zV0bv6xpOHA7yqiUZXYafhAFq/awOJVG9wQOI6TKdLuWdwKTOpORaqNJ59zHCerxJ0jmEpYDAYh2dy+wI2VUqoa+JaVjuNklbhzBD/Ke90GvGhmL1dAn6rhaSYcx8kqsVxDFpLPPUvIPDoSqGqSuErgGUgdx8kqsQyBpJOB6cBHCPsWPyapJtJQ59jJRwSO42SUuK6hrwNTzGwJtC8Wuxe4uVKK9TTtiefcEDiOkzHiRg3V5YxAxLIEsn2CHSND8MbajbRt2VplbRzHcXqOuCOCOyXdBVwfHZ8C/K0yKlWH/g11jBkygKVrN7J07aZ2V5HjOE6tU2rP4onAjmb2FUkfBo4CBDwCXNcD+vUoTcMHsnTtRhavWu+GwHGczFDKvXMZsAbAzP5sZheY2RcJo4HLKq1cT+ORQ47jZJFShmB8lDOoA2bWQti2sqbITRh75JDjOFmilCHoyj8yqDsV6Q3s5GkmHMfJIKUMwQxJZxeelHQWYQvKmsL3JXAcJ4uUiho6H7hF0mlsa/gnA/2BD1VSsWrgcwSO42SRLg2Bmb0OHCHpGGD/6PRfzezvFdesCjQND96uxas98ZzjONkh7n4E9xO2mKxpcpPFr6/ayNatRl2dqqyR4zhO5amp1cHlMqh/PSMa+7Fpy1aWt9ZcXj3HcZyiuCEowHMOOY6TNdwQFOBZSB3HyRpuCArwLSsdx8kabggK8C0rHcfJGm4ICvAtKx3HyRpuCArwRWWO42QNNwQFuCFwHCdruCEoID9qyMyqrI3jOE7lcUNQwNABDQzuX8/6zVtYvaGt2uo4juNUHDcEBUhy95DjOJnCDUER2pPPeQip4zgZwA1BEXxE4DhOlqiYIZC0m6T7Jc2V9LSk86LzoyTdI2le9DyyUjqkxbesdBwnS1RyRNAGfMnM9gXeAnxO0n7AV4H7zGwScF903KvwEYHjOFmiYobAzBab2ePR6zXAXGAX4IPAtdFl1wInVkqHtHi+IcdxsoR6IlZe0njgIcIuZ4vMbETeeyvMbDv3kKRzgHMAmpqamqdOnZqq7tbWVhobGxPJLFy5mS/fs4zdhzXw3bc2JpYvt36Xd3mXd/nukJ88efJMM5tc8kIzq+gDGELY7/jD0fHKgvdXlCqjubnZ0tLS0pJYZtnajTbuwtvtgIvuTCVfbv0u7/Iu7/LdIQ+0WIx2uqJRQ5L6AX8CrjOzP0enX5fUFL3fBCyppA5pGNnYj/4Ndaze0Mb6tq3VVsdxHKeiVDJqSMCvgblmdmneW7cBZ0SvzwBurZQOaZHUHjm0fL0bAsdxaptKjgiOBE4HjpU0O3q8B/g+8A5J84B3RMe9jlzk0LLWLVXWxHEcp7I0VKpgM3sYUCdvH1eperuLXOSQjwgcx6l1fGVxJ7SPCNb7iMBxnNrGDUEnNA1zQ+A4TjZwQ9AJO0WJ55a1umvIcZzaxg1BJ7hryHGcrOCGoBOa2g2Bjwgcx6lt3BB0wpghA6ivE6s3bmX1hs3VVsdxHKdiuCHohPo6cej4UQBc8/AL1VXGcRyngrgh6ILzj58EwNX/WMDK1k1V1sZxHKcyuCHogsP2GM1BO/ZnzcY2rnxoQbXVcRzHqQhuCEpw6v5DAPjttBd4Y83GKmvjOI7T/bghKMGkUf05ft8dWb95C798YH611XEcx+l23BDE4Esn7AXAdY8u4tWV66usjeM4TvfihiAG+zYN430HNrFpy1Z+fr+PChzHqS3cEMTk/OP3ok5w44yXWLSstdrqOI7jdBtuCGIycYchfOjgXWnbalx237+qrY7jOE634YYgAecdN4mGOvGXWa8wf8maaqvjOI7TLbghSMDuoxs5ZcpubDX4yb3zqq2O4zhOt+CGICGfP3Yi/Rvq+OuTi3nm1dXVVsdxHKds3BAkpGn4ID522DgALr3nuSpr4ziOUz5uCFLw2aP3ZFC/eu6du4RZi1ZUWx3HcZyycEOQgrFDB/CJI8cDcOk9HkHkOE7fxg1BSs552x4MHdDAP+Yt5dEFy6qtjuM4TmrcEKRkRGN/PvXWPQD48d3PYWZV1shxHCcdbgjK4JNHjWdkYz9mvLCCh+YtrbY6juM4qXBDUAZDB/bjM2/fE/BRgeM4fRc3BGXy8cPHM2bIAJ58eRV3P/N6tdVxHMdJjBuCMhnUv57PHxNGBZfe/S+2bvVRgeM4fQs3BN3AqYftzs7DB/Lc62u4fc7iaqvjOI6TCDcE3cCAhnrOPS5sdH/ZPf+ibcvWKmvkOI4THzcE3cRJzbsybnQjC5au48+zXqm2Oo7jOLGpmCGQ9BtJSyQ9lXdulKR7JM2LnkdWqv6epl99HecfH0YFP713HpvafFTgOE7foJIjgt8C7yo491XgPjObBNwXHdcMHzhoFybuMIRXVq7nhpaXqq2O4zhOLCpmCMzsIWB5wekPAtdGr68FTqxU/dWgvk5c8I6w0f3P/z6PDZu3VFkjx3Gc0qiSi6AkjQduN7P9o+OVZjYi7/0VZlbUPSTpHOAcgKampuapU6em0qG1tZXGxsZUsmnkt5rx/+5dxsKVbZxx0FCO31U9Wr/Lu7zLu3yOyZMnzzSzySUvNLOKPYDxwFN5xysL3l8Rp5zm5mZLS0tLS2rZtPL3zX3Nxl14ux18yd32j0en93j9Lu/yLu/yZmZAi8VoYxtSm5p0vC6pycwWS2oClvRw/T3CMXvvwMG7j2DWopX85dl17L3fxtRlrdywhTfWpJff2OYL3BzH6ZqeNgS3AWcA34+eb+3h+nsESXz5hL057erHuHnuOm7+7r3lFTg1vXyd4ICWaRw2YRSHjh/FlPGjGN7Yrzx9HMepKSpmCCRdDxwNjJH0MnARwQDcKOksYBHwkUrVX22O2HM0p0zejTvnvEK/fum/5s2b28qSX75uE0+8tJInXlrJVQ8tQIK9dxwaDMOE0UyZMJIdhg5MXb7jOH2fihkCMzu1k7eOq1SdvQlJ/OCkAzl5wmaam5tTlzNz5syy5B9+bAY2ajzTFy7nsYXLmf3SSp59bQ3PvraGax95EYA9xgzm0Amj2h+7jkw/OeU4Tt+jp11DTg8zqKGO5kljeeuksQBs2LyFJ19exfSFy3hs4XJmvriCBUvXsWDpOv44I6x92GXEIKaMH8mhE0YzuLWNQ8yQVM2P4ThOBXFDkDEG9qtv7/l/HmjbspWnX13dPmKY8cJyXlm5nldmr+cvs18F4JKH7+0wYthnp2HU17lhcJxawQ1Bxmmor+Og3UZw0G4jOPtte7B1q/GvJWuCYViwnGnzXmfZuk3c8dRr3PHUawAMHdjAlPHbDMMBuwynX72nrXKcvoobAqcDdXVin52Gsc9Ow/j44eNpaWlh9Ph9211J0xcu5+UV6/n7s0v4+7Mh+ndQv3oOGTei3TgcvNtIBvWvr/IncRwnLm4InC6RxIQxg5kwZjCnTNkdgFdWrmdG5EqavnAZz7+xjmnzlzFt/jIA+tWLA3cdwaETRjGqbSOTNmxm2EAPWXWc3oobAicxu4wYxC4H78KJB+8CwNK1G/MMw3LmvraamS+uYOaLKwD43rS72W/nYRw6fjSHThjFlPEjGT1kQDU/guM4ebghcMpmzJABvPuAJt59QBMAq9ZvZuaLwTA88NRLPL+ijadeWc1Tr6zmN9MWAjBphyEdJqCbhg+q5kdwnEzjhsDpdoYP6sex++zIsfvsyAk7tLLvAQcxe9HK9hHD44tWMG/JWuYtWct1jy0CYPdRjcEoRPMM40Y3esiq4/QQbgicitPYv4EjJo7hiIljANjYtoWnXlnVbhhaXljBouWtLFreys0zXwZgh6EDOHTCKIZvXcuMtc+nrvuVl6srv+z1VgY2rfKQW6dX44bA6XEGNNTTPG4UzeNG8R9Hw5atxtzFq9snn2e8sIIlazZy+5OLg8BTz5ZX4Zzqyv/v4w97yK3Tq3FD4FSd+jqx/y7D2X+X4Zx11ATMjOffWMujC5YzY+5Cdtpxx9Rlv/b661WVf/bFV3l+tToNuc1NoB+8+wgG9vOQW6c6uCFweh2SmLjDUCbuMJR9+y2luXnf1GXNnNnaC+SbY4XcHhSF3B46YRTN40Yy1ENunR7CDYHj9ACFIbdvrNlIywsdQ25bXlxBy4sr+OUDz1MneNPOw9mx/yZ2emlO6nrfeGMVYzMsv2r5ahb3e5VDx49ih2GeZbcz3BA4ThUYO7TzkNvpC5cz5+VVzHllFXMAFi4qr7Lnsy0/9V+zAJgwZnB7VFrIsjvII9Mi3BA4Ti8gP+QWoHVTG7MXreT+mXPZfffdUpe7aNFLmZZ/ev6LvLp5EDNfWM7CpetYuHQdN7SELLs7Dx8YGYUwT7Pn2MGZNQxuCBynF5ILuR2wqpHm5vGpy5nZf5nLNzfTtmUrzyzumGX31VUb+MvsV9uz7I4e3D9a+R5GDPs2DUtdb1/DDYHjODVPQ30dB+46ggN3HcGn3hqy7M5bsrZDMsUlazZ2zLI7oIHxw+sY8+T01PWuWrWa4WXKnzt4Ccfss0PqMuLghsBxnMxRVyf23mkoe+80lNMPH4+Z8eKyVqYvXM70F4JhWLS8lTlLgCVvlFfZa+XJv7JyfXn1x8ANgeM4mUcS48cMZvyYwZw8JcxJLF61ntv/MYs9J+6Zutz58+czceLEsuSP3ntsavm4uCFwHMcpQtPwQRzSNIDmfdIvKBy+7uWy5XtiD3Ff4+44jpNx3BA4juNkHDcEjuM4GccNgeM4TsZxQ+A4jpNx3BA4juNkHDcEjuM4GUdmVm0dSiLpDeDFlOJjgKVlVO/yLu/yLt9X5ceZWekVaWZW0w+gxeVd3uVdPovycR/uGnIcx8k4bggcx3EyThYMwVUu7/Iu7/IZlY9Fn5gsdhzHcSpHFkYEjuM4The4IXAcx8k4bggcx3EyjhsCx3HaUWC3auvR25DUr9o6VJKaNASSJsQ5V6KMekk7S9o99+g+DTut86+STpM0uNJ1ldBjsKS66PVekj6Q9I8gaRdJR0h6W+6RUpeRkg6Mee0+ea8HFLz3ljT1l4ukPSV9Q9JTZZTxDkn3JLg+9e9nIXrkLylVzddhnKTjo9eDJA1NIFv2/7c7iIzisZKuBl5OWUadpGHdrFr30xOr1nr6ATxe5NzMBPLnEpZ1Pw3MiR5PxpT9ITAM6AfcF5XzsZiyHwSuB5YANwAnAv1TfP7zIh0E/Bp4HDghgfxMoBHYBXgJuAW4LoH8D4AXgL8BU6PHbQnkH4j0HwUsivS5NMnvXngPFLsnuijnw8A8YBWwGlgDrE4g3wScD0wHNgAXAQfEkDsW+BewFvg9sB/QEn3+D/fg7/cLYErS+y5P/mxgBvB8dDwJuC+BfFn/307KvCrBtYcBP43uvbXAGcDIBPJ/iO7fwcCzwGLgKwnkd4z+t3dEx/sBZ5Xz+UvWWcnCe/oB7AP8G/B89GfOPc4Enk5QznxgdEodZkfPHwKujRqzJxKWMQg4JfoDvwb8BnhHAvknoud3ArcBByVsCB+Pns8F/l/0elYC+eeAAWX8jrOi508BF0evSxrifB0L9U2o/3xg3xR6nw38PWrMvwMcCCxM8rmBo4EBhE7AauC8FHqU+/s9A2yJ/kdPkqAjFMnPBvoX/B5zYsiV9f+N/mvFHqOBl2PIf5fQAbgvuvdGJ/n98j9/9HwacCmhU5jk+7sDODnvf9wQ5/sr51Frm9fvDbwPGAG8P+/8GsKfNC4vEXqDacgNwd8DXG9myyUlKsDM1hNGBDdEbpFrCb2S+phF5Cp8D3CNmT2hZEpI0uGEG/ms6FySe2UB4XvYmEAmnwZJTYQ/w9cTyFknr4sdd8XrZjY3wfU5fgE8Avy7mbUASEpSr5nZA9Hrv0h6w8x+mkKPcn+/d6eoM5+NZrYpd8tJaiDe91/u/zeXnDL/XrfoeIcY8ucQOjFXALeb2YaEv1+OfpEr7kTg52a2OWEbMMbMbpT0NQAza5O0JYUesakpQ2BmtwK3SjrczB4po6gFwAOS/kpeY2Zml8aQnSrpWWA98B+SxhLcA7GRtCOhEfwowc1wE/CJBEXMlHQ3MAH4WuSf3ZpA/nzga8AtZva0pD2A+xPItwKzJd1Hx+/vCzHlLwHuAqaZ2Yyo/nkx5HaV9DPCHz/3muh4l9jaQ4ukGwi+8nz9/1xCbmfgI8Cl0W94I9s6BnEYIenDecfKP45Rf46yfj8ze1HSUcAkM7smuoeHxJUHHpT0n8AgSe8A/oPgHixVb7n/3wXAcWa2qPANSS/FkN8JOAE4FbhM0v2Ez9BgZm0J9LiS4Bp9AnhI0jiSdSzXSRpNZDyj+a20HdNY1OTKYkm7ApcDRxK+zIcJQ+xYEz6SLip23swujik/kuBT3iKpERhmZq/FkDubcBPuTXALXW9m0+LUWVBOHfBmYIGZrYxuql3M7MmE5Qw2s3Up6j+j2HkzuzZpWd1Rb9L6JV1TXNw+mUCXXQmG/FSCv/4WM/vPFPWmqj8qL+3vdxEwGdjbzPaStDNwk5kdGVO+jjASOYFghO8CrraYjU1kuH4KvIXw/30E+KKZLSgh9zngYTN7osh755rZ5XHqj64fSBidnAocRZjj+PeYshPMbGHesYCJZhanM4OkQwjt1/7AU8BY4KSk/99EVNLvVK0HcA+hB90QPc4E7klRzlBgSEKZjxd7xJS9hvDnqeuG72AX4AjgbblHAtnDCX7iRdHxQcAvE9bfP7qR9wf6JZTdi+CnfSo6PhD4RkzZsYRGbES178M8nfYGLopxXewJ4Ur+fgQfv+jo40/i4x4M1Ocd1wONCeQfBU7P+/9+DHisir/fUOCMBNeXPdkdfe43pfn/pHnUlGsojx3MLL939VtJ58cVlrQ/8DvCRBOSlhIa86djiE/Jez0QOI4QtfN/pQTN7BORP/W9eaGQc4E7LcHQVNIPCJPNuUk/CD2rh2IWcRnbJpqxMMcQO/xT0tGEeY0XCA3KbpLOMLO49f8v8BXCEBsze1LSHwgTsF3V+yngvwmTjRMknWNmt8XVO6+cVCPKEt9RHNfMN4C47p+uKOv3AzaZmeX840oeznwfcDwh4gZC8MPdhI5JHGRmv8s7/r2kzyfUoWOB0jvMrMsQXEkf7+LtkqOZ6D/7JmB4gYtvGKEtiEWBLMBeklYRJoyXxC0nCbVqCN6Q9DFCKCaE4d2yBPJXAReY2f3Q3rD9LzFuZDM7N/9Y0nCCUSlJNAS/nxBuNovQiL4P+LGkY8zs1Zj6n0h/DpSYAAAcmUlEQVQY1qedrMXMXiqY4EoyWfVjQrjqcxBi2Qm/RXNM+UYzm15QfxxDeD7wJjN7I3IvXEfUGCbkGkII4Eei449F595RQu4rRc4ZoUe+K/En+8umzN/vRklXEuYszgY+Sbj/4zLQzHJGADNbG7lI43K/pK8CfyR8f6cAf5U0KipveYKycvwaKLUWaEqRcyJMXO9C6c5cdwWrnEUY1eU6D0cTRkl7SbqkwEh2C7VqCD4J/Bz4CeFG+md0Li6Dc0YAwMweSNErytFKiKOOw38DV5jZZfknJX0B+B4hcigO5UbtvCTpCMAk9Qe+QBiZxKVfzggAmNm/lGxB2lJJe7JtsuwkgnEsxSYzeyOqc4EKFpUlYGyaEaWZ5f/5iSZcv07QPU6Pdh9JxfzACsVbrIV1lPn7mdmPokne1YTG7VuletMFrJN0iJk9DiCpmRA8EZdToudPF5z/JOGe2KOYkKTOjL4IoaBdkt+Ji/z6pwEXEhrh78aQ765gla2E8OXXI112JEQyHUYY1bshiIOFqIEPlFHEAknfZNsX/jFgYRfXtyNpKtuGkXWExSA3xqz3LWZ2ZuFJM/uZpOeKXN8Z5UbtfIYwWbcLYUXl3cDnEtTfIunXbPv+TiMscorL5wijsn0kvUL47j8WQy4/Umi74wSff2k5I0pJxwHfJNwH/52gEV1Ix55kWsr9/TCzeyQ9RtRGSBqVoCd+PnCTpNwItoltjXucutOuIn4r4T5ZW3BewKFxCohcs2cCXwIeI0zSJvnvAcyPoqbGk9fGWvzJ/vE5IxCxBNjLQij65oS6xKImDUEU7nY26X+ITwIXE/y1IljhuOGbP8p73Qa8WMq3nEdXvabWmGVAcIekcYkAYGZLCY13Wj5LaHi+wLbv75cJ6l8AHB+NwurMbE1M0ULXTBLjk0+qEaWk9xJGAKuAr1vyiK+NZvZiQplCHeqB080s9e8n6dOEEN71hN6p6KInXoiFkN99CKMJAc+aWckGTNKxZvb3Ij7yXLml5k8eBVrN7MEiZZdszKOoo/MIcxzvKuO3uBX4B3AvyVxyOf4h6XZC2DiERXYPRf+HlSl16pJaDR/9J+GHmEneD2Fmf6pwvfXAXWZ2fEr5BcCXi70F/NDM9kxQVn9C9A3Ac3H+iHmyEwirUsfT0ZDGGmVFN+wGM9sSHdcTVhp3acwkXdDV+xZvHUexcgcC7zezm0peXAaSthJ64E9QZHKx1PcnaS3wrXwRQoqShy0vHDGGHg+Y2dFxry8iPw84POoQpC3jCLa/f7r0sUu62MwuShu+K2kSIVBkWsH5twKvmtnzJeS3Enrfb9Dx90vkmpM028zeHOfaTuRFaPyPjOp+GPiTVbCxrskRAWGy8cKkQpIuM7PzC9w77ZT6I1tYN9AqabiZpVkA8iCduwbiRtx0R9TOXwiTa1NJthAtR9qokVxisr0JE3e5Uc37SfD5od345BYHvZPQMejSEEj6f2b2Q0mXU/z3L+VaOiaJjkX4H7Z9BznGA1+X9G0z+2PMcqZJ+jlhdXr7OoKczz4Gz5NsBNoBSb8D9iSEoeZHrXVpCCIjUEfIsRPXnZrPT4BiazXWEyKpSrnduiux3e2S3mNmf0sjHDX4N0ePHqFWRwTfAf6Z9IeQ1GxmMyW9vdj7xYacRcq4kbAQ5h46/gnj+qfLRtJMQpqDDlE7ZhYrakfSY2Z2WBn1b9cjStJLUlgV/W85l5DCyuibzOxdMWTfBvw78F5C0rcjgT1KjUYi2feb2VR1w4K4yD1JbvK6HKJomXvN7JCY1xcLVTUzOzam/MGEKKnHSDHHJGkusF/aHqykh8wscbZaSU+Z2f6dvDfHzA5IqU898FEzuy7m9WsIayk2RY/ciCJWFlKFlcSXA/sS1uPUA+viyqehVkcE5wH/KWkjsJmYP4SZ5XzKb7aCHC+SziP02Evx1+iRmG50jZQbtfNThdWld9OxIYjboyw3amR3wh8oxyZCz7hLJL1MyBh5BSHb4xpJC+MYAQAzmxo9p1oBHQ3pLyJECAmok9QGXG5ml6QpM9JneVR23OvLHZlcSUieN4d0I8KnCOka4kR6FeMeSV9m+xFNqcnqrmL1B5WqVCFd9OcIk+y3ETpznye4a2cTwpFLYmaxU253ws8Jq9JvIiyO/Dgwscwyu6QmDUGxHyLJH4kQplmY7OvMIueK1V1OGoUfEW64OwgNcLJsddsoN2rnAMLKzmPZ1hBYdByHsqJGCHpPl3RLVO+HiLEgD/gTYQ3FKcAWSbeSLNkcsF3kV45VhJTQV5pZZ7mjzieMQKbkfPoK6xmukPRFM/tJUl2iMo4FViS4/lvFzicwRm1m1mWnpARjgGckTadjRyJuJF9uLiA/0inOZPUMSWebWYc1D5LOIt79/zvC9/wIIfvoVwg98g+a2ew4ikf15UJPJ5jZfyls9NNkZtPjlmFm8yXVR/Ns10TznhWjVl1Dl5jZt/KO64DflYqkkHQqwa1wFMGnnGMosKWrSWBJN5rZyZLmUNy/XHKiSdKbCT2BdxFu3OsJOU4S/UgK8fOfiz5He9SOxVxgppA070Az21Ty4s7L6EfCqJEC+UMI4YAAD5nZrJhyIvjqTyVkXx1GWKDzN8tb5FSijJ8SUlXkwkdPIaQDH0TIG3V6J3KzCOnClxacHwvcbWYHl6i32L0zCniVsLL92Zj6fynvMJczZ26pydY8+e8SsnhOpWNDHit8tBzXajkoxNvfQhhB5hr+yYTG/ENWIt9XvvsocgctBXa3+FFruXKuIHSgjjWzfRVyj91tZsUWrBWTf4gwx3Y14b5bDJxpZgcl0SORzjVqCH5LiJT5XtQo3kTI//HtEnLjCBNG3wO+mvfWGkKulU5Xt0pqMrPFkXtnOiGVdTuWMBQtiro4lXBDXGgJUiWkjdrJk78BONcSLmdXmeF/koaZ2erIJ15MPtGK0sgYvZtgXE8wszEx5bbzUefOSXrazN7UiVxXPupO38u7ZlzBKQOWWYrEcQXlDiBsDPTOmNcXi1AyM4sVPhqVMY6QvfRehVXF9XEbVIUwzuvMbGV0PBI41cxihSBLOoaQowfCPgZ/jyn3eP48TOFxXHJykmbljL+kJ+I25NF39zrBgH0RGA78wkpEPZVDTbqGCDH/1ynk8z6GEIVQclgeNdYvEpZ3J8LMcv7QoQQf63LCEvmbrePikJJEPciDCS6alwkhbUkoN9fLjsCzkmaQbGj/doJvuVh0hlE6j84fCL3XmRQJ3yNmHHt7hWEUchtwm6SSPuI8xkra3aJ0xgrblOaMSFejpLTvAck7CwloJMF3Z+kXdAGgkJbiHMJoZk+Cz/1XhLxbcTjbzH6Rp8+KqMxYhsBCVoAkadNzHCRpdfRahBTUq0k42QtsjjpfuZXxY0k213JiNEe5gbCeKTdHmWZviljU1Iggcifk6EdokKcRQiFjT3Z2x6y9woYypxDigV/uyq2UJ/OJSGYgIXTsxqS98qiccqN2qjK0L5fO3HI54rjnonLeQ2i4nic0AhMIOfUfIDRSl3Uit4W8yc38twj5d3pkA/SC76Ge4Ob6LyuRhrncEV1eObMJK3kfy+sRx47aUUizcVDOJRo1qk92NhLrbUg6jfA/PoQQxn0SIXturHUsxUYi+aOLSlBrI4IfFxyvIKR4+DHJJju7Y9Z+CcG/t4x4uyNBMFhzCJEv7wROyJ/jTjDZVlbUjpk9GPlbcz7N6XEMkrop6knSfWZ2XKlzRXhf7nJC5NZ74tRXiJn9TWFx0j5sm+PITRAXNQKRXI8llSvB+/JetxF2XIuTtK/cEV2OtDuU5biLkPjuV5HcZ4A7E8hXFTO7TiGE+zjC/XOixdjxLm+OcoI65k0aSrKkmYmpKUNg5YfN5ZeVatZe0mcJvYGxhF792Wb2TMxqu0v/sqJ2JJ1MWNz0AOFGvlzSV8ys1AKXssLmFFYANwJjIr9wzgoOI+z+1SX5rhVJqdM1RD7tC4BxZna2pEmS9jaz29OUVwW+UzihLel3nU1y57DyF3TleFApdijL40KCa+mzhHvgbsLEaV/idULASQPhe2jvmHXBPwkTw2Po2KldQ9g7umLUlGsoh0Lq54sIG7JAiP+/xGKu9i1n1l7S94E/WoJwszzZdr90uZQTtSPpCUL0y5LoeCxhQVPFohaies4jGLGdCZEyOVYD/2tmP09QVqqJvkj2BsI8xcfNbP9ofuGRuK61alNk0rOB4FrZL6Z8qgVdefJl7VBWUNYoYFer5O5c3Yyk/yKEmz/PtpGQWcwFfdWgVg3BnwiLWnIx/acTfI5FfZ9F5IvN2v/SzOZXQN38etv/wJL+ZGb/llC+u3y8Hfy50R/7iQQ+3lRbDebJJ9pWME8uv+G/joLEeQnmiFrMbHLaqI9qEQVH/CchOCAXISbCRPVVZva1mOV8k+BKTLqgq1hZiRtySQ8Qsgc3ENbVvAE8aOWtbegxFBLcHWApw6+j/+8PCC5lkXyyOjE15RrKY8+CRvTiaAIrLksJue03RLL1QNrc9knIX0CWKEImort8vHdKuouOcfRJ0nX8AfgFYSEYhPmW6wn51ONwpcIeDLle6QOEhVylRjW5uSARRnI/Kng/bo9sUzQKyE1W7kn6vR16DDP7HvA9Sd+L2+h3Qi7v/38UnI91TxZryCUlaciHWwgj/hRwTeSy6jMjAkIndATJo/1y/JCQJDHJHiBlUauGYL2ko8zsYQBJR5IsxUG54ZdpsU5exxM2uyh6jpsyu7NyviIpP/vhVWZ2S4Iiyt1q8JeEqK9cuODphLQRnyohdyHwUi6UVyFn0L8Rku99O07FCjOcvyJMTu4m6TrC93BmAv2rzXTlJT6UNAI42sz+ElN+P4IROIpwH/6D8J3EpdyGvEFSE3AyIa13X+N7wCxJT5FuZfXrPWkEoHZdQ28muIWGExqy5QQf/xMx5csKv0xLXvih2H54X3Jo2F1RO+USzZOspONWgwMIo4SSLoZibpg4rhlJjwPHW8jN87ao/nOBNxN2fDoppv4zCf7ttxC++0etjJTMPU0n92/s8EOFxImr2ZZb51RghJmdHFN+DuH7u5awL8MMSU8mCN/9CGFjn2lm9tnI1fg/SV2l1ULS04TQ9Q65mixm+LXCyvadCFmA8w1Jd+xnXZSaHBFEE7UHKSSRwsxWlxAppNykaanohvDDcpNdAd3io0y11WAeWyTtadFKyqghiLPBR32ekTmFMJL5E/CnhK7BRwkZS1MlD+wF1BU5l+S/vneB0b0/CiCIyyWECeJpkRHYA5gXV9hCvP1NeccLCCO7vsJSM/tZ6cs6ZRihE3hC3rkkrt3E1NSIoBvj2KcQepMdwi9tW3bSmkbSfHrYR1lQ/3GENMgLCEZoHPAJy9tHuhO5pwiZY9sU8iWdY9EeDIqR4iGvnGcIm/q8yLYRmsXt0VYbSb8hjMh+QWhAzgVGWpFtUDuR/y3wKzN7NDo+DDjDzArnDCqCQtr0K4Ado6itA4EPmNl3eqL+cpF0KaEnfxvpsvf2OLVmCC6KXhbd2MTMSvmY88sqK2laNemGqJ1pZnZkGfXn4vB3N7NzFBZnJYrDV8iPk//9l5yslfR1wiKypYRU1oeYmUmaCFwb9zNp+5w/QEVTQHQrCrmmvkmY58rF4X/HYuYsUthPYG/CwkYI3+VcgpujpEEstyGX9CAh8+eVeVFbsQ15tVH5+0EMJITfvom81NoWf6vdxNSUIcihlBubdFf4ZbWR9CihN5iL+vkoIYlcrKidcn2U5cbhq8iCLmIaEoX0IE2EbI/ronN7AUN6c4+sN9GZIcxRyiCW25BLmmFmUwrCdys+R9dbkHQT8CxhlfElhDDouWZ2XqXqrMk5AlJubEL3hV9Wm3Kjdsr1Ue5pZqcoLJnHzNZH0ThxuYZgSHLJ/14m+IxLGoKcO6Pg3L8S1N3niXqkxVKhx+qRdsPIp9HMphf85HFSXORYGoXs5sJ3TyL9Jjc9hqSPmdnvO3NRJwjWmGhmH5H0QTO7VtIfCHMuFaNWDUGqjU26K/yyF3C/pK/SMWrnr9HinpJRO93w+cuNwy/XkGSdL+e9HkiYaE3SEJdLuQ3554CrgH0kvQIsBD7W7Vp2P4Oj53KDNnJu6JWS9iesiRlfZpldUpOuIWhfZZpoY5PeEn5ZLiqeTz6HWYm88uX4eKMG+3SCj3M/gn/6SEL47gMx9f8nIWHXNAt53fck7Ll8aBx5Z3sUFnQVzSpbgbr2IDTkRxASPy4ETks60ojmOuos4cYw1URh8ekXLOVudFEZnyLstncgYXQ8BPimmV3ZPVoWqbMWDYFC/vjtsBJ5fPImm4tiZheXo1dfoRt8vGXF4SskKvsGKQ1J1lHHjX3qgGbgZ2a2dw/UXQecZGY3pm3IVf5Wm1VF0v1WRgJMbUt22WPUqmvor2zzkQ4CJgDPEWbhO6VWGvpuiNop18dbVhy+md0TLQ7LGZLzkhgSp31jHxF+t4WEEVrFMbOt0XzUjXGjlIqQL9e+1WbZyvUc/5T0c7bP1RQ3WGGhpDsj+b9bD/TWa3JEUEjkJvq0mRUucOrs+rLCL6tNN0Tt3AF8nhBpdUjk4z3LzN4dUz5VHL46Jo3bDo/6KU3UIz/czKZVUYduS1oXlZdoq81q0w3ho4MIASsfJYzmphIyGj/cfVoW1JkFQwB0yOwZ49qywi+rjcrMnlmujzdtHH4nf6A88d6bxrc3IekRM0u83Wo31r+Q4lFLaRIporA3xXQzm1Subn2N6LP/lPD/q9jGRzXpGiqY9K0jbBn3RpIiygy/rDblRu2cSMg2ej/h+1sHHC9ppsXYZyFt+GE5flWnA3crJA38c0+4FYpQVtI6Fd9qs0/MDwAo7O7338DOZvZuSfsRRmm/TlDG2wnRfu8GZhAS8FWMmhwRFEz6thGyT/7Jtm03WEq+rKRp1aSbonb+QNii8zaCW+e9hJtxH4K76Ifdr3mH+j8HXGdmK6PjkcCpZhZr8/KsI2kNIZSxjbABesXz2RfUX27SuvwRZZKtNnsFkWv1GkLCvYMUNgaaZfH381hISN99I8EllnauJTY1aQhySBqc5kssN/yy2nRD1M5dhJXZa6PjIYRtNz8EzLSYO12lpdgqUlV4826n+yjmhozjmiyIdtqO3twBy6fcldGShlnyRJllUauuocMJG8EPAXaXdBBhsjhW0iwzm1BJ/XqAcrNnFq7M3kxI97BeUk9s0FInSTm3RhSb3b8H6q0JJN1nZseVOldBZkl6i3VMWhdn8jo/2qmQOFlrewvrJI1mm2v2LUDJbXIlXZ4ns937ZvaF7lVzGzVpCIDLgHcSJZ0zsycU8tPHohvCL6vNMcCnJaXNnvkH4FFJt0bH7weuj+LCn+l2bbfnLuBGSb8i/DE+Q9goxukChWRljcCYyJ2Wa02GEfaB7ikOAz4uqUPSupzvv7P7sAY6YDkuILQ9e0qaRpjjiLMXRkv0fCTBrXtDdPwRgpGsGDXpGpL0mJkdVkbUTF/fvLzs7JkKezAcRWhMHjazlhIi3UYUAvlpwuriXPbMq3t6kU1fQ9J5wPmERv8Vog4AsIawN8MvekiP1EnrIn/6FjMzSbsRjMr8OEEKvYnoc+Sy5z5nCbIXR9FzJ+RkFDIh313JYIpaHRG8JOkIwCT1B75AsgUpfTrXTdqonYIyZlLhXkgXdW8lpLi4ohr191XM7KfAT6OVuZdZ2C7ym4SouUd6UI9U95+kswkbIq2V9F+E1e2PAwdL+o2Z/aAb1awY0cisQ9SUpF/FDVYhGPKhhJ0VIbi4Kzqiq1VD8BlC7O0uhMyVdxMSWcWlT25eXit0dxx6BjnJzC6RdBTwDuDHBKPa29fBnA/sSWgE5xLmpZZGrtoZBCPRF/g/wijs8uj4VEIizI/ElP8+YZ4lt67m7cTcczstNWkIogiZ09LIRj3/vr55eV9nct7rgYQ/UJcRJU4Hci609xJ2GrtV0rerqE9cNpnZCmCFpPm5SDcza5W0qYRsb6KsrT7N7Joocu90gkG8k227JVaEmjQEkn4IfIewzP1O4CDgfDP7fSnZyDd5Hh3DLz3XTQ9iZssKTl0m6WGgaDIyZztekXQlYYeyH0QpGortY9zbGCTpYIKu/fNSjoi8nbr6AGmjpoiu/xRwHrArYT3BWwiuvYqtrK/VyeLZZvZmSR8irJL9InB/gsniXwC/NbMZldTTKY465hyqI4wQPhv398s6kSvlXcAcM5snqQk4wMzurrJqXVIkxUiucRL0nZXnKn+rzzmErXYfjdqxfYCLzeyUSulckyMCoF/0/F5CHvvlCed6yw2/dMrjx2xrBHIrw+P6VzOPmbWSt5ucmS2mD+zwlWvoo/m5whQVfSlwoMstcWOwwcw2SELSADN7VlJFU4jXqiGYGlnlDcBnJY2NXsclVpZNp2K8m7Cr1ni23aMfpQ/lm3HK4lpCioqfRcenEiZgK5pvpxuZZGb35p+QdIaZXRtT/mVJIwh7ht8jaQUVniOoVdfQIEIa5bcRVsjOJsSh9/pekQMKudhXEkIH29cOmNmPq6aU02OkTVHRW5D0EPA0YcvQIcDVwEYzi7OorLCstwPDgTvNrGIT5rU6Isj1KHJbS55KCCftKz2KrLOrmZU7vHb6LmVNtvYC3g58idABBfiWmV3fxfWdYmYPdptWXVCrhqCs8C2n6vxT0gFmNqfaijhVIVWKil7ESMJneJ4Q+TMuP3dWb6RWDUFf71FknaOAM6OFZRvxyfqs0ddHg48C3zez30Ru6h8Q2p8jqqtW59TqHEFZ4VtOdemOXEmOUy0k7U5wD02IVnjvDow3s4eqrFqn1KohSJ30ynEcpxwkXUHodB5rZvtGmWDvNrMpVVatU2rSNeQNveM4VeQwMztE0iwAM1sRJb/stfSFZeeO4zh9ic3RZkq5pJVjCSOEXosbAsdxnO7lZ8AtwA6Svgs8TNjMvtdSk3MEjuM41STKD5TbWOk+M0uyH0qP44bAcRwn47hryHEcJ+O4IXAcx8k4bgiczCHp65KelvSkpNnRyvNK1fWApMmlr3Sc6lGT6wgcpzMkHQ68DzjEzDZKGgP06hhvx6k0PiJwskYTsNTMNkLY39rMXpX0LUkzJD0l6apo7+pcj/4nkh6SNFfSFEl/ljRP0neia8ZLelbStdEo4+Zol7AOSDpB0iOSHpd0k6Qh0fnvS3omkv1RD34XjgO4IXCyx93AbpL+JemXUb53gJ+b2RQz2x8YRBg15NhkZm8DfgXcCnwO2J+QGG90dM3ewFVRHqvVhB222olGHt8AjjezQ4AW4AJJo4APAW+KZL9Tgc/sOF3ihsDJFGa2FmgGzgHeAG6QdCZwjKTHolTHxwJvyhO7LXqeAzxtZoujEcUCYLfovZfMLJfh9veEDKr5vAXYD5gmaTZwBjCOYDQ2AFdL+jDQ2m0f1nFi4nMETuYwsy3AA8ADUcP/aeBAYLKZvSTp28DAPJGN0fPWvNe549x/qHBBTuGxgHvM7NRCfSQdSlh89FHCznrHJvxIjlMWPiJwMoWkvSVNyjv1ZuC56PXSyG+feEtBYPdoIhrCjngPF7z/KHCkpImRHo2S9orqG25mfwPOj/RxnB7FRwRO1hgCXB5tDt4GzCe4iVYSXD8vADNSlDsXOEPSlcA84Ir8N83sjcgFdb2kAdHpbwBrgFslDSSMGr6Yom7HKQtPMeE4ZSJpPHB7NNHsOH0Odw05juNkHB8ROI7jZBwfETiO42QcNwSO4zgZxw2B4zhOxnFD4DiOk3HcEDiO42Sc/w+rohPBIuNKNAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "freq = nltk.FreqDist(words)\n",
    "freq.plot(20, cumulative=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FreqDist({'pseudoknot': 55, 'replication': 17, 'virus': 9, 'FMDV': 9, 'replicons': 9, 'genomes': 8, 'replicon': 7, 'nucleotides': 6, 'SHAPE': 6, 'region': 6, ...})"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method FreqDist.freq of FreqDist({'pseudoknot': 55, 'replication': 17, 'virus': 9, 'FMDV': 9, 'replicons': 9, 'genomes': 8, 'replicon': 7, 'nucleotides': 6, 'SHAPE': 6, 'region': 6, ...})>"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TEST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "paper_info = get_paper_data(cfg['data-path'] + biorxiv, papers[10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paper_id': '03ea3a614b56409d3f099c9ad764864293132540',\n",
       " 'metadata': {'title': 'Live-cell single RNA imaging reveals bursts of translational frameshifting',\n",
       "  'authors': [{'first': 'Kenneth',\n",
       "    'middle': [],\n",
       "    'last': 'Lyon',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Luis',\n",
       "    'middle': ['U'],\n",
       "    'last': 'Aguilera',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Tatsuya',\n",
       "    'middle': [],\n",
       "    'last': 'Morisaki',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Brian',\n",
       "    'middle': [],\n",
       "    'last': 'Munsky',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Timothy',\n",
       "    'middle': ['J'],\n",
       "    'last': 'Stasevich',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Colorado State University',\n",
       "     'location': {'addrLine': 'Fort Collins',\n",
       "      'postCode': '80523',\n",
       "      'region': 'CO',\n",
       "      'country': 'USA'}},\n",
       "    'email': 'tim.stasevich@colostate.edu'}]},\n",
       " 'abstract': [{'text': 'Ribosomal frameshifting during the translation of RNA is implicated in both human disease and viral infection. While previous work has uncovered many mechanistic details about single RNA frameshifting kinetics in vitro, very little is known about how single RNA frameshift in living systems. To confront this problem, we have developed technology to quantify live-cell single RNA translation dynamics in frameshifted open reading frames. Applying this technology to RNA encoding the HIV-1 frameshift sequence reveals a small subset (~8%) of the translating pool robustly frameshift in living cells. Frameshifting RNA are preferentially in multi-RNA \"translation factories,\" are translated at about the same All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint 2 rate as non-frameshifting RNA (~2 aa/sec), and can continuously frameshift for more than four rounds of translation. Fits to a bursty model of frameshifting constrain frameshifting kinetic rates and demonstrate how ribosomal traffic jams contribute to the persistence of the frameshifting state. These data provide novel insight into retroviral frameshifting and could lead to new strategies to perturb the process in living cells.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'}],\n",
       " 'body_text': [{'text': 'Frameshifting is a fundamental biological process in which a ribosome translating an RNA slips by +/-1 nucleotides, resulting in the translation of an entirely different peptide sequence from that point forward. While frameshifting is generally detrimental to protein fidelity (Belew et al., 2014; Choi et al., 2015) , the process effectively creates two distinct proteins from a single RNA (Clark et al., 2007; Meydan et al., 2017; Yordanova et al., 2015) . Viruses exploit this aspect of frameshifting to minimize their genomes and to successfully replicate in host cells (Brierley and Dos Ramos, 2006; Brierley et al., 1989; Caliskan et al., 2015; Cardno et al., 2015; Mouzakis et al., 2013) . A prototypical example is HIV, which utilizes frameshifting to translate the gag-pol proteins from a single viral RNA (Guerrero et al., 2015) .',\n",
       "   'cite_spans': [{'start': 277,\n",
       "     'end': 297,\n",
       "     'text': '(Belew et al., 2014;',\n",
       "     'ref_id': 'BIBREF4'},\n",
       "    {'start': 298,\n",
       "     'end': 316,\n",
       "     'text': 'Choi et al., 2015)',\n",
       "     'ref_id': 'BIBREF13'},\n",
       "    {'start': 387,\n",
       "     'end': 411,\n",
       "     'text': 'RNA (Clark et al., 2007;',\n",
       "     'ref_id': None},\n",
       "    {'start': 412, 'end': 432, 'text': 'Meydan et al., 2017;', 'ref_id': None},\n",
       "    {'start': 433,\n",
       "     'end': 456,\n",
       "     'text': 'Yordanova et al., 2015)',\n",
       "     'ref_id': None},\n",
       "    {'start': 574,\n",
       "     'end': 604,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 605,\n",
       "     'end': 627,\n",
       "     'text': 'Brierley et al., 1989;',\n",
       "     'ref_id': 'BIBREF6'},\n",
       "    {'start': 628,\n",
       "     'end': 650,\n",
       "     'text': 'Caliskan et al., 2015;',\n",
       "     'ref_id': 'BIBREF9'},\n",
       "    {'start': 651,\n",
       "     'end': 671,\n",
       "     'text': 'Cardno et al., 2015;',\n",
       "     'ref_id': 'BIBREF10'},\n",
       "    {'start': 672,\n",
       "     'end': 694,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None},\n",
       "    {'start': 811,\n",
       "     'end': 838,\n",
       "     'text': 'RNA (Guerrero et al., 2015)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'INTRODUCTION'},\n",
       "  {'text': 'Although frameshifting has been extensively studied in vitro and in bulk assays (Caliskan et al., 2014; Charbonneau et al., 2012; Chen et al., 2014; Lopinski et al., 2000; Mouzakis et al., 2013; Ritchie et al., 2017) , the process has never been observed at the single molecule level in living cells. This leaves many basic questions about frameshifting unresolved. In particular, it is not clear how heterogenous frameshifting is from one RNA to another, nor is it clear if single RNA continuously frameshift in a constitutive fashion or if instead they frameshift in definable bursts, as has been observed for transcription (Lionnet and Singer, 2012) and 0-frame translation (Wu et al., 2016) . Finally, the localization of frameshifting has never been investigated, so it is not clear if frameshifting occurs all throughout the cell or is instead preferentially localized to specific sub-cellular regions. In the case of HIV-1 gag-pol, for example, previous assays have shown that 5-10% of translated protein product is frameshifted (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . This is thought to occur when ribosomes translate through a specialized frameshift sequence (FSS) containing a stem loop structure that slows down incoming ribosomes and causes them to slip back one nucleotide on a slippery sequence preceding the stem loop. Whether or not this occurs constitutively and with equal probability on all HIV-1 RNA or if instead it occurs on a specialized subset that are in the right place, at the right time, and with the right factors remains to be determined.',\n",
       "   'cite_spans': [{'start': 80,\n",
       "     'end': 103,\n",
       "     'text': '(Caliskan et al., 2014;',\n",
       "     'ref_id': None},\n",
       "    {'start': 104,\n",
       "     'end': 129,\n",
       "     'text': 'Charbonneau et al., 2012;',\n",
       "     'ref_id': 'BIBREF11'},\n",
       "    {'start': 130,\n",
       "     'end': 148,\n",
       "     'text': 'Chen et al., 2014;',\n",
       "     'ref_id': 'BIBREF12'},\n",
       "    {'start': 149,\n",
       "     'end': 171,\n",
       "     'text': 'Lopinski et al., 2000;',\n",
       "     'ref_id': None},\n",
       "    {'start': 172,\n",
       "     'end': 194,\n",
       "     'text': 'Mouzakis et al., 2013;',\n",
       "     'ref_id': None},\n",
       "    {'start': 195,\n",
       "     'end': 216,\n",
       "     'text': 'Ritchie et al., 2017)',\n",
       "     'ref_id': None},\n",
       "    {'start': 626,\n",
       "     'end': 652,\n",
       "     'text': '(Lionnet and Singer, 2012)',\n",
       "     'ref_id': None},\n",
       "    {'start': 677, 'end': 694, 'text': '(Wu et al., 2016)', 'ref_id': None},\n",
       "    {'start': 1036,\n",
       "     'end': 1066,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 1067,\n",
       "     'end': 1087,\n",
       "     'text': 'Dulude et al., 2002;',\n",
       "     'ref_id': None},\n",
       "    {'start': 1088,\n",
       "     'end': 1112,\n",
       "     'text': 'Grentzmann et al., 1998;',\n",
       "     'ref_id': None},\n",
       "    {'start': 1113,\n",
       "     'end': 1135,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'INTRODUCTION'},\n",
       "  {'text': 'To directly address these sorts of questions, we have developed technology to visualize and quantify single RNA frameshifting dynamics in living cells. Using multi-frame repeat epitopes, complementary high-affinity fluorescent probes that selectively bind the epitopes, and multicolor single-molecule microscopy, we are able to simultaneously monitor the translation of single RNA into two unique nascent polypeptide chains encoded in shifted open reading frames. Application to the HIV-1 FSS uncovers unexpected heterogeneity in the production of frameshifted product and implicates a novel bursty frameshifting mechanism. Besides frameshifting, our technology can now be used to examine other translational regulatory dynamics, including upstream open reading frame selection, non-canonical initiation, and ribosomal shunting. We anticipate multi-frame nascent chain tracking will be a powerful new tool to dissect complex translational regulatory dynamics in living cells and organisms.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'INTRODUCTION'},\n",
       "  {'text': 'A multi-frame tag to monitor single RNA translation in two reading frames simultaneously We created a multi-frame (MF) tag to monitor, in living cells, the translation of single RNAs with overlapping open reading frames (ORFs). The tag builds off earlier technology to visualize translation using repeat FLAG or SunTag epitopes labeled by fluorescent Fab or scFv, respectively (Lyon and Stasevich, 2017; Morisaki and Stasevich, 2018) . In the MF tag, FLAG epitopes in the 0 frame are separated from one another by SunTag epitopes in the -1 frame. With this arrangement, single RNAs with ribosomes translating the 0 frame will produce FLAG epitopes labeled by Fab, while those with ribosomes translating the -1 frame will produce SunTag epitopes labeled by scFv. Thus, depending on the chosen frame(s), polysomes will appear all green (all ribosomes translating the 0 frame), all blue (all ribosomes translating the -1 frame), or some combination of the two (Fig. 1A) .',\n",
       "   'cite_spans': [{'start': 377,\n",
       "     'end': 403,\n",
       "     'text': '(Lyon and Stasevich, 2017;',\n",
       "     'ref_id': None},\n",
       "    {'start': 404,\n",
       "     'end': 433,\n",
       "     'text': 'Morisaki and Stasevich, 2018)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 957,\n",
       "     'end': 966,\n",
       "     'text': '(Fig. 1A)',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'RESULTS'},\n",
       "  {'text': 'To ensure both frames of the MF tag encode functional proteins and have similar ribosomal occupancy, we inserted the first exon of the human GNAS1 locus downstream of the FLAG and SunTag epitopes. The GNAS1 locus contains two overlapping ORFs of roughly equivalent lengths that encode peripheral membrane proteins in the 0 and -1 frames: XXLb1 and AlexX (Abramowitz et al., 2004; Aydin et al., 2009) (Fig. 1A ). In combination with the epitopes, this arrangement has several advantages: 1) epitope densities are maximal since -1 frame epitopes act as linkers for the 0 frame and vice versa; 2) signals are digital, so frameshifted and non-frameshifted species are marked by two distinct probes/colors, and 3) epitopes are placed in nearly equivalent positions, so signals appear at roughly the same time and with similar amplification when translated with similar kinetics.',\n",
       "   'cite_spans': [{'start': 354,\n",
       "     'end': 379,\n",
       "     'text': '(Abramowitz et al., 2004;',\n",
       "     'ref_id': None},\n",
       "    {'start': 380,\n",
       "     'end': 399,\n",
       "     'text': 'Aydin et al., 2009)',\n",
       "     'ref_id': 'BIBREF2'}],\n",
       "   'ref_spans': [{'start': 400,\n",
       "     'end': 408,\n",
       "     'text': '(Fig. 1A',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'RESULTS'},\n",
       "  {'text': 'As a first application of the MF tag, we focused on -1 programmed ribosomal frameshifting caused by the HIV-1 frameshift sequence (FSS). We inserted the FSS upstream of our MF tag and transiently transfected the resulting construct into U-2 OS cells. The FSS contains a slippery poly-U stretch nine nucleotides upstream of a stem loop. Two to ten hours post transfection, we observed cells with tens or hundreds of individual RNA diffusing throughout the nucleus and cytoplasm (Fig. 1B) . Nascent Chain To confirm these RNA were active translation sites, or polysomes, we performed two experiments. First, we re-imaged cells 12-24 hours after transfection. At these later time points, Fab and scFv began to accumulate in the cell membrane ( Fig. S1 and Movie this frame encoding a functional protein, as demonstrated by shifting the sequence by one nucleotide into the 0 frame ( Fig. S1 and Movie S2, right panels). Second, we treated cells with the translational inhibitor puromycin. Just minutes after treatment, we observed a dramatic decrease in the number of Fab-and/or scFv-labeled RNA, consistent with the premature release of nascent chains ( Fig. 1D and Movie S3). Together, these data provide strong evidence that we are able to detect single RNA frameshifting dynamics with the MF tag.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 477,\n",
       "     'end': 486,\n",
       "     'text': '(Fig. 1B)',\n",
       "     'ref_id': 'FIGREF9'},\n",
       "    {'start': 741, 'end': 748, 'text': 'Fig. S1', 'ref_id': 'FIGREF9'},\n",
       "    {'start': 879, 'end': 886, 'text': 'Fig. S1', 'ref_id': 'FIGREF9'},\n",
       "    {'start': 1151, 'end': 1158, 'text': 'Fig. 1D', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'RESULTS'},\n",
       "  {'text': 'The HIV-1 FSS structure has been previously shown to produce frameshifted protein with an efficiency of 5-10% based on the dual luciferase assay and similar bulk assays (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . However, it remains unclear how this percentage is established. One possibility is that all RNA behave more or less the same and their ribosomes frameshift with 5-10% probability. On the opposite end of the spectrum, it is possible that RNA display a high degree of heterogeneity, so that just 5-10% of RNA have ribosomes that frameshift with nearly 100% probability. A third possibility is that frameshifting is common on all RNA, but frameshifted proteins are less stable and degraded faster than nonframeshifted proteins.',\n",
       "   'cite_spans': [{'start': 169,\n",
       "     'end': 199,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 200, 'end': 220, 'text': 'Dulude et al., 2002;', 'ref_id': None},\n",
       "    {'start': 221,\n",
       "     'end': 245,\n",
       "     'text': 'Grentzmann et al., 1998;',\n",
       "     'ref_id': None},\n",
       "    {'start': 246,\n",
       "     'end': 268,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'As a first step to quantify frameshifting dynamics at the single-molecule level, we tracked thousands of individual RNA approximately two to ten hours post transfection with and without the FSS sequence present (±FSS; Fig. 2A -C). In the +FSS cells, we found 92 ± 1.3% of translation sites were translating the canonical 0 frame alone, while 6.2 ± 1.1% were translating both the 0 and -1 frames. Only rarely did we observe translation All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 218,\n",
       "     'end': 225,\n",
       "     'text': 'Fig. 2A',\n",
       "     'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint sites translating just the -1 frame (1.6 ± 0.5%, Fig. S2 and Movie S4). To ensure these results were not influenced by the multi-frame tag, we reversed the FLAG and SunTag epitopes in the tag and repeated experiments. This gave nearly the same percentages, confirming the tag order and/or epitope positioning did not bias measurements ( Fig. S3 and Movie S5). We then repeated experiments in cells transfected with the control -FSS construct. In this case, we observed virtually no frameshifting sites (0.9 ± 0.7%) ( Fig. 2C and Movie S6). Taken together, these data suggest the FSS alone causes ~8% of translation sites to frameshift.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 195,\n",
       "     'end': 202,\n",
       "     'text': 'Fig. S2',\n",
       "     'ref_id': 'FIGREF10'},\n",
       "    {'start': 483, 'end': 490, 'text': 'Fig. S3', 'ref_id': 'FIGREF12'},\n",
       "    {'start': 663, 'end': 670, 'text': 'Fig. 2C', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'To further characterize the efficiency of frameshifting, we quantified the precise number of frameshifted versus non-frameshifted ribosomes per translation site. To do this, we re-imaged the MF tag at high laser powers so we could simultaneously visualize single mature XXLb1 or AlexX proteins in comparison to their nascent chains within translation sites (Fig. S4 ). After renormalizing the fluorescence, we found frameshifting sites with translation in both the 0 and -1 frames had 7.8 ± 0.7 ribosomes total, 1.9 ± 0.3 of which were frameshifted, while 0-frame only translation sites had 3.2 ± 0.1 ribosomes and -1-frame only translation sites had 3.4 ± 1.4 frameshifted ribosomes (Fig. 2B,D) .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 357,\n",
       "     'end': 365,\n",
       "     'text': '(Fig. S4',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 684, 'end': 695, 'text': '(Fig. 2B,D)', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'Thus, although just ~8% of translation sites contained frameshifted ribosomes, within these subset of sites, anywhere between 30 to 100% of ribosomes were frameshifted.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'These data support a heterogenous RNA model in which frameshifting occurs on a small subset of RNA with high probability.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Using the multi-frame tag to quantify HIV-1 frameshifting efficiency'},\n",
       "  {'text': 'Given the heterogeneity of observed frameshifting, we hypothesized that the frameshifting state could be stimulated by a specific sub-cellular environment. To test this All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint hypothesis, we performed a statistical analysis of all tracks to see if any biophysical parameters correlated with frameshifting. This revealed frameshifting sites diffuse more slowly than other translation sites or RNA (Fig. 2E ) and were also slightly more likely to be near the nuclear periphery (Fig. 2F) . These results are consistent with a slight preference for frameshifting in the perinuclear endoplasmic reticulum, where RNA are less mobile and more efficiently translated (Voigt et al., 2017; Wang et al., 2016) .',\n",
       "   'cite_spans': [{'start': 629,\n",
       "     'end': 649,\n",
       "     'text': '(Voigt et al., 2017;',\n",
       "     'ref_id': None},\n",
       "    {'start': 650, 'end': 668, 'text': 'Wang et al., 2016)', 'ref_id': None}],\n",
       "   'ref_spans': [{'start': 366,\n",
       "     'end': 374,\n",
       "     'text': '(Fig. 2E',\n",
       "     'ref_id': 'FIGREF10'},\n",
       "    {'start': 445, 'end': 454, 'text': '(Fig. 2F)', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'Of the parameters we quantified, one of the strongest correlates of frameshifting was RNA intensity. Specifically, sites translating the -1 frame had an average RNA intensity that was nearly 30% brighter than RNA only spots ( Fig. 2B and S5A ). In addition, these brighter frameshifting sites were translation-dependent, dissociating upon puromycin treatment (Movie S7). We wondered if this might be an artifact due to the aggregation of probes. Self-aggregation of Fab was unlikely as this would cause 0-frame translation sites to also be brighter, which was not the case (Fig. 2B) . Similarly, it was unlikely to be due to self-aggregation of scFv, as this would cause 0-frame translation sites to be brighter in the reverse MF tag, which was also not the case (Fig. S5B ). This left the possibility that Fab aggregate with scFv. To rule this out, we reimaged the MF tag without Fab. As frameshifting translation sites remained brighter than other sites ( Fig.   S5C ), we conclude the brighter RNA signal is not a tagging artifact, but instead represents a propensity for frameshifting sites to associate with other translating RNA in higher-order complexes reminiscent of \"translation factories\" (Morisaki et al., 2016; Pichon et al., 2016) .',\n",
       "   'cite_spans': [{'start': 1200,\n",
       "     'end': 1223,\n",
       "     'text': '(Morisaki et al., 2016;',\n",
       "     'ref_id': None},\n",
       "    {'start': 1224,\n",
       "     'end': 1244,\n",
       "     'text': 'Pichon et al., 2016)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 226,\n",
       "     'end': 241,\n",
       "     'text': 'Fig. 2B and S5A',\n",
       "     'ref_id': 'FIGREF10'},\n",
       "    {'start': 573, 'end': 582, 'text': '(Fig. 2B)', 'ref_id': 'FIGREF10'},\n",
       "    {'start': 763, 'end': 772, 'text': '(Fig. S5B', 'ref_id': 'FIGREF14'},\n",
       "    {'start': 958, 'end': 968, 'text': 'Fig.   S5C', 'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'We did not observe factories in cells transfected with the control -FSS construct ( Fig. S5D ). We therefore reasoned the FSS itself might facilitate frameshifting in factories. All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 84,\n",
       "     'end': 92,\n",
       "     'text': 'Fig. S5D',\n",
       "     'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https: //doi.org/10.1101/478040 doi: bioRxiv preprint To test this model, we co-transfected a short oligo RNA containing the FSS alone into cells expressing the +FSS MF tag. This led to a significant increase in the fraction of frameshifting sites translating just the -1 frame, from 1.6 to 5.6% when 1 \\uf06dg oligo was added, and up to 9.5% when 4 \\uf06dg oligo was added (Fig. 3A,B) . The significant increase in the -1 only sites suggests RNA can exist in a state in which all or nearly all ribosomes are frameshifted. Moreover, frameshifting continued to occur preferentially within factories, having RNA signals from 30 to 50% brighter than RNA-only spots (Fig. 3C) , even though the overall fraction of RNA within factories remained unchanged (Fig. S6 ).',\n",
       "   'cite_spans': [{'start': 100,\n",
       "     'end': 146,\n",
       "     'text': '//doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 457,\n",
       "     'end': 468,\n",
       "     'text': '(Fig. 3A,B)',\n",
       "     'ref_id': 'FIGREF12'},\n",
       "    {'start': 745, 'end': 754, 'text': '(Fig. 3C)', 'ref_id': 'FIGREF12'},\n",
       "    {'start': 833, 'end': 841, 'text': '(Fig. S6', 'ref_id': 'FIGREF16'}],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'Importantly, in control cells expressing the -FSS construct, we did not see an increase in frameshifting upon oligo RNA co-transfection. Collectively, these data provide further support for a model in which the HIV-1 FSS itself facilitates frameshifting in factories.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Frameshifting occurs preferentially in multi-RNA \"translation factories\"'},\n",
       "  {'text': 'The ~8% of frameshifted translation sites we observed is consistent with previous measurements of 5-10% frameshifted protein product (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . All else equal, this implies that frameshifting alone can explain the steady-state levels of frameshifted protein, without the need for other regulatory mechanisms, such as protein degradation.',\n",
       "   'cite_spans': [{'start': 133,\n",
       "     'end': 163,\n",
       "     'text': '(Brierley and Dos Ramos, 2006;',\n",
       "     'ref_id': 'BIBREF5'},\n",
       "    {'start': 164, 'end': 184, 'text': 'Dulude et al., 2002;', 'ref_id': None},\n",
       "    {'start': 185,\n",
       "     'end': 209,\n",
       "     'text': 'Grentzmann et al., 1998;',\n",
       "     'ref_id': None},\n",
       "    {'start': 210,\n",
       "     'end': 232,\n",
       "     'text': 'Mouzakis et al., 2013)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translational output of frameshifted ribosomes'},\n",
       "  {'text': 'To test this hypothesis, we measured the elongation rate of frameshifted ribosomes compared to non-frameshifted ribosomes. Fits to the linear portion of run-off curves yielded similar post-tag run-off times of ~386 s and ~475 s, corresponding to average elongation rates of 2.7 ± 0.3 aa/sec and 2.2 ± 0.6 aa/sec, respectively ( Fig. 4A and Fig.   S7 ). Fluorescence recovery after photobleaching experiments further confirmed these measurements without the use of drugs ( Fig. S8 and Movie S8). Based on these similar All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 328,\n",
       "     'end': 349,\n",
       "     'text': 'Fig. 4A and Fig.   S7',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 472, 'end': 479, 'text': 'Fig. S8', 'ref_id': 'FIGREF24'}],\n",
       "   'section': 'Translational output of frameshifted ribosomes'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint elongation rates for both non-frameshifting and frameshifting sites, a single round of translation would take ~9 minutes. Accounting for the number of ribosomes per translation site and their relative fractions, we calculate a cell with 100 RNA would produce ~115 frameshifted protein per hour compared to ~2,220 canonical proteins. In other words, frameshifted proteins would account for ~5% of the total, in agreement with earlier measurements. Thus, the dynamics of the FSS sequence alone can be sufficient to account for the steady-state levels of frameshifted protein in living cells, without the need for additional regulatory mechanisms.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translational output of frameshifted ribosomes'},\n",
       "  {'text': 'Despite the similar elongation rates, we noticed a slight (~60 s) delay in the run-off response at frameshifting translation sites (Sun sites Fig 4A) . Given earlier work showing the potential of the FSS to pause ribosomes (Dulude et al., 2002) , we envisioned this delay could be due to a queue or traffic jam of ribosomes upstream of the FSS within frameshifting sites. As the backed-up ribosomes clear the traffic jam, they replenish the loss of ribosomes running-off. Only after the traffic jam is fully cleared does the number of ribosomes beyond the FSS (with labeled nascent chains) begin to decay.',\n",
       "   'cite_spans': [{'start': 223,\n",
       "     'end': 244,\n",
       "     'text': '(Dulude et al., 2002)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 142,\n",
       "     'end': 149,\n",
       "     'text': 'Fig 4A)',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Evidence for ribosomal traffic jams at HIV-1 frameshifting translation sites'},\n",
       "  {'text': 'To test this possibility, we added a 10x HA epitope repeat upstream of the FSS (Fig. 4B ). This served two purposes: first, it allowed us to monitor both ribosomes upstream of the FSS (translating HA epitopes) and ribosomes downstream (translating either 0 frame FLAG or -1 frame SunTag epitopes); second, the arrangement more closely mimicked the natural placement of the FSS between the gag-pol polyproteins. In particular, the additional sequence space upstream of the FSS could accommodate longer ribosomal traffic jams, should they occur. We hypothesized that longer ribosomal All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 79,\n",
       "     'end': 87,\n",
       "     'text': '(Fig. 4B',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Evidence for ribosomal traffic jams at HIV-1 frameshifting translation sites'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint traffic jams would lead to longer run-off delays. Consistent with this, ribosomes within frameshifting sites took much longer to run-off (Fig. 4B) , with frameshifted ribosome levels fluctuating but remaining overall steady for upwards of 3000 s, despite an overall ribosome loss (Figs. S9 and S10). We observed this trend even at the single-molecule level, where the total number of ribosomes (marked by HA) dropped, but the number of Such a release can be seen in the single-molecule track at the ~1000 s time point, when the frameshift signal gets significantly brighter. Interestingly, this increase in brightness coincided with the association of an additional non-translating +FSS RNA (gray arrows in Fig. 4C ). This is consistent with the hypothesis that multi-RNA factories stimulate frameshifting. Although difficult to capture, we observed this type of stimulated frameshifting burst in another single frameshifting RNA track as well (Fig. S11 ).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 283,\n",
       "     'end': 292,\n",
       "     'text': '(Fig. 4B)',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 853, 'end': 860, 'text': 'Fig. 4C', 'ref_id': 'FIGREF13'},\n",
       "    {'start': 1090, 'end': 1099, 'text': '(Fig. S11', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Evidence for ribosomal traffic jams at HIV-1 frameshifting translation sites'},\n",
       "  {'text': 'To quantify the kinetics of frameshifting, we developed two candidate models and attempted to fit each to our three main observations: (i) the percentages of frameshifting versus non-frameshifting translation sites (Fig. 4C ), (ii) ribosomal occupancies (Fig 4D) , and (iii) run-off kinetics (Figs. 4E,F). Both models include initiation of ribosomes, codondependent elongation of proteins along the RNA template, and ribosomal exclusion to block ribosomes from passing one another or occupying the same place on the RNA.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 215,\n",
       "     'end': 223,\n",
       "     'text': '(Fig. 4C',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 254, 'end': 262, 'text': '(Fig 4D)', 'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The only difference in the two models is the treatment for how ribosomes shift from the 0 to the -1 frame.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The first model assumes constitutive frameshifting, in which all ribosomes frameshift at the FSS with equal probability. This model could capture either observation (i) or (ii), but not both simultaneously; frameshifting either led to excessively large fractions of frameshifting sites or excessively small ribosomal loading, in disagreement with our observations that a relatively small fraction of RNA frameshift with relatively high ribosomal occupancies. Even with addition of distinct pauses in elongation at the FSS in both frames, the constitutive model was unable to fit our data (Fig. S12 ).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 588,\n",
       "     'end': 597,\n",
       "     'text': '(Fig. S12',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': \"The second model is inspired by two-state gene models that are commonly used to describe heterogeneous transcription (Munsky et al., 2012) . In this `bursty' model, RNA stochastically switch between non-frameshifting and frameshifting states in which either 0% or 100% of ribosomes produce frameshifted proteins (Fig. 5A ). In the bursty model, the RNA frameshift state is assumed to switch ON and OFF at rates and , respectively, and the steady state fraction of RNA in the ON state is given by = /( + ). We estimated f from the observed fraction of frameshifting translation sites (Fig. 2C ).\",\n",
       "   'cite_spans': [{'start': 117,\n",
       "     'end': 138,\n",
       "     'text': '(Munsky et al., 2012)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 312,\n",
       "     'end': 320,\n",
       "     'text': '(Fig. 5A',\n",
       "     'ref_id': 'FIGREF14'},\n",
       "    {'start': 583, 'end': 591, 'text': '(Fig. 2C', 'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'To estimate the timescale of and , we tracked translation sites for longer periods of time. To achieve this tracking, we doubled the number of epitopes in the multiframe reporter (creating the +FSS 2x multi-frame tag) and changed our imaging strategy to sample the RNA signal at all time points and the 0 and -1 translation signals once every fifth time point. This arrangement substantially reduced photobleaching and allowed us to continuously track and monitor the translational status of single translation sites in 3D',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'for nearly an hour. Fig. 5B shows the frameshifting state survival times for the seven translation sites we tracked in this manner, including one site that frameshifted for longer All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 20,\n",
       "     'end': 27,\n",
       "     'text': 'Fig. 5B',\n",
       "     'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint than 40 minutes ( Fig. S13 and Movie S10), representing at least four rounds of translation at our measured elongation rate of ~2.4 aa/sec. Remarkably, this frameshifting translation site associated with another for a large part of the 40-minute imaging window, which supports our hypothesis that the FSS facilitates frameshifting in multi-RNA sites. From the frameshift state survival times, we fit the rate of to be ~0.0013 sec -1 (Fig. 5B S12 ). To account for the different run-off delays seen at frameshifting and non-frameshifting sites (seen in Fig. 4A ,B), the model required elongation pauses of 1/ ~ 48 s at non-frameshifting sites and 1/ * ~ 90 s at frameshifting sites. We explored if codon usage could also explain the differences in run-off times. However, according to the codon adaptation index (Gorgoni et al., 2016; Sharp and Li, 1987) , which is related to the speed at which each codon is translated in the simulation, there is no notable difference between the 0 and -1 frames ( Fig. S14) . Moreover, the distinct pauses predicted by the model are comparable to those previously measured using in vitro and in vivo bulk assays (Lopinski et al., 2000) .',\n",
       "   'cite_spans': [{'start': 957,\n",
       "     'end': 979,\n",
       "     'text': '(Gorgoni et al., 2016;',\n",
       "     'ref_id': None},\n",
       "    {'start': 980, 'end': 999, 'text': 'Sharp and Li, 1987)', 'ref_id': None},\n",
       "    {'start': 1294,\n",
       "     'end': 1317,\n",
       "     'text': '(Lopinski et al., 2000)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 164,\n",
       "     'end': 172,\n",
       "     'text': 'Fig. S13',\n",
       "     'ref_id': 'FIGREF9'},\n",
       "    {'start': 579, 'end': 587, 'text': '(Fig. 5B', 'ref_id': 'FIGREF14'},\n",
       "    {'start': 588, 'end': 591, 'text': 'S12', 'ref_id': 'FIGREF9'},\n",
       "    {'start': 698, 'end': 705, 'text': 'Fig. 4A', 'ref_id': 'FIGREF13'},\n",
       "    {'start': 1146, 'end': 1155, 'text': 'Fig. S14)', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Because the estimated average pause time, 1/ * , is greater than the average initiation time, 1/ , ribosomes could initiate faster than they clear the FSS with an excess rate of − * = 0.009 per second and create upstream traffic jams in frameshifting sites. These traffic jams would continue to build for as long as the RNA remains in the frameshifted state, or approximately 1/ = 770 s on average. This would allow for the accumulation of approximately ( − * ) / ~ 7 ribosomes (on the HA multi-frame tag, Table 1 ). Occasionally, traffic jams can extend all the way back to the start codon, as seen in a sample simulation (Movie S11). However, because these traffic jams occur upstream of the FSS within the HA epitope repeats, the nascent chain intensity at translation sites is only partially correlated with ribosomal occupancy. As a result, the model predicts frameshifting sites are slightly brighter than non-frameshifting sites ( Fig.   5G ), which we also observe experimentally (Fig. S15 ). Moreover, due to the long time it can take to clear traffic jams, frameshifting can persist for hours following the global shut down of translation initiation (with harringtonine, for example), as can be seen in simulations of the best-fit model (Movie S12) and consistent with what we observed in Figs. 4C and S11. Thus, the final bursty model suggests a mechanism by which frameshifting can persist for long periods in the absence of translation initiation.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 652,\n",
       "     'end': 659,\n",
       "     'text': 'Table 1',\n",
       "     'ref_id': 'TABREF1'},\n",
       "    {'start': 1084, 'end': 1093, 'text': 'Fig.   5G', 'ref_id': 'FIGREF14'},\n",
       "    {'start': 1134, 'end': 1143, 'text': '(Fig. S15', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Computational modeling of HIV-1 frameshifting bursts at the single RNA level'},\n",
       "  {'text': 'Frameshifting is a common tactic used by viruses to minimize their genomes for faster, more efficient replication in host cells, effectively getting two viral proteins for the price of one viral RNA. While the general architecture of frameshift sequences is well characterized and the dynamics of frameshifting have been measured with single-All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint nucleotide precision in vitro, until now frameshifting had not been directly observed in a living system. To achieve this, we created a multi-frame repeat epitope tag that can light up single RNA translation sites in different colors depending on which open reading frame is being translated. Together with sensitive single-molecule microscopy and computational modeling, we have demonstrated five novel aspects of HIV-1 frameshifting:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': '(i) frameshifted proteins originate from a small subset of RNA that frameshift with high efficiency; (ii) frameshifting occurs preferentially in multi-RNA translation factories that are facilitated in part by the frameshift sequence; (iii) frameshifting occurs in bursts on single RNA that can last for several rounds of translation; (iv) ribosomes that frameshift are paused for longer at the frameshifting sequence than ribosomes that do not frameshift;',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'and (v) pauses at the frameshift sequence induce ribosomal traffic jams that can maintain the production of frameshifted protein despite global inhibition of translation. Fig. 6 summarizes our findings.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 171,\n",
       "     'end': 177,\n",
       "     'text': 'Fig. 6',\n",
       "     'ref_id': 'FIGREF16'}],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'In contrast to constitutive frameshifting on any RNA, our data indicate that frameshifting occurs in bursts on a subset of RNA. Bursty expression has been demonstrated by others, both at single transcription sites as well as at translation sites in bacteria (Lionnet and Singer, 2012) and eukaryotes (Wu et al., 2016) . The origin of frameshifting bursts is difficult to pinpoint. It is tempting to speculate that there is a unique structure the RNA takes that enhances frameshifting, particularly given the distinct pause times in our best-fit model. According to our observations, multi-RNA translation factories appear to be more susceptible to frameshift bursts. Such factories have been previously observed (Morisaki et al., 2016; Pichon et al., 2016) . Local concentration of translation factors is higher within factories, so translation kinetics are enhanced (Pichon et al., All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 300,\n",
       "     'end': 317,\n",
       "     'text': '(Wu et al., 2016)',\n",
       "     'ref_id': None},\n",
       "    {'start': 712,\n",
       "     'end': 735,\n",
       "     'text': '(Morisaki et al., 2016;',\n",
       "     'ref_id': None},\n",
       "    {'start': 736,\n",
       "     'end': 756,\n",
       "     'text': 'Pichon et al., 2016)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint 2016). However, the crowding of machinery at these factory sites may cause them to be more prone to errors and even more so with increased interactions with other FSSencoding RNA, the FSS being a proverbial wrench within translational factory machinery.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'Such low-fidelity factories may benefit viral robustness, where mutations are desirable.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'Consistent with this idea, HIV-1 RNAs prefer a dimeric state which may promote translation factory formation (Barajas et al., 2018) .',\n",
       "   'cite_spans': [{'start': 109,\n",
       "     'end': 131,\n",
       "     'text': '(Barajas et al., 2018)',\n",
       "     'ref_id': 'BIBREF3'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'In this study, we observed frameshifting in the context of factories in three different circumstances. First, in puromycin experiments, we observed frameshifting sites separate into two frameshifting sites upon translation shut-down (Movie S7). Second, we observed on two occasions non-translating RNA interact with a frameshifting site, after which the frameshifting signal at the site markedly increased (Figs. 4E and S11). Third, when we co-transfected FSS oligo RNA into cells (Fig. 3) , frameshifting increased in proportion to the amount we co-transfected. Collectively, these data suggest a causal ordering, whereby interactions between FSS-encoding sequences lead to enhanced frameshifting.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 481,\n",
       "     'end': 489,\n",
       "     'text': '(Fig. 3)',\n",
       "     'ref_id': 'FIGREF12'}],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'According to our best-fit model, pauses always occur at the frameshift sequence, with longer pauses associated with frameshifting RNA compared to non-frameshifting RNA. Pausing in and of itself is therefore only a weak predictor of frameshifting, as others have shown (Kontos et al., 2001; Tu et al., 1992) . Nevertheless, the longer pauses associated with frameshifting suggest the frameshift sequence can adopt more than one The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint an elevated ribosomal occupancy as far back as ~400 codons from the FSS, a length that coincides almost perfectly with the length of the FSS-upstream sequence in HIV-1 gagpol. This is not unprecedented, as ribosome profiling experiments have also found evidence for relatively high ribosomal densities as far back as the start codon of the gag protein (Napthine et al., 2017) . These data suggest the strength of the HIV-1 pause may have evolved to on occasion produce the longest possible traffic jams. As clearance of these jams takes time, frameshifting can persist on RNA for hours despite a global shutdown of translation initiation. In effect, the jam acts like a battery that continually fuels the Beyond the imaging of HIV-1 translational frameshifting, the multi-frame tag can now be used in a variety of other contexts. For example, it can immediately be used to investigate frameshifting dynamics along other viral RNA sequences, as well as frameshifting thought to occur along endogenous human mRNA (Cardno et al., 2015) .',\n",
       "   'cite_spans': [{'start': 268,\n",
       "     'end': 289,\n",
       "     'text': '(Kontos et al., 2001;',\n",
       "     'ref_id': None},\n",
       "    {'start': 290, 'end': 306, 'text': 'Tu et al., 1992)', 'ref_id': None},\n",
       "    {'start': 925,\n",
       "     'end': 948,\n",
       "     'text': '(Napthine et al., 2017)',\n",
       "     'ref_id': None},\n",
       "    {'start': 1584,\n",
       "     'end': 1605,\n",
       "     'text': '(Cardno et al., 2015)',\n",
       "     'ref_id': 'BIBREF10'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'Likewise, the multi-frame tag can be used to examine other non-canonical translation processes involving more than one open reading frame, including start-codon selection, leaky scanning, ribosomal shunting, and general translation fidelity. In fact, in an accompanying paper, a similar multi-frame tag was used to examine upstream and downstream open reading frame selection (Boersma et al.) . Like us, the authors also saw a high degree of heterogeneity between translating RNAs, with bursts of translation All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 376,\n",
       "     'end': 392,\n",
       "     'text': '(Boersma et al.)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint initiation in multiple open reading frames similar to the bursts of frameshifting we observed at the HIV-1 frameshift sequence. Translational heterogeneity may therefore be far more common than originally appreciated, particularly when it comes to non-canonical translation. We therefore believe multi-color single molecule imaging of both canonical and non-canonical translation will become a powerful new tool for dissecting complex RNA regulatory dynamics in a variety of important contexts. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Dulude, D., Baril, M., and Brakier-Gingras, L. (2002) . Characterization of the frameshift stimulatory signal controlling a programmed -1 ribosomal frameshift in the human immunodeficiency virus type 1. Nucleic Acids Res. 30, 5094-5102. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 890,\n",
       "     'end': 931,\n",
       "     'text': 'Baril, M., and Brakier-Gingras, L. (2002)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. is the translation initiation rate, is the translation elongation rate, is the rate at which RNA switch to the frameshifting state, is the rate at which RNA switch to the non-frameshifting state, is the pause rate at the FSS in the non-frameshifting state, * is the pause rate at the FSS in the frameshifting All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint longer ribosomal traffic jams. These allow frameshifting RNA to continue to be translated for up to an hour longer than non-frameshifting RNA upon global translation initiation shutdown.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint RNA (0, 1, and 4 mg) encoding the frameshift sequence were co-transfected into cells with the +FSS multi-frame reporter or the control -FSS reporter. The bar graph above shows the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). (B)',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The distributions of RNA signals from the four experiments normalized to their mean value. In all +FSS experiments, the fraction of bright RNA (defined here as having an intensity greater than or equal to 1.6 times the mean) remained constant. The -FSS experiment had fewer bright RNA, suggesting the FSS sequence facilitates incorporation into multi-RNA sites. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'For completeness, an additional experiment was performed to generate the ribosome run-off curve from FLAG epitopes in the HA multi-frame tag. Because we can only image two epitopes at the same time (since RNA is imaged in a third color), we examined FLAG and Sun epitopes (Sun epitopes were required to distinguish frameshifting and nonframeshifting sites). The FLAG run-off from this experiment is on the right (green triangles). This curve is compared to the run-off curve of HA in HA only sites (orange squares) from Figure 4B . Both experiments represent non-frameshifting sites (no Sun signal detected). Error bars represent S.E.M.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 520,\n",
       "     'end': 529,\n",
       "     'text': 'Figure 4B',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Figure S10. Ribosome run-off from FLAG epitopes in HA multi-frame tag'},\n",
       "  {'text': 'A single frameshifting translation site encoding the HA multi-frame tag was tracked after harringtonine addition. On the top, a montage of image trims shows the detected RNA-, HA-, and Sun shown signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint ribosomes) is plotted through time. At about the fourth timepoint, another non-translating RNA interacts with the frameshifting RNA, leading to a burst of frameshifting signal (Sun).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions. and frameshifting sites (gray), plotted with data in Fig. 4A (black triangles and gray circles). On the right, simulated run-off (solid lines) from the start site (AUG) of all ribosomes in non-frameshifting (black) and frameshifting sites (gray), plotted with data in The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint channels separated to highlight two frameshifted RNA that split from a single translation site. On the right, the position of the spot is plotted through time. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 139,\n",
       "     'end': 146,\n",
       "     'text': 'Fig. 4A',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'Intensity distribution of non-frameshifted HA signals (black) versus frameshifted HA signals (gray) produced from the HA multi-frame tag.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'},\n",
       "  {'text': 'Movie S1. Dynamics of cell presented in Figure 1B Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 4 seconds for a total of 76 seconds. Circled regions of interest correspond to species identified in Figure 1B and C. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 40,\n",
       "     'end': 49,\n",
       "     'text': 'Figure 1B',\n",
       "     'ref_id': 'FIGREF9'},\n",
       "    {'start': 410, 'end': 419, 'text': 'Figure 1B', 'ref_id': 'FIGREF9'}],\n",
       "   'section': 'SUPPLEMENTARY MOVIE LEGENDS'},\n",
       "  {'text': 'Single plane movies showing example U-2 OS cells expressing either +FSS multi-frame tag (left), -FSS control multi-frame tag (middle), or -FSS (+1 nt) tag with XXLb1 shifted into the 0 frame (right). All cells were bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, AlexX), and scFv-sfGFP (Ch3 blue, XXLb1). Images were acquired every 7 seconds for a total of 42 seconds. Scale bar, 10 µm. Figure 1D Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 112 seconds prior to puromycin and 280 seconds following puromycin. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 420,\n",
       "     'end': 429,\n",
       "     'text': 'Figure 1D',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Movie S2. Dynamics of cells presented in Figure S1'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S4. Dynamics of cell presented in Figure S2'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 63 seconds. Circled regions of interest correspond to species identified in Fig. S2 . Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 506,\n",
       "     'end': 513,\n",
       "     'text': 'Fig. S2',\n",
       "     'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Movie S4. Dynamics of cell presented in Figure S2'},\n",
       "  {'text': \"Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the +FSS reverse multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, -1 ORF), and scFv-sfGFP (Ch3 blue, 0 ORF). Images were acquired every 7 seconds for a total of 63 seconds. Circled regions of interest correspond to RNA with no translation (numbered '1'), 0 and -1 frame translation site (numbered '2'), and 0 frame only translation site (numbered '3'). Scale bar, 10 µm.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S5. Representative cell transfected with the +FSS reverse multi-frame tag'},\n",
       "  {'text': \"Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the -FSS control tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 63 seconds. Circled regions of interest correspond to RNA with no translation (numbered '1'), and 0 frame only translation site (numbered '2'-'8').\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S6. Representative cell transfected with the -FSS control tag'},\n",
       "  {'text': 'Scale bar, 10 µm. Figure   1D All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 18,\n",
       "     'end': 29,\n",
       "     'text': 'Figure   1D',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Movie S6. Representative cell transfected with the -FSS control tag'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the + FSS multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 7 seconds for a total of 217 (of 280) seconds total post puromycin addition. This movie shows the same cell from Movie S3, but now zoomed in to accentuate a frameshifting site splitting (red arrows). Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S7. Zoom-in showing frameshifting site presented as a montage in'},\n",
       "  {'text': 'Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the + FSS 2x multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), FLAG Fab Cy3 (Ch2 green, 0 ORF), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired every 10 seconds for a total of 100 seconds pre-FRAP and 790 seconds post-FRAP. Control spot (far left red arrow) and two frameshifting sites (center arrows) were tracked through time to measure their fluorescence (0 frame translation) after photobleaching. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S8. Dynamics of Fluorescence Recovery After Photobleaching (FRAP) experiment shown in Figure S8'},\n",
       "  {'text': 'Max projection of a 13 z-stack movie showing an example U-2 OS cell expressing the HA multi-frame tag and bead-loaded with probes: MCP-Halo JF646 (Ch1 red, RNA), HA Fab Cy3 (Ch2 green, all ORFs), and scFv-sfGFP (Ch3 blue, -1 ORF). Images were acquired post harringtonine every 120 seconds for a total of 3480 seconds. Scale bar, 10 µm.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'Movie S10. Dynamics of tracked frameshifting sites presented in Figure S13 All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 64,\n",
       "     'end': 74,\n",
       "     'text': 'Figure S13',\n",
       "     'ref_id': 'FIGREF9'}],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. Movie S12. Sample simulation of best-fit bursty model of frameshifted and nonframeshifted translation sites during harringtonine run-off All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': \"The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint A best-fit bursty model simulation of multiple non-frameshifting (top four labeled 'nonFS Spots') and frameshifting (bottom four labeled 'FS Spots') translation sites encoding the HA multi-frame tag. In the simulation, harringtonine is added after 1000 s and the simulation continues for an additional 2000 s. Individual ribosomes (gray dots) are shown moving along the reporters (gray lines), with green, blue, and cyan squares marking the positions of FLAG, SunTag, and HA epitopes, respectively. As the ribosomes move along the repeat epitopes, the translated nascent chains appear as green (FLAG-tagged nascent chains), blue (Sun-tagged nascent chains), and cyan (HA-tagged nascent chains) dots that increase in size as more epitopes are translated. When the RNA is in the non-frameshifting state, the FSS is labeled 'FSS(OF)' in green; when the RNA is in the frameshifting state, the FSS is labeled 'FSS(-1F)' in blue. The simulated time is shown to the right of each reporter.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Movie S9. Dynamics of tracked frameshifting site presented in Figure 4C'},\n",
       "  {'text': 'Requests for further information and resources should be directed to and will be fulfilled by the corresponding authors, Brian Munsky (brian.munsky@colostate.edu) and Tim Stasevich (tim.stasevich@colostate.edu). Key plasmids will be deposited on Addgene.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'CONTACT FOR REAGENTS AND RESOURCE SHARING'},\n",
       "  {'text': \"The HIV-1 frameshift sequence (FSS) followed by either the multi-frame (MF) tag or the reverse multi-frame (revMF) tag were synthesized by GeneArt® gene synthesis service (Thermo Fisher Scientific). The gene fragments were flanked by NotI and NheI, and fused to the upstream of the beta-actin zipcode and 24x MS2 stem loops in the 3' UTR of plasmid pUB_smFLAG_ActB_MS2 (Plasmid #81083, addgene) to obtain the FSS-MF and FSS-revMF, respectively. To double the MF tag, the MF tag region was digested out from FSS-MF with XbaI and AgeI, and then ligated into FSS-MF flanked with NheI and\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': \"AgeI. The open reading frame encoding human XXLb1/AlexX (Abramowitz et al., 2004; Aydin et al., 2009 ) was amplified from U-2 OS cells cDNA with the primers: 5'-GTT GTC ATA TGG GCG TGC GCA ACT -3'; 5'-GAT GTA GCT AGC CTA GAA GCA GCA GGC GGT G -3'. The amplified XXLb1/AlexX was flanked with NsiI and NheI, and then inserted into the C-terminal region of the FSS-MF, FSS-2xMF and FSS-revMF to obtain FSS-MF-AlexX (i.e. the +FSS multi-frame tag), FSS-2xMF-AlexX (i.e. the +FSS 2x multi-frame tag), and FSS-revMF-AlexX (i.e. the +FSS reverse multi-frame tag), respectively. To All rights reserved. No reuse allowed without permission.\",\n",
       "   'cite_spans': [{'start': 56,\n",
       "     'end': 81,\n",
       "     'text': '(Abramowitz et al., 2004;',\n",
       "     'ref_id': None},\n",
       "    {'start': 82,\n",
       "     'end': 100,\n",
       "     'text': 'Aydin et al., 2009',\n",
       "     'ref_id': 'BIBREF2'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': \"The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Fab generation and dye-conjugation Fab generation was done using the Pierce mouse IgG1 preparation kit (Thermo Fisher Scientific) per the manufacturer's instructions. Briefly, beads conjugated with ficin were incubated in 25 mM cysteine to digest FLAG (Wako, 012-22384 Anti DYKDDDDK mouse IgG2b monoclonal) or HA (Sigma-Aldrich, H3663 HA-7 IgG1 mouse monoclonal) antibodies to generate Fab. Fab were separated from the digested Fc region using a NAb Protein A column (Thermo Scientific, product # 1860592). Fab were concentrated to ~1 mg/ml and conjugated to either Cy3 or Alexa Fluor 488 (A488). Cy3 Nhydroxysuccinimide ester (Invitrogen) or A488 tetrafluorophenyl ester (Invitrogen) was suspended in DMSO and stored at -20°C. 100 µg of Fab were mixed with 10 µL of 1M NaHCO3, to a final volume of 100 µL. 2.66 µl of Cy3 (or 5 \\uf06dl of A488) was added to this 100 µL mixture and incubated for 2 hours at room temperature with end-over-end rotation.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'The dye conjugated Fab were eluted from a PBS equilibrated PD-mini G-25 desalting column (GE Healthcare) to remove unconjugated dye. Dye conjugated Fabs then were concentrated in an Ultrafree 0.5 filter (10k-cut off; Millipore) to 1 mg/ml. This conjugation and concentration process was repeated on occasion to ensure a degree of labeling close to one. The ratio of Fab:dye, , was determined using the absorbance at 280 and 550 nm or 495 nm, the extinction coefficient of IgG at 280 nm, Ɛ , the extinction coefficient of the dye, Ɛ , provided by the manufacturer, and the dye correction factor at 280 nm, , provided by the manufacturer. The degree of labeling, , was calculated with the following formula:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint sfGFP and 33 µg/ml of purified MCP-HaloTag protein were mixed with PBS to a final volume of 4 µl. Second, in cell culture hood, DMEM was aspirated from the MatTek chamber and the 4 µl mix was pipetted on top of cells. Third, ~106 µm glass beads (Sigma Aldrich) were evenly sprinkled over the cells. Fourth, the chamber was tapped carefully ~10 times on the cell culture hood bench top. Fifth, DMEM was immediately added back to the cells. Sixth, cells were returned to the incubator for at least an hour to recover from the loading procedure. In most experiments, we also bead-loaded DNA, which we added to the initial 4 µl mix (so that DNA had a final concentration close to 1 mg/ml). On occasion, DNA was transiently transfected ~2 hours before bead-loading. Around one hour before experiments began, bead-loaded cells were washed with phenol-red-free complete DMEM to remove glass beads, and 200 nM of JF646-HaloTag ligand (a cell permeable fluorogenic ligand (Grimm et al., 2015) ) was added to label MCP-HaloTag protein. After 30 mins of incubation, cells were washed three times using phenol-red-free complete DMEM to remove any unconjugated fluorophores. Cells were then immediately imaged for experiments.',\n",
       "   'cite_spans': [{'start': 1200,\n",
       "     'end': 1220,\n",
       "     'text': '(Grimm et al., 2015)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Plasmid Construction'},\n",
       "  {'text': 'To track single molecule translation sites, a custom-built widefield fluorescence microscope based on a highly inclined illumination scheme (Tokunaga et al., 2008 ) was used (Morisaki et al., 2016 . Briefly, the excitation beams, 488, 561 and 637 nm solidstate lasers (Vortran), were coupled and focused at the back focal plane of the objective lens (60X, NA 1.49 oil immersion objective, Olympus). The emission signals were split by an imaging grade, ultra-flat dichroic mirror (T660lpxr, Chroma) and detected using two All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [{'start': 140,\n",
       "     'end': 162,\n",
       "     'text': '(Tokunaga et al., 2008',\n",
       "     'ref_id': None},\n",
       "    {'start': 163,\n",
       "     'end': 196,\n",
       "     'text': ') was used (Morisaki et al., 2016',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Single molecule tracking microscopy'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. Readout time for the cameras from the combination of our imaging size, readout mode, and the vertical shift speed was 23.36 msec, resulting in an imaging rate of 13 Hz (70 msec per image). The excitation laser lines were digitally synched to ensure they only illuminated cells when the camera was exposing in order to avoid excessive photobleaching. To capture the entire volume of the cytoplasm of U-2 OS cells, 13 zstacks with step size of 500 nm (6 µm in total) were imaged using the piezoelectric stage such that the z-position changed every 2 images (one image for Cy3 and one for All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Single molecule tracking microscopy'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint sfGFP/A488 + JF646). The position of the filter wheel was changed during the camera readout time. This resulted in a total cellular imaging rate of 0.5 Hz (2 sec per volume for 3-colors). Note that all colors described in the text and that are shown in the figures are based on the color of the excitation laser: RNA in red (JF646) and protein in green (Cy3) or blue (sfGFP/A488).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Single molecule tracking microscopy'},\n",
       "  {'text': 'Images were first pre-processed using either Fiji (Schindelin et al., 2012) or a customwritten batch processing Mathematica code (Wolfram Research) to make 2D maximum intensity projections from 3D images. Pre-processed images were then analyzed with a custom-written Mathematica code to detect and track particles. Specifically, particles were emphasized with a band-pass filter so the positions could be detected using the built-in Mathmatica routine ComponentMeasurements \"IntensityCentroid\". Detected particles were linked through time by allowing a maximum displacement of 5 pixels between consecutive frames. Particle tracks lasting at least 5-10 frames were selected and their precise coordinates were determined by fitting (using the built-in Mathematica routine NonlinearModelFit) the original 2D maximum intenstiy projected images to a 2D Gaussians of the following form:',\n",
       "   'cite_spans': [{'start': 45,\n",
       "     'end': 75,\n",
       "     'text': 'Fiji (Schindelin et al., 2012)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'where is the background intensity, the particle peak intensity, ( , ) the spread of the particle, and ( 0 , 0 ) the particle location. The offset between the two cameras was All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint registered using the built-in Mathematica routine FindGeometricTransform to find the transform function that best aligned the fitted positions of 100 nm diameter Tetraspeck beads evenly spread out across the image field-of-view. We did not register the images, but only the fitted positions in order to avoid introducing any distortion into images. This is why a slight offset can be observed between the red and the green/blue particles even though they are within a diffraction limited spot, according to our registration.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'For visuallization and some quantification, average intensity image trims were created by averaging images of all detected particles of a given species (each centered by their intensity centroid). To compute the average number of ribosomes in a translation site (see Fig. S4 ), the average image trims of translation sites and single mature proteins were fitted to Eq. (2). The average intensities were calculated by integrating the fitted Gaussian. The ratio of the average intensity of translation sites to that of single mature proteins was then calculated to give the average number of ribosomes in a translation site. The average intensity of single RNAs were also calculated using the same procedure ( Fig. 3C , S5, and S6). Fits were susceptible to noise, so we also used an alternative strategy to determine the average intensity of translation sites and RNA that was robust to noise (in all Figs aside from Fig. 3C , S4, S5, and S6). Specifically, the intensity of centered images of RNA or translation sites was calculated to be the average intensity within a centered radial spot of four pixels in diameter minus the average background intensity from a centered ring with an outer diameter of twelve pixels and an inner diameter of eight pixels. Mathematica source code is available upon request.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 267,\n",
       "     'end': 274,\n",
       "     'text': 'Fig. S4',\n",
       "     'ref_id': 'FIGREF13'},\n",
       "    {'start': 708, 'end': 715, 'text': 'Fig. 3C', 'ref_id': 'FIGREF12'},\n",
       "    {'start': 916, 'end': 923, 'text': 'Fig. 3C', 'ref_id': 'FIGREF12'}],\n",
       "   'section': 'Particle tracking'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translation Site Species Identification'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https: //doi.org/10.1101/478040 doi: bioRxiv preprint After RNA particles were identified and tracked using the custom Mathematica code described above, an average centered image of the first five frames from each track was created for RNA (JF646), 0 ORF (FLAG-Cy3 in the +FSS multi-frame tag), and -1 ORF (scFv-sfGFP in the +FSS multi-frame tag). The trims were then hand checked to remove any trims with artifacts, e.g., smears or non-diffraction limited spots. Next, a custom Mathematica code was used to detect particles in the 0 ORF or -1 ORF trim channels, sorting the spots into RNA only, 0 frame translation sites (0 TS only), 0 and -1 TS, and -1 only TS. For all cases, RNA always had to be present. Finally, frameshifting translation sites (the 0 and -1 TS or the -1 only TS) were validated by eye, to further remove artifacts.',\n",
       "   'cite_spans': [{'start': 100,\n",
       "     'end': 146,\n",
       "     'text': '//doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translation Site Species Identification'},\n",
       "  {'text': 'For example, RNA that briefly colocalized with a mature protein puncta were removed at this stage. After all sites were validated, the total count of each type of species was used to determine the percentage of non-translating RNA (no TS), 0 frame translation sites (0 TS), 0 and -1 TS, and -1 only TS.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Translation Site Species Identification'},\n",
       "  {'text': 'To confirm active translation elongation, puromycin (Sigma Aldrich) was used to release nascent chains from elongating ribosomes, leading to a rapid loss of nascent chain signal at translation sites. Bead-loaded cells with visible translation sites were imaged at a rate of one volume every seven seconds. After acquiring 16 pre-images, cells were treated with a final concentration of 100 µg/ml puromycin and continuously imaged for an additional 100 time points. As a control, the same imaging conditions were performed except that the cells were treated with vehicle (H2O). In this case, nascent chain signals All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Puromycin treatment'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint did not disappear (data not shown). Both frameshifting and non-frameshifting translation sites were monitored through time using the tracking code described above.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Puromycin treatment'},\n",
       "  {'text': \"Oligo FSS RNA co-transfection An uncapped RNA oligo containing the FSS from HIV-1 was synthesized from IDTDNA with the following RNA sequence: 5'-CUG GCC UUC CCU UGU GGG AAG GCC AGA UCU UCC CUA AAA AA -3'. Co-transfection of FSS-MF-AlexX and the RNA oligo was carried out via bead-loading, as describe above. Briefly, 1 µg plasmid FSS-MF-AlexX construct DNA, 100 µg/ml of fluorescently labeled Fab, 250 µg/ml of purified (GCN4) scFv-sfGFP or 1 µg of (GCN4) scFv-sfGFP plasmid, 1 or 4 µg of RNA oligo, and 33 µg/ml of purified MCP-HaloTag protein were mixed with PBS to a final volume of ~4 µl of PBS.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Puromycin treatment'},\n",
       "  {'text': 'To measure the average elongation rate, harringtonine (Cayman Chemical) was used to block translation initiation and iduce the run-off of all actively elongating ribosomes, leading to a gradual loss of nascent chain signal at translation sites. Beadloaded cells with visible translation sites were imaged as described above except that cell volumes were acquired every 60 or 120 seconds. After acquiring 5 pre-images, cells were treated with a final concentration of 3 µg/ml Harringtonine and continuously imaged for 30 more time points. As a control, the same imaging conditions were performed except that cells were treated with vehicle (DMSO). In this case, nascent chain signals did not decay (data not shown). The intensities of translation sites measured, as described above. The All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint intensity of all translation sites in each frame (and in all cells) were then totaled to produce the run-off curve. Run-off curves were normalized to the mean of the total intensity of the first four time points after treatment of harringtonine began. These curves were then fit to a linear regression to estimate run-off times (see Fig. S7 ). The linear portion of the runoff decay begins when the normalized run-off intensity reaches a fraction 0 , where',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 479,\n",
       "     'end': 486,\n",
       "     'text': 'Fig. S7',\n",
       "     'ref_id': 'FIGREF23'}],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'where 1 is the length of untagged portion of the open reading frame and 2 is the length of the tagged portion (i.e. the length of the repeated epitopes), as described (Morisaki and Stasevich, 2018). The linear portion of the decay was then interpolated to background levels to estimate the run-off time . The elongation rate was calculated as follows:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'Fluorescence recovery after photobleaching (FRAP)',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'To confirm elongation rates of frameshifting translation sites, FRAP experiments were performed. Bead-loaded cells with visible translation sites were imaged once every 10 seconds, with an intentional photobleach at frame 10. The photobleach was performed with a 405 nm laser focused to a spot roughly 5 \\uf06dm in diameter and operating at a minimal power such that the nascent chain signal did not completely vanish. This allowed us to track the translation site continously throughout the experiment. Following the intentional photobleach, the fluorescence recovery of translation sites within the bleach zone were monitored for an additional 80 time points. These translation sites were tracked and their intensities quantified, as described above. To correct for unintentional photobleaching, All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint the intensity of a control translation site was also tracked and quantified. The loss of signal from the control translation site was fit to a single exponential decay and this decay was divided out from the FRAP recovery curves of the intentionally photobleached translation sites. The FRAP recovery curve can be thought of as the inverse of the harringtonine runoff curve (Morisaki and Stasevich, 2018) . In this way, the FRAP recovery curve was fit to determine the average translation elongation rate (Fig. S8 ).',\n",
       "   'cite_spans': [{'start': 520,\n",
       "     'end': 550,\n",
       "     'text': '(Morisaki and Stasevich, 2018)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [{'start': 651,\n",
       "     'end': 659,\n",
       "     'text': '(Fig. S8',\n",
       "     'ref_id': 'FIGREF24'}],\n",
       "   'section': 'Ribosome Run-Off Experiments and fits'},\n",
       "  {'text': 'The frameshift models describe the stochastic dynamics of nascent translation with single-codon resolution. Both the constitutive and bursting models are formulated to track an arbitrary number of individual ribosomes that can perform three possible reactions: (i)',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': \"A new ribosome can initiate translation at the start codon at rate kini. (ii) An existing ribosome can elongate at rate ke to incorporate one amino acid into the nascent peptide chain. The rate ke is sequence-specific with each codon's rate scaled by the genomic copy of the corresponding codon (Nakamura et al., 1999) . When the ribosome completes the final amino acid, translation is terminated and that ribosome is eliminated. (iii) A ribosome at the frameshift sequence can shift from the 0 to the -1 frame. In addition to these three reactions, ribosomes at the frameshift sequence can pause for an average time of 1/ * or 1/ for the shifted and non-shifted states. For the bursting model, each RNA is assumed to switch back and forth between non-frameshifting and frameshifting states with rates and . With these mechanisms and parameters, the models can be analyzed using either simplified approximations or detailed simulations.\",\n",
       "   'cite_spans': [{'start': 295,\n",
       "     'end': 318,\n",
       "     'text': '(Nakamura et al., 1999)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Several approximate features can be derived directly from the bursting model parameters (see Table 1 ). The average burst time or time spent in a frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 239,\n",
       "     'end': 246,\n",
       "     'text': 'Table 1',\n",
       "     'ref_id': 'TABREF1'}],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The average time spent in a non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The fraction of RNA in a frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The number of ribosomes that initiate during the frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The number of ribosomes that initiate during the non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The time for a ribosome to clear the FSS in the frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'where the second term accounts for ribosome pileup in the 9-codon ribosomal exclusion (Ingolia et al., 2009) , , region upstream from the FSS. Similarly, the time for a ribosome to clear the FSS in the non-frameshifting state is:',\n",
       "   'cite_spans': [{'start': 86,\n",
       "     'end': 108,\n",
       "     'text': '(Ingolia et al., 2009)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint One can approximate = 1 or 0 if initiation is faster ( > ) or slower ( > )',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'than the FSS pause. The number of ribosomes to pass the FSS in the frameshifting state',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The number of ribosomes to pass the FSS in the non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The average number of ribosomes to accumulate upstream of the FSS in the frameshifting state is: * = * − *',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'and the number to clear in a non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'In equations (13) and (14), positive values of correspond to accumulation of ribosomes in the traffic jam, and negative values correspond to clearance of ribosomes. The average elongation time for a ribosome in the frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'and the number to clear in a non-frameshifting state is:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'where is the gene length in codons.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': \"Simulations were started at t = -10,000 seconds to approximate steady state at t = 0 using the Gillespie algorithm (Gillespie, 1976) . Ribosome densities were found by collecting position statistics for multiple simulations. Simulated ribosome numbers and positions and multi-frame tag probe locations were combined to estimate translation site intensities. Harringtonine assays were simulated by preventing the initiation reaction at the time of treatment. Parameter estimation was performed using genetic algorithms and a multiple-objective cost function that considers the frameshifting efficiency, the number of ribosomes per RNA and the Harringtonine assays. A detailed description of the computational methods and codes is given in the 'Computational Details' section below.\",\n",
       "   'cite_spans': [{'start': 115,\n",
       "     'end': 132,\n",
       "     'text': '(Gillespie, 1976)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'General aspects of modeling frameshift kinetics'},\n",
       "  {'text': 'Both the constitutive model and the bursting model consist of three general ribosomal reaction types: initiation ( 0 ) , elongation ( or ̂) and frameshifting ( or ̂) as depicted in Eq. 17,',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'Ribosome Initiation. Ribosomes are assumed to bind to the RNA with rate . To account for the fact that ribosomes are large biomolecules that occupy around 20 to 30 All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint nuclear bases on the RNA (Ingolia et. al, 2009) , initiation is blocked by any downstream ribosome within = 9 codons (in either frame). Therefore, initiation rate is set at:',\n",
       "   'cite_spans': [{'start': 171,\n",
       "     'end': 193,\n",
       "     'text': '(Ingolia et. al, 2009)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': \"Ribosome Elongation. Each ribosome moves along the RNA codon by codon in the 5' to 3' direction. The elongation rate for each ℎ codon, ̂ is assumed to be:\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'where denotes the codon usage frequency in the human genome obtained from (Nakamura et. al, 2000) , and ̅ represents the average codon usage frequency in the human genome. The fit parameter specifies the average elongation rate. Ribosomal termination is assumed to be equivalent to elongation of the final codon. ',\n",
       "   'cite_spans': [{'start': 74,\n",
       "     'end': 97,\n",
       "     'text': '(Nakamura et. al, 2000)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'or the ribosome can continue in the -1 frame at rate: All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'For the bursting model, the decision to continue in the 0 or -1 frame depends upon the frameshifting state with the rate given by: ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'Relating model dynamics to experimental fluorescence intensity. To relate ribosome occupancy to experimental translation spots, we proscribe an assumed fluorescence to each ribosome based upon its position. This intensity is proportional to the number of peptide probes upstream from the ribosome location and in the appropriate frame or frames. Ribosomes in the 0 frame include all upstream probes in the 0 frame. Ribosomes in the -1 frame include probes in the 0 frame between the start codon and the FSS (if any) and probes in the -1 frame between the FSS and the current ribosome position. The total intensity vector for the ℎ color is given by',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'where ( ) denotes the position (i.e., frame and codon location) of the ℎ ribosome, and ( ) is the corresponding intensity of the ℎ color.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Computational Details'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Fraction of translation spots. Experimental data was measured between 2 to 6 hours post bead-loading for a period of 120 seconds. To reproduce these experimental data, the model was solved using 100 trajectories of 120 seconds starting at steady state. Spots were classified as 0 frame only ( 0 ) , -1 frame only ( −1 ) and both frames ( ) .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Following the experimental observation that frameshifting spots commonly consist of more than one RNA, were equally added to the 0 and −1 . The percentage of nonframeshifting translation spots was calculated as follows:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'where 0 is the sum of 0 and is the total number of translating spots.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The percentage of frameshifting translation spots was calculated as follows: * = ( −1 + ( 1 2 ⁄ ) ) • 100,',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'where −1 is the sum of −1 .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Number of ribosomes per RNA. Experimental observations allowed to quantify the number of ribosomes in frameshifting and non-frameshifting translation spots ( Figure 2 in the main text). To relate model simulations and data, the number of ribosomes per RNA was estimated by directly by adding the number of ribosomes in each RNA at steady state.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 158,\n",
       "     'end': 166,\n",
       "     'text': 'Figure 2',\n",
       "     'ref_id': 'FIGREF10'}],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Harringtonine assays. Harringtonine inhibits translation by binding to the ribosomal 60S sub-unit, which blocks new initiation events. Experimental data showed that Harringtonine causes the intensity in translating spots to drop to a basal intensity value after a run-off time (Figure 4 in main text). To mimic the effects of Harringtonine in our model we modified the initiation rate as:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 423,\n",
       "     'end': 432,\n",
       "     'text': '(Figure 4',\n",
       "     'ref_id': 'FIGREF13'}],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': '< and codons {1, … , } unoccupied,',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': \"where is the time of application of Harringtonine. After Harringtonine application, spots simulated from the original construct were classified as 0 frame only ('0F') or both 0 frame and -1 frames ('BF'). After classification, average spot intensities were quantified as:\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': \"where (.) is experimental background expression obtained at the end of the experimental time of the run-off assays. Similarly, spots simulated for the extended construct with upstream HA tags and downstream -1 tags were classified as HA in non-shifted spots ('HA') or HA in shifted spots ('HA*').\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint ( ) = 〈 ( )〉 + , (29) * ( ) = 〈 * ( )〉 + * .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Parameter estimation. For each model, we sought to find a single parameter set that reproduces all experimental data. Given the different sources of experimental data we estimated the parameter values using a multi-objective optimization strategy to simultaneously compare the Harringtonine run-off data, the fractions of shifted and nonshifted spots and the ribosome occupancy, as follows:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'where experimental data are denoted as ̃ and simulations results as ̂. Weights ( , and ) were used to balance constraints by the different experiments and time points.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'represents the number of experimental points in the Harringtonine assays ( = 30), is the number of studied frames ( = 2), is the number of different gene constructs ( = 2), and is the number of different data types ( = 3). The weight used on the Harringtonine data set was defined as = 1 ( • • ) ⁄ . The weight in the fraction of spots per frame was defined as = 1 ( + * ) ⁄ . And the weight for the number of ribosomes was defined as: = 1 ( + * ) ⁄ .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Parameter searches. Parameter searches consisted of optimization routines based on genetic algorithms (GA) using a population of 100 individuals and running the search for 20 generations.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The constitutive model has a total of four parameters ( , , and * ), whereas the bursting model has two additional parameters ( , ). The optimization strategy consisted in to fit four free parameters ( , , and * ) in the models, parameters ( , ) were independently determined from data in Fig. 5B ,C). The parameter set that best reproduces the data was selected as:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 289,\n",
       "     'end': 296,\n",
       "     'text': 'Fig. 5B',\n",
       "     'ref_id': 'FIGREF14'}],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': '= argmax − log 10 ( ) .',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'Optimized parameter values are given in full detail in All rights reserved. No reuse allowed without permission.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'},\n",
       "  {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Comparison of Data and Models'}],\n",
       " 'bib_entries': {'BIBREF1': {'ref_id': 'b1',\n",
       "   'title': 'XLas, the extra-long form of the -subunit of the Gs G protein, is significantly longer than suspected, and so is its companion Alex',\n",
       "   'authors': [],\n",
       "   'year': None,\n",
       "   'venue': 'Proc. Natl. Acad. Sci',\n",
       "   'volume': '101',\n",
       "   'issn': '',\n",
       "   'pages': '8366--8371',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF2': {'ref_id': 'b2',\n",
       "   'title': 'Extralarge XLαs (XXLαs), a Variant of Stimulatory G Protein α-Subunit (Gsα), Is a Distinct',\n",
       "   'authors': [{'first': 'C', 'middle': [], 'last': 'Aydin', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Aytan', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['J'], 'last': 'Mahon', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['A W'], 'last': 'Tawfeek', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['W'], 'last': 'Kowall', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Dedeoglu', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Bastepe', 'suffix': ''}],\n",
       "   'year': 2009,\n",
       "   'venue': 'Membrane-Anchored GNAS Product that Can Mimic Gsα. Endocrinology',\n",
       "   'volume': '150',\n",
       "   'issn': '',\n",
       "   'pages': '3567--3575',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF3': {'ref_id': 'b3',\n",
       "   'title': 'Identifying the assembly intermediate in which Gag first associates with unspliced HIV-1 RNA suggests a novel model for HIV-1 RNA packaging',\n",
       "   'authors': [{'first': 'B',\n",
       "     'middle': ['C'],\n",
       "     'last': 'Barajas',\n",
       "     'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Tanaka', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['A'], 'last': 'Robinson', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['J'], 'last': 'Phuong', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Chutiraka', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['C'], 'last': 'Reed', 'suffix': ''},\n",
       "    {'first': 'Lingappa', 'middle': [], 'last': '', 'suffix': ''}],\n",
       "   'year': 2018,\n",
       "   'venue': 'J.R',\n",
       "   'volume': '',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF4': {'ref_id': 'b4',\n",
       "   'title': 'Ribosomal frameshifting in the CCR5 mRNA is regulated by miRNAs and the NMD pathway',\n",
       "   'authors': [{'first': 'A', 'middle': ['T'], 'last': 'Belew', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Meskauskas', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Musalgaonkar', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '512',\n",
       "   'issn': '',\n",
       "   'pages': '265--269',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF5': {'ref_id': 'b5',\n",
       "   'title': 'Programmed ribosomal frameshifting in HIV-1 and the SARS-CoV',\n",
       "   'authors': [{'first': 'I', 'middle': [], 'last': 'Brierley', 'suffix': ''},\n",
       "    {'first': 'Dos', 'middle': [], 'last': 'Ramos', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': ['J'], 'last': '', 'suffix': ''}],\n",
       "   'year': 2006,\n",
       "   'venue': 'Virus Res',\n",
       "   'volume': '119',\n",
       "   'issn': '',\n",
       "   'pages': '29--42',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF6': {'ref_id': 'b6',\n",
       "   'title': 'Characterization of an Efficient Coronavirus Ribosomal Frameshifting Signal : Requirement for an RNA Pseudoknot',\n",
       "   'authors': [{'first': 'I', 'middle': [], 'last': 'Brierley', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Digard', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Stephen', 'suffix': ''}],\n",
       "   'year': 1989,\n",
       "   'venue': '',\n",
       "   'volume': '57',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF8': {'ref_id': 'b8',\n",
       "   'title': 'Programmed -1 frameshifting by kinetic partitioning during impeded translocation',\n",
       "   'authors': [],\n",
       "   'year': None,\n",
       "   'venue': 'Cell',\n",
       "   'volume': '157',\n",
       "   'issn': '',\n",
       "   'pages': '1619--1631',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF9': {'ref_id': 'b9',\n",
       "   'title': 'Changed in translation: mRNA recoding by −1 programmed ribosomal frameshifting',\n",
       "   'authors': [{'first': 'N', 'middle': [], 'last': 'Caliskan', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Peske', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['V'], 'last': 'Rodnina', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'Trends Biochem. Sci',\n",
       "   'volume': '40',\n",
       "   'issn': '',\n",
       "   'pages': '265--274',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF10': {'ref_id': 'b10',\n",
       "   'title': 'HIV-1 and Human PEG10 Frameshift Elements Are Functionally Distinct and Distinguished by Novel Small Molecule Modulators',\n",
       "   'authors': [{'first': 'T', 'middle': ['S'], 'last': 'Cardno', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Shimaki', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['E'], 'last': 'Sleebs', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Lackovic', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['P'], 'last': 'Parisot', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['M'], 'last': 'Moss', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Crowe-Mcauliffe', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['F'], 'last': 'Mathew', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['D'], 'last': 'Edgar', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Kleffmann', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'PLoS One',\n",
       "   'volume': '10',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF11': {'ref_id': 'b11',\n",
       "   'title': \"The 5' UTR of HIV-1 full-length mRNA and the Tat viral protein modulate the programmed -1 ribosomal frameshift that generates HIV-1 enzymes\",\n",
       "   'authors': [{'first': 'J',\n",
       "     'middle': [],\n",
       "     'last': 'Charbonneau',\n",
       "     'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Gendron', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Ferbeyre', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Brakier-Gingras', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'RNA',\n",
       "   'volume': '18',\n",
       "   'issn': '',\n",
       "   'pages': '519--529',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF12': {'ref_id': 'b12',\n",
       "   'title': 'Dynamic pathways of −1 translational frameshifting',\n",
       "   'authors': [{'first': 'J', 'middle': [], 'last': 'Chen', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Petrov', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Johansson', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Tsai', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['E'], 'last': 'O&apos;leary', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['D'], 'last': 'Puglisi', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '512',\n",
       "   'issn': '',\n",
       "   'pages': '328--332',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF13': {'ref_id': 'b13',\n",
       "   'title': 'Frameshift mutations in mammalian target of rapamycin pathway genes and their regional heterogeneity in sporadic colorectal cancers',\n",
       "   'authors': [{'first': 'M', 'middle': ['R'], 'last': 'Choi', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['J'], 'last': 'Yoo', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['H'], 'last': 'An', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['H'], 'last': 'Lee', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'Hum. Pathol',\n",
       "   'volume': '46',\n",
       "   'issn': '',\n",
       "   'pages': '753--760',\n",
       "   'other_ids': {}}},\n",
       " 'ref_entries': {'FIGREF0': {'text': 'Tracking (NCT) (Morisaki et al., 2016) of the RNA revealed a high degree of RNA-to-RNA heterogeneity, with a subset of RNA labeled by Fab onlyindicative of FLAG epitopes from translation in the 0 frameand a smaller subset labeled by both Fab and scFvindicative of both FLAG and SunTag epitopes from canonical and frameshifted translation in the 0 and -1 frames, respectively (Fig. 1B,C and Movie S1).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF1': {'text': 'frameshifted ribosomes (marked by the SunTag) fluctuated wildly up and down (Fig. 4C, and Movie S9). These fluctuations appear to arise from the stochastic release of stalled ribosomes within the traffic jam and their subsequent translation of frameshifted epitopes.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF2': {'text': 'state or structure, similar to what has been shown in vitro with sequences encoding pseudoknots (Houck-Loomis et al., 2011). Longer pauses result in longer ribosomal traffic jams. Our model predicts these jams can extend all the way back to the initiation site, involving up to ~40 ribosomes, as shown in the lower panel of Fig. 5F. Our model predicts All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF3': {'text': 'production of downstream frameshifted protein. This unique mechanism would allow viral proteins to continue to increase in numbers during cellular stress. An open question is how these long traffic jams manage to evade protein quality control (Joazeiro, 2017; Juszkiewicz and Hegde, 2017), which recently was shown to target the interface of jammed ribosomes (Juszkiewicz et al., 2018).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF4': {'text': 'We thank Dr. Hataichanok (Mam) Scherman of the Colorado State University Protein Expression/Purification Facility and Amanda Koch for cloning and purification of the SunTag scFv-sfGFP and MCP-Halo. We thank all members of the Stasevich and Munsky labs for helpful discussion and comments. This work is supported by the W.M. Keck Foundation. AUTHOR CONTRIBUTIONS Conceptualization, TJS, KL, and BM; Methodology: KL, LA, TM, BM, and TJS; Software, KL, LA, TM, BM, and TJS; Formal Analysis, KL and LA; Experimentation, KL, TM, and TJS; Computational modeling, LA and BM; Resources, BM and TJS; Writing, Review, and Editing, KL, LA, TM, BM, and TJS; Supervision, BM and TJS; Funding Acquisition, BM and TJS.All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF5': {'text': ', M.B., Jänicke, M., Gottesbühren, U., Kleffmann, T., Legge, M., Poole, E.S., and Tate, W.P. (2007). Mammalian gene PEG10 expresses two reading frames by high efficiency -1 Frameshifting in embryonic-associated tissues. J. Biol. Chem. 282, 37359-37369.All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF6': {'text': \"Edelstein, A.D.,Tsuchida, M.A., Amodaj, N., Pinkard, H., Vale, R.D., and Stuurman, N.  (2014). Advanced methods of microscope control using μManager software. J. Biol. Methods 1, 10.Gillespie, D.T. (1976). A general method for numerically simulating the stochastic time evolution of coupled chemical reactions. J. Comput. Phys.Gorgoni, B., Ciandrini, L., McFarland, M.R., Romano, M.C., and Stansfield, I. (2016). Identification of the mRNA targets of tRNA-specific regulation using genome-wide simulation of translation. Nucleic Acids Res.Grentzmann, G., Ingram, J.A., Kelly, P.J., Gesteland, R.F., and Atkins, J.F. (1998). A dual-luciferase reporter system for studying recoding signals. RNA 4, 479-486. Grimm, J.B., English, B.P., Chen, J., Slaughter, J.P., Zhang, Z., Revyakin, A., Patel, R., Macklin, J.J., Normanno, D., Singer, R.H., et al. (2015). A general method to improve fluorophores for live-cell and single-molecule microscopy. Nat. Methods. Guerrero, S., Batisse, J., Libre, C., Bernacchi, S., Marquet, R., and Paillart, J.-C. (2015). HIV-1 replication and the cellular eukaryotic translation apparatus. Viruses 7, 199-218. Hayashi-Takanaka, Y., Yamagata, K., Wakayama, T., Stasevich, T.J., Kainuma, T., Tsurimoto, T., Tachibana, M., Shinkai, Y., Kurumizaka, H., Nozaki, N., et al. (2011). Tracking epigenetic histone modifications in single cells using Fab-based live endogenous modification labeling. Nucleic Acids Res. 39, 6475-6488. Houck-Loomis, B., Durney, M.A., Salguero, C., Shankar, N., Nagle, J.M., Goff, S.P., and D'Souza, V.M. (2011). An equilibrium-dependent retroviral mRNA switch regulates translational recoding. Nature 480, 561-564. Ingolia, N.T., Ghaemmaghami, S., Newman, J.R.S., and Weissman, J.S. (2009). Genome-wide analysis in vivo of translation with nucleotide resolution using ribosome profiling. Science (80-. ). Joazeiro, C.A.P. (2017). Ribosomal Stalling During Translation : Providing Substrates for Ribosome-Associated Protein Quality Control. 1-26. Juszkiewicz, S., and Hegde, R.S. (2017). Initiation of Quality Control during Poly(A) Translation Requires Site-Specific Ribosome Ubiquitination. Mol. Cell 65, 743-750.e4. Juszkiewicz, S., Chandrasekaran, V., Lin, Z., Kraatz, S., Ramakrishnan, V., and Hegde, R.S. (2018). ZNF598 Is a Quality Control Sensor of Collided Ribosomes. Mol. Cell 1-13.Kontos, H., Napthine, S.,and Brierley, I. (2001). Ribosomal pausing at a frameshifter RNA pseudoknot is sensitive to reading phase but shows little correlation with frameshift efficiency.Mol. Cell. Biol. 21, 8657-8670.   Lionnet, T., and Singer, R.H. (2012). Transcription goes digital. EMBO Rep. 13, 313-321. All rights reserved. No reuse allowed without permission.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF7': {'text': 'S., Klepacki, D., Karthikeyan, S., Margus, T., Thomas, P., Jones, J.E., Khan, Y., Briggs, J., Dinman, J.D.,Vázquez-Laslop, N., et al. (2017). Programmed Ribosomal Frameshifting Generates a Copper Transporter and a Copper Chaperone from the Same Gene. Mol. Cell 65, 207-219. Morisaki, T., and Stasevich, T.J. (2018). Quantifying Single mRNA Translation Kinetics in Living Cells. Cold Spring Harb. Perspect. Biol. 10, a032078. Morisaki, T., Lyon, K., DeLuca, K.F., DeLuca, J.G., English, B.P., Zhang, Z., Lavis, L.D., Grimm, J.B., Viswanathan, S., Looger, L.L., et al. (2016). Real-time quantification of single RNA translation dynamics in living cells. Science 352, 1425-1429. Mouzakis, K.D., Lang, A.L., Vander Meulen, K. a., Easterday, P.D., and Butcher, S.E. (2013). HIV-1 frameshift efficiency is primarily determined by the stability of base pairs positioned at the mRNA entrance channel of the ribosome. Nucleic Acids Res. 41, 1901-1913. Munsky, B., Neuert, G., and Van Oudenaarden, A. (2012). Using gene expression noise to understand gene regulation. Science (80-. ). Nakamura, Y., Gojobori, T., and Ikemura, T. (1999). Codon usage tabulated from the international DNA sequence databases; its status 1999. Nucleic Acids Res. Napthine, S., Ling, R., Finch, L.K., Jones, J.D., Bell, S., Brierley, I., and Firth, A.E. (2017). Protein-directed ribosomal frameshifting temporally regulates gene expression. Nat. Commun. 8, 1-11. Pichon, X., Bastide, A., Safieddine, A., Chouaib, R., Samacoits, A., Basyuk, E., Peter, M., Mueller, F., and Bertrand, E. (2016). Visualization of single endogenous polysomes reveals the dynamics of translation in live human cells. J. Cell Biol. 214, 769-781. Ritchie, D.B., Tittle, C., Rezajooei, N., Rouleau, L., Cappellano, T.R., Sikkema, W., and Woodside, M.T. (2017). Folding Heterogeneity in HIV-1 Frameshifting Hairpin. Biophys. J. 112, 368a.Schindelin, J.,Arganda-Carreras, I., Frise, E., Kaynig, V., Longair, M., Pietzsch, T.,  Preibisch, S., Rueden, C., Saalfeld, S., Schmid, B., et al. (2012). Fiji: An open-source platform for biological-image analysis. Nat. Methods 9, 676-682.Sharp, P.M., and Li, W.H. (1987). The codon adaptation index-a measure of directional synonymous codon usage bias, and its potential applications. Nucleic Acids Res.Tokunaga, M., Imamoto, N., and Sakata-Sogawa, K. (2008). Highly inclined thin illumination enables clear single-molecule imaging in cells. Nat. Methods 5, 159-161. All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF8': {'text': 'for frameshifting. Proc. Natl. Acad. Sci. U. S.A. 89, 8636-8640.   Viswanathan, S., Williams, M.E., Bloss, E.B., Stasevich, T.J., Speer, C.M., Nern, A.,  Pfeiffer, B.D., Hooks, B.M., Li, W.-P., English, B.P., et al. (2015). High-performance probes for light and electron microscopy. Nat. Methods 12, 568-576.Voigt, F., Zhang, H., Cui, X.A., Triebold, D., Liu, A.X., Eglinger, J., Lee, E.S., Chao, J.A., and Palazzo, A.F. (2017). Single-Molecule Quantification of Translation-Dependent Association of mRNAs with the Endoplasmic Reticulum. Cell Rep. 21, 3740-3753. Wang, C., Han, B., Zhou, R., and Zhuang, X. (2016). Real-Time Imaging of Translation on Single mRNA Transcripts in Live Cells. Cell 165, 990-1001. Wu, B., Eliscovich, C., Yoon, Y.J., and Singer, R.H. (2016). Translation dynamics of single mRNAs in live cells and neurons. Science 352, 1430-1435. Yordanova, M.M., Wu, C., Andreev, D.E., Sachs, M.S., and Atkins, J.F. (2015). A nascent peptide signal responsive to endogenous levels of polyamines acts to stimulate regulatory frameshifting on antizyme mRNA. J. Biol. Chem. 290, 17863-17878.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF9': {'text': \"A multi-frame tag to image single RNA frameshifting dynamics in living cells (A) The multi-frame (MF) tag contains 12 repeated FLAG epitopes in the 0 frame interspaced between 12 repeated SunTag v4 epitopes in the -1 frame. Depending on which frame is translated, nascent epitopes are labeled by fluorescent anti-FLAG antibody fragments (Cy3-Fab, green 'Y') or anti-SunTag single chain variable fragments (scFv-GFP, blue 'Y'). Following the repeat epitopes is exon 1 of the GNAS locus, in which the peripheral membrane proteins AlexX (689 aa) and XXLb1 (690 aa) were placed in the 0 and -1 frames, respectively. Preceding the multi-frame tag is a multi-frame element (MFE). In this study, the HIV-1 frameshift sequence was used as the MFE. To facilitate single-RNA tracking, a 24X MS2 stem-loop tag was also placed in the 3' UTR. This tag is labeled by MCP-HaloTag (with JF646-HaloTag ligand, red). (B) A representative cell ~10 hours after transient transfection with the multi-frame tag depicted in A. The red circle (labeled '1') highlights a non-translating RNA, the yellow circle (labeled '2') highlights a 0frame translation site (TS), and the white circle (labeled '3') highlights a 0 & -1 TS. Scale bar, 10 µm. (C) Montages showing the temporal evolution of the RNA spots circled in B. (D) A representative montage showing the loss of signal from the 0 and -1 open reading frames upon addition of the translational inhibitor puromycin (100 µg/mL).\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF10': {'text': 'Quantification of HIV-1 stimulated frameshifting (A) A schematic of the multi-frame (+FSS) and control (-FSS) reporters. (B) Average image trims of all non-translating RNA sites (no TS), 0 frame translation sites (TS), 0 & -1 TS, and -1 TS, with their respective merges. (C) The number of RNA detected per cell All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF11': {'text': 'RNA). The pie charts show the percentage of translating species per transfected cell. The number of detected RNA for each species is shown above each bar. Error bars represent S.E.M. among cells. (D) The number of detected ribosomes per translation site for the +FSS and -FSS reporters. Error bars represent S.E.M. among RNA. (E) The mean squared displacement (MSD) of tracked RNA species as a function of time. Error bars represent S.E.M. among RNA. (F) The average distance (µm) of detected translation sites from the nucleus. An outline of a representative cell on the right shows all detected RNA within the cell and their measured distance from the nuclear border (inner curve).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF12': {'text': 'The frameshift sequence stimulates frameshifting in multi-RNA translation factories (A) Cells were co-transfected with increasing concentrations of short oligo RNA encoding just the frameshift sequence, together with either the multi-frame reporter (+FSS) or control reporter (-FSS). (B) The percentage of translation sites translating just the 0 frame (green), the 0 and -1 frame (cyan), or just the -1 frame (dark blue) at all oligo concentrations for the +FSS and -FSS reporters. (C) The ratio of the intensity of RNA signals within frameshifting sites to non-frameshifting RNA at all oligo concentrations for the +FSS and -FSS reporters. In all bar graphs, the total number of each type of RNA species is shown above each bar and the error represents S.E.M. between cells.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF13': {'text': 'Ribosomal run-off at frameshifting and non-frameshifting translation sites showing harringtonine-induced ribosomal run-off from the +FSS 2x multiframe RNA reporter with FLAG (green) and SunTag epitopes (blue) in the 0 and -1 frames. The intensity of nascent chain signals within non-frameshifting translation sites(marked by just anti-FLAG Fab, green triangles, 217 translation sites initially in 19 cells) and frameshifting sites (marked by anti-SunTag scFv, cyan circles, 32 translation sites initially) drops with time as ribosomes complete translation, i.e. run-off. (B) Same as A, but with a modified multi-frame reporter with extra HA epitopes (orange, HA multi-frame tag) upstream of the frameshift sequence. The intensity of nascent chain signals from non-frameshifting translation sites (marked by just anti-HA Fab, orange triangles, 175 translation sites initially in 27 cells) and frameshifting translation sites (marked by anti-SunTag scFv, orange circles, 42 translation sites initially) drops with time as ribosomes run-off. (C) A sample single translation site encoding the modified HA multi-frame reporter (shown in B) after addition of harringtonine. A montage of image trims shows the detected RNA-, HA-, and Sun-signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting ribosomes) is plotted through time. Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF14': {'text': 'A model for bursty frameshifting (A) A schematic of the model:',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF15': {'text': 'sites through time (gray dots) is fit with a single exponential decay (black line). (C-F) Simultaneous fit of all data. (C) A bar graph comparing the measured (black) and best-fit model predicted (gray) percentage of non-frameshifting (0F) and frameshifting (-1F) translation sites. Error bars represent S.E.M. among cells. (D) A bar graph comparing the measured (black) and best-fit model predicted (gray) number of ribosomes within non-frameshifting (0F) and frameshifting (-1F) translation sites. Error bars represent S.E.M. among RNA. (E) Best-fit model (solid lines) of the data from Error bars represent S.E.M. among RNA. Below, the predicted ribosomal occupancy along the original multi-frame reporter is shown. The positions of the FSS (red) and FLAG (green) and SunTag (blue) epitopes are shown in color. (F) Best-fit model prediction of the data from Figure 4B. Error bars represent S.E.M. among RNA. Below, the predicted ribosomal occupancy along the modified HA-tagged multi-frame reporter is shown. The positions of the FSS (red) and HA (orange), FLAG (green), and SunTag (blue) epitopes are shown in color. (G) Best-fit model prediction of the intensity distribution of HA signals for frameshifting (gray) and non-frameshifting sites (black) encoding the HA multi-frame tag.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF16': {'text': 'The lifecycle of RNA encoding the HIV-1 frameshift sequence RNA (red) encoding the HIV-1 frameshift sequence (stem loop) are activated for canonical 0-frame translation. Occasionally, they switch to a frameshifting state (~8%) that can last from tens of minutes to an hour or so. Transitions to this state are stimulated through interactions with other FSS-encoding RNA, either by chance or within multi-RNA translation factories. When in the frameshifting state, longer pauses at the FSS induce All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF17': {'text': 'Accumulation of mature AlexX and XXLb1 proteins from multi-frame tags Mature AlexX (upper) and XXLb1 (lower) membrane proteins accumulate in distinct punctae that coalesce to form elongated structures that undulate dynamically. These structures are enriched at the top and bottom of cells and around the edge of the cytoplasm (arrows), consistent with membrane localization. Movie S2 shows the dynamics of these structures. In the +FSS multi-frame reporter, mature AlexX comes from 0 frame canonical translation, while mature XXLb1 comes from -1 frameshifting translation. In the -FSS control reporter, only mature AlexX accumulates, indicating little to no frameshifting. The -FSS (+1 nt) column shows accumulation of XXLb1 when an extra nucleotide is inserted in the -FSS control reporter following the start codon. This extra nucleotide pushes XXLb1 into the 0 frame, demonstrating it can be expressed.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF18': {'text': 'A cell with all possible types of translation sites A sample cell exhibiting all four possible translation sites (TS). (1) untranslated TS; (2) 0 frame only TS; (3) 0 and -1 frame translated TS; and (4) -1 frame only TS. Scale bar = 10 µm. Movie S4 showcases the dynamics of these spots.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF19': {'text': 'Epitope order in tags has little impact on percentage of frameshifting RNA The percentage of RNA translating the -1 frame only (dark blue), the 0 and -1 frames (cyan), and the 0 frame only (green) for the +FSS multi-frame tag and the +FSS reverse tag (in which FLAG and SunTag epitopes are reversed). Pie charts show the percentage of translating species per cell. Error bars represent S.E.M. among cells. All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF20': {'text': 'Calibrating translation site intensity to the number of ribosomes(A) Example cell where single mature membrane proteins were detected (green circles), along with translation sites (co-localized RNA and protein; purple circles). (B) Average images of detected spots (33 mature proteins and 35 translation sites) were fit to a Gaussian to determine their intensity ratio, shown in the bar graph on the right. This ratio provides a calibration factor to convert translation site intensites to the number of riboeoms. The error bar represents the propagated 90% confidence interval from the fit.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF21': {'text': 'Frameshifting sites have brighter RNA signals (A-E) Average RNA signals for non-translating RNA (no TS), 0 only translation sites (0 TS), and -1 frameshifting translation sites (-1 TS). The number of RNA used to generate each average image is shown. The bar graphs below show the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). Experiments were done using (A) the +FSS multi-frame tag, (B) the +FSS reverse tag, (C) the +FSS multi-frame tag imaged without anti-FLAG Fab, and (D) the -FSS control tag. Error bars represent the fitted 90%confidence interval. Aside from the -FSS control, frameshifting sites (-1 TS) have significantly brighter RNA signals compared to canonical translation sites (0 TS). Note the frameshifting observed in the -FSS control represents less than 1% of translating RNA and can be considered background frameshifting.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF22': {'text': 'Upon oligo RNA co-transfection, stimulated frameshifting still occurs at bright RNA sites, even though the fraction of bright RNA sites remains unchanged (A) Average RNA signals from non-translating RNA (no TS), 0-frame only translation sites (0 TS), and frameshifting translation sites (-1 TS) when different concentrations of oligo All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF23': {'text': 'Fits to linear portion of ribosomal run-offs provide elongation rate estimates The total intensity of detected non-frameshifting and frameshifting translation sites decays with time after addition (at t = 0 sec) of the translational initiation inhibitor harringtonine. Data is taken from Fig. 4A. If L2 is the length of the tagged portion of the open reading frame and L1 is the length of the non-tagged portion, then the linear (post-tag) portion of the decay begins from approximately L1/(L1+L2/2). This portion of the curve provides an estimate of the elongation rate during the part of the run-off where no new epitopes arebeing translated. Therefore, the run-off fit is independent of the frameshifting kinetics that occur upstream of the epitopes. Curves are normalized to their initial values.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF24': {'text': 'Fluroescence recovery after photobleach of frameshifting sites (A) Fluorescence recovery after photobleaching (FRAP) experiments were performed at frameshifting translation sites (yellow circle marks the photobleach spot). The fluorescence recovery of the 0-frame signal within these sites was quantified as a function , with sample pre, post, and recovery frames shown above. (B) The average FRAP recovery time can be fit to estimate the elongation rate. This rate is similar to what was measured with harringtonine in Fig. S7.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF25': {'text': 'Ribosome run-off from the HA multi-frame tag Ribosome run-off curve showing Sun signal in HA&Sun sites from the experiment performed in Figure 4B. The Sun signal comes from frameshifted ribosomes that have run past the frameshift sequence (FSS). Error bars represent S.E.M.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF26': {'text': 'Comparison between the bursting and constitutive models Model dynamics for the bursting (A,C,E) and constitutive model (B,D,F). Simulations were performed using the best parameter values obtained from the optimization process. (A,B) Simulated time courses representing single molecule fluctuation dynamics from 100 translation sites (a sample trace is shown in bold). Green and blue lines represent the translation of FLAG (0 frame) and Sun (-1 frame) epitopes in the +FSS multi-frame tag, respectively. (C,D) On the left, bar graphs showing the experimental (black) and simulated (gray) percentages of 0 and -1 frame translation. On the right, the number of ribosomes translating either the 0 or -1 frame. (E,F) On the left, simulated run-off (solid lines) from the frameshift sequence (FSS) of all ribosomes in non-frameshifting (black)',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF27': {'text': 'black triangles and gray circles). Error bars represent the standard error of the mean (S.E.M.). Details can be found in the Supplementary Methods.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF28': {'text': 'Single frameshifting RNA track persisting for >35 minutesA sample frameshifting translation site encoding the +FSS 2x multi-frame tag was tracked for >35 minutes. Left shows the cell at the 1200 second time point. The zoom shows the All rights reserved. No reuse allowed without permission.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF29': {'text': 'Codon usage plays a minor role in traffic jam formation (A) The codon adaptation index (CAI) for the 0 frame (top) and -1 frame (bottom) of the +FSS multi-frame tag. The y-axis shows the CAI calculated using the codon frequency in the human genome. The x-axis shows the length of the genes, in codons. In the plots, rare codons have low CAIs and common codons have high CAIs. The vertical red line represents the location of the frameshift sequence (FSS). Similar CAIs are obtained for the sequences in the 0 frame (CAI0F = 0.76) and the -1 frame (CAI-1F = 0.74).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF30': {'text': \"for 2275 seconds total. Left is the merge, right is Ch3 only. Scale bar, 10 µm. Movie S11. Sample simulation of best-fit bursty model with large ribosomal traffic jam A best-fit bursty model simulation of a translation site encoding the HA multi-frame tag. On the top, the simulated intensity of the site is tracked through time, with green showing signal from FLAG epitopes downstream of the FSS, blue showing signal from SunTag epitopes downstream of the FSS, and cyan showing signal from HA epitopes upstream of the FSS. On the bottom, individual ribosomes (gray dots) are shown moving along the reporter (gray line), with green, blue, and cyan squares marking the positions of FLAG,SunTag, and HA epitopes, respectively. As the ribosomes move along the repeat epitopes, the translated nascent chains appear as green (FLAG-tagged nascent chains), blue (Sun-tagged nascent chains), and cyan (HA-tagged nascent chains) dots that increase in size as more epitopes are translated. When the RNA is in the nonframeshifting state, the FSS is labeled 'FSS(OF)' in green; when the RNA is in the frameshifting state, the FSS is labeled 'FSS(-1F)' in blue. The simulated time is shown to the right of the reporter. A large ribosomal traffic jam is created when the RNA switches into the frameshifting state.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF31': {'text': \"-FSS-MF-AlexX (i.e. the HA multi-frame tag), the spaghetti monster HA (smHA)(Viswanathan et al., 2015)  was flanked with NotI and PstI, and then inserted into the N-terminal region of FSS-MF-AlexX. For the control constructs, FSS was removed using KpnI and XbaI. To keep the same frame for MF and AlexX, the following sequencewas ligated between KpnI and XbaI to obtain MF-AlexX (i.e. the -FSS control tag): 5'-GGT ACC GGG AAT TTT CTT CAG AGC AGA CCA GAG CCA ACA GCC GCA CCG TTT CTA GA -3'. To shift the -1 frame into 0 frame for MF and AlexX, the following sequence was ligated between KpnI and XbaI to obtain MF-AlexX (the -FSS(+1nt) tag): 5'-CGG GAA TTT TCT TCA GAG CAG ACC AGA GCC AAC AGC CGC ACC GTT CT -3'. scFv-sfGFP was amplified from pHR-scFv-GCN4-sfGFP-GB1-dWPRE (Plasmid #60907, addgene) using primers: 5'-GCG CGC ATA TGA TGG GCC CCG ACA TC -3'; 5'-GCC GGA ATT CGC CGC CTT CGG TTA CCG TGA AGG T -3'. The amplified scFv-sfGFP was flanked with NdeI and EcoRI, and then inserted into a pET21 vector backbone for expression and purification from E.coli. For HA multi-frame tag experiments, the HA epitope encoded in the scFv-sfGFP plasmid (Plasmid #60907) from Addgene was removed by site-directed mutagenesis with QuikChange Lightning (Agilent Technologies) per manufacturer's instruction using primers: 5'-CCT CCG CCT CCA CCA GCG TAA TCT GAA CTA GCG GTT CTG CCG CTG CTC ACG GTC ACC AGG GTG CCC -3'; 5'-GGG CAC CCT GGT GAC CGT GAG CAG CGG CAG AAC CGC TAG TTC AGA TTA CGC TGG TGG AGG CGG AGG -3'.All rights reserved. No reuse allowed without permission.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF32': {'text': \"tagged MCP or scFv-sfGFP was purified over a Ni-NTA-agarose (Qiagen) per the manufacturer's instructions with minor modifications. Briefly, bacteria were lysed in a PBS-based buffer with a complete set of protease inhibitors (Roche). Binding to the Ni-NTA resin was done in the presence of 10 mM imidazole. The resin was washed with 20 and 50 mM imidazole in PBS. The protein was then eluted in 300 mM imidazole in PBS. The eluted his-tagged MCP was dialyzed in a HEPES-based buffer (10% glycerol, 25 mM HEPES pH 7.9, 12.5 mM MgCl2, 100 mM KCl, 0.1 mM EDTA, 0.01 % NP-40 detergent, and 1 mM DTT), snap-frozen in liquid nitrogen, and stored at -80C. Cell culture, transfection, and bead-loading U-2 OS cells were grown using DMEM (Thermo Scientific) supplemented with: 10% (v/v) FBS, 1 mM L-glutamine and 1% (v/v) Penicillin-streptomycin. Before experiments, cells were plated on a 35 mm MatTek chamber (MatTek) and DNA was either transiently transfected with Lipofectamine LTX (Thermo Scientific) per the manufacturer's instructions or transiently transfected via bead-loading. As described previously (Hayashi-Takanaka et al., 2011; Morisaki et al., 2016) , bead-loading involved the following six steps: First, 100 µg/ml of fluorescently labeled Fab, 250 µg/ml of purified (GCN4) scFv-All rights reserved. No reuse allowed without permission.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF33': {'text': '100X images with 130 nm/pixel). With this setting, one camera detected farred signals and the other detected either red or green signals. Far red signals were detected with the 637 nm laser and the 731/137 nm emission filter (FF01-731/137/25, Semrock). Red and green signals were separated by the combination of the excitation lasers and the emission filters installed in a filter wheel (HS-625 HSFW TTL, Finger Lakes Instrumentation); namely, the 561 nm laser and 593/46 nm emission filter (FF01-593/46-25, Semrock) were used for Cy3 imaging, and the 488 nm laser and 510/42 nm emission filter (FF01-510/42-25, Semrock) were used for sfGFP or A488 imaging. Live cells were placed into a stage top incubator set to a temperature of 37°C and supplemented with 5% CO2 (Okolab) on a piezoelectric stage (PZU-2150, Applied Scientific Instrumentation). The focus was maintained using the CRISP Autofocus System (CRISP-890, Applied Scientific Instrumentation). The lasers, the cameras, the filter wheel, and the piezoelectric stage were synchronized via an Arduino Mega board (Arduino). Image acquisition was done with open source Micro-Manager software (Edelstein et al., 2014). Imaging size was set to 512 x 512 pixels 2 (66.6 x 66.6 µm 2 ), and exposure time was set to 53.64 msec.',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF34': {'text': 'When the ribosome reaches the frameshift site, , it pauses, and may shift from the 0 frame to the -1 frame. For the constitutive model, the ribosome can continue in the 0 frame with rate',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'TABREF1': {'text': 'Bursty Frameshifting Model Parameters',\n",
       "   'latex': None,\n",
       "   'type': 'table'},\n",
       "  'TABREF4': {'text': \"table 1 for the bursting model. Simulation details. To simulated the model's stochastic dynamics, we used the direct method from Gillespie's algorithm (Gillespie 1976) coded in Matlab. Genetic Algorithms were coded in Matlab. Simulations were performed on the W. M. Keck High Performance Computing Cluster at Colorado State University.\",\n",
       "   'latex': None,\n",
       "   'type': 'table'}},\n",
       " 'back_matter': [{'text': \" Figure 1 . A multi-frame tag to image single RNA frameshifting dynamics in living cells (A) The multi-frame (MF) tag contains 12 repeated FLAG epitopes in the 0 frame interspaced between 12 repeated SunTag v4 epitopes in the -1 frame. Depending on which frame is translated, nascent epitopes are labeled by fluorescent anti-FLAG antibody fragments (Cy3-Fab, green 'Y') or anti-SunTag single chain variable fragments (scFv-GFP, blue 'Y'). Following the repeat epitopes is exon 1 of the GNAS locus, in which the peripheral membrane proteins AlexX (689 aa) and XXLb1 (690 aa) were placed in the 0 and -1 frames, respectively. Preceding the multi-frame tag is a multiframe element (MFE) . In this study, the HIV-1 frameshift sequence was used as the MFE. To facilitate single-RNA tracking, a 24X MS2 stem-loop tag was also placed in the 3' UTR. This tag is labeled by MCP-HaloTag (with JF646-HaloTag ligand, red). (B) A representative cell~10 hours after transient transfection with the multi-frame tag depicted in A. The red circle (labeled '1') highlights a non-translating RNA, the yellow circle (labeled '2') highlights a 0-frame translation site . The frameshift sequence stimulates frameshifting in multi-RNA translation factories (A) Cells were co-transfected with increasing concentrations of short oligo RNA encoding just the frameshift sequence, together with either the multi-frame reporter (+FSS) or control reporter (-FSS). (B) The percentage of translation sites translating just the 0 frame (green), the 0 and -1 frame (cyan), or just the -1 frame (dark blue) at all oligo concentrations for the +FSS and -FSS reporters. (C) The ratio of the intensity of RNA signals within frameshifting sites to non-frameshifting RNA at all oligo concentrations for the +FSS and -FSS reporters. In all bar graphs, the total number of each type of RNA species is shown above each bar and the error represents S.E.M. between cells. harringtonine-induced ribosomal run-off from the +FSS 2x multi-frame RNA reporter with FLAG (green) and SunTag epitopes (blue) in the 0 and -1 frames. The intensity of nascent chain signals within non-frameshifting translation sites (marked by just anti-FLAG Fab, green triangles, 217 translation sites initially in 19 cells) and frameshifting sites (marked by anti-SunTag scFv, cyan circles, 32 translation sites initially) drops with time as ribosomes complete translation, i.e. run-off. (B) Same as A, but with a modified multi-frame reporter with extra HA epitopes (orange, HA multi-frame tag) upstream of the frameshift sequence. The intensity of nascent chain signals from non-frameshifting translation sites (marked by just anti-HA Fab, orange triangles, 175 translation sites initially in 27 cells) and frameshifting translation sites (marked by anti-SunTag scFv, orange circles, 42 translation sites initially) drops with time as ribosomes run-off. (C) A sample single translation site encoding the modified HA multi-frame reporter (shown in B) after addition of harringtonine. A montage of image trims shows the detected RNA-, HA-, and Sun-signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting ribosomes) is plotted through time. Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions.\",\n",
       "   'cite_spans': [{'start': 678, 'end': 683, 'text': '(MFE)', 'ref_id': None}],\n",
       "   'ref_spans': [{'start': 1, 'end': 9, 'text': 'Figure 1', 'ref_id': None}],\n",
       "   'section': 'annex'},\n",
       "  {'text': '(A) A schematic of the model: k ini is the translation initiation rate, k e is the translation elongation rate, k on is the rate at which RNA switch to the frameshifting state, k off is the rate at which RNA switch to the non-frameshifting state, k FSS is the pause rate at the FSS in the non-frameshifting state, k* FSS is the pause rate at the FSS in the frameshifting state, and k t is the termination rate (assumed equal to k el ). Occasionally, they switch to a frameshifting state (~8%) that can last from tens of minutes to an hour or so. Transitions to this state are stimulated through interactions with other FSS-encoding RNA, either by chance or within multi-RNA translation factories. When in the frameshifting state, longer pauses at the FSS induce longer ribosomal traffic jams. These allow frameshifting RNA to continue to be translated for up to an hour longer than non-frameshifting RNA upon global translation initiation shut-down.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure 5. A model for bursty frameshifting'},\n",
       "  {'text': 'Mature AlexX (upper) and XXLb1 (lower) membrane proteins accumulate in distinct punctae that coalesce to form elongated structures that undulate dynamically. These structures are enriched at the top and bottom of cells and around the edge of the cytoplasm (arrows), consistent with membrane localization. Movie S2 shows the dynamics of these structures. In the +FSS multi-frame reporter, mature AlexX comes from 0 frame canonical translation, while mature XXLb1 comes from -1 frameshifting translation. In the -FSS control reporter, only mature AlexX accumulates, indicating little to no frameshifting. The -FSS (+1 nt) column shows accumulation of XXLb1 when an extra nucleotide is inserted in the -FSS control reporter following the start codon. This extra nucleotide pushes XXLb1 into the 0 frame, demonstrating it can be expressed. ',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Fig. S1. Accumulation of mature AlexX and XXLb1 proteins from multi-frame tags'},\n",
       "  {'text': 'The percentage of RNA translating the -1 frame only (dark blue), the 0 and -1 frames (cyan), and the 0 frame only (green) for the +FSS multi-frame tag and the +FSS reverse tag (in which FLAG and SunTag epitopes are reversed). Pie charts show the percentage of translating species per cell. Error bars represent S.E.M. among cells.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S3. Epitope order in tags has little impact on percentage of frameshifting RNA'},\n",
       "  {'text': '(A) Example cell where single mature membrane proteins were detected (green circles), along with translation sites (co-localized RNA and protein; purple circles). (B) Average images of detected spots (33 mature proteins and 35 translation sites) were fit to a Gaussian to determine their intensity ratio, shown in the bar graph on the right. This ratio provides a calibration factor to convert translation site intensites to the number of riboeoms. The error bar represents the propagated 90% confidence interval from the fit. Error bars represent the fitted 90% confidence interval. Aside from the -FSS control, frameshifting sites (-1 TS) have significantly brighter RNA signals compared to canonical translation sites (0 TS). Note the frameshifting observed in the -FSS control represents less than 1% of translating RNA and can be considered background frameshifting. Figure S6 . Upon oligo RNA co-transfection, stimulated frameshifting still occurs at bright RNA sites, even though the fraction of bright RNA sites remains unchanged (A) Average RNA signals from non-translating RNA (no TS), 0-frame only translation sites (0 TS), and frameshifting translation sites (-1 TS) when different concentrations of oligo RNA (0, 1, and 4 mg) encoding the frameshift sequence were co-transfected into cells with the +FSS multi-frame reporter or the control -FSS reporter. The bar graph above shows the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). (B) The distributions of RNA signals from the four experiments normalized to their mean value. In all +FSS experiments, the fraction of bright RNA (defined here as having an intensity greater than or equal to 1.6 times the mean) remained constant. The -FSS experiment had fewer bright RNA, suggesting the FSS sequence facilitates incorporation into multi-RNA sites.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 872,\n",
       "     'end': 881,\n",
       "     'text': 'Figure S6',\n",
       "     'ref_id': None}],\n",
       "   'section': 'Figure S4. Calibrating translation site intensity to the number of ribosomes'},\n",
       "  {'text': 'The total intensity of detected non-frameshifting and frameshifting translation sites decays with time after addition (at t = 0 sec) of the translational initiation inhibitor harringtonine. Data is taken from Fig. 4A . If L2 is the length of the tagged portion of the open reading frame and L1 is the length of the non-tagged portion, then the linear (post-tag) portion of the decay begins from approximately L1/(L1+L2/2). This portion of the curve provides an estimate of the elongation rate during the part of the run-off where no new epitopes are being translated. Therefore, the run-off fit is independent of the frameshifting kinetics that occur upstream of the epitopes. Curves are normalized to their initial values. Figure 4B . The Sun signal comes from frameshifted ribosomes that have run past the frameshift sequence (FSS). Error bars represent S.E.M. Figure S10 . Ribosome run-off from FLAG epitopes in HA multi-frame tag For completeness, an additional experiment was performed to generate the ribosome run-off curve from FLAG epitopes in the HA multi-frame tag. Because we can only image two epitopes at the same time (since RNA is imaged in a third color), we examined FLAG and Sun epitopes (Sun epitopes were required to distinguish frameshifting and non-frameshifting sites). The FLAG run-off from this experiment is on the right (green triangles). This curve is compared to the run-off curve of HA in HA only sites (orange squares) from Figure 4B . Both experiments represent nonframeshifting sites (no Sun signal detected). Error bars represent S.E.M.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 209, 'end': 216, 'text': 'Fig. 4A', 'ref_id': None},\n",
       "    {'start': 724, 'end': 733, 'text': 'Figure 4B', 'ref_id': None},\n",
       "    {'start': 863, 'end': 873, 'text': 'Figure S10', 'ref_id': None},\n",
       "    {'start': 1455, 'end': 1464, 'text': 'Figure 4B', 'ref_id': None}],\n",
       "   'section': 'Figure S7. Fits to linear portion of ribosomal run-offs provide elongation rate estimates'},\n",
       "  {'text': 'A single frameshifting translation site encoding the HA multi-frame tag was tracked after harringtonine addition. On the top, a montage of image trims shows the detected RNA-, HA-, and Sun shown signals through time. Below, the positions of the detected signals within the site are plotted through time. On the right, the normalized total intensity of the HA Fab signal (marking all ribosomes) and the SunTag scFv signal (marking frameshifting ribosomes) is plotted through time. At about the fourth timepoint, another non-translating RNA interacts with the frameshifting RNA, leading to a burst of frameshifting signal (Sun). Gray arrows signify a burst of frameshifting, coinciding with multi-RNA interactions.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S11. Track of a stimulated frameshifting burst at a single translation site'},\n",
       "  {'text': 'Model dynamics for the bursting (A,C,E) and constitutive model (B,D,F). Simulations were performed using the best parameter values obtained from the optimization process. (A,B) Simulated time courses representing single molecule fluctuation dynamics from 100 translation sites (a sample trace is shown in bold). Green and blue lines represent the translation of FLAG (0 frame) and Sun (-1 frame) epitopes in the +FSS multi-frame tag, respectively. (C,D) On the left, bar graphs showing the experimental (black) and simulated (gray) percentages of 0 and -1 frame translation. On the right, the number of ribosomes translating either the 0 or -1 frame. (E,F) On the left, simulated run-off (solid lines) from the frameshift sequence (FSS) of all ribosomes in nonframeshifting (black) and frameshifting sites (gray), plotted with data in Fig. 4A (black triangles and gray circles). On the right, simulated run-off (solid lines) from the start site (AUG) of all ribosomes in non-frameshifting (black) and frameshifting sites (gray), plotted with data in Fig. 4B (black triangles and gray circles). Error bars represent the standard error of the mean (S.E.M.). Details can be found in the Supplementary Methods. Figure S13 . Single frameshifting RNA track persisting for >35 minutes A sample frameshifting translation site encoding the +FSS 2x multi-frame tag was tracked for >35 minutes. Left shows the cell at the 1200 second time point. The zoom shows the channels separated to highlight two frameshifted RNA that split from a single translation site. On the right, the position of the spot is plotted through time.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 835, 'end': 842, 'text': 'Fig. 4A', 'ref_id': None},\n",
       "    {'start': 1050, 'end': 1057, 'text': 'Fig. 4B', 'ref_id': None},\n",
       "    {'start': 1207, 'end': 1217, 'text': 'Figure S13', 'ref_id': None}],\n",
       "   'section': 'Figure S12. Comparison between the bursting and constitutive models'},\n",
       "  {'text': 'The codon adaptation index (CAI) for the 0 frame (top) and -1 frame (bottom) of the +FSS multiframe tag. The y-axis shows the CAI calculated using the codon frequency in the human genome. The x-axis shows the length of the genes, in codons. In the plots, rare codons have low CAIs and common codons have high CAIs. The vertical red line represents the location of the frameshift sequence (FSS). Similar CAIs are obtained for the sequences in the 0 frame (CAI 0F = 0.76) and the -1 frame (CAI -1F = 0.74).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S14. Codon usage plays a minor role in traffic jam formation'},\n",
       "  {'text': 'Intensity distribution of non-frameshifted HA signals (black) versus frameshifted HA signals (gray) produced from the HA multi-frame tag.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Figure S15. HA signal intensity distributions of non-frameshifting and frameshifting sites'}]}"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_keys(['INTRODUCTION', 'RESULTS', 'DISCUSSION'])\n"
     ]
    }
   ],
   "source": [
    "def get_sections_from_body(body):\n",
    "    sections = {}\n",
    "    for section in body:\n",
    "        if section['section'].isupper():\n",
    "            if section['section'] not in sections:\n",
    "                sections[section['section']] = ''\n",
    "            else:\n",
    "                sections[section['section']] += section['text']\n",
    "    return sections\n",
    "\n",
    "print(sections.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'INTRODUCTION': 'Although frameshifting has been extensively studied in vitro and in bulk assays (Caliskan et al., 2014; Charbonneau et al., 2012; Chen et al., 2014; Lopinski et al., 2000; Mouzakis et al., 2013; Ritchie et al., 2017) , the process has never been observed at the single molecule level in living cells. This leaves many basic questions about frameshifting unresolved. In particular, it is not clear how heterogenous frameshifting is from one RNA to another, nor is it clear if single RNA continuously frameshift in a constitutive fashion or if instead they frameshift in definable bursts, as has been observed for transcription (Lionnet and Singer, 2012) and 0-frame translation (Wu et al., 2016) . Finally, the localization of frameshifting has never been investigated, so it is not clear if frameshifting occurs all throughout the cell or is instead preferentially localized to specific sub-cellular regions. In the case of HIV-1 gag-pol, for example, previous assays have shown that 5-10% of translated protein product is frameshifted (Brierley and Dos Ramos, 2006; Dulude et al., 2002; Grentzmann et al., 1998; Mouzakis et al., 2013) . This is thought to occur when ribosomes translate through a specialized frameshift sequence (FSS) containing a stem loop structure that slows down incoming ribosomes and causes them to slip back one nucleotide on a slippery sequence preceding the stem loop. Whether or not this occurs constitutively and with equal probability on all HIV-1 RNA or if instead it occurs on a specialized subset that are in the right place, at the right time, and with the right factors remains to be determined.To directly address these sorts of questions, we have developed technology to visualize and quantify single RNA frameshifting dynamics in living cells. Using multi-frame repeat epitopes, complementary high-affinity fluorescent probes that selectively bind the epitopes, and multicolor single-molecule microscopy, we are able to simultaneously monitor the translation of single RNA into two unique nascent polypeptide chains encoded in shifted open reading frames. Application to the HIV-1 FSS uncovers unexpected heterogeneity in the production of frameshifted product and implicates a novel bursty frameshifting mechanism. Besides frameshifting, our technology can now be used to examine other translational regulatory dynamics, including upstream open reading frame selection, non-canonical initiation, and ribosomal shunting. We anticipate multi-frame nascent chain tracking will be a powerful new tool to dissect complex translational regulatory dynamics in living cells and organisms.',\n",
       " 'RESULTS': 'To ensure both frames of the MF tag encode functional proteins and have similar ribosomal occupancy, we inserted the first exon of the human GNAS1 locus downstream of the FLAG and SunTag epitopes. The GNAS1 locus contains two overlapping ORFs of roughly equivalent lengths that encode peripheral membrane proteins in the 0 and -1 frames: XXLb1 and AlexX (Abramowitz et al., 2004; Aydin et al., 2009) (Fig. 1A ). In combination with the epitopes, this arrangement has several advantages: 1) epitope densities are maximal since -1 frame epitopes act as linkers for the 0 frame and vice versa; 2) signals are digital, so frameshifted and non-frameshifted species are marked by two distinct probes/colors, and 3) epitopes are placed in nearly equivalent positions, so signals appear at roughly the same time and with similar amplification when translated with similar kinetics.As a first application of the MF tag, we focused on -1 programmed ribosomal frameshifting caused by the HIV-1 frameshift sequence (FSS). We inserted the FSS upstream of our MF tag and transiently transfected the resulting construct into U-2 OS cells. The FSS contains a slippery poly-U stretch nine nucleotides upstream of a stem loop. Two to ten hours post transfection, we observed cells with tens or hundreds of individual RNA diffusing throughout the nucleus and cytoplasm (Fig. 1B) . Nascent Chain To confirm these RNA were active translation sites, or polysomes, we performed two experiments. First, we re-imaged cells 12-24 hours after transfection. At these later time points, Fab and scFv began to accumulate in the cell membrane ( Fig. S1 and Movie this frame encoding a functional protein, as demonstrated by shifting the sequence by one nucleotide into the 0 frame ( Fig. S1 and Movie S2, right panels). Second, we treated cells with the translational inhibitor puromycin. Just minutes after treatment, we observed a dramatic decrease in the number of Fab-and/or scFv-labeled RNA, consistent with the premature release of nascent chains ( Fig. 1D and Movie S3). Together, these data provide strong evidence that we are able to detect single RNA frameshifting dynamics with the MF tag.',\n",
       " 'DISCUSSION': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint nucleotide precision in vitro, until now frameshifting had not been directly observed in a living system. To achieve this, we created a multi-frame repeat epitope tag that can light up single RNA translation sites in different colors depending on which open reading frame is being translated. Together with sensitive single-molecule microscopy and computational modeling, we have demonstrated five novel aspects of HIV-1 frameshifting:(i) frameshifted proteins originate from a small subset of RNA that frameshift with high efficiency; (ii) frameshifting occurs preferentially in multi-RNA translation factories that are facilitated in part by the frameshift sequence; (iii) frameshifting occurs in bursts on single RNA that can last for several rounds of translation; (iv) ribosomes that frameshift are paused for longer at the frameshifting sequence than ribosomes that do not frameshift;and (v) pauses at the frameshift sequence induce ribosomal traffic jams that can maintain the production of frameshifted protein despite global inhibition of translation. Fig. 6 summarizes our findings.In contrast to constitutive frameshifting on any RNA, our data indicate that frameshifting occurs in bursts on a subset of RNA. Bursty expression has been demonstrated by others, both at single transcription sites as well as at translation sites in bacteria (Lionnet and Singer, 2012) and eukaryotes (Wu et al., 2016) . The origin of frameshifting bursts is difficult to pinpoint. It is tempting to speculate that there is a unique structure the RNA takes that enhances frameshifting, particularly given the distinct pause times in our best-fit model. According to our observations, multi-RNA translation factories appear to be more susceptible to frameshift bursts. Such factories have been previously observed (Morisaki et al., 2016; Pichon et al., 2016) . Local concentration of translation factors is higher within factories, so translation kinetics are enhanced (Pichon et al., All rights reserved. No reuse allowed without permission.The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint 2016). However, the crowding of machinery at these factory sites may cause them to be more prone to errors and even more so with increased interactions with other FSSencoding RNA, the FSS being a proverbial wrench within translational factory machinery.Such low-fidelity factories may benefit viral robustness, where mutations are desirable.Consistent with this idea, HIV-1 RNAs prefer a dimeric state which may promote translation factory formation (Barajas et al., 2018) .In this study, we observed frameshifting in the context of factories in three different circumstances. First, in puromycin experiments, we observed frameshifting sites separate into two frameshifting sites upon translation shut-down (Movie S7). Second, we observed on two occasions non-translating RNA interact with a frameshifting site, after which the frameshifting signal at the site markedly increased (Figs. 4E and S11). Third, when we co-transfected FSS oligo RNA into cells (Fig. 3) , frameshifting increased in proportion to the amount we co-transfected. Collectively, these data suggest a causal ordering, whereby interactions between FSS-encoding sequences lead to enhanced frameshifting.According to our best-fit model, pauses always occur at the frameshift sequence, with longer pauses associated with frameshifting RNA compared to non-frameshifting RNA. Pausing in and of itself is therefore only a weak predictor of frameshifting, as others have shown (Kontos et al., 2001; Tu et al., 1992) . Nevertheless, the longer pauses associated with frameshifting suggest the frameshift sequence can adopt more than one The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint an elevated ribosomal occupancy as far back as ~400 codons from the FSS, a length that coincides almost perfectly with the length of the FSS-upstream sequence in HIV-1 gagpol. This is not unprecedented, as ribosome profiling experiments have also found evidence for relatively high ribosomal densities as far back as the start codon of the gag protein (Napthine et al., 2017) . These data suggest the strength of the HIV-1 pause may have evolved to on occasion produce the longest possible traffic jams. As clearance of these jams takes time, frameshifting can persist on RNA for hours despite a global shutdown of translation initiation. In effect, the jam acts like a battery that continually fuels the Beyond the imaging of HIV-1 translational frameshifting, the multi-frame tag can now be used in a variety of other contexts. For example, it can immediately be used to investigate frameshifting dynamics along other viral RNA sequences, as well as frameshifting thought to occur along endogenous human mRNA (Cardno et al., 2015) .Likewise, the multi-frame tag can be used to examine other non-canonical translation processes involving more than one open reading frame, including start-codon selection, leaky scanning, ribosomal shunting, and general translation fidelity. In fact, in an accompanying paper, a similar multi-frame tag was used to examine upstream and downstream open reading frame selection (Boersma et al.) . Like us, the authors also saw a high degree of heterogeneity between translating RNAs, with bursts of translation All rights reserved. No reuse allowed without permission.The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint initiation in multiple open reading frames similar to the bursts of frameshifting we observed at the HIV-1 frameshift sequence. Translational heterogeneity may therefore be far more common than originally appreciated, particularly when it comes to non-canonical translation. We therefore believe multi-color single molecule imaging of both canonical and non-canonical translation will become a powerful new tool for dissecting complex RNA regulatory dynamics in a variety of important contexts. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint Dulude, D., Baril, M., and Brakier-Gingras, L. (2002) . Characterization of the frameshift stimulatory signal controlling a programmed -1 ribosomal frameshift in the human immunodeficiency virus type 1. Nucleic Acids Res. 30, 5094-5102. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. All rights reserved. No reuse allowed without permission.The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. is the translation initiation rate, is the translation elongation rate, is the rate at which RNA switch to the frameshifting state, is the rate at which RNA switch to the non-frameshifting state, is the pause rate at the FSS in the non-frameshifting state, * is the pause rate at the FSS in the frameshifting All rights reserved. No reuse allowed without permission.The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint longer ribosomal traffic jams. These allow frameshifting RNA to continue to be translated for up to an hour longer than non-frameshifting RNA upon global translation initiation shutdown.All rights reserved. No reuse allowed without permission.The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint All rights reserved. No reuse allowed without permission.The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/478040 doi: bioRxiv preprint RNA (0, 1, and 4 mg) encoding the frameshift sequence were co-transfected into cells with the +FSS multi-frame reporter or the control -FSS reporter. The bar graph above shows the Gaussian fit intensity (normalized to non-translating RNA, i.e. 0 TS sites). (B)The distributions of RNA signals from the four experiments normalized to their mean value. In all +FSS experiments, the fraction of bright RNA (defined here as having an intensity greater than or equal to 1.6 times the mean) remained constant. The -FSS experiment had fewer bright RNA, suggesting the FSS sequence facilitates incorporation into multi-RNA sites. '}"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Introduction'"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "txt = 'INTRODUCTION'\n",
    "txt[0] + txt[1:].lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: 0015023cc06b5362d332b3baf348d11567ca2fbb\n",
      "\n",
      "Title: The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3\n",
      "\n",
      "Authors: [{'first': 'Joseph', 'middle': ['C'], 'last': 'Ward', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Lidia', 'middle': [], 'last': 'Lasecka-Dykes', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Chris', 'middle': [], 'last': 'Neil', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Oluwapelumi', 'middle': [], 'last': 'Adeyemi', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Sarah', 'middle': [], 'last': '', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': '', 'middle': [], 'last': 'Gold', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Niall', 'middle': [], 'last': 'Mclean', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Caroline', 'middle': [], 'last': 'Wright', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Morgan', 'middle': ['R'], 'last': 'Herod', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'David', 'middle': [], 'last': 'Kealy', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Emma', 'middle': [], 'last': '', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Warner', 'middle': [], 'last': '', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Donald', 'middle': ['P'], 'last': 'King', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Tobias', 'middle': ['J'], 'last': 'Tuthill', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'David', 'middle': ['J'], 'last': 'Rowlands', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Nicola', 'middle': ['J'], 'last': '', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Stonehouse', 'middle': [], 'last': 'A#', 'suffix': '', 'affiliation': {}, 'email': ''}]\n",
      "\n",
      "Abstract: [{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]\n"
     ]
    }
   ],
   "source": [
    "print('ID: {}'.format(paper_info['paper_id']))\n",
    "print('\\nTitle: {}'.format(paper_info['metadata']['title']))\n",
    "print('\\nAuthors: {}'.format(paper_info['metadata']['authors']))\n",
    "print('\\nAbstract: {}'.format(paper_info['abstract']))\n",
    "sections = get_sections_from_body(paper_info['body_text'])\n",
    "for section in sections.keys():\n",
    "    print('\\n{}: {}'.format(section[0] + section[1:].lower(),sections[section]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys([])"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sections.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
