{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import json\n",
    "import pandas as pd\n",
    "import yaml\n",
    "from itertools import chain\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_file_names(path):\n",
    "    # get list of papers .json from path\n",
    "    file_names = []\n",
    "    for file_name in os.listdir(path):\n",
    "        file_names.append(file_name)\n",
    "    return file_names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scraper\n",
    "\n",
    "Scraper to fetch article data online based on article information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# init session\n",
    "session = requests.session()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\david\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:3: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "# import configurations\n",
    "with open('config.yaml','r') as ymlfile:\n",
    "    cfg = yaml.load(ymlfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load metadata \n",
    "source = 'biorxiv'\n",
    "url = cfg['data-path'] + '2020-03-13/all_sources_metadata_2020-03-13.csv'\n",
    "df_meta = pd.read_csv(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "sources = df_meta['source_x'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1236 paper ids from CZI.\n",
      "There are 27337 paper ids from PMC.\n",
      "There are 566 paper ids from biorxiv.\n",
      "There are 361 paper ids from medrxiv.\n"
     ]
    }
   ],
   "source": [
    "for source in sources:\n",
    "    #print(source)\n",
    "    papers_df = df_meta[df_meta['source_x']==source][['sha','doi']]\n",
    "    papers_array = papers_df.to_numpy()\n",
    "    print('There are {} paper ids from {}.'.format(len(papers_array),source))\n",
    "    \n",
    "    # get papers relative to source"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "source_x\n",
       "CZI         1198\n",
       "PMC        16593\n",
       "biorxiv        0\n",
       "medrxiv        0\n",
       "Name: journal, dtype: int64"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.groupby(['source_x'])['journal'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1236</th>\n",
       "      <td>03203ab50eb64271a9e825f94a1b1a6c46ea14b3</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Recombination Every Day: Abundant Recombinatio...</td>\n",
       "      <td>http://dx.doi.org/10.1371/journal.pbio.0030089</td>\n",
       "      <td>PMC1054884</td>\n",
       "      <td>15737066.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>Viral recombination can dramatically impact ev...</td>\n",
       "      <td>2005 Mar 1</td>\n",
       "      <td>['Froissart, Remy', 'Roze, Denis', 'Uzest, Mar...</td>\n",
       "      <td>PLoS Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1237</th>\n",
       "      <td>98a3b0606a67d829816c1d934e2d1a7196985151</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Prospective evaluation of an internet-linked h...</td>\n",
       "      <td>http://dx.doi.org/10.1186/cc2967</td>\n",
       "      <td>PMC1065064</td>\n",
       "      <td>15566586.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>INTRODUCTION: Critical care physicians may ben...</td>\n",
       "      <td>2004 Oct 14</td>\n",
       "      <td>['Lapinsky, Stephen E', 'Wax, Randy', 'Showalt...</td>\n",
       "      <td>Crit Care</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1238</th>\n",
       "      <td>d617306cda56236d02117ae7a5fc5e7fcd015554</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Subversion of Cellular Autophagosomal Machiner...</td>\n",
       "      <td>http://dx.doi.org/10.1371/journal.pbio.0030156</td>\n",
       "      <td>PMC1084330</td>\n",
       "      <td>15884975.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>Infection of human cells with poliovirus induc...</td>\n",
       "      <td>2005 May 26</td>\n",
       "      <td>['Jackson, William T', 'Giddings, Thomas H', '...</td>\n",
       "      <td>PLoS Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1239</th>\n",
       "      <td>d619c3ceec4db4f3f350c3d5fb3842bd83f04a80</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The immediate effects of the severe acute resp...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2458-5-30</td>\n",
       "      <td>PMC1084353</td>\n",
       "      <td>15804368.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: When an emerging infectious diseas...</td>\n",
       "      <td>2005 Apr 4</td>\n",
       "      <td>['Lee, Cheng-Hua', 'Huang, Nicole', 'Chang, Ho...</td>\n",
       "      <td>BMC Public Health</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1240</th>\n",
       "      <td>d0c6b0c2d387baae89eb2898969913218b3bedff</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Molecular advances in the cell biology of SARS...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1743-422X-2-35</td>\n",
       "      <td>PMC1087510</td>\n",
       "      <td>15833113.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>In the aftermath of the SARS epidemic, there h...</td>\n",
       "      <td>2005 Apr 15</td>\n",
       "      <td>['Stark, Caren J', 'Atreya, CD']</td>\n",
       "      <td>Virol J</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1241</th>\n",
       "      <td>e6184d2db86268ba31e49b03a5aab475d5ce5ca6</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Individual sequences in large sets of gene seq...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-90</td>\n",
       "      <td>PMC1090557</td>\n",
       "      <td>15817134.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Most current DNA diagnostic tests ...</td>\n",
       "      <td>2005 Apr 8</td>\n",
       "      <td>['Gibbs, Mark J', 'Armstrong, John S', 'Gibbs,...</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1242</th>\n",
       "      <td>c6008b68c8b16e3a6a48a2cb892bac5c9353df86</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Absence of association between angiotensin con...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2334-5-26</td>\n",
       "      <td>PMC1090578</td>\n",
       "      <td>15819995.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: It has been postulated that geneti...</td>\n",
       "      <td>2005 Apr 9</td>\n",
       "      <td>['Chan, KC Allen', 'Tang, Nelson LS', 'Hui, Da...</td>\n",
       "      <td>BMC Infect Dis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1243</th>\n",
       "      <td>6f07f87e8ef78f0416556e69c88247e588f9192c</td>\n",
       "      <td>PMC</td>\n",
       "      <td>A Three-Stemmed mRNA Pseudoknot in the SARS Co...</td>\n",
       "      <td>http://dx.doi.org/10.1371/journal.pbio.0030172</td>\n",
       "      <td>PMC1110908</td>\n",
       "      <td>15884978.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>A wide range of RNA viruses use programmed −1 ...</td>\n",
       "      <td>2005 Jun 17</td>\n",
       "      <td>['Plant, Ewan P', 'Pérez-Alvarado, Gabriela C'...</td>\n",
       "      <td>PLoS Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>99b74061d99f96f6842cf3efea27058d680ed188</td>\n",
       "      <td>PMC</td>\n",
       "      <td>New Frameshifting Pseudoknot Found in SARS Virus</td>\n",
       "      <td>http://dx.doi.org/10.1371/journal.pbio.0030199</td>\n",
       "      <td>PMC1110910</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2005 Jun 17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLoS Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>9ffde004c991e9cf3c63e9143946a64ffaa9ee2a</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>02b715cc786b21f3e45f79afe877f1c02e9bfd08</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>cdcdd72d94ba5194a20b73cd5332a76f2105df94</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>201de588b3a7dbcf1319dd201fd9ee806f1557d0</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1249</th>\n",
       "      <td>7178530e60694f264910bd591cc70a707d498bfa</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1250</th>\n",
       "      <td>8e6d79c06c141e0f13bb4ff2a007ba37b992829e</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1251</th>\n",
       "      <td>52885ace333b00bb8b096a70aa7f33383f456275</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1252</th>\n",
       "      <td>01d8f392a96f0b16e1a92bd00b001f82a2507018</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1253</th>\n",
       "      <td>e875c30d3311bba4f2fc9534c477c61c8d33db84</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1254</th>\n",
       "      <td>6fc5b537c91bd586409a0a74bde7eb19071467f4</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1255</th>\n",
       "      <td>fadbf9aa0290ab261722dffc3fb1d6fd486ce0a7</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Microbial Rosetta Stone Database: A compil...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2180-5-19</td>\n",
       "      <td>PMC1127111</td>\n",
       "      <td>15850481.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: Thousands of different microorgani...</td>\n",
       "      <td>2005 Apr 25</td>\n",
       "      <td>['Ecker, David J', 'Sampath, Rangarajan', 'Wil...</td>\n",
       "      <td>BMC Microbiol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1256</th>\n",
       "      <td>c6c8b82bc5a800425b075540eff41b0af719f80f</td>\n",
       "      <td>PMC</td>\n",
       "      <td>HIV Epidemiology in Africa: Weak Variables and...</td>\n",
       "      <td>http://dx.doi.org/10.1371/journal.pmed.0020137</td>\n",
       "      <td>PMC1140948</td>\n",
       "      <td>15916469.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2005 May 31</td>\n",
       "      <td>['Brody, Stuart', 'Potterat, John J']</td>\n",
       "      <td>PLoS Med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>914c8d326beaccabd37199a9903b27fa7f9b0586</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Pseudoknots: RNA Structures with Diverse Funct...</td>\n",
       "      <td>http://dx.doi.org/10.1371/journal.pbio.0030213</td>\n",
       "      <td>PMC1149493</td>\n",
       "      <td>15941360.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>Just as proteins form distinct structural moti...</td>\n",
       "      <td>2005 Jun 14</td>\n",
       "      <td>['Staple, David W', 'Butcher, Samuel E']</td>\n",
       "      <td>PLoS Biol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1258</th>\n",
       "      <td>198ed53d956a2707716d175cc6516bb720f52d82</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1259</th>\n",
       "      <td>3c4e7b941e30be5bcd8c5222305bb53b5374f70c</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1260</th>\n",
       "      <td>c74cbed3524e1dd5d93d9db9d71c9f91ba8561e0</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1261</th>\n",
       "      <td>9c86de3d1107dd2c0fb8d9bb03fff120c41d6559</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1262</th>\n",
       "      <td>283fd3dad298787130f03473882a3dbf22420dc4</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1263</th>\n",
       "      <td>ce545fcd547a3bf435499dd84b5c87332a5c5879</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1264</th>\n",
       "      <td>5baa1fa52d730151424cc2a70242c7937016b815</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1265</th>\n",
       "      <td>c163182040644bbb1350c3d3acacacc431fb07aa</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Integration of the Gene Ontology into an objec...</td>\n",
       "      <td>http://dx.doi.org/10.1186/1471-2105-6-113</td>\n",
       "      <td>PMC1156866</td>\n",
       "      <td>15885145.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>BACKGROUND: To standardize gene product descri...</td>\n",
       "      <td>2005 May 10</td>\n",
       "      <td>['Shegogue, Daniel', 'Zheng, W Jim']</td>\n",
       "      <td>BMC Bioinformatics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28543</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Studies of respiratory viruses in personnel at...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2130878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Thirteen men wintering on an Antarctic base we...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Holmes, M. J.', 'Allen, T. R.', 'Bradburne, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28544</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Infectious diseases: annual review of signific...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2466919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reimann, H. A.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28545</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Infective hepatitis and the handling of specim...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2156316</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Marmion, B. P.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28546</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The role of Australia antigen.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1930761</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28547</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Cirrhosis Associated with the Australia Antige...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1820311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A 19-week-old English girl developed acute vir...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Wright, Ralph', 'Perkins, J. R.', 'Bower, B....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28548</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Viruses associated with acute respiratory infe...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2130859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>All respiratory illnesses which were reported ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Higgins, P. G.', 'Ellis, Eileen M.', 'Woolle...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28549</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Review: antigens and viruses in acute hepatitis.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1712859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Zuckerman, A J', 'Taylor, P E', 'Bird, R G']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28550</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Genetic Aspects of Viral Diseases: The Lilly L...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC5366631</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fenner, F. J.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28551</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Viruses and immunosuppressant drugs.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1701623</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sirtori, C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28552</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Australian Antigen and Autoantibodies in Chron...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1700632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The sera of 110 patients with chronic hepatiti...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vischer, T. L.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28553</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Presence of Particles other than the Australia...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1699390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Examination of serum specimens from a patient ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Zuckerman, A. J.', 'Taylor, Patricia E.']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28554</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Australia antigen and hepatitis.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1699357</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28555</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Viral hepatitis and tests for the Australia (h...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2427559</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>”Australia” antigen has been shown to be close...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28556</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Standardized Viral Hemagglutination and Hemagg...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC378095</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Standardized hemagglutination and hemagglutina...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Hierholzer, John C.', 'Suggs, Morris T.', 'H...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28557</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The General Practitioner and the Virus Laboratory</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1810642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Higgins, P G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28558</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Relationship of Phagocytic Activity to Pathoge...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2072135</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The pathogenicity of mouse hepatitis virus (MH...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Lavelle, G. C.', 'Starr, T. J.']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28559</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Virus prevalence in Scotland.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2635177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Grist, N. R.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28560</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Effects of corticosteroids on mouse hepatitis ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1552950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Datta, D V', 'Isselbacher, K J']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28561</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Mouse hepatitis virus (MHV3) infection in chro...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1750354</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Warren, K S', 'Rosenthal, M S', 'Domingo, E O']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28562</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Activity of nicotinamide–adenine dinucleotide ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1198772</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1. The activity of NAD pyrophosphorylase is lo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Stirpe, F.', 'Corte, E. Della']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28563</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Growth in suckling-mouse brain of \"IBV-like\" v...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC223830</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['McIntosh, K', 'Becker, W B', 'Chanock, R M']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28564</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Electron microscopy of the hepatocellular and ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1965313</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Ruebner, B. H.', 'Hirano, T.', 'Slusser, R. ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28565</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The induction of hepatitis by prior partial he...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1907275</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Ruffolo, P. R.', 'Margolis, G.', 'Kilham, L.']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28566</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Electron micrographic features of acute murine...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1920454</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Papadimitriou, J. M.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28567</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Effect of certain murine pathogens on phagocyt...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2094603</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Gledhill, A. W.', 'Bilbey, D. L.', 'Niven, J...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28568</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>EFFECT OF CORTISONE ON GENETIC RESISTANCE TO M...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC300229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Gallily, Ruth', 'Warwick, Anne', 'Bang, Fred...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28569</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Effect of a Murine Hepatitis Virus on the ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1949530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Jones, Wallace A.', 'Cohen, Richard B.']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28570</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The Interference of Mouse Hepatitis Virus with...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2083466</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gledhill, A. W.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28571</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>INTERFERENCE BETWEEN INFLUENZA VIRUS AND INFEC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC169263</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>['Groupé, Vincent', 'Pugh, Leonora H.']</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28572</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Infectious Bronchitis and Newcastle Disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1791257</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Beaudette, F. R.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27337 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sha source_x  \\\n",
       "1236   03203ab50eb64271a9e825f94a1b1a6c46ea14b3      PMC   \n",
       "1237   98a3b0606a67d829816c1d934e2d1a7196985151      PMC   \n",
       "1238   d617306cda56236d02117ae7a5fc5e7fcd015554      PMC   \n",
       "1239   d619c3ceec4db4f3f350c3d5fb3842bd83f04a80      PMC   \n",
       "1240   d0c6b0c2d387baae89eb2898969913218b3bedff      PMC   \n",
       "1241   e6184d2db86268ba31e49b03a5aab475d5ce5ca6      PMC   \n",
       "1242   c6008b68c8b16e3a6a48a2cb892bac5c9353df86      PMC   \n",
       "1243   6f07f87e8ef78f0416556e69c88247e588f9192c      PMC   \n",
       "1244   99b74061d99f96f6842cf3efea27058d680ed188      PMC   \n",
       "1245   9ffde004c991e9cf3c63e9143946a64ffaa9ee2a      PMC   \n",
       "1246   02b715cc786b21f3e45f79afe877f1c02e9bfd08      PMC   \n",
       "1247   cdcdd72d94ba5194a20b73cd5332a76f2105df94      PMC   \n",
       "1248   201de588b3a7dbcf1319dd201fd9ee806f1557d0      PMC   \n",
       "1249   7178530e60694f264910bd591cc70a707d498bfa      PMC   \n",
       "1250   8e6d79c06c141e0f13bb4ff2a007ba37b992829e      PMC   \n",
       "1251   52885ace333b00bb8b096a70aa7f33383f456275      PMC   \n",
       "1252   01d8f392a96f0b16e1a92bd00b001f82a2507018      PMC   \n",
       "1253   e875c30d3311bba4f2fc9534c477c61c8d33db84      PMC   \n",
       "1254   6fc5b537c91bd586409a0a74bde7eb19071467f4      PMC   \n",
       "1255   fadbf9aa0290ab261722dffc3fb1d6fd486ce0a7      PMC   \n",
       "1256   c6c8b82bc5a800425b075540eff41b0af719f80f      PMC   \n",
       "1257   914c8d326beaccabd37199a9903b27fa7f9b0586      PMC   \n",
       "1258   198ed53d956a2707716d175cc6516bb720f52d82      PMC   \n",
       "1259   3c4e7b941e30be5bcd8c5222305bb53b5374f70c      PMC   \n",
       "1260   c74cbed3524e1dd5d93d9db9d71c9f91ba8561e0      PMC   \n",
       "1261   9c86de3d1107dd2c0fb8d9bb03fff120c41d6559      PMC   \n",
       "1262   283fd3dad298787130f03473882a3dbf22420dc4      PMC   \n",
       "1263   ce545fcd547a3bf435499dd84b5c87332a5c5879      PMC   \n",
       "1264   5baa1fa52d730151424cc2a70242c7937016b815      PMC   \n",
       "1265   c163182040644bbb1350c3d3acacacc431fb07aa      PMC   \n",
       "...                                         ...      ...   \n",
       "28543                                       NaN      PMC   \n",
       "28544                                       NaN      PMC   \n",
       "28545                                       NaN      PMC   \n",
       "28546                                       NaN      PMC   \n",
       "28547                                       NaN      PMC   \n",
       "28548                                       NaN      PMC   \n",
       "28549                                       NaN      PMC   \n",
       "28550                                       NaN      PMC   \n",
       "28551                                       NaN      PMC   \n",
       "28552                                       NaN      PMC   \n",
       "28553                                       NaN      PMC   \n",
       "28554                                       NaN      PMC   \n",
       "28555                                       NaN      PMC   \n",
       "28556                                       NaN      PMC   \n",
       "28557                                       NaN      PMC   \n",
       "28558                                       NaN      PMC   \n",
       "28559                                       NaN      PMC   \n",
       "28560                                       NaN      PMC   \n",
       "28561                                       NaN      PMC   \n",
       "28562                                       NaN      PMC   \n",
       "28563                                       NaN      PMC   \n",
       "28564                                       NaN      PMC   \n",
       "28565                                       NaN      PMC   \n",
       "28566                                       NaN      PMC   \n",
       "28567                                       NaN      PMC   \n",
       "28568                                       NaN      PMC   \n",
       "28569                                       NaN      PMC   \n",
       "28570                                       NaN      PMC   \n",
       "28571                                       NaN      PMC   \n",
       "28572                                       NaN      PMC   \n",
       "\n",
       "                                                   title  \\\n",
       "1236   Recombination Every Day: Abundant Recombinatio...   \n",
       "1237   Prospective evaluation of an internet-linked h...   \n",
       "1238   Subversion of Cellular Autophagosomal Machiner...   \n",
       "1239   The immediate effects of the severe acute resp...   \n",
       "1240   Molecular advances in the cell biology of SARS...   \n",
       "1241   Individual sequences in large sets of gene seq...   \n",
       "1242   Absence of association between angiotensin con...   \n",
       "1243   A Three-Stemmed mRNA Pseudoknot in the SARS Co...   \n",
       "1244    New Frameshifting Pseudoknot Found in SARS Virus   \n",
       "1245   The Microbial Rosetta Stone Database: A compil...   \n",
       "1246   The Microbial Rosetta Stone Database: A compil...   \n",
       "1247   The Microbial Rosetta Stone Database: A compil...   \n",
       "1248   The Microbial Rosetta Stone Database: A compil...   \n",
       "1249   The Microbial Rosetta Stone Database: A compil...   \n",
       "1250   The Microbial Rosetta Stone Database: A compil...   \n",
       "1251   The Microbial Rosetta Stone Database: A compil...   \n",
       "1252   The Microbial Rosetta Stone Database: A compil...   \n",
       "1253   The Microbial Rosetta Stone Database: A compil...   \n",
       "1254   The Microbial Rosetta Stone Database: A compil...   \n",
       "1255   The Microbial Rosetta Stone Database: A compil...   \n",
       "1256   HIV Epidemiology in Africa: Weak Variables and...   \n",
       "1257   Pseudoknots: RNA Structures with Diverse Funct...   \n",
       "1258   Integration of the Gene Ontology into an objec...   \n",
       "1259   Integration of the Gene Ontology into an objec...   \n",
       "1260   Integration of the Gene Ontology into an objec...   \n",
       "1261   Integration of the Gene Ontology into an objec...   \n",
       "1262   Integration of the Gene Ontology into an objec...   \n",
       "1263   Integration of the Gene Ontology into an objec...   \n",
       "1264   Integration of the Gene Ontology into an objec...   \n",
       "1265   Integration of the Gene Ontology into an objec...   \n",
       "...                                                  ...   \n",
       "28543  Studies of respiratory viruses in personnel at...   \n",
       "28544  Infectious diseases: annual review of signific...   \n",
       "28545  Infective hepatitis and the handling of specim...   \n",
       "28546                     The role of Australia antigen.   \n",
       "28547  Cirrhosis Associated with the Australia Antige...   \n",
       "28548  Viruses associated with acute respiratory infe...   \n",
       "28549   Review: antigens and viruses in acute hepatitis.   \n",
       "28550  Genetic Aspects of Viral Diseases: The Lilly L...   \n",
       "28551               Viruses and immunosuppressant drugs.   \n",
       "28552  Australian Antigen and Autoantibodies in Chron...   \n",
       "28553  Presence of Particles other than the Australia...   \n",
       "28554                   Australia antigen and hepatitis.   \n",
       "28555  Viral hepatitis and tests for the Australia (h...   \n",
       "28556  Standardized Viral Hemagglutination and Hemagg...   \n",
       "28557  The General Practitioner and the Virus Laboratory   \n",
       "28558  Relationship of Phagocytic Activity to Pathoge...   \n",
       "28559                      Virus prevalence in Scotland.   \n",
       "28560  Effects of corticosteroids on mouse hepatitis ...   \n",
       "28561  Mouse hepatitis virus (MHV3) infection in chro...   \n",
       "28562  Activity of nicotinamide–adenine dinucleotide ...   \n",
       "28563  Growth in suckling-mouse brain of \"IBV-like\" v...   \n",
       "28564  Electron microscopy of the hepatocellular and ...   \n",
       "28565  The induction of hepatitis by prior partial he...   \n",
       "28566  Electron micrographic features of acute murine...   \n",
       "28567  Effect of certain murine pathogens on phagocyt...   \n",
       "28568  EFFECT OF CORTISONE ON GENETIC RESISTANCE TO M...   \n",
       "28569  The Effect of a Murine Hepatitis Virus on the ...   \n",
       "28570  The Interference of Mouse Hepatitis Virus with...   \n",
       "28571  INTERFERENCE BETWEEN INFLUENZA VIRUS AND INFEC...   \n",
       "28572        Infectious Bronchitis and Newcastle Disease   \n",
       "\n",
       "                                                  doi       pmcid   pubmed_id  \\\n",
       "1236   http://dx.doi.org/10.1371/journal.pbio.0030089  PMC1054884  15737066.0   \n",
       "1237                 http://dx.doi.org/10.1186/cc2967  PMC1065064  15566586.0   \n",
       "1238   http://dx.doi.org/10.1371/journal.pbio.0030156  PMC1084330  15884975.0   \n",
       "1239         http://dx.doi.org/10.1186/1471-2458-5-30  PMC1084353  15804368.0   \n",
       "1240         http://dx.doi.org/10.1186/1743-422X-2-35  PMC1087510  15833113.0   \n",
       "1241         http://dx.doi.org/10.1186/1471-2105-6-90  PMC1090557  15817134.0   \n",
       "1242         http://dx.doi.org/10.1186/1471-2334-5-26  PMC1090578  15819995.0   \n",
       "1243   http://dx.doi.org/10.1371/journal.pbio.0030172  PMC1110908  15884978.0   \n",
       "1244   http://dx.doi.org/10.1371/journal.pbio.0030199  PMC1110910         NaN   \n",
       "1245         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1246         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1247         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1248         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1249         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1250         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1251         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1252         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1253         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1254         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1255         http://dx.doi.org/10.1186/1471-2180-5-19  PMC1127111  15850481.0   \n",
       "1256   http://dx.doi.org/10.1371/journal.pmed.0020137  PMC1140948  15916469.0   \n",
       "1257   http://dx.doi.org/10.1371/journal.pbio.0030213  PMC1149493  15941360.0   \n",
       "1258        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1259        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1260        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1261        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1262        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1263        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1264        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "1265        http://dx.doi.org/10.1186/1471-2105-6-113  PMC1156866  15885145.0   \n",
       "...                                               ...         ...         ...   \n",
       "28543                                             NaN  PMC2130878         NaN   \n",
       "28544                                             NaN  PMC2466919         NaN   \n",
       "28545                                             NaN  PMC2156316         NaN   \n",
       "28546                                             NaN  PMC1930761         NaN   \n",
       "28547                                             NaN  PMC1820311         NaN   \n",
       "28548                                             NaN  PMC2130859         NaN   \n",
       "28549                                             NaN  PMC1712859         NaN   \n",
       "28550                                             NaN  PMC5366631         NaN   \n",
       "28551                                             NaN  PMC1701623         NaN   \n",
       "28552                                             NaN  PMC1700632         NaN   \n",
       "28553                                             NaN  PMC1699390         NaN   \n",
       "28554                                             NaN  PMC1699357         NaN   \n",
       "28555                                             NaN  PMC2427559         NaN   \n",
       "28556                                             NaN   PMC378095         NaN   \n",
       "28557                                             NaN  PMC1810642         NaN   \n",
       "28558                                             NaN  PMC2072135         NaN   \n",
       "28559                                             NaN  PMC2635177         NaN   \n",
       "28560                                             NaN  PMC1552950         NaN   \n",
       "28561                                             NaN  PMC1750354         NaN   \n",
       "28562                                             NaN  PMC1198772         NaN   \n",
       "28563                                             NaN   PMC223830         NaN   \n",
       "28564                                             NaN  PMC1965313         NaN   \n",
       "28565                                             NaN  PMC1907275         NaN   \n",
       "28566                                             NaN  PMC1920454         NaN   \n",
       "28567                                             NaN  PMC2094603         NaN   \n",
       "28568                                             NaN   PMC300229         NaN   \n",
       "28569                                             NaN  PMC1949530         NaN   \n",
       "28570                                             NaN  PMC2083466         NaN   \n",
       "28571                                             NaN   PMC169263         NaN   \n",
       "28572                                             NaN  PMC1791257         NaN   \n",
       "\n",
       "      license                                           abstract publish_time  \\\n",
       "1236    CC BY  Viral recombination can dramatically impact ev...   2005 Mar 1   \n",
       "1237    CC BY  INTRODUCTION: Critical care physicians may ben...  2004 Oct 14   \n",
       "1238    CC BY  Infection of human cells with poliovirus induc...  2005 May 26   \n",
       "1239    CC BY  BACKGROUND: When an emerging infectious diseas...   2005 Apr 4   \n",
       "1240    CC BY  In the aftermath of the SARS epidemic, there h...  2005 Apr 15   \n",
       "1241    CC BY  BACKGROUND: Most current DNA diagnostic tests ...   2005 Apr 8   \n",
       "1242    CC BY  BACKGROUND: It has been postulated that geneti...   2005 Apr 9   \n",
       "1243    CC BY  A wide range of RNA viruses use programmed −1 ...  2005 Jun 17   \n",
       "1244    CC BY                                                NaN  2005 Jun 17   \n",
       "1245    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1246    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1247    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1248    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1249    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1250    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1251    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1252    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1253    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1254    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1255    CC BY  BACKGROUND: Thousands of different microorgani...  2005 Apr 25   \n",
       "1256    CC BY                                                NaN  2005 May 31   \n",
       "1257    CC BY  Just as proteins form distinct structural moti...  2005 Jun 14   \n",
       "1258    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1259    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1260    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1261    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1262    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1263    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1264    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "1265    CC BY  BACKGROUND: To standardize gene product descri...  2005 May 10   \n",
       "...       ...                                                ...          ...   \n",
       "28543     NaN  Thirteen men wintering on an Antarctic base we...          NaN   \n",
       "28544     NaN                                                NaN          NaN   \n",
       "28545     NaN                                                NaN          NaN   \n",
       "28546     NaN                                                NaN          NaN   \n",
       "28547     NaN  A 19-week-old English girl developed acute vir...          NaN   \n",
       "28548     NaN  All respiratory illnesses which were reported ...          NaN   \n",
       "28549     NaN                                                NaN          NaN   \n",
       "28550     NaN                                                NaN          NaN   \n",
       "28551     NaN                                                NaN          NaN   \n",
       "28552     NaN  The sera of 110 patients with chronic hepatiti...          NaN   \n",
       "28553     NaN  Examination of serum specimens from a patient ...          NaN   \n",
       "28554     NaN                                                NaN          NaN   \n",
       "28555     NaN  ”Australia” antigen has been shown to be close...          NaN   \n",
       "28556     NaN  Standardized hemagglutination and hemagglutina...          NaN   \n",
       "28557     NaN                                                NaN          NaN   \n",
       "28558     NaN  The pathogenicity of mouse hepatitis virus (MH...          NaN   \n",
       "28559     NaN                                                NaN          NaN   \n",
       "28560     NaN                                                NaN          NaN   \n",
       "28561     NaN                                                NaN          NaN   \n",
       "28562     NaN  1. The activity of NAD pyrophosphorylase is lo...          NaN   \n",
       "28563     NaN                                                NaN          NaN   \n",
       "28564     NaN                                                NaN          NaN   \n",
       "28565     NaN                                                NaN          NaN   \n",
       "28566     NaN                                                NaN          NaN   \n",
       "28567     NaN                                                NaN          NaN   \n",
       "28568     NaN                                                NaN          NaN   \n",
       "28569     NaN                                                NaN          NaN   \n",
       "28570     NaN                                                NaN          NaN   \n",
       "28571     NaN                                                NaN          NaN   \n",
       "28572     NaN                                                NaN          NaN   \n",
       "\n",
       "                                                 authors             journal  \\\n",
       "1236   ['Froissart, Remy', 'Roze, Denis', 'Uzest, Mar...           PLoS Biol   \n",
       "1237   ['Lapinsky, Stephen E', 'Wax, Randy', 'Showalt...           Crit Care   \n",
       "1238   ['Jackson, William T', 'Giddings, Thomas H', '...           PLoS Biol   \n",
       "1239   ['Lee, Cheng-Hua', 'Huang, Nicole', 'Chang, Ho...   BMC Public Health   \n",
       "1240                    ['Stark, Caren J', 'Atreya, CD']             Virol J   \n",
       "1241   ['Gibbs, Mark J', 'Armstrong, John S', 'Gibbs,...  BMC Bioinformatics   \n",
       "1242   ['Chan, KC Allen', 'Tang, Nelson LS', 'Hui, Da...      BMC Infect Dis   \n",
       "1243   ['Plant, Ewan P', 'Pérez-Alvarado, Gabriela C'...           PLoS Biol   \n",
       "1244                                                 NaN           PLoS Biol   \n",
       "1245   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1246   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1247   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1248   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1249   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1250   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1251   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1252   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1253   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1254   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1255   ['Ecker, David J', 'Sampath, Rangarajan', 'Wil...       BMC Microbiol   \n",
       "1256               ['Brody, Stuart', 'Potterat, John J']            PLoS Med   \n",
       "1257            ['Staple, David W', 'Butcher, Samuel E']           PLoS Biol   \n",
       "1258                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1259                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1260                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1261                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1262                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1263                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1264                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "1265                ['Shegogue, Daniel', 'Zheng, W Jim']  BMC Bioinformatics   \n",
       "...                                                  ...                 ...   \n",
       "28543  ['Holmes, M. J.', 'Allen, T. R.', 'Bradburne, ...                 NaN   \n",
       "28544                                     Reimann, H. A.                 NaN   \n",
       "28545                                     Marmion, B. P.                 NaN   \n",
       "28546                                                NaN                 NaN   \n",
       "28547  ['Wright, Ralph', 'Perkins, J. R.', 'Bower, B....                 NaN   \n",
       "28548  ['Higgins, P. G.', 'Ellis, Eileen M.', 'Woolle...                 NaN   \n",
       "28549     ['Zuckerman, A J', 'Taylor, P E', 'Bird, R G']                 NaN   \n",
       "28550                                      Fenner, F. J.                 NaN   \n",
       "28551                                         Sirtori, C                 NaN   \n",
       "28552                                     Vischer, T. L.                 NaN   \n",
       "28553        ['Zuckerman, A. J.', 'Taylor, Patricia E.']                 NaN   \n",
       "28554                                                NaN                 NaN   \n",
       "28555                                                NaN                 NaN   \n",
       "28556  ['Hierholzer, John C.', 'Suggs, Morris T.', 'H...                 NaN   \n",
       "28557                                       Higgins, P G                 NaN   \n",
       "28558                 ['Lavelle, G. C.', 'Starr, T. J.']                 NaN   \n",
       "28559                                       Grist, N. R.                 NaN   \n",
       "28560                 ['Datta, D V', 'Isselbacher, K J']                 NaN   \n",
       "28561  ['Warren, K S', 'Rosenthal, M S', 'Domingo, E O']                 NaN   \n",
       "28562                  ['Stirpe, F.', 'Corte, E. Della']                 NaN   \n",
       "28563     ['McIntosh, K', 'Becker, W B', 'Chanock, R M']                 NaN   \n",
       "28564  ['Ruebner, B. H.', 'Hirano, T.', 'Slusser, R. ...                 NaN   \n",
       "28565   ['Ruffolo, P. R.', 'Margolis, G.', 'Kilham, L.']                 NaN   \n",
       "28566                               Papadimitriou, J. M.                 NaN   \n",
       "28567  ['Gledhill, A. W.', 'Bilbey, D. L.', 'Niven, J...                 NaN   \n",
       "28568  ['Gallily, Ruth', 'Warwick, Anne', 'Bang, Fred...                 NaN   \n",
       "28569         ['Jones, Wallace A.', 'Cohen, Richard B.']                 NaN   \n",
       "28570                                    Gledhill, A. W.                 NaN   \n",
       "28571            ['Groupé, Vincent', 'Pugh, Leonora H.']                 NaN   \n",
       "28572                                   Beaudette, F. R.                 NaN   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "1236                           NaN            NaN          True  \n",
       "1237                           NaN            NaN          True  \n",
       "1238                           NaN            NaN          True  \n",
       "1239                           NaN            NaN          True  \n",
       "1240                           NaN            NaN          True  \n",
       "1241                           NaN            NaN          True  \n",
       "1242                           NaN            NaN          True  \n",
       "1243                           NaN            NaN          True  \n",
       "1244                           NaN            NaN         False  \n",
       "1245                           NaN            NaN          True  \n",
       "1246                           NaN            NaN         False  \n",
       "1247                           NaN            NaN         False  \n",
       "1248                           NaN            NaN         False  \n",
       "1249                           NaN            NaN         False  \n",
       "1250                           NaN            NaN         False  \n",
       "1251                           NaN            NaN         False  \n",
       "1252                           NaN            NaN         False  \n",
       "1253                           NaN            NaN         False  \n",
       "1254                           NaN            NaN         False  \n",
       "1255                           NaN            NaN         False  \n",
       "1256                           NaN            NaN          True  \n",
       "1257                           NaN            NaN          True  \n",
       "1258                           NaN            NaN          True  \n",
       "1259                           NaN            NaN         False  \n",
       "1260                           NaN            NaN         False  \n",
       "1261                           NaN            NaN         False  \n",
       "1262                           NaN            NaN         False  \n",
       "1263                           NaN            NaN         False  \n",
       "1264                           NaN            NaN         False  \n",
       "1265                           NaN            NaN         False  \n",
       "...                            ...            ...           ...  \n",
       "28543                          NaN            NaN           NaN  \n",
       "28544                          NaN            NaN           NaN  \n",
       "28545                          NaN            NaN           NaN  \n",
       "28546                          NaN            NaN           NaN  \n",
       "28547                          NaN            NaN           NaN  \n",
       "28548                          NaN            NaN           NaN  \n",
       "28549                          NaN            NaN           NaN  \n",
       "28550                          NaN            NaN           NaN  \n",
       "28551                          NaN            NaN           NaN  \n",
       "28552                          NaN            NaN           NaN  \n",
       "28553                          NaN            NaN           NaN  \n",
       "28554                          NaN            NaN           NaN  \n",
       "28555                          NaN            NaN           NaN  \n",
       "28556                          NaN            NaN           NaN  \n",
       "28557                          NaN            NaN           NaN  \n",
       "28558                          NaN            NaN           NaN  \n",
       "28559                          NaN            NaN           NaN  \n",
       "28560                          NaN            NaN           NaN  \n",
       "28561                          NaN            NaN           NaN  \n",
       "28562                          NaN            NaN           NaN  \n",
       "28563                          NaN            NaN           NaN  \n",
       "28564                          NaN            NaN           NaN  \n",
       "28565                          NaN            NaN           NaN  \n",
       "28566                          NaN            NaN           NaN  \n",
       "28567                          NaN            NaN           NaN  \n",
       "28568                          NaN            NaN           NaN  \n",
       "28569                          NaN            NaN           NaN  \n",
       "28570                          NaN            NaN           NaN  \n",
       "28571                          NaN            NaN           NaN  \n",
       "28572                          NaN            NaN           NaN  \n",
       "\n",
       "[27337 rows x 14 columns]"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['source_x']=='PMC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intensive Care Med\n",
      "Cell Discovery\n",
      "Journal of Clinical Medicine\n",
      "J Clin Med\n",
      "Infectious Disease Modelling\n",
      "The Lancet. Global health\n",
      "Journal of Travel Medicine\n",
      "mBio\n",
      "Global Health Research and Policy\n",
      "Eurosurveillance\n",
      "Bioinformatics (Oxford, England)\n",
      "Journal of Korean Medical Science\n",
      "Pathogens\n",
      "EClinicalMedicine\n",
      "Infection, Genetics and Evolution\n",
      "Microbes and Infection\n",
      "Cell Research\n",
      "Journal of travel medicine\n",
      "Hong Kong Medical Journal\n",
      "Innovative Biosystems and Bioengineering\n",
      "Precision Clinical Medicine\n",
      "Frontiers in Microbiology\n",
      "International Journal of Environmental Research and Public Health\n",
      "Protein Cell\n",
      "Critical Care\n",
      "Osong Public Health and Research Perspectives\n",
      "Viruses\n",
      "Intensive Care Medicine\n",
      "Data in Brief\n",
      "Epidemiol Infect\n",
      "Nature\n",
      "National Science Review\n",
      "Biosafety and Health\n",
      "International Journal of Infectious Diseases\n",
      "The Lancet Global Health\n",
      "Infection Control & Hospital Epidemiology\n",
      "Journal of Immunology Research\n",
      "Clinical & Translational Immunology\n",
      "The Lancet Planetary Health\n",
      "Chin Med J (Engl)\n",
      "Journal of the Chinese Medical Association : JCMA\n",
      "Revista Peruana de Investigación en Salud\n",
      "Military Medical Research\n",
      "Healthcare (Basel)\n",
      "Infection Prevention in Practice\n",
      "Eye\n",
      "J Chin Med Assoc\n",
      "Infectious Diseases of Poverty\n",
      "F1000Research\n",
      "Swiss Med Wkly\n",
      "Journal of Epidemiology and Global Health\n",
      "Journal of Infection in Developing Countries\n",
      "Antiviral Research\n",
      "The Lancet\n",
      "International Journal of Oral Science\n",
      "Chinese medical journal\n",
      "Archives of disease in childhood. Fetal and neonatal edition\n",
      "Journal of Translational Medicine\n",
      "Euro surveillance : bulletin Europeen sur les maladies transmissibles = European communicable disease bulletin\n",
      "Healthcare\n",
      "Journal of the Formosan Medical Association\n",
      "One Health\n",
      "The Lancet Public Health\n",
      "The Lancet Digital Health\n",
      "Journal of Sport and Health Science\n",
      "Korean journal of radiology\n",
      "Pediatric Reports. 2020;12(1)\n",
      "Global Social Welfare\n",
      "Journal of Microbiology, Immunology and Infection\n",
      "Signal Transduction and Targeted Therapy\n",
      "Journal of the American College of Emergency Physicians Open\n",
      "Vaccine\n",
      "Nat Rev Microbiol\n",
      "BJGP open\n",
      "Infectious Disease Reports\n",
      "Revista Argentina de Microbiología\n",
      "Korean J Radiol\n",
      "Forensic Science International: Mind and Law\n",
      "Open Forum Infectious Diseases\n",
      "BMC Medicine\n",
      "Acta Pharmaceutica Sinica B\n",
      "Biochemical and Biophysical Research Communications\n",
      "Virus research\n",
      "BMJ\n",
      "J Korean Med Sci\n",
      "Journal of Integrative Medicine\n",
      "Nature Microbiology\n",
      "Influenza and Other Respiratory Viruses\n",
      "Global Biosecurity\n",
      "Clin Chem\n",
      "Medicine (Baltimore)\n",
      "Juntendo Medical Journal\n",
      "Int J Occup Environ Med\n",
      "Emerg Infect Dis\n",
      "Emerging infectious diseases\n",
      "Zhonghua Yi Xue Za Zhi\n",
      "JAMA - Journal of the American Medical Association\n",
      "Jama\n",
      "JAMA\n",
      "Chinese Journal of Infectious Diseases\n",
      "Chinese Journal of Digestion\n",
      "Chinese Critical Care Medicine\n",
      "Alcoholism & Drug Abuse Weekly\n",
      "Am J Hematol\n",
      "ChemBioChem\n",
      "Chemistry & Industry\n",
      "Cytometry Part A\n",
      "Drug Dev Res\n",
      "European journal of immunology\n",
      "Chinese Journal of Experimental Ophthalmology\n",
      "Journal of medical virology\n",
      "Journal of Medical Virology\n",
      "J Med Virol\n",
      "Chinese Journal of Orthopaedic Trauma\n",
      "Chinese Journal of Hospital Administration\n",
      "Medicinal Research Reviews\n",
      "Mental Health Weekly\n",
      "Pediatric Pulmonology\n",
      "Ultrasound in Obstetrics & Gynecology\n",
      "Methods in molecular biology (Clifton, N.J.)\n",
      "Herz\n",
      "nan\n",
      "Chinese Journal of Laboratory Medicine\n",
      "European Journal of Nuclear Medicine and Molecular Imaging\n",
      "Eur J Nucl Med Mol Imaging\n",
      "Chinese Journal of Gastrointestinal Surgery\n",
      "Cardiovascular and interventional radiology\n",
      "European Radiology\n",
      "Journal of Neurology\n",
      "Graefe's Archive for Clinical and Experimental Ophthalmology\n",
      "Arch Virol\n",
      "Clin Oral Investig\n",
      "Journal of Public Health\n",
      "Health Care Management Science\n",
      "Journal of Medical Systems\n",
      "Virus Genes\n",
      "CME\n",
      "Sci China Life Sci\n",
      "Science China Life Sciences\n",
      "Journal of General Internal Medicine\n",
      "Chin J Integr Med\n",
      "Virologica Sinica\n",
      "Virol Sin\n",
      "Neuroscience Bulletin\n",
      "Building Simulation\n",
      "World J Pediatr\n",
      "Zhongguo Dang Dai Er Ke Za Zhi\n",
      "World Journal of Pediatrics\n",
      "World journal of pediatrics : WJP\n",
      "Can J Anaesth\n",
      "Canadian Journal of Anesthesia/Journal canadien d'anesthésie\n",
      "VirusDisease\n",
      "MMW - Fortschritte der Medizin\n",
      "Infection\n",
      "Current Tropical Medicine Reports\n",
      "Anaesthesia Critical Care & Pain Medicine\n",
      "Academic Radiology\n",
      "American Journal of Infection Control\n",
      "American Journal of Obstetrics and Gynecology\n",
      "Asian Journal of Psychiatry\n",
      "N Z Med J\n",
      "Biosensors and Bioelectronics\n",
      "British journal of anaesthesia\n",
      "Clinica Chimica Acta\n",
      "Cell\n",
      "Chest\n",
      "Cell Host & Microbe\n",
      "Clinical Imaging\n",
      "Clinical Microbiology and Infection\n",
      "Computers in Biology and Medicine\n",
      "Infez Med\n",
      "Complementary Therapies in Clinical Practice\n",
      "Drug Discovery Today\n",
      "European Journal of Pharmaceutical Sciences\n",
      "Health Policy and Technology\n",
      "Intensive and Critical Care Nursing\n",
      "International journal of antimicrobial agents\n",
      "Zhejiang Da Xue Xue Bao Yi Xue Ban\n",
      "International Journal of Antimicrobial Agents\n",
      "International Journal of Hygiene and Environmental Health\n",
      "Int J Hyg Environ Health\n",
      "The Journal of the Association of Physicians of India\n",
      "International Journal of Pharmaceutics\n",
      "International Journal of Surgery\n",
      "Indian Journal of Tuberculosis\n",
      "Journal of the American Academy of Dermatology\n",
      "Journal of the American College of Radiology : JACR\n",
      "Chinese Journal of Stomatology\n",
      "Journal of Anxiety Disorders\n",
      "Journal of Autoimmunity\n",
      "Journal of Clinical Virology\n",
      "Journal of Global Antimicrobial Resistance\n",
      "Journal of Genetics and Genomics\n",
      "Journal of Hospital Infection\n",
      "J Hosp Infect\n",
      "Journal of Infection\n",
      "Journal of Pharmaceutical Analysis\n",
      "Chinese Journal of Burns\n",
      "Journal of Thoracic Oncology\n",
      "Chinese Journal of Surgery\n",
      "Kidney International\n",
      "Life Sciences\n",
      "Mayo Clinic Proceedings\n",
      "Med Clin (Barc)\n",
      "Deutsche Apotheker Zeitung\n",
      "Midwifery\n",
      "Molecular Immunology\n",
      "Nursing for Women's Health\n",
      "Progress in Cardiovascular Diseases\n",
      "Pharmacological Research\n",
      "Science Magazine\n",
      "Psychiatry Research\n",
      "Public Health\n",
      "Pulmonology\n",
      "Chinese Journal of Radiology\n",
      "Rev Clin Esp\n",
      "Research in Social and Administrative Pharmacy\n",
      "Science Bulletin\n",
      "Social Science & Medicine\n",
      "Talanta\n",
      "Trends in Microbiology\n",
      "Travel Medicine and Infectious Disease\n",
      "Travel medicine and infectious disease\n",
      "Travel Med Infect Dis\n",
      "Transfusion Medicine Reviews\n",
      "Veterinary Microbiology\n",
      "Virus Research\n",
      "Lancet\n",
      "Chinese Journal of Clinical Infectious Diseases\n",
      "Chinese Journal of Emergency Medicine\n",
      "Chinese Journal of Experimental and Clinical Virology\n",
      "New Scientist\n",
      "The Lancet Oncology\n",
      "The Lancet Infectious Diseases\n",
      "Lancet Infect Dis\n",
      "The Lancet Respiratory Medicine\n",
      "The Lancet Psychiatry\n",
      "The Lancet Child & Adolescent Health\n",
      "The Lancet Gastroenterology & Hepatology\n",
      "Infect Control Hosp Epidemiol\n",
      "Journal of Medicinal Chemistry\n",
      "ACS Infect Dis\n",
      "C&EN Global Enterprise\n",
      "Nature Reviews Drug Discovery\n",
      "Nature Biotechnology\n",
      "Nature Medicine\n",
      "Nat Med\n",
      "Harefuah\n",
      "Cell Res\n",
      "Cellular & Molecular Immunology AB  -The novel coronavirus was denoted as 2019-nCoV by WHO (https://www.who.int/emergencies/diseases/novel-coronavirus-2019) and the Wuhan pneumonia was named as novel coronavirus-infected pneumonia (NCIP) by Chinese scient\n",
      "Cellular & Molecular Immunology\n",
      "Nature Ecology & Evolution\n",
      "Nat Microbiol\n",
      "Nature Reviews Cardiology\n",
      "Nature reviews. Microbiology\n",
      "Nature Reviews Microbiology\n",
      "Nature medicine\n",
      "Biochem J\n",
      "Gastroenterology\n",
      "Journal of Cardiothoracic and Vascular Anesthesia\n",
      "New England Journal of Medicine\n",
      "The New England journal of medicine\n",
      "N Engl J Med\n",
      "Proc Natl Acad Sci U S A\n",
      "J Biol Chem\n",
      "Applicable Analysis\n",
      "New Zealand Veterinary Journal\n",
      "Journal of Histotechnology\n",
      "Veterinary Quarterly\n",
      "Baylor University Medical Center Proceedings\n",
      "Health Communication\n",
      "Anatolia\n",
      "Pathog Glob Health\n",
      "Human Vaccines & Immunotherapeutics\n",
      "Emerging microbes & infections\n",
      "Emerg Microbes Infect\n",
      "Population Health Management\n",
      "Surg Infect (Larchmt)\n",
      "Am J Clin Pathol\n",
      "Zhonghua Nei Ke Za Zhi\n",
      "Annals of Work Exposures and Health\n",
      "Clinical Infectious Diseases,\n",
      "Clinical Infectious Diseases\n",
      "Clinical infectious diseases : an official publication of the Infectious Diseases Society of America\n",
      "Clin Infect Dis\n",
      "Clinical Chemistry\n",
      "Chinese Journal of Trauma\n",
      "Pathogens and Disease\n",
      "International Journal of Epidemiology\n",
      "The Journal of Infectious Diseases\n",
      "Journal of Infectious Diseases\n",
      "Journal of the Pediatric Infectious Diseases Society\n",
      "J Travel Med\n",
      "Occupational Medicine\n",
      "QJM\n",
      "Chinese Pediatric Emergency Medicine\n",
      "Current opinion in pulmonary medicine\n",
      "Invest Radiol\n",
      "Investigative radiology\n",
      "Chinese Medical Journal\n",
      "Chinese Journal of Behavioral Medicine and Brain Science\n",
      "Microbiology and Immunology\n",
      "Economic Outlook\n",
      "BJOG: An International Journal of Obstetrics & Gynaecology\n",
      "J Dig Dis\n",
      "Microbial Biotechnology\n",
      "Academic Emergency Medicine\n",
      "African Journal of Ecology\n",
      "Am J Transplant\n",
      "Allergy\n",
      "Acta Obstetricia et Gynecologica Scandinavica\n",
      "British Journal of Dermatology\n",
      "European Journal of Clinical Investigation\n",
      "European journal of clinical investigation\n",
      "International Journal of Clinical Practice\n",
      "Int J Clin Pract\n",
      "Int Nurs Rev\n",
      "Chinese Journal of Anesthesiology\n",
      "Chinese Journal of Hepatology\n",
      "J Evid Based Med\n",
      "Journal of Thrombosis and Haemostasis\n",
      "Journal of Zoology\n",
      "Letters in applied microbiology\n",
      "Psychiatry and Clinical Neurosciences\n",
      "Respirology\n",
      "Transbound Emerg Dis\n",
      "Transboundary and Emerging Diseases\n",
      "Tropical Medicine & International Health\n",
      "Vox Sanguinis\n",
      "Science\n",
      "Science (New York, N.Y.)\n",
      "Sccience\n",
      "Journal of clinical microbiology\n",
      "Journal of Virology\n",
      "Br J Ophthalmol\n",
      "BMJ (Clinical research ed.)\n",
      "Gut\n",
      "The Veterinary record\n",
      "Radiology\n",
      "Chinese Journal of Microbiology and Immunology\n",
      "Singapore Med J\n",
      "American journal of respiratory and critical care medicine\n",
      "Environment and Planning A: Economy and Space\n",
      "Canadian Association of Radiologists Journal\n",
      "Indoor and Built Environment\n",
      "Chinese Journal of Geriatrics\n",
      "The European respiratory journal\n",
      "Tropical Medicine and Health\n",
      "Vet Ital\n",
      "Veterinaria italiana\n",
      "Asian Pacific journal of allergy and immunology\n",
      "Br J Nurs\n",
      "Southwest Journal of Pulmonary and Critical Care. 2020;20(2):69-70\n",
      "Chinese Journal of Organ Transplantation\n",
      "CMAJ : Canadian Medical Association journal = journal de l'Association medicale canadienne\n",
      "Clin Chem Lab Med\n",
      "Diagnosis\n",
      "MMWR Morb Mortal Wkly Rep\n",
      "MMWR. Morbidity and mortality weekly report\n",
      "MMWR and Morbidity and Mortality Weekly Report\n",
      "Yi chuan = Hereditas\n",
      "Laeknabladid\n",
      "The Journal of Information Science and Technology Association\n",
      "Acta Med Port\n",
      "Ann Palliat Med\n",
      "Annals of Translational Medicine\n",
      "Am J Public Health\n",
      "Postępy Mikrobiologii\n",
      "SSRN\n",
      "Current topics in medicinal chemistry\n",
      "Reviews on recent clinical trials\n",
      "American Journal of Roentgenology\n",
      "AJR Am J Roentgenol\n",
      "AJR. American journal of roentgenology\n",
      "J Biol Regul Homeost Agents\n",
      "Bulletin of the World Health Organization\n",
      "Chinese medical sciences journal = Chung-kuo i hsueh k'o hsueh tsa chih\n",
      "European review for medical and pharmacological sciences\n",
      "Eastern Mediterranean Health Journal\n",
      "Chinese Journal of Perinatal Medicine\n",
      "Journal of Nepal Medical Association\n",
      "Commun Dis Intell (2018)\n",
      "6\n",
      "Biology\n",
      "Biomolecules\n",
      "Chinese Journal of Applied Clinical Pediatrics\n",
      "International Journal of Molecular Sciences\n",
      "Journal of Risk and Financial Management\n",
      "Zhonghua Wei Chang Wai Ke Za Zhi\n",
      "Zhonghua yi xue za zhi\n",
      "Zhonghua wai ke za zhi [Chinese journal of surgery]\n",
      "Zhonghua Wai Ke Za Zhi\n",
      "Zhonghua Er Ke Za Zhi\n",
      "[Zhonghua yan ke za zhi] Chinese journal of ophthalmology\n",
      "Zhonghua Yan Ke Za Zhi\n",
      "Zhonghua Kou Qiang Yi Xue Za Zhi\n",
      "Zhonghua Jie He He Hu Xi Za Zhi\n",
      "Zhonghua jie he he hu xi za zhi = Zhonghua jiehe he huxi zazhi = Chinese journal of tuberculosis and respiratory diseases\n",
      "Zhonghua Xin Xue Guan Bing Za Zhi\n",
      "Zhonghua xin xue guan bing za zhi\n",
      "Zhonghua Yu Fang Yi Xue Za Zhi\n",
      "Zhonghua yu fang yi xue za zhi [Chinese journal of preventive medicine]\n",
      "Zhonghua Bing Li Xue Za Zhi\n",
      "Zhonghua Zhong Liu Za Zhi\n",
      "Zhonghua zhong liu za zhi [Chinese journal of oncology]\n",
      "Zhonghua Liu Xing Bing Xue Za Zhi\n",
      "Zhonghua liu xing bing xue za zhi = Zhonghua liuxingbingxue zazhi\n",
      "Zhonghua Lao Dong Wei Sheng Zhi Ye Bing Za Zhi\n",
      "Zhonghua shao shang za zhi = Zhonghua shaoshang zazhi = Chinese journal of burns\n",
      "Zhonghua Shao Shang Za Zhi\n",
      "Zhonghua xue ye xue za zhi = Zhonghua xueyexue zazhi\n",
      "Chinese Journal of Preventive Medicine\n",
      "Chinese Journal of Pediatrics\n",
      "Zhonghua Gan Zang Bing Za Zhi\n",
      "Zhonghua Er Bi Yan Hou Tou Jing Wai Ke Za Zhi\n",
      "Zhongguo Fei Ai Za Zhi\n",
      "J Prev Med Public Health\n",
      "Biomol Ther (Seoul)\n",
      "Infect Chemother\n",
      "Asian Pacific Journal of Tropical Medicine\n",
      "Epidemiol Health\n",
      "Swiss medical weekly\n",
      "BioScience TrendsAB - As of January 22, 2020, a total of 571 cases of the 2019-new coronavirus (2019-nCoV) have been reported in 25 provinces (districts and cities) in China. At present, there is no vaccine or antiviral treatment for human and animal coro\n",
      "BioScience Trends\n",
      "Biosci Trends\n",
      "Drug Discoveries & Therapeutics\n",
      "Medical Journal of Australia\n",
      "Ann Intern Med\n",
      "Chinese Journal of Evidence-Based Medicine\n",
      "Clinical medicine (London, England)\n",
      "Japanese Journal of Infectious Diseases\n",
      "International Journal of Interactive Multimedia and Artificial Intelligence\n",
      "Rev Med Suisse\n",
      "Boletín bibliográfico de la Biblioteca Médica Nacional\n",
      "MMWR\n",
      "Emergency medicine practice\n",
      "Iranian South Medical Journal\n",
      "Chinese Journal of Obstetrics and Gynecology\n",
      "Nota Técnica-SVS/SES-RJ\n",
      "Chinese Journal of Oncology\n",
      "International Journal of Traditional Chinese Medicine\n",
      "Archives of Academic Emergency Medicine\n",
      "Infezioni in Medicina\n",
      "PLoS Biol\n",
      "Crit Care\n",
      "BMC Public Health\n",
      "Virol J\n",
      "BMC Bioinformatics\n",
      "BMC Infect Dis\n",
      "BMC Microbiol\n",
      "PLoS Med\n",
      "BMC Genomics\n",
      "J Inflamm (Lond)\n",
      "Respir Res\n",
      "BMC Med Inform Decis Mak\n",
      "BMC Evol Biol\n",
      "Retrovirology\n",
      "BMC Pulm Med\n",
      "Global Health\n",
      "Hum Resour Health\n",
      "Genome Biol\n",
      "Int J Health Geogr\n",
      "Environ Health Perspect\n",
      "Genet Vaccines Ther\n",
      "Environ Health\n",
      "BMC Health Serv Res\n",
      "Immunome Res\n",
      "PLoS Pathog\n",
      "BMC Mol Biol\n",
      "BMC Dev Biol\n",
      "BMC Biotechnol\n",
      "BMC Vet Res\n",
      "J Autoimmune Dis\n",
      "Microb Cell Fact\n",
      "BMC Med Ethics\n",
      "Emerg Themes Epidemiol\n",
      "PLoS Comput Biol\n",
      "Arthritis Res Ther\n",
      "J Med Internet Res\n",
      "J Ethnobiol Ethnomed\n",
      "Health Res Policy Syst\n",
      "BMC Immunol\n",
      "BMC Musculoskelet Disord\n",
      "PLoS One\n",
      "BMC Med\n",
      "Acta Vet Scand\n",
      "Infect Agent Cancer\n",
      "BMC Biol\n",
      "Theor Biol Med Model\n",
      "J Transl Med\n",
      "PLoS Negl Trop Dis\n",
      "J Immune Based Ther Vaccines\n",
      "J Orthop Surg\n",
      "Mol Vis\n",
      "Clin Dev Immunol\n",
      "Source Code Biol Med\n",
      "Dev Immunol\n",
      "Immun Ageing\n",
      "J R Soc Interface\n",
      "BMC Med Res Methodol\n",
      "Comp Funct Genomics\n",
      "Proc Biol Sci\n",
      "Reprod Biol Endocrinol\n",
      "PLoS Genet\n",
      "Cases J\n",
      "BMC Res Notes\n",
      "Ann Clin Microbiol Antimicrob\n",
      "BMC Med Genomics\n",
      "J Gen Virol\n",
      "J Occup Med Toxicol\n",
      "Asia Pac Fam Med\n",
      "J Neuroinflammation\n",
      "Evol Bioinform Online\n",
      "Int J Mol Sci\n",
      "Curr Genomics\n",
      "Interdiscip Perspect Infect Dis\n",
      "Clin Mol Allergy\n",
      "Proteome Sci\n",
      "Plant J\n",
      "Aust New Zealand Health Policy\n",
      "Open Virol J\n",
      "BMC Physiol\n",
      "Parasit Vectors\n",
      "J Mol Biol\n",
      "BMC Plant Biol\n",
      "Biomark Insights\n",
      "Biotechnol J\n",
      "Transfus Med\n",
      "Acta Crystallogr D Biol Crystallogr\n",
      "Ital J Pediatr\n",
      "Curr Oncol\n",
      "Case Rep Med\n",
      "J Hematol Oncol\n",
      "J Biomed Sci\n",
      "BMC Pregnancy Childbirth\n",
      "BMC Gastroenterol\n",
      "Health Qual Life Outcomes\n",
      "J Health Popul Nutr\n",
      "AIDS Res Ther\n",
      "Thorax\n",
      "Implement Sci\n",
      "Curr Pharm Des\n",
      "Mol Microbiol\n",
      "Aging (Albany NY)\n",
      "BMC Med Genet\n",
      "Alcohol Res Health\n",
      "Brain Pathol\n",
      "Bioinform Biol Insights\n",
      "Adv Bioinformatics\n",
      "Int J Environ Res Public Health\n",
      "BMC Struct Biol\n",
      "BMC Cell Biol\n",
      "J Oncol\n",
      "BMC Cancer\n",
      "BMC Genet\n",
      "Biol Direct\n",
      "Mediators Inflamm\n",
      "Algorithms Mol Biol\n",
      "BMC Pediatr\n",
      "Cancer Cell Int\n",
      "Vet Med Int\n",
      "BMC Syst Biol\n",
      "Int J Microbiol\n",
      "Mar Drugs\n",
      "Parasite Immunol\n",
      "Hepat Res Treat\n",
      "J Cardiovasc Dis Res\n",
      "Biochem Res Int\n",
      "Epidemiol Perspect Innov\n",
      "Int J Alzheimers Dis\n",
      "BMC Proc\n",
      "Mol Cancer\n",
      "BMC Fam Pract\n",
      "Vet Res\n",
      "Evid Based Complement Alternat Med\n",
      "J Pregnancy\n",
      "J Drug Deliv\n",
      "Trials\n",
      "BMC Neurol\n",
      "Biomed Imaging Interv J\n",
      "Malar J\n",
      "Pulm Med\n",
      "J Med Case Reports\n",
      "J Viral Hepat\n",
      "Philos Trans R Soc Lond B Biol Sci\n",
      "J Parasitol Res\n",
      "Clin Trials\n",
      "Toxins (Basel)\n",
      "Comput Math Methods Med\n",
      "Ann Intensive Care\n",
      "J Cardiothorac Surg\n",
      "Sci Pharm\n",
      "Clin Proteomics\n",
      "Emerg Health Threats J\n",
      "Acta Crystallogr Sect F Struct Biol Cryst Commun\n",
      "Ir Vet J\n",
      "J Trop Med\n",
      "BMC Emerg Med\n",
      "Int J Hypertens\n",
      "Pan Afr Med J\n",
      "Adv Prev Med\n",
      "Sensors (Basel)\n",
      "BMC Int Health Hum Rights\n",
      "Beilstein J Org Chem\n",
      "Nutrients\n",
      "Adv Virol\n",
      "ScientificWorldJournal\n",
      "Scand J Trauma Resusc Emerg Med\n",
      "BMC Complement Altern Med\n",
      "Biol Proced Online\n",
      "J Clin Med Res\n",
      "Int J Pept\n",
      "Int Arch Med\n",
      "Int J Cell Biol\n",
      "J Pathog\n",
      "Acta Crystallogr Sect E Struct Rep Online\n",
      "Acta Naturae\n",
      "Open Biol\n",
      "Int J Emerg Med\n",
      "J Nanobiotechnology\n",
      "Stem Cells Int\n",
      "J Clin Virol\n",
      "Am J Trop Med Hyg\n",
      "Mol Biol Int\n",
      "Pediatr Rheumatol Online J\n",
      "Front Microbiol\n",
      "Case Rep Rheumatol\n",
      "Crit Care Res Pract\n",
      "Front Cell Infect Microbiol\n",
      "Antimicrob Resist Infect Control\n",
      "Arch Public Health\n",
      "Int J Inflam\n",
      "Influenza Res Treat\n",
      "Iran J Parasitol\n",
      "J Glob Health\n",
      "eLife\n",
      "Case Rep Pulmonol\n",
      "Brain Behav\n",
      "Int J Equity Health\n",
      "Isr J Health Policy Res\n",
      "Cough\n",
      "Front Genet\n",
      "J Infect Dis\n",
      "Case Rep Transplant\n",
      "BMC Biochem\n",
      "J Clin Bioinforma\n",
      "Transl Neurodegener\n",
      "Cell Commun Signal\n",
      "Bioinformatics\n",
      "Front Immunol\n",
      "Iran J Radiol\n",
      "Case Rep Infect Dis\n",
      "Front Plant Sci\n",
      "J Ophthalmol\n",
      "Microbiology\n",
      "Antiinflamm Antiallergy Agents Med Chem\n",
      "Mol Oncol\n",
      "Orphanet J Rare Dis\n",
      "Cell Biosci\n",
      "Glob Health Action\n",
      "Front Neural Circuits\n",
      "Genome Announc\n",
      "Stand Genomic Sci\n",
      "Int J Mol Med\n",
      "Chem Cent J\n",
      "Springerplus\n",
      "Biomed Res Int\n",
      "J Pept Sci\n",
      "Curr Protein Pept Sci\n",
      "J Biomed Res\n",
      "Zoonoses Public Health\n",
      "Int J Pediatr\n",
      "Scientifica (Cairo)\n",
      "Cilia\n",
      "Int J Biomed Sci\n",
      "BMC Neurosci\n",
      "Curr Neuropharmacol\n",
      "Croat Med J\n",
      "World Allergy Organ J\n",
      "Eur J Immunol\n",
      "ISRN Vet Sci\n",
      "Clin Transl Med\n",
      "Virology\n",
      "Parasite\n",
      "J Toxicol\n",
      "J Mol Signal\n",
      "G3 (Bethesda)\n",
      "Infect Dis Poverty\n",
      "Rambam Maimonides Med J\n",
      "Gut Pathog\n",
      "Diagn Pathol\n",
      "Nucleic Acids Res\n",
      "J Clin Microbiol\n",
      "Cancers (Basel)\n",
      "Vet J\n",
      "PeerJ\n",
      "Cell Microbiol\n",
      "J Epidemiol\n",
      "Sci Rep\n",
      "Pharmaceuticals (Basel)\n",
      "Exp Ther Med\n",
      "Iran J Pharm Res\n",
      "J Virol\n",
      "ASN Neuro\n",
      "EMBO Mol Med\n",
      "BMC Nephrol\n",
      "Infect Ecol Epidemiol\n",
      "Mol Genet Genomic Med\n",
      "Front Physiol\n",
      "Front Endocrinol (Lausanne)\n",
      "F1000Res\n",
      "Acta Neuropathol Commun\n",
      "EURASIP J Bioinform Syst Biol\n",
      "Cells\n",
      "European J Med Plants\n",
      "Cell Death Dis\n",
      "Front Public Health\n",
      "Genes (Basel)\n",
      "Microbiologyopen\n",
      "Jundishapur J Nat Pharm Prod\n",
      "Front Cell Neurosci\n",
      "Health Psychol Behav Med\n",
      "Database (Oxford)\n",
      "Genome Med\n",
      "J Immunol Res\n",
      "BMC Psychiatry\n",
      "Colomb Med (Cali).; 44(2):121-128\n",
      "Case Rep Crit Care\n",
      "Cardiovasc Diabetol\n",
      "Autoimmune Dis\n",
      "J Epidemiol Community Health\n",
      "World J Surg Oncol\n",
      "Crit Ultrasound J\n",
      "AMB Express\n",
      "Influenza Other Respir Viruses\n",
      "J Diabetes Res\n",
      "Genome Biol Evol\n",
      "J R Soc Med\n",
      "JMIR Mhealth Uhealth\n",
      "Front Aging Neurosci\n",
      "BMC Med Educ\n",
      "Am J Bioeth\n",
      "Bioengineered\n",
      "Mediterr J Hematol Infect Dis\n",
      "Chin Med\n",
      "Glob Health Sci Pract\n",
      "Mol Hum Reprod\n",
      "J Proteomics Bioinform\n",
      "Physiol Rep\n",
      "Pharmaceutics\n",
      "Int J Endocrinol\n",
      "J Vis Exp\n",
      "Iran J Reprod Med\n",
      "Jundishapur J Microbiol\n",
      "Bull World Health Organ\n",
      "IUCrJ\n",
      "Evol Appl\n",
      "Front Oncol\n",
      "Cell Regen (Lond)\n",
      "New Phytol\n",
      "Multidiscip Respir Med\n",
      "Stem Cell Res Ther\n",
      "DNA Res\n",
      "Biosensors (Basel)\n",
      "Daru\n",
      "Biology (Basel)\n",
      "Front Bioeng Biotechnol\n",
      "Vet Res Forum\n",
      "Animal\n",
      "Vet Microbiol\n",
      "Breast Cancer Res\n",
      "Mol Med Rep\n",
      "eLife.; 4:e05564\n",
      "Nat Commun\n",
      "Mil Med Res\n",
      "Vet Rec\n",
      "Health Policy Plan\n",
      "Mem Inst Oswaldo Cruz\n",
      "West J Emerg Med\n",
      "Oncotarget\n",
      "Oxid Med Cell Longev\n",
      "ISRN Biotechnol\n",
      "Cell J\n",
      "Part Fibre Toxicol\n",
      "J Exp Clin Cancer Res\n",
      "BMC Nurs\n",
      "Sci Data\n",
      "Fibrogenesis Tissue Repair\n",
      "J Biomark\n",
      "Comput Struct Biotechnol J\n",
      "Body Soc\n",
      "Vaccines (Basel)\n",
      "Animals (Basel)\n",
      "Virus Res\n",
      "J Clin Cell Immunol\n",
      "Nanoscale Res Lett\n",
      "In Silico Pharmacol\n",
      "Front Med (Lausanne)\n",
      "Case Rep Pediatr\n",
      "J Glob Health.; 5(2):020404\n",
      "J Anim Sci Biotechnol\n",
      "EMBO Rep\n",
      "Int J Circumpolar Health\n",
      "Exp Mol Med\n",
      "Microbes Environ\n",
      "J Cell Sci\n",
      "Int J Genomics\n",
      "J Vet Med\n",
      "Comput Intell Neurosci\n",
      "J Glob Health.; 5(1):010408\n",
      "J Anim Sci Technol\n",
      "Life Sci Soc Policy\n",
      "J Glob Health.; 5(2):050418\n",
      "MethodsX\n",
      "Cureus.; 7(9):e332\n",
      "eLife.; 4:e09015\n",
      "Molecules\n",
      "Syst Rev\n",
      "Glob Soc Policy\n",
      "Pak J Med Sci\n",
      "EJNMMI Res\n",
      "Bull Math Biol\n",
      "Immunity\n",
      "Front Vet Sci\n",
      "J Glob Health.; 5(2):020417\n",
      "Eur Clin Respir J\n",
      "Genomics Proteomics Bioinformatics\n",
      "Front Mol Biosci\n",
      "Front Pharmacol\n",
      "BioData Min\n",
      "Microbiome\n",
      "Ecohealth\n",
      "Cell Tissue Res\n",
      "eLife.; 4:e11785\n",
      "Sci Bull (Beijing)\n",
      "3 Biotech\n",
      "Expert Rev Vaccines\n",
      "Int J Bacteriol\n",
      "Front Pediatr\n",
      "eLife.; 4:e09186\n",
      "Vet World\n",
      "Glob J Health Sci\n",
      "J Anal Methods Chem\n",
      "Intensive Care Med Exp\n",
      "Evol Med Public Health\n",
      "Virus Evol\n",
      "Harm Reduct J\n",
      "Dev Psychol\n",
      "New Microbes New Infect\n",
      "eLife.; 5:e11275\n",
      "Cell Chem Biol\n",
      "Semin Immunopathol\n",
      "mSphere\n",
      "Case Rep Neurol Med\n",
      "Plant Biotechnol J\n",
      "eLife.; 5:e12469\n",
      "Einstein (Sao Paulo)\n",
      "Appl Microbiol Biotechnol\n",
      "Lipids Health Dis\n",
      "eLife.; 5:e14228\n",
      "ISRN Mol Biol\n",
      "Health Technol (Berl)\n",
      "Can Respir J\n",
      "Biodivers Data J\n",
      "Int Braz J Urol\n",
      "Aging Cell\n",
      "J Clin Immunol\n",
      "BMC Anesthesiol\n",
      "Environ Sci Pollut Res Int\n",
      "Curr Treat Options Infect Dis\n",
      "Int J Med Chem\n",
      "Eye Vis (Lond)\n",
      "J Immunol\n",
      "Int J Ment Health Syst\n",
      "Front Hum Neurosci\n",
      "Cell Death Discov\n",
      "eLife.; 5:e16965\n",
      "Poult Sci\n",
      "Infect Immun\n",
      "Microarrays (Basel)\n",
      "Inf Serv Use\n",
      "Virol Mycol\n",
      "Dis Markers\n",
      "Rev Lat Am Enfermagem\n",
      "Exp Neurol\n",
      "eLife.; 5:e18311\n",
      "Microorganisms\n",
      "J Clin Exp Dent\n",
      "J Med Life\n",
      "Western Pac Surveill Response J\n",
      "Exp Transl Stroke Med\n",
      "J Healthc Eng\n",
      "Front Psychol\n",
      "mSystems\n",
      "Cell Mol Life Sci\n",
      "Antimicrob Agents Chemother\n",
      "Int J Health Policy Manag\n",
      "R Soc Open Sci\n",
      "Iran J Child Neurol\n",
      "eLife.; 5:e17636\n",
      "Monatsh Chem\n",
      "Bundesgesundheitsblatt Gesundheitsforschung Gesundheitsschutz\n",
      "Alzheimers Res Ther\n",
      "BMC Hematol\n",
      "Hum Vaccin Immunother\n",
      "Hum Ecol Interdiscip J\n",
      "eLife.; 5:e18491\n",
      "J Nucleic Acids\n",
      "Acta Neuropathol\n",
      "J Nat Med\n",
      "J Venom Anim Toxins Incl Trop Dis\n",
      "Proteomes\n",
      "FEBS Open Bio\n",
      "BMJ Open\n",
      "J Glob Health.; 6(2):020704\n",
      "Biosci Rep\n",
      "eNeuro\n",
      "Trop Anim Health Prod\n",
      "Protein Sci\n",
      "Nanomaterials (Basel)\n",
      "Cardiol Res\n",
      "Crim Justice Ethics\n",
      "Parasitol Res\n",
      "Health Secur\n",
      "Microb Genom\n",
      "Med Educ Online\n",
      "Front Neurosci\n",
      "Disaster Mil Med\n",
      "BMC Oral Health\n",
      "Front Neurol\n",
      "Cell Rep\n",
      "J Cell Mol Med\n",
      "Z Gesundh Wiss\n",
      "J Environ Public Health\n",
      "Zool Res\n",
      "eLife.; 6:e23172\n",
      "Appl Environ Microbiol\n",
      "EXCLI J\n",
      "Front Mol Neurosci\n",
      "Porcine Health Manag\n",
      "Lancet Glob Health\n",
      "Euro Surveill\n",
      "Conserv Physiol\n",
      "Chem Sci\n",
      "Parasite.; 24:12\n",
      "Emerg Med Int\n",
      "eLife.; 6:e25687\n",
      "Trends Microbiol\n",
      "JMM Case Rep\n",
      "Biomedicines\n",
      "Clin Vaccine Immunol\n",
      "Med Hist\n",
      "eLife.; 6:e22069\n",
      "Genom Data\n",
      "Afr J Lab Med\n",
      "Stem Cells Transl Med\n",
      "Heart Rhythm\n",
      "Balkan Med J\n",
      "Open Forum Infect Dis\n",
      "Diseases\n",
      "Pharmacol Res Perspect\n",
      "Biochim Biophys Acta\n",
      "Gigascience\n",
      "Biomed Eng Online\n",
      "Pneumonia (Nathan)\n",
      "Natl Sci Rev\n",
      "Respir Med\n",
      "Cell Discov\n",
      "eLife.; 6:e26875\n",
      "Genetics\n",
      "Food Nutr Res\n",
      "Trop Dis Travel Med Vaccines\n",
      "Brain Sci\n",
      "Lung Cancer Int\n",
      "Int J Telerehabil\n",
      "Mol Neurodegener\n",
      "J Exp Med\n",
      "Burns Trauma\n",
      "Iran J Biotechnol\n",
      "Health Expect\n",
      "Biochem Biophys Rep\n",
      "Eur J Clin Microbiol Infect Dis\n",
      "Adv Sci (Weinh)\n",
      "Pathog Immun\n",
      "Vet Sci\n",
      "Biol Res\n",
      "Pediatr Radiol\n",
      "J Pharm Sci\n",
      "eLife.; 6:e26738\n",
      "Diagnostics (Basel)\n",
      "JMIR Public Health Surveill\n",
      "NPJ Vaccines\n",
      "J Intensive Care\n",
      "SAHARA J\n",
      "Autophagy\n",
      "Contrast Media Mol Imaging\n",
      "Aliment Pharmacol Ther\n",
      "Microb Cell.; 4(10):311-330\n",
      "Cent Asian J Glob Health\n",
      "Cancer Immunol Immunother\n",
      "NPJ Genom Med\n",
      "NPJ Syst Biol Appl\n",
      "Glob Health Res Policy\n",
      "Pract Lab Med\n",
      "AIMS Public Health\n",
      "eLife.; 6:e30700\n",
      "Front Cell Dev Biol\n",
      "Confl Health\n",
      "J Emerg Dis Virol\n",
      "J Allergy Clin Immunol\n",
      "Iran J Microbiol\n",
      "Cancer Med\n",
      "Front Synaptic Neurosci\n",
      "J Glob Health.; 7(2):020102\n",
      "Heart Fail Rev\n",
      "J Biol Eng\n",
      "Mol Psychiatry\n",
      "Front Neuroanat\n",
      "eLife.; 7:e31257\n",
      "Mol Cell Proteomics\n",
      "eLife.; 7:e31662\n",
      "Res Integr Peer Rev\n",
      "Wellcome Open Res\n",
      "Front Zool\n",
      "Clin Sci (Lond)\n",
      "Noncoding RNA\n",
      "Syst Biol\n",
      "Int J Epidemiol\n",
      "Mol Biol Evol\n",
      "eLife.; 7:e31919\n",
      "Anal Biochem\n",
      "Clin Transl Immunology\n",
      "Can J Infect Dis Med Microbiol\n",
      "Rev Bras Ter Intensiva\n",
      "Int J Infect Dis\n",
      "Mol Cell\n",
      "J Pediatric Infect Dis Soc\n",
      "J Appl Oral Sci\n",
      "Bone Res\n",
      "Trends Immunol\n",
      "Virulence\n",
      "Clin Transl Allergy\n",
      "Curr Opin Microbiol\n",
      "Clin Pract Cases Emerg Med\n",
      "Epidemics\n",
      "Cureus.; 10(3):e2387\n",
      "Int J Public Health\n",
      "Vet Med Sci\n",
      "Aging Dis\n",
      "J Neurovirol\n",
      "Data Brief\n",
      "Dev World Bioeth\n",
      "J Enzyme Inhib Med Chem\n",
      "Med Sci (Basel)\n",
      "J Orthop Surg Res\n",
      "PPAR Res\n",
      "J Behav Decis Mak\n",
      "Plant Mol Biol\n",
      "Trop Med Infect Dis\n",
      "Respirol Case Rep\n",
      "Proteomics\n",
      "Curationis\n",
      "Mycopathologia\n",
      "Antonie Van Leeuwenhoek\n",
      "Ther Clin Risk Manag\n",
      "J Med Case Rep\n",
      "Mycology\n",
      "Mol Ecol Resour\n",
      "Adv Exp Med Biol\n",
      "Acta Crystallogr D Struct Biol\n",
      "Pharm Biol\n",
      "Mamm Genome\n",
      "Genet Mol Biol\n",
      "J S Afr Vet Assoc\n",
      "Trans R Soc Trop Med Hyg\n",
      "J Med Libr Assoc\n",
      "Hum Gene Ther\n",
      "Mol Biol Rep\n",
      "Medicines (Basel)\n",
      "Paediatr Int Child Health\n",
      "Histochem Cell Biol\n",
      "BJGP Open\n",
      "Environ Int\n",
      "Rev Paul Pediatr\n",
      "eLife.; 7:e37663\n",
      "J Glob Health.; 8(2):020310\n",
      "Commun Biol\n",
      "Microbiol Resour Announc\n",
      "Appl Netw Sci\n",
      "GMS Hyg Infect Control\n",
      "Onderstepoort J Vet Res\n",
      "eLife.; 7:e41426\n",
      "Health Sci Rep\n",
      "Environ Health Prev Med\n",
      "Nucleic Acid Ther\n",
      "Inflamm Regen\n",
      "eLife.; 7:e39494\n",
      "eLife.; 7:e40977\n",
      "Iran Biomed J\n",
      "High Throughput\n",
      "Micromachines (Basel)\n",
      "Materials (Basel)\n",
      "Microb Ecol\n",
      "Case Rep Oncol Med\n",
      "Curr Clin Microbiol Rep\n",
      "Transgenic Res\n",
      "Biol Open\n",
      "eLife.; 8:e42037\n",
      "Immunology\n",
      "Antivir Chem Chemother\n",
      "Animal Model Exp Med\n",
      "J Lipid Res\n",
      "Iran J Public Health\n",
      "BMJ Glob Health\n",
      "J Community Health\n",
      "Polymers (Basel)\n",
      "J Immunol Sci\n",
      "eLife.; 8:e43983\n",
      "Medchemcomm\n",
      "Life Sci Alliance\n",
      "BMC Mol Cell Biol\n",
      "Int J Parasitol Drugs Drug Resist\n",
      "EMBO J\n",
      "BMC Geriatr\n",
      "Front Chem\n",
      "eLife.; 8:e43764\n",
      "Methods Protoc\n",
      "Tanaffos\n",
      "Am J Reprod Immunol\n",
      "Thromb J\n",
      "Brief Bioinform\n",
      "eLife.; 8:e44345\n",
      "Int J Qual Stud Health Well-being\n",
      "eLife.; 8:e45205\n",
      "Pilot Feasibility Stud\n",
      "Orthop Surg\n",
      "Cureus.; 11(4):e4577\n",
      "Glob Chall\n",
      "Med J Aust\n",
      "eLife.; 8:e47117\n",
      "Antibodies (Basel)\n",
      "Rev Panam Salud Publica\n",
      "Biometals\n",
      "Trop Med Health\n",
      "Am J Respir Crit Care Med\n",
      "Antioxidants (Basel)\n",
      "Medicina (Kaunas)\n",
      "Biochim Biophys Acta Gen Subj\n",
      "Microb Cell.; 6(9):373-396\n",
      "eLife.; 8:e47615\n",
      "Theranostics\n",
      "J Innate Immun\n",
      "eLife.; 8:e46681\n",
      "Ann Glob Health.; 84(4):710-716\n",
      "Int Health\n",
      "Research (Wash D C)\n",
      "Int J Biol Sci\n",
      "ISME J\n",
      "Ecol Evol\n",
      "Int J Parasitol Parasites Wildl\n",
      "eLife.; 8:e47261\n",
      "Asian-Australas J Anim Sci\n",
      "Vet Q\n",
      "Metabolites\n",
      "Transl Psychiatry\n",
      "Clin Otolaryngol\n",
      "Risk Anal\n",
      "J Cancer\n",
      "JAMA Pediatr\n",
      "J Clin Lab Anal\n",
      "Case Rep Vet Med\n",
      "J Hum Genet\n",
      "Vet Med (Auckl)\n",
      "Anim Genet\n",
      "Heliyon\n",
      "Brief Funct Genomics\n",
      "Insects\n",
      "Lancet Planet Health\n",
      "J Mol Evol\n",
      "Anim Front\n",
      "Children (Basel)\n",
      "J Fungi (Basel)\n",
      "Infect Dis Model\n",
      "Antibiotics (Basel)\n",
      "eLife.; 8:e51230\n",
      "Front Neurorobot\n",
      "Hortic Res\n",
      "Epidemiol Infect.; 147:e310\n",
      "Mol Ecol\n",
      "Signal Transduct Target Ther\n",
      "Epidemiol Infect.; 148:e4\n",
      "Epidemiol Infect.; 148:e22\n",
      "Mass Spectrom Rev\n",
      "Cell Biol Int\n",
      "Int J Oral Sci\n",
      "Primate Biol\n",
      "Ann Glob Health.; 85(1):3\n",
      "eLife.; 9:e55570\n",
      "Yale J Biol Med\n",
      "J Cell Biol\n",
      "Protein Eng Des Sel\n",
      "J Public Health (Oxf)\n",
      "Yonsei Med J\n",
      "Open Biochem J\n",
      "Mcgill J Med\n",
      "Med Law Rev\n",
      "J Mol Genet Med\n",
      "Open Med\n",
      "Mol Syst Biol\n",
      "Curr Chem Genomics\n",
      "Open AIDS J\n",
      "Epidemiol Rev\n",
      "Mol Biol Cell\n",
      "Ann Oncol\n",
      "Clinics (Sao Paulo)\n",
      "Open Cardiovasc Med J\n",
      "Bone Marrow Transplant\n",
      "Int J Med Sci\n",
      "Healthc Inform Res\n",
      "Glycobiology\n",
      "Korean J Lab Med\n",
      "Open Respir Med J\n",
      "Hum Reprod\n",
      "F1000 Biol Rep\n",
      "Korean J Pediatr\n",
      "Ann Saudi Med\n",
      "GMS Curr Top Otorhinolaryngol Head Neck Surg\n",
      "Asia Pac Allergy\n",
      "Expert Rev Mol Med\n",
      "J Toxicol Pathol\n",
      "Lab Anim Res\n",
      "Ann Lab Med\n",
      "Arch Med Sci\n",
      "Immune Netw\n",
      "Am J Epidemiol\n",
      "Allergy Asthma Immunol Res\n",
      "J Geriatr Cardiol\n",
      "Pathobiol Aging Age Relat Dis\n",
      "Saf Health Work\n",
      "Mol Ther Nucleic Acids\n",
      "Bacteriophage\n",
      "Tuberc Respir Dis (Seoul)\n",
      "JRSM Short Rep\n",
      "RNA Biol\n",
      "Ups J Med Sci\n",
      "Oncoimmunology\n",
      "J Radiat Res\n",
      "Eur Respir J\n",
      "J Vet Sci\n",
      "Clin Exp Vaccine Res\n",
      "Expert Opin Drug Deliv\n",
      "Osong Public Health Res Perspect\n",
      "CBE Life Sci Educ\n",
      "Braz J Microbiol\n",
      "Drug Des Devel Ther\n",
      "Tissue Barriers\n",
      "Acta Inform Med\n",
      "F1000Prime Rep\n",
      "Acta Pharmacol Sin\n",
      "AIDS\n",
      "Clin Interv Aging\n",
      "Braz J Med Biol Res\n",
      "Mol Endocrinol\n",
      "J Microbiol Biol Educ\n",
      "Arch Dis Child\n",
      "Int J Nanomedicine\n",
      "MAbs\n",
      "J Vet Med Sci\n",
      "Leukemia\n",
      "Pharmgenomics Pers Med\n",
      "Curr Opin HIV AIDS\n",
      "Mol Cells\n",
      "Stem Cell Reports\n",
      "Online J Public Health Inform\n",
      "J Blood Med\n",
      "J Bras Pneumol\n",
      "Curr Opin Infect Dis\n",
      "Rev Inst Med Trop Sao Paulo\n",
      "Korean Circ J\n",
      "World J Emerg Med\n",
      "Clin Nutr Res\n",
      "Pharm Pract (Granada)\n",
      "J Public Health Res\n",
      "ISBT Sci Ser\n",
      "Int J Gen Med\n",
      "Int J Chron Obstruct Pulmon Dis\n",
      "Shanghai Arch Psychiatry\n",
      "Cancer Res Treat\n",
      "Neuropsychiatr Dis Treat\n",
      "Infect Drug Resist\n",
      "FEMS Microbiol Rev\n",
      "Heart Lung Vessel\n",
      "Hepatology\n",
      "J Pathol\n",
      "Photochem Photobiol\n",
      "Ann Med Surg (Lond)\n",
      "Hepat Mon\n",
      "Angew Chem Int Ed Engl\n",
      "J Vaccines Immun\n",
      "Adv Appl Bioinform Chem\n",
      "Korean J Intern Med\n",
      "J Adv Res\n",
      "Med J Islam Repub Iran\n",
      "Cytokine Growth Factor Rev\n",
      "Nurs Midwifery Stud\n",
      "Diabetes Spectr\n",
      "Chem Biol\n",
      "Med Mycol Case Rep\n",
      "Trop Doct\n",
      "Saudi Med J\n",
      "Mol Ther Methods Clin Dev\n",
      "Rev Saude Publica\n",
      "Clin Kidney J\n",
      "Cent Eur J Immunol\n",
      "J Menopausal Med\n",
      "Cochrane Database Syst Rev\n",
      "Ann Neurosci\n",
      "J Periodontal Implant Sci\n",
      "Breathe (Sheff)\n",
      "Acta Otolaryngol\n",
      "J Exerc Rehabil\n",
      "Gene Ther\n",
      "Vet Rec Open\n",
      "Acta Pharm Sin B\n",
      "Int Med Case Rep J\n",
      "Pediatr Gastroenterol Hepatol Nutr\n",
      "Clin Endosc\n",
      "Emerg (Tehran)\n",
      "Cell Cycle\n",
      "Can J Respir Ther\n",
      "EBioMedicine\n",
      "Public Health Ethics\n",
      "Sci Adv\n",
      "Immunol Rev\n",
      "IDCases\n",
      "Expert Rev Anti Infect Ther\n",
      "J Occup Environ Med\n",
      "Appl Transl Genom\n",
      "Mucosal Immunol\n",
      "Clin Epidemiol\n",
      "Cell Mol Immunol\n",
      "J Prev Med Hyg\n",
      "J Interferon Cytokine Res\n",
      "Genomics Inform\n",
      "J Virus Erad.; 2(1):1-4\n",
      "J Stroke\n",
      "Glia\n",
      "Mol Ther Oncolytics\n",
      "Zdr Varst\n",
      "Arch Pharm (Weinheim)\n",
      "SAGE Open Med Case Rep\n",
      "J Vet Intern Med\n",
      "Risk Manag Healthc Policy\n",
      "Turk J Emerg Med\n",
      "Saudi J Biol Sci\n",
      "Korean J Anesthesiol\n",
      "Dev Reprod\n",
      "Acta Orthop\n",
      "Meta Gene\n",
      "Pediatr Pulmonol\n",
      "Pediatr Infect Dis J\n",
      "Int J MCH AIDS\n",
      "Korean J Med Educ\n",
      "Perspect Public Health\n",
      "Patient Prefer Adherence\n",
      "EJIFCC\n",
      "Shock\n",
      "Curr Top Med Chem\n",
      "ERJ Open Res\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Healthc Policy\n",
      "Ecol Lett\n",
      "J Community Hosp Intern Med Perspect\n",
      "Cell Host Microbe\n",
      "J Pharmacopuncture\n",
      "Iran J Basic Med Sci\n",
      "J Ginseng Res\n",
      "Clin Exp Emerg Med\n",
      "J Virus Erad.; 2(Suppl 1):21-52\n",
      "Int J Cancer\n",
      "Cell Biochem Funct\n",
      "J Clin Transl Hepatol\n",
      "World J Gastroenterol\n",
      "World J Clin Pediatr\n",
      "Antiviral Res\n",
      "Cold Spring Harb Mol Case Stud\n",
      "Asia Pac J Oncol Nurs\n",
      "J Autoimmun\n",
      "Indian Heart J\n",
      "Mutagenesis\n",
      "Ann Rehabil Med\n",
      "Iran Red Crescent Med J\n",
      "World J Crit Care Med\n",
      "Korean J Thorac Cardiovasc Surg\n",
      "Intrinsically Disord Proteins\n",
      "Disaster Health\n",
      "Anatol J Cardiol\n",
      "Intern Med\n",
      "Curr Drug Targets\n",
      "J Public Health Africa\n",
      "J Asthma Allergy\n",
      "World J Clin Cases\n",
      "JFMS Open Rep\n",
      "Int J Surg Case Rep\n",
      "Reumatologia\n",
      "Local Reg Anesth\n",
      "J Cell Death\n",
      "Curr HIV Res\n",
      "Comb Chem High Throughput Screen\n",
      "Mult Scler J Exp Transl Clin\n",
      "Respir Med Case Rep\n",
      "Autops Case Rep\n",
      "Saudi Pharm J\n",
      "Haematologica\n",
      "Health Hum Rights\n",
      "Glob Pediatr Health\n",
      "Kidney Res Clin Pract\n",
      "Clin Case Rep\n",
      "Viral Immunol\n",
      "Curr Ther Res Clin Exp\n",
      "Acad Pathol\n",
      "J Clin Neurol\n",
      "Ann Neurol\n",
      "J Public Health Manag Pract\n",
      "Case Rep Nephrol Dial\n",
      "Adv Dis Control Prev\n",
      "Med Princ Pract\n",
      "Toxicol Rep\n",
      "Eur J Microbiol Immunol (Bp)\n",
      "Prion\n",
      "Mol Metab\n",
      "Chronic Dis Transl Med\n",
      "Nanotheranostics\n",
      "J Neurosci Res\n",
      "Int J Health Sci (Qassim)\n",
      "Med Sci Monit\n",
      "Korean J Fam Med\n",
      "Ind Health\n",
      "Hepatol Commun\n",
      "J Gen Fam Med\n",
      "Drugs\n",
      "J Med Ethics\n",
      "Integr Cancer Ther\n",
      "J Pharm Anal\n",
      "J Crit Care Med (Targu Mures)\n",
      "Korean J Parasitol\n",
      "Future Virol\n",
      "Appl Health Econ Health Policy\n",
      "Inquiry\n",
      "Microbiol Insights\n",
      "Lancet Respir Med\n",
      "J Struct Biol\n",
      "J Feline Med Surg\n",
      "Open Med (Wars)\n",
      "Synth Syst Biotechnol\n",
      "J Orthop Translat\n",
      "Crit Care Med\n",
      "Psychiatry Investig\n",
      "J Med Genet\n",
      "Oncogene\n",
      "Korean J Ophthalmol\n",
      "Res Rep Trop Med\n",
      "Ecol Modell\n",
      "Korean J Food Sci Anim Resour\n",
      "JACC Basic Transl Sci\n",
      "Transplant Direct\n",
      "Noncoding RNA Res\n",
      "Anim Nutr\n",
      "Oncol Rep\n",
      "Yearb Med Inform\n",
      "Cancer Manag Res\n",
      "BMB Rep\n",
      "Biomed J\n",
      "Acta Otorhinolaryngol Ital\n",
      "Int J Vet Sci Med\n",
      "Rev Esp Quimioter\n",
      "J Alzheimers Dis Rep.; 1(1):125-157\n",
      "Acute Med Surg\n",
      "J Tissue Eng\n",
      "Biologics\n",
      "Future Microbiol\n",
      "Saudi J Med Med Sci\n",
      "Open Vet J\n",
      "Chemistry\n",
      "Afr J Emerg Med\n",
      "AIDS Res Hum Retroviruses\n",
      "World J Otorhinolaryngol Head Neck Surg\n",
      "J Vet Res\n",
      "Antib Ther\n",
      "Allergy Rhinol (Providence)\n",
      "Oncol Lett\n",
      "Eur J Case Rep Intern Med\n",
      "MedEdPORTAL.; 12:10433\n",
      "J Insect Sci\n",
      "Biotechnol Rep (Amst)\n",
      "Vector Borne Zoonotic Dis\n",
      "Brain\n",
      "J Dent Sci\n",
      "Clin Exp Immunol\n",
      "J Audiol Otol\n",
      "Ann Clin Transl Neurol\n",
      "Redox Biol\n",
      "Ulster Med J\n",
      "Acta Biomed\n",
      "Oman Med J\n",
      "Adv J Emerg Med\n",
      "Microbiol Immunol\n",
      "Int J Nurs Sci\n",
      "HeartRhythm Case Rep\n",
      "Plast Reconstr Surg Glob Open\n",
      "J Antimicrob Chemother\n",
      "Evolution\n",
      "Vaccine X\n",
      "Epidemiology\n",
      "Technol Cancer Res Treat\n",
      "J Taibah Univ Med Sci\n",
      "J Adv Vet Anim Res\n",
      "iScience\n",
      "BMJ Open Qual\n",
      "Transfusion\n",
      "Int Psychiatry\n",
      "Ther Adv Chronic Dis\n",
      "Clin Pract\n",
      "FEBS Lett\n",
      "Bioethics\n",
      "Pharmacol Rev\n",
      "Korean J Crit Care Med\n",
      "Plant Pathol J\n",
      "Nanotechnol Sci Appl\n",
      "Br J Sports Med\n",
      "Korean J Neurotrauma\n",
      "Acute Crit Care\n",
      "J Appl Microbiol\n",
      "Ann N Y Acad Sci\n",
      "J Small Anim Pract\n",
      "Cardiovasc Hematol Agents Med Chem\n",
      "SSM Popul Health\n",
      "Ci Ji Yi Xue Za Zhi\n",
      "CNS Oncol.; 8(4):CNS45\n",
      "Cancer Sci\n",
      "Biomed Hub\n",
      "Pediatr Qual Saf\n",
      "Onco Targets Ther\n",
      "Aten Primaria\n",
      "JHEP Rep\n",
      "Int J Oncol\n",
      "Clin Appl Thromb Hemost\n",
      "Exp Dermatol\n",
      "J Sport Health Sci\n",
      "Pediatr Rep\n",
      "Interv Med Appl Sci.; 11(1):17-20\n",
      "Fam Med Community Health\n",
      "BMJ Open Respir Res\n",
      "Prev Med Rep\n",
      "Prev Chronic Dis\n",
      "J Biomed Biotechnol\n",
      "J Gen Intern Med\n",
      "Insect Mol Biol\n",
      "Eur J Pediatr\n",
      "Br J Cancer\n",
      "Nat Struct Mol Biol\n",
      "Nat Methods\n",
      "J Biol\n",
      "Int J Behav Med\n",
      "Genet Sel Evol\n",
      "J Neurol\n",
      "Clin Ophthalmol\n",
      "Lab Invest\n",
      "Eur Biophys J\n",
      "J Biomol NMR\n",
      "Immunol Cell Biol\n",
      "J Int AIDS Soc\n",
      "Nat Biotechnol\n",
      "J Math Biol\n",
      "Immunogenetics\n",
      "Biochemistry\n",
      "Food Environ Virol\n",
      "Amino Acids\n",
      "Eur J Dent\n",
      "Nat Immunol\n",
      "Hum Genomics Proteomics\n",
      "J Multidiscip Healthc\n",
      "Protoplasma\n",
      "Pharm Res\n",
      "Eur J Trauma Emerg Surg\n",
      "Drug Target Insights\n",
      "Syst Synth Biol\n",
      "Age (Dordr)\n",
      "Hematol Rev\n",
      "J Mol Model\n",
      "Insights Imaging\n",
      "Anal Bioanal Chem\n",
      "Eur Spine J\n",
      "Biol Trace Elem Res\n",
      "Inflamm Res\n",
      "Naunyn Schmiedebergs Arch Pharmacol\n",
      "Sci Eng Ethics\n",
      "Hum Genomics\n",
      "Drug Deliv Transl Res\n",
      "Bioinformation\n",
      "Pediatr Res\n",
      "Appl Clin Genet\n",
      "J Chem Inf Model\n",
      "Cell Biochem Biophys\n",
      "BioDrugs\n",
      "Biol Invasions\n",
      "Microb Biotechnol\n",
      "Toxicol Res\n",
      "Infect Dis Rep\n",
      "Glob Adv Health Med\n",
      "J Med Chem\n",
      "J Immunoassay Immunochem\n",
      "Eur J Histochem\n",
      "Mol Pharm\n",
      "Sports Health\n",
      "Ann Hematol\n",
      "Chem Rev\n",
      "Clin J Gastroenterol\n",
      "Anal Chem\n",
      "Biomed Eng Comput Biol\n",
      "J Phys Chem B\n",
      "Mikrochim Acta\n",
      "Clin Pharmacol Ther\n",
      "ACS Chem Biol\n",
      "Glob Public Health\n",
      "Environ Health Insights\n",
      "Int Arch Otorhinolaryngol\n",
      "Genes Immun\n",
      "Med Microbiol Immunol\n",
      "J Neuropathol Exp Neurol\n",
      "Auto Immun Highlights\n",
      "J Stat Phys\n",
      "Ophthalmol Ther\n",
      "ACS Synth Biol\n",
      "Dig Dis Sci\n",
      "Infect Dis Ther\n",
      "Nat Chem\n",
      "ACS Appl Mater Interfaces\n",
      "Plasmonics\n",
      "Environ Sci Technol Lett\n",
      "J Occup Environ Hyg\n",
      "Cancer Inform\n",
      "Qatar Med J\n",
      "JBI Database System Rev Implement Rep\n",
      "Avicenna J Phytomed\n",
      "J Technol Transf\n",
      "Crit Rev Microbiol\n",
      "Bone Joint Res\n",
      "Biochem Insights\n",
      "Diabetes\n",
      "J Reprod Infertil\n",
      "Afr J Tradit Complement Altern Med\n",
      "ACS Cent Sci\n",
      "AMIA Jt Summits Transl Sci Proc\n",
      "J Proteome Res\n",
      "Afr J Infect Dis\n",
      "Proc Jpn Acad Ser B Phys Biol Sci\n",
      "Ethiop J Health Sci\n",
      "Nat Nanotechnol\n",
      "ACS Omega\n",
      "Drugs Context\n",
      "Open Access Maced J Med Sci\n",
      "Cell Mol Biol Lett\n",
      "Nat Chem Biol\n",
      "Mol Immunol\n",
      "Asian Pac J Cancer Prev\n",
      "Afr Health Sci\n",
      "Res Vet Sci\n",
      "J Lab Physicians\n",
      "Am J Physiol Lung Cell Mol Physiol\n",
      "J Family Community Med\n",
      "Curr Opin Virol\n",
      "Dis Model Mech\n",
      "JMIR Med Inform\n",
      "Ann Thorac Med\n",
      "Indian J Community Med\n",
      "J Family Med Prim Care\n",
      "J Educ Health Promot\n",
      "Indian J Crit Care Med\n",
      "JMIR Res Protoc\n",
      "J Theor Biol\n",
      "Indian J Med Res\n",
      "J Glob Infect Dis\n",
      "J Pharm Bioallied Sci\n",
      "Urol Ann\n",
      "J Emerg Trauma Shock\n",
      "Curr Opin Chem Biol\n",
      "J Postgrad Med\n",
      "Adv Biomed Res\n",
      "J Int Soc Prev Community Dent\n",
      "eLife.; 7:e37324\n",
      "JMIR Med Educ\n",
      "FASEB J\n",
      "Indian J Dermatol\n",
      "Pharmacogn Mag\n",
      "Ann Indian Acad Neurol\n",
      "Indian J Radiol Imaging\n",
      "Res Pharm Sci\n",
      "Int J Prev Med\n",
      "Indian Dermatol Online J\n",
      "J Res Med Sci\n",
      "Indian J Ophthalmol\n",
      "PLoS Curr\n",
      "Am J Physiol Gastrointest Liver Physiol\n",
      "Indian J Anaesth\n",
      "J Oral Maxillofac Pathol\n",
      "Trends Ecol Evol\n",
      "J Pediatr Neurosci\n",
      "J Tradit Complement Med\n",
      "Int J Crit Illn Inj Sci\n",
      "Neurobiol Aging\n",
      "Ann Med Health Sci Res\n",
      "Medicine (Abingdon)\n",
      "Pharmacogn Rev\n",
      "N Am J Med Sci\n",
      "Prog Lipid Res\n",
      "J Public Health Policy\n",
      "J Nat Sci Biol Med\n",
      "J Pharmacol Pharmacother\n",
      "Iran J Nurs Midwifery Res\n",
      "J Adv Pharm Technol Res\n",
      "Niger Med J\n",
      "Lung India\n",
      "Asian J Transfus Sci\n",
      "J Med Signals Sens\n",
      "J Virol Methods\n",
      "Int Immunopharmacol\n",
      "Immunol Lett\n",
      "Int J Ayurveda Res\n",
      "Pharmacognosy Res\n",
      "Soc Sci Med\n",
      "J Leukoc Biol\n",
      "Indian J Pharm Sci\n",
      "Mens Sana Monogr\n"
     ]
    }
   ],
   "source": [
    "journals = df_meta['journal'].unique()\n",
    "for journal in journals:\n",
    "    print(journal)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c630ebcdf30652f0422c3ec12a00b50241dc9bd9</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) as a SA...</td>\n",
       "      <td>10.1007/s00134-020-05985-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125455.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>2.002765e+09</td>\n",
       "      <td>#3252</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>53eccda7977a31e3d0f565c884da036b1e85438e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comparative genetic analysis of the novel coro...</td>\n",
       "      <td>10.1038/s41421-020-0147-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...</td>\n",
       "      <td>Cell Discovery</td>\n",
       "      <td>3.003431e+09</td>\n",
       "      <td>#1861</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>210a892deb1c61577f6fba58505fd65356ce6636</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Incubation Period and Other Epidemiological Ch...</td>\n",
       "      <td>10.3390/jcm9020538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The geographic spread of 2019 novel coronaviru...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...</td>\n",
       "      <td>Journal of Clinical Medicine</td>\n",
       "      <td>3.006065e+09</td>\n",
       "      <td>#1043</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>e3b40cc8e0e137c416b4a2273a4dca94ae8178cc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Characteristics of and Public Health Responses...</td>\n",
       "      <td>10.3390/jcm9020575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32093211.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, cases of unidentified pneumo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Deng, Sheng-Qun; Peng, Hong-Juan</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>1.776631e+08</td>\n",
       "      <td>#1999</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92c2c9839304b4f2bc1276d41b1aa885d8b364fd</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Imaging changes in severe COVID-19 pneumonia</td>\n",
       "      <td>10.1007/s00134-020-05976-w</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125453.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Wei</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>3.006643e+09</td>\n",
       "      <td>#3242</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0df0d5270a9399cf4e23c0cdd877a80616a9725e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>An updated estimation of the risk of transmiss...</td>\n",
       "      <td>10.1016/j.idm.2020.02.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The basic reproduction number of an infectious...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#729</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>f24242580be243d5fc3f432915d86af6854bb8b7</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Real-time forecasts of the 2019-nCoV epidemic ...</td>\n",
       "      <td>10.1016/j.idm.2020.02.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The initial cluster of severe pneumonia cases ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#865</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>d13a685f861b0f1ba05afa6e005311ad1820fd3a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>RETRACTED: Chinese medical staff request inter...</td>\n",
       "      <td>10.1016/s2214-109x(20)30065-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32105614.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zeng, Yingchun; Zhen, Yan</td>\n",
       "      <td>The Lancet. Global health</td>\n",
       "      <td>2.627046e+09</td>\n",
       "      <td>#5386</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19 outbreak on the Diamond Princess crui...</td>\n",
       "      <td>10.1093/jtm/taaa030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Cruise ships carry a large number of people in...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.</td>\n",
       "      <td>Journal of Travel Medicine</td>\n",
       "      <td>3.006304e+09</td>\n",
       "      <td>#2926</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>e9239100c5493ea914dc23c3d7a262f4326022ac</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Distinct Roles for Sialoside and Protein Recep...</td>\n",
       "      <td>10.1128/mBio.02764-19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Coronaviruses (CoVs) are common human and anim...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Qing, Enya; Hantak, Michael; Perlman, Stanley;...</td>\n",
       "      <td>mBio</td>\n",
       "      <td>3.005811e+09</td>\n",
       "      <td>#2427</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>469ed0f00c09e2637351c9735c306f27acf3aace</td>\n",
       "      <td>CZI</td>\n",
       "      <td>First two months of the 2019 Coronavirus Disea...</td>\n",
       "      <td>10.1186/s41256-020-00137-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Similar to outbreaks of many other infectious ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Xinguang; Yu, Bin</td>\n",
       "      <td>Global Health Research and Policy</td>\n",
       "      <td>3.006646e+09</td>\n",
       "      <td>#5595</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4e550e034ccca6fa2a91e481ddba24db67bc9ae5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness of airport screening at detectin...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.5.2000080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>We simulated 100 2019-nCoV infected travellers...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Quilty, Billy J; Clifford, Sam; group2, CMMID ...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005152e+09</td>\n",
       "      <td>#682</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4bbb0c59babc718f67953fae032dad6ae0d7aeb1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Genome Detective Coronavirus Typing Tool for r...</td>\n",
       "      <td>10.1093/bioinformatics/btaa145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32108862.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>SUMMARY: Genome Detective is a web-based, user...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>3.003549e+09</td>\n",
       "      <td>#2776</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>c821803c55c6aad89b6d0c1d3ba252051e464017</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Case of the Index Patient Who Caused Tertiary ...</td>\n",
       "      <td>10.3346/jkms.2020.35.e79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Since mid-December of 2019, coronavirus diseas...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...</td>\n",
       "      <td>Journal of Korean Medical Science</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#3770</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>c3bee2a4caca614b34f92c17b643b854dcdab28d</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Emergence of Novel Coronavirus 2019-nCoV: Need...</td>\n",
       "      <td>10.3390/pathogens9020148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Novel Coronavirus (2019-nCoV) is an emerging p...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...</td>\n",
       "      <td>Pathogens</td>\n",
       "      <td>3.003886e+09</td>\n",
       "      <td>#1502</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>715842fa536064980818ad7e31ce511272a4b6bc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The coronavirus 2019-nCoV epidemic: Is hindsig...</td>\n",
       "      <td>10.1016/j.eclinm.2020.100289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...</td>\n",
       "      <td>EClinicalMedicine</td>\n",
       "      <td>3.004913e+09</td>\n",
       "      <td>#3333</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Nonstructural proteins NS7b and NS8 are likely...</td>\n",
       "      <td>10.1016/j.meegid.2020.104272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The seventh novel human infecting Betacoronavi...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro</td>\n",
       "      <td>Infection, Genetics and Evolution</td>\n",
       "      <td>2.286137e+09</td>\n",
       "      <td>#4581</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>06c89f69aa7b5f9648d2c1543b8246fe9c3610cf</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Pathogenicity and Transmissibility of 2019-nCo...</td>\n",
       "      <td>10.1016/j.micinf.2020.01.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>A zoonotic coronavirus, labeled as 2019-nCoV b...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Jieliang</td>\n",
       "      <td>Microbes and Infection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#236</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>acb678bdd7634055de18d0b89bb6a4890e6a0306</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Remdesivir and chloroquine effectively inhibit...</td>\n",
       "      <td>10.1038/s41422-020-0282-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...</td>\n",
       "      <td>Cell Research</td>\n",
       "      <td>3.005213e+09</td>\n",
       "      <td>#205</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>47772f3e98d8c61bb9782531a0338ba85f27bf3f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential for global spread of a novel coronav...</td>\n",
       "      <td>10.1093/jtm/taaa011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...</td>\n",
       "      <td>Journal of travel medicine</td>\n",
       "      <td>3.004048e+09</td>\n",
       "      <td>#102</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Severe acute respiratory symptoms and suspecte...</td>\n",
       "      <td>10.12809/hkmj208375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Hon, K. L.; Leung, K. K. Y.</td>\n",
       "      <td>Hong Kong Medical Journal</td>\n",
       "      <td>2.029408e+09</td>\n",
       "      <td>#3395</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4ff89a71126d2932544a8337ba28787fde5f02a8</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Statistics-Based Predictions of Coronavirus Ep...</td>\n",
       "      <td>10.20535/ibb.2020.4.1.195074</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Information about the open-access article 'Sta...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Nesteruk, Igor</td>\n",
       "      <td>Innovative Biosystems and Bioengineering</td>\n",
       "      <td>3.006305e+09</td>\n",
       "      <td>#1218</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ac37ba61c91bb6939a507dbf4efe2119ddafeb9c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Early transmission patterns of coronavirus dis...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.8.2000097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#2582</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0</td>\n",
       "      <td>CZI</td>\n",
       "      <td>HRCT Imaging Features in Representative Import...</td>\n",
       "      <td>10.1093/pcmedi/pbaa004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>With the spread of novel coronavirus (2019-nCo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...</td>\n",
       "      <td>Precision Clinical Medicine</td>\n",
       "      <td>3.006468e+09</td>\n",
       "      <td>#781</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>c9fee561c2a3834645dbb61dc4ae6448051da492</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comprehensive Genomic Characterization Analysi...</td>\n",
       "      <td>10.3389/fmicb.2019.03036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Porcine delta coronavirus (PDCoV) is a novel e...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...</td>\n",
       "      <td>Frontiers in Microbiology</td>\n",
       "      <td>3.003968e+09</td>\n",
       "      <td>#5462</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>655537fc8cc52bccf43cf7189ab060d3097caa7a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential Factors Influencing Repeated SARS Ou...</td>\n",
       "      <td>10.3390/ijerph17051633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Within last 17 years two widespread epidemics ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....</td>\n",
       "      <td>International Journal of Environmental Researc...</td>\n",
       "      <td>2.615949e+09</td>\n",
       "      <td>#3296</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>bb7cd48a85b72f8d73e57d655c6d8a034fce3778</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Puzzle of highly pathogenic human coronaviruse...</td>\n",
       "      <td>10.1007/s13238-020-00693-y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32088858.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Li, Jing; Liu, Wenjun</td>\n",
       "      <td>Protein Cell</td>\n",
       "      <td>2.766462e+09</td>\n",
       "      <td>#1815</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3c42431c5b95e707486ce7441ac139071e07b706</td>\n",
       "      <td>CZI</td>\n",
       "      <td>RETRACTION: Retraction-Chinese medical staff r...</td>\n",
       "      <td>10.1016/s2214-109x(20)30076-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32113504.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>The Editors Of The Lancet Global, Health</td>\n",
       "      <td>The Lancet. Global health</td>\n",
       "      <td>2.794879e+09</td>\n",
       "      <td>#5486</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>6a93283b499ae5bc6aaf29f14e701dc8f25138ea</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Critical care response to a hospital outbreak ...</td>\n",
       "      <td>10.1186/s13054-020-2786-x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liu, Yong; Li, Jinxiu; Feng, Yongwen</td>\n",
       "      <td>Critical Care</td>\n",
       "      <td>2.537201e+09</td>\n",
       "      <td>#1225</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>6fc52ed878c271020a2a375bb6e4b12943a7666c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness for the Response to COVID-19: Th...</td>\n",
       "      <td>10.24171/j.phrp.2020.11.1.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>In the current issue of Osong Public Health an...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cho, Hae-Wol</td>\n",
       "      <td>Osong Public Health and Research Perspectives</td>\n",
       "      <td>3.005888e+09</td>\n",
       "      <td>#2662</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1206</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Daily briefing: World’s biggest physics meetin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The March Meeting of the APS was cancelled jus...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Graham, Flora</td>\n",
       "      <td>Nature</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#3172</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1207</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>How Coronaviruses Cause Infection—from Colds t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The novel coronavirus outbreak raises question...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Makin, Simon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#294</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1208</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Advices on the prevention and control of nosoc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The pneumonia caused by the novel coronavirus ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XU, Hongzhen; CHEN, Shuohui; FU, Junfen; SHU, ...</td>\n",
       "      <td>Chinese Journal of Hospital Administration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#2106</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1209</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>‘This beast is moving very fast.’ Will the new...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The silver lining of the epidemic is that scie...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cohen, Kai Kupferschmidt; Jon</td>\n",
       "      <td>Science</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#332</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1210</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo Coronavírus: atendimento a pessoas com su...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Brasil. Ministerio da, Saude</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#1866</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1211</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo coronavirus (2019 nCov) Medidas de preven...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>São Paulo Secretaria da, Saúde</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#922</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1212</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The prevention and control of a new coronaviru...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32057210.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>During a short period of time, the outbreak of...</td>\n",
       "      <td>2020</td>\n",
       "      <td>LI, Zhi yong; MENG, Liu yan</td>\n",
       "      <td>Chinese Journal of Stomatology</td>\n",
       "      <td>2.377782e+09</td>\n",
       "      <td>#1228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19: another infectious disease emerging ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32078596.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Murdoch, David R.; French, Nigel P.</td>\n",
       "      <td>N Z Med J</td>\n",
       "      <td>2.185518e+09</td>\n",
       "      <td>#1541</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Dynamic changes of chest CT imaging in patient...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32096366.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OBJECTIVE: To analyze the dynamic changes of c...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Jincheng; Liu, Jinpeng; Wang, Yuanyuan; ...</td>\n",
       "      <td>Zhejiang Da Xue Xue Bao Yi Xue Ban</td>\n",
       "      <td>2.349103e+09</td>\n",
       "      <td>#1922</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1215</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus COVID-19: an overview for em...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32105049.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prior to the global outbreak of SARS-CoV in 20...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Giwa, A.; Desai, A.</td>\n",
       "      <td>Emergency medicine practice</td>\n",
       "      <td>3.005414e+09</td>\n",
       "      <td>#2815</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Coronavirus 9-nCoV: recomendaciones para aerop...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ante la situación mundial, en relación a 2019-...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Argentina. Ministerio de, Salud</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.604377e+09</td>\n",
       "      <td>#801</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel Coronavirus Disease 2019 (COVID-19): An ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Background: At the beginning of the New Year 2...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Keshavarz, Ahmad Tavakoli; Katayon, Vahdat; Mo...</td>\n",
       "      <td>Iranian South Medical Journal</td>\n",
       "      <td>6.213808e+08</td>\n",
       "      <td>#4536</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Coronavirus just caused the American Physical ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Citing the growing threat of the coronavirus, ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cho, Adrian</td>\n",
       "      <td>Science Magazine</td>\n",
       "      <td>2.595675e+09</td>\n",
       "      <td>#2772</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Mensajero de la Salud: nuevo coronavirus 2019-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Declaratoria de Emergencia en Salud Pública de...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Mexico. Secretaria de la, Salud</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.003591e+09</td>\n",
       "      <td>#1803</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The management of biosafety risk in clinical l...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>During the outbreak of coronavirus disease-19 ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XIAO, Yuling; LU, Xiaojun; KANG, Mei; LI, Dong...</td>\n",
       "      <td>Chinese Journal of Laboratory Medicine</td>\n",
       "      <td>2.194450e+09</td>\n",
       "      <td>#2110</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Lineamiento estandarizado para la Vigilancia E...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Este documento describe la situación epidemiol...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Mexico. Secretaria de Salud. Subsecretaria de ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.241123e+09</td>\n",
       "      <td>#1802</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Singapore claims first use of antibody test to...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In what appears to be a first, disease tracker...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Normile, Dennis</td>\n",
       "      <td>Science Magazine</td>\n",
       "      <td>2.855668e+09</td>\n",
       "      <td>#2584</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The network investigation on knowledge, attitu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To analyze the current situation of ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>CHEN, Yan</td>\n",
       "      <td>Chinese Journal of Preventive Medicine</td>\n",
       "      <td>2.363411e+09</td>\n",
       "      <td>#2333</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Investigation on demands for antenatal care se...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To identify problems and demands for...</td>\n",
       "      <td>2020</td>\n",
       "      <td>DU, Li; GU, Yibin; CUI, Mengqing; LI, Wenxian;...</td>\n",
       "      <td>Chinese Journal of Obstetrics and Gynecology</td>\n",
       "      <td>1.891210e+09</td>\n",
       "      <td>#2305</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1225</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Analysis of early chest high resolution CT ima...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To investigate the first chest HRCT ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>LIU, Haifeng; ZHANG, Dongyou; YANG, Yi; LONG, ...</td>\n",
       "      <td>Chinese Journal of Radiology</td>\n",
       "      <td>2.418494e+09</td>\n",
       "      <td>#2187</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1226</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo Coronavirus (nCoV)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Os Coronavírus (CoV) compõem uma grande famíli...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rio de Janeiro . Secretaria de Estado de Saúde...</td>\n",
       "      <td>Nota Técnica-SVS/SES-RJ</td>\n",
       "      <td>2.518955e+09</td>\n",
       "      <td>#216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1227</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Difficulties and strategies of public hospital...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Outbreak of the novel coronavirus pneumonia (N...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XU, Dong; HU, Yu; DING, Ning; XIA, Jiahong; ZH...</td>\n",
       "      <td>Chinese Journal of Hospital Administration</td>\n",
       "      <td>2.348898e+09</td>\n",
       "      <td>#2108</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Get Real</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Since this is going to be a post about the cor...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lowe, Derek</td>\n",
       "      <td>Science</td>\n",
       "      <td>2.343732e+09</td>\n",
       "      <td>#3808</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>How Ophthalmologists Should Understand and Res...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The new coronavirus pneumonia that first appea...</td>\n",
       "      <td>2020</td>\n",
       "      <td>ZHIJIE, Li</td>\n",
       "      <td>Chinese Journal of Experimental Ophthalmology</td>\n",
       "      <td>3.005943e+09</td>\n",
       "      <td>#2051</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1230</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Experts proposal and frequently asked question...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The outbreak of novel coronavirus pneumonia (N...</td>\n",
       "      <td>2020</td>\n",
       "      <td>ZHANG, Lei; CAO, Qing; WANG, Ying; LU, Quan; H...</td>\n",
       "      <td>Chinese Journal of Applied Clinical Pediatrics</td>\n",
       "      <td>2.416639e+09</td>\n",
       "      <td>#2063</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1231</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Surgical treatment strategy for digestive syst...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The outbreak of novel coronavirus pneumonia oc...</td>\n",
       "      <td>2020</td>\n",
       "      <td>MA, Fuhai; HU, Haitao; TIAN, Yantao</td>\n",
       "      <td>Chinese Journal of Oncology</td>\n",
       "      <td>2.366820e+09</td>\n",
       "      <td>#2176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1232</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus pneumonia in the primary gen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>To observe the curative effect of TCM syndrome...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liao, Rongye; Yang, Jie; Cao, Zhi; Wang, Jun</td>\n",
       "      <td>International Journal of Traditional Chinese M...</td>\n",
       "      <td>2.616653e+09</td>\n",
       "      <td>#1962</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>A Novel Coronavirus Outbreak from Wuhan City i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Alavi-Moghaddam, Mostafa</td>\n",
       "      <td>Archives of Academic Emergency Medicine</td>\n",
       "      <td>3.006237e+09</td>\n",
       "      <td>#3509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1234</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>History is repeating itself: Probable zoonotic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rodriguez-Morales, A. J.; Bonilla-Aldana, D. K...</td>\n",
       "      <td>Infezioni in Medicina</td>\n",
       "      <td>3.005681e+09</td>\n",
       "      <td>#926</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1235</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Epidemiological and Clinical Characteristics o...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chen, Nanshan and Zhou, Min and Dong, Xuan and...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.004581e+09</td>\n",
       "      <td>#22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1236 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           sha source_x  \\\n",
       "0     c630ebcdf30652f0422c3ec12a00b50241dc9bd9      CZI   \n",
       "1     53eccda7977a31e3d0f565c884da036b1e85438e      CZI   \n",
       "2     210a892deb1c61577f6fba58505fd65356ce6636      CZI   \n",
       "3     e3b40cc8e0e137c416b4a2273a4dca94ae8178cc      CZI   \n",
       "4     92c2c9839304b4f2bc1276d41b1aa885d8b364fd      CZI   \n",
       "5     0df0d5270a9399cf4e23c0cdd877a80616a9725e      CZI   \n",
       "6     f24242580be243d5fc3f432915d86af6854bb8b7      CZI   \n",
       "7     d13a685f861b0f1ba05afa6e005311ad1820fd3a      CZI   \n",
       "8     e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1      CZI   \n",
       "9     e9239100c5493ea914dc23c3d7a262f4326022ac      CZI   \n",
       "10    469ed0f00c09e2637351c9735c306f27acf3aace      CZI   \n",
       "11    4e550e034ccca6fa2a91e481ddba24db67bc9ae5      CZI   \n",
       "12    4bbb0c59babc718f67953fae032dad6ae0d7aeb1      CZI   \n",
       "13    c821803c55c6aad89b6d0c1d3ba252051e464017      CZI   \n",
       "14    c3bee2a4caca614b34f92c17b643b854dcdab28d      CZI   \n",
       "15    715842fa536064980818ad7e31ce511272a4b6bc      CZI   \n",
       "16    601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f      CZI   \n",
       "17    06c89f69aa7b5f9648d2c1543b8246fe9c3610cf      CZI   \n",
       "18    acb678bdd7634055de18d0b89bb6a4890e6a0306      CZI   \n",
       "19    47772f3e98d8c61bb9782531a0338ba85f27bf3f      CZI   \n",
       "20    544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5      CZI   \n",
       "21    4ff89a71126d2932544a8337ba28787fde5f02a8      CZI   \n",
       "22    ac37ba61c91bb6939a507dbf4efe2119ddafeb9c      CZI   \n",
       "23    6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0      CZI   \n",
       "24    c9fee561c2a3834645dbb61dc4ae6448051da492      CZI   \n",
       "25    655537fc8cc52bccf43cf7189ab060d3097caa7a      CZI   \n",
       "26    bb7cd48a85b72f8d73e57d655c6d8a034fce3778      CZI   \n",
       "27    3c42431c5b95e707486ce7441ac139071e07b706      CZI   \n",
       "28    6a93283b499ae5bc6aaf29f14e701dc8f25138ea      CZI   \n",
       "29    6fc52ed878c271020a2a375bb6e4b12943a7666c      CZI   \n",
       "...                                        ...      ...   \n",
       "1206                                       NaN      CZI   \n",
       "1207                                       NaN      CZI   \n",
       "1208                                       NaN      CZI   \n",
       "1209                                       NaN      CZI   \n",
       "1210                                       NaN      CZI   \n",
       "1211                                       NaN      CZI   \n",
       "1212                                       NaN      CZI   \n",
       "1213                                       NaN      CZI   \n",
       "1214                                       NaN      CZI   \n",
       "1215                                       NaN      CZI   \n",
       "1216                                       NaN      CZI   \n",
       "1217                                       NaN      CZI   \n",
       "1218                                       NaN      CZI   \n",
       "1219                                       NaN      CZI   \n",
       "1220                                       NaN      CZI   \n",
       "1221                                       NaN      CZI   \n",
       "1222                                       NaN      CZI   \n",
       "1223                                       NaN      CZI   \n",
       "1224                                       NaN      CZI   \n",
       "1225                                       NaN      CZI   \n",
       "1226                                       NaN      CZI   \n",
       "1227                                       NaN      CZI   \n",
       "1228                                       NaN      CZI   \n",
       "1229                                       NaN      CZI   \n",
       "1230                                       NaN      CZI   \n",
       "1231                                       NaN      CZI   \n",
       "1232                                       NaN      CZI   \n",
       "1233                                       NaN      CZI   \n",
       "1234                                       NaN      CZI   \n",
       "1235                                       NaN      CZI   \n",
       "\n",
       "                                                  title  \\\n",
       "0     Angiotensin-converting enzyme 2 (ACE2) as a SA...   \n",
       "1     Comparative genetic analysis of the novel coro...   \n",
       "2     Incubation Period and Other Epidemiological Ch...   \n",
       "3     Characteristics of and Public Health Responses...   \n",
       "4          Imaging changes in severe COVID-19 pneumonia   \n",
       "5     An updated estimation of the risk of transmiss...   \n",
       "6     Real-time forecasts of the 2019-nCoV epidemic ...   \n",
       "7     RETRACTED: Chinese medical staff request inter...   \n",
       "8     COVID-19 outbreak on the Diamond Princess crui...   \n",
       "9     Distinct Roles for Sialoside and Protein Recep...   \n",
       "10    First two months of the 2019 Coronavirus Disea...   \n",
       "11    Effectiveness of airport screening at detectin...   \n",
       "12    Genome Detective Coronavirus Typing Tool for r...   \n",
       "13    Case of the Index Patient Who Caused Tertiary ...   \n",
       "14    Emergence of Novel Coronavirus 2019-nCoV: Need...   \n",
       "15    The coronavirus 2019-nCoV epidemic: Is hindsig...   \n",
       "16    Nonstructural proteins NS7b and NS8 are likely...   \n",
       "17    Pathogenicity and Transmissibility of 2019-nCo...   \n",
       "18    Remdesivir and chloroquine effectively inhibit...   \n",
       "19    Potential for global spread of a novel coronav...   \n",
       "20    Severe acute respiratory symptoms and suspecte...   \n",
       "21    Statistics-Based Predictions of Coronavirus Ep...   \n",
       "22    Early transmission patterns of coronavirus dis...   \n",
       "23    HRCT Imaging Features in Representative Import...   \n",
       "24    Comprehensive Genomic Characterization Analysi...   \n",
       "25    Potential Factors Influencing Repeated SARS Ou...   \n",
       "26    Puzzle of highly pathogenic human coronaviruse...   \n",
       "27    RETRACTION: Retraction-Chinese medical staff r...   \n",
       "28    Critical care response to a hospital outbreak ...   \n",
       "29    Effectiveness for the Response to COVID-19: Th...   \n",
       "...                                                 ...   \n",
       "1206  Daily briefing: World’s biggest physics meetin...   \n",
       "1207  How Coronaviruses Cause Infection—from Colds t...   \n",
       "1208  Advices on the prevention and control of nosoc...   \n",
       "1209  ‘This beast is moving very fast.’ Will the new...   \n",
       "1210  Novo Coronavírus: atendimento a pessoas com su...   \n",
       "1211  Novo coronavirus (2019 nCov) Medidas de preven...   \n",
       "1212  The prevention and control of a new coronaviru...   \n",
       "1213  COVID-19: another infectious disease emerging ...   \n",
       "1214  Dynamic changes of chest CT imaging in patient...   \n",
       "1215  Novel coronavirus COVID-19: an overview for em...   \n",
       "1216  Coronavirus 9-nCoV: recomendaciones para aerop...   \n",
       "1217  Novel Coronavirus Disease 2019 (COVID-19): An ...   \n",
       "1218  Coronavirus just caused the American Physical ...   \n",
       "1219  Mensajero de la Salud: nuevo coronavirus 2019-...   \n",
       "1220  The management of biosafety risk in clinical l...   \n",
       "1221  Lineamiento estandarizado para la Vigilancia E...   \n",
       "1222  Singapore claims first use of antibody test to...   \n",
       "1223  The network investigation on knowledge, attitu...   \n",
       "1224  Investigation on demands for antenatal care se...   \n",
       "1225  Analysis of early chest high resolution CT ima...   \n",
       "1226                            Novo Coronavirus (nCoV)   \n",
       "1227  Difficulties and strategies of public hospital...   \n",
       "1228                                           Get Real   \n",
       "1229  How Ophthalmologists Should Understand and Res...   \n",
       "1230  Experts proposal and frequently asked question...   \n",
       "1231  Surgical treatment strategy for digestive syst...   \n",
       "1232  Novel coronavirus pneumonia in the primary gen...   \n",
       "1233  A Novel Coronavirus Outbreak from Wuhan City i...   \n",
       "1234  History is repeating itself: Probable zoonotic...   \n",
       "1235  Epidemiological and Clinical Characteristics o...   \n",
       "\n",
       "                                         doi pmcid   pubmed_id      license  \\\n",
       "0                 10.1007/s00134-020-05985-9   NaN  32125455.0     cc-by-nc   \n",
       "1                  10.1038/s41421-020-0147-1   NaN         NaN        cc-by   \n",
       "2                         10.3390/jcm9020538   NaN         NaN        cc-by   \n",
       "3                         10.3390/jcm9020575   NaN  32093211.0        cc-by   \n",
       "4                 10.1007/s00134-020-05976-w   NaN  32125453.0     cc-by-nc   \n",
       "5                  10.1016/j.idm.2020.02.001   NaN         NaN  cc-by-nc-nd   \n",
       "6                  10.1016/j.idm.2020.02.002   NaN         NaN  cc-by-nc-nd   \n",
       "7              10.1016/s2214-109x(20)30065-6   NaN  32105614.0        cc-by   \n",
       "8                        10.1093/jtm/taaa030   NaN         NaN     cc-by-nc   \n",
       "9                      10.1128/mBio.02764-19   NaN         NaN        cc-by   \n",
       "10                10.1186/s41256-020-00137-4   NaN         NaN        cc-by   \n",
       "11    10.2807/1560-7917.ES.2020.25.5.2000080   NaN         NaN        cc-by   \n",
       "12            10.1093/bioinformatics/btaa145   NaN  32108862.0     cc-by-nc   \n",
       "13                  10.3346/jkms.2020.35.e79   NaN         NaN     cc-by-nc   \n",
       "14                  10.3390/pathogens9020148   NaN         NaN        cc-by   \n",
       "15              10.1016/j.eclinm.2020.100289   NaN         NaN  cc-by-nc-nd   \n",
       "16              10.1016/j.meegid.2020.104272   NaN         NaN  cc-by-nc-nd   \n",
       "17              10.1016/j.micinf.2020.01.004   NaN         NaN  cc-by-nc-nd   \n",
       "18                 10.1038/s41422-020-0282-0   NaN         NaN        cc-by   \n",
       "19                       10.1093/jtm/taaa011   NaN         NaN     cc-by-nc   \n",
       "20                       10.12809/hkmj208375   NaN         NaN  cc-by-nc-nd   \n",
       "21              10.20535/ibb.2020.4.1.195074   NaN         NaN        cc-by   \n",
       "22    10.2807/1560-7917.ES.2020.25.8.2000097   NaN         NaN        cc-by   \n",
       "23                    10.1093/pcmedi/pbaa004   NaN         NaN     cc-by-nc   \n",
       "24                  10.3389/fmicb.2019.03036   NaN         NaN        cc-by   \n",
       "25                    10.3390/ijerph17051633   NaN         NaN        cc-by   \n",
       "26                10.1007/s13238-020-00693-y   NaN  32088858.0        cc-by   \n",
       "27             10.1016/s2214-109x(20)30076-0   NaN  32113504.0        cc-by   \n",
       "28                 10.1186/s13054-020-2786-x   NaN         NaN        cc-by   \n",
       "29              10.24171/j.phrp.2020.11.1.01   NaN         NaN  cc-by-nc-nd   \n",
       "...                                      ...   ...         ...          ...   \n",
       "1206                                     NaN   NaN         NaN          NaN   \n",
       "1207                                     NaN   NaN         NaN          NaN   \n",
       "1208                                     NaN   NaN         NaN          NaN   \n",
       "1209                                     NaN   NaN         NaN          NaN   \n",
       "1210                                     NaN   NaN         NaN          NaN   \n",
       "1211                                     NaN   NaN         NaN          NaN   \n",
       "1212                                     NaN   NaN  32057210.0          NaN   \n",
       "1213                                     NaN   NaN  32078596.0          NaN   \n",
       "1214                                     NaN   NaN  32096366.0          NaN   \n",
       "1215                                     NaN   NaN  32105049.0          NaN   \n",
       "1216                                     NaN   NaN         NaN          NaN   \n",
       "1217                                     NaN   NaN         NaN          NaN   \n",
       "1218                                     NaN   NaN         NaN          NaN   \n",
       "1219                                     NaN   NaN         NaN          NaN   \n",
       "1220                                     NaN   NaN         NaN          NaN   \n",
       "1221                                     NaN   NaN         NaN          NaN   \n",
       "1222                                     NaN   NaN         NaN          NaN   \n",
       "1223                                     NaN   NaN         NaN          NaN   \n",
       "1224                                     NaN   NaN         NaN          NaN   \n",
       "1225                                     NaN   NaN         NaN          NaN   \n",
       "1226                                     NaN   NaN         NaN          NaN   \n",
       "1227                                     NaN   NaN         NaN          NaN   \n",
       "1228                                     NaN   NaN         NaN          NaN   \n",
       "1229                                     NaN   NaN         NaN          NaN   \n",
       "1230                                     NaN   NaN         NaN          NaN   \n",
       "1231                                     NaN   NaN         NaN          NaN   \n",
       "1232                                     NaN   NaN         NaN          NaN   \n",
       "1233                                     NaN   NaN         NaN          NaN   \n",
       "1234                                     NaN   NaN         NaN          NaN   \n",
       "1235                                     NaN   NaN         NaN          NaN   \n",
       "\n",
       "                                               abstract publish_time  \\\n",
       "0                                                   NaN         2020   \n",
       "1                                                   NaN         2020   \n",
       "2     The geographic spread of 2019 novel coronaviru...         2020   \n",
       "3     In December 2019, cases of unidentified pneumo...         2020   \n",
       "4                                                   NaN         2020   \n",
       "5     The basic reproduction number of an infectious...         2020   \n",
       "6     The initial cluster of severe pneumonia cases ...         2020   \n",
       "7                                                   NaN         2020   \n",
       "8     Cruise ships carry a large number of people in...         2020   \n",
       "9     Coronaviruses (CoVs) are common human and anim...         2020   \n",
       "10    Similar to outbreaks of many other infectious ...         2020   \n",
       "11    We simulated 100 2019-nCoV infected travellers...         2020   \n",
       "12    SUMMARY: Genome Detective is a web-based, user...         2020   \n",
       "13    Since mid-December of 2019, coronavirus diseas...         2020   \n",
       "14    Novel Coronavirus (2019-nCoV) is an emerging p...         2020   \n",
       "15                                                  NaN         2020   \n",
       "16    The seventh novel human infecting Betacoronavi...         2020   \n",
       "17    A zoonotic coronavirus, labeled as 2019-nCoV b...         2020   \n",
       "18                                                  NaN         2020   \n",
       "19                                                  NaN         2020   \n",
       "20                                                  NaN         2020   \n",
       "21    Information about the open-access article 'Sta...         2020   \n",
       "22                                                  NaN         2020   \n",
       "23    With the spread of novel coronavirus (2019-nCo...         2020   \n",
       "24    Porcine delta coronavirus (PDCoV) is a novel e...         2020   \n",
       "25    Within last 17 years two widespread epidemics ...         2020   \n",
       "26                                                  NaN         2020   \n",
       "27                                                  NaN         2020   \n",
       "28                                                  NaN         2020   \n",
       "29    In the current issue of Osong Public Health an...         2020   \n",
       "...                                                 ...          ...   \n",
       "1206  The March Meeting of the APS was cancelled jus...         2020   \n",
       "1207  The novel coronavirus outbreak raises question...         2020   \n",
       "1208  The pneumonia caused by the novel coronavirus ...         2020   \n",
       "1209  The silver lining of the epidemic is that scie...         2020   \n",
       "1210                                                NaN         2020   \n",
       "1211                                                NaN         2020   \n",
       "1212  During a short period of time, the outbreak of...         2020   \n",
       "1213                                                NaN         2020   \n",
       "1214  OBJECTIVE: To analyze the dynamic changes of c...         2020   \n",
       "1215  Prior to the global outbreak of SARS-CoV in 20...         2020   \n",
       "1216  Ante la situación mundial, en relación a 2019-...         2020   \n",
       "1217  Background: At the beginning of the New Year 2...         2020   \n",
       "1218  Citing the growing threat of the coronavirus, ...         2020   \n",
       "1219  Declaratoria de Emergencia en Salud Pública de...         2020   \n",
       "1220  During the outbreak of coronavirus disease-19 ...         2020   \n",
       "1221  Este documento describe la situación epidemiol...         2020   \n",
       "1222  In what appears to be a first, disease tracker...         2020   \n",
       "1223  Objective To analyze the current situation of ...         2020   \n",
       "1224  Objective To identify problems and demands for...         2020   \n",
       "1225  Objective To investigate the first chest HRCT ...         2020   \n",
       "1226  Os Coronavírus (CoV) compõem uma grande famíli...         2020   \n",
       "1227  Outbreak of the novel coronavirus pneumonia (N...         2020   \n",
       "1228  Since this is going to be a post about the cor...         2020   \n",
       "1229  The new coronavirus pneumonia that first appea...         2020   \n",
       "1230  The outbreak of novel coronavirus pneumonia (N...         2020   \n",
       "1231  The outbreak of novel coronavirus pneumonia oc...         2020   \n",
       "1232  To observe the curative effect of TCM syndrome...         2020   \n",
       "1233                                                NaN         2020   \n",
       "1234                                                NaN         2020   \n",
       "1235                                                NaN          NaN   \n",
       "\n",
       "                                                authors  \\\n",
       "0     Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...   \n",
       "1     Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...   \n",
       "2     Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...   \n",
       "3                      Deng, Sheng-Qun; Peng, Hong-Juan   \n",
       "4                                            Zhang, Wei   \n",
       "5     Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...   \n",
       "6     Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...   \n",
       "7                             Zeng, Yingchun; Zhen, Yan   \n",
       "8             Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.   \n",
       "9     Qing, Enya; Hantak, Michael; Perlman, Stanley;...   \n",
       "10                              Chen, Xinguang; Yu, Bin   \n",
       "11    Quilty, Billy J; Clifford, Sam; group2, CMMID ...   \n",
       "12    Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...   \n",
       "13    Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...   \n",
       "14    Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...   \n",
       "15    Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...   \n",
       "16      Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro   \n",
       "17                                       Chen, Jieliang   \n",
       "18    Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...   \n",
       "19    Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...   \n",
       "20                          Hon, K. L.; Leung, K. K. Y.   \n",
       "21                                       Nesteruk, Igor   \n",
       "22    Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...   \n",
       "23    Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...   \n",
       "24    Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...   \n",
       "25    Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....   \n",
       "26                                Li, Jing; Liu, Wenjun   \n",
       "27             The Editors Of The Lancet Global, Health   \n",
       "28                 Liu, Yong; Li, Jinxiu; Feng, Yongwen   \n",
       "29                                         Cho, Hae-Wol   \n",
       "...                                                 ...   \n",
       "1206                                      Graham, Flora   \n",
       "1207                                       Makin, Simon   \n",
       "1208  XU, Hongzhen; CHEN, Shuohui; FU, Junfen; SHU, ...   \n",
       "1209                      Cohen, Kai Kupferschmidt; Jon   \n",
       "1210                       Brasil. Ministerio da, Saude   \n",
       "1211                     São Paulo Secretaria da, Saúde   \n",
       "1212                        LI, Zhi yong; MENG, Liu yan   \n",
       "1213                Murdoch, David R.; French, Nigel P.   \n",
       "1214  Wang, Jincheng; Liu, Jinpeng; Wang, Yuanyuan; ...   \n",
       "1215                                Giwa, A.; Desai, A.   \n",
       "1216                    Argentina. Ministerio de, Salud   \n",
       "1217  Keshavarz, Ahmad Tavakoli; Katayon, Vahdat; Mo...   \n",
       "1218                                        Cho, Adrian   \n",
       "1219                    Mexico. Secretaria de la, Salud   \n",
       "1220  XIAO, Yuling; LU, Xiaojun; KANG, Mei; LI, Dong...   \n",
       "1221  Mexico. Secretaria de Salud. Subsecretaria de ...   \n",
       "1222                                    Normile, Dennis   \n",
       "1223                                          CHEN, Yan   \n",
       "1224  DU, Li; GU, Yibin; CUI, Mengqing; LI, Wenxian;...   \n",
       "1225  LIU, Haifeng; ZHANG, Dongyou; YANG, Yi; LONG, ...   \n",
       "1226  Rio de Janeiro . Secretaria de Estado de Saúde...   \n",
       "1227  XU, Dong; HU, Yu; DING, Ning; XIA, Jiahong; ZH...   \n",
       "1228                                        Lowe, Derek   \n",
       "1229                                         ZHIJIE, Li   \n",
       "1230  ZHANG, Lei; CAO, Qing; WANG, Ying; LU, Quan; H...   \n",
       "1231                MA, Fuhai; HU, Haitao; TIAN, Yantao   \n",
       "1232       Liao, Rongye; Yang, Jie; Cao, Zhi; Wang, Jun   \n",
       "1233                           Alavi-Moghaddam, Mostafa   \n",
       "1234  Rodriguez-Morales, A. J.; Bonilla-Aldana, D. K...   \n",
       "1235  Chen, Nanshan and Zhou, Min and Dong, Xuan and...   \n",
       "\n",
       "                                                journal  \\\n",
       "0                                    Intensive Care Med   \n",
       "1                                        Cell Discovery   \n",
       "2                          Journal of Clinical Medicine   \n",
       "3                                            J Clin Med   \n",
       "4                                    Intensive Care Med   \n",
       "5                          Infectious Disease Modelling   \n",
       "6                          Infectious Disease Modelling   \n",
       "7                             The Lancet. Global health   \n",
       "8                            Journal of Travel Medicine   \n",
       "9                                                  mBio   \n",
       "10                    Global Health Research and Policy   \n",
       "11                                     Eurosurveillance   \n",
       "12                     Bioinformatics (Oxford, England)   \n",
       "13                    Journal of Korean Medical Science   \n",
       "14                                            Pathogens   \n",
       "15                                    EClinicalMedicine   \n",
       "16                    Infection, Genetics and Evolution   \n",
       "17                               Microbes and Infection   \n",
       "18                                        Cell Research   \n",
       "19                           Journal of travel medicine   \n",
       "20                            Hong Kong Medical Journal   \n",
       "21             Innovative Biosystems and Bioengineering   \n",
       "22                                     Eurosurveillance   \n",
       "23                          Precision Clinical Medicine   \n",
       "24                            Frontiers in Microbiology   \n",
       "25    International Journal of Environmental Researc...   \n",
       "26                                         Protein Cell   \n",
       "27                            The Lancet. Global health   \n",
       "28                                        Critical Care   \n",
       "29        Osong Public Health and Research Perspectives   \n",
       "...                                                 ...   \n",
       "1206                                             Nature   \n",
       "1207                                                NaN   \n",
       "1208         Chinese Journal of Hospital Administration   \n",
       "1209                                            Science   \n",
       "1210                                                NaN   \n",
       "1211                                                NaN   \n",
       "1212                     Chinese Journal of Stomatology   \n",
       "1213                                          N Z Med J   \n",
       "1214                 Zhejiang Da Xue Xue Bao Yi Xue Ban   \n",
       "1215                        Emergency medicine practice   \n",
       "1216                                                NaN   \n",
       "1217                      Iranian South Medical Journal   \n",
       "1218                                   Science Magazine   \n",
       "1219                                                NaN   \n",
       "1220             Chinese Journal of Laboratory Medicine   \n",
       "1221                                                NaN   \n",
       "1222                                   Science Magazine   \n",
       "1223             Chinese Journal of Preventive Medicine   \n",
       "1224       Chinese Journal of Obstetrics and Gynecology   \n",
       "1225                       Chinese Journal of Radiology   \n",
       "1226                            Nota Técnica-SVS/SES-RJ   \n",
       "1227         Chinese Journal of Hospital Administration   \n",
       "1228                                            Science   \n",
       "1229      Chinese Journal of Experimental Ophthalmology   \n",
       "1230     Chinese Journal of Applied Clinical Pediatrics   \n",
       "1231                        Chinese Journal of Oncology   \n",
       "1232  International Journal of Traditional Chinese M...   \n",
       "1233            Archives of Academic Emergency Medicine   \n",
       "1234                              Infezioni in Medicina   \n",
       "1235                                                NaN   \n",
       "\n",
       "      Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "0                    2.002765e+09          #3252          True  \n",
       "1                    3.003431e+09          #1861          True  \n",
       "2                    3.006065e+09          #1043          True  \n",
       "3                    1.776631e+08          #1999          True  \n",
       "4                    3.006643e+09          #3242         False  \n",
       "5                    3.006029e+09           #729          True  \n",
       "6                    3.006029e+09           #865          True  \n",
       "7                    2.627046e+09          #5386         False  \n",
       "8                    3.006304e+09          #2926          True  \n",
       "9                    3.005811e+09          #2427          True  \n",
       "10                   3.006646e+09          #5595          True  \n",
       "11                   3.005152e+09           #682          True  \n",
       "12                   3.003549e+09          #2776          True  \n",
       "13                   3.005657e+09          #3770          True  \n",
       "14                   3.003886e+09          #1502          True  \n",
       "15                   3.004913e+09          #3333          True  \n",
       "16                   2.286137e+09          #4581          True  \n",
       "17                            NaN           #236          True  \n",
       "18                   3.005213e+09           #205          True  \n",
       "19                   3.004048e+09           #102          True  \n",
       "20                   2.029408e+09          #3395          True  \n",
       "21                   3.006305e+09          #1218          True  \n",
       "22                   3.005657e+09          #2582          True  \n",
       "23                   3.006468e+09           #781          True  \n",
       "24                   3.003968e+09          #5462          True  \n",
       "25                   2.615949e+09          #3296          True  \n",
       "26                   2.766462e+09          #1815          True  \n",
       "27                   2.794879e+09          #5486         False  \n",
       "28                   2.537201e+09          #1225          True  \n",
       "29                   3.005888e+09          #2662         False  \n",
       "...                           ...            ...           ...  \n",
       "1206                          NaN          #3172           NaN  \n",
       "1207                          NaN           #294           NaN  \n",
       "1208                          NaN          #2106           NaN  \n",
       "1209                          NaN           #332           NaN  \n",
       "1210                          NaN          #1866           NaN  \n",
       "1211                          NaN           #922           NaN  \n",
       "1212                 2.377782e+09          #1228           NaN  \n",
       "1213                 2.185518e+09          #1541           NaN  \n",
       "1214                 2.349103e+09          #1922           NaN  \n",
       "1215                 3.005414e+09          #2815           NaN  \n",
       "1216                 1.604377e+09           #801           NaN  \n",
       "1217                 6.213808e+08          #4536           NaN  \n",
       "1218                 2.595675e+09          #2772           NaN  \n",
       "1219                 3.003591e+09          #1803           NaN  \n",
       "1220                 2.194450e+09          #2110           NaN  \n",
       "1221                 2.241123e+09          #1802           NaN  \n",
       "1222                 2.855668e+09          #2584           NaN  \n",
       "1223                 2.363411e+09          #2333           NaN  \n",
       "1224                 1.891210e+09          #2305           NaN  \n",
       "1225                 2.418494e+09          #2187           NaN  \n",
       "1226                 2.518955e+09           #216           NaN  \n",
       "1227                 2.348898e+09          #2108           NaN  \n",
       "1228                 2.343732e+09          #3808           NaN  \n",
       "1229                 3.005943e+09          #2051           NaN  \n",
       "1230                 2.416639e+09          #2063           NaN  \n",
       "1231                 2.366820e+09          #2176           NaN  \n",
       "1232                 2.616653e+09          #1962           NaN  \n",
       "1233                 3.006237e+09          #3509           NaN  \n",
       "1234                 3.005681e+09           #926           NaN  \n",
       "1235                 3.004581e+09            #22           NaN  \n",
       "\n",
       "[1236 rows x 14 columns]"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[(df_meta['source_x']=='CZI')&(df_meta['source_x'] != 'NaN')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>29139</th>\n",
       "      <td>090b6c8b3df30bc248221869f673a2d970caa1b9</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/19008417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29140</th>\n",
       "      <td>10525ac89e46be4cb9cb9fd1131d28411a902047</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/19011940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29141</th>\n",
       "      <td>b0899a264af548a89d649154aec569889717b295</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2019.12.17.19013490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29142</th>\n",
       "      <td>e93ba7d8a047795d5ec114741f32f5b18e8567c7</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.15.19015693</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29143</th>\n",
       "      <td>6cb02c7565f6f74a3d165e14196de5e9e87d2d04</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.23.20018549</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29144</th>\n",
       "      <td>cbc05d14c57b91081970a232ab83bc993f998fe2</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.26.20018754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29145</th>\n",
       "      <td>fdf006a84a946f24f5905dcea8c5c1ee266c26d2</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.26.20018887</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29146</th>\n",
       "      <td>25d49e49a4cb420ec8ed2a703c0ed88d7cd5d0d0</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.27.20018952</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29147</th>\n",
       "      <td>12fac9aedb1a09a3922a3c084ce4723708e463d6</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.27.20018986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29148</th>\n",
       "      <td>30b5c7b8faf95265f67ae59f4686eaf9b2772893</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.28.20019224</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29149</th>\n",
       "      <td>957e71389292e0881dff2bbe0e87fe6bf8b2d381</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.28.20019299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29150</th>\n",
       "      <td>2afff5fe89a3e3ec270a41c2b929dfeec68d7d50</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.29.20019547</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29151</th>\n",
       "      <td>a6af28a7f0a0d8d1e4d5b06f5ae38412298bbcb4</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.30.20019612</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29152</th>\n",
       "      <td>de3f8aaa209f08308aacb81b3a6fcd8db3e805de</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.30.20019828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29153</th>\n",
       "      <td>f17c6ac4986aecba2dc22f67b32f2d16b369117b</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.30.20019836</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29154</th>\n",
       "      <td>261df5b4f76de3cb7aa2e8dbeac321221f065df1</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.30.20019844</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29155</th>\n",
       "      <td>0ff4ad5359ee2df3568605385a8616d8da66bf2c</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.30.20019877</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29156</th>\n",
       "      <td>a01672035e0b9d21f9a934b9c9071610513325f2</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.31.20019265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29157</th>\n",
       "      <td>a44334e676e43c1889d282b0f0a8365f0f1e0c52</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.31.20019901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29158</th>\n",
       "      <td>898880683d9a962428033fd7500f74647c4a36f0</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.01.31.20019935</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29159</th>\n",
       "      <td>97e0830fe6c8bf6facb896320dd1b8d31ab49a54</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.01.20019984</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29160</th>\n",
       "      <td>7b40dfdf9a60cfe4a199cb967ee458b7dfaabe0c</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.02.20020016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29161</th>\n",
       "      <td>3990e8112c87103e73c37a3f89155b3e53326fd4</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20019141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29162</th>\n",
       "      <td>a1bff76ce360e8990b0a4ee2a5228a6e6e63d9c1</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20019497</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29163</th>\n",
       "      <td>5d3612cd95331a2f0c43ef9d03ce583b5ff8c995</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20020206</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29164</th>\n",
       "      <td>b0c409979750e2975f6c707cf4a9a44ee6c9e748</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20020248</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29165</th>\n",
       "      <td>21100aba41a4bfb48d7dc37f1bf5dbb38bf3867a</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20020263</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29166</th>\n",
       "      <td>5c525cc04075b7c82e52b1e80d96bb394b1a0907</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20020271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29167</th>\n",
       "      <td>b51d1c7287ac75790a52752489577d33d522898b</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.03.20020289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29168</th>\n",
       "      <td>1b531d266b4933bf8040739ed199914655b075d6</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.02.04.20020339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29470</th>\n",
       "      <td>083dd400b7854ffa908a21641197855aee981ea5</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20032003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29471</th>\n",
       "      <td>93d8f1552a81587e2f88e38580cbe4753b9934f1</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.05.20032011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29472</th>\n",
       "      <td>dd2cd004a197872cfce1c31a9685d6972f838efa</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20029793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29473</th>\n",
       "      <td>e53f5a6c4cbac340f0d97a4ba3289c30b7222dd5</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29474</th>\n",
       "      <td>fd847a97e6fce134345b25ae9240c4ebb680cca5</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29475</th>\n",
       "      <td>610549c879c62efc42f2ad32f63c65b25d8931c7</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29476</th>\n",
       "      <td>088d5c4a4f212be9590b6ce3d70a8e01847e76cf</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031880</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29477</th>\n",
       "      <td>fd96fc1859201e13e35526c188e022fa6e335327</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20031955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29478</th>\n",
       "      <td>26e75d3c815aae7fd9b094c3e5c74d3f7132ca13</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29479</th>\n",
       "      <td>bd08e2cbbc561e52823aedd0180f844e0c6cd2a6</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29480</th>\n",
       "      <td>ea17aad7692bb5040f3b2ba1d01b8249e4969469</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032334</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29481</th>\n",
       "      <td>6d3b3f4ab80a61c45f82c61c6c756cfc6ddf4bf2</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29482</th>\n",
       "      <td>3ba79e5d013c7c3cccd1449ccd16b93c61a4c576</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.06.20032425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29483</th>\n",
       "      <td>c1a08ce8f2c5d5311b19822c3df5aa9d25c4d93c</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20031393</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29484</th>\n",
       "      <td>c6012a6141f5ad05ac983aa9f1a7c57938363b16</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20031575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29485</th>\n",
       "      <td>74bee8bf3229f4c28991a7231c02d911d24770e9</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032326</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29486</th>\n",
       "      <td>ec90b0b861d31f910433d388370344024ca7c35d</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032524</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29487</th>\n",
       "      <td>970a28c4a3c1fbcb5322772956955b8c4a3bb257</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29488</th>\n",
       "      <td>abe590b95aa51e7309844156acdae4085870ea33</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29489</th>\n",
       "      <td>9a077c5ff1dcbb9104feeaa832e73ccdcf5fdcc9</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.07.20032672</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29490</th>\n",
       "      <td>6cf87a546884756094da0d300e85a061c2cc43ea</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20029710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29491</th>\n",
       "      <td>43f760b5cb9c3b430c6d0b31fa90d42e71a6429c</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20030643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29492</th>\n",
       "      <td>901572f1595cb6ea733858d3fc15b8ea46054061</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20031229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29493</th>\n",
       "      <td>4c84d1153daa335f5ff6d61db665024b32ac8195</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20031658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29494</th>\n",
       "      <td>e5f19b6daf956e815c779228cc0cad1293d65bbb</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29495</th>\n",
       "      <td>c42a617a00afe6a36bde0a8e3638e0f55bfee4f7</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032847</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29496</th>\n",
       "      <td>c4ce14ce42fa4360dfe3515ec9d1584847381c27</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032854</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29497</th>\n",
       "      <td>c41e09a32be90c84cea0616bb1c726aecba721e0</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20032219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29498</th>\n",
       "      <td>ca88735399ff43d0e673876200655099f06f5567</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20033183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29499</th>\n",
       "      <td>0b48e1bfcdff9a42c88cc80a98661feb2703390d</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20033217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>361 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sha source_x title  \\\n",
       "29139  090b6c8b3df30bc248221869f673a2d970caa1b9  medrxiv   NaN   \n",
       "29140  10525ac89e46be4cb9cb9fd1131d28411a902047  medrxiv   NaN   \n",
       "29141  b0899a264af548a89d649154aec569889717b295  medrxiv   NaN   \n",
       "29142  e93ba7d8a047795d5ec114741f32f5b18e8567c7  medrxiv   NaN   \n",
       "29143  6cb02c7565f6f74a3d165e14196de5e9e87d2d04  medrxiv   NaN   \n",
       "29144  cbc05d14c57b91081970a232ab83bc993f998fe2  medrxiv   NaN   \n",
       "29145  fdf006a84a946f24f5905dcea8c5c1ee266c26d2  medrxiv   NaN   \n",
       "29146  25d49e49a4cb420ec8ed2a703c0ed88d7cd5d0d0  medrxiv   NaN   \n",
       "29147  12fac9aedb1a09a3922a3c084ce4723708e463d6  medrxiv   NaN   \n",
       "29148  30b5c7b8faf95265f67ae59f4686eaf9b2772893  medrxiv   NaN   \n",
       "29149  957e71389292e0881dff2bbe0e87fe6bf8b2d381  medrxiv   NaN   \n",
       "29150  2afff5fe89a3e3ec270a41c2b929dfeec68d7d50  medrxiv   NaN   \n",
       "29151  a6af28a7f0a0d8d1e4d5b06f5ae38412298bbcb4  medrxiv   NaN   \n",
       "29152  de3f8aaa209f08308aacb81b3a6fcd8db3e805de  medrxiv   NaN   \n",
       "29153  f17c6ac4986aecba2dc22f67b32f2d16b369117b  medrxiv   NaN   \n",
       "29154  261df5b4f76de3cb7aa2e8dbeac321221f065df1  medrxiv   NaN   \n",
       "29155  0ff4ad5359ee2df3568605385a8616d8da66bf2c  medrxiv   NaN   \n",
       "29156  a01672035e0b9d21f9a934b9c9071610513325f2  medrxiv   NaN   \n",
       "29157  a44334e676e43c1889d282b0f0a8365f0f1e0c52  medrxiv   NaN   \n",
       "29158  898880683d9a962428033fd7500f74647c4a36f0  medrxiv   NaN   \n",
       "29159  97e0830fe6c8bf6facb896320dd1b8d31ab49a54  medrxiv   NaN   \n",
       "29160  7b40dfdf9a60cfe4a199cb967ee458b7dfaabe0c  medrxiv   NaN   \n",
       "29161  3990e8112c87103e73c37a3f89155b3e53326fd4  medrxiv   NaN   \n",
       "29162  a1bff76ce360e8990b0a4ee2a5228a6e6e63d9c1  medrxiv   NaN   \n",
       "29163  5d3612cd95331a2f0c43ef9d03ce583b5ff8c995  medrxiv   NaN   \n",
       "29164  b0c409979750e2975f6c707cf4a9a44ee6c9e748  medrxiv   NaN   \n",
       "29165  21100aba41a4bfb48d7dc37f1bf5dbb38bf3867a  medrxiv   NaN   \n",
       "29166  5c525cc04075b7c82e52b1e80d96bb394b1a0907  medrxiv   NaN   \n",
       "29167  b51d1c7287ac75790a52752489577d33d522898b  medrxiv   NaN   \n",
       "29168  1b531d266b4933bf8040739ed199914655b075d6  medrxiv   NaN   \n",
       "...                                         ...      ...   ...   \n",
       "29470  083dd400b7854ffa908a21641197855aee981ea5  medrxiv   NaN   \n",
       "29471  93d8f1552a81587e2f88e38580cbe4753b9934f1  medrxiv   NaN   \n",
       "29472  dd2cd004a197872cfce1c31a9685d6972f838efa  medrxiv   NaN   \n",
       "29473  e53f5a6c4cbac340f0d97a4ba3289c30b7222dd5  medrxiv   NaN   \n",
       "29474  fd847a97e6fce134345b25ae9240c4ebb680cca5  medrxiv   NaN   \n",
       "29475  610549c879c62efc42f2ad32f63c65b25d8931c7  medrxiv   NaN   \n",
       "29476  088d5c4a4f212be9590b6ce3d70a8e01847e76cf  medrxiv   NaN   \n",
       "29477  fd96fc1859201e13e35526c188e022fa6e335327  medrxiv   NaN   \n",
       "29478  26e75d3c815aae7fd9b094c3e5c74d3f7132ca13  medrxiv   NaN   \n",
       "29479  bd08e2cbbc561e52823aedd0180f844e0c6cd2a6  medrxiv   NaN   \n",
       "29480  ea17aad7692bb5040f3b2ba1d01b8249e4969469  medrxiv   NaN   \n",
       "29481  6d3b3f4ab80a61c45f82c61c6c756cfc6ddf4bf2  medrxiv   NaN   \n",
       "29482  3ba79e5d013c7c3cccd1449ccd16b93c61a4c576  medrxiv   NaN   \n",
       "29483  c1a08ce8f2c5d5311b19822c3df5aa9d25c4d93c  medrxiv   NaN   \n",
       "29484  c6012a6141f5ad05ac983aa9f1a7c57938363b16  medrxiv   NaN   \n",
       "29485  74bee8bf3229f4c28991a7231c02d911d24770e9  medrxiv   NaN   \n",
       "29486  ec90b0b861d31f910433d388370344024ca7c35d  medrxiv   NaN   \n",
       "29487  970a28c4a3c1fbcb5322772956955b8c4a3bb257  medrxiv   NaN   \n",
       "29488  abe590b95aa51e7309844156acdae4085870ea33  medrxiv   NaN   \n",
       "29489  9a077c5ff1dcbb9104feeaa832e73ccdcf5fdcc9  medrxiv   NaN   \n",
       "29490  6cf87a546884756094da0d300e85a061c2cc43ea  medrxiv   NaN   \n",
       "29491  43f760b5cb9c3b430c6d0b31fa90d42e71a6429c  medrxiv   NaN   \n",
       "29492  901572f1595cb6ea733858d3fc15b8ea46054061  medrxiv   NaN   \n",
       "29493  4c84d1153daa335f5ff6d61db665024b32ac8195  medrxiv   NaN   \n",
       "29494  e5f19b6daf956e815c779228cc0cad1293d65bbb  medrxiv   NaN   \n",
       "29495  c42a617a00afe6a36bde0a8e3638e0f55bfee4f7  medrxiv   NaN   \n",
       "29496  c4ce14ce42fa4360dfe3515ec9d1584847381c27  medrxiv   NaN   \n",
       "29497  c41e09a32be90c84cea0616bb1c726aecba721e0  medrxiv   NaN   \n",
       "29498  ca88735399ff43d0e673876200655099f06f5567  medrxiv   NaN   \n",
       "29499  0b48e1bfcdff9a42c88cc80a98661feb2703390d  medrxiv   NaN   \n",
       "\n",
       "                                       doi pmcid  pubmed_id  \\\n",
       "29139             doi.org/10.1101/19008417   NaN        NaN   \n",
       "29140             doi.org/10.1101/19011940   NaN        NaN   \n",
       "29141  doi.org/10.1101/2019.12.17.19013490   NaN        NaN   \n",
       "29142  doi.org/10.1101/2020.01.15.19015693   NaN        NaN   \n",
       "29143  doi.org/10.1101/2020.01.23.20018549   NaN        NaN   \n",
       "29144  doi.org/10.1101/2020.01.26.20018754   NaN        NaN   \n",
       "29145  doi.org/10.1101/2020.01.26.20018887   NaN        NaN   \n",
       "29146  doi.org/10.1101/2020.01.27.20018952   NaN        NaN   \n",
       "29147  doi.org/10.1101/2020.01.27.20018986   NaN        NaN   \n",
       "29148  doi.org/10.1101/2020.01.28.20019224   NaN        NaN   \n",
       "29149  doi.org/10.1101/2020.01.28.20019299   NaN        NaN   \n",
       "29150  doi.org/10.1101/2020.01.29.20019547   NaN        NaN   \n",
       "29151  doi.org/10.1101/2020.01.30.20019612   NaN        NaN   \n",
       "29152  doi.org/10.1101/2020.01.30.20019828   NaN        NaN   \n",
       "29153  doi.org/10.1101/2020.01.30.20019836   NaN        NaN   \n",
       "29154  doi.org/10.1101/2020.01.30.20019844   NaN        NaN   \n",
       "29155  doi.org/10.1101/2020.01.30.20019877   NaN        NaN   \n",
       "29156  doi.org/10.1101/2020.01.31.20019265   NaN        NaN   \n",
       "29157  doi.org/10.1101/2020.01.31.20019901   NaN        NaN   \n",
       "29158  doi.org/10.1101/2020.01.31.20019935   NaN        NaN   \n",
       "29159  doi.org/10.1101/2020.02.01.20019984   NaN        NaN   \n",
       "29160  doi.org/10.1101/2020.02.02.20020016   NaN        NaN   \n",
       "29161  doi.org/10.1101/2020.02.03.20019141   NaN        NaN   \n",
       "29162  doi.org/10.1101/2020.02.03.20019497   NaN        NaN   \n",
       "29163  doi.org/10.1101/2020.02.03.20020206   NaN        NaN   \n",
       "29164  doi.org/10.1101/2020.02.03.20020248   NaN        NaN   \n",
       "29165  doi.org/10.1101/2020.02.03.20020263   NaN        NaN   \n",
       "29166  doi.org/10.1101/2020.02.03.20020271   NaN        NaN   \n",
       "29167  doi.org/10.1101/2020.02.03.20020289   NaN        NaN   \n",
       "29168  doi.org/10.1101/2020.02.04.20020339   NaN        NaN   \n",
       "...                                    ...   ...        ...   \n",
       "29470  doi.org/10.1101/2020.03.05.20032003   NaN        NaN   \n",
       "29471  doi.org/10.1101/2020.03.05.20032011   NaN        NaN   \n",
       "29472  doi.org/10.1101/2020.03.06.20029793   NaN        NaN   \n",
       "29473  doi.org/10.1101/2020.03.06.20031278   NaN        NaN   \n",
       "29474  doi.org/10.1101/2020.03.06.20031377   NaN        NaN   \n",
       "29475  doi.org/10.1101/2020.03.06.20031856   NaN        NaN   \n",
       "29476  doi.org/10.1101/2020.03.06.20031880   NaN        NaN   \n",
       "29477  doi.org/10.1101/2020.03.06.20031955   NaN        NaN   \n",
       "29478  doi.org/10.1101/2020.03.06.20032144   NaN        NaN   \n",
       "29479  doi.org/10.1101/2020.03.06.20032177   NaN        NaN   \n",
       "29480  doi.org/10.1101/2020.03.06.20032334   NaN        NaN   \n",
       "29481  doi.org/10.1101/2020.03.06.20032417   NaN        NaN   \n",
       "29482  doi.org/10.1101/2020.03.06.20032425   NaN        NaN   \n",
       "29483  doi.org/10.1101/2020.03.07.20031393   NaN        NaN   \n",
       "29484  doi.org/10.1101/2020.03.07.20031575   NaN        NaN   \n",
       "29485  doi.org/10.1101/2020.03.07.20032326   NaN        NaN   \n",
       "29486  doi.org/10.1101/2020.03.07.20032524   NaN        NaN   \n",
       "29487  doi.org/10.1101/2020.03.07.20032573   NaN        NaN   \n",
       "29488  doi.org/10.1101/2020.03.07.20032599   NaN        NaN   \n",
       "29489  doi.org/10.1101/2020.03.07.20032672   NaN        NaN   \n",
       "29490  doi.org/10.1101/2020.03.08.20029710   NaN        NaN   \n",
       "29491  doi.org/10.1101/2020.03.08.20030643   NaN        NaN   \n",
       "29492  doi.org/10.1101/2020.03.08.20031229   NaN        NaN   \n",
       "29493  doi.org/10.1101/2020.03.08.20031658   NaN        NaN   \n",
       "29494  doi.org/10.1101/2020.03.08.20032821   NaN        NaN   \n",
       "29495  doi.org/10.1101/2020.03.08.20032847   NaN        NaN   \n",
       "29496  doi.org/10.1101/2020.03.08.20032854   NaN        NaN   \n",
       "29497  doi.org/10.1101/2020.03.09.20032219   NaN        NaN   \n",
       "29498  doi.org/10.1101/2020.03.09.20033183   NaN        NaN   \n",
       "29499  doi.org/10.1101/2020.03.09.20033217   NaN        NaN   \n",
       "\n",
       "                                               license abstract publish_time  \\\n",
       "29139  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29140  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29141  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29142  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29143  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29144  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29145  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29146  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29147  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29148  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29149  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29150  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29151  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29152  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29153  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29154  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29155  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29156  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29157  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29158  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29159  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29160  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29161  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29162  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29163  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29164  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29165  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29166  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29167  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29168  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "...                                                ...      ...          ...   \n",
       "29470  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29471  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29472  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29473  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29474  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29475  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29476  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29477  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29478  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29479  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29480  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29481  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29482  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29483  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29484  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29485  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29486  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29487  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29488  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29489  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29490  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29491  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29492  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29493  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29494  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29495  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29496  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29497  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29498  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29499  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "\n",
       "      authors journal  Microsoft Academic Paper ID WHO #Covidence  \\\n",
       "29139     NaN     NaN                          NaN            NaN   \n",
       "29140     NaN     NaN                          NaN            NaN   \n",
       "29141     NaN     NaN                          NaN            NaN   \n",
       "29142     NaN     NaN                          NaN            NaN   \n",
       "29143     NaN     NaN                          NaN            NaN   \n",
       "29144     NaN     NaN                          NaN            NaN   \n",
       "29145     NaN     NaN                          NaN            NaN   \n",
       "29146     NaN     NaN                          NaN            NaN   \n",
       "29147     NaN     NaN                          NaN            NaN   \n",
       "29148     NaN     NaN                          NaN            NaN   \n",
       "29149     NaN     NaN                          NaN            NaN   \n",
       "29150     NaN     NaN                          NaN            NaN   \n",
       "29151     NaN     NaN                          NaN            NaN   \n",
       "29152     NaN     NaN                          NaN            NaN   \n",
       "29153     NaN     NaN                          NaN            NaN   \n",
       "29154     NaN     NaN                          NaN            NaN   \n",
       "29155     NaN     NaN                          NaN            NaN   \n",
       "29156     NaN     NaN                          NaN            NaN   \n",
       "29157     NaN     NaN                          NaN            NaN   \n",
       "29158     NaN     NaN                          NaN            NaN   \n",
       "29159     NaN     NaN                          NaN            NaN   \n",
       "29160     NaN     NaN                          NaN            NaN   \n",
       "29161     NaN     NaN                          NaN            NaN   \n",
       "29162     NaN     NaN                          NaN            NaN   \n",
       "29163     NaN     NaN                          NaN            NaN   \n",
       "29164     NaN     NaN                          NaN            NaN   \n",
       "29165     NaN     NaN                          NaN            NaN   \n",
       "29166     NaN     NaN                          NaN            NaN   \n",
       "29167     NaN     NaN                          NaN            NaN   \n",
       "29168     NaN     NaN                          NaN            NaN   \n",
       "...       ...     ...                          ...            ...   \n",
       "29470     NaN     NaN                          NaN            NaN   \n",
       "29471     NaN     NaN                          NaN            NaN   \n",
       "29472     NaN     NaN                          NaN            NaN   \n",
       "29473     NaN     NaN                          NaN            NaN   \n",
       "29474     NaN     NaN                          NaN            NaN   \n",
       "29475     NaN     NaN                          NaN            NaN   \n",
       "29476     NaN     NaN                          NaN            NaN   \n",
       "29477     NaN     NaN                          NaN            NaN   \n",
       "29478     NaN     NaN                          NaN            NaN   \n",
       "29479     NaN     NaN                          NaN            NaN   \n",
       "29480     NaN     NaN                          NaN            NaN   \n",
       "29481     NaN     NaN                          NaN            NaN   \n",
       "29482     NaN     NaN                          NaN            NaN   \n",
       "29483     NaN     NaN                          NaN            NaN   \n",
       "29484     NaN     NaN                          NaN            NaN   \n",
       "29485     NaN     NaN                          NaN            NaN   \n",
       "29486     NaN     NaN                          NaN            NaN   \n",
       "29487     NaN     NaN                          NaN            NaN   \n",
       "29488     NaN     NaN                          NaN            NaN   \n",
       "29489     NaN     NaN                          NaN            NaN   \n",
       "29490     NaN     NaN                          NaN            NaN   \n",
       "29491     NaN     NaN                          NaN            NaN   \n",
       "29492     NaN     NaN                          NaN            NaN   \n",
       "29493     NaN     NaN                          NaN            NaN   \n",
       "29494     NaN     NaN                          NaN            NaN   \n",
       "29495     NaN     NaN                          NaN            NaN   \n",
       "29496     NaN     NaN                          NaN            NaN   \n",
       "29497     NaN     NaN                          NaN            NaN   \n",
       "29498     NaN     NaN                          NaN            NaN   \n",
       "29499     NaN     NaN                          NaN            NaN   \n",
       "\n",
       "      has_full_text  \n",
       "29139          True  \n",
       "29140          True  \n",
       "29141          True  \n",
       "29142          True  \n",
       "29143          True  \n",
       "29144          True  \n",
       "29145         False  \n",
       "29146         False  \n",
       "29147          True  \n",
       "29148          True  \n",
       "29149          True  \n",
       "29150          True  \n",
       "29151          True  \n",
       "29152         False  \n",
       "29153          True  \n",
       "29154         False  \n",
       "29155          True  \n",
       "29156          True  \n",
       "29157          True  \n",
       "29158          True  \n",
       "29159          True  \n",
       "29160          True  \n",
       "29161         False  \n",
       "29162          True  \n",
       "29163          True  \n",
       "29164         False  \n",
       "29165          True  \n",
       "29166          True  \n",
       "29167          True  \n",
       "29168          True  \n",
       "...             ...  \n",
       "29470         False  \n",
       "29471          True  \n",
       "29472          True  \n",
       "29473          True  \n",
       "29474          True  \n",
       "29475          True  \n",
       "29476         False  \n",
       "29477          True  \n",
       "29478          True  \n",
       "29479         False  \n",
       "29480          True  \n",
       "29481          True  \n",
       "29482         False  \n",
       "29483          True  \n",
       "29484          True  \n",
       "29485          True  \n",
       "29486          True  \n",
       "29487          True  \n",
       "29488          True  \n",
       "29489          True  \n",
       "29490          True  \n",
       "29491          True  \n",
       "29492          True  \n",
       "29493          True  \n",
       "29494          True  \n",
       "29495          True  \n",
       "29496          True  \n",
       "29497         False  \n",
       "29498          True  \n",
       "29499          True  \n",
       "\n",
       "[361 rows x 14 columns]"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['source_x']=='medrxiv']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scrape papers from bioRxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "def biorxiv_html_paper_data_extractor(soup):\n",
    "    '''\n",
    "        Extract paper information from html webpage of biorxiv\n",
    "    '''\n",
    "    paper = {}\n",
    "    paper['title'] = soup.h1.text\n",
    "\n",
    "    # paper authors\n",
    "    authors = []\n",
    "    for author in soup.find('span',attrs={'class':'highwire-citation-authors'}).find_all('span',attrs={'class':'highwire-citation-author'}):\n",
    "        author = author.find('span',attrs={'class':'nlm-given-names'}).text + ' ' + author.find('span',attrs={'class':'nlm-surname'}).text\n",
    "        authors.append(author)\n",
    "    #print(authors)\n",
    "    paper['authors'] = authors\n",
    "\n",
    "    sections = {}\n",
    "    for section in soup.find_all('div',attrs={'class':'section'}):\n",
    "        section_name = section.find('h2').text\n",
    "        if section_name != 'References':\n",
    "            #print(section_name)\n",
    "            sections[section_name] = ''\n",
    "            for p in section.find_all('p'):\n",
    "                sections[section_name] += p.text + '\\n'\n",
    "            #print(sections[section_name])\n",
    "            #print()\n",
    "    paper['sections'] = sections\n",
    "\n",
    "    # get tables\n",
    "    tables = []\n",
    "    html_tables = tmp_soup.find_all('div',attrs={'class':'table'})\n",
    "    for html_table in html_tables:\n",
    "        table = {}\n",
    "        table['id'] = html_table['id']\n",
    "        \n",
    "        table_label = html_table.find('span',attrs={'class':'table-label'})\n",
    "        if table_label:\n",
    "            table['label'] = table_label.text\n",
    "            \n",
    "        table_caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "        if table_caption:\n",
    "            table['caption'] = table_caption.text\n",
    "            \n",
    "        table['description'] = ''\n",
    "        for p in html_table.find_all('p'):\n",
    "            table['description'] += p.text + '\\n'\n",
    "        table['image'] = None\n",
    "        tables.append(table)\n",
    "        \n",
    "    paper['tables'] = tables\n",
    "    \n",
    "    # get figures\n",
    "    figures = []\n",
    "    html_figures = tmp_soup.find_all('div',attrs={'class':'fig'})\n",
    "    for html_figure in html_figures:\n",
    "        figure = {}\n",
    "        \n",
    "        figure_label = html_figure.find('span',attrs={'class':'fig-label'})\n",
    "        if figure_label:\n",
    "            figure['label'] = figure_label.text\n",
    "            \n",
    "        figure_title = html_figure.find('span',attrs={'class':'caption-title'})\n",
    "        if figure_title:\n",
    "            figure['title'] = title.text\n",
    "            \n",
    "        figure['description'] = ''\n",
    "        figure_ps = html_figure.find_all('p')\n",
    "        for p in figure_ps:\n",
    "            figure['description'] += p.text + '\\n'\n",
    "        figures.append(figure)\n",
    "        figure_img = html_figure.find('img')\n",
    "        if figure_img:\n",
    "            figure['img'] = figure_img['data-src']\n",
    "    \n",
    "    paper['figures'] = figures\n",
    "    \n",
    "    # getting references data\n",
    "    references = []\n",
    "    for ref in soup.find('div',attrs={'class':'section ref-list'}).find_all('li'):\n",
    "        reference = {}\n",
    "        # label\n",
    "        try:\n",
    "            reference['label'] = ref.find('span',attrs={'class':'ref-label'}).text\n",
    "        except:\n",
    "            reference['label'] = None\n",
    "\n",
    "        # authors\n",
    "        author_names = []\n",
    "        try:\n",
    "            authors = ref.find('cite').find_all('span',attrs={'class':'cit-auth'})\n",
    "            for author in authors:\n",
    "                surname = author.find('span',attrs={'class':'cit-name-surname'}).text \n",
    "                given_names = author.find('span',attrs={'class':'cit-name-given-names'}).text\n",
    "                author_name = surname + ' ' + given_names\n",
    "                author_names.append(author_name)\n",
    "            reference['authors'] = author_names\n",
    "        except:\n",
    "            reference['authors'] = author_names\n",
    "\n",
    "        # date\n",
    "        try:\n",
    "            reference['date'] = ref.find('cite').find('span',attrs={'class':'cit-pub-date'}).text\n",
    "        except:\n",
    "            reference['date'] = None\n",
    "\n",
    "        # title\n",
    "        try:\n",
    "            reference['title'] = ref.find('cite').find('span',attrs={'class':'cit-article-title'}).text\n",
    "        except:\n",
    "            reference['title'] = ''\n",
    "\n",
    "        # links\n",
    "        ref_links = []\n",
    "        try:\n",
    "            links = ref.find('div',attrs={'class':'cit-extra'}).find_all('a')\n",
    "            for link in links:\n",
    "                if link.text != 'OpenUrl':\n",
    "                    link_name = link.text\n",
    "                    link_url = link['href']\n",
    "                    ref_links.append((link_name,link_url))\n",
    "            reference['links'] = ref_links\n",
    "        except:\n",
    "            reference['links'] = ref_links\n",
    "        references.append(reference)\n",
    "    paper['references'] = references\n",
    "    #print(references)\n",
    "    return paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "source = 'biorxiv'\n",
    "papers_df = df_meta[df_meta['source_x']==source][['sha','doi']]\n",
    "papers_array = papers_df.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/001727v1.full\n",
      "Paper ID: f056da9c64fbf00a4645ae326e8a4339d015d155 saved to data/biorxiv.\n",
      "Added 1 papers to data/biorxiv.\n"
     ]
    }
   ],
   "source": [
    "# get papers from data/biorxiv\n",
    "existing_paper_ids = set([name.strip('.json') for name in get_file_names('data/biorxiv')])\n",
    "\n",
    "# go through all papers\n",
    "new_paper_cnt = 0\n",
    "for row in papers_array[:1]:\n",
    "    if row[0] not in existing_paper_ids:\n",
    "        paper = {}\n",
    "        paper['source'] = 'biorxiv'\n",
    "        paper['id'] = row[0]\n",
    "        paper['url'] = 'https://biorxiv.org/content' + row[1].strip('doi.org') + 'v1.full'\n",
    "        print('Fetching paper data at {}'.format(paper['url']))\n",
    "        html = session.get(paper['url'])\n",
    "        tmp_soup = BeautifulSoup(html.content,'html.parser')\n",
    "        paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "        paper = dict(chain(paper.items(),paper_webdata.items()))    \n",
    "        with open('data/biorxiv/{}.json'.format(paper['id']), 'w') as fp:\n",
    "            json.dump(paper, fp, indent=4)\n",
    "        print('Paper ID: {} saved to data/biorxiv.'.format(paper['id']))\n",
    "        new_paper_cnt += 1\n",
    "print('Added {} papers to data/biorxiv.'.format(new_paper_cnt))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://biorxiv.org/content' + papers_array[0][1].strip('doi.org') + 'v1.full'\n",
    "html = session.get(url)\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'label': 'Figure 1.', 'description': 'A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.\\n', 'img': 'https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F1.medium.gif'}\n",
      "\n",
      "{'label': 'Figure 2.', 'description': 'Relationship of reported value for each program (horizontal axis, log scale) to the empirically-determined Positive Predictive Value (PPV), shown on the vertical axis. While the exact values depend on the test data used, the general values at significant cutoff values (0.8, 0.9, 0.95 PPV) remain relatively constant across different datasets (data not shown).\\n', 'img': 'https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F2.medium.gif'}\n",
      "\n",
      "{'label': 'Figure 3.', 'description': 'The Positive Predictive Value (PPV, vertical axis) is shown for each organism (boxes on right), at each level of known abundance (horizontal axis, see Table 2), and each program (boxes at top), across a maximum of 50 replicates (indicated by the size of each point). Note that the reference database for MetaPhlAn does not include viruses, and the reference database for Kraken does not include RNA viruses\\n', 'img': 'https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F3.medium.gif'}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get figures\n",
    "figures = []\n",
    "html_figures = tmp_soup.find_all('div',attrs={'class':'fig'})\n",
    "for html_figure in html_figures:\n",
    "    figure = {}\n",
    "    figure_label = html_figure.find('span',attrs={'class':'fig-label'})\n",
    "    if figure_label:\n",
    "        figure['label'] = figure_label.text\n",
    "    figure_title = html_figure.find('span',attrs={'class':'caption-title'})\n",
    "    if figure_title:\n",
    "        figure['title'] = title.text\n",
    "    figure['description'] = ''\n",
    "    figure_ps = html_figure.find_all('p')\n",
    "    for p in figure_ps:\n",
    "        figure['description'] += p.text + '\\n'\n",
    "    figures.append(figure)\n",
    "    figure_img = html_figure.find('img')\n",
    "    if figure_img:\n",
    "        figure['img'] = figure_img['data-src']\n",
    "\n",
    "for figure in figures:\n",
    "    print(figure)\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'T1', 'label': 'Table 1.', 'description': 'Summary of methods for metagenomic classification.\\n', 'image': None}\n",
      "{'id': 'T2', 'label': 'Table 2.', 'caption': 'The abundance of each target organism in each set of simulated datasets. Each set is indicated by the number in the top row, and was generated with 50 replicates.', 'description': '', 'image': None}\n"
     ]
    }
   ],
   "source": [
    "# get tables\n",
    "tables = []\n",
    "html_tables = tmp_soup.find_all('div',attrs={'class':'table'})\n",
    "for html_table in html_tables:\n",
    "    table = {}\n",
    "    table['id'] = html_table['id']\n",
    "    label = html_table.find('span',attrs={'class':'table-label'})\n",
    "    if label:\n",
    "        table['label'] = label.text\n",
    "    caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "    if caption:\n",
    "        table['caption'] = caption.text\n",
    "    table['description'] = ''\n",
    "    for p in html_table.find_all('p'):\n",
    "        table['description'] += p.text + '\\n'\n",
    "    table['image'] = None\n",
    "    tables.append(table)\n",
    "\n",
    "for table in tables:\n",
    "    print(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<p class=\"first-child\" id=\"p-9\">A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.</p>]"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "html_figures[0].find_all('p')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scrape papers from medrxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get papers from data/medrxiv\n",
    "existing_paper_ids = set([name.strip('.json') for name in get_file_names('data/medrxiv')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scrape papers from CZI\n",
    "Journals: 1198"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c630ebcdf30652f0422c3ec12a00b50241dc9bd9</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) as a SA...</td>\n",
       "      <td>10.1007/s00134-020-05985-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125455.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>2.002765e+09</td>\n",
       "      <td>#3252</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>53eccda7977a31e3d0f565c884da036b1e85438e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comparative genetic analysis of the novel coro...</td>\n",
       "      <td>10.1038/s41421-020-0147-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...</td>\n",
       "      <td>Cell Discovery</td>\n",
       "      <td>3.003431e+09</td>\n",
       "      <td>#1861</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>210a892deb1c61577f6fba58505fd65356ce6636</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Incubation Period and Other Epidemiological Ch...</td>\n",
       "      <td>10.3390/jcm9020538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The geographic spread of 2019 novel coronaviru...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...</td>\n",
       "      <td>Journal of Clinical Medicine</td>\n",
       "      <td>3.006065e+09</td>\n",
       "      <td>#1043</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>e3b40cc8e0e137c416b4a2273a4dca94ae8178cc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Characteristics of and Public Health Responses...</td>\n",
       "      <td>10.3390/jcm9020575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32093211.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, cases of unidentified pneumo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Deng, Sheng-Qun; Peng, Hong-Juan</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>1.776631e+08</td>\n",
       "      <td>#1999</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92c2c9839304b4f2bc1276d41b1aa885d8b364fd</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Imaging changes in severe COVID-19 pneumonia</td>\n",
       "      <td>10.1007/s00134-020-05976-w</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125453.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Wei</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>3.006643e+09</td>\n",
       "      <td>#3242</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0df0d5270a9399cf4e23c0cdd877a80616a9725e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>An updated estimation of the risk of transmiss...</td>\n",
       "      <td>10.1016/j.idm.2020.02.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The basic reproduction number of an infectious...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#729</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>f24242580be243d5fc3f432915d86af6854bb8b7</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Real-time forecasts of the 2019-nCoV epidemic ...</td>\n",
       "      <td>10.1016/j.idm.2020.02.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The initial cluster of severe pneumonia cases ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#865</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>d13a685f861b0f1ba05afa6e005311ad1820fd3a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>RETRACTED: Chinese medical staff request inter...</td>\n",
       "      <td>10.1016/s2214-109x(20)30065-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32105614.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zeng, Yingchun; Zhen, Yan</td>\n",
       "      <td>The Lancet. Global health</td>\n",
       "      <td>2.627046e+09</td>\n",
       "      <td>#5386</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19 outbreak on the Diamond Princess crui...</td>\n",
       "      <td>10.1093/jtm/taaa030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Cruise ships carry a large number of people in...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.</td>\n",
       "      <td>Journal of Travel Medicine</td>\n",
       "      <td>3.006304e+09</td>\n",
       "      <td>#2926</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>e9239100c5493ea914dc23c3d7a262f4326022ac</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Distinct Roles for Sialoside and Protein Recep...</td>\n",
       "      <td>10.1128/mBio.02764-19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Coronaviruses (CoVs) are common human and anim...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Qing, Enya; Hantak, Michael; Perlman, Stanley;...</td>\n",
       "      <td>mBio</td>\n",
       "      <td>3.005811e+09</td>\n",
       "      <td>#2427</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>469ed0f00c09e2637351c9735c306f27acf3aace</td>\n",
       "      <td>CZI</td>\n",
       "      <td>First two months of the 2019 Coronavirus Disea...</td>\n",
       "      <td>10.1186/s41256-020-00137-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Similar to outbreaks of many other infectious ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Xinguang; Yu, Bin</td>\n",
       "      <td>Global Health Research and Policy</td>\n",
       "      <td>3.006646e+09</td>\n",
       "      <td>#5595</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4e550e034ccca6fa2a91e481ddba24db67bc9ae5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness of airport screening at detectin...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.5.2000080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>We simulated 100 2019-nCoV infected travellers...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Quilty, Billy J; Clifford, Sam; group2, CMMID ...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005152e+09</td>\n",
       "      <td>#682</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4bbb0c59babc718f67953fae032dad6ae0d7aeb1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Genome Detective Coronavirus Typing Tool for r...</td>\n",
       "      <td>10.1093/bioinformatics/btaa145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32108862.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>SUMMARY: Genome Detective is a web-based, user...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>3.003549e+09</td>\n",
       "      <td>#2776</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>c821803c55c6aad89b6d0c1d3ba252051e464017</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Case of the Index Patient Who Caused Tertiary ...</td>\n",
       "      <td>10.3346/jkms.2020.35.e79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Since mid-December of 2019, coronavirus diseas...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...</td>\n",
       "      <td>Journal of Korean Medical Science</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#3770</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>c3bee2a4caca614b34f92c17b643b854dcdab28d</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Emergence of Novel Coronavirus 2019-nCoV: Need...</td>\n",
       "      <td>10.3390/pathogens9020148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Novel Coronavirus (2019-nCoV) is an emerging p...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...</td>\n",
       "      <td>Pathogens</td>\n",
       "      <td>3.003886e+09</td>\n",
       "      <td>#1502</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>715842fa536064980818ad7e31ce511272a4b6bc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The coronavirus 2019-nCoV epidemic: Is hindsig...</td>\n",
       "      <td>10.1016/j.eclinm.2020.100289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...</td>\n",
       "      <td>EClinicalMedicine</td>\n",
       "      <td>3.004913e+09</td>\n",
       "      <td>#3333</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Nonstructural proteins NS7b and NS8 are likely...</td>\n",
       "      <td>10.1016/j.meegid.2020.104272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The seventh novel human infecting Betacoronavi...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro</td>\n",
       "      <td>Infection, Genetics and Evolution</td>\n",
       "      <td>2.286137e+09</td>\n",
       "      <td>#4581</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>06c89f69aa7b5f9648d2c1543b8246fe9c3610cf</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Pathogenicity and Transmissibility of 2019-nCo...</td>\n",
       "      <td>10.1016/j.micinf.2020.01.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>A zoonotic coronavirus, labeled as 2019-nCoV b...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Jieliang</td>\n",
       "      <td>Microbes and Infection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#236</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>acb678bdd7634055de18d0b89bb6a4890e6a0306</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Remdesivir and chloroquine effectively inhibit...</td>\n",
       "      <td>10.1038/s41422-020-0282-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...</td>\n",
       "      <td>Cell Research</td>\n",
       "      <td>3.005213e+09</td>\n",
       "      <td>#205</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>47772f3e98d8c61bb9782531a0338ba85f27bf3f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential for global spread of a novel coronav...</td>\n",
       "      <td>10.1093/jtm/taaa011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...</td>\n",
       "      <td>Journal of travel medicine</td>\n",
       "      <td>3.004048e+09</td>\n",
       "      <td>#102</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Severe acute respiratory symptoms and suspecte...</td>\n",
       "      <td>10.12809/hkmj208375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Hon, K. L.; Leung, K. K. Y.</td>\n",
       "      <td>Hong Kong Medical Journal</td>\n",
       "      <td>2.029408e+09</td>\n",
       "      <td>#3395</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4ff89a71126d2932544a8337ba28787fde5f02a8</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Statistics-Based Predictions of Coronavirus Ep...</td>\n",
       "      <td>10.20535/ibb.2020.4.1.195074</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Information about the open-access article 'Sta...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Nesteruk, Igor</td>\n",
       "      <td>Innovative Biosystems and Bioengineering</td>\n",
       "      <td>3.006305e+09</td>\n",
       "      <td>#1218</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ac37ba61c91bb6939a507dbf4efe2119ddafeb9c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Early transmission patterns of coronavirus dis...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.8.2000097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#2582</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0</td>\n",
       "      <td>CZI</td>\n",
       "      <td>HRCT Imaging Features in Representative Import...</td>\n",
       "      <td>10.1093/pcmedi/pbaa004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>With the spread of novel coronavirus (2019-nCo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...</td>\n",
       "      <td>Precision Clinical Medicine</td>\n",
       "      <td>3.006468e+09</td>\n",
       "      <td>#781</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>c9fee561c2a3834645dbb61dc4ae6448051da492</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comprehensive Genomic Characterization Analysi...</td>\n",
       "      <td>10.3389/fmicb.2019.03036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Porcine delta coronavirus (PDCoV) is a novel e...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...</td>\n",
       "      <td>Frontiers in Microbiology</td>\n",
       "      <td>3.003968e+09</td>\n",
       "      <td>#5462</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>655537fc8cc52bccf43cf7189ab060d3097caa7a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential Factors Influencing Repeated SARS Ou...</td>\n",
       "      <td>10.3390/ijerph17051633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Within last 17 years two widespread epidemics ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....</td>\n",
       "      <td>International Journal of Environmental Researc...</td>\n",
       "      <td>2.615949e+09</td>\n",
       "      <td>#3296</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>bb7cd48a85b72f8d73e57d655c6d8a034fce3778</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Puzzle of highly pathogenic human coronaviruse...</td>\n",
       "      <td>10.1007/s13238-020-00693-y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32088858.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Li, Jing; Liu, Wenjun</td>\n",
       "      <td>Protein Cell</td>\n",
       "      <td>2.766462e+09</td>\n",
       "      <td>#1815</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3c42431c5b95e707486ce7441ac139071e07b706</td>\n",
       "      <td>CZI</td>\n",
       "      <td>RETRACTION: Retraction-Chinese medical staff r...</td>\n",
       "      <td>10.1016/s2214-109x(20)30076-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32113504.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>The Editors Of The Lancet Global, Health</td>\n",
       "      <td>The Lancet. Global health</td>\n",
       "      <td>2.794879e+09</td>\n",
       "      <td>#5486</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>6a93283b499ae5bc6aaf29f14e701dc8f25138ea</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Critical care response to a hospital outbreak ...</td>\n",
       "      <td>10.1186/s13054-020-2786-x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liu, Yong; Li, Jinxiu; Feng, Yongwen</td>\n",
       "      <td>Critical Care</td>\n",
       "      <td>2.537201e+09</td>\n",
       "      <td>#1225</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>6fc52ed878c271020a2a375bb6e4b12943a7666c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness for the Response to COVID-19: Th...</td>\n",
       "      <td>10.24171/j.phrp.2020.11.1.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>In the current issue of Osong Public Health an...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cho, Hae-Wol</td>\n",
       "      <td>Osong Public Health and Research Perspectives</td>\n",
       "      <td>3.005888e+09</td>\n",
       "      <td>#2662</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1206</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Daily briefing: World’s biggest physics meetin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The March Meeting of the APS was cancelled jus...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Graham, Flora</td>\n",
       "      <td>Nature</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#3172</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1207</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>How Coronaviruses Cause Infection—from Colds t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The novel coronavirus outbreak raises question...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Makin, Simon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#294</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1208</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Advices on the prevention and control of nosoc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The pneumonia caused by the novel coronavirus ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XU, Hongzhen; CHEN, Shuohui; FU, Junfen; SHU, ...</td>\n",
       "      <td>Chinese Journal of Hospital Administration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#2106</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1209</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>‘This beast is moving very fast.’ Will the new...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The silver lining of the epidemic is that scie...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cohen, Kai Kupferschmidt; Jon</td>\n",
       "      <td>Science</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#332</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1210</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo Coronavírus: atendimento a pessoas com su...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Brasil. Ministerio da, Saude</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#1866</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1211</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo coronavirus (2019 nCov) Medidas de preven...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>São Paulo Secretaria da, Saúde</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#922</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1212</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The prevention and control of a new coronaviru...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32057210.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>During a short period of time, the outbreak of...</td>\n",
       "      <td>2020</td>\n",
       "      <td>LI, Zhi yong; MENG, Liu yan</td>\n",
       "      <td>Chinese Journal of Stomatology</td>\n",
       "      <td>2.377782e+09</td>\n",
       "      <td>#1228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19: another infectious disease emerging ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32078596.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Murdoch, David R.; French, Nigel P.</td>\n",
       "      <td>N Z Med J</td>\n",
       "      <td>2.185518e+09</td>\n",
       "      <td>#1541</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Dynamic changes of chest CT imaging in patient...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32096366.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OBJECTIVE: To analyze the dynamic changes of c...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Jincheng; Liu, Jinpeng; Wang, Yuanyuan; ...</td>\n",
       "      <td>Zhejiang Da Xue Xue Bao Yi Xue Ban</td>\n",
       "      <td>2.349103e+09</td>\n",
       "      <td>#1922</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1215</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus COVID-19: an overview for em...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32105049.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prior to the global outbreak of SARS-CoV in 20...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Giwa, A.; Desai, A.</td>\n",
       "      <td>Emergency medicine practice</td>\n",
       "      <td>3.005414e+09</td>\n",
       "      <td>#2815</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Coronavirus 9-nCoV: recomendaciones para aerop...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ante la situación mundial, en relación a 2019-...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Argentina. Ministerio de, Salud</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.604377e+09</td>\n",
       "      <td>#801</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel Coronavirus Disease 2019 (COVID-19): An ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Background: At the beginning of the New Year 2...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Keshavarz, Ahmad Tavakoli; Katayon, Vahdat; Mo...</td>\n",
       "      <td>Iranian South Medical Journal</td>\n",
       "      <td>6.213808e+08</td>\n",
       "      <td>#4536</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Coronavirus just caused the American Physical ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Citing the growing threat of the coronavirus, ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cho, Adrian</td>\n",
       "      <td>Science Magazine</td>\n",
       "      <td>2.595675e+09</td>\n",
       "      <td>#2772</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Mensajero de la Salud: nuevo coronavirus 2019-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Declaratoria de Emergencia en Salud Pública de...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Mexico. Secretaria de la, Salud</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.003591e+09</td>\n",
       "      <td>#1803</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The management of biosafety risk in clinical l...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>During the outbreak of coronavirus disease-19 ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XIAO, Yuling; LU, Xiaojun; KANG, Mei; LI, Dong...</td>\n",
       "      <td>Chinese Journal of Laboratory Medicine</td>\n",
       "      <td>2.194450e+09</td>\n",
       "      <td>#2110</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Lineamiento estandarizado para la Vigilancia E...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Este documento describe la situación epidemiol...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Mexico. Secretaria de Salud. Subsecretaria de ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.241123e+09</td>\n",
       "      <td>#1802</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Singapore claims first use of antibody test to...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In what appears to be a first, disease tracker...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Normile, Dennis</td>\n",
       "      <td>Science Magazine</td>\n",
       "      <td>2.855668e+09</td>\n",
       "      <td>#2584</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The network investigation on knowledge, attitu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To analyze the current situation of ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>CHEN, Yan</td>\n",
       "      <td>Chinese Journal of Preventive Medicine</td>\n",
       "      <td>2.363411e+09</td>\n",
       "      <td>#2333</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Investigation on demands for antenatal care se...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To identify problems and demands for...</td>\n",
       "      <td>2020</td>\n",
       "      <td>DU, Li; GU, Yibin; CUI, Mengqing; LI, Wenxian;...</td>\n",
       "      <td>Chinese Journal of Obstetrics and Gynecology</td>\n",
       "      <td>1.891210e+09</td>\n",
       "      <td>#2305</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1225</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Analysis of early chest high resolution CT ima...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To investigate the first chest HRCT ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>LIU, Haifeng; ZHANG, Dongyou; YANG, Yi; LONG, ...</td>\n",
       "      <td>Chinese Journal of Radiology</td>\n",
       "      <td>2.418494e+09</td>\n",
       "      <td>#2187</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1226</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo Coronavirus (nCoV)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Os Coronavírus (CoV) compõem uma grande famíli...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rio de Janeiro . Secretaria de Estado de Saúde...</td>\n",
       "      <td>Nota Técnica-SVS/SES-RJ</td>\n",
       "      <td>2.518955e+09</td>\n",
       "      <td>#216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1227</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Difficulties and strategies of public hospital...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Outbreak of the novel coronavirus pneumonia (N...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XU, Dong; HU, Yu; DING, Ning; XIA, Jiahong; ZH...</td>\n",
       "      <td>Chinese Journal of Hospital Administration</td>\n",
       "      <td>2.348898e+09</td>\n",
       "      <td>#2108</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Get Real</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Since this is going to be a post about the cor...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lowe, Derek</td>\n",
       "      <td>Science</td>\n",
       "      <td>2.343732e+09</td>\n",
       "      <td>#3808</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>How Ophthalmologists Should Understand and Res...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The new coronavirus pneumonia that first appea...</td>\n",
       "      <td>2020</td>\n",
       "      <td>ZHIJIE, Li</td>\n",
       "      <td>Chinese Journal of Experimental Ophthalmology</td>\n",
       "      <td>3.005943e+09</td>\n",
       "      <td>#2051</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1230</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Experts proposal and frequently asked question...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The outbreak of novel coronavirus pneumonia (N...</td>\n",
       "      <td>2020</td>\n",
       "      <td>ZHANG, Lei; CAO, Qing; WANG, Ying; LU, Quan; H...</td>\n",
       "      <td>Chinese Journal of Applied Clinical Pediatrics</td>\n",
       "      <td>2.416639e+09</td>\n",
       "      <td>#2063</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1231</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Surgical treatment strategy for digestive syst...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The outbreak of novel coronavirus pneumonia oc...</td>\n",
       "      <td>2020</td>\n",
       "      <td>MA, Fuhai; HU, Haitao; TIAN, Yantao</td>\n",
       "      <td>Chinese Journal of Oncology</td>\n",
       "      <td>2.366820e+09</td>\n",
       "      <td>#2176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1232</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus pneumonia in the primary gen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>To observe the curative effect of TCM syndrome...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liao, Rongye; Yang, Jie; Cao, Zhi; Wang, Jun</td>\n",
       "      <td>International Journal of Traditional Chinese M...</td>\n",
       "      <td>2.616653e+09</td>\n",
       "      <td>#1962</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>A Novel Coronavirus Outbreak from Wuhan City i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Alavi-Moghaddam, Mostafa</td>\n",
       "      <td>Archives of Academic Emergency Medicine</td>\n",
       "      <td>3.006237e+09</td>\n",
       "      <td>#3509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1234</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>History is repeating itself: Probable zoonotic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rodriguez-Morales, A. J.; Bonilla-Aldana, D. K...</td>\n",
       "      <td>Infezioni in Medicina</td>\n",
       "      <td>3.005681e+09</td>\n",
       "      <td>#926</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1235</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Epidemiological and Clinical Characteristics o...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chen, Nanshan and Zhou, Min and Dong, Xuan and...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.004581e+09</td>\n",
       "      <td>#22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1236 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           sha source_x  \\\n",
       "0     c630ebcdf30652f0422c3ec12a00b50241dc9bd9      CZI   \n",
       "1     53eccda7977a31e3d0f565c884da036b1e85438e      CZI   \n",
       "2     210a892deb1c61577f6fba58505fd65356ce6636      CZI   \n",
       "3     e3b40cc8e0e137c416b4a2273a4dca94ae8178cc      CZI   \n",
       "4     92c2c9839304b4f2bc1276d41b1aa885d8b364fd      CZI   \n",
       "5     0df0d5270a9399cf4e23c0cdd877a80616a9725e      CZI   \n",
       "6     f24242580be243d5fc3f432915d86af6854bb8b7      CZI   \n",
       "7     d13a685f861b0f1ba05afa6e005311ad1820fd3a      CZI   \n",
       "8     e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1      CZI   \n",
       "9     e9239100c5493ea914dc23c3d7a262f4326022ac      CZI   \n",
       "10    469ed0f00c09e2637351c9735c306f27acf3aace      CZI   \n",
       "11    4e550e034ccca6fa2a91e481ddba24db67bc9ae5      CZI   \n",
       "12    4bbb0c59babc718f67953fae032dad6ae0d7aeb1      CZI   \n",
       "13    c821803c55c6aad89b6d0c1d3ba252051e464017      CZI   \n",
       "14    c3bee2a4caca614b34f92c17b643b854dcdab28d      CZI   \n",
       "15    715842fa536064980818ad7e31ce511272a4b6bc      CZI   \n",
       "16    601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f      CZI   \n",
       "17    06c89f69aa7b5f9648d2c1543b8246fe9c3610cf      CZI   \n",
       "18    acb678bdd7634055de18d0b89bb6a4890e6a0306      CZI   \n",
       "19    47772f3e98d8c61bb9782531a0338ba85f27bf3f      CZI   \n",
       "20    544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5      CZI   \n",
       "21    4ff89a71126d2932544a8337ba28787fde5f02a8      CZI   \n",
       "22    ac37ba61c91bb6939a507dbf4efe2119ddafeb9c      CZI   \n",
       "23    6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0      CZI   \n",
       "24    c9fee561c2a3834645dbb61dc4ae6448051da492      CZI   \n",
       "25    655537fc8cc52bccf43cf7189ab060d3097caa7a      CZI   \n",
       "26    bb7cd48a85b72f8d73e57d655c6d8a034fce3778      CZI   \n",
       "27    3c42431c5b95e707486ce7441ac139071e07b706      CZI   \n",
       "28    6a93283b499ae5bc6aaf29f14e701dc8f25138ea      CZI   \n",
       "29    6fc52ed878c271020a2a375bb6e4b12943a7666c      CZI   \n",
       "...                                        ...      ...   \n",
       "1206                                       NaN      CZI   \n",
       "1207                                       NaN      CZI   \n",
       "1208                                       NaN      CZI   \n",
       "1209                                       NaN      CZI   \n",
       "1210                                       NaN      CZI   \n",
       "1211                                       NaN      CZI   \n",
       "1212                                       NaN      CZI   \n",
       "1213                                       NaN      CZI   \n",
       "1214                                       NaN      CZI   \n",
       "1215                                       NaN      CZI   \n",
       "1216                                       NaN      CZI   \n",
       "1217                                       NaN      CZI   \n",
       "1218                                       NaN      CZI   \n",
       "1219                                       NaN      CZI   \n",
       "1220                                       NaN      CZI   \n",
       "1221                                       NaN      CZI   \n",
       "1222                                       NaN      CZI   \n",
       "1223                                       NaN      CZI   \n",
       "1224                                       NaN      CZI   \n",
       "1225                                       NaN      CZI   \n",
       "1226                                       NaN      CZI   \n",
       "1227                                       NaN      CZI   \n",
       "1228                                       NaN      CZI   \n",
       "1229                                       NaN      CZI   \n",
       "1230                                       NaN      CZI   \n",
       "1231                                       NaN      CZI   \n",
       "1232                                       NaN      CZI   \n",
       "1233                                       NaN      CZI   \n",
       "1234                                       NaN      CZI   \n",
       "1235                                       NaN      CZI   \n",
       "\n",
       "                                                  title  \\\n",
       "0     Angiotensin-converting enzyme 2 (ACE2) as a SA...   \n",
       "1     Comparative genetic analysis of the novel coro...   \n",
       "2     Incubation Period and Other Epidemiological Ch...   \n",
       "3     Characteristics of and Public Health Responses...   \n",
       "4          Imaging changes in severe COVID-19 pneumonia   \n",
       "5     An updated estimation of the risk of transmiss...   \n",
       "6     Real-time forecasts of the 2019-nCoV epidemic ...   \n",
       "7     RETRACTED: Chinese medical staff request inter...   \n",
       "8     COVID-19 outbreak on the Diamond Princess crui...   \n",
       "9     Distinct Roles for Sialoside and Protein Recep...   \n",
       "10    First two months of the 2019 Coronavirus Disea...   \n",
       "11    Effectiveness of airport screening at detectin...   \n",
       "12    Genome Detective Coronavirus Typing Tool for r...   \n",
       "13    Case of the Index Patient Who Caused Tertiary ...   \n",
       "14    Emergence of Novel Coronavirus 2019-nCoV: Need...   \n",
       "15    The coronavirus 2019-nCoV epidemic: Is hindsig...   \n",
       "16    Nonstructural proteins NS7b and NS8 are likely...   \n",
       "17    Pathogenicity and Transmissibility of 2019-nCo...   \n",
       "18    Remdesivir and chloroquine effectively inhibit...   \n",
       "19    Potential for global spread of a novel coronav...   \n",
       "20    Severe acute respiratory symptoms and suspecte...   \n",
       "21    Statistics-Based Predictions of Coronavirus Ep...   \n",
       "22    Early transmission patterns of coronavirus dis...   \n",
       "23    HRCT Imaging Features in Representative Import...   \n",
       "24    Comprehensive Genomic Characterization Analysi...   \n",
       "25    Potential Factors Influencing Repeated SARS Ou...   \n",
       "26    Puzzle of highly pathogenic human coronaviruse...   \n",
       "27    RETRACTION: Retraction-Chinese medical staff r...   \n",
       "28    Critical care response to a hospital outbreak ...   \n",
       "29    Effectiveness for the Response to COVID-19: Th...   \n",
       "...                                                 ...   \n",
       "1206  Daily briefing: World’s biggest physics meetin...   \n",
       "1207  How Coronaviruses Cause Infection—from Colds t...   \n",
       "1208  Advices on the prevention and control of nosoc...   \n",
       "1209  ‘This beast is moving very fast.’ Will the new...   \n",
       "1210  Novo Coronavírus: atendimento a pessoas com su...   \n",
       "1211  Novo coronavirus (2019 nCov) Medidas de preven...   \n",
       "1212  The prevention and control of a new coronaviru...   \n",
       "1213  COVID-19: another infectious disease emerging ...   \n",
       "1214  Dynamic changes of chest CT imaging in patient...   \n",
       "1215  Novel coronavirus COVID-19: an overview for em...   \n",
       "1216  Coronavirus 9-nCoV: recomendaciones para aerop...   \n",
       "1217  Novel Coronavirus Disease 2019 (COVID-19): An ...   \n",
       "1218  Coronavirus just caused the American Physical ...   \n",
       "1219  Mensajero de la Salud: nuevo coronavirus 2019-...   \n",
       "1220  The management of biosafety risk in clinical l...   \n",
       "1221  Lineamiento estandarizado para la Vigilancia E...   \n",
       "1222  Singapore claims first use of antibody test to...   \n",
       "1223  The network investigation on knowledge, attitu...   \n",
       "1224  Investigation on demands for antenatal care se...   \n",
       "1225  Analysis of early chest high resolution CT ima...   \n",
       "1226                            Novo Coronavirus (nCoV)   \n",
       "1227  Difficulties and strategies of public hospital...   \n",
       "1228                                           Get Real   \n",
       "1229  How Ophthalmologists Should Understand and Res...   \n",
       "1230  Experts proposal and frequently asked question...   \n",
       "1231  Surgical treatment strategy for digestive syst...   \n",
       "1232  Novel coronavirus pneumonia in the primary gen...   \n",
       "1233  A Novel Coronavirus Outbreak from Wuhan City i...   \n",
       "1234  History is repeating itself: Probable zoonotic...   \n",
       "1235  Epidemiological and Clinical Characteristics o...   \n",
       "\n",
       "                                         doi pmcid   pubmed_id      license  \\\n",
       "0                 10.1007/s00134-020-05985-9   NaN  32125455.0     cc-by-nc   \n",
       "1                  10.1038/s41421-020-0147-1   NaN         NaN        cc-by   \n",
       "2                         10.3390/jcm9020538   NaN         NaN        cc-by   \n",
       "3                         10.3390/jcm9020575   NaN  32093211.0        cc-by   \n",
       "4                 10.1007/s00134-020-05976-w   NaN  32125453.0     cc-by-nc   \n",
       "5                  10.1016/j.idm.2020.02.001   NaN         NaN  cc-by-nc-nd   \n",
       "6                  10.1016/j.idm.2020.02.002   NaN         NaN  cc-by-nc-nd   \n",
       "7              10.1016/s2214-109x(20)30065-6   NaN  32105614.0        cc-by   \n",
       "8                        10.1093/jtm/taaa030   NaN         NaN     cc-by-nc   \n",
       "9                      10.1128/mBio.02764-19   NaN         NaN        cc-by   \n",
       "10                10.1186/s41256-020-00137-4   NaN         NaN        cc-by   \n",
       "11    10.2807/1560-7917.ES.2020.25.5.2000080   NaN         NaN        cc-by   \n",
       "12            10.1093/bioinformatics/btaa145   NaN  32108862.0     cc-by-nc   \n",
       "13                  10.3346/jkms.2020.35.e79   NaN         NaN     cc-by-nc   \n",
       "14                  10.3390/pathogens9020148   NaN         NaN        cc-by   \n",
       "15              10.1016/j.eclinm.2020.100289   NaN         NaN  cc-by-nc-nd   \n",
       "16              10.1016/j.meegid.2020.104272   NaN         NaN  cc-by-nc-nd   \n",
       "17              10.1016/j.micinf.2020.01.004   NaN         NaN  cc-by-nc-nd   \n",
       "18                 10.1038/s41422-020-0282-0   NaN         NaN        cc-by   \n",
       "19                       10.1093/jtm/taaa011   NaN         NaN     cc-by-nc   \n",
       "20                       10.12809/hkmj208375   NaN         NaN  cc-by-nc-nd   \n",
       "21              10.20535/ibb.2020.4.1.195074   NaN         NaN        cc-by   \n",
       "22    10.2807/1560-7917.ES.2020.25.8.2000097   NaN         NaN        cc-by   \n",
       "23                    10.1093/pcmedi/pbaa004   NaN         NaN     cc-by-nc   \n",
       "24                  10.3389/fmicb.2019.03036   NaN         NaN        cc-by   \n",
       "25                    10.3390/ijerph17051633   NaN         NaN        cc-by   \n",
       "26                10.1007/s13238-020-00693-y   NaN  32088858.0        cc-by   \n",
       "27             10.1016/s2214-109x(20)30076-0   NaN  32113504.0        cc-by   \n",
       "28                 10.1186/s13054-020-2786-x   NaN         NaN        cc-by   \n",
       "29              10.24171/j.phrp.2020.11.1.01   NaN         NaN  cc-by-nc-nd   \n",
       "...                                      ...   ...         ...          ...   \n",
       "1206                                     NaN   NaN         NaN          NaN   \n",
       "1207                                     NaN   NaN         NaN          NaN   \n",
       "1208                                     NaN   NaN         NaN          NaN   \n",
       "1209                                     NaN   NaN         NaN          NaN   \n",
       "1210                                     NaN   NaN         NaN          NaN   \n",
       "1211                                     NaN   NaN         NaN          NaN   \n",
       "1212                                     NaN   NaN  32057210.0          NaN   \n",
       "1213                                     NaN   NaN  32078596.0          NaN   \n",
       "1214                                     NaN   NaN  32096366.0          NaN   \n",
       "1215                                     NaN   NaN  32105049.0          NaN   \n",
       "1216                                     NaN   NaN         NaN          NaN   \n",
       "1217                                     NaN   NaN         NaN          NaN   \n",
       "1218                                     NaN   NaN         NaN          NaN   \n",
       "1219                                     NaN   NaN         NaN          NaN   \n",
       "1220                                     NaN   NaN         NaN          NaN   \n",
       "1221                                     NaN   NaN         NaN          NaN   \n",
       "1222                                     NaN   NaN         NaN          NaN   \n",
       "1223                                     NaN   NaN         NaN          NaN   \n",
       "1224                                     NaN   NaN         NaN          NaN   \n",
       "1225                                     NaN   NaN         NaN          NaN   \n",
       "1226                                     NaN   NaN         NaN          NaN   \n",
       "1227                                     NaN   NaN         NaN          NaN   \n",
       "1228                                     NaN   NaN         NaN          NaN   \n",
       "1229                                     NaN   NaN         NaN          NaN   \n",
       "1230                                     NaN   NaN         NaN          NaN   \n",
       "1231                                     NaN   NaN         NaN          NaN   \n",
       "1232                                     NaN   NaN         NaN          NaN   \n",
       "1233                                     NaN   NaN         NaN          NaN   \n",
       "1234                                     NaN   NaN         NaN          NaN   \n",
       "1235                                     NaN   NaN         NaN          NaN   \n",
       "\n",
       "                                               abstract publish_time  \\\n",
       "0                                                   NaN         2020   \n",
       "1                                                   NaN         2020   \n",
       "2     The geographic spread of 2019 novel coronaviru...         2020   \n",
       "3     In December 2019, cases of unidentified pneumo...         2020   \n",
       "4                                                   NaN         2020   \n",
       "5     The basic reproduction number of an infectious...         2020   \n",
       "6     The initial cluster of severe pneumonia cases ...         2020   \n",
       "7                                                   NaN         2020   \n",
       "8     Cruise ships carry a large number of people in...         2020   \n",
       "9     Coronaviruses (CoVs) are common human and anim...         2020   \n",
       "10    Similar to outbreaks of many other infectious ...         2020   \n",
       "11    We simulated 100 2019-nCoV infected travellers...         2020   \n",
       "12    SUMMARY: Genome Detective is a web-based, user...         2020   \n",
       "13    Since mid-December of 2019, coronavirus diseas...         2020   \n",
       "14    Novel Coronavirus (2019-nCoV) is an emerging p...         2020   \n",
       "15                                                  NaN         2020   \n",
       "16    The seventh novel human infecting Betacoronavi...         2020   \n",
       "17    A zoonotic coronavirus, labeled as 2019-nCoV b...         2020   \n",
       "18                                                  NaN         2020   \n",
       "19                                                  NaN         2020   \n",
       "20                                                  NaN         2020   \n",
       "21    Information about the open-access article 'Sta...         2020   \n",
       "22                                                  NaN         2020   \n",
       "23    With the spread of novel coronavirus (2019-nCo...         2020   \n",
       "24    Porcine delta coronavirus (PDCoV) is a novel e...         2020   \n",
       "25    Within last 17 years two widespread epidemics ...         2020   \n",
       "26                                                  NaN         2020   \n",
       "27                                                  NaN         2020   \n",
       "28                                                  NaN         2020   \n",
       "29    In the current issue of Osong Public Health an...         2020   \n",
       "...                                                 ...          ...   \n",
       "1206  The March Meeting of the APS was cancelled jus...         2020   \n",
       "1207  The novel coronavirus outbreak raises question...         2020   \n",
       "1208  The pneumonia caused by the novel coronavirus ...         2020   \n",
       "1209  The silver lining of the epidemic is that scie...         2020   \n",
       "1210                                                NaN         2020   \n",
       "1211                                                NaN         2020   \n",
       "1212  During a short period of time, the outbreak of...         2020   \n",
       "1213                                                NaN         2020   \n",
       "1214  OBJECTIVE: To analyze the dynamic changes of c...         2020   \n",
       "1215  Prior to the global outbreak of SARS-CoV in 20...         2020   \n",
       "1216  Ante la situación mundial, en relación a 2019-...         2020   \n",
       "1217  Background: At the beginning of the New Year 2...         2020   \n",
       "1218  Citing the growing threat of the coronavirus, ...         2020   \n",
       "1219  Declaratoria de Emergencia en Salud Pública de...         2020   \n",
       "1220  During the outbreak of coronavirus disease-19 ...         2020   \n",
       "1221  Este documento describe la situación epidemiol...         2020   \n",
       "1222  In what appears to be a first, disease tracker...         2020   \n",
       "1223  Objective To analyze the current situation of ...         2020   \n",
       "1224  Objective To identify problems and demands for...         2020   \n",
       "1225  Objective To investigate the first chest HRCT ...         2020   \n",
       "1226  Os Coronavírus (CoV) compõem uma grande famíli...         2020   \n",
       "1227  Outbreak of the novel coronavirus pneumonia (N...         2020   \n",
       "1228  Since this is going to be a post about the cor...         2020   \n",
       "1229  The new coronavirus pneumonia that first appea...         2020   \n",
       "1230  The outbreak of novel coronavirus pneumonia (N...         2020   \n",
       "1231  The outbreak of novel coronavirus pneumonia oc...         2020   \n",
       "1232  To observe the curative effect of TCM syndrome...         2020   \n",
       "1233                                                NaN         2020   \n",
       "1234                                                NaN         2020   \n",
       "1235                                                NaN          NaN   \n",
       "\n",
       "                                                authors  \\\n",
       "0     Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...   \n",
       "1     Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...   \n",
       "2     Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...   \n",
       "3                      Deng, Sheng-Qun; Peng, Hong-Juan   \n",
       "4                                            Zhang, Wei   \n",
       "5     Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...   \n",
       "6     Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...   \n",
       "7                             Zeng, Yingchun; Zhen, Yan   \n",
       "8             Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.   \n",
       "9     Qing, Enya; Hantak, Michael; Perlman, Stanley;...   \n",
       "10                              Chen, Xinguang; Yu, Bin   \n",
       "11    Quilty, Billy J; Clifford, Sam; group2, CMMID ...   \n",
       "12    Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...   \n",
       "13    Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...   \n",
       "14    Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...   \n",
       "15    Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...   \n",
       "16      Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro   \n",
       "17                                       Chen, Jieliang   \n",
       "18    Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...   \n",
       "19    Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...   \n",
       "20                          Hon, K. L.; Leung, K. K. Y.   \n",
       "21                                       Nesteruk, Igor   \n",
       "22    Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...   \n",
       "23    Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...   \n",
       "24    Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...   \n",
       "25    Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....   \n",
       "26                                Li, Jing; Liu, Wenjun   \n",
       "27             The Editors Of The Lancet Global, Health   \n",
       "28                 Liu, Yong; Li, Jinxiu; Feng, Yongwen   \n",
       "29                                         Cho, Hae-Wol   \n",
       "...                                                 ...   \n",
       "1206                                      Graham, Flora   \n",
       "1207                                       Makin, Simon   \n",
       "1208  XU, Hongzhen; CHEN, Shuohui; FU, Junfen; SHU, ...   \n",
       "1209                      Cohen, Kai Kupferschmidt; Jon   \n",
       "1210                       Brasil. Ministerio da, Saude   \n",
       "1211                     São Paulo Secretaria da, Saúde   \n",
       "1212                        LI, Zhi yong; MENG, Liu yan   \n",
       "1213                Murdoch, David R.; French, Nigel P.   \n",
       "1214  Wang, Jincheng; Liu, Jinpeng; Wang, Yuanyuan; ...   \n",
       "1215                                Giwa, A.; Desai, A.   \n",
       "1216                    Argentina. Ministerio de, Salud   \n",
       "1217  Keshavarz, Ahmad Tavakoli; Katayon, Vahdat; Mo...   \n",
       "1218                                        Cho, Adrian   \n",
       "1219                    Mexico. Secretaria de la, Salud   \n",
       "1220  XIAO, Yuling; LU, Xiaojun; KANG, Mei; LI, Dong...   \n",
       "1221  Mexico. Secretaria de Salud. Subsecretaria de ...   \n",
       "1222                                    Normile, Dennis   \n",
       "1223                                          CHEN, Yan   \n",
       "1224  DU, Li; GU, Yibin; CUI, Mengqing; LI, Wenxian;...   \n",
       "1225  LIU, Haifeng; ZHANG, Dongyou; YANG, Yi; LONG, ...   \n",
       "1226  Rio de Janeiro . Secretaria de Estado de Saúde...   \n",
       "1227  XU, Dong; HU, Yu; DING, Ning; XIA, Jiahong; ZH...   \n",
       "1228                                        Lowe, Derek   \n",
       "1229                                         ZHIJIE, Li   \n",
       "1230  ZHANG, Lei; CAO, Qing; WANG, Ying; LU, Quan; H...   \n",
       "1231                MA, Fuhai; HU, Haitao; TIAN, Yantao   \n",
       "1232       Liao, Rongye; Yang, Jie; Cao, Zhi; Wang, Jun   \n",
       "1233                           Alavi-Moghaddam, Mostafa   \n",
       "1234  Rodriguez-Morales, A. J.; Bonilla-Aldana, D. K...   \n",
       "1235  Chen, Nanshan and Zhou, Min and Dong, Xuan and...   \n",
       "\n",
       "                                                journal  \\\n",
       "0                                    Intensive Care Med   \n",
       "1                                        Cell Discovery   \n",
       "2                          Journal of Clinical Medicine   \n",
       "3                                            J Clin Med   \n",
       "4                                    Intensive Care Med   \n",
       "5                          Infectious Disease Modelling   \n",
       "6                          Infectious Disease Modelling   \n",
       "7                             The Lancet. Global health   \n",
       "8                            Journal of Travel Medicine   \n",
       "9                                                  mBio   \n",
       "10                    Global Health Research and Policy   \n",
       "11                                     Eurosurveillance   \n",
       "12                     Bioinformatics (Oxford, England)   \n",
       "13                    Journal of Korean Medical Science   \n",
       "14                                            Pathogens   \n",
       "15                                    EClinicalMedicine   \n",
       "16                    Infection, Genetics and Evolution   \n",
       "17                               Microbes and Infection   \n",
       "18                                        Cell Research   \n",
       "19                           Journal of travel medicine   \n",
       "20                            Hong Kong Medical Journal   \n",
       "21             Innovative Biosystems and Bioengineering   \n",
       "22                                     Eurosurveillance   \n",
       "23                          Precision Clinical Medicine   \n",
       "24                            Frontiers in Microbiology   \n",
       "25    International Journal of Environmental Researc...   \n",
       "26                                         Protein Cell   \n",
       "27                            The Lancet. Global health   \n",
       "28                                        Critical Care   \n",
       "29        Osong Public Health and Research Perspectives   \n",
       "...                                                 ...   \n",
       "1206                                             Nature   \n",
       "1207                                                NaN   \n",
       "1208         Chinese Journal of Hospital Administration   \n",
       "1209                                            Science   \n",
       "1210                                                NaN   \n",
       "1211                                                NaN   \n",
       "1212                     Chinese Journal of Stomatology   \n",
       "1213                                          N Z Med J   \n",
       "1214                 Zhejiang Da Xue Xue Bao Yi Xue Ban   \n",
       "1215                        Emergency medicine practice   \n",
       "1216                                                NaN   \n",
       "1217                      Iranian South Medical Journal   \n",
       "1218                                   Science Magazine   \n",
       "1219                                                NaN   \n",
       "1220             Chinese Journal of Laboratory Medicine   \n",
       "1221                                                NaN   \n",
       "1222                                   Science Magazine   \n",
       "1223             Chinese Journal of Preventive Medicine   \n",
       "1224       Chinese Journal of Obstetrics and Gynecology   \n",
       "1225                       Chinese Journal of Radiology   \n",
       "1226                            Nota Técnica-SVS/SES-RJ   \n",
       "1227         Chinese Journal of Hospital Administration   \n",
       "1228                                            Science   \n",
       "1229      Chinese Journal of Experimental Ophthalmology   \n",
       "1230     Chinese Journal of Applied Clinical Pediatrics   \n",
       "1231                        Chinese Journal of Oncology   \n",
       "1232  International Journal of Traditional Chinese M...   \n",
       "1233            Archives of Academic Emergency Medicine   \n",
       "1234                              Infezioni in Medicina   \n",
       "1235                                                NaN   \n",
       "\n",
       "      Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "0                    2.002765e+09          #3252          True  \n",
       "1                    3.003431e+09          #1861          True  \n",
       "2                    3.006065e+09          #1043          True  \n",
       "3                    1.776631e+08          #1999          True  \n",
       "4                    3.006643e+09          #3242         False  \n",
       "5                    3.006029e+09           #729          True  \n",
       "6                    3.006029e+09           #865          True  \n",
       "7                    2.627046e+09          #5386         False  \n",
       "8                    3.006304e+09          #2926          True  \n",
       "9                    3.005811e+09          #2427          True  \n",
       "10                   3.006646e+09          #5595          True  \n",
       "11                   3.005152e+09           #682          True  \n",
       "12                   3.003549e+09          #2776          True  \n",
       "13                   3.005657e+09          #3770          True  \n",
       "14                   3.003886e+09          #1502          True  \n",
       "15                   3.004913e+09          #3333          True  \n",
       "16                   2.286137e+09          #4581          True  \n",
       "17                            NaN           #236          True  \n",
       "18                   3.005213e+09           #205          True  \n",
       "19                   3.004048e+09           #102          True  \n",
       "20                   2.029408e+09          #3395          True  \n",
       "21                   3.006305e+09          #1218          True  \n",
       "22                   3.005657e+09          #2582          True  \n",
       "23                   3.006468e+09           #781          True  \n",
       "24                   3.003968e+09          #5462          True  \n",
       "25                   2.615949e+09          #3296          True  \n",
       "26                   2.766462e+09          #1815          True  \n",
       "27                   2.794879e+09          #5486         False  \n",
       "28                   2.537201e+09          #1225          True  \n",
       "29                   3.005888e+09          #2662         False  \n",
       "...                           ...            ...           ...  \n",
       "1206                          NaN          #3172           NaN  \n",
       "1207                          NaN           #294           NaN  \n",
       "1208                          NaN          #2106           NaN  \n",
       "1209                          NaN           #332           NaN  \n",
       "1210                          NaN          #1866           NaN  \n",
       "1211                          NaN           #922           NaN  \n",
       "1212                 2.377782e+09          #1228           NaN  \n",
       "1213                 2.185518e+09          #1541           NaN  \n",
       "1214                 2.349103e+09          #1922           NaN  \n",
       "1215                 3.005414e+09          #2815           NaN  \n",
       "1216                 1.604377e+09           #801           NaN  \n",
       "1217                 6.213808e+08          #4536           NaN  \n",
       "1218                 2.595675e+09          #2772           NaN  \n",
       "1219                 3.003591e+09          #1803           NaN  \n",
       "1220                 2.194450e+09          #2110           NaN  \n",
       "1221                 2.241123e+09          #1802           NaN  \n",
       "1222                 2.855668e+09          #2584           NaN  \n",
       "1223                 2.363411e+09          #2333           NaN  \n",
       "1224                 1.891210e+09          #2305           NaN  \n",
       "1225                 2.418494e+09          #2187           NaN  \n",
       "1226                 2.518955e+09           #216           NaN  \n",
       "1227                 2.348898e+09          #2108           NaN  \n",
       "1228                 2.343732e+09          #3808           NaN  \n",
       "1229                 3.005943e+09          #2051           NaN  \n",
       "1230                 2.416639e+09          #2063           NaN  \n",
       "1231                 2.366820e+09          #2176           NaN  \n",
       "1232                 2.616653e+09          #1962           NaN  \n",
       "1233                 3.006237e+09          #3509           NaN  \n",
       "1234                 3.005681e+09           #926           NaN  \n",
       "1235                 3.004581e+09            #22           NaN  \n",
       "\n",
       "[1236 rows x 14 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['source_x']=='CZI']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scrape papers from PMC\n",
    "Journals: 16593"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [],
   "source": [
    "papers_df = df_meta[(df_meta['source_x']=='PMC')&(df_meta['sha'].notnull())]\n",
    "papers_array = papers_df[['sha','doi','pmcid','publish_time','journal']].to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884\n"
     ]
    }
   ],
   "source": [
    "pmc_base_url = 'https://www.ncbi.nlm.nih.gov/pmc/articles/'\n",
    "for row in papers_array[:1]:\n",
    "    paper = {}\n",
    "    paper['id'] = row[0]\n",
    "    paper['doi'] = row[1]\n",
    "    paper['PMCID'] = row[2]\n",
    "    paper['url'] = pmc_base_url + paper['PMCID']\n",
    "    paper['date'] = row[3]\n",
    "    paper['journal'] = row[4]\n",
    "    print(paper['url'])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884'"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper['url']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "html = session.get(paper['url'])\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
       "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n",
       "\n",
       "<html lang=\"en\" xml:lang=\"en\" xmlns=\"http://www.w3.org/1999/xhtml\">\n",
       "<head><meta content=\"text/html; charset=utf-8\" http-equiv=\"Content-Type\"/>\n",
       "<!-- AppResources meta begin -->\n",
       "<script type=\"text/javascript\">var ncbi_startTime = new Date();</script>\n",
       "<!-- AppResources meta end -->\n",
       "<!-- TemplateResources meta begin -->\n",
       "<meta content=\"\" name=\"paf_template\"/>\n",
       "<!-- TemplateResources meta end -->\n",
       "<!-- Logger begin -->\n",
       "<meta content=\"pmc\" name=\"ncbi_db\"/><meta content=\"article\" name=\"ncbi_pdid\"/><meta content=\"\" name=\"ncbi_acc\"/><meta content=\"plosbiol\" name=\"ncbi_domain\"/><meta content=\"record\" name=\"ncbi_report\"/><meta content=\"fulltext\" name=\"ncbi_type\"/><meta content=\"\" name=\"ncbi_objectid\"/><meta content=\"/articles/PMC1054884/\" name=\"ncbi_pcid\"/><meta content=\"pmc\" name=\"ncbi_app\"/>\n",
       "<!-- Logger end -->\n",
       "<title>Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection</title>\n",
       "<!-- AppResources external_resources begin -->\n",
       "<link href=\"/core/jig/1.14.8/css/jig.min.css\" rel=\"stylesheet\"/><script src=\"/core/jig/1.14.8/js/jig.min.js\" type=\"text/javascript\"></script>\n",
       "<!-- AppResources external_resources end -->\n",
       "<!-- Page meta begin -->\n",
       "<meta content=\"INDEX,NOFOLLOW,NOARCHIVE\" name=\"robots\"/><link href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884/\" rel=\"canonical\"/><link href=\"http://purl.org/DC/elements/1.0/\" rel=\"schema.DC\"/><meta content=\"PLoS Biology\" name=\"citation_journal_title\"/><meta content=\"Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\" name=\"citation_title\"/><meta content=\"Remy Froissart, Denis Roze, Marilyne Uzest, Lionel Galibert, Stephane Blanc, Yannis Michalakis\" name=\"citation_authors\"/><meta content=\"March 2005\" name=\"citation_date\"/><meta content=\"3\" name=\"citation_issue\"/><meta content=\"3\" name=\"citation_volume\"/><meta content=\"10.1371/journal.pbio.0030089\" name=\"citation_doi\"/><meta content=\"/pmc/articles/PMC1054884/?report=abstract\" name=\"citation_abstract_html_url\"/><meta content=\"15737066\" name=\"citation_pmid\"/><meta content=\"Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\" name=\"DC.Title\"/><meta content=\"Text\" name=\"DC.Type\"/><meta content=\"Public Library of Science\" name=\"DC.Publisher\"/><meta content=\"Remy Froissart\" name=\"DC.Contributor\"/><meta content=\"Denis Roze\" name=\"DC.Contributor\"/><meta content=\"Marilyne Uzest\" name=\"DC.Contributor\"/><meta content=\"Lionel Galibert\" name=\"DC.Contributor\"/><meta content=\"Stephane Blanc\" name=\"DC.Contributor\"/><meta content=\"Yannis Michalakis\" name=\"DC.Contributor\"/><meta content=\"2005 Mar\" name=\"DC.Date\"/><meta content=\"10.1371/journal.pbio.0030089\" name=\"DC.Identifier\"/><meta content=\"en\" name=\"DC.Language\"/><meta content=\"Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\" property=\"og:title\"/><meta content=\"article\" property=\"og:type\"/><meta content=\"Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections ...\" property=\"og:description\"/><meta content=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884/\" property=\"og:url\"/><meta content=\"PubMed Central (PMC)\" property=\"og:site_name\"/><meta content=\"https://www.ncbi.nlm.nih.gov/corehtml/pmc/pmcgifs/pmc-logo-share.png\" property=\"og:image\"/><meta content=\"summary\" name=\"twitter:card\"/><meta content=\"@ncbi\" name=\"twitter:site\"/><link href=\"/corehtml/pmc/css/3.14.1/pmc.min.css\" rel=\"stylesheet\" type=\"text/css\"/><link href=\"/corehtml/pmc/css/3.14.1/pmc_extras_prnt.min.css\" media=\"print\" rel=\"stylesheet\" type=\"text/css\"/><script src=\"/corehtml/pmc/js/common.min.js\" type=\"text/javascript\">//</script><script src=\"/corehtml/pmc/js/NcbiTagServer.min.js\" type=\"text/javascript\">//</script><meta content=\"backend:'https://api.ncbi.nlm.nih.gov/lit/ctxp/v1/pmc/'\" name=\"citationexporter\"/><script src=\"https://www.ncbi.nlm.nih.gov/corehtml/pmc/ctxp/jquery.citationexporter.min.js\" type=\"text/javascript\">//</script><link href=\"https://www.ncbi.nlm.nih.gov/corehtml/pmc/ctxp/citationexporter.css\" rel=\"stylesheet\" type=\"text/css\"/><script src=\"/core/mathjax/2.6.1/MathJax.js?config=/corehtml/pmc/js/mathjax-config-classic.3.4.js\" type=\"text/javascript\"></script><script type=\"text/javascript\">window.name=\"mainwindow\";</script><style type=\"text/css\">.pmc-wm {background:transparent repeat-y top left;background-image:url(/corehtml/pmc/pmcgifs/wm-plosbiol.gif);background-size: auto, contain}</style><style type=\"text/css\">.print-view{display:block}</style><style type=\"text/css\">\n",
       "        div.pmc_para_cit li.highlight,\n",
       "        div.pmc_para_cit li.highlight .one_line_source\n",
       "        { background: #E0E0E0; }\n",
       "        a.bibr.highlight { background: #E0E0E0; } \n",
       "      </style><meta content=\"\" name=\"cited_in_systematic_reviews\"/><link href=\"/pmc/articles/PMC1054884/epub/\" rel=\"alternate\" type=\"application/epub+zip\"/><link href=\"/pmc/articles/PMC1054884/pdf/pbio.0030089.pdf\" rel=\"alternate\" type=\"application/pdf\"/>\n",
       "<!-- Page meta end -->\n",
       "<link href=\"//www.ncbi.nlm.nih.gov/favicon.ico\" rel=\"shortcut icon\"/><meta content=\"8A1B0C78E7A1C97100000000031B031B.m_8\" name=\"ncbi_phid\"/>\n",
       "<meta content=\"origin-when-cross-origin\" name=\"referrer\"/><link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/css/3852956/3985586/3808861/4121862/3974050/3917732/251717/4048120/3846471/14534/45193/4113719/3849091/3984811/3751656/4033350/3840896/3577051/3852958/3881636/3579733/4062871/12930/3964959/3855473/4047625/3854974/3854955/4076335/4128070/9685/3549676/3609192/3609193/3609213/3395586/4143404.css\" rel=\"stylesheet\" type=\"text/css\"/><link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/css/3411343/3882866/4157116.css\" media=\"print\" rel=\"stylesheet\" type=\"text/css\"/></head>\n",
       "<body class=\"article\">\n",
       "<div class=\"grid\">\n",
       "<div class=\"col twelve_col nomargin shadow\">\n",
       "<!-- System messages like service outage or JS required; this is handled by the TemplateResources portlet -->\n",
       "<div class=\"sysmessages\">\n",
       "<noscript>\n",
       "<p class=\"nojs\">\n",
       "<strong>Warning:</strong>\n",
       "\tThe NCBI web site requires JavaScript to function. \n",
       "\t<a href=\"/guide/browsers/#enablejs\" target=\"_blank\" title=\"Learn how to enable JavaScript\">more...</a>\n",
       "</p>\n",
       "</noscript>\n",
       "</div>\n",
       "<!--/.sysmessage-->\n",
       "<div class=\"wrap\">\n",
       "<div class=\"page\">\n",
       "<div class=\"top\">\n",
       "<div class=\"universal_header\" id=\"universal_header\"><ul class=\"inline_list jig-ncbimenu ui-ncbimenu resources_list\" id=\"navcontent\"><li class=\"ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt\"><a accesskey=\"1\" class=\"ui-ncbimenu-link-first\" href=\"/\" id=\"ncbihome\" role=\"banner\" title=\"NCBI Home\"><span class=\"offscreen_noflow\">NCBI</span><img alt=\"NCBI Logo\" class=\"ncbi_logo\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/28977\" title=\"NCBI\"/></a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"3\" href=\"#maincontent\" tabindex=\"0\" title=\"Skip to the content\">Skip to main\n",
       "                        content</a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"4\" href=\"#navcontent\" tabindex=\"0\" title=\"Skip to the navigation\">Skip to\n",
       "                        navigation</a></li><li class=\"topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset\" id=\"resource-menu\"><a class=\"ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown\" href=\"/static/header_footer_ajax/submenu/#resources\">Resources</a></li><li class=\"topmenu ui-helper-reset ui-ncbimenu-item-first\" id=\"all-howtos-menu\"><a class=\"ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown\" href=\"/static/header_footer_ajax/submenu/#howto\">How To</a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"0\" href=\"/guide/browsers/#accesskeys\" tabindex=\"0\" title=\"About My NCBI Accesskeys\">About NCBI Accesskeys</a></li></ul><div class=\"myncbi\"><span id=\"myncbiusername\" style=\"display:none\"><a href=\"/account/settings/\" id=\"mnu\" title=\"Edit account settings\"></a></span><a accesskey=\"2\" href=\"/myncbi/\" id=\"myncbi\" style=\"display:none\">My NCBI</a><a href=\"/account/\" id=\"sign_in\">Sign in to NCBI</a><a href=\"/account/signout/\" id=\"sign_out\" style=\"display:none\">Sign Out</a></div></div>\n",
       "<div class=\"header\">\n",
       "<div class=\"res_logo\">\n",
       "<h1 class=\"img_logo\"><a class=\"pmc_logo offscreen\" href=\"/pmc/\">PMC</a></h1>\n",
       "<div class=\"NLMLogo\">\n",
       "<a href=\"https://www.nlm.nih.gov/\" title=\"US National Library of Medicine\">US National Library of Medicine</a>\n",
       "<br/>\n",
       "<a href=\"https://www.nih.gov/\" title=\"National Institutes of Health\">National Institutes of Health</a>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"search\"><form action=\"/pmc/\" method=\"get\"><div class=\"search_form\"><label class=\"offscreen_noflow\" for=\"database\">Search database</label><select id=\"database\"><optgroup label=\"Recent\"><option class=\"last\" data-ac_dict=\"pmc-search-autocomplete\" selected=\"selected\" value=\"pmc\">PMC</option></optgroup><optgroup label=\"All\"><option value=\"gquery\">All Databases</option><option value=\"assembly\">Assembly</option><option value=\"biocollections\">Biocollections</option><option value=\"bioproject\">BioProject</option><option value=\"biosample\">BioSample</option><option value=\"biosystems\">BioSystems</option><option value=\"books\">Books</option><option value=\"clinvar\">ClinVar</option><option value=\"cdd\">Conserved Domains</option><option value=\"gap\">dbGaP</option><option value=\"dbvar\">dbVar</option><option value=\"gene\">Gene</option><option value=\"genome\">Genome</option><option value=\"gds\">GEO DataSets</option><option value=\"geoprofiles\">GEO Profiles</option><option value=\"gtr\">GTR</option><option value=\"homologene\">HomoloGene</option><option value=\"ipg\">Identical Protein Groups</option><option value=\"medgen\">MedGen</option><option value=\"mesh\">MeSH</option><option value=\"ncbisearch\">NCBI Web Site</option><option value=\"nlmcatalog\">NLM Catalog</option><option value=\"nuccore\">Nucleotide</option><option value=\"omim\">OMIM</option><option data-ac_dict=\"pmc-search-autocomplete\" value=\"pmc\">PMC</option><option value=\"popset\">PopSet</option><option value=\"probe\">Probe</option><option value=\"protein\">Protein</option><option value=\"proteinclusters\">Protein Clusters</option><option value=\"pcassay\">PubChem BioAssay</option><option value=\"pccompound\">PubChem Compound</option><option value=\"pcsubstance\">PubChem Substance</option><option value=\"pubmed\">PubMed</option><option value=\"snp\">SNP</option><option value=\"sparcle\">Sparcle</option><option value=\"sra\">SRA</option><option value=\"structure\">Structure</option><option value=\"taxonomy\">Taxonomy</option><option value=\"toolkit\">ToolKit</option><option value=\"toolkitall\">ToolKitAll</option><option value=\"toolkitbookgh\">ToolKitBookgh</option></optgroup></select><div class=\"nowrap\"><label accesskey=\"/\" class=\"offscreen_noflow\" for=\"term\">Search term</label><div class=\"nowrap\"><input autocomplete=\"off\" class=\"jig-ncbiclearbutton jig-ncbiautocomplete\" data-jigconfig=\"dictionary:'pmc-search-autocomplete',disableUrl:'NcbiSearchBarAutoComplCtrl'\" data-sbconfig=\"ds:'no',pjs:'no',afs:'yes'\" id=\"term\" name=\"term\" title=\"Search PMC. Use up and down arrows to choose an item from the autocomplete.\" type=\"text\" value=\"\"/></div><button class=\"button_search nowrap\" cmd=\"go\" id=\"search\" type=\"submit\">Search</button></div></div></form><ul class=\"searchlinks inline_list\"><li>\n",
       "<a href=\"/pmc/advanced/\">Advanced</a>\n",
       "</li><li>\n",
       "<a href=\"/pmc/journals/\">Journal list</a>\n",
       "</li><li class=\"help\">\n",
       "<a href=\"/books/NBK3825/\" target=\"_blank\">Help</a>\n",
       "</li></ul></div>\n",
       "</div>\n",
       "<!--<component id=\"Page\" label=\"headcontent\"/>-->\n",
       "</div>\n",
       "<div class=\"content\">\n",
       "<!-- site messages -->\n",
       "<div class=\"container\">\n",
       "<div class=\"content eight_col col\" id=\"maincontent\">\n",
       "<div class=\"navlink-box\">\n",
       "<ul class=\"page-breadcrumbs inline_list small\"><li class=\"journal-list\"><a class=\"navlink\" href=\"/pmc/journals/\">Journal List</a></li><li class=\"archive\"><a class=\"navlink\" href=\"/pmc/journals/212/\">PLoS Biol</a></li><li class=\"issue-page\"><a class=\"navlink\" href=\"/pmc/issues/115975/\">v.3(3); 2005 Mar</a></li><li class=\"accid\">PMC1054884</li></ul>\n",
       "</div>\n",
       "<!-- Journal banner -->\n",
       "<div class=\"pmc-page-banner whole_rhythm\"><div><img alt=\"Logo of plosbiol\" src=\"/corehtml/pmc/pmcgifs/logo-plosbiol.gif\" usemap=\"#logo-imagemap\"/><map id=\"logo-imagemap\" name=\"logo-imagemap\"><area alt=\"PLoS Biology\" coords=\"0,0,438,55\" href=\"http://www.plosbiology.org/\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"PLoS Biology\"/><area alt=\"Submit to PLoS\" coords=\"106,56,202,74\" href=\"http://www.plosbiology.org/static/checklist.action\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Submit to PLoS\"/><area alt=\"Get E-mail Alerts\" coords=\"204,56,302,74\" href=\"http://journals.plos.org/alerts\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Get E-mail Alerts\"/><area alt=\"Contact Us\" coords=\"304,56,374,74\" href=\"http://www.plosbiology.org/static/contact.action\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Contact Us\"/><area alt=\"Public Library of Science (PLoS)\" coords=\"440,0,500,55\" href=\"http://www.plos.org/\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Public Library of Science (PLoS)\"/><area alt=\"View this Article\" coords=\"0,56,104,74\" href=\"//dx.doi.org/10.1371/journal.pbio.0030089\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Content%20Provider%7CArticle%7CFullText\" shape=\"rect\" target=\"pmc_ext\" title=\"View this Article\"/></map></div> </div>\n",
       "<!--component id='MainPortlet' label='search-reference'/-->\n",
       "<!-- Book content -->\n",
       "<div class=\"\">\n",
       "<div class=\"hide-overflow article lit-style content pmc-wm slang-all page-box\"><!--main-content--><div class=\"jig-ncbiinpagenav\" data-jigconfig=\"smoothScroll: false, allHeadingLevels: ['h2'], headingExclude: ':hidden'\"><div class=\"fm-sec half_rhythm no_top_margin\"><div class=\"fm-citation half_rhythm no_top_margin clearfix\"><div class=\"inline_block eight_col va_top\"><div><div><span class=\"cit\"><span id=\"pmcmata\">PLoS Biol</span>. 2005 Mar; 3(3): e89. </span></div><div><span class=\"fm-vol-iss-date\">Published online 2005 Mar 1. </span> <span class=\"doi\">doi: <a href=\"//dx.doi.org/10.1371%2Fjournal.pbio.0030089\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CCrosslink%7CDOI\" target=\"pmc_ext\">10.1371/journal.pbio.0030089</a></span></div></div></div><div class=\"inline_block four_col va_top show-overflow align_right\"><div class=\"fm-citation-ids\"><div class=\"fm-citation-pmcid\"><span class=\"fm-citation-ids-label\">PMCID: </span><span>PMC1054884</span></div><div class=\"fm-citation-pmid\">PMID: <a href=\"/pubmed/15737066\">15737066</a></div></div></div></div><h1 class=\"content-title\">Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection</h1><div class=\"half_rhythm\"><div class=\"contrib-group fm-author\"><a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550468180560\" href=\"/pubmed/?term=Froissart%20R%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Remy Froissart</a>,<sup>\n",
       "1\n",
       ",</sup><sup>¤</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550464448704\" href=\"/pubmed/?term=Roze%20D%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Denis Roze</a>,<sup>\n",
       "2\n",
       "</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550502491840\" href=\"/pubmed/?term=Uzest%20M%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Marilyne Uzest</a>,<sup>\n",
       "1\n",
       "</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550463591280\" href=\"/pubmed/?term=Galibert%20L%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Lionel Galibert</a>,<sup>\n",
       "1\n",
       "</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550504176336\" href=\"/pubmed/?term=Blanc%20S%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Stephane Blanc</a>,<sup>\n",
       "1\n",
       "</sup> and  <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550504173968\" href=\"/pubmed/?term=Michalakis%20Y%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Yannis Michalakis</a><sup><img alt=\"corresponding author\" src=\"/corehtml/pmc/pmcgifs/corrauth.gif\"/></sup><sup>\n",
       "2\n",
       "</sup></div><div class=\"contrib-group aff-tip\" style=\"display:none\"><div id=\"_co_idm140550468180560\"><h3 class=\"no_margin\">Remy Froissart</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Froissart%20R%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Remy Froissart</a></div></div><div id=\"_co_idm140550464448704\"><h3 class=\"no_margin\">Denis Roze</h3><p><strong>2</strong>Génétique et Evolution des Maladies Infectieuses, Unité Mixte de Recherche Centre National de la Recherche Scientifique (CNRS)–Institut de Recherche pour le Développement (IRD) 2724, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Roze%20D%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Denis Roze</a></div></div><div id=\"_co_idm140550502491840\"><h3 class=\"no_margin\">Marilyne Uzest</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Uzest%20M%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Marilyne Uzest</a></div></div><div id=\"_co_idm140550463591280\"><h3 class=\"no_margin\">Lionel Galibert</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Galibert%20L%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Lionel Galibert</a></div></div><div id=\"_co_idm140550504176336\"><h3 class=\"no_margin\">Stephane Blanc</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Blanc%20S%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Stephane Blanc</a></div></div><div id=\"_co_idm140550504173968\"><h3 class=\"no_margin\">Yannis Michalakis</h3><p><strong>2</strong>Génétique et Evolution des Maladies Infectieuses, Unité Mixte de Recherche Centre National de la Recherche Scientifique (CNRS)–Institut de Recherche pour le Développement (IRD) 2724, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Michalakis%20Y%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Yannis Michalakis</a></div></div></div></div><div class=\"contrib-group half_rhythm fm-editor\">Roger Hull, <span class=\"fm-role\">Academic Editor</span><sup></sup></div><div class=\"fm-panel half_rhythm\"><div class=\"togglers\"><a class=\"pmctoggle\" href=\"#\" rid=\"idm140550461914704_ai idm140550468180688_ai idm140550465846224_ai\">Author information</a> <a class=\"pmctoggle\" href=\"#\" rid=\"idm140550461914704_an\">Article notes</a> <a class=\"pmctoggle\" href=\"#\" rid=\"idm140550461914704_cpl\">Copyright and License information</a> <a href=\"/pmc/about/disclaimer/\">Disclaimer</a></div><div class=\"fm-authors-info fm-panel hide half_rhythm\" id=\"idm140550461914704_ai\" style=\"display:none\"><div class=\"fm-affl\" id=\"aff1\" lang=\"en\"><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</div><div class=\"fm-affl\" id=\"aff2\" lang=\"en\"><strong>2</strong>Génétique et Evolution des Maladies Infectieuses, Unité Mixte de Recherche Centre National de la Recherche Scientifique (CNRS)–Institut de Recherche pour le Développement (IRD) 2724, Montpellier, France</div><div class=\"fm-affl\" id=\"edit1\" lang=\"en\">John Innes Center, United Kingdom</div><div><sup><img alt=\"corresponding author\" src=\"/corehtml/pmc/pmcgifs/corrauth.gif\"/></sup>Corresponding author.</div><div><span class=\"contrib-email\" id=\"idm140550504173968\">Yannis Michalakis: <a class=\"oemail\" data-email=\"rf.dri.lpm@sikalahciM.sinnaY\" href=\"mailto:dev@null\">rf.dri.lpm@sikalahciM.sinnaY</a></span> </div></div><div class=\"fm-article-notes fm-panel hide half_rhythm\" id=\"idm140550461914704_an\" style=\"display:none\"><div class=\"fm-pubdate half_rhythm\">Received 2004 Sep 15; Accepted 2005 Jan 9.</div></div><div class=\"permissions fm-panel half_rhythm hide\" id=\"idm140550461914704_cpl\" style=\"display:none\"><div class=\"fm-copyright half_rhythm\"><a href=\"/pmc/about/copyright/\">Copyright</a> : © 2005 Froissart et al.</div><div class=\"license half_rhythm\">This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.</div></div></div><div class=\"links-box whole_rhythm\" id=\"pmclinksbox\"><div class=\"fm-panel\"><div>See \"<a href=\"/pmc/articles/PMC1054890/\">Recombination as a Way of Life: Viruses Do It Every Day</a>\" , e117.</div><div style=\"border-top: 1px solid rgb(102, 102, 153); margin: 5px 10% 3px;\"></div><div>This article has been <a href=\"/pmc/articles/PMC1054884/citedby/\">cited by</a> other articles in PMC.</div></div></div></div><div class=\"sec\"></div><div class=\"tsec sec\" id=\"idm140550464314896\" lang=\"en\"><h2 class=\"head no_bottom_margin\" id=\"idm140550464314896title\">Abstract</h2><!--article-meta--><div><p class=\"p p-first-last\" id=\"__p1\">Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections occur. While the recombination rate has been recently evaluated in experimentally co-infected cell cultures for several viruses, direct quantification at the most biologically significant level, that of a host infection, is still lacking. This study fills this gap using the cauliflower mosaic virus as a model. We distributed four neutral markers along the viral genome, and co-inoculated host plants with marker-containing and wild-type viruses. The frequency of recombinant genomes was evaluated 21 d post-inoculation. On average, over 50% of viral genomes recovered after a single host infection were recombinants, clearly indicating that recombination is very frequent in this virus. Estimates of the recombination rate show that all regions of the genome are equally affected by this process. Assuming that ten viral replication cycles occurred during our experiment—based on data on the timing of coat protein detection—the per base and replication cycle recombination rate was on the order of 2 × 10<sup>−5</sup> to 4 × 10<sup>−5</sup>. This first determination of a virus recombination rate during a single multi-cellular host infection indicates that recombination is very frequent in the everyday life of this virus.</p></div></div><div class=\"tsec sec\" id=\"s1\"><h2 class=\"head no_bottom_margin\" id=\"s1title\">Introduction</h2><p class=\"p p-first\" id=\"__p3\">As increasing numbers of full-length viral sequences become available, recombinant or mosaic viruses are being recognized more frequently [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b01\" id=\"__tag_577434177\" rid=\"pbio-0030089-b01\">1</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b02\" id=\"__tag_577434170\" rid=\"pbio-0030089-b02\">2</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b03\" id=\"__tag_577434160\" rid=\"pbio-0030089-b03\">3</a>]. Recombination events have been demonstrated to be associated with viruses expanding their host range [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b04\" id=\"__tag_577434146\" rid=\"pbio-0030089-b04\">4</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b05\" id=\"__tag_577434174\" rid=\"pbio-0030089-b05\">5</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b06\" id=\"__tag_577434196\" rid=\"pbio-0030089-b06\">6</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b07\" id=\"__tag_577434152\" rid=\"pbio-0030089-b07\">7</a>] or increasing their virulence [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b08\" id=\"__tag_577434186\" rid=\"pbio-0030089-b08\">8</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b09\" id=\"__tag_577434183\" rid=\"pbio-0030089-b09\">9</a>], thus accompanying, or perhaps even being at the origin of, major changes during virus adaptation. It remains unclear, however, whether recombination events represent a highly frequent and significant phenomenon in the everyday life of these viruses.</p><p id=\"__p4\">Viruses can exchange genetic material when at least two different viral genomes co-infect the same host cell. Progeny can then become hybrid through different mechanisms, such as reassortment of segments when the parental genomes are fragmented [<a class=\"bibr popnode\" href=\"#pbio-0030089-b10\" rid=\"pbio-0030089-b10\">10</a>], intra-molecular recombination when polymerases switch templates (in RNA viruses) [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b11\" id=\"__tag_577434201\" rid=\"pbio-0030089-b11\">11</a>], or homologous or non-homologous recombination (in both RNA and DNA viruses). Quantification of viral recombination in multi-cellular organisms has been attempted under two distinct experimental approaches: in vitro (in cell cultures) [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b12\" id=\"__tag_577434150\" rid=\"pbio-0030089-b12\">12</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b13\" id=\"__tag_577434184\" rid=\"pbio-0030089-b13\">13</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b14\" id=\"__tag_577434151\" rid=\"pbio-0030089-b14\">14</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b15\" id=\"__tag_577434159\" rid=\"pbio-0030089-b15\">15</a>], and in vivo (in live hosts) [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b16\" id=\"__tag_577434172\" rid=\"pbio-0030089-b16\">16</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b17\" id=\"__tag_577434162\" rid=\"pbio-0030089-b17\">17</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b18\" id=\"__tag_577434188\" rid=\"pbio-0030089-b18\">18</a>]. The in vitro approach, which has so far been applied only to animal viruses, allows the establishment of the “intrinsic” recombination rate in experimentally co-infected cells in cell cultures [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b14\" id=\"__tag_577434195\" rid=\"pbio-0030089-b14\">14</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b15\" id=\"__tag_577434164\" rid=\"pbio-0030089-b15\">15</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b19\" id=\"__tag_577434148\" rid=\"pbio-0030089-b19\">19</a>]. However, it does not necessarily reflect the situation in entire, living hosts, where the frequency of co-infected cells is poorly known and depends on many factors such as the size of the pathogen population, the relative frequency and distribution of the different variants, and host defense mechanisms preventing secondary infection of cells. The in vivo experimental approach is closer to biological conditions and may thus be more informative of what actually happens in “the real world.”</p><p id=\"__p5\">However, as discussed below, numerous experimental constraints have so far precluded an actual quantification of the baseline rate of recombination. First, many experimental designs have used extreme positive selection, where only recombinant genomes were viable (e.g., [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b13\" id=\"__tag_577434180\" rid=\"pbio-0030089-b13\">13</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b20\" id=\"__tag_577434202\" rid=\"pbio-0030089-b20\">20</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434147\" rid=\"pbio-0030089-b21\">21</a>]). Other studies did not use complementation techniques but detected recombinants by PCR within infected hosts or tissues [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b18\" id=\"__tag_577434193\" rid=\"pbio-0030089-b18\">18</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b22\" id=\"__tag_577434178\" rid=\"pbio-0030089-b22\">22</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b23\" id=\"__tag_577434191\" rid=\"pbio-0030089-b23\">23</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b24\" id=\"__tag_577434197\" rid=\"pbio-0030089-b24\">24</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b25\" id=\"__tag_577434187\" rid=\"pbio-0030089-b25\">25</a>], which provides information on their presence but not on their frequency in the viral population. So far, no quantitative PCR or other quantitative method has been applied to evaluate the number of recombinants appearing in an experimentally infected live host. Finally, recent methods based on sequence analysis inferred the population recombination rate, rather than the individual recombination rate [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b01\" id=\"__tag_577434176\" rid=\"pbio-0030089-b01\">1</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b26\" id=\"__tag_577434169\" rid=\"pbio-0030089-b26\">26</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b27\" id=\"__tag_577434167\" rid=\"pbio-0030089-b27\">27</a>]. While results from these methods certainly take in vivo recombination into account, there are other caveats: isolates have often been collected in different hosts—sometimes in different geographical regions—and sometimes the selective neutrality of sequence variation on which these estimates are based is not clearly established. Estimates from such studies by essence address the estimation of the recombination rate at a different evolutionary scale.</p><p id=\"__p6\">Taken together, the currently available information indicates that no viral recombination rate has ever been estimated directly at time and space scales corresponding to a single multi-cellular host infection, although this level is most significant for the biology and evolution of viruses. This study intends to fill this gap by evaluating the recombination frequency of the cauliflower mosaic virus (CaMV) during a single passage in one of its host plants (the turnip <em class=\"genus-species\">Brassica rapa</em>).</p><p id=\"__p7\">CaMV is a pararetrovirus, which is a major grouping containing hepadnaviruses (e.g., hepatitis B virus), badnaviruses (e.g., banana streak virus), and caulimoviruses (e.g., CaMV). Pararetroviruses are characterized by a non-segmented double-stranded DNA genome. After entering the host cell nucleus, the viral DNA accumulates as a minichromosome [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b28\" id=\"__tag_577434200\" rid=\"pbio-0030089-b28\">28</a>] whose transcription is ensured by the host RNA polymerase II [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b29\" id=\"__tag_577434185\" rid=\"pbio-0030089-b29\">29</a>]. The CaMV genome consists in approximately 8,000 bp and encodes six viral gene products that have been detected <em>in planta</em> (<a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>) [<a class=\"bibr popnode\" href=\"#pbio-0030089-b30\" rid=\"pbio-0030089-b30\">30</a>]. Viral proteins P1 to P6 are expressed from two major transcripts, namely a 19S RNA, encoding P6, and a 35S RNA corresponding to the entire genome and serving as mRNA for proteins P1–P5 [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b31\" id=\"__tag_577434144\" rid=\"pbio-0030089-b31\">31</a>]. Using the pre-genomic 35S RNA as a matrix, the protein P5 (product of gene V) reverse-transcribes the genome into genomic DNA that is concomitantly encapsidated [<a class=\"bibr popnode\" href=\"#pbio-0030089-b30\" rid=\"pbio-0030089-b30\">30</a>].</p><!--fig ft0--><!--fig mode=article f1--><div class=\"fig iconblock whole_rhythm clearfix\" co-legend-rid=\"lgnd_pbio-0030089-g001\" id=\"pbio-0030089-g001\"><a href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><!--fig/graphic|fig/alternatives/graphic mode=\"anchored\" m1--><div class=\"figure\" data-largeobj=\"\" data-largeobj-link-rid=\"largeobj_idm140550461561296\"><a class=\"inline_block ts_canvas\" href=\"/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=1065690_pbio.0030089.g001.jpg\" target=\"tileshopwindow\"><div class=\"ts_bar small\" title=\"Click on image to zoom\"></div><img alt=\"An external file that holds a picture, illustration, etc.\n",
       "Object name is pbio.0030089.g001.jpg\" class=\"tileshop\" src=\"/pmc/articles/PMC1054884/bin/pbio.0030089.g001.jpg\" title=\"Click on image to zoom\"/></a></div><div class=\"largeobj-link align_right\" id=\"largeobj_idm140550461561296\" style=\"display: none\"><a href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/?report=objectonly\" target=\"object\">Open in a separate window</a></div></a><div class=\"icnblk_cntnt\" id=\"lgnd_pbio-0030089-g001\"><div><a class=\"figpopup\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\">Figure 1</a></div><!--caption a7--><div class=\"caption\"><!--caption a8--><strong>Genetic Map of CaMV</strong><p id=\"__p8\">The CaMV genome is a circular double-stranded DNA of 8,024 bp, represented in the figure by a double line. The thick arrows with different textures represent the organization of open reading frames I to VI, encoding proteins detected <em>in planta</em>. Markers a, b, c, and d were engineered at the positions indicated (see <a class=\"sec\" href=\"#s4\" rid=\"s4\">Materials and Methods</a> for precise positions). The inner black arrows represent monocistronic 19S RNA and polycistronic 35S RNA produced by the cellular machinery. The nucleotide position 0 (numbering according to [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b44\" id=\"__tag_577434194\" rid=\"pbio-0030089-b44\">44</a>]) indicates the origin of replication via reverse transcription, which occurs in the direction indicated by the dotted outermost circle-like arrow. Reverse transcription is accomplished by the viral reverse transcriptase, using the 35S RNA as template [<a class=\"bibr popnode\" href=\"#pbio-0030089-b49\" rid=\"pbio-0030089-b49\">49</a>].</p></div></div></div><p id=\"__p9\">The detection of CaMV recombinants in turnip hosts has been reported numerous times. Some studies have demonstrated the appearance of infectious recombinant viral genomes after inoculation (i) of a host plant with two infectious or non-infectious parental clones [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434158\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode\" href=\"#pbio-0030089-b32\" rid=\"pbio-0030089-b32\">32</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b33\" id=\"__tag_577434161\" rid=\"pbio-0030089-b33\">33</a>,<a class=\"bibr popnode\" href=\"#pbio-0030089-b34\" rid=\"pbio-0030089-b34\">34</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b35\" id=\"__tag_577434198\" rid=\"pbio-0030089-b35\">35</a>] or (ii) of a transgenic plant containing one CaMV transgene with a CaMV genome missing the corresponding genomic region [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434179\" rid=\"pbio-0030089-b36\">36</a>]. While the former revealed inter-genomic viral recombination, the latter demonstrated that CaMV can also recombine with transgenes within the host's genome. Another study based on phylogenetic analyses of various CaMV strains has clearly suggested different origins for different genomic regions and, hence, multiple recombination events during the evolution of this virus [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b37\" id=\"__tag_577434156\" rid=\"pbio-0030089-b37\">37</a>]. Indirect experimental evidence has indicated that, in some cases, CaMV recombination could occur within the host nucleus, between different viral minichromosomes, presumably through the action of the DNA repair cellular machinery [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434190\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b35\" id=\"__tag_577434182\" rid=\"pbio-0030089-b35\">35</a>]. Nevertheless, the mechanism of “template switching” during reverse transcription, predominant in all retroviruses, most certainly also applies to pararetroviruses. For this reason, and on the basis of numerous experimental data, CaMV is generally believed to recombine mostly in the cytoplasm of the host cell, by “legal” template switching between two pre-genomic RNA molecules [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434153\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b35\" id=\"__tag_577434143\" rid=\"pbio-0030089-b35\">35</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434173\" rid=\"pbio-0030089-b36\">36</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b38\" id=\"__tag_577434192\" rid=\"pbio-0030089-b38\">38</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b39\" id=\"__tag_577434171\" rid=\"pbio-0030089-b39\">39</a>], or “illegal” template switching between the 19S and the 35S RNA [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434142\" rid=\"pbio-0030089-b36\">36</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b40\" id=\"__tag_577434140\" rid=\"pbio-0030089-b40\">40</a>]. Under this hypothesis, recombination in CaMV could therefore be considered as operating on a linear template during reverse transcription, with the 5′ and 3′ extremities later ligated to circularize the genomic DNA (position 0 in <a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>). The above cited studies clearly demonstrate that CaMV is able to recombine. However, since these studies are based on complementation techniques, non-quantitative detection, or phylogenetically based inferences of recombination, they do not inform us on whether recombination is an exceptional event or an “everyday” process shaping the genetic composition of CaMV populations.</p><p class=\"p p-last\" id=\"__p10\">In the present work, we aimed at answering this question. To this end, we have constructed a CaMV genome with four genetic markers, demonstrated to be neutral in competition experiments. By co-inoculating host plants with equal amounts of wild-type and marker-containing CaMV particles, we have generated mixed populations in which impressive proportions of recombinants—distributed in several different classes corresponding to exchange of different genomic regions—have been detected and quantified. Altogether, the recombinant genomes averaged over 50% of the population. Further analysis of these data, assuming a number of viral replications during the infection period ranging from five to 20, indicates that the per nucleotide per replication cycle recombination rate of CaMV is of the same order of magnitude, i.e., on the order of a few 10<sup>−5</sup>, across the entire genome. We thereby provide the first quantification, to our knowledge, of the recombination rate in a virus population during a single passage in a single host.</p></div><div class=\"tsec sec\" id=\"s2\"><h2 class=\"head no_bottom_margin\" id=\"s2title\">Results</h2><div class=\"sec sec-first\" id=\"s2a\"><h3 id=\"s2atitle\">Recombinant Frequency in CaMV Populations from Co-Infected Plants</h3><p class=\"p p-first\" id=\"__p11\">From <a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>, and supposing that all marker-containing genomic regions can recombine, we could predict the detection and quantification of seven classes of recombinant genotypes: +bcd/a+++, a+cd/+b++, ab+d/++c+, abc+/+++d, ++cd/ab++, a++d/+bc+, and a+c+/+b+d. Indeed, all classes were detected, and their frequencies in the ten CaMV populations analyzed are summarized in <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/\" rid-figpopup=\"pbio-0030089-t001\" rid-ob=\"ob-pbio-0030089-t001\" target=\"table\"><span>Table 1</span></a>.</p><!--table ft1--><!--table-wrap mode=\"anchored\" t5--><div class=\"table-wrap table anchored whole_rhythm\" id=\"pbio-0030089-t001\"><h3>Table 1</h3><!--caption a7--><div class=\"caption\"><!--caption a8--><strong>Quantification of Recombinant Genomes in CaMV Populations</strong></div><div class=\"figure\" data-largeobj=\"\" data-largeobj-link-rid=\"largeobj_idm140550466509952\"><a class=\"inline_block ts_canvas\" href=\"/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=1065690_pbio.0030089.t001.jpg\" target=\"tileshopwindow\"><div class=\"ts_bar small\" title=\"Click on image to zoom\"></div><img alt=\"An external file that holds a picture, illustration, etc.\n",
       "Object name is pbio.0030089.t001.jpg\" class=\"tileshop\" src=\"/pmc/articles/PMC1054884/bin/pbio.0030089.t001.jpg\" title=\"Click on image to zoom\"/></a></div><div class=\"largeobj-link align_right\" id=\"largeobj_idm140550466509952\" style=\"display: none\"><a href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/?report=objectonly\" target=\"object\">Open in a separate window</a></div><div class=\"tblwrap-foot\"><div id=\"nt101\"><p class=\"p p-first-last\" id=\"__p12\">\n",
       "<sup>a</sup> Viral genomes were cloned and analyzed from ten of 24 co-infected plants</p></div><div id=\"nt102\"><p class=\"p p-first-last\" id=\"__p13\">\n",
       "<sup>b</sup> Seven possible classes of recombinants were predicted, their respective frequency in the population is expressed in percentage</p></div><div id=\"nt103\"><p class=\"p p-first-last\" id=\"__p14\">\n",
       "<sup>c</sup> The proportions of recombinants from the seven classes were added to estimate the total percentage of recombinant genomes within each tested plant</p></div></div></div><p class=\"p p-last\" id=\"__p15\">Altogether, the proportion of recombinant genomes found in the mixed viral populations was astonishingly high and very similar in the ten co-infected plants analyzed (<a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/\" rid-figpopup=\"pbio-0030089-t001\" rid-ob=\"ob-pbio-0030089-t001\" target=\"table\"><span>Table 1</span></a>, last column), ranging between 44% (plant 5) to 60% (plants 7, 12, and 20), with a mean frequency (± standard error) of 53.8% ± 2.0%. This result indicates that recombination events are very frequent during the invasion of the host plant by CaMV and represents, to our knowledge, the first direct quantification of viral recombination during the infection of a live multi-cellular host.</p></div><div class=\"sec sec-last\" id=\"s2b\"><h3 id=\"s2btitle\">Probability of Recombination between Various Pairs of Markers</h3><p class=\"p p-first\" id=\"__p16\">The inferred per generation recombination and interference rates, assuming that CaMV undergoes ten replication cycles during the 21 d between infection and sampling, are given for each of the ten plants in <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Table 2</span></a>. Recombination rates between adjacent markers are large, on the order of 0.05 to 0.1. Taking the distance in nucleotides between markers into account yields an average recombination rate per nucleotide and generation on the order of 4 × 10<sup>−5</sup>. Interestingly, this recombination rate does not vary throughout the genome (Kruskal–Wallis test, <em>p</em> = 0.16).</p><!--table ft1--><!--table-wrap mode=\"anchored\" t5--><div class=\"table-wrap table anchored whole_rhythm\" id=\"pbio-0030089-t002\"><h3>Table 2</h3><!--caption a7--><div class=\"caption\"><!--caption a8--><strong>Recombination Parameters for the Viral Populations Sampled from Ten Infected Plants</strong></div><div class=\"figure\" data-largeobj=\"\" data-largeobj-link-rid=\"largeobj_idm140550465918608\"><a class=\"inline_block ts_canvas\" href=\"/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=1065690_pbio.0030089.t002.jpg\" target=\"tileshopwindow\"><div class=\"ts_bar small\" title=\"Click on image to zoom\"></div><img alt=\"An external file that holds a picture, illustration, etc.\n",
       "Object name is pbio.0030089.t002.jpg\" class=\"tileshop\" src=\"/pmc/articles/PMC1054884/bin/pbio.0030089.t002.jpg\" title=\"Click on image to zoom\"/></a></div><div class=\"largeobj-link align_right\" id=\"largeobj_idm140550465918608\" style=\"display: none\"><a href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/?report=objectonly\" target=\"object\">Open in a separate window</a></div><div class=\"tblwrap-foot\"><div id=\"nt201\"><p class=\"p p-first-last\" id=\"__p17\">The various parameters are as follows: <em>r</em>\n",
       "<sub>1</sub>, recombination rate between markers a and b; <em>r</em>\n",
       "<sub>2</sub>, recombination rate between markers b and c; <em>r</em>\n",
       "<sub>3</sub>, recombination rate between markers c and d; <em>i</em>\n",
       "<sub>12</sub>, interference between crossovers in segments a–b and b–c; <em>i</em>\n",
       "<sub>23</sub>, interference between crossovers in segments b–c and c–d; <em>i</em>\n",
       "<sub>13</sub>, interference between crossovers in segments a–b and c–d; <em>i</em>\n",
       "<sub>123</sub>, second-order interference accounting for residual interference. The recombination rates are the maximum likelihood estimates (± 95% confidence intervals). The interference parameters were obtained numerically as explained in the <a class=\"sec\" href=\"#s4\" rid=\"s4\">Materials and Methods</a>\n",
       "</p></div></div></div><p id=\"__p18\">To relax the assumption of the number of replications during the 21 d, we calculated the recombination parameters assuming five or 20 generations. The effect of the number of generations on the estimates is linear: doubling the number of generations results in a halving of the recombination rate (detailed results not shown). For example, the average recombination rates <em>r</em>\n",
       "<sub>1</sub>, <em>r</em>\n",
       "<sub>2</sub>, and <em>r</em>\n",
       "<sub>3</sub> assuming 20 generations were equal to 0.05, 0.04, and 0.025, respectively (compare with values in <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Table 2</span></a>), yielding per nucleotide per generation recombination rates of 1.9 × 10<sup>−5</sup>, 2.2 × 10<sup>−5</sup> and 1.6 × 10<sup>−5</sup>.</p><p class=\"p p-last\" id=\"__p19\">Inspection of <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Table 2</span></a> also shows that first-order interference coefficients were in general negative, indicating that a crossing over in one genomic segment increases the probability that a crossing over will occur in another genomic segment, while the second-order coefficient parameter had an average value close to zero with a large variance. The mechanism leading to these results will be discussed in the following section.</p></div></div><div class=\"tsec sec\" id=\"s3\"><h2 class=\"head no_bottom_margin\" id=\"s3title\">Discussion</h2><p class=\"p p-first\" id=\"__p20\">One major breakthrough in the work presented here lies in the space and time scales at which the experiments were performed. Indeed, the processes occurring within the course of a single infection of one multi-cellular host are of obvious biological relevance for any disease. Previous studies on viral recombination suffered from major drawbacks in this respect, basing their conclusions on experiments relying on complementation among non-infectious viruses or between viruses with undetermined relative fitness, on phylogenetically based analyses, or on experiments in cell cultures. For reasons detailed in the Introduction, the first two methods either do not provide information on the frequency of recombination, but only its occurrence, or address the question at a different temporal, and often spatial, scale. Results from cell cultures, on the other hand, impose cell co-infection by different viral variants, potentially overestimating the frequency of recombination events. Our study circumvents these limitations by analyzing viral genotypes sampled from infected plants after the course of a single infection, and therefore the invasion and co-infection of cells in various organs and tissues is very close to natural.</p><p id=\"__p21\">More than half of the genomes (53.8% ± 2.0%; see <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/\" rid-figpopup=\"pbio-0030089-t001\" rid-ob=\"ob-pbio-0030089-t001\" target=\"table\"><span>Table 1</span></a>) present in a CaMV population after a single passage in its host plant were identified as recombinants, and these data allowed us to infer a per nucleotide per generation recombination rate on the order of 2 × 10<sup>−5</sup> to 4 × 10<sup>−5</sup>. The time length of one generation, i.e., the time required for a given genome to go from one replication to the next, is totally unknown in plant viruses. The only experimental data available on CaMV are based on the kinetics of gene expression in infected protoplasts, where the capsid protein is produced between 48 and 72 h [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b40\" id=\"__tag_577434145\" rid=\"pbio-0030089-b40\">40</a>]. The reverse transcription and the encapsidation of genomic DNA being two coupled phenomena [<a class=\"bibr popnode\" href=\"#pbio-0030089-b30\" rid=\"pbio-0030089-b30\">30</a>], we judged it reasonable to assume a generation time of 2 d and, thus, an average of ten generations during our experiments. In case this estimate is mistaken, we have verified a linear relationship between <em>r</em> and the number of generations, thereby allowing an immediate adjustment of <em>r</em> if the CaMV generation time is more precisely established. At this point, we must consider that all cloned genomes may not have been through all the successive replication events potentially allowed by the timing of our experiments. It was previously shown that about 95% of CaMV mature virus particles accumulate in compact inclusion bodies [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b41\" id=\"__tag_577434168\" rid=\"pbio-0030089-b41\">41</a>], where they may be sequestered for a long time, as such inclusions are very frequent in all infected cells, including those in leaves that have been invaded by the virus population for several weeks. The viral population may thus present an age structure that could bias the estimation of the recombination rate. In order to minimize this bias, the clones we analyzed were collected in one young newly formed leaf, where the chances of finding genomes from “unsequestered lines” were assumed to be higher. In any case, our data analysis is conservative, since this age structure can only lead to an underestimation of the recombination rate.</p><p id=\"__p22\">Our results show that interferences between pairs of loci are negative: a recombination event between two loci apparently increases the probability of recombination between another pair of loci. We believe that the most parsimonious explanation of these negative interferences is based on the way the infection builds up within plant hosts. Indeed, one can divide infected host cells into those infected by a single virus genotype and those infected by more than one viral genotype. In the former, analogous to clonal propagation, recombination is undetectable. In the latter, recombination is not only detectable but, as our results indicate, very frequent. Samples consisting of viruses resulting from a mixture of these two types of host cell infections will thus contain viruses with no recombination and viruses with several recombination events, thus yielding an impression of negative interference. These conceptual arguments are supported by mathematical models. It is indeed easy to show (detailed results not shown) that if a proportion <em>F</em> of the population reproduces clonally, analogous to single infections, while the remaining reproduces panmictically, negative interferences could be inferred even if they do not exist. For example, assuming a three-locus model with real recombination rates <em>r</em>\n",
       "<sub>1</sub> and <em>r</em>\n",
       "<sub>2</sub> and interference <em>i</em>\n",
       "<sub>12</sub>, the “apparent” recombination and interference parameters, would be <em>r</em>\n",
       "<sub>1</sub> = (1 − <em>F</em>)r<sub>1</sub>, <em>r</em>\n",
       "<sub>2</sub> = (1 − <em>F</em>)<em>r</em>\n",
       "<sub>2</sub>, and <em>i</em>\n",
       "<sub>12</sub> = −(<em>F</em> − <em>i</em>\n",
       "<sub>12</sub>)/(1 − <em>F</em>). Interestingly, this example also shows that our estimates of the recombination rate are conservative: that a fraction <em>F</em> of host cells are singly infected while others are multiply infected leads to an underestimation of the recombination rate.</p><p id=\"__p23\">As judged by <em>r</em>\n",
       "<sub>1</sub>, <em>r</em>\n",
       "<sub>2</sub>, and <em>r</em>\n",
       "<sub>3</sub>, calculated between markers a–b, b–c, and c–d, respectively, we found evidence for recombination through the entire CaMV genome. The values for <em>r</em>\n",
       "<sub>1,</sub>\n",
       "<em>r</em>\n",
       "<sub>2</sub>, and <em>r</em>\n",
       "<sub>3</sub> are remarkably similar, hence the recombination sites seem to be evenly distributed along the genome. We considered the template-switching model as the major way recombinants are created in CaMV. As already mentioned in the Introduction, hot spots of template switching have been predicted at the position of the 5′ extremities of the 35S and 19S RNAs [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434166\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434165\" rid=\"pbio-0030089-b36\">36</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b42\" id=\"__tag_577434175\" rid=\"pbio-0030089-b42\">42</a>]. If other recombination mechanisms, such as that associated with second-strand DNA synthesis or with the host cell DNA repair machinery, act significantly, hot spots would be expected at the positions of the sequence interruption Δ1, Δ2, and Δ3 [<a class=\"bibr popnode\" href=\"#pbio-0030089-b43\" rid=\"pbio-0030089-b43\">43</a>]. Due to the design of our experiment and the position of the four markers, we have no information on putative hot spots at positions corresponding to the 5′ end of the 35S RNA and to Δ1 (at nucleotide position 0). Nevertheless, the putative hot spots at the 5′ end of the 19S RNA and at Δ2 and Δ3 (nucleotide positions 4,220 and 1,635, respectively) fall between marker pairs c–d, b–c, and a–b, respectively. Our results indicate that either these hot spots are quantitatively equivalent—though predicted by different recombination mechanisms—or, more likely, that they simply do not exist. Whatever the explanation, what we observe is that the CaMV can exchange any portion of its genome, and thus any gene thereof, with an astonishingly high frequency during the course of a single host infection.</p><p id=\"__p24\">To our knowledge, the viral recombination rate has never previously been quantified experimentally for a plant virus [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b03\" id=\"__tag_577434199\" rid=\"pbio-0030089-b03\">3</a>]. In contrast, retroviruses and particularly HIV-1 have been extensively investigated in that sense. As we have already discussed for these latter cases, the quantification of the intrinsic recombination rate was carried out in artificially co-infected cell cultures. The estimated intrinsic per nucleotide per generation recombination rate in HIV-1 is on the order of 10<sup>−4</sup> [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b14\" id=\"__tag_577434155\" rid=\"pbio-0030089-b14\">14</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b15\" id=\"__tag_577434163\" rid=\"pbio-0030089-b15\">15</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b19\" id=\"__tag_577434157\" rid=\"pbio-0030089-b19\">19</a>], less than one order of magnitude higher than our estimation for CaMV. Because for various reasons detailed above we probably underestimate the within-host CaMV recombination rate, we believe that the intrinsic recombination rate in CaMV is higher and perhaps on the order of that of HIV.</p><p class=\"p p-last\" id=\"__p25\">Other pararetroviruses such as plant badnaviruses or vertebrate hepadnaviruses have a similar cycle within their host cells, including steps of nuclear minichromosome, genomic size RNA synthesis, and reverse transcription and encapsidation. Nevertheless, vertebrate hepadnaviruses (e.g., hepatitis B virus) infect hosts that are very different from plants in their biology and physiology, and this could lead to a totally different frequency of cell co-infection during the development of the virus populations. Thus, even though our results can be informative for other pararetroviruses because of the viruses' shared biological characteristics, they should not be extrapolated to vertebrate pararetroviruses without caution.</p></div><div class=\"tsec sec\" id=\"s4\"><h2 class=\"head no_bottom_margin\" id=\"s4title\">Materials and Methods</h2><div class=\"sec sec-first-last\" id=\"s4a\"><div class=\"sec sec-first\" id=\"s4a1\"><p></p><h4 class=\"inline\" id=\"s4a1title\">Viral isolates </h4><p class=\"p p-first-last\" id=\"__p26\">We used the plasmid pCa37, which is the complete genome of the CaMV isolate Cabb-S, cloned into the pBR322 plasmid at the unique SalI restriction site [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b44\" id=\"__tag_577434154\" rid=\"pbio-0030089-b44\">44</a>]. To analyze recombination in different regions of the genome, we introduced four genetic markers: a, b, c, and d, at the positions 881, 3,539, 5,365, and 6,943, respectively, thus approximately at four cardinal points of the CaMV circular double-stranded DNA of 8,024 bp (<a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>). All markers, each corresponding to a single nucleotide change, were introduced by PCR-directed mutagenesis in pCa37, and resulted in the duplication of previously unique restriction sites BsiWI, PstI, MluI, and SacI in a plasmid designated pMark-S. Because, in this study, we targeted the possible exchange of genes between viral genomes, all markers a, b, c, and d were introduced within coding regions corresponding to open reading frames I, IV, V, and VI, respectively. Another important concern was to quantify recombination in the absence of selection, i.e., to create neutral markers. Consequently all markers consist of synonymous mutations (see below).</p></div><div class=\"sec\" id=\"s4a2\"><p></p><h4 class=\"inline\" id=\"s4a2title\">Production of viral particles and co-inoculation </h4><p class=\"p p-first\" id=\"__p27\">To generate the parental virus particles, plasmids pCa37 and pMark-S were mechanically inoculated into individual plants as previously described [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b33\" id=\"__tag_577434189\" rid=\"pbio-0030089-b33\">33</a>]. All plants were turnips (<em class=\"genus-species\">B. rapa</em> cv, “Just Right”) grown under glasshouse conditions at 23 °C with a 16/8 (light/dark) photoperiod. Thirty days post-inoculation, all symptomatic leaves were harvested and viral particles were purified as described earlier [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b45\" id=\"__tag_577434149\" rid=\"pbio-0030089-b45\">45</a>].</p><p class=\"p p-last\" id=\"__p28\">The resulting preparations of parental viruses, designated Cabb-S and Mark-S, were quantified by spectrometry using the formula described by Hull et al. [<a class=\"bibr popnode\" href=\"#pbio-0030089-b46\" rid=\"pbio-0030089-b46\">46</a>]. We fixed the initial frequency of markers to a value of 0.5, and a solution containing 0.1 mg/ml of virus particles of both Cabb-S and Mark-S at a 1:1 ratio was prepared. Plantlets were co-infected by mechanical inoculation of two to three leaves with 20 μl of this virus solution, using abrasive Celite AFA (Fluka, Ronkonkoma, New York, United States). The mixed CaMV population was allowed to grow during 21 d of systemic infection.</p></div><div class=\"sec\" id=\"s4a3\"><p></p><h4 class=\"inline\" id=\"s4a3title\">Estimation of marker frequency within mixed virus populations </h4><p class=\"p p-first-last\" id=\"__p29\">We designed an experimental protocol for quantifying marker frequency within a mixed Cabb-S/Mark-S virus population after a single passage in a host plant. Twenty-four individual plants, inoculated as above with equal amounts of Cabb-S and Mark-S, were harvested 21 d post-inoculation, when symptoms were fully developed. The viral DNA was purified from 200 mg of young newly formed infected leaves according to the protocol described previously [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b47\" id=\"__tag_577434141\" rid=\"pbio-0030089-b47\">47</a>]. After the precipitation step of this protocol, the viral DNA was resuspended and further purified with the Wizard DNA clean-up kit (Promega, Fitchburg, Wisconsin, United States) in TE 1X (100 mM Tris-HCl and 10 mM EDTA [pH 8]). Aliquots of viral DNA preparations were digested by restriction enzymes corresponding either to marker a, b, c, or d and submitted to a 1% agarose gel electrophoresis, colored by ethydium bromide and exposed to UV. Each individual restriction enzyme cut once in Cabb-S DNA and twice in Mark-S, thus generating DNA fragments of different sizes attributable to one or the other in the mixed population of CaMV genomes. After scanning the agarose gels, we estimated the relative frequency of the two genotypes in each viral DNA preparation and at each marker position, by densitometry using the NIH 1.62 Image program. The statistical analyses of the frequency of the four markers are described below.</p></div><div class=\"sec\" id=\"s4a4\"><p></p><h4 class=\"inline\" id=\"s4a4title\">Isolation of individual CaMV genomes and identification of recombinants </h4><p class=\"p p-first-last\" id=\"__p30\">To identify and quantify the recombinants within the CaMV mixed populations, aliquots from ten of the 24 viral DNA preparations described above were digested by the restriction enzyme SalI, and directly cloned into pUC19 at the corresponding site. In each of the ten viral populations analyzed, 50 full-genome-length clones were digested separately by BsiWI, PstI, MluI, and SacI, to test for the presence of marker a, b, c, and d, respectively. In this experiment, with the marker representing an additional restriction site, we could easily distinguish between the Cabb-S and the Mark-S genotype at all four marker positions, upon agarose gel (1%) electrophoresis of the digested clones. Clones with none or all four markers were parental genotypes, whereas clones harboring 1, 2, or 3 markers were clearly recombinants. Due to the very high number of recombinants detected, markers eventually appearing or disappearing due to spontaneous mutations were neglected.</p></div><div class=\"sec sec-last\" id=\"s4a5\"><p></p><h4 class=\"inline\" id=\"s4a5title\">Statistical analysis </h4><p class=\"p p-first\" id=\"__p31\">Here we present the different methods we used to quantify recombination in the CaMV genome. Because all these methods assume that the different markers are neutral, we first discuss assumption.</p><p id=\"__p32\">We used two datasets to test the neutrality of markers, both resulting from plants co-infected with a 1:1 ratio of Mark-S and Cabb-S. The first consisted of viral DNA densitometry data derived from 24 plants (described above), where for each plant we have an estimate of the frequency of each marker in the genome population. The second consisted of the restriction of 50 individual full-genome-length viral clones obtained from one co-infected plant (described above), yielding an estimate of the frequency of each marker, and this was repeated on ten different plants. The frequencies of the different markers were 0.508, 0.501, 0.516, and 0.507 for markers a, b, c, and d in the first dataset and 0.521, 0.518, 0.514, and 0.524 in the second dataset. We tested whether these frequencies were significantly different from the expected value under neutrality, 0.5, using either <em>t</em>-tests, for datasets where normality could not be rejected (seven out of eight cases), or Wilcoxon signed-rank non-parametric tests otherwise (marker c in the first dataset). In all cases <em>p</em>-values were larger than 0.05.</p><p id=\"__p33\">There are several cautionary remarks regarding these analyses. First, in all cases we found an excess of markers. Unfortunately, the two datasets cannot be regarded as independent because, even though the methods through which the frequency estimates were obtained were different, the plants used in the second dataset were a subset of the plants of the first. We thus have only four independent estimates in each case, and there is minimal power to detect significant deviations from neutrality with such a small sample size. It should be noted at this stage that deviations from the expected value could also be caused either by slight deviations from the 1:1 ratio in the infecting mixed solution, or by deviations from that ratio in the frequency of the viral particles that actually get into the plants. Second, because of the relatively small sample sizes and low statistical power, the tests presented above could have detected only large deviations.</p><p id=\"__p34\">The results clearly show, however, precisely that the markers do not have large effects, if any, and that therefore recombination estimates would be affected only very slightly by any hypothetical selective effects of the markers. Because of this, along with the fact that the introduced markers provoke silent substitutions in the CaMV genome, we assumed that markers were effectively neutral in the rest of the analysis.</p><p id=\"__p35\">The dataset used to estimate the recombination frequency consisted of the 500 full-genome-length viral clones (50 from each of ten co-infected plants) individually genotyped for each of the four markers. As discussed in detail in the Results, recombination was very frequent and concerned all four markers. Indeed, approximately half of the genotyped clones exhibited a recombinant genotype. It was therefore meaningful to try to obtain quantitative estimates of recombination from our data.</p><p id=\"__p36\">Our aim was to analyze viral recombination in a live host. Consequently, we had to deal with the fact that more than one viral replication cycle occurred during the 21 d that infection lasted in our experiment (we had to wait that long for the disease to develop and to be able to recover sufficient amounts of viral DNA from each infection). Based on the kinetics of gene expression [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b40\" id=\"__tag_577434181\" rid=\"pbio-0030089-b40\">40</a>], we postulate that each replication cycle lasts between 2 and 3 d, and that therefore seven to ten cycles occurred between infection and the sampling time. In case this assumption is incorrect, we did calculations assuming five, seven, ten, or 20 replication cycles during these 21 d. As shown, the results were not affected qualitatively, and only slightly quantitatively. It is important to note that we assumed that recombination occurred through a template-switching mechanism, and that therefore, from a recombination point of view, the CaMV genome is linear. The reverse transcription starts and finishes at the position 0 in <a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>, which is the point of circularization of the DNA genome. This implies that changes between contiguous markers a–b, b–c, and c–d can be considered as true recombination whereas those between a and d cannot, as they may simply stem from circularization of DNA, during the synthesis of which the polymerase has switched template once anywhere between a–b, b–c, or c–d.</p><p id=\"__p37\">To estimate the recombination rate between markers, we wrote recurrence equations describing the change in frequency of each genotype over one generation, assuming random mating and no selection (i.e., the standard Wright–Fisher population genetics model). We then expressed the frequency of all possible genotypes <em>n</em> generations later as a function of their initial frequency and of the recombination parameters. Subsequently we calculated the maximum likelihood estimates of the recombination parameters and their asymptotic variances given initial frequencies (we assumed that the two “parental” genotypes, Mark-S and Cabb-S, had equal initial frequencies of 0.5 and that all other genotypes had initial frequencies of zero) and frequencies after <em>n</em> generations (the observed frequencies; as stated above we used different values of <em>n</em>). All algebraic and numerical calculations were carried out with the software Mathematica.</p><p id=\"__p38\">The recombination parameters are the recombination rates between two adjacent loci, e.g., <em>r</em>\n",
       "<sub>1</sub> for the recombination rate between markers a and b, and the interference coefficients, e.g., <em>i</em>\n",
       "<sub>12</sub> for interference between recombination events in the segments between markers a and b and b and c. To define these parameters we followed Christiansen [<a class=\"bibr popnode\" href=\"#pbio-0030089-b48\" rid=\"pbio-0030089-b48\">48</a>], and in particular the recombination distributions for two, three, and four loci (respectively, <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Tables 2</span></a>.7, 2.8, and 2.9 of [<a class=\"bibr popnode\" href=\"#pbio-0030089-b48\" rid=\"pbio-0030089-b48\">48</a>]). It is important to realize that given the definitions of these parameters, the estimator of the recombination rate between two loci is not affected by the number of loci considered. In other words, we obtain the same estimation of the recombination rate between markers a and b whether we consider genotypic frequencies at just these two loci, or the frequencies at these two loci plus a third locus, or the complete information to which we have access, the four-marker genotypes. Information on additional loci only affects the estimates of the interference coefficients.</p><p class=\"p p-last\" id=\"__p39\">It proved impossible to carry out the calculations for four loci algebraically. Instead, we used a computer program to calculate the expected genotypic frequencies at all four loci after <em>n</em> generations, given the above stated initial frequencies and specified recombination parameters. For each combination of recombination parameters we calculated a Euclidean distance between the vector of the expected genotypic frequencies and the observed genotypic frequencies, and considered that the estimated recombination parameters were those yielding the minimal Euclidean distance. In all cases, the estimated recombination rates between pairs of loci were equal to the second decimal to those estimated algebraically from data for three or two loci.</p></div></div></div><div class=\"tsec sec\" id=\"idm140550462018496\"><h2 class=\"head no_bottom_margin\" id=\"idm140550462018496title\">Acknowledgments</h2><div class=\"sec\"><p id=\"__p40\">\n",
       "<strong>Competing interests.</strong> The authors have declared that no competing interests exist.</p></div></div><div class=\"tsec sec\" id=\"idm140550460948112\"><h2 class=\"head no_bottom_margin\" id=\"idm140550460948112title\">Abbreviation</h2><div class=\"bk-sec\"><div><table class=\"default-table glossary\"><tbody><tr><td>CaMV</td><td>cauliflower mosaic virus</td></tr></tbody></table></div></div></div><div class=\"tsec sec headless whole_rhythm\" id=\"idm140550466165536\"><!--back/fn-group--><div class=\"fm-sec half_rhythm small\"><div><p class=\"p p-first-last\" id=\"__p41\">\n",
       "<strong>Author contributions.</strong> RF, SB, and YM conceived and designed the experiments. RF, MU, LG, and SB performed the experiments. RF, DR, SB, and YM analyzed the data. RF, DR, SB, and YM wrote the paper.</p></div></div></div><div class=\"tsec sec headless whole_rhythm\" id=\"idm140550466163728\"><!--back/fn-group--><div class=\"fm-sec half_rhythm small\"><p class=\"fn sec\" id=\"n2\"><p class=\"p p-first-last\" id=\"__p42\">¤Current address: Institut National de la Santé et de la Recherche Médicale (INSERM), Equipe Ecologie et Evolution des micro-organismes E 0339, Paris, France</p></p><p class=\"fn sec\" id=\"n3\"><p class=\"p p-first-last\" id=\"__p43\">Citation: Froissart R, Roze D, Uzest M, Galibert L, Blanc S, et al. (2005) Recombination every day: Abundant recombination in a virus during a single multi-cellular host infection. PLoS Biol 3(3): e89.</p></p></div></div><div class=\"tsec sec\" id=\"idm140550502083280\"><h2 class=\"head no_bottom_margin\" id=\"idm140550502083280title\">References</h2><div class=\"ref-list-sec sec\" id=\"reference-list\"><ul class=\"back-ref-list\" style=\"list-style-type:decimal;\"><li id=\"pbio-0030089-b01\"><span class=\"element-citation\">Awadalla P. The evolutionary genomics of pathogen recombination. <span><span class=\"ref-journal\">Nat Rev Genet. </span>2003;<span class=\"ref-vol\">4</span>:50–60.</span> [<a href=\"/pubmed/12509753\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Nat+Rev+Genet&amp;title=The+evolutionary+genomics+of+pathogen+recombination&amp;author=P+Awadalla&amp;volume=4&amp;publication_year=2003&amp;pages=50-60&amp;pmid=12509753&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b02\"><span class=\"element-citation\">Hendrix RW, Hatfull GF, Smith MC. Bacteriophages with tails: Chasing their origins and evolution. <span><span class=\"ref-journal\">Res Microbiol. </span>2003;<span class=\"ref-vol\">154</span>:253–257.</span> [<a href=\"/pubmed/12798229\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Res+Microbiol&amp;title=Bacteriophages+with+tails:+Chasing+their+origins+and+evolution&amp;author=RW+Hendrix&amp;author=GF+Hatfull&amp;author=MC+Smith&amp;volume=154&amp;publication_year=2003&amp;pages=253-257&amp;pmid=12798229&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b03\"><span class=\"element-citation\">Garcia-Arenal F, Fraile A, Malpica JM. Variability and genetic structure of plant virus populations. <span><span class=\"ref-journal\">Annu Rev Phytopathol. </span>2001;<span class=\"ref-vol\">39</span>:157–186.</span> [<a href=\"/pubmed/11701863\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Annu+Rev+Phytopathol&amp;title=Variability+and+genetic+structure+of+plant+virus+populations&amp;author=F+Garcia-Arenal&amp;author=A+Fraile&amp;author=JM+Malpica&amp;volume=39&amp;publication_year=2001&amp;pages=157-186&amp;pmid=11701863&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b04\"><span class=\"element-citation\">Brown DW. Threat to humans from virus infections of non-human primates. <span><span class=\"ref-journal\">Rev Med Virol. </span>1997;<span class=\"ref-vol\">7</span>:239–246.</span> [<a href=\"/pubmed/10398488\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Rev+Med+Virol&amp;title=Threat+to+humans+from+virus+infections+of+non-human+primates&amp;author=DW+Brown&amp;volume=7&amp;publication_year=1997&amp;pages=239-246&amp;pmid=10398488&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b05\"><span class=\"element-citation\">Gibbs MJ, Weiller GF. Evidence that a plant virus switched hosts to infect a vertebrate and then recombined with a vertebrate-infecting virus. <span><span class=\"ref-journal\">Proc Natl Acad Sci U S A. </span>1999;<span class=\"ref-vol\">96</span>:8022–8027.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC22181/\">PMC free article</a>]</span> [<a href=\"/pubmed/10393941\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Proc+Natl+Acad+Sci+U+S+A&amp;title=Evidence+that+a+plant+virus+switched+hosts+to+infect+a+vertebrate+and+then+recombined+with+a+vertebrate-infecting+virus&amp;author=MJ+Gibbs&amp;author=GF+Weiller&amp;volume=96&amp;publication_year=1999&amp;pages=8022-8027&amp;pmid=10393941&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b06\"><span class=\"element-citation\">Malim MH, Emerman M. HIV-1 sequence variation: Drift, shift, and attenuation. <span><span class=\"ref-journal\">Cell. </span>2001;<span class=\"ref-vol\">104</span>:469–472.</span> [<a href=\"/pubmed/11239404\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=HIV-1+sequence+variation:+Drift,+shift,+and+attenuation&amp;author=MH+Malim&amp;author=M+Emerman&amp;volume=104&amp;publication_year=2001&amp;pages=469-472&amp;pmid=11239404&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b07\"><span class=\"element-citation\">Hu WS, Rhodes T, Dang Q, Pathak V. Retroviral recombination: Review of genetic analyses. <span><span class=\"ref-journal\">Front Biosci. </span>2003;<span class=\"ref-vol\">8</span>:D143–D155.</span> [<a href=\"/pubmed/12456341\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Front+Biosci&amp;title=Retroviral+recombination:+Review+of+genetic+analyses&amp;author=WS+Hu&amp;author=T+Rhodes&amp;author=Q+Dang&amp;author=V+Pathak&amp;volume=8&amp;publication_year=2003&amp;pages=D143-D155&amp;pmid=12456341&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b08\"><span class=\"element-citation\">Gibbs MJ, Armstrong JS, Gibbs AJ. Recombination in the hemagglutinin gene of the 1918 “Spanish flu” <span><span class=\"ref-journal\">Science. </span>2001;<span class=\"ref-vol\">293</span>:1842–1845.</span> [<a href=\"/pubmed/11546876\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Science&amp;title=Recombination+in+the+hemagglutinin+gene+of+the+1918+“Spanish+flu”&amp;author=MJ+Gibbs&amp;author=JS+Armstrong&amp;author=AJ+Gibbs&amp;volume=293&amp;publication_year=2001&amp;pages=1842-1845&amp;pmid=11546876&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b09\"><span class=\"element-citation\">Rest JS, Mindell DP. SARS associated coronavirus has a recombinant polymerase and coronaviruses have a history of host-shifting. <span><span class=\"ref-journal\">Infect Genet Evol. </span>2003;<span class=\"ref-vol\">3</span>:219–225.</span> [<a href=\"/pubmed/14522185\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Infect+Genet+Evol&amp;title=SARS+associated+coronavirus+has+a+recombinant+polymerase+and+coronaviruses+have+a+history+of+host-shifting&amp;author=JS+Rest&amp;author=DP+Mindell&amp;volume=3&amp;publication_year=2003&amp;pages=219-225&amp;pmid=14522185&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b10\"><span class=\"element-citation\">Chao L. Evolution of genetic exchange in RNA viruses. In: Morse SS, editor. The evolutionary biology of viruses. New York: Raven Press. pp. <span>1994:233–250.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar?q=Chao+L+Evolution+of+genetic+exchange+in+RNA+viruses.+In:+Morse+SS,+editor.+The+evolutionary+biology+of+viruses.+New+York:+Raven+Press.+pp+1994+233+250+\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b11\"><span class=\"element-citation\">Worobey M, Holmes EC. Evolutionary aspects of recombination in RNA viruses. <span><span class=\"ref-journal\">J Gen Virol. </span>1999;<span class=\"ref-vol\">80</span>:2535–2543.</span> [<a href=\"/pubmed/10573145\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Evolutionary+aspects+of+recombination+in+RNA+viruses&amp;author=M+Worobey&amp;author=EC+Holmes&amp;volume=80&amp;publication_year=1999&amp;pages=2535-2543&amp;pmid=10573145&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b12\"><span class=\"element-citation\">Kirkegaard K, Baltimore D. The mechanism of RNA recombination in poliovirus. <span><span class=\"ref-journal\">Cell. </span>1986;<span class=\"ref-vol\">47</span>:433–443.</span> [<a href=\"/pubmed/3021340\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=The+mechanism+of+RNA+recombination+in+poliovirus&amp;author=K+Kirkegaard&amp;author=D+Baltimore&amp;volume=47&amp;publication_year=1986&amp;pages=433-443&amp;pmid=3021340&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b13\"><span class=\"element-citation\">Clavel F, Hoggan MD, Willey RL, Strebel K, Martin MA, et al.  Genetic recombination of human immunodeficiency virus. <span><span class=\"ref-journal\">J Virol. </span>1989;<span class=\"ref-vol\">63</span>:1455–1459.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC247851/\">PMC free article</a>]</span> [<a href=\"/pubmed/2915387\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Genetic+recombination+of+human+immunodeficiency+virus&amp;author=F+Clavel&amp;author=MD+Hoggan&amp;author=RL+Willey&amp;author=K+Strebel&amp;author=MA+Martin&amp;volume=63&amp;publication_year=1989&amp;pages=1455-1459&amp;pmid=2915387&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b14\"><span class=\"element-citation\">Jetzt AE, Yu H, Klarmann GJ, Ron Y, Preston BD, et al.  High rate of recombination throughout the human immunodeficiency virus type 1 genome. <span><span class=\"ref-journal\">J Virol. </span>2000;<span class=\"ref-vol\">74</span>:1234–1240.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC111457/\">PMC free article</a>]</span> [<a href=\"/pubmed/10627533\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=High+rate+of+recombination+throughout+the+human+immunodeficiency+virus+type+1+genome&amp;author=AE+Jetzt&amp;author=H+Yu&amp;author=GJ+Klarmann&amp;author=Y+Ron&amp;author=BD+Preston&amp;volume=74&amp;publication_year=2000&amp;pages=1234-1240&amp;pmid=10627533&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b15\"><span class=\"element-citation\">Rhodes T, Wargo H, Hu WS. High rates of human immunodeficiency virus type 1 recombination: Near-random segregation of markers one kilobase apart in one round of viral replication. <span><span class=\"ref-journal\">J Virol. </span>2003;<span class=\"ref-vol\">77</span>:11193–11200.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC224990/\">PMC free article</a>]</span> [<a href=\"/pubmed/14512567\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=High+rates+of+human+immunodeficiency+virus+type+1+recombination:+Near-random+segregation+of+markers+one+kilobase+apart+in+one+round+of+viral+replication&amp;author=T+Rhodes&amp;author=H+Wargo&amp;author=WS+Hu&amp;volume=77&amp;publication_year=2003&amp;pages=11193-11200&amp;pmid=14512567&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b16\"><span class=\"element-citation\">Keck JG, Matsushima GK, Makino S, Fleming JO, Vannier DM, et al.  In vivo RNA-RNA recombination of coronavirus in mouse brain. <span><span class=\"ref-journal\">J Virol. </span>1988;<span class=\"ref-vol\">62</span>:1810–1813.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC253235/\">PMC free article</a>]</span> [<a href=\"/pubmed/2833625\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=In+vivo+RNA-RNA+recombination+of+coronavirus+in+mouse+brain&amp;author=JG+Keck&amp;author=GK+Matsushima&amp;author=S+Makino&amp;author=JO+Fleming&amp;author=DM+Vannier&amp;volume=62&amp;publication_year=1988&amp;pages=1810-1813&amp;pmid=2833625&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b17\"><span class=\"element-citation\">Fraile A, Alonso-Prados JL, Aranda MA, Bernal JJ, Malpica JM, et al.  Genetic exchange by recombination or reassortment is infrequent in natural populations of a tripartite RNA plant virus. <span><span class=\"ref-journal\">J Virol. </span>1997;<span class=\"ref-vol\">71</span>:934–940.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC191141/\">PMC free article</a>]</span> [<a href=\"/pubmed/8995610\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Genetic+exchange+by+recombination+or+reassortment+is+infrequent+in+natural+populations+of+a+tripartite+RNA+plant+virus&amp;author=A+Fraile&amp;author=JL+Alonso-Prados&amp;author=MA+Aranda&amp;author=JJ+Bernal&amp;author=JM+Malpica&amp;volume=71&amp;publication_year=1997&amp;pages=934-940&amp;pmid=8995610&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b18\"><span class=\"element-citation\">Bruyere A, Wantroba M, Flasinski S, Dzianott A, Bujarski JJ. Frequent homologous recombination events between molecules of one RNA component in a multipartite RNA virus. <span><span class=\"ref-journal\">J Virol. </span>2000;<span class=\"ref-vol\">74</span>:4214–4219.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC111936/\">PMC free article</a>]</span> [<a href=\"/pubmed/10756034\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Frequent+homologous+recombination+events+between+molecules+of+one+RNA+component+in+a+multipartite+RNA+virus&amp;author=A+Bruyere&amp;author=M+Wantroba&amp;author=S+Flasinski&amp;author=A+Dzianott&amp;author=JJ+Bujarski&amp;volume=74&amp;publication_year=2000&amp;pages=4214-4219&amp;pmid=10756034&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b19\"><span class=\"element-citation\">Zhuang J, Jetzt AE, Sun G, Yu H, Klarmann G, et al.  Human immunodeficiency virus type 1 recombination: Rate, fidelity, and putative hot spots. <span><span class=\"ref-journal\">J Virol. </span>2002;<span class=\"ref-vol\">76</span>:11273–11282.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC136766/\">PMC free article</a>]</span> [<a href=\"/pubmed/12388687\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Human+immunodeficiency+virus+type+1+recombination:+Rate,+fidelity,+and+putative+hot+spots&amp;author=J+Zhuang&amp;author=AE+Jetzt&amp;author=G+Sun&amp;author=H+Yu&amp;author=G+Klarmann&amp;volume=76&amp;publication_year=2002&amp;pages=11273-11282&amp;pmid=12388687&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b20\"><span class=\"element-citation\">Bujarski JJ, Kaesberg P. Genetic recombination between RNA components of a multipartite plant virus. <span><span class=\"ref-journal\">Nature. </span>1986;<span class=\"ref-vol\">321</span>:528–531.</span> [<a href=\"/pubmed/2423879\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genetic+recombination+between+RNA+components+of+a+multipartite+plant+virus&amp;author=JJ+Bujarski&amp;author=P+Kaesberg&amp;volume=321&amp;publication_year=1986&amp;pages=528-531&amp;pmid=2423879&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b21\"><span class=\"element-citation\">Vaden VR, Melcher U. Recombination sites in cauliflower mosaic virus DNAs: Implications for mechanisms of recombination. <span><span class=\"ref-journal\">Virology. </span>1990;<span class=\"ref-vol\">177</span>:717–726.</span> [<a href=\"/pubmed/2371775\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Recombination+sites+in+cauliflower+mosaic+virus+DNAs:+Implications+for+mechanisms+of+recombination&amp;author=VR+Vaden&amp;author=U+Melcher&amp;volume=177&amp;publication_year=1990&amp;pages=717-726&amp;pmid=2371775&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b22\"><span class=\"element-citation\">Banner LR, Lai MM. Random nature of coronavirus RNA recombination in the absence of selection pressure. <span><span class=\"ref-journal\">Virology. </span>1991;<span class=\"ref-vol\">185</span>:441–445.</span> [<a href=\"/pubmed/1656597\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Random+nature+of+coronavirus+RNA+recombination+in+the+absence+of+selection+pressure&amp;author=LR+Banner&amp;author=MM+Lai&amp;volume=185&amp;publication_year=1991&amp;pages=441-445&amp;pmid=1656597&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b23\"><span class=\"element-citation\">Kottier SA, Cavanagh D, Britton P. Experimental evidence of recombination in coronavirus infectious bronchitis virus. <span><span class=\"ref-journal\">Virology. </span>1995;<span class=\"ref-vol\">213</span>:569–580.</span> [<a href=\"/pubmed/7491781\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Experimental+evidence+of+recombination+in+coronavirus+infectious+bronchitis+virus&amp;author=SA+Kottier&amp;author=D+Cavanagh&amp;author=P+Britton&amp;volume=213&amp;publication_year=1995&amp;pages=569-580&amp;pmid=7491781&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b24\"><span class=\"element-citation\">Revers F, Le Gall O, Candresse T, Le Romancer M, Dunez J. Frequent occurrence of recombinant potyvirus isolates. <span><span class=\"ref-journal\">J Gen Virol. </span>1996;<span class=\"ref-vol\">77</span>:1953–1965.</span> [<a href=\"/pubmed/8760448\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Frequent+occurrence+of+recombinant+potyvirus+isolates&amp;author=F+Revers&amp;author=O+Le+Gall&amp;author=T+Candresse&amp;author=M+Le+Romancer&amp;author=J+Dunez&amp;volume=77&amp;publication_year=1996&amp;pages=1953-1965&amp;pmid=8760448&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b25\"><span class=\"element-citation\">Aaziz R, Tepfer M. Recombination between genomic RNAs of two cucumoviruses under conditions of minimal selection pressure. <span><span class=\"ref-journal\">Virology. </span>1999;<span class=\"ref-vol\">263</span>:282–289.</span> [<a href=\"/pubmed/10544102\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Recombination+between+genomic+RNAs+of+two+cucumoviruses+under+conditions+of+minimal+selection+pressure&amp;author=R+Aaziz&amp;author=M+Tepfer&amp;volume=263&amp;publication_year=1999&amp;pages=282-289&amp;pmid=10544102&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b26\"><span class=\"element-citation\">McVean G, Awadalla P, Fearnhead P. A coalescent-based method for detecting and estimating recombination from gene sequences. <span><span class=\"ref-journal\">Genetics. </span>2002;<span class=\"ref-vol\">160</span>:1231–1241.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC1462015/\">PMC free article</a>]</span> [<a href=\"/pubmed/11901136\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Genetics&amp;title=A+coalescent-based+method+for+detecting+and+estimating+recombination+from+gene+sequences&amp;author=G+McVean&amp;author=P+Awadalla&amp;author=P+Fearnhead&amp;volume=160&amp;publication_year=2002&amp;pages=1231-1241&amp;pmid=11901136&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b27\"><span class=\"element-citation\">Stumpf MP, McVean GA. Estimating recombination rates from population-genetic data. <span><span class=\"ref-journal\">Nat Rev Genet. </span>2003;<span class=\"ref-vol\">4</span>:959–968.</span> [<a href=\"/pubmed/14631356\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Nat+Rev+Genet&amp;title=Estimating+recombination+rates+from+population-genetic+data&amp;author=MP+Stumpf&amp;author=GA+McVean&amp;volume=4&amp;publication_year=2003&amp;pages=959-968&amp;pmid=14631356&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b28\"><span class=\"element-citation\">Rothnie HM, Chapdelaine Y, Hohn T. Pararetroviruses and retroviruses: A comparative review of viral structure and gene expression strategies. <span><span class=\"ref-journal\">Adv Virus Res. </span>1994;<span class=\"ref-vol\">44</span>:1–67.</span> [<a href=\"/pubmed/7817872\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Adv+Virus+Res&amp;title=Pararetroviruses+and+retroviruses:+A+comparative+review+of+viral+structure+and+gene+expression+strategies&amp;author=HM+Rothnie&amp;author=Y+Chapdelaine&amp;author=T+Hohn&amp;volume=44&amp;publication_year=1994&amp;pages=1-67&amp;pmid=7817872&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b29\"><span class=\"element-citation\">Mason WS, Taylor JM, Hull R. Retroid virus genome replication. <span><span class=\"ref-journal\">Adv Virus Res. </span>1987;<span class=\"ref-vol\">32</span>:35–96.</span> [<a href=\"/pubmed/2441578\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Adv+Virus+Res&amp;title=Retroid+virus+genome+replication&amp;author=WS+Mason&amp;author=JM+Taylor&amp;author=R+Hull&amp;volume=32&amp;publication_year=1987&amp;pages=35-96&amp;pmid=2441578&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b30\"><span class=\"element-citation\">Hohn T, Fütterer J. The proteins and functions of plant pararetroviruses: Knowns and unknowns. <span><span class=\"ref-journal\">Crit Rev Plant Sci. </span>1997;<span class=\"ref-vol\">16</span>:133–167.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Crit+Rev+Plant+Sci&amp;title=The+proteins+and+functions+of+plant+pararetroviruses:+Knowns+and+unknowns&amp;author=T+Hohn&amp;author=J+Fütterer&amp;volume=16&amp;publication_year=1997&amp;pages=133-167&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b31\"><span class=\"element-citation\">Fütterer J, Hohn T. Translation in plants—Rules and exceptions. <span><span class=\"ref-journal\">Plant Mol Biol. </span>1996;<span class=\"ref-vol\">32</span>:159–189.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC7089402/\">PMC free article</a>]</span> [<a href=\"/pubmed/8980479\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Plant+Mol+Biol&amp;title=Translation+in+plants—Rules+and+exceptions&amp;author=J+Fütterer&amp;author=T+Hohn&amp;volume=32&amp;publication_year=1996&amp;pages=159-189&amp;pmid=8980479&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b32\"><span class=\"element-citation\">Melcher U, Choe IS, Lebeurier G, Richards K, Essenberg RC. Selective allele loss and interference between cauliflower mosaic virus DNAs. <span><span class=\"ref-journal\">Mol Gen Genet. </span>1986;<span class=\"ref-vol\">203</span>:230–236.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Mol+Gen+Genet&amp;title=Selective+allele+loss+and+interference+between+cauliflower+mosaic+virus+DNAs&amp;author=U+Melcher&amp;author=IS+Choe&amp;author=G+Lebeurier&amp;author=K+Richards&amp;author=RC+Essenberg&amp;volume=203&amp;publication_year=1986&amp;pages=230-236&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b33\"><span class=\"element-citation\">Melcher U, Steffens DL, Lyttle DJ, Lebeurier G, Lin H, et al.  Infectious and non-infectious mutants of cauliflower mosaic virus DNA. <span><span class=\"ref-journal\">J Gen Virol. </span>1986;<span class=\"ref-vol\">67</span>:1491–1498.</span> [<a href=\"/pubmed/3723112\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Infectious+and+non-infectious+mutants+of+cauliflower+mosaic+virus+DNA&amp;author=U+Melcher&amp;author=DL+Steffens&amp;author=DJ+Lyttle&amp;author=G+Lebeurier&amp;author=H+Lin&amp;volume=67&amp;publication_year=1986&amp;pages=1491-1498&amp;pmid=3723112&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b34\"><span class=\"element-citation\">Zhang XS, Melcher U. Competition between isolates and variants of cauliflower mosaic virus in infected turnip plants. <span><span class=\"ref-journal\">J Gen Virol. </span>1990;<span class=\"ref-vol\">70</span>:3427–3437.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Competition+between+isolates+and+variants+of+cauliflower+mosaic+virus+in+infected+turnip+plants&amp;author=XS+Zhang&amp;author=U+Melcher&amp;volume=70&amp;publication_year=1990&amp;pages=3427-3437&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b35\"><span class=\"element-citation\">Choe S, Melcher U, Richards K, Lebeurier G, Essenberg RC. Recombination between mutant CaMV DNAs. <span><span class=\"ref-journal\">Plant Mol Biol. </span>1985;<span class=\"ref-vol\">5</span>:281–289.</span> [<a href=\"/pubmed/24306919\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Plant+Mol+Biol&amp;title=Recombination+between+mutant+CaMV+DNAs&amp;author=S+Choe&amp;author=U+Melcher&amp;author=K+Richards&amp;author=G+Lebeurier&amp;author=RC+Essenberg&amp;volume=5&amp;publication_year=1985&amp;pages=281-289&amp;pmid=24306919&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b36\"><span class=\"element-citation\">Gal S, Pisan B, Hohn T, Grimsley N, Hohn B. Agroinfection of transgenic plants leads to viable cauliflower mosaic virus by intermolecular recombination. <span><span class=\"ref-journal\">Virology. </span>1992;<span class=\"ref-vol\">187</span>:525–533.</span> [<a href=\"/pubmed/1546451\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Agroinfection+of+transgenic+plants+leads+to+viable+cauliflower+mosaic+virus+by+intermolecular+recombination&amp;author=S+Gal&amp;author=B+Pisan&amp;author=T+Hohn&amp;author=N+Grimsley&amp;author=B+Hohn&amp;volume=187&amp;publication_year=1992&amp;pages=525-533&amp;pmid=1546451&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b37\"><span class=\"element-citation\">Chenault KD, Melcher U. Phylogenetic relationships reveal recombination among isolates of cauliflower mosaic virus. <span><span class=\"ref-journal\">J Mol Evol. </span>1994;<span class=\"ref-vol\">39</span>:496–505.</span> [<a href=\"/pubmed/7807539\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Mol+Evol&amp;title=Phylogenetic+relationships+reveal+recombination+among+isolates+of+cauliflower+mosaic+virus&amp;author=KD+Chenault&amp;author=U+Melcher&amp;volume=39&amp;publication_year=1994&amp;pages=496-505&amp;pmid=7807539&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b38\"><span class=\"element-citation\">Geldreich A, Lebeurier G, Hirth L. In vivo dimerization of cauliflower mosaic virus DNA can explain recombination. <span><span class=\"ref-journal\">Gene. </span>1986;<span class=\"ref-vol\">48</span>:277–286.</span> [<a href=\"/pubmed/3557131\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Gene&amp;title=In+vivo+dimerization+of+cauliflower+mosaic+virus+DNA+can+explain+recombination&amp;author=A+Geldreich&amp;author=G+Lebeurier&amp;author=L+Hirth&amp;volume=48&amp;publication_year=1986&amp;pages=277-286&amp;pmid=3557131&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b39\"><span class=\"element-citation\">Grimsley N, Hohn T, Hohn B. Recombination in a plant virus: Template-switching in cauliflower mosaic virus. <span><span class=\"ref-journal\">EMBO J. </span>1986;<span class=\"ref-vol\">5</span>:641–646.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC1166838/\">PMC free article</a>]</span> [<a href=\"/pubmed/16453678\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=EMBO+J&amp;title=Recombination+in+a+plant+virus:+Template-switching+in+cauliflower+mosaic+virus&amp;author=N+Grimsley&amp;author=T+Hohn&amp;author=B+Hohn&amp;volume=5&amp;publication_year=1986&amp;pages=641-646&amp;pmid=16453678&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b40\"><span class=\"element-citation\">Kobayashi K, Nakayashiki H, Tsuge S, Mise K, Furusawa I. Accumulation kinetics of viral gene products in cauliflower mosaic virus-infected turnip protoplasts. <span><span class=\"ref-journal\">Microbiol Immunol. </span>1998;<span class=\"ref-vol\">42</span>:65–69.</span> [<a href=\"/pubmed/9525783\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Microbiol+Immunol&amp;title=Accumulation+kinetics+of+viral+gene+products+in+cauliflower+mosaic+virus-infected+turnip+protoplasts&amp;author=K+Kobayashi&amp;author=H+Nakayashiki&amp;author=S+Tsuge&amp;author=K+Mise&amp;author=I+Furusawa&amp;volume=42&amp;publication_year=1998&amp;pages=65-69&amp;pmid=9525783&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b41\"><span class=\"element-citation\">Drucker M, Froissart R, Hebrard E, Uzest M, Ravallec M, et al.  Intracellular distribution of viral gene products regulates a complex mechanism of cauliflower mosaic virus acquisition by its aphid vector. <span><span class=\"ref-journal\">Proc Natl Acad Sci U S A. </span>2002;<span class=\"ref-vol\">99</span>:2422–2427.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC122380/\">PMC free article</a>]</span> [<a href=\"/pubmed/11842201\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Proc+Natl+Acad+Sci+U+S+A&amp;title=Intracellular+distribution+of+viral+gene+products+regulates+a+complex+mechanism+of+cauliflower+mosaic+virus+acquisition+by+its+aphid+vector&amp;author=M+Drucker&amp;author=R+Froissart&amp;author=E+Hebrard&amp;author=M+Uzest&amp;author=M+Ravallec&amp;volume=99&amp;publication_year=2002&amp;pages=2422-2427&amp;pmid=11842201&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b42\"><span class=\"element-citation\">Schoelz JE, Wintermantel WM. Expansion of viral host range through complementation and recombination in transgenic plants. <span><span class=\"ref-journal\">Plant Cell. </span>1993;<span class=\"ref-vol\">5</span>:1669–1679.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC160395/\">PMC free article</a>]</span> [<a href=\"/pubmed/12271051\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Plant+Cell&amp;title=Expansion+of+viral+host+range+through+complementation+and+recombination+in+transgenic+plants&amp;author=JE+Schoelz&amp;author=WM+Wintermantel&amp;volume=5&amp;publication_year=1993&amp;pages=1669-1679&amp;pmid=12271051&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b43\"><span class=\"element-citation\">Hull R.  <span class=\"ref-journal\">Matthews' plant virology, 4th ed.</span> San Diego: Academic Press; 2001. 1,001 pp. <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?title=Matthews'+plant+virology,+4th+ed&amp;author=R+Hull&amp;publication_year=2001&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b44\"><span class=\"element-citation\">Franck A, Guilley H, Jonard G, Richards K, Hirth L. Nucleotide sequence of cauliflower mosaic virus DNA. <span><span class=\"ref-journal\">Cell. </span>1980;<span class=\"ref-vol\">21</span>:285–294.</span> [<a href=\"/pubmed/7407912\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=Nucleotide+sequence+of+cauliflower+mosaic+virus+DNA&amp;author=A+Franck&amp;author=H+Guilley&amp;author=G+Jonard&amp;author=K+Richards&amp;author=L+Hirth&amp;volume=21&amp;publication_year=1980&amp;pages=285-294&amp;pmid=7407912&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b45\"><span class=\"element-citation\">Leh V, Jacquot E, Geldreich A, Hermann T, Leclerc D, et al.  Aphid transmission of cauliflower mosaic virus requires the viral PIII protein. <span><span class=\"ref-journal\">EMBO J. </span>1999;<span class=\"ref-vol\">18</span>:7077–7085.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC1171770/\">PMC free article</a>]</span> [<a href=\"/pubmed/10601029\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=EMBO+J&amp;title=Aphid+transmission+of+cauliflower+mosaic+virus+requires+the+viral+PIII+protein&amp;author=V+Leh&amp;author=E+Jacquot&amp;author=A+Geldreich&amp;author=T+Hermann&amp;author=D+Leclerc&amp;volume=18&amp;publication_year=1999&amp;pages=7077-7085&amp;pmid=10601029&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b46\"><span class=\"element-citation\">Hull R, Shepherd RJ, Harvey JD. Cauliflower mosaic virus: An improved purification procedure and some properties of the virus particule. <span><span class=\"ref-journal\">J Gen Virol. </span>1976;<span class=\"ref-vol\">31</span>:93–100.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Cauliflower+mosaic+virus:+An+improved+purification+procedure+and+some+properties+of+the+virus+particule&amp;author=R+Hull&amp;author=RJ+Shepherd&amp;author=JD+Harvey&amp;volume=31&amp;publication_year=1976&amp;pages=93-100&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b47\"><span class=\"element-citation\">Gardner RC, Shepherd RJ. A procedure for rapid isolation and analysis of cauliflower mosaic virus DNA. <span><span class=\"ref-journal\">Virology. </span>1980;<span class=\"ref-vol\">106</span>:159–161.</span> [<a href=\"/pubmed/18631716\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=A+procedure+for+rapid+isolation+and+analysis+of+cauliflower+mosaic+virus+DNA&amp;author=RC+Gardner&amp;author=RJ+Shepherd&amp;volume=106&amp;publication_year=1980&amp;pages=159-161&amp;pmid=18631716&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b48\"><span class=\"element-citation\">Christiansen FB.  <span class=\"ref-journal\">Population genetics of multiple loci.</span> New York: John Wiley and Sons; 2000. 365 pp. <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?title=Population+genetics+of+multiple+loci&amp;author=FB+Christiansen&amp;publication_year=2000&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b49\"><span class=\"element-citation\">Bonneville JM, Hohn T.  <span class=\"ref-journal\">A reverse transcriptase for cauliflower mosaic virus. The state of the art, 1992. In: Shalka N, Goff S, editors. Reverse transcriptase.</span> Plainview (New York): Cold Spring Harbor Laboratory Press; 1993. pp. 357–390. <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?title=A+reverse+transcriptase+for+cauliflower+mosaic+virus.+The+state+of+the+art,+1992.+In:+Shalka+N,+Goff+S,+editors.+Reverse+transcriptase&amp;author=JM+Bonneville&amp;author=T+Hohn&amp;publication_year=1993&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li></ul></div></div></div><!--post-content--><div class=\"courtesy-note whole_rhythm small\"><hr/>Articles from <span class=\"acknowledgment-journal-title\">PLoS Biology</span> are provided here courtesy of <strong>Public Library of Science</strong></div></div>\n",
       "</div>\n",
       "<!-- Book content -->\n",
       "</div>\n",
       "<div class=\"four_col col last\" id=\"rightcolumn\">\n",
       "<!-- Custom content above discovery portlets -->\n",
       "<div class=\"col6\">\n",
       "</div>\n",
       "<div xmlns:np=\"http://ncbi.gov/portal/XSLT/namespace\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"><div class=\"format-menu\"><h2>Formats:</h2><ul><li class=\"selected\">Article</li> | <li><a href=\"/pmc/articles/PMC1054884/?report=reader\">PubReader</a></li> | <li class=\"epub-link\"><a href=\"/pmc/articles/PMC1054884/epub/\">ePub (beta)</a></li> | <li><a href=\"/pmc/articles/PMC1054884/pdf/pbio.0030089.pdf\">PDF (129K)</a></li> | <li><a class=\"citationexporter ctxp\" data-citationid=\"PMC1054884\" href=\"#\">Citation</a></li></ul></div></div><div class=\"share-buttons\" xmlns:np=\"http://ncbi.gov/portal/XSLT/namespace\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"><h2>Share</h2><ul><li class=\"facebook\"><a href=\"https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F\"><img alt=\"Share on Facebook\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047626\"/>\n",
       "                             Facebook\n",
       "                        </a></li><li class=\"twitter\"><a href=\"https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F&amp;text=Recombination%20Every%20Day%3A%20Abundant%20Recombination%20in%20a%20Virus%20during%20a%20Single%20Multi-Cellular%20Host%20Infection\"><img alt=\"Share on Twitter\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047627\"/>\n",
       "                             Twitter\n",
       "                        </a></li><li class=\"gplus\"><a href=\"https://plus.google.com/share?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F\"><img alt=\"Share on Google Plus\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047628\"/>\n",
       "                             Google+\n",
       "                        </a></li></ul></div>\n",
       "<div data-accid=\"PMC1054884\" data-aid=\"1054884\" data-aiid=\"1065690\" data-domain=\"plosbiol\" data-domainid=\"212\" data-iid=\"115975\" data-md5=\"a9769365bbb44745ffca360502c0eeca\" data-pmid=\"15737066\" id=\"ajax-portlets\"></div>\n",
       "<!-- Custom content below discovery portlets -->\n",
       "<div class=\"col7\">\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<!-- Custom content after all -->\n",
       "<div class=\"col8\">\n",
       "</div>\n",
       "<div class=\"col9\">\n",
       "</div>\n",
       "<script src=\"/corehtml/pmc/js/jquery.scrollTo-1.4.2.js\"></script>\n",
       "<script>\n",
       "    (function($){\n",
       "        $('.skiplink').each(function(i, item){\n",
       "            var href = $($(item).attr('href'));\n",
       "            href.attr('tabindex', '-1').addClass('skiptarget'); // ensure the target can receive focus\n",
       "            $(item).on('click', function(event){\n",
       "                event.preventDefault();\n",
       "                $.scrollTo(href, 0, {\n",
       "                    onAfter: function(){\n",
       "                        href.focus();\n",
       "                    }\n",
       "                });\n",
       "            });\n",
       "        });\n",
       "    })(jQuery);\n",
       "</script>\n",
       "<div id=\"body-link-poppers\"></div>\n",
       "</div>\n",
       "<div class=\"bottom\">\n",
       "<div id=\"NCBIFooter_dynamic\">\n",
       "<a class=\"help_desk\" href=\"\" id=\"help-desk-link\" target=\"_blank\">Support Center</a>\n",
       "<a class=\"help_desk\" href=\"https://support.ncbi.nlm.nih.gov/ics/support/KBList.asp?Time=2020-03-24T11:10:12-04:00&amp;Snapshot=%2Fprojects%2FPMC%2FPMCViewer@4.46&amp;Host=ptpmc102&amp;ncbi_phid=8A1B0C78E7A1C97100000000031B031B&amp;ncbi_session=8A1AB3A7E7A22D31_0927SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F&amp;Db=pmc&amp;folderID=132&amp;Ncbi_App=pmc&amp;Page=literature&amp;style=classic&amp;deptID=28049\" id=\"help-desk-link\" target=\"_blank\">Support Center</a>\n",
       "</div>\n",
       "<div class=\"footer\" id=\"footer\">\n",
       "<div class=\"subfooter\"> </div><script src=\"/portal/portal3rc.fcgi/static/js/preloaderWidget.js\" type=\"text/javascript\"> </script>\n",
       "<div class=\"offscreen_noflow\" id=\"external-disclaimer\">\n",
       "        External link. Please review our <a href=\"https://www.nlm.nih.gov/privacy.html\">privacy policy</a>.\n",
       "    </div>\n",
       "<div class=\"contact_info\" id=\"ncbifooter\">\n",
       "<div id=\"footer-contents-right\">\n",
       "<div id=\"nlm_thumb_logo\">\n",
       "<a href=\"https://www.nlm.nih.gov\" title=\"NLM\">NLM</a>\n",
       "</div>\n",
       "<div id=\"nih_thumb_logo\">\n",
       "<a href=\"https://www.nih.gov\" title=\"NIH\">NIH</a>\n",
       "</div>\n",
       "<div id=\"hhs_thumb_logo\">\n",
       "<a href=\"https://www.hhs.gov\" title=\"DHHS\">DHHS</a>\n",
       "</div>\n",
       "<div id=\"usagov_thumb_logo\">\n",
       "<a href=\"https://www.usa.gov\" title=\"USA.gov\">USA.gov</a>\n",
       "</div>\n",
       "</div>\n",
       "<div id=\"footer-contents-left\">\n",
       "<p class=\"address vcard\">\n",
       "<span class=\"url\">\n",
       "<a class=\"fn url newdomain\" href=\"https://www.ncbi.nlm.nih.gov\">National Center for\n",
       "                        Biotechnology Information</a>,\n",
       "                </span> <span class=\"org url newdomain\"><a href=\"https://www.nlm.nih.gov/\">U.S. National Library of Medicine</a></span>\n",
       "<span class=\"adr\">\n",
       "<span class=\"street-address\">8600 Rockville Pike</span>, <span class=\"locality\">Bethesda</span>\n",
       "<span class=\"region\">MD</span>, <span class=\"postal-code\">20894</span>\n",
       "<span class=\"country-name\">USA</span>\n",
       "</span>\n",
       "</p>\n",
       "<a href=\"/home/about/policies.shtml\">Policies and Guidelines</a> | <a href=\"/home/about/contact.shtml\">Contact</a>\n",
       "</div>\n",
       "</div>\n",
       "<script src=\"/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIConfigJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js?v=1\" type=\"text/javascript\"> </script>\n",
       "<script src=\"/portal/portal3rc.fcgi/static/js/hfjs2.js\" type=\"text/javascript\"> </script>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<!--/.page-->\n",
       "</div>\n",
       "<!--/.wrap-->\n",
       "</div><!-- /.twelve_col -->\n",
       "</div>\n",
       "<!-- /.grid -->\n",
       "<span class=\"PAFAppResources\"></span>\n",
       "<!-- BESelector tab -->\n",
       "<noscript><img alt=\"statistics\" src=\"/stat?jsdisabled=true&amp;ncbi_db=pmc&amp;ncbi_pdid=article&amp;ncbi_acc=&amp;ncbi_domain=plosbiol&amp;ncbi_report=record&amp;ncbi_type=fulltext&amp;ncbi_objectid=&amp;ncbi_pcid=/articles/PMC1054884/&amp;ncbi_app=pmc\"/></noscript>\n",
       "<!-- usually for JS scripts at page bottom -->\n",
       "<!--<component id=\"PageFixtures\" label=\"styles\"></component>-->\n",
       "<!-- 8A1AB3A7E7A22D31_0927SID /projects/PMC/PMCViewer@4.46 ptpmc102 v4.1.r585844 Mon, May 06 2019 02:53:16 -->\n",
       "<script snapshot=\"pmc\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/js/3879255/4121861/3818874/4168176/3821238/4117325/4087685/4072593/4076480/3921943/4105668/4065628.js\" type=\"text/javascript\"></script></body>\n",
       "</html>"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_soup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection'"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# title\n",
    "tmp_soup.find('h1',attrs={'class':'content-title'}).text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Remy Froissart', 'Denis Roze', 'Marilyne Uzest', 'Lionel Galibert', 'Stephane Blanc', 'Yannis Michalakis']\n"
     ]
    }
   ],
   "source": [
    "# authors\n",
    "authors = []\n",
    "for a in tmp_soup.find('div',attrs={'class':'contrib-group fm-author'}).find_all('a'):\n",
    "    authors.append(a.text)\n",
    "print(authors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abstract\n",
      "Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections occur. While the recombination rate has been recently evaluated in experimentally co-infected cell cultures for several viruses, direct quantification at the most biologically significant level, that of a host infection, is still lacking. This study fills this gap using the cauliflower mosaic virus as a model. We distributed four neutral markers along the viral genome, and co-inoculated host plants with marker-containing and wild-type viruses. The frequency of recombinant genomes was evaluated 21 d post-inoculation. On average, over 50% of viral genomes recovered after a single host infection were recombinants, clearly indicating that recombination is very frequent in this virus. Estimates of the recombination rate show that all regions of the genome are equally affected by this process. Assuming that ten viral replication cycles occurred during our experiment—based on data on the timing of coat protein detection—the per base and replication cycle recombination rate was on the order of 2 × 10−5 to 4 × 10−5. This first determination of a virus recombination rate during a single multi-cellular host infection indicates that recombination is very frequent in the everyday life of this virus.\n",
      "\n",
      "Introduction\n",
      "As increasing numbers of full-length viral sequences become available, recombinant or mosaic viruses are being recognized more frequently [1,2,3]. Recombination events have been demonstrated to be associated with viruses expanding their host range [4,5,6,7] or increasing their virulence [8,9], thus accompanying, or perhaps even being at the origin of, major changes during virus adaptation. It remains unclear, however, whether recombination events represent a highly frequent and significant phenomenon in the everyday life of these viruses.\n",
      "Viruses can exchange genetic material when at least two different viral genomes co-infect the same host cell. Progeny can then become hybrid through different mechanisms, such as reassortment of segments when the parental genomes are fragmented [10], intra-molecular recombination when polymerases switch templates (in RNA viruses) [11], or homologous or non-homologous recombination (in both RNA and DNA viruses). Quantification of viral recombination in multi-cellular organisms has been attempted under two distinct experimental approaches: in vitro (in cell cultures) [12,13,14,15], and in vivo (in live hosts) [16,17,18]. The in vitro approach, which has so far been applied only to animal viruses, allows the establishment of the “intrinsic” recombination rate in experimentally co-infected cells in cell cultures [14,15,19]. However, it does not necessarily reflect the situation in entire, living hosts, where the frequency of co-infected cells is poorly known and depends on many factors such as the size of the pathogen population, the relative frequency and distribution of the different variants, and host defense mechanisms preventing secondary infection of cells. The in vivo experimental approach is closer to biological conditions and may thus be more informative of what actually happens in “the real world.”\n",
      "However, as discussed below, numerous experimental constraints have so far precluded an actual quantification of the baseline rate of recombination. First, many experimental designs have used extreme positive selection, where only recombinant genomes were viable (e.g., [13,20,21]). Other studies did not use complementation techniques but detected recombinants by PCR within infected hosts or tissues [18,22,23,24,25], which provides information on their presence but not on their frequency in the viral population. So far, no quantitative PCR or other quantitative method has been applied to evaluate the number of recombinants appearing in an experimentally infected live host. Finally, recent methods based on sequence analysis inferred the population recombination rate, rather than the individual recombination rate [1,26,27]. While results from these methods certainly take in vivo recombination into account, there are other caveats: isolates have often been collected in different hosts—sometimes in different geographical regions—and sometimes the selective neutrality of sequence variation on which these estimates are based is not clearly established. Estimates from such studies by essence address the estimation of the recombination rate at a different evolutionary scale.\n",
      "Taken together, the currently available information indicates that no viral recombination rate has ever been estimated directly at time and space scales corresponding to a single multi-cellular host infection, although this level is most significant for the biology and evolution of viruses. This study intends to fill this gap by evaluating the recombination frequency of the cauliflower mosaic virus (CaMV) during a single passage in one of its host plants (the turnip Brassica rapa).\n",
      "CaMV is a pararetrovirus, which is a major grouping containing hepadnaviruses (e.g., hepatitis B virus), badnaviruses (e.g., banana streak virus), and caulimoviruses (e.g., CaMV). Pararetroviruses are characterized by a non-segmented double-stranded DNA genome. After entering the host cell nucleus, the viral DNA accumulates as a minichromosome [28] whose transcription is ensured by the host RNA polymerase II [29]. The CaMV genome consists in approximately 8,000 bp and encodes six viral gene products that have been detected in planta (Figure 1) [30]. Viral proteins P1 to P6 are expressed from two major transcripts, namely a 19S RNA, encoding P6, and a 35S RNA corresponding to the entire genome and serving as mRNA for proteins P1–P5 [31]. Using the pre-genomic 35S RNA as a matrix, the protein P5 (product of gene V) reverse-transcribes the genome into genomic DNA that is concomitantly encapsidated [30].\n",
      "The CaMV genome is a circular double-stranded DNA of 8,024 bp, represented in the figure by a double line. The thick arrows with different textures represent the organization of open reading frames I to VI, encoding proteins detected in planta. Markers a, b, c, and d were engineered at the positions indicated (see Materials and Methods for precise positions). The inner black arrows represent monocistronic 19S RNA and polycistronic 35S RNA produced by the cellular machinery. The nucleotide position 0 (numbering according to [44]) indicates the origin of replication via reverse transcription, which occurs in the direction indicated by the dotted outermost circle-like arrow. Reverse transcription is accomplished by the viral reverse transcriptase, using the 35S RNA as template [49].\n",
      "The detection of CaMV recombinants in turnip hosts has been reported numerous times. Some studies have demonstrated the appearance of infectious recombinant viral genomes after inoculation (i) of a host plant with two infectious or non-infectious parental clones [21,32,33,34,35] or (ii) of a transgenic plant containing one CaMV transgene with a CaMV genome missing the corresponding genomic region [36]. While the former revealed inter-genomic viral recombination, the latter demonstrated that CaMV can also recombine with transgenes within the host's genome. Another study based on phylogenetic analyses of various CaMV strains has clearly suggested different origins for different genomic regions and, hence, multiple recombination events during the evolution of this virus [37]. Indirect experimental evidence has indicated that, in some cases, CaMV recombination could occur within the host nucleus, between different viral minichromosomes, presumably through the action of the DNA repair cellular machinery [21,35]. Nevertheless, the mechanism of “template switching” during reverse transcription, predominant in all retroviruses, most certainly also applies to pararetroviruses. For this reason, and on the basis of numerous experimental data, CaMV is generally believed to recombine mostly in the cytoplasm of the host cell, by “legal” template switching between two pre-genomic RNA molecules [21,35,36,38,39], or “illegal” template switching between the 19S and the 35S RNA [36,40]. Under this hypothesis, recombination in CaMV could therefore be considered as operating on a linear template during reverse transcription, with the 5′ and 3′ extremities later ligated to circularize the genomic DNA (position 0 in Figure 1). The above cited studies clearly demonstrate that CaMV is able to recombine. However, since these studies are based on complementation techniques, non-quantitative detection, or phylogenetically based inferences of recombination, they do not inform us on whether recombination is an exceptional event or an “everyday” process shaping the genetic composition of CaMV populations.\n",
      "In the present work, we aimed at answering this question. To this end, we have constructed a CaMV genome with four genetic markers, demonstrated to be neutral in competition experiments. By co-inoculating host plants with equal amounts of wild-type and marker-containing CaMV particles, we have generated mixed populations in which impressive proportions of recombinants—distributed in several different classes corresponding to exchange of different genomic regions—have been detected and quantified. Altogether, the recombinant genomes averaged over 50% of the population. Further analysis of these data, assuming a number of viral replications during the infection period ranging from five to 20, indicates that the per nucleotide per replication cycle recombination rate of CaMV is of the same order of magnitude, i.e., on the order of a few 10−5, across the entire genome. We thereby provide the first quantification, to our knowledge, of the recombination rate in a virus population during a single passage in a single host.\n",
      "\n",
      "Results\n",
      "From Figure 1, and supposing that all marker-containing genomic regions can recombine, we could predict the detection and quantification of seven classes of recombinant genotypes: +bcd/a+++, a+cd/+b++, ab+d/++c+, abc+/+++d, ++cd/ab++, a++d/+bc+, and a+c+/+b+d. Indeed, all classes were detected, and their frequencies in the ten CaMV populations analyzed are summarized in Table 1.\n",
      "\n",
      "a Viral genomes were cloned and analyzed from ten of 24 co-infected plants\n",
      "\n",
      "b Seven possible classes of recombinants were predicted, their respective frequency in the population is expressed in percentage\n",
      "\n",
      "c The proportions of recombinants from the seven classes were added to estimate the total percentage of recombinant genomes within each tested plant\n",
      "Altogether, the proportion of recombinant genomes found in the mixed viral populations was astonishingly high and very similar in the ten co-infected plants analyzed (Table 1, last column), ranging between 44% (plant 5) to 60% (plants 7, 12, and 20), with a mean frequency (± standard error) of 53.8% ± 2.0%. This result indicates that recombination events are very frequent during the invasion of the host plant by CaMV and represents, to our knowledge, the first direct quantification of viral recombination during the infection of a live multi-cellular host.\n",
      "The inferred per generation recombination and interference rates, assuming that CaMV undergoes ten replication cycles during the 21 d between infection and sampling, are given for each of the ten plants in Table 2. Recombination rates between adjacent markers are large, on the order of 0.05 to 0.1. Taking the distance in nucleotides between markers into account yields an average recombination rate per nucleotide and generation on the order of 4 × 10−5. Interestingly, this recombination rate does not vary throughout the genome (Kruskal–Wallis test, p = 0.16).\n",
      "The various parameters are as follows: r\n",
      "1, recombination rate between markers a and b; r\n",
      "2, recombination rate between markers b and c; r\n",
      "3, recombination rate between markers c and d; i\n",
      "12, interference between crossovers in segments a–b and b–c; i\n",
      "23, interference between crossovers in segments b–c and c–d; i\n",
      "13, interference between crossovers in segments a–b and c–d; i\n",
      "123, second-order interference accounting for residual interference. The recombination rates are the maximum likelihood estimates (± 95% confidence intervals). The interference parameters were obtained numerically as explained in the Materials and Methods\n",
      "\n",
      "To relax the assumption of the number of replications during the 21 d, we calculated the recombination parameters assuming five or 20 generations. The effect of the number of generations on the estimates is linear: doubling the number of generations results in a halving of the recombination rate (detailed results not shown). For example, the average recombination rates r\n",
      "1, r\n",
      "2, and r\n",
      "3 assuming 20 generations were equal to 0.05, 0.04, and 0.025, respectively (compare with values in Table 2), yielding per nucleotide per generation recombination rates of 1.9 × 10−5, 2.2 × 10−5 and 1.6 × 10−5.\n",
      "Inspection of Table 2 also shows that first-order interference coefficients were in general negative, indicating that a crossing over in one genomic segment increases the probability that a crossing over will occur in another genomic segment, while the second-order coefficient parameter had an average value close to zero with a large variance. The mechanism leading to these results will be discussed in the following section.\n",
      "\n",
      "Discussion\n",
      "One major breakthrough in the work presented here lies in the space and time scales at which the experiments were performed. Indeed, the processes occurring within the course of a single infection of one multi-cellular host are of obvious biological relevance for any disease. Previous studies on viral recombination suffered from major drawbacks in this respect, basing their conclusions on experiments relying on complementation among non-infectious viruses or between viruses with undetermined relative fitness, on phylogenetically based analyses, or on experiments in cell cultures. For reasons detailed in the Introduction, the first two methods either do not provide information on the frequency of recombination, but only its occurrence, or address the question at a different temporal, and often spatial, scale. Results from cell cultures, on the other hand, impose cell co-infection by different viral variants, potentially overestimating the frequency of recombination events. Our study circumvents these limitations by analyzing viral genotypes sampled from infected plants after the course of a single infection, and therefore the invasion and co-infection of cells in various organs and tissues is very close to natural.\n",
      "More than half of the genomes (53.8% ± 2.0%; see Table 1) present in a CaMV population after a single passage in its host plant were identified as recombinants, and these data allowed us to infer a per nucleotide per generation recombination rate on the order of 2 × 10−5 to 4 × 10−5. The time length of one generation, i.e., the time required for a given genome to go from one replication to the next, is totally unknown in plant viruses. The only experimental data available on CaMV are based on the kinetics of gene expression in infected protoplasts, where the capsid protein is produced between 48 and 72 h [40]. The reverse transcription and the encapsidation of genomic DNA being two coupled phenomena [30], we judged it reasonable to assume a generation time of 2 d and, thus, an average of ten generations during our experiments. In case this estimate is mistaken, we have verified a linear relationship between r and the number of generations, thereby allowing an immediate adjustment of r if the CaMV generation time is more precisely established. At this point, we must consider that all cloned genomes may not have been through all the successive replication events potentially allowed by the timing of our experiments. It was previously shown that about 95% of CaMV mature virus particles accumulate in compact inclusion bodies [41], where they may be sequestered for a long time, as such inclusions are very frequent in all infected cells, including those in leaves that have been invaded by the virus population for several weeks. The viral population may thus present an age structure that could bias the estimation of the recombination rate. In order to minimize this bias, the clones we analyzed were collected in one young newly formed leaf, where the chances of finding genomes from “unsequestered lines” were assumed to be higher. In any case, our data analysis is conservative, since this age structure can only lead to an underestimation of the recombination rate.\n",
      "Our results show that interferences between pairs of loci are negative: a recombination event between two loci apparently increases the probability of recombination between another pair of loci. We believe that the most parsimonious explanation of these negative interferences is based on the way the infection builds up within plant hosts. Indeed, one can divide infected host cells into those infected by a single virus genotype and those infected by more than one viral genotype. In the former, analogous to clonal propagation, recombination is undetectable. In the latter, recombination is not only detectable but, as our results indicate, very frequent. Samples consisting of viruses resulting from a mixture of these two types of host cell infections will thus contain viruses with no recombination and viruses with several recombination events, thus yielding an impression of negative interference. These conceptual arguments are supported by mathematical models. It is indeed easy to show (detailed results not shown) that if a proportion F of the population reproduces clonally, analogous to single infections, while the remaining reproduces panmictically, negative interferences could be inferred even if they do not exist. For example, assuming a three-locus model with real recombination rates r\n",
      "1 and r\n",
      "2 and interference i\n",
      "12, the “apparent” recombination and interference parameters, would be r\n",
      "1 = (1 − F)r1, r\n",
      "2 = (1 − F)r\n",
      "2, and i\n",
      "12 = −(F − i\n",
      "12)/(1 − F). Interestingly, this example also shows that our estimates of the recombination rate are conservative: that a fraction F of host cells are singly infected while others are multiply infected leads to an underestimation of the recombination rate.\n",
      "As judged by r\n",
      "1, r\n",
      "2, and r\n",
      "3, calculated between markers a–b, b–c, and c–d, respectively, we found evidence for recombination through the entire CaMV genome. The values for r\n",
      "1,\n",
      "r\n",
      "2, and r\n",
      "3 are remarkably similar, hence the recombination sites seem to be evenly distributed along the genome. We considered the template-switching model as the major way recombinants are created in CaMV. As already mentioned in the Introduction, hot spots of template switching have been predicted at the position of the 5′ extremities of the 35S and 19S RNAs [21,36,42]. If other recombination mechanisms, such as that associated with second-strand DNA synthesis or with the host cell DNA repair machinery, act significantly, hot spots would be expected at the positions of the sequence interruption Δ1, Δ2, and Δ3 [43]. Due to the design of our experiment and the position of the four markers, we have no information on putative hot spots at positions corresponding to the 5′ end of the 35S RNA and to Δ1 (at nucleotide position 0). Nevertheless, the putative hot spots at the 5′ end of the 19S RNA and at Δ2 and Δ3 (nucleotide positions 4,220 and 1,635, respectively) fall between marker pairs c–d, b–c, and a–b, respectively. Our results indicate that either these hot spots are quantitatively equivalent—though predicted by different recombination mechanisms—or, more likely, that they simply do not exist. Whatever the explanation, what we observe is that the CaMV can exchange any portion of its genome, and thus any gene thereof, with an astonishingly high frequency during the course of a single host infection.\n",
      "To our knowledge, the viral recombination rate has never previously been quantified experimentally for a plant virus [3]. In contrast, retroviruses and particularly HIV-1 have been extensively investigated in that sense. As we have already discussed for these latter cases, the quantification of the intrinsic recombination rate was carried out in artificially co-infected cell cultures. The estimated intrinsic per nucleotide per generation recombination rate in HIV-1 is on the order of 10−4 [14,15,19], less than one order of magnitude higher than our estimation for CaMV. Because for various reasons detailed above we probably underestimate the within-host CaMV recombination rate, we believe that the intrinsic recombination rate in CaMV is higher and perhaps on the order of that of HIV.\n",
      "Other pararetroviruses such as plant badnaviruses or vertebrate hepadnaviruses have a similar cycle within their host cells, including steps of nuclear minichromosome, genomic size RNA synthesis, and reverse transcription and encapsidation. Nevertheless, vertebrate hepadnaviruses (e.g., hepatitis B virus) infect hosts that are very different from plants in their biology and physiology, and this could lead to a totally different frequency of cell co-infection during the development of the virus populations. Thus, even though our results can be informative for other pararetroviruses because of the viruses' shared biological characteristics, they should not be extrapolated to vertebrate pararetroviruses without caution.\n",
      "\n",
      "Materials and Methods\n",
      "\n",
      "We used the plasmid pCa37, which is the complete genome of the CaMV isolate Cabb-S, cloned into the pBR322 plasmid at the unique SalI restriction site [44]. To analyze recombination in different regions of the genome, we introduced four genetic markers: a, b, c, and d, at the positions 881, 3,539, 5,365, and 6,943, respectively, thus approximately at four cardinal points of the CaMV circular double-stranded DNA of 8,024 bp (Figure 1). All markers, each corresponding to a single nucleotide change, were introduced by PCR-directed mutagenesis in pCa37, and resulted in the duplication of previously unique restriction sites BsiWI, PstI, MluI, and SacI in a plasmid designated pMark-S. Because, in this study, we targeted the possible exchange of genes between viral genomes, all markers a, b, c, and d were introduced within coding regions corresponding to open reading frames I, IV, V, and VI, respectively. Another important concern was to quantify recombination in the absence of selection, i.e., to create neutral markers. Consequently all markers consist of synonymous mutations (see below).\n",
      "\n",
      "To generate the parental virus particles, plasmids pCa37 and pMark-S were mechanically inoculated into individual plants as previously described [33]. All plants were turnips (B. rapa cv, “Just Right”) grown under glasshouse conditions at 23 °C with a 16/8 (light/dark) photoperiod. Thirty days post-inoculation, all symptomatic leaves were harvested and viral particles were purified as described earlier [45].\n",
      "The resulting preparations of parental viruses, designated Cabb-S and Mark-S, were quantified by spectrometry using the formula described by Hull et al. [46]. We fixed the initial frequency of markers to a value of 0.5, and a solution containing 0.1 mg/ml of virus particles of both Cabb-S and Mark-S at a 1:1 ratio was prepared. Plantlets were co-infected by mechanical inoculation of two to three leaves with 20 μl of this virus solution, using abrasive Celite AFA (Fluka, Ronkonkoma, New York, United States). The mixed CaMV population was allowed to grow during 21 d of systemic infection.\n",
      "\n",
      "We designed an experimental protocol for quantifying marker frequency within a mixed Cabb-S/Mark-S virus population after a single passage in a host plant. Twenty-four individual plants, inoculated as above with equal amounts of Cabb-S and Mark-S, were harvested 21 d post-inoculation, when symptoms were fully developed. The viral DNA was purified from 200 mg of young newly formed infected leaves according to the protocol described previously [47]. After the precipitation step of this protocol, the viral DNA was resuspended and further purified with the Wizard DNA clean-up kit (Promega, Fitchburg, Wisconsin, United States) in TE 1X (100 mM Tris-HCl and 10 mM EDTA [pH 8]). Aliquots of viral DNA preparations were digested by restriction enzymes corresponding either to marker a, b, c, or d and submitted to a 1% agarose gel electrophoresis, colored by ethydium bromide and exposed to UV. Each individual restriction enzyme cut once in Cabb-S DNA and twice in Mark-S, thus generating DNA fragments of different sizes attributable to one or the other in the mixed population of CaMV genomes. After scanning the agarose gels, we estimated the relative frequency of the two genotypes in each viral DNA preparation and at each marker position, by densitometry using the NIH 1.62 Image program. The statistical analyses of the frequency of the four markers are described below.\n",
      "\n",
      "To identify and quantify the recombinants within the CaMV mixed populations, aliquots from ten of the 24 viral DNA preparations described above were digested by the restriction enzyme SalI, and directly cloned into pUC19 at the corresponding site. In each of the ten viral populations analyzed, 50 full-genome-length clones were digested separately by BsiWI, PstI, MluI, and SacI, to test for the presence of marker a, b, c, and d, respectively. In this experiment, with the marker representing an additional restriction site, we could easily distinguish between the Cabb-S and the Mark-S genotype at all four marker positions, upon agarose gel (1%) electrophoresis of the digested clones. Clones with none or all four markers were parental genotypes, whereas clones harboring 1, 2, or 3 markers were clearly recombinants. Due to the very high number of recombinants detected, markers eventually appearing or disappearing due to spontaneous mutations were neglected.\n",
      "\n",
      "Here we present the different methods we used to quantify recombination in the CaMV genome. Because all these methods assume that the different markers are neutral, we first discuss assumption.\n",
      "We used two datasets to test the neutrality of markers, both resulting from plants co-infected with a 1:1 ratio of Mark-S and Cabb-S. The first consisted of viral DNA densitometry data derived from 24 plants (described above), where for each plant we have an estimate of the frequency of each marker in the genome population. The second consisted of the restriction of 50 individual full-genome-length viral clones obtained from one co-infected plant (described above), yielding an estimate of the frequency of each marker, and this was repeated on ten different plants. The frequencies of the different markers were 0.508, 0.501, 0.516, and 0.507 for markers a, b, c, and d in the first dataset and 0.521, 0.518, 0.514, and 0.524 in the second dataset. We tested whether these frequencies were significantly different from the expected value under neutrality, 0.5, using either t-tests, for datasets where normality could not be rejected (seven out of eight cases), or Wilcoxon signed-rank non-parametric tests otherwise (marker c in the first dataset). In all cases p-values were larger than 0.05.\n",
      "There are several cautionary remarks regarding these analyses. First, in all cases we found an excess of markers. Unfortunately, the two datasets cannot be regarded as independent because, even though the methods through which the frequency estimates were obtained were different, the plants used in the second dataset were a subset of the plants of the first. We thus have only four independent estimates in each case, and there is minimal power to detect significant deviations from neutrality with such a small sample size. It should be noted at this stage that deviations from the expected value could also be caused either by slight deviations from the 1:1 ratio in the infecting mixed solution, or by deviations from that ratio in the frequency of the viral particles that actually get into the plants. Second, because of the relatively small sample sizes and low statistical power, the tests presented above could have detected only large deviations.\n",
      "The results clearly show, however, precisely that the markers do not have large effects, if any, and that therefore recombination estimates would be affected only very slightly by any hypothetical selective effects of the markers. Because of this, along with the fact that the introduced markers provoke silent substitutions in the CaMV genome, we assumed that markers were effectively neutral in the rest of the analysis.\n",
      "The dataset used to estimate the recombination frequency consisted of the 500 full-genome-length viral clones (50 from each of ten co-infected plants) individually genotyped for each of the four markers. As discussed in detail in the Results, recombination was very frequent and concerned all four markers. Indeed, approximately half of the genotyped clones exhibited a recombinant genotype. It was therefore meaningful to try to obtain quantitative estimates of recombination from our data.\n",
      "Our aim was to analyze viral recombination in a live host. Consequently, we had to deal with the fact that more than one viral replication cycle occurred during the 21 d that infection lasted in our experiment (we had to wait that long for the disease to develop and to be able to recover sufficient amounts of viral DNA from each infection). Based on the kinetics of gene expression [40], we postulate that each replication cycle lasts between 2 and 3 d, and that therefore seven to ten cycles occurred between infection and the sampling time. In case this assumption is incorrect, we did calculations assuming five, seven, ten, or 20 replication cycles during these 21 d. As shown, the results were not affected qualitatively, and only slightly quantitatively. It is important to note that we assumed that recombination occurred through a template-switching mechanism, and that therefore, from a recombination point of view, the CaMV genome is linear. The reverse transcription starts and finishes at the position 0 in Figure 1, which is the point of circularization of the DNA genome. This implies that changes between contiguous markers a–b, b–c, and c–d can be considered as true recombination whereas those between a and d cannot, as they may simply stem from circularization of DNA, during the synthesis of which the polymerase has switched template once anywhere between a–b, b–c, or c–d.\n",
      "To estimate the recombination rate between markers, we wrote recurrence equations describing the change in frequency of each genotype over one generation, assuming random mating and no selection (i.e., the standard Wright–Fisher population genetics model). We then expressed the frequency of all possible genotypes n generations later as a function of their initial frequency and of the recombination parameters. Subsequently we calculated the maximum likelihood estimates of the recombination parameters and their asymptotic variances given initial frequencies (we assumed that the two “parental” genotypes, Mark-S and Cabb-S, had equal initial frequencies of 0.5 and that all other genotypes had initial frequencies of zero) and frequencies after n generations (the observed frequencies; as stated above we used different values of n). All algebraic and numerical calculations were carried out with the software Mathematica.\n",
      "The recombination parameters are the recombination rates between two adjacent loci, e.g., r\n",
      "1 for the recombination rate between markers a and b, and the interference coefficients, e.g., i\n",
      "12 for interference between recombination events in the segments between markers a and b and b and c. To define these parameters we followed Christiansen [48], and in particular the recombination distributions for two, three, and four loci (respectively, Tables 2.7, 2.8, and 2.9 of [48]). It is important to realize that given the definitions of these parameters, the estimator of the recombination rate between two loci is not affected by the number of loci considered. In other words, we obtain the same estimation of the recombination rate between markers a and b whether we consider genotypic frequencies at just these two loci, or the frequencies at these two loci plus a third locus, or the complete information to which we have access, the four-marker genotypes. Information on additional loci only affects the estimates of the interference coefficients.\n",
      "It proved impossible to carry out the calculations for four loci algebraically. Instead, we used a computer program to calculate the expected genotypic frequencies at all four loci after n generations, given the above stated initial frequencies and specified recombination parameters. For each combination of recombination parameters we calculated a Euclidean distance between the vector of the expected genotypic frequencies and the observed genotypic frequencies, and considered that the estimated recombination parameters were those yielding the minimal Euclidean distance. In all cases, the estimated recombination rates between pairs of loci were equal to the second decimal to those estimated algebraically from data for three or two loci.\n",
      "\n",
      "Acknowledgments\n",
      "\n",
      "Competing interests. The authors have declared that no competing interests exist.\n",
      "\n",
      "Abbreviation\n",
      "\n",
      "References\n",
      "\n",
      "Formats:\n",
      "\n",
      "Share\n",
      "\n"
     ]
    }
   ],
   "source": [
    "sections = {}\n",
    "headers = tmp_soup.find_all('h2')\n",
    "for header in headers:\n",
    "    print(header.text)\n",
    "    paragraphs = header.parent.find_all('p')\n",
    "    for paragraph in paragraphs:\n",
    "        print(paragraph.text)\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
