{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import json\n",
    "import pandas as pd\n",
    "import yaml\n",
    "from itertools import chain\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "def get_file_names(path):\n",
    "    # get list of papers .json from path\n",
    "    file_names = []\n",
    "    for file_name in os.listdir(path):\n",
    "        file_names.append(file_name)\n",
    "    return file_names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scraper\n",
    "\n",
    "Scraper to fetch article data online based on article information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# init session\n",
    "session = requests.session()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\david\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:3: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "# import configurations\n",
    "with open('config.yaml','r') as ymlfile:\n",
    "    cfg = yaml.load(ymlfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load metadata\n",
    "df_meta = pd.read_csv('metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 19471 paper ids from Elsevier.\n",
      "There are 153 paper ids from CZI.\n",
      "There are 1154 paper ids from WHO.\n",
      "There are 22309 paper ids from PMC.\n",
      "There are 83 paper ids from PMC_new.\n",
      "There are 605 paper ids from biorxiv.\n",
      "There are 445 paper ids from medrxiv.\n"
     ]
    }
   ],
   "source": [
    "sources = df_meta['source_x'].unique()\n",
    "for source in sources:\n",
    "    #print(source)\n",
    "    papers_df = df_meta[df_meta['source_x']==source][['sha','doi']]\n",
    "    papers_array = papers_df.to_numpy()\n",
    "    print('There are {} paper ids from {}.'.format(len(papers_array),source))\n",
    "    \n",
    "    # get papers relative to source"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "source_x\n",
       "CZI           153\n",
       "Elsevier    19409\n",
       "PMC         12425\n",
       "PMC_new        82\n",
       "WHO          1104\n",
       "biorxiv         0\n",
       "medrxiv         0\n",
       "Name: journal, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.groupby(['source_x'])['journal'].count()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scrape papers from bioRxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def scrape_biorxiv(paper):\n",
    "    print('Fetching paper data at {}'.format(paper['url']))\n",
    "    html = session.get(paper['url'])\n",
    "    tmp_soup = BeautifulSoup(html.content, 'html.parser')\n",
    "    paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "    paper = dict(chain(paper.items(),paper_webdata.items()))\n",
    "    return paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_authors(authors_html):\n",
    "    ''' HTML list of authors\n",
    "    '''\n",
    "    authors = []\n",
    "    for author in authors_html:\n",
    "        try:\n",
    "            given_name = author.find('span',attrs={'class':'nlm-given-names'}).text\n",
    "        except:\n",
    "            given_name = ''\n",
    "        try:\n",
    "            surname = author.find('span',attrs={'class':'nlm-surname'}).text\n",
    "        except:\n",
    "            surname = ''\n",
    "        authors.append(given_name + ' ' + surname)\n",
    "    return authors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_references(references_html):\n",
    "    ''' HTML list of references\n",
    "    soup.find('div',attrs={'class':'section ref-list'}).find_all('li')\n",
    "    '''\n",
    "    references = []\n",
    "    for ref in references_html:\n",
    "        reference = {}\n",
    "        # label\n",
    "        try:\n",
    "            reference['label'] = ref.find('span',attrs={'class':'ref-label'}).text\n",
    "        except:\n",
    "            reference['label'] = None\n",
    "\n",
    "        # authors\n",
    "        author_names = []\n",
    "        try:\n",
    "            authors = ref.find('cite').find_all('span',attrs={'class':'cit-auth'})\n",
    "            for author in authors:\n",
    "                surname = author.find('span',attrs={'class':'cit-name-surname'}).text\n",
    "                given_names = author.find('span',attrs={'class':'cit-name-given-names'}).text\n",
    "                author_name = surname + ' ' + given_names\n",
    "                author_names.append(author_name)\n",
    "            reference['authors'] = author_names\n",
    "        except:\n",
    "            reference['authors'] = author_names\n",
    "\n",
    "        # date\n",
    "        try:\n",
    "            reference['date'] = ref.find('cite').find('span',attrs={'class':'cit-pub-date'}).text\n",
    "        except:\n",
    "            reference['date'] = None\n",
    "\n",
    "        # title\n",
    "        try:\n",
    "            reference['title'] = ref.find('cite').find('span',attrs={'class':'cit-article-title'}).text\n",
    "        except:\n",
    "            reference['title'] = ''\n",
    "\n",
    "        # links\n",
    "        ref_links = []\n",
    "        try:\n",
    "            links = ref.find('div',attrs={'class':'cit-extra'}).find_all('a')\n",
    "            for link in links:\n",
    "                if link.text != 'OpenUrl':\n",
    "                    link_name = link.text\n",
    "                    link_url = link['href']\n",
    "                    ref_links.append((link_name,link_url))\n",
    "            reference['links'] = ref_links\n",
    "        except:\n",
    "            reference['links'] = ref_links\n",
    "        references.append(reference)\n",
    "    return references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_sections(sections_html):\n",
    "    '''\n",
    "    '''\n",
    "    sections = []\n",
    "    for section in sections_html:\n",
    "        # ignore references\n",
    "        if 'ref-list' not in section.attrs['class'] :\n",
    "            sections.append(explore_section(section))\n",
    "    return sections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_tables(html_tables):\n",
    "    tables = []\n",
    "    for html_table in html_tables:\n",
    "        table = {}\n",
    "        table['id'] = html_table['id']\n",
    "\n",
    "        table_label = html_table.find('span',attrs={'class':'table-label'})\n",
    "        if table_label:\n",
    "            table['label'] = table_label.text\n",
    "\n",
    "        table_caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "        if table_caption:\n",
    "            table['caption'] = table_caption.text\n",
    "\n",
    "        table['description'] = ''\n",
    "        for p in html_table.find_all('p'):\n",
    "            table['description'] += p.text + '\\n'\n",
    "        table['image'] = None\n",
    "        tables.append(table)\n",
    "    return tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def get_biorxiv_figures(figures_html):\n",
    "    figures = []\n",
    "    for html_figure in figures_html:\n",
    "        figure = {}\n",
    "\n",
    "        figure_label = html_figure.find('span',attrs={'class':'fig-label'})\n",
    "        if figure_label:\n",
    "            figure['label'] = figure_label.text\n",
    "\n",
    "        figure_title = html_figure.find('span',attrs={'class':'caption-title'})\n",
    "        if figure_title:\n",
    "            figure['title'] = figure_title.text\n",
    "\n",
    "        figure['description'] = ''\n",
    "        figure_ps = html_figure.find_all('p')\n",
    "        for p in figure_ps:\n",
    "            figure['description'] += p.text + '\\n'\n",
    "        figures.append(figure)\n",
    "        figure_img = html_figure.find('img')\n",
    "        if figure_img:\n",
    "            figure['img'] = figure_img['data-src']\n",
    "    return figures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def process_paragraph(paragraph):\n",
    "    ''' Removes and modifies html tags in paragraphs\n",
    "    '''\n",
    "    for anchor in paragraph.find_all('a'):\n",
    "        anchor.replace_with('[' + anchor['href'].strip('#') + ',' + anchor.text + ']')\n",
    "    return paragraph.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def explore_section(root_tag):\n",
    "    '''Recursively explore section and all subsections for content\n",
    "    '''\n",
    "    content = {}\n",
    "    content['title'] = ''\n",
    "    content['content'] = []\n",
    "    for child in root_tag.findChildren(recursive=False):\n",
    "        if child.name == 'h2' or child.name == 'h3' or child.name == 'h4':\n",
    "            content['title'] = child.text\n",
    "        elif child.name == 'p':\n",
    "            content['content'].append(process_paragraph(child))\n",
    "        if child.name == 'div':\n",
    "            if 'class' in child.attrs:\n",
    "                if 'subsection' in child['class']:\n",
    "                    # if subsection explore recursively\n",
    "                    content['content'].append(explore_section(child))\n",
    "            elif 'id' in child.attrs.keys():\n",
    "                if 'T' in child['id']:\n",
    "                    content.append({'id':child['id'],'type':'table'})\n",
    "                elif 'F' in child['id']:\n",
    "                    content.append({'id':child['id'],'type':'figure'})\n",
    "    else:\n",
    "        return content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "def biorxiv_html_paper_data_extractor(soup):\n",
    "    '''\n",
    "        Extract paper information from html webpage of biorxiv\n",
    "    '''\n",
    "    paper = {}\n",
    "    paper['title'] = soup.h1.text\n",
    "\n",
    "    authors_html = soup.find('span',attrs={'class':'highwire-citation-authors'}) \\\n",
    "                       .find_all('span',attrs={'class':'highwire-citation-author'})\n",
    "    paper['authors'] = get_biorxiv_authors(authors_html)\n",
    "\n",
    "    sections_html = soup.find_all('div',attrs={'class':'section'})\n",
    "    paper['sections'] = get_biorxiv_sections(sections_html)\n",
    "\n",
    "    tables_html = soup.find_all('div',attrs={'class':'table'})\n",
    "    paper['tables'] = get_biorxiv_tables(tables_html)\n",
    "\n",
    "    # get figures\n",
    "    figures_html = soup.find_all('div',attrs={'class':'fig'})\n",
    "    \n",
    "\n",
    "    paper['figures'] = get_biorxiv_figures(figures_html)\n",
    "\n",
    "    # get references\n",
    "    references_html = soup.find('div',attrs={'class':'section ref-list'}).find_all('li')\n",
    "    paper['references'] = get_biorxiv_references(references_html)\n",
    "    \n",
    "    return paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "papers_df = df_meta[df_meta['source_x']=='biorxiv'][['sha','doi']]\n",
    "papers_array = papers_df.to_numpy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Single Paper Test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['f056da9c64fbf00a4645ae326e8a4339d015d155' '10.1101/001727']\n"
     ]
    }
   ],
   "source": [
    "row = papers_array[0]\n",
    "print(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/001727v1.full\n"
     ]
    }
   ],
   "source": [
    "paper = {}\n",
    "paper['source'] = 'biorxiv'\n",
    "paper['id'] = row[0]\n",
    "paper['url'] = 'https://biorxiv.org/content/' + row[1].strip('doi.org') + 'v1.full'\n",
    "print('Fetching paper data at {}'.format(paper['url']))\n",
    "html = session.get(paper['url'])\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')\n",
    "paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "paper = dict(chain(paper.items(),paper_webdata.items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Abstract 1\n",
      "1 Introduction 2\n",
      "2 Methods 2\n",
      "3 Results 4\n",
      "4 Discussion 4\n",
      "5 Appendix 1: Target Pathogen Database 308\n",
      "6 Appendix 2: Viral Database 1717\n",
      "7 Appendix 3: Bacterial Database 1068\n",
      "8 References 0\n"
     ]
    }
   ],
   "source": [
    "i = 0\n",
    "for section in paper['sections']:\n",
    "    print(i,section['title'], len(section['content']))\n",
    "    i += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'label': '',\n",
       "  'authors': ['Ames SK',\n",
       "   'Hysom DA',\n",
       "   'Gardner SN',\n",
       "   'Lloyd GS',\n",
       "   'Gokhale MB',\n",
       "   'Allen JE'],\n",
       "  'date': '2013',\n",
       "  'title': 'Scalable metagenomic taxonomy classification using a reference genome database',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1093/bioinformatics/btt389&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=23828782&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom')]},\n",
       " {'label': '',\n",
       "  'authors': ['Bazinet AL', 'Cummings MP'],\n",
       "  'date': '2012',\n",
       "  'title': 'A comparative evaluation of sequence classification programs',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1186/1471-2105-13-92&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=22574964&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom')]},\n",
       " {'label': '',\n",
       "  'authors': ['Berendzen J',\n",
       "   'Bruno WJ',\n",
       "   'Cohn JD',\n",
       "   'Hengartner NW',\n",
       "   'Kuske CR',\n",
       "   'McMahon BH',\n",
       "   'Wolinsky MA',\n",
       "   'Xie G'],\n",
       "  'date': '2012',\n",
       "  'title': 'Rapid phylogenetic and functional classification of short genomic fragments with signature peptides',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1186/1756-0500-5-460&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=22925230&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom')]},\n",
       " {'label': '',\n",
       "  'authors': ['Brady A', 'Salzberg SL'],\n",
       "  'date': '2009',\n",
       "  'title': 'Phymm and PhymmBL: metagenomic phylogenetic classification with interpolated Markov models',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1038/nmeth.1358&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=19648916&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom'),\n",
       "   ('Web of Science',\n",
       "    '/lookup/external-ref?access_num=000269382900018&link_type=ISI')]},\n",
       " {'label': '',\n",
       "  'authors': ['Huson DH', 'Mitra S', 'Weber N', 'Ruscheweyh H', 'Schuster SC'],\n",
       "  'date': '2011',\n",
       "  'title': 'Integrative analysis of environmental sequences using MEGAN4',\n",
       "  'links': [('Abstract/FREE Full Text',\n",
       "    '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NjoiZ2Vub21lIjtzOjU6InJlc2lkIjtzOjk6IjIxLzkvMTU1MiI7czo0OiJhdG9tIjtzOjM3OiIvYmlvcnhpdi9lYXJseS8yMDE0LzAxLzEwLzAwMTcyNy5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=')]},\n",
       " {'label': '',\n",
       "  'authors': ['Korbel JO',\n",
       "   'Abyzov A',\n",
       "   'Mu XJ',\n",
       "   'Carriero N',\n",
       "   'Cayting P',\n",
       "   'Zhang Z',\n",
       "   'Snyder M',\n",
       "   'Gerstein MB'],\n",
       "  'date': '2009',\n",
       "  'title': 'PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1186/gb-2009-10-2-r23&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=19236709&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom')]},\n",
       " {'label': '',\n",
       "  'authors': ['Langmead B', 'Salzberg S'],\n",
       "  'date': '2012',\n",
       "  'title': 'Fast gapped-read alignment with Bowtie 2',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1038/nmeth.1923&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=22388286&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom')]},\n",
       " {'label': '',\n",
       "  'authors': ['Liu B', 'Gibbons T', 'Ghodsi M', 'Treangen T', 'Pop M'],\n",
       "  'date': '2011',\n",
       "  'title': 'Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1186/1471-2164-12-S2-S4&link_type=DOI')]},\n",
       " {'label': '',\n",
       "  'authors': ['Richter DC', 'Ott F', 'Auch AF', 'Schmid R', 'Huson DH'],\n",
       "  'date': '2008',\n",
       "  'title': 'MetaSim: a sequencing simulator for genomics and metagenomics',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1371/journal.pone.0003373&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=18841204&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom')]},\n",
       " {'label': '',\n",
       "  'authors': ['Schloissnig S',\n",
       "   'Arumugam M',\n",
       "   'Sunagawa S',\n",
       "   'Mitreva M',\n",
       "   'Tap J',\n",
       "   'Zhu A',\n",
       "   'Waller A',\n",
       "   'Mende DR',\n",
       "   'Kultima JR',\n",
       "   'Martin J',\n",
       "   'Kota K',\n",
       "   'Sunyaev SR',\n",
       "   'Weinstock GM',\n",
       "   'Bork P'],\n",
       "  'date': '2013',\n",
       "  'title': 'Genomic variation landscape of the human gut microbiome',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1038/nature11711&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=23222524&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom'),\n",
       "   ('Web of Science',\n",
       "    '/lookup/external-ref?access_num=000312933800027&link_type=ISI')]},\n",
       " {'label': '',\n",
       "  'authors': ['Segata N',\n",
       "   'Boernigen D',\n",
       "   'Tickle TL',\n",
       "   'Morgan XC',\n",
       "   'Garrett WS',\n",
       "   'Huttenhower C'],\n",
       "  'date': '2013',\n",
       "  'title': 'Computational meta’omics for microbial community studies',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1038/msb.2013.22&link_type=DOI')]},\n",
       " {'label': '',\n",
       "  'authors': ['Segata N',\n",
       "   'Waldron L',\n",
       "   'Ballarini A',\n",
       "   'Narasimhan V',\n",
       "   'Jousson O',\n",
       "   'Huttenhower C'],\n",
       "  'date': '2012',\n",
       "  'title': 'Metagenomic microbial community profiling using unique clade-specific marker genes',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1038/nmeth.2066&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=22688413&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom'),\n",
       "   ('Web of Science',\n",
       "    '/lookup/external-ref?access_num=000307015700022&link_type=ISI')]},\n",
       " {'label': '',\n",
       "  'authors': ['Sunagawa S',\n",
       "   'Mende DR',\n",
       "   'Zeller G',\n",
       "   'Izquierdo-Carrasco F',\n",
       "   'Berger SA',\n",
       "   'Kultima JR',\n",
       "   'Coelho LP',\n",
       "   'Arumugam M',\n",
       "   'Tap J',\n",
       "   'Nielsen HB',\n",
       "   'Rasmussen S',\n",
       "   'Brunak S',\n",
       "   'Pedersen O',\n",
       "   'Guarner F',\n",
       "   'de Vos WM',\n",
       "   'Wang J',\n",
       "   'Li J',\n",
       "   'Doré J',\n",
       "   'Ehrlich SD',\n",
       "   'Stamatakis A',\n",
       "   'Bork P'],\n",
       "  'date': '2013',\n",
       "  'title': 'Metagenomic species profiling using universal phylogenetic marker genes',\n",
       "  'links': [('CrossRef',\n",
       "    '/lookup/external-ref?access_num=10.1038/nmeth.2693&link_type=DOI'),\n",
       "   ('PubMed',\n",
       "    '/lookup/external-ref?access_num=24141494&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom'),\n",
       "   ('Web of Science',\n",
       "    '/lookup/external-ref?access_num=000327698100019&link_type=ISI')]},\n",
       " {'label': '',\n",
       "  'authors': ['Wood DE', 'Salzberg SL'],\n",
       "  'date': None,\n",
       "  'title': 'Ultrafast metagenomic sequence classification using exact alignments',\n",
       "  'links': []},\n",
       " {'label': '',\n",
       "  'authors': ['Wu GD',\n",
       "   'Chen J',\n",
       "   'Hoffmann C',\n",
       "   'Bittinger K',\n",
       "   'Chen YY',\n",
       "   'Keilbaugh SA',\n",
       "   'Bewtra M',\n",
       "   'Knights D',\n",
       "   'Walters WA',\n",
       "   'Knight R',\n",
       "   'Sinha R',\n",
       "   'Gilroy E',\n",
       "   'Gupta K',\n",
       "   'Baldassano R',\n",
       "   'Nessel L',\n",
       "   'Li H',\n",
       "   'Bushman FD',\n",
       "   'Lewis JD'],\n",
       "  'date': '2011',\n",
       "  'title': 'Linking long-term dietary patterns with gut microbial enterotypes',\n",
       "  'links': [('Abstract/FREE Full Text',\n",
       "    '/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEyOiIzMzQvNjA1Mi8xMDUiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNC8wMS8xMC8wMDE3MjcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9')]}]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper['references']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "section = explore_section(methods)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Metagenomic classification methods are based on a wide variety of theoretical underpinnings. The basic varieties include alignment of reads to various nucleotide databases or exact matching to nucleotide or protein signature sequences (or kmers). A representative set of recent methods are described in [T1,Table 1] (also see [ref-2,Bazinet & Cummings 2012]).'"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "section['content'][0]['content'][0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get All Papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/010389v1.full\n",
      "Paper ID: eccef80cfbe078235df22398f195d5db462d8000 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/012070v1.full\n",
      "Paper ID: c41fdb2efd6d61384a92a84cbba3f8233629a41b saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/018481v1.full\n",
      "Paper ID: 33565294e6bc67fb7ee14dcae6cfdb08148f4ea5 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/027722v1.full\n",
      "Paper ID: 1f9d3f9a1a0e8db6a086e0a2b5ba50cf9f235dae saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/029397v1.full\n",
      "Paper ID: 01e3b313e78a352593be2ff64927192af66619b5 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/030742v1.full\n",
      "Paper ID: nan saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/020834v1.full\n",
      "Paper ID: 3461d71f6890f7e5ba53bf168be3945cdb16d901 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/017178v1.full\n",
      "Paper ID: 1dd898b5ca1ae70ec0e3cad89fc87a165002a99e saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/041558v1.full\n",
      "Paper ID: 8cfb8dfc2f8b2c6e1d60ff48da02de6670694869 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/044594v1.full\n",
      "Paper ID: 6499f5c865c3db3bdfc1e969ea940a80d547a5bb saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/046532v1.full\n",
      "Paper ID: a59906b732bf4a489e282c3e4f499d4166c622e7 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/053967v1.full\n",
      "Paper ID: 1c0b25c8e3c89243fd8797e2e964e12689bc8556 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/054072v1.full\n",
      "Paper ID: d1b26388670660f66249b55ece2beccee4c18d1a saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/048801v1.full\n",
      "Paper ID: 99dab788e984601ee4efcd1138926caf10c00dcf saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/060434v1.full\n",
      "Paper ID: nan saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/061788v1.full\n",
      "Paper ID: 53442eacc3f233078507fa37b78267399e8c1e3b saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/041509v1.full\n",
      "Paper ID: 3b73c8abb7528c958cec0bae1890828af007b599 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/053421v1.full\n",
      "Paper ID: 97339c9d365437d54ea33be1a1b634d35f460208 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/065250v1.full\n",
      "Paper ID: c04e10858012bdd666d50c0d69c3d1e7224ccbea saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/069013v1.full\n",
      "Paper ID: ffbf8ea9948d73572fd052a74afa01b19e6758a3 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/069799v1.full\n",
      "Paper ID: fd3fc2c49f5cc27e4262261d0c9045911d65cb6e saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/070656v1.full\n",
      "Paper ID: 44a440cc1c135c938d2216ea672f8ef4f9c01296 saved to data/biorxiv.\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/073098v1.full\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'text'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-53-07640c99b4ca>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     13\u001b[0m         \u001b[0mhtml\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m         \u001b[0mtmp_soup\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhtml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcontent\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'html.parser'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m         \u001b[0mpaper_webdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbiorxiv_html_paper_data_extractor\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtmp_soup\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m         \u001b[0mpaper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mchain\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mpaper_webdata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m         \u001b[1;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'data/biorxiv/{}.json'\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'id'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'w'\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mfp\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-52-beba7bd08c13>\u001b[0m in \u001b[0;36mbiorxiv_html_paper_data_extractor\u001b[1;34m(soup)\u001b[0m\n\u001b[0;32m      9\u001b[0m     \u001b[0mauthors\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mauthor\u001b[0m \u001b[1;32min\u001b[0m \u001b[0msoup\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'highwire-citation-authors'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind_all\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'highwire-citation-author'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 11\u001b[1;33m         \u001b[0mauthor\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mauthor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'nlm-given-names'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m' '\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mauthor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'nlm-surname'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     12\u001b[0m         \u001b[0mauthors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mauthor\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     13\u001b[0m     \u001b[1;31m#print(authors)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'text'"
     ]
    }
   ],
   "source": [
    "# get papers from data/biorxiv\n",
    "existing_paper_ids = set([name.strip('.json') for name in get_file_names('data/biorxiv')])\n",
    "\n",
    "# go through all papers\n",
    "# new_paper_cnt = 0\n",
    "for row in papers_array:\n",
    "    if row[0] not in existing_paper_ids:\n",
    "        paper = {}\n",
    "        paper['source'] = 'biorxiv'\n",
    "        paper['id'] = row[0]\n",
    "        paper['url'] = 'https://biorxiv.org/content/' + row[1].strip('doi.org') + 'v1.full'\n",
    "        print('Fetching paper data at {}'.format(paper['url']))\n",
    "        html = session.get(paper['url'])\n",
    "        tmp_soup = BeautifulSoup(html.content,'html.parser')\n",
    "        paper_webdata = biorxiv_html_paper_data_extractor(tmp_soup)\n",
    "        paper = dict(chain(paper.items(),paper_webdata.items()))    \n",
    "        with open('data/biorxiv/{}.json'.format(paper['id']), 'w') as fp:\n",
    "            json.dump(paper, fp, indent=4)\n",
    "        print('Paper ID: {} saved to data/biorxiv.'.format(paper['id']))\n",
    "        new_paper_cnt += 1\n",
    "        print('-'*60)\n",
    "print('Added {} papers to data/biorxiv.'.format(new_paper_cnt))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://biorxiv.org/content/10.1101/010389v1.full'\n",
    "html = session.get(url)\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Abstract': 'Background Developing methods to reconstruct transmission histories for viral outbreaks could provide critical information to support locating sources of disease transmission. Phylogenetic methods used to measure the degree of relatedness among sequenced viral samples have proven useful in identifying potential outbreak sources. The complex nature of infectious disease, however, makes it difficult to assign a rigorously defined quantitative confidence value assessing the likelihood of a true direct transmission event using genetic data alone.\\nResults A new method is presented to calculate a confidence value assessing the likelihood of a transmission event using both phylogenetic inference and limited knowledge of incubation and infectious duration times. The method is applied to simulations of a foot and mouth disease (FMD) outbreak to demonstrate how the combination of both phylogenetic and epidemiology data can be used to strengthen the assessment of the likelihood of direct transmission over methods using just phylogenetic data or infection timing data alone. The method is applied to a previous FMD outbreak to identify areas where over confidence in previously inferred direct transmission may exist.\\nConclusion Combining knowledge from viral evolution and epidemiology within a single integrated transmission inference framework is an important approach to assess the potential likelihood of transmission events and makes clear how specific features of a virus’ spread through the course of an outbreak will directly determine the potential for confidence in inferred host transmission links.\\n', 'Background': 'Developing methods to protect against RNA virus outbreaks in the face of constantly changing disease dynamics remains one of the great public health challenges of our time. One small piece of this problem involves determining the host-pathogen transmission history to identify an outbreak’s origin. In recent cases such as the 2003 SARS outbreak [1, 2], H5N1 influenza [3] infection in humans and Marburgvirus [4], phylogenetic inference has been used to help identify potential outbreak sources. Typically, a collection of sequences from candidate related sources are compared to infer transmission links between the most closely related sequences. These methods have been used beyond the public health domain to provide evidence in HIV [5, 6] and other forensic investigations [7] to help infer when a suspect is involved in infecting a victim.\\nSchematic of transmission links overlaid on phylogenetic trees T0 (left tree) and T1 (right tree). Transmission links are shown as directed edges connecting two infected hosts X0 to X1 and Y0 to Y1 respectively.\\nWhile phylogenetic data helps establish the evolutionary relationships among the sequenced viral samples, there are no well established tools to translate these results to explicitly infer host to host transmission links with confidence values that accurately capture the uncertainty of host transmission inference. The result is a sampling and transmission inference process that could lead to false confidence when using genetic data to establish host to host transmission [8, 9].\\nComplementary to the viral genome sequencing advances is contact tracing work to find host to host transmission patterns independent of molecular genetic sequence data [10–13]. Previously a transmission inference framework was described that estimates the probability of viral transmission between two hosts using both viral sequence samples and contact tracing data [14, 15]. The method uses labeled training sets from past outbreaks to correlate models of viral evolution associated with direct transmission and indirect transmission to assign a posterior probability that gives a confidence value for inferring a transmission relationship between two infected hosts. One of the limitations of the approach is the need for a robust training set, which can limit application to novel outbreaks. Even when training data is available, a training set over-represented with local outbreak features could skew confidence values. Thus, there is a need to develop additional tools to mitigate limits introduced from training. One feature that can be difficult to accurately account for with limited exemplars is time series data. The objective of this article is to present an approach to incorporate time series data with molecular genetic sequence data to measure the uncertainty of host to host transmission events in the face of limited training data and variability in host infection times.\\nFigure 1 shows a simple schematic to illustrate the variable host infection time transmission inference problem. The figure shows two simple trees, T0 and T1 (left and right tree respectively) where the infected hosts are overlaid as circles on the trees to show the transmission event from host X0 to X1 in T0 and Y0 to Y1 in T1. The larger circle for Y1 represents a longer infection time and the two trees represent the evolutionary relationship between the sequences. Assuming comparable homogeneous nucleotide substitution rates, the longer branch length in T1 could be explained by the longer infection time in host Y1 prior to sampling. Including the direct transmission branch lengths from T0 and T1 in a single genetic based transmission inference model leads to a distribution of genetic distances associated with direct transmission for a wider range of values. Thus, estimating accurate confidence values means ensuring that a representative sampling from the true distribution of infection times can be obtained. To measure the impact of this problem, a new method is introduced to incorporate variation in infection times when assigning a final posterior probability confidence value. The proposed method forgoes the use of training data by relying on prior knowledge of viral infection times to act as a constraint in conjunction with phylogenetic data. The results show that with knowledge of infection times confidence in transmission inference can be high and even with limited knowledge of timing patterns, high confidence values can be obtained that exceed the training based methods. Thus, this new transmission inference approach allows for the use of clinical observations of infection to be combined with inferred viral evolution when training data is not available to measure uncertainty associated with specific transmission hypotheses.\\nEfforts to integrate models of epidemiology and genetic evolution have been ongoing for some time [16–19], yet less work has focused on the specific problem of measuring the certainty of inferring individual host transmission events. One related effort is a method called SeqTrack, which was used to examine the origins of the 2009 H1N1 outbreak [20]. This work is similar to our previous transmission inference [14] approach with a few key differences. Rather than inferring a host transmission graph, a genealogy tree is inferred where nodes represent sequence samples and edges between nodes link a sequence to putative descendants. Sample collection times exclude some candidate edges and edge weights represent genetic distances between two sequences. The genealogy tree is inferred by finding the directed graph analog of the minimum spanning tree. Given the elapsed time between two sequences and a per day mutation rate, a confidence value is assigned for direct genealogy linkages. The framework presented here is a more general framework where variable within host evolution times are assumed to play a role. A second closely related effort is the work to recover transmission links between farms applied to the 2001 FMD outbreak [21].\\nFMD samples were collected and sequenced and an inferred phylogenetic tree was used in conjunction with farm infection times to evaluate the most likely transmission tree. Data from that study is used here and a comparison with this work will be covered in detail later in the paper. While the Cottam et al. approach evaluates a putative outbreak tree overlaid onto a single phylogenetic tree, our main contribution is to provide a framework that models features of transmission inference not previously considered. In particular, uncertainty in transmission inference is newly captured by explicitly considering unobserved nodes in the outbreak tree and uncertainty of phylogeny through the use of multiple potentially competing phylogenetic inferences. The result is a more realistic measure of uncertainty when assigning a transmission link. The framework’s added use of competing transmission hypotheses does, however, come with an increased computational cost. The remainder of the paper details the modifications to our original framework, shows how new information is used in simulations and real data, details important differences with the Cottam et al. approach and introduces a way for dealing with limitations introduced by the framework’s increased computational cost.\\n', 'Results': 'The previous framework is briefly reviewed to motivate the notation used to incorporate outbreak timing data. For any two infected hosts A and B, distance δ = δ(A, B) defines a genetic distance between samples taken from the two hosts. In the simplest form one sequenced sample is available for each host and the genetic relationship is described by a Jukes-Cantor like distance measure (δ can take on more complex forms as will be described later). The variable M represents the number of edges (ignoring direction) linking two nodes in an outbreak tree where nodes represent infected entities and edges represent directed transmission events from a source to a recipient. For the direct transmission hypothesis M=1 the posterior probability is defined to be:\\n\\n\\nwhere  for some maximum transmission chain length n. P (δ|M) is estimated from labeled training data using density estimators or fitting the observed data to a distribution such as Poisson, Negative Binomial or Gamma.\\nThe modified framework explicitly combines knowledge of viral infection times within the genetic based transmission inference framework. This is accomplished by adding a parameter σ and computing the value P (δ, σ|M) rather than P (δ|M). This reflects the desire to estimate the probability of observing the genetic relation δ in conjunction with infection related timing values for the transmission hypothesis M. The σ parameter values are motivated by a SEIR type infection concept [22], which describes a host in one of four states 1) Susceptible (S) to infection 2) Exposed (E) to an infection (but not infectious) 3) Infectious (I) - able to transmit virus to other hosts and 4) Recovered state (R) - where the host is no longer able to transmit the virus. The SEIR model describes outbreak dynamics in a homogeneous mixing population and since the focus of the framework is to predict individual host to host transmission relationships a pre-determined outbreak tree is used, which excludes hosts in the susceptible state. Figure 2 shows the graphical representation of two infected hosts A and B linked by a direct transmission event. Each circle represents a discrete time point marking the transition between states or time intervals. Transmission time points marked red in Figure 2 denote the time of transmission from Host A to Host B. Sample time points marked in purple represent times when viral samples are collected for sequencing. The representation is analogous to a phylogenetic tree where branches (or tree edges) represent time intervals and nodes represent fixed time points. Here edges or transmission events between infected nodes represent instantaneous time events and the host nodes represent time intervals defined by a linear chain of discrete events. The motivation for this representation is to employ a sufficiently realistic epidemiology model to describe the transmission relationships of interest and use existing methods for modeling viral evolution by mapping them to the epidemiology model.\\nHost infection time model. The small circle nodes denote discrete time events and edges denote time intervals. The large blue circles denote a single infected host.\\nThe modified framework reports the probability of jointly observing both the inferred genetic relationship between sequences from host A and B and the timing events associated with the viral infections of the two hosts. A key phylogenetic inferred parameter becomes the estimated “wall clock” time of the inferred most recent common ancestor between the time stamped sequence samples taken from the two hosts and potentially using samples from the larger outbreak. Here the Bayesian estimation software BEAST [23] is used to estimate a series of candidate phylogenetic relationships and inferred times, weighted by a likelihood value for each estimate. Transmission hypothesis testing determines whether the constraints implied by the hypothesis are consistent with the inferred most recent common ancestor times. Next, a description for estimating the probability of direct transmission is given, which is then extended to evaluate any transmission relationship between two nodes in an outbreak tree.\\nProbability distributions describe the expected exposure time and the expected infectious time in the host. An additional key requirement is the assignment of a fixed time point to each observed host that is relative to some global time clock. The point in time context can always be satisfied by assuming that the time is noted when a sample is collected from each infected individual. An alternative is to assume a fixed recovery time. For notational simplicity, a fixed recovery time is used although a comparable procedure is available for working from sample times as the reference. With host X recovery time RX, infectious time duration infX occurring with probability P (infX) and exposure time expX occurring with probability P (expX), the direct transmission hypothesis that host A transmitted the virus to host B is calculated by evaluating all pairwise combinations of expx and infx for each host. Candidate transmission events are marked by cases where the infectious time interval for host A overlaps with a potential initial infection time for host B and the inferred most recent common ancestor time for the sequence samples occurs after the initial infection time of A and prior to the initial infection time of B.\\nFigure 3 shows the notation and the constraints for the most recent common ancestor time (MRCAT). The initial infection time EB of host B is EB = RB − infB − expB and similarly the initial infection time EA for host A is EA = RA − infA − expA, which leads to the constraints defined by the specific transmission hypothesis. The transmission constraint C1(EB, RA, IA) returns 1 if IA ≤ EB < RA and 0 otherwise. The most recent common ancestor constraint C2(M RCAT, EB, EA) returns 1 if EA ≤ M RCAT < EB and 0 otherwise. The probability of EB and EA is defined to be P (EB|expB, infB, RB) = P (expB) × P (infB) and P (EA|expA, infA, RA) = P (expA) × P (infA) respectively. Now the genetic relation parameter δ is a set of estimates M RCATi for 1 ≤ i ≤ q where q is a finite integer determined during an independent phylogenetic parameter estimation process. (For example, using Bayesian genetic inference software BEAST, a default value could be q = 10, 000 estimates.) The direct transmission hypothesis is tested by finding the infection time intervals that are consistent with the inferred most recent common ancestor times and is described as follows:\\n\\n\\nand\\n\\n\\nwhere Z is a normalizing constant that reflects the maximum probability value of the individual timing interval events independent of constraints.\\nExample mapping of a two leaf phylogenetic tree onto the host node n (A or B). En is the initial infection time, In is initial infectious time, TA is transmission time from host A to host B. Sn,1 is the sample collection time (for the first of potentially multiple samples) and Rn is the initial recovery time. G – SA, 1 and G − SB,1 are genetic sequences for the sampled isolates. MRCAT is an inferred time that occurs within the time interval: EA ≤ M RCAT < TA.\\nTo maintain consistency with previous descriptions of the original framework [14], the transmission hypothesis M up to now was described by an integer value, which reports the number of edges in a tree separating the two nodes in question. This is sufficient for genetic only transmission inference when genetic distance is computed without regard to transmission direction. Thus, different sub tree configurations containing the same number of edges are treated identically without regard to transmission direction.\\nFigure 4 shows how the time series process of outbreak events distinguishes transmission direction. For a sub tree connecting two nodes with k edges, there are k distinct sub trees with each sub tree representing a distinct potential source node for the infection. For example, the transmission sub tree with two edges separating host A and B has two possible sub trees to describe the transmission relationship between the two hosts, one where the source node is A and one sub tree where A and B receive the infection through direct transmission from a common source. In general for a transmission hypothesis linking two nodes by k edges, there are k distinct sub trees that must be evaluated.\\nTransmission chains. The nodes labeled “s” represent the putative infection source. Blue nodes represent the two hosts (A and B) with sequenced samples. Green nodes reflect putative infected (hidden) intermediate hosts. Examples of transmission chains from length 1 (the direct transmission case) to 4 are shown.\\nThe color coding of nodes in Figure 4 highlights the two distinct node types in the hypothesis testing framework. The hypothesis test considers an outbreak tree with “observed” nodes and “hidden” nodes. The hidden nodes are marked green in the figure and represent infected individuals where no direct observable data is available. The hidden nodes explicitly model the probability of the alternative transmission hypothesis linking two observed nodes A and B. To evaluate the M > 1 case an evaluation procedure for each sub tree tM ∊ TM (A, B) is used. Thus, P (σ, δ|M > 1) becomes the value of the most probable alternative hypothesis:\\n\\n\\nand the calculation of P (σ, δ|tM) invokes a recursive tree traversal procedure to check the possible transmission timing values that are consistent with the transmission hypothesis tM . Hidden nodes cannot be tied to a global reference time but rather occur within a time range that is relative to the reference time points specified by the observed nodes and satisfy the transmission and phylogenetic constraints. A tree traversal process sets the range of possible infection times using the reference time points for the observed nodes. One of two traversal procedures is invoked depending on the sub tree structure: directed linear chain, where an observed node A is the source and the last node in the transmission chain is the other observed node B (see for example the bottom panel in Figure 4). The alternative traversal process is invoked when the two observed nodes are linked by transmission from a common source as shown in Figure 5. Figure 5 shows an example of the tree traversal process for one specific instance of tM where M = 3. The common source sub tree traversal algorithm starts with each observed node (leaf) independently and moves up the tree to the root. Each node n is assigned an earliest infection time Emin,n and a latest infection time Emax,n. The parent of node n is assigned an earliest and latest infection time for each minimum and maximum future initial infection time at node n. The total number of possible initial infection time ranges for the source node is 2d/2 where d is the length of the chain from the source to the leaf. The set of initial timing event ranges derived from one observable node are compared with the set of infection time ranges derived from the other observable node to determine whether there is a candidate infection time interval that supports the tree’s transmission hypothesis linking both observed nodes.\\nTree traversal procedure (right panel) for the transmission hypothesis shown in the left panel. Left panel shows a transmission hypothesis linking two observed infected nodes A and B (marked in blue), connected by a common unobserved labeled source node marked 0 and two intermediate source nodes marked 1 and 2. The right panel shows the post order tree traversal, where the observed time stamp for each observed node serves as the starting point and the possible infection times for each source node are computed by moving up the tree.\\nThe consistency function g(tM, INC, INF, MRCAT) checks the sub tree tM assuming each node n is assigned a fixed incubation time interval incn ∊ INC and a fixed infectious time interval infn ∊ INF and returns 1 if the constraints are met and 0 otherwise. The final probability calculation is the set of timing intervals that maximize the product of the individual timing probabilities over all nodes:\\nA similar procedure is applied for the linear transmission chain case. Here the algorithm starts with the observed source node and propagates candidate infection timing intervals forward to the next recipient node (rather than backwards to the parent node) until the final observable node is reached and the set of possible infection timing intervals is compared with terminal node timing to determine if a candidate timing interval supports the proposed transmission chain hypothesis.\\nThe key drawback to the method is computational cost. There are 2d+1 timing assignments that must be considered for each combination of timing intervals. If the number of possible distinct values for incn and infn is x the number of timing interval combinations becomes xM+1. Thus, there is a practical limit to the maximal value for M and x. Measuring the potential linkages for distantly related host pairs can be determined by viewing the estimated probability values of linkage up to the maximum M and when all probability values are low, M can be inferred to be too small to allow transmission inference.\\nTo evaluate the potential strength of inference from combining sequence and timing data, a simulation was used to generate genetic sequences from pre-defined outbreak conditions. The simulation uses a generative stochastic process for P (σ, δ|tM=1) and takes the following parameters as input:\\nA tree of infected hosts and their respective transmission links.\\nTwo probability density functions Pinc and Pinf determining the length of time intervals incn and infn respectively for each node n in a outbreak tree.\\nA transmission probability density function Ptrans determining the probability of transmitting the virus from host A to host B at time t within time interval infA.\\nA sample probability density function Psample determining when a sequence sample is collected within time interval infA for host A.\\nA per day mutation rate and an initial source index sequence.\\nA single run simulates an outbreak by propagating sequences across the pre-specified outbreak tree. The index case node A’s initial infection time is set to 0 (e.g. EA = 0) and the initial infectious time IA and recovery time RA are determined by values drawn from the Pinc and Pinf distributions. Secondary infected host B’s initial infection time EB is determined by drawing from the Ptrans distribution to fix the time within the infectious interval from IA to RA that the actual transmission event takes place. The index sequence evolves daily according to the per day mutation rate with distinct sequence stored for each host n for each day from En to Rn. When host A transmits the virus to host B at time t, the sequence in host A at time t becomes the seed sequence for host B. A single sequence is used to represent each host node, the chosen sampled sequence is determined by Psample, which gives a discrete time point within the infectious time interval from In to Rn in host node n. Sampled sequences are assumed to be taken following presentation of symptoms in the infected individual. This simulation approach is similar to a related effort to use an epidemic simulation model to construct large scale seasonal outbreaks of measles with viral genome sequence data [24]. Simulations were used to evaluate different viral sampling strategies for their proficiency in reconstructing viral population histories. A key difference with the work presented here is the focus on recovering individual host transmission chains rather than homogeneous host population infection models [25].\\nUsing the simulated sampled sequences from host nodes, phylogenetic trees were generated with BEAST running 1,000,000 iterations with 100,000 initial runs discarded and evaluating 10,000 sampled trees. The HKY nucleotide substitution model was used with fixed variation, and a relaxed molecular clock [26] with exponential distribution and a fixed population size coalescent model. The genetic distance between two sequences was taken to be the sum of the absolute time difference between two leaves in the tree and the most recent common ancestor. The final distance value is the weighted average over all available inferred trees. The same set of most recent common ancestor times was used as input to the method for integrating time with genetic data. For the test results reported below, the probability distributions for the incubation time and infectious duration time were taken to be uniform across a pre-determined range described below. The simulator reflects reported features from the 2001 UK FMD outbreak described in [21]. Infected hosts here do not reflect individual animals but rather infected farms. An average incubation time of 5 days is assumed with a variation between 2 and 14 days determined by a gamma distribution. The infectious time interval varied from 1 to 10 days drawn from a distribution model on previously described time intervals given by [21]. The objective was to evaluate the impact of combining the genetic data with timing information and apply the evaluation methods to the original 2001 UK FMD outbreak data to assess the confidence of the associated transmission links predicted using similar evidence but using the new inference approach. A per day mutation rate of 2.226 × 10−5 is used following a previously published report [10].\\nFigure 6 shows the tree used for the simulation. The tree is based on a subset of the 2001 outbreak given in [21]. The nodes labeled with two character codes were added to the published tree to connect nodes thought to be linked indirectly such that the total time duration distribution generated from repeated simulations covered the same time frame of 110 days reported for the outbreak. Results are based on 50 independent outbreak simulations. In the genetic distance only based framework labeled training data is used to estimate the likelihood of observing a genetic distance under the different possible transmission hypotheses. In the modified framework, a prior belief of how the virus spreads replaces the evidence taken from a training set. To evaluate the two approaches, 5 fold cross-validation takes the 50 randomly generated outbreaks and partitions the outbreaks into 5 non overlapping sets of 10 distinct outbreaks to evaluate predicted transmission link accuracy using the remaining 40 outbreaks for training. Evaluations are repeated 5 times to report performance for all 50 outbreaks. When training data is not needed, statistics are simply collected on all 50 outbreaks in one step. Three basic prediction methods are compared - using genetic data only, using the epidemiology timing data only and using both lines of evidence together to inform the final prediction.\\nTree used to generate simulated sequences and reflect known transmission links. Two character codes reflect hypothetical farms included in the tree and edges represent transmission relationships between infected farms.\\nPerformance with knowledge of timing. ROC curves showing the ability to correctly infer direct transmission cases (Sensitivity) versus falsely labeling indirect transmission pairs as direct transmission cases (False Positive Rate) using different confidence cut off values. Three methods are shown, using exact knowledge of timing data alone (Timing-Only), using only genetic data (MRCAT-Only) or combining the two and assuming exact knowledge of infection times.\\nFigure 7 shows a ROC curve for the rate of true direct transmission links identified (y-axis) versus the rate of host nodes falsely predicted to be linked (x-axis) using different prediction cutoff values. First consider prediction accuracy with near perfect knowledge of the timing events for the observed host nodes shown by the Timing-Only plot in Figure 7. The linear plot reflects the fact that the method generates a single prediction value so that above a certain cutoff either all of the true direct transmission links are predicted at the expense of falsely predicting transmission links for 3% of the other non-linked host node pairs or using a lower cutoff yields no predicted links. Recall that the calculation is meant to represent the most probable collection of incubation and infectious times consistent with the transmission hypothesis relative to the most probable collection of incubation times independent of the hypothesis. Here just a single set of incubation and infectious duration times are considered (the true time) and thus every pair of nodes with a timing pattern consistent with the predicted transmission event occurs with probability 1. Thus, the false positive rate reflects the fact that roughly 3% of node pairs that are not linked have a timing pattern consistent with the potential for direct transmission. These pairs likely refer to “siblings” or other nodes that are close by in the tree. This provides a baseline measure for quantifying the difficulty of the inference problem and shows that when exact timing knowledge is available the possible transmission inferences become highly constrained. The MRCAT-Only line shows the other end of the spectrum where there are no explicit timing constraints to prevent inference of direct transmission. The plot shows that a low confidence threshold is required to correctly identify all true direct transmission links, which shows that there is a wide overlap between the genetic distance range for both direct and indirectly linked hosts. Combining the two approaches shows that the already low false positive rate from the highly constrained timing data can be reduced still further by adding the constraints introduced by the timing data inferred from the time stamped genetic sequence data.\\nLimited timing knowledge. ROC curves showing the ability to correctly infer direct transmission cases (Sensitivity) versus falsely labeling indirect transmission pairs as direct transmission cases (False Positive Rate) using different confidence cut off values. Four methods are shown, using only timing data (Timing-Only), using only genetic data (MRCAT-Only) or combining the two assuming either a 2-14 day incubation period (Combined:2-14) or a 2-9 day incubation period (Combined:2-9).\\nFigure 7 illustrates how the algorithm works but assumes unrealistically precise knowledge of incubation and infectious duration times. Figure 8 shows the same comparison, but reducing the knowledge of incubation and infectious times. There are two distinct parameter settings shown in the Figure 8. One setting where every observed node is assumed to have an incubation time that could range from 2 to 9 days with equal probability and the estimated infectious duration for each node is based on the known window but assigned and uncertainty of plus or minus one day (Combined:2-9 in Figure 8). The second setting is identical except the incubation time is assumed to extend from 2 to 14 days (Combined:2-14 in Figure 8). The Timing-Only plot uses the 2 to 9 day incubation window and shows that (without using the genetic data) the range of possible timing scenarios more than triples the number of node pairs that are falsely linked by direct transmission from 3% to 10%. Combining the data sources shows improved inference with consistently higher sensitivity levels and less false positives. The results also show there are clear performance trade offs illustrating the underlying difficulty of the problem. On the one hand capping the maximal incubation time at 9 days greatly constrains the number of plausible transmission hypotheses, thus limiting the number of false positives. The trade off is that for the transmission cases where the incubation time actually exceeds 9 days the true positives can be missed, thus there is a hard upper bound on the sensitivity rate that is less than 1. Further relaxing restrictions to allow for the longer incubation times of up to 14 days allows detection of all true positives, but the number of false positives increases too. The key here seems to be the uncertainty associated with assigning a time to the most recent common ancestor. Longer incubation times open more plausible phylogenetic scenarios where two hosts are falsely linked by direct transmission but are actually linked by a shared infectious source. One potential way to strengthen inference could be to collect more sequences per farm spread over multiple time points, which in some situations may reduce the range of most likely most recent common ancestor times.\\nA second experiment was conducted to re-examine the transmission relationships reported by Cottom et al. on samples taken from a subset of farms infected during the 2001 United Kingdom outbreak. Although the true outbreak tree cannot be empirically determined, re-examination of this dataset proved useful in determining whether our new transmission inference method using similar information as input would lead to similar conclusions. Moreover, differences in output should highlight important differences in the two approaches and help assess the strengths and weaknesses of the different approaches.\\nFigure 9 shows the higher confidence inferred transmission links for the published 2001 outbreak data with whole viral genome sequence data available for each farm. An incubation window of 2 to 14 days and an estimated infectious time interval with a 1 day uncertainty was used to find the most likely transmission chain. Cottom et al. similarly apply a transmission inference algorithm to the data using a closely related approach that takes both estimated infection times and genetic sequence data as input. Cottom et al. use a statistical parsimony tree relating the collection of sequences as the basis for building putative transmission trees. The algorithm starts at the leaves of the parsimony tree and assigns the inferred common ancestor nodes to one of the farms (with a descendant leaf) as the putative source. All possible tree labellings are evaluated using an infection time likelihood score, which similar to our approach estimates likely infection times based on estimated incubation and infectious duration times. Several important differences between the two approaches emerge after examining the differences in output. One difference is that our method incorporates putative unobserved farms into transmission hypothesis tests, which help prevent over confidence in certain transmission inferences. As will be discussed in the next section, this can also form the basis for supporting a more flexible framework for evaluating more sparsely sampled outbreaks. A second difference is that multiple phylogenetic tree topologies are used in the form of multiple most recent common ancestor time estimates, which better reflect the uncertainty of phylogenetic inference. Finally, a key difference is in the information representation format. The Cottom et al. approach assigns outbreak tree nodes to phylogenetic tree nodes whereas our method assigns phylogenetic tree nodes to outbreak tree nodes, which represent variable length time. The significance of these differences is that the new framework can test many additional plausible transmission hypotheses, which would otherwise be ignored.\\nA most likely inferred transmission tree with links comparing the tree to previous reports. Red lines show novel links and dashed lines show previously reported alternative links. Each node represents an infected farm.\\nLikelihood for different transmission hypothesis for C to P and O to P, with different transmission chain lengths M.\\nDespite the differences, there was largely strong consensus for predictions between the two methods with all of the farms linked with high confidence in [21] predicted in the new method except the K to B link (results not shown). The contiguous edges marked in black show transmission inferences that match [21]. Linkage between K and B required increasing the infectious window beyond the initial reported estimate. To make a complete comparison the discrepancy is noted and the infectious window was increased to 4 days to accommodate the possibility of a transmission link from K to B. The two dashed edges show links made by the new inference tool not present in the most probable transmission tree reported in [21]. The two red edges show the direct transmission inferences in the most likely Cottam outbreak tree that are not present in the most probable tree reported here. Next the key differences in the two approaches are discussed with respect to differences in observed transmission inference.\\nTransmission hypotheses include the potential for intermediate unobserved farm involvement to estimate confidence values.\\nFor farm P, the new method only slightly favors an alternative source - C over O (0.57 versus 0.59) making it impossible to preclude either farm as the likely source. The favoring of C over O can be explained in part by differences in the alternative hypothesis likelihood scores. Figure 10 shows the likelihood scores for each of the individual transmission hypothesis linking C to P and O to P. The x-axis is labeled by the number of intermediate host nodes (+ 1). Adjacent unlabeled values have the same M value but a different sub tree structure. The figure shows that the direct hypothesis scores are nearly identical but the alternative hypothesis for M=2 shows that an indirect link for O to P is more plausible. Thus, the explicit scoring of alternative transmission hypotheses involving hypothetical infected farms highlight potential transmission inference ambiguity. In the case of the Cottam tree, the potential for an O to P link is not discussed, however, it appears that this possibility may be precluded due to the structure of the parsimony tree.\\nInferred most recent common ancestor times. Range of inferred most recent common ancestor times (MRCAT) between the B,O sequence pairs (red) and the K,O sequence pairs. Shaded area under the two curves show the time intervals consistent with the respective transmission hypotheses.\\nUse of multiple most recent common ancestor time estimates in addition to transmission time is included in the transmission likelihood calculation.\\nFor farm O, the current method more clearly favors B as the source over K. Figure 11 shows weighted probability density estimates for the most recent common ancestor times for B and O (red) and K and O (blue). The red and blue shaded area under the curves show the range of predicted initial infection times consistent with the MRCAT and the direct transmission hypothesis for B to O and K to O respectively.\\nPortion of the maximum credibility tree. Phylogenetic tree leaves represent sequence taken from four farms: O, C, D, M and P. The putative transmission chain O to C to P to M is overlaid onto the phylogenetic tree, with each farm assigned to an ancestral node.\\nThe red area exceeds the blue area indicating that the inferred MRCA times between B and O are more frequently consistent with the B to O direct transmission hypothesis than the K to O alternative. Similar to the previous case, it appears that the inferred tree used in Cottam may preclude the possibility of a K to B to O transmission path.\\nFarm transmission is modeled to occur instantaneously (with respect to time) and the variable length evolution occurs within the host.\\nThe maximum credibility tree shown in Figure 12 indicates the limitation of using the inferred ancestral nodes as the basis for reconstruction of the outbreak tree. The transmission chain O to C to P to M includes a 0 length branch from C to P that suggests a transmission from farm C to farm P with no genetic variation in the virus, which would be rejected as unlikely given minimum incubation times and the expected mutation rate. In the new method, however, since farm infections are represented by time intervals, the initial infection time of farm C can be shown to occur higher up the tree between two phylogenetic tree nodes, which would imply some mutation to occur on farm C prior to transmission to farm P. Thus, mapping the inferred phylogeny onto the outbreak tree rather than assigning outbreak nodes to phylogenetic tree nodes allows for the consideration of transmission hypotheses that would otherwise be excluded.\\nA strength of the Cottam et al. approach is that the transmission tree is built using all the sequences collectively by iteratively labeling the internal tree nodes rather than treating pairs of sequences independently, thus some transmission hypotheses conceivably could be more readily rejected by looking at multiple farms simultaneously. The new method shows the use of multiple tree topologies in a way that is similar to other recent applications [27] and demonstrates a natural way to incorporate multiple tree topologies in a single sequence comparison procedure. One potential improvement to the Cottam et al. approach would be to include additional candidate tree topologies for evaluation and report likelihood values that are averaged across multiple trees.\\nWhile it may be possible to densely sample from an outbreak in select circumstances, in many cases only sparse sampling is possible and this is likely to be another limitation of the published Cottam et al. approach, which used dense farm sampling. Integrating phylogenetic data with infection timing data to infer more distant transmission linkages, however, is expected to be considerably more difficult. Intuitively, distinguishing between a transmission chain of length 20 from a chain of length 25 is not likely given the range of possible infection times. This is not a limitation of the method but a limitation of exploitable transmission inference knowledge. Nevertheless, a potential contribution of the method presented here is that if it is possible to relax the requirements of an exhaustive search of infection times, the framework could be used to more precisely quantify the uncertainty associated with making inherently more difficult transmission inferences and thus shed light on the conditions where more distant transmission links can be made.\\nTo investigate this problem we introduce two random sampling steps to the algorithm, which when not implemented would otherwise preclude consideration of longer transmission chains. The first sampling step is to draw k different timing samples from the underlying incubation and infectious time duration distributions Pinc and Pinf respectively. For the results described here k was set to 50. The second random sampling step is in the tree traversal procedure illustrated in Figure 5, which is designed to determine whether there is a sequence of timing events for a particular transmission chain and a given collection of infection time durations that is consistent with the inferred most recent common ancestor time. Rather than an exhaustive search, a fixed number of random trials are taken. This approach is thus biased against inferring links requiring rarely occurring timing intervals that would be required to support the most recent common ancestor time. We note that additional sampling strategies could be implemented to make this approach more robust, however, the current modifications should be sufficient for evaluating the potential to apply this approach to larger sparsely sampled outbreaks.\\nFor evaluation a variation on the previously described FMD outbreak simulation is used. Rather than using the smaller fixed tree topology, a tree topology was generated at random starting with an index node and randomly generating child nodes using a branching factor of 1.5, a maximal outbreak duration of 110 days and a maximal node size of 2,000. 100 trees were generated and the 40 trials that generated trees with 2,000 nodes were selected for further analysis. For each tree a pair of nodes was chosen at random to represent each path length present in the tree starting from 1 up to the maximal observed path length. To limit computational costs, only odd path lengths were evaluated.\\nAn example maximum credibility tree for a sparsely sampled simulated FMD outbreak. Tree leaves are labeled with numbers corresponding to nodes taken from a 2,000 node simulated outbreak tree. Select node pairs with known transmission relationships were evaluated. The selected node pairs are labeled by their transmission links using blue edges to link pairs and the path length marked in blue. The two sub trees are connected by a common root (not shown for image compactness).\\nFigure 13 shows an example consensus tree for the selected pairs taken from an outbreak chosen at random. The directed edges labeled in blue show the path lengths connecting the selected nodes with the path length labels from 1 to the maximum observed (23) for this particular outbreak tree. The tree infers evolutionary relationships that are generally consistent with what would be expected given the different transmission chain lengths observed. For example, a path length of 23 separates node 628 from 1740 and indeed the two nodes end up in two different subclades of the tree. The key question becomes can additional information be obtained by incorporating knowledge of infection times and alternative transmission hypothesis testing with the phylogenetic data to better capture the range of possible relationships among the infected farms. To answer this, we first show the likelihood evaluation for the node pairs taken from the tree in Figure 13 and then show the same likelihood evaluations averaged across all 40 randomized outbreak trials. To limit computational costs, evaluated path lengths start with 1 and 5 and continue in intervals of 5 up to 35. The likelihood value P (δ|M) is reported as the maximal likelihood value for all alternative sub trees of path length M. Figure 14 shows the value of P (δ|M) for each pair of nodes with node pairs labeled by their true transmission path length. For the sampled outbreak shown in Figure 13, with the exception of M=11, node pairs for path lengths up to 15 show P (δ|M) values that favor a specific hypothesis or a small subset of hypotheses. As the true path length increases, the average likelihood value drops and the distinction between different path lengths becomes less clear. For the node pair separated by path length of 15 for example, there are clearly higher values for inferring a path length from between 5 and 20, but there is limited distinction within this range. For the most part the framework largely reinforces what is visualized using the consensus phylogenetic tree. An interesting case, however, is the node pair (1255, 1888) linked by a transmission chain of 9, compared with node pair (713, 1119) separated by chain length 13. In the framework, both pairs of nodes have likelihood values that peak at M=5, however the alternative transmission hypotheses for linking (713, 1119) with M=10 or M=15 are much higher than they are for the (1255, 1881) case. This is an example where the relationships inferred by the tree may be misconstrued without the explicit use of a transmission inference framework. It is important to note that the transmission framework does not resolve all conflicts. For the M=11 case for node pair (138, 184), the nodes show up in two different subclades suggesting a more distant transmission relationship, than might be expected by the arrangement of other pairs in the same subclade and separated by longer transmission chains. In this case, the transmission inference framework also suggests a more distant transmission link, given the relatively uniform distribution of low P (δ|M) values.\\nLikelihood values are given for the 12 selected transmission pairs shown in Figure 13. Each node pair is evaluated for different values of M from 1 to 35 and the true path length for each node pair is shown in the legend.\\nLikelihood values for all node pairs for a select path length averaged across all 40 trials. Plot shows path lengths where the average likelihood value peaks at M=5 or less.\\nFigure 15 and Figure 16 show P (δ|M) values averaged over the 40 random trials. For clarity the two figures are used to separate the distributions into two classes, those with peaks at M ≤ 5 and those that peak at M > 5. (Note that the maximal path length observed was 37, however, due to the 2,000 tree node limit, longer path lengths occur less frequently in the 40 trial experiment and thus path lengths with less than 40 observations are not shown.) The results show that the framework assigns a likelihood score that roughly reflects the true underlying path length. In particular, all path lengths less than 10 clearly peak at M=1 or M=5 in the figure. The results do suggest room for improvement, however, since path lengths for M=11-17 still show their maximum likelihood value at M=5. A possible reason for this is that for longer transmission chains the space of inconsistent infection times may expand at a rate that is not proportional to the increase in the number consistent infection times. Since the method uses a fixed random sample size this would favor shorter more constrained transmission hypotheses. Nevertheless, even where the likelihood value peaks at M=5, for longer transmission chains the likelihood values for M > 5 come increasingly closer to the peak until there is finally a peak shift at path length 19 from M=5 to M=10. It is interesting to note that as the path length increases, the peak likelihood value decreases quantifying the increasing ambiguity associated with making long transmission chain inferences. Despite this difficulty, the results indicate that in many cases it may be possible to assign a transmission relationship using a more inclusive class of transmission chain lengths. For example, nodes linked by a transmission path length less than 10 appear to be fairly distinct from those separated by path lengths greater than 20. These distinctions could prove to be of critical importance, particularly when evaluated in the context of an underlying contact network.\\nLikelihood values for node pairs connected by longer path lengths. Plot is similar to Figure 15 but shows path lengths where the average likelihood value peaks at M > 5.\\n', 'Discussion': 'It is important to note that while the presented method is designed to use prior knowledge of viral infection timing patterns, uniform probability distributions were used since an accurate estimate of the distribution is not likely in the general case. More generally since the method requires both viral evolution and epidemiology knowledge there are significant outstanding challenges not addressed in this report. The method makes the simplifying assumption of a neutral evolution model without recombination. This assumption may be appropriate for certain acute viral infection types where host immunity does not play a significant role, however, chronic infections with wider persistent circulation of multiple subtypes increase the chances to misinterpret the transmission process. Additional work is need to ensure that any violation of assumptions of evolution are detected and (ideally) corrected. The model currently lacks inter-species evolution modeling and as sequencing technology advances there will be opportunities to better measure within host viral populations [28] and the affects on the genetic features of the recipient host’s initial seeding population. A consequence of this limited knowledge is reliance on a potentially simplifying feature of viral population structure. When testing direct transmission hypotheses the most recent common ancestor between sequences sampled from the source and recipient are assumed to reside in the source host. However, with a sufficiently large and genetically diverse host population it is at least conceivable that the most recent common ancestor resides in an earlier predecessor node.\\nWith respect to epidemiology modeling, the paper assumes there are accurate values for P (M), which describe the prior knowledge about the relative frequency of the different types of transmission chains. There has been considerable progress in this area in recent years linking contact networks with the more traditional stochastic compartmentalized transmission models to yield estimates [11, 29] that can be inferred using knowledge gleaned from previous outbreaks. Moreover, there are other potentially useful epidemiological models that could be substituted for the SEIR approach presented here [30]. Another important area of development is the use of sequence data and a coalescent model to estimate epidemiological parameters. Recent work demonstrates the potential to estimate parameters for a SIR model from genetic sequence data [31], which could provide an important basis for specifying the epidemiological parameters for transmission inference, when limited non-genome sequence derived observation is available. A strength of the approach is to provide a framework that can readily make use of increasing amounts of knowledge for a particular outbreak. For example, awareness of infections may increase during the course of an outbreak and thus favor using two different priors for the infectious duration times, one at an early outbreak stage and another at a later stage of the outbreak. Other clinical features of infection can be taken into account either by changing prior likelihoods for alternatives in the underlying outbreak tree structure or by making adjustment to priors in the timing distributions. In some cases prior knowledge could be used to narrow the window of possible infection times and greatly strengthen the confidence of transmission inference. In other cases prior knowledge may end up increasing the window of possible infection times and thus highlight the inherent complexity of reconstructing transmission events.\\n', 'Conclusion': 'A new method is presented that takes as input a collection of time-stamped viral genomes and basic knowledge about the timing features of the infection, the underlying transmission network and returns confidence values that assess the probability of correctly inferring host transmission links. Simulations of an FMD outbreak show how the combined use of the epidemiology based timing constraints with the phylogenetic inferred most recent common ancestor times improve confidence in transmission inference over methods that rely on just one individual feature. The simulations show that labeled training data measuring genetic distance between direct and indirect transmission cases can be used without the additional knowledge of the features of the viral transmission, however, appropriate training data in some cases may not be available. The approach presented is expected to provide a basis for building flexible tools to more readily map inferred viral evolutionary relationships onto epidemiological models to more accurately quantify host transmission relationships.\\n', 'Footnotes': 'Email: Jonathan E Allen*- allen99{at}llnl.gov; Stephan P Velsko - velsko2{at}llnl.gov;\\n', 'Reference': ''}\n"
     ]
    }
   ],
   "source": [
    "sections = {}\n",
    "for section in tmp_soup.find_all('div',attrs={'class':'section'}):\n",
    "    section_header = section.find('h2')\n",
    "    if section_header:\n",
    "        section_name = section_header.text\n",
    "    \n",
    "    if section_header and section_name != 'References' and section_name != 'Acknowledgments':\n",
    "        #print(section_name)\n",
    "        sections[section_name] = ''\n",
    "        for p in section.find_all('p'):\n",
    "            sections[section_name] += p.text + '\\n'\n",
    "        #print(sections[section_name])\n",
    "        #print()\n",
    "print(sections)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'T1', 'label': 'Table 1.', 'description': 'Summary of methods for metagenomic classification.\\n', 'image': None}\n",
      "{'id': 'T2', 'label': 'Table 2.', 'caption': 'The abundance of each target organism in each set of simulated datasets. Each set is indicated by the number in the top row, and was generated with 50 replicates.', 'description': '', 'image': None}\n"
     ]
    }
   ],
   "source": [
    "# get tables\n",
    "tables = []\n",
    "html_tables = tmp_soup.find_all('div',attrs={'class':'table'})\n",
    "for html_table in html_tables:\n",
    "    table = {}\n",
    "    table['id'] = html_table['id']\n",
    "    label = html_table.find('span',attrs={'class':'table-label'})\n",
    "    if label:\n",
    "        table['label'] = label.text\n",
    "    caption = html_table.find('span',attrs={'class':'caption-title'})\n",
    "    if caption:\n",
    "        table['caption'] = caption.text\n",
    "    table['description'] = ''\n",
    "    for p in html_table.find_all('p'):\n",
    "        table['description'] += p.text + '\\n'\n",
    "    table['image'] = None\n",
    "    tables.append(table)\n",
    "\n",
    "for table in tables:\n",
    "    print(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<p class=\"first-child\" id=\"p-9\">A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.</p>]"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "html_figures[0].find_all('p')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Scrape papers from medrxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# get papers from data/medrxiv\n",
    "existing_paper_ids = set([name.strip('.json') for name in get_file_names('data/medrxiv')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Scrape papers from CZI\n",
    "Journals: 1198"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "collapsed": true,
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c630ebcdf30652f0422c3ec12a00b50241dc9bd9</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) as a SA...</td>\n",
       "      <td>10.1007/s00134-020-05985-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125455.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>2.002765e+09</td>\n",
       "      <td>#3252</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>53eccda7977a31e3d0f565c884da036b1e85438e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comparative genetic analysis of the novel coro...</td>\n",
       "      <td>10.1038/s41421-020-0147-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...</td>\n",
       "      <td>Cell Discovery</td>\n",
       "      <td>3.003431e+09</td>\n",
       "      <td>#1861</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>210a892deb1c61577f6fba58505fd65356ce6636</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Incubation Period and Other Epidemiological Ch...</td>\n",
       "      <td>10.3390/jcm9020538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The geographic spread of 2019 novel coronaviru...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...</td>\n",
       "      <td>Journal of Clinical Medicine</td>\n",
       "      <td>3.006065e+09</td>\n",
       "      <td>#1043</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>e3b40cc8e0e137c416b4a2273a4dca94ae8178cc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Characteristics of and Public Health Responses...</td>\n",
       "      <td>10.3390/jcm9020575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32093211.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, cases of unidentified pneumo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Deng, Sheng-Qun; Peng, Hong-Juan</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>1.776631e+08</td>\n",
       "      <td>#1999</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92c2c9839304b4f2bc1276d41b1aa885d8b364fd</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Imaging changes in severe COVID-19 pneumonia</td>\n",
       "      <td>10.1007/s00134-020-05976-w</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125453.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Wei</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>3.006643e+09</td>\n",
       "      <td>#3242</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0df0d5270a9399cf4e23c0cdd877a80616a9725e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>An updated estimation of the risk of transmiss...</td>\n",
       "      <td>10.1016/j.idm.2020.02.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The basic reproduction number of an infectious...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#729</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>f24242580be243d5fc3f432915d86af6854bb8b7</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Real-time forecasts of the 2019-nCoV epidemic ...</td>\n",
       "      <td>10.1016/j.idm.2020.02.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The initial cluster of severe pneumonia cases ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...</td>\n",
       "      <td>Infectious Disease Modelling</td>\n",
       "      <td>3.006029e+09</td>\n",
       "      <td>#865</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>d13a685f861b0f1ba05afa6e005311ad1820fd3a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>RETRACTED: Chinese medical staff request inter...</td>\n",
       "      <td>10.1016/s2214-109x(20)30065-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32105614.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zeng, Yingchun; Zhen, Yan</td>\n",
       "      <td>The Lancet. Global health</td>\n",
       "      <td>2.627046e+09</td>\n",
       "      <td>#5386</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19 outbreak on the Diamond Princess crui...</td>\n",
       "      <td>10.1093/jtm/taaa030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Cruise ships carry a large number of people in...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.</td>\n",
       "      <td>Journal of Travel Medicine</td>\n",
       "      <td>3.006304e+09</td>\n",
       "      <td>#2926</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>e9239100c5493ea914dc23c3d7a262f4326022ac</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Distinct Roles for Sialoside and Protein Recep...</td>\n",
       "      <td>10.1128/mBio.02764-19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Coronaviruses (CoVs) are common human and anim...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Qing, Enya; Hantak, Michael; Perlman, Stanley;...</td>\n",
       "      <td>mBio</td>\n",
       "      <td>3.005811e+09</td>\n",
       "      <td>#2427</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>469ed0f00c09e2637351c9735c306f27acf3aace</td>\n",
       "      <td>CZI</td>\n",
       "      <td>First two months of the 2019 Coronavirus Disea...</td>\n",
       "      <td>10.1186/s41256-020-00137-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Similar to outbreaks of many other infectious ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Xinguang; Yu, Bin</td>\n",
       "      <td>Global Health Research and Policy</td>\n",
       "      <td>3.006646e+09</td>\n",
       "      <td>#5595</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4e550e034ccca6fa2a91e481ddba24db67bc9ae5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness of airport screening at detectin...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.5.2000080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>We simulated 100 2019-nCoV infected travellers...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Quilty, Billy J; Clifford, Sam; group2, CMMID ...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005152e+09</td>\n",
       "      <td>#682</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>4bbb0c59babc718f67953fae032dad6ae0d7aeb1</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Genome Detective Coronavirus Typing Tool for r...</td>\n",
       "      <td>10.1093/bioinformatics/btaa145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32108862.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>SUMMARY: Genome Detective is a web-based, user...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...</td>\n",
       "      <td>Bioinformatics (Oxford, England)</td>\n",
       "      <td>3.003549e+09</td>\n",
       "      <td>#2776</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>c821803c55c6aad89b6d0c1d3ba252051e464017</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Case of the Index Patient Who Caused Tertiary ...</td>\n",
       "      <td>10.3346/jkms.2020.35.e79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>Since mid-December of 2019, coronavirus diseas...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...</td>\n",
       "      <td>Journal of Korean Medical Science</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#3770</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>c3bee2a4caca614b34f92c17b643b854dcdab28d</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Emergence of Novel Coronavirus 2019-nCoV: Need...</td>\n",
       "      <td>10.3390/pathogens9020148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Novel Coronavirus (2019-nCoV) is an emerging p...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...</td>\n",
       "      <td>Pathogens</td>\n",
       "      <td>3.003886e+09</td>\n",
       "      <td>#1502</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>715842fa536064980818ad7e31ce511272a4b6bc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The coronavirus 2019-nCoV epidemic: Is hindsig...</td>\n",
       "      <td>10.1016/j.eclinm.2020.100289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...</td>\n",
       "      <td>EClinicalMedicine</td>\n",
       "      <td>3.004913e+09</td>\n",
       "      <td>#3333</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Nonstructural proteins NS7b and NS8 are likely...</td>\n",
       "      <td>10.1016/j.meegid.2020.104272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>The seventh novel human infecting Betacoronavi...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro</td>\n",
       "      <td>Infection, Genetics and Evolution</td>\n",
       "      <td>2.286137e+09</td>\n",
       "      <td>#4581</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>06c89f69aa7b5f9648d2c1543b8246fe9c3610cf</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Pathogenicity and Transmissibility of 2019-nCo...</td>\n",
       "      <td>10.1016/j.micinf.2020.01.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>A zoonotic coronavirus, labeled as 2019-nCoV b...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chen, Jieliang</td>\n",
       "      <td>Microbes and Infection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#236</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>acb678bdd7634055de18d0b89bb6a4890e6a0306</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Remdesivir and chloroquine effectively inhibit...</td>\n",
       "      <td>10.1038/s41422-020-0282-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...</td>\n",
       "      <td>Cell Research</td>\n",
       "      <td>3.005213e+09</td>\n",
       "      <td>#205</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>47772f3e98d8c61bb9782531a0338ba85f27bf3f</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential for global spread of a novel coronav...</td>\n",
       "      <td>10.1093/jtm/taaa011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...</td>\n",
       "      <td>Journal of travel medicine</td>\n",
       "      <td>3.004048e+09</td>\n",
       "      <td>#102</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Severe acute respiratory symptoms and suspecte...</td>\n",
       "      <td>10.12809/hkmj208375</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Hon, K. L.; Leung, K. K. Y.</td>\n",
       "      <td>Hong Kong Medical Journal</td>\n",
       "      <td>2.029408e+09</td>\n",
       "      <td>#3395</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4ff89a71126d2932544a8337ba28787fde5f02a8</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Statistics-Based Predictions of Coronavirus Ep...</td>\n",
       "      <td>10.20535/ibb.2020.4.1.195074</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Information about the open-access article 'Sta...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Nesteruk, Igor</td>\n",
       "      <td>Innovative Biosystems and Bioengineering</td>\n",
       "      <td>3.006305e+09</td>\n",
       "      <td>#1218</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ac37ba61c91bb6939a507dbf4efe2119ddafeb9c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Early transmission patterns of coronavirus dis...</td>\n",
       "      <td>10.2807/1560-7917.ES.2020.25.8.2000097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...</td>\n",
       "      <td>Eurosurveillance</td>\n",
       "      <td>3.005657e+09</td>\n",
       "      <td>#2582</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0</td>\n",
       "      <td>CZI</td>\n",
       "      <td>HRCT Imaging Features in Representative Import...</td>\n",
       "      <td>10.1093/pcmedi/pbaa004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>With the spread of novel coronavirus (2019-nCo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...</td>\n",
       "      <td>Precision Clinical Medicine</td>\n",
       "      <td>3.006468e+09</td>\n",
       "      <td>#781</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>c9fee561c2a3834645dbb61dc4ae6448051da492</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comprehensive Genomic Characterization Analysi...</td>\n",
       "      <td>10.3389/fmicb.2019.03036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Porcine delta coronavirus (PDCoV) is a novel e...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...</td>\n",
       "      <td>Frontiers in Microbiology</td>\n",
       "      <td>3.003968e+09</td>\n",
       "      <td>#5462</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>655537fc8cc52bccf43cf7189ab060d3097caa7a</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Potential Factors Influencing Repeated SARS Ou...</td>\n",
       "      <td>10.3390/ijerph17051633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Within last 17 years two widespread epidemics ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....</td>\n",
       "      <td>International Journal of Environmental Researc...</td>\n",
       "      <td>2.615949e+09</td>\n",
       "      <td>#3296</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>bb7cd48a85b72f8d73e57d655c6d8a034fce3778</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Puzzle of highly pathogenic human coronaviruse...</td>\n",
       "      <td>10.1007/s13238-020-00693-y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32088858.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Li, Jing; Liu, Wenjun</td>\n",
       "      <td>Protein Cell</td>\n",
       "      <td>2.766462e+09</td>\n",
       "      <td>#1815</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3c42431c5b95e707486ce7441ac139071e07b706</td>\n",
       "      <td>CZI</td>\n",
       "      <td>RETRACTION: Retraction-Chinese medical staff r...</td>\n",
       "      <td>10.1016/s2214-109x(20)30076-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32113504.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>The Editors Of The Lancet Global, Health</td>\n",
       "      <td>The Lancet. Global health</td>\n",
       "      <td>2.794879e+09</td>\n",
       "      <td>#5486</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>6a93283b499ae5bc6aaf29f14e701dc8f25138ea</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Critical care response to a hospital outbreak ...</td>\n",
       "      <td>10.1186/s13054-020-2786-x</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liu, Yong; Li, Jinxiu; Feng, Yongwen</td>\n",
       "      <td>Critical Care</td>\n",
       "      <td>2.537201e+09</td>\n",
       "      <td>#1225</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>6fc52ed878c271020a2a375bb6e4b12943a7666c</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Effectiveness for the Response to COVID-19: Th...</td>\n",
       "      <td>10.24171/j.phrp.2020.11.1.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>In the current issue of Osong Public Health an...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cho, Hae-Wol</td>\n",
       "      <td>Osong Public Health and Research Perspectives</td>\n",
       "      <td>3.005888e+09</td>\n",
       "      <td>#2662</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1206</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Daily briefing: World’s biggest physics meetin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The March Meeting of the APS was cancelled jus...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Graham, Flora</td>\n",
       "      <td>Nature</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#3172</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1207</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>How Coronaviruses Cause Infection—from Colds t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The novel coronavirus outbreak raises question...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Makin, Simon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#294</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1208</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Advices on the prevention and control of nosoc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The pneumonia caused by the novel coronavirus ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XU, Hongzhen; CHEN, Shuohui; FU, Junfen; SHU, ...</td>\n",
       "      <td>Chinese Journal of Hospital Administration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#2106</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1209</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>‘This beast is moving very fast.’ Will the new...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The silver lining of the epidemic is that scie...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cohen, Kai Kupferschmidt; Jon</td>\n",
       "      <td>Science</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#332</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1210</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo Coronavírus: atendimento a pessoas com su...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Brasil. Ministerio da, Saude</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#1866</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1211</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo coronavirus (2019 nCov) Medidas de preven...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>São Paulo Secretaria da, Saúde</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#922</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1212</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The prevention and control of a new coronaviru...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32057210.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>During a short period of time, the outbreak of...</td>\n",
       "      <td>2020</td>\n",
       "      <td>LI, Zhi yong; MENG, Liu yan</td>\n",
       "      <td>Chinese Journal of Stomatology</td>\n",
       "      <td>2.377782e+09</td>\n",
       "      <td>#1228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>COVID-19: another infectious disease emerging ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32078596.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Murdoch, David R.; French, Nigel P.</td>\n",
       "      <td>N Z Med J</td>\n",
       "      <td>2.185518e+09</td>\n",
       "      <td>#1541</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Dynamic changes of chest CT imaging in patient...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32096366.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OBJECTIVE: To analyze the dynamic changes of c...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Wang, Jincheng; Liu, Jinpeng; Wang, Yuanyuan; ...</td>\n",
       "      <td>Zhejiang Da Xue Xue Bao Yi Xue Ban</td>\n",
       "      <td>2.349103e+09</td>\n",
       "      <td>#1922</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1215</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus COVID-19: an overview for em...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32105049.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prior to the global outbreak of SARS-CoV in 20...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Giwa, A.; Desai, A.</td>\n",
       "      <td>Emergency medicine practice</td>\n",
       "      <td>3.005414e+09</td>\n",
       "      <td>#2815</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Coronavirus 9-nCoV: recomendaciones para aerop...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ante la situación mundial, en relación a 2019-...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Argentina. Ministerio de, Salud</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.604377e+09</td>\n",
       "      <td>#801</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel Coronavirus Disease 2019 (COVID-19): An ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Background: At the beginning of the New Year 2...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Keshavarz, Ahmad Tavakoli; Katayon, Vahdat; Mo...</td>\n",
       "      <td>Iranian South Medical Journal</td>\n",
       "      <td>6.213808e+08</td>\n",
       "      <td>#4536</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Coronavirus just caused the American Physical ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Citing the growing threat of the coronavirus, ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cho, Adrian</td>\n",
       "      <td>Science Magazine</td>\n",
       "      <td>2.595675e+09</td>\n",
       "      <td>#2772</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Mensajero de la Salud: nuevo coronavirus 2019-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Declaratoria de Emergencia en Salud Pública de...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Mexico. Secretaria de la, Salud</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.003591e+09</td>\n",
       "      <td>#1803</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The management of biosafety risk in clinical l...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>During the outbreak of coronavirus disease-19 ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XIAO, Yuling; LU, Xiaojun; KANG, Mei; LI, Dong...</td>\n",
       "      <td>Chinese Journal of Laboratory Medicine</td>\n",
       "      <td>2.194450e+09</td>\n",
       "      <td>#2110</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Lineamiento estandarizado para la Vigilancia E...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Este documento describe la situación epidemiol...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Mexico. Secretaria de Salud. Subsecretaria de ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.241123e+09</td>\n",
       "      <td>#1802</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Singapore claims first use of antibody test to...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In what appears to be a first, disease tracker...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Normile, Dennis</td>\n",
       "      <td>Science Magazine</td>\n",
       "      <td>2.855668e+09</td>\n",
       "      <td>#2584</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>The network investigation on knowledge, attitu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To analyze the current situation of ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>CHEN, Yan</td>\n",
       "      <td>Chinese Journal of Preventive Medicine</td>\n",
       "      <td>2.363411e+09</td>\n",
       "      <td>#2333</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Investigation on demands for antenatal care se...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To identify problems and demands for...</td>\n",
       "      <td>2020</td>\n",
       "      <td>DU, Li; GU, Yibin; CUI, Mengqing; LI, Wenxian;...</td>\n",
       "      <td>Chinese Journal of Obstetrics and Gynecology</td>\n",
       "      <td>1.891210e+09</td>\n",
       "      <td>#2305</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1225</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Analysis of early chest high resolution CT ima...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Objective To investigate the first chest HRCT ...</td>\n",
       "      <td>2020</td>\n",
       "      <td>LIU, Haifeng; ZHANG, Dongyou; YANG, Yi; LONG, ...</td>\n",
       "      <td>Chinese Journal of Radiology</td>\n",
       "      <td>2.418494e+09</td>\n",
       "      <td>#2187</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1226</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novo Coronavirus (nCoV)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Os Coronavírus (CoV) compõem uma grande famíli...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rio de Janeiro . Secretaria de Estado de Saúde...</td>\n",
       "      <td>Nota Técnica-SVS/SES-RJ</td>\n",
       "      <td>2.518955e+09</td>\n",
       "      <td>#216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1227</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Difficulties and strategies of public hospital...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Outbreak of the novel coronavirus pneumonia (N...</td>\n",
       "      <td>2020</td>\n",
       "      <td>XU, Dong; HU, Yu; DING, Ning; XIA, Jiahong; ZH...</td>\n",
       "      <td>Chinese Journal of Hospital Administration</td>\n",
       "      <td>2.348898e+09</td>\n",
       "      <td>#2108</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Get Real</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Since this is going to be a post about the cor...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Lowe, Derek</td>\n",
       "      <td>Science</td>\n",
       "      <td>2.343732e+09</td>\n",
       "      <td>#3808</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>How Ophthalmologists Should Understand and Res...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The new coronavirus pneumonia that first appea...</td>\n",
       "      <td>2020</td>\n",
       "      <td>ZHIJIE, Li</td>\n",
       "      <td>Chinese Journal of Experimental Ophthalmology</td>\n",
       "      <td>3.005943e+09</td>\n",
       "      <td>#2051</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1230</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Experts proposal and frequently asked question...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The outbreak of novel coronavirus pneumonia (N...</td>\n",
       "      <td>2020</td>\n",
       "      <td>ZHANG, Lei; CAO, Qing; WANG, Ying; LU, Quan; H...</td>\n",
       "      <td>Chinese Journal of Applied Clinical Pediatrics</td>\n",
       "      <td>2.416639e+09</td>\n",
       "      <td>#2063</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1231</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Surgical treatment strategy for digestive syst...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The outbreak of novel coronavirus pneumonia oc...</td>\n",
       "      <td>2020</td>\n",
       "      <td>MA, Fuhai; HU, Haitao; TIAN, Yantao</td>\n",
       "      <td>Chinese Journal of Oncology</td>\n",
       "      <td>2.366820e+09</td>\n",
       "      <td>#2176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1232</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Novel coronavirus pneumonia in the primary gen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>To observe the curative effect of TCM syndrome...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Liao, Rongye; Yang, Jie; Cao, Zhi; Wang, Jun</td>\n",
       "      <td>International Journal of Traditional Chinese M...</td>\n",
       "      <td>2.616653e+09</td>\n",
       "      <td>#1962</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>A Novel Coronavirus Outbreak from Wuhan City i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Alavi-Moghaddam, Mostafa</td>\n",
       "      <td>Archives of Academic Emergency Medicine</td>\n",
       "      <td>3.006237e+09</td>\n",
       "      <td>#3509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1234</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>History is repeating itself: Probable zoonotic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Rodriguez-Morales, A. J.; Bonilla-Aldana, D. K...</td>\n",
       "      <td>Infezioni in Medicina</td>\n",
       "      <td>3.005681e+09</td>\n",
       "      <td>#926</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1235</th>\n",
       "      <td>NaN</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Epidemiological and Clinical Characteristics o...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chen, Nanshan and Zhou, Min and Dong, Xuan and...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.004581e+09</td>\n",
       "      <td>#22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1236 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           sha source_x  \\\n",
       "0     c630ebcdf30652f0422c3ec12a00b50241dc9bd9      CZI   \n",
       "1     53eccda7977a31e3d0f565c884da036b1e85438e      CZI   \n",
       "2     210a892deb1c61577f6fba58505fd65356ce6636      CZI   \n",
       "3     e3b40cc8e0e137c416b4a2273a4dca94ae8178cc      CZI   \n",
       "4     92c2c9839304b4f2bc1276d41b1aa885d8b364fd      CZI   \n",
       "5     0df0d5270a9399cf4e23c0cdd877a80616a9725e      CZI   \n",
       "6     f24242580be243d5fc3f432915d86af6854bb8b7      CZI   \n",
       "7     d13a685f861b0f1ba05afa6e005311ad1820fd3a      CZI   \n",
       "8     e1b336d8be1a4c0ccc5a1bf41e48b3b004d3ece1      CZI   \n",
       "9     e9239100c5493ea914dc23c3d7a262f4326022ac      CZI   \n",
       "10    469ed0f00c09e2637351c9735c306f27acf3aace      CZI   \n",
       "11    4e550e034ccca6fa2a91e481ddba24db67bc9ae5      CZI   \n",
       "12    4bbb0c59babc718f67953fae032dad6ae0d7aeb1      CZI   \n",
       "13    c821803c55c6aad89b6d0c1d3ba252051e464017      CZI   \n",
       "14    c3bee2a4caca614b34f92c17b643b854dcdab28d      CZI   \n",
       "15    715842fa536064980818ad7e31ce511272a4b6bc      CZI   \n",
       "16    601d3c7ae4ebcd2d835e9cf8d7427ebd0b5db83f      CZI   \n",
       "17    06c89f69aa7b5f9648d2c1543b8246fe9c3610cf      CZI   \n",
       "18    acb678bdd7634055de18d0b89bb6a4890e6a0306      CZI   \n",
       "19    47772f3e98d8c61bb9782531a0338ba85f27bf3f      CZI   \n",
       "20    544a170a18bae9bf44c531c2b0b4bdc5e85ea8f5      CZI   \n",
       "21    4ff89a71126d2932544a8337ba28787fde5f02a8      CZI   \n",
       "22    ac37ba61c91bb6939a507dbf4efe2119ddafeb9c      CZI   \n",
       "23    6dfe2e77c8175b3eb38d6fc3ccc440fb81a0e2f0      CZI   \n",
       "24    c9fee561c2a3834645dbb61dc4ae6448051da492      CZI   \n",
       "25    655537fc8cc52bccf43cf7189ab060d3097caa7a      CZI   \n",
       "26    bb7cd48a85b72f8d73e57d655c6d8a034fce3778      CZI   \n",
       "27    3c42431c5b95e707486ce7441ac139071e07b706      CZI   \n",
       "28    6a93283b499ae5bc6aaf29f14e701dc8f25138ea      CZI   \n",
       "29    6fc52ed878c271020a2a375bb6e4b12943a7666c      CZI   \n",
       "...                                        ...      ...   \n",
       "1206                                       NaN      CZI   \n",
       "1207                                       NaN      CZI   \n",
       "1208                                       NaN      CZI   \n",
       "1209                                       NaN      CZI   \n",
       "1210                                       NaN      CZI   \n",
       "1211                                       NaN      CZI   \n",
       "1212                                       NaN      CZI   \n",
       "1213                                       NaN      CZI   \n",
       "1214                                       NaN      CZI   \n",
       "1215                                       NaN      CZI   \n",
       "1216                                       NaN      CZI   \n",
       "1217                                       NaN      CZI   \n",
       "1218                                       NaN      CZI   \n",
       "1219                                       NaN      CZI   \n",
       "1220                                       NaN      CZI   \n",
       "1221                                       NaN      CZI   \n",
       "1222                                       NaN      CZI   \n",
       "1223                                       NaN      CZI   \n",
       "1224                                       NaN      CZI   \n",
       "1225                                       NaN      CZI   \n",
       "1226                                       NaN      CZI   \n",
       "1227                                       NaN      CZI   \n",
       "1228                                       NaN      CZI   \n",
       "1229                                       NaN      CZI   \n",
       "1230                                       NaN      CZI   \n",
       "1231                                       NaN      CZI   \n",
       "1232                                       NaN      CZI   \n",
       "1233                                       NaN      CZI   \n",
       "1234                                       NaN      CZI   \n",
       "1235                                       NaN      CZI   \n",
       "\n",
       "                                                  title  \\\n",
       "0     Angiotensin-converting enzyme 2 (ACE2) as a SA...   \n",
       "1     Comparative genetic analysis of the novel coro...   \n",
       "2     Incubation Period and Other Epidemiological Ch...   \n",
       "3     Characteristics of and Public Health Responses...   \n",
       "4          Imaging changes in severe COVID-19 pneumonia   \n",
       "5     An updated estimation of the risk of transmiss...   \n",
       "6     Real-time forecasts of the 2019-nCoV epidemic ...   \n",
       "7     RETRACTED: Chinese medical staff request inter...   \n",
       "8     COVID-19 outbreak on the Diamond Princess crui...   \n",
       "9     Distinct Roles for Sialoside and Protein Recep...   \n",
       "10    First two months of the 2019 Coronavirus Disea...   \n",
       "11    Effectiveness of airport screening at detectin...   \n",
       "12    Genome Detective Coronavirus Typing Tool for r...   \n",
       "13    Case of the Index Patient Who Caused Tertiary ...   \n",
       "14    Emergence of Novel Coronavirus 2019-nCoV: Need...   \n",
       "15    The coronavirus 2019-nCoV epidemic: Is hindsig...   \n",
       "16    Nonstructural proteins NS7b and NS8 are likely...   \n",
       "17    Pathogenicity and Transmissibility of 2019-nCo...   \n",
       "18    Remdesivir and chloroquine effectively inhibit...   \n",
       "19    Potential for global spread of a novel coronav...   \n",
       "20    Severe acute respiratory symptoms and suspecte...   \n",
       "21    Statistics-Based Predictions of Coronavirus Ep...   \n",
       "22    Early transmission patterns of coronavirus dis...   \n",
       "23    HRCT Imaging Features in Representative Import...   \n",
       "24    Comprehensive Genomic Characterization Analysi...   \n",
       "25    Potential Factors Influencing Repeated SARS Ou...   \n",
       "26    Puzzle of highly pathogenic human coronaviruse...   \n",
       "27    RETRACTION: Retraction-Chinese medical staff r...   \n",
       "28    Critical care response to a hospital outbreak ...   \n",
       "29    Effectiveness for the Response to COVID-19: Th...   \n",
       "...                                                 ...   \n",
       "1206  Daily briefing: World’s biggest physics meetin...   \n",
       "1207  How Coronaviruses Cause Infection—from Colds t...   \n",
       "1208  Advices on the prevention and control of nosoc...   \n",
       "1209  ‘This beast is moving very fast.’ Will the new...   \n",
       "1210  Novo Coronavírus: atendimento a pessoas com su...   \n",
       "1211  Novo coronavirus (2019 nCov) Medidas de preven...   \n",
       "1212  The prevention and control of a new coronaviru...   \n",
       "1213  COVID-19: another infectious disease emerging ...   \n",
       "1214  Dynamic changes of chest CT imaging in patient...   \n",
       "1215  Novel coronavirus COVID-19: an overview for em...   \n",
       "1216  Coronavirus 9-nCoV: recomendaciones para aerop...   \n",
       "1217  Novel Coronavirus Disease 2019 (COVID-19): An ...   \n",
       "1218  Coronavirus just caused the American Physical ...   \n",
       "1219  Mensajero de la Salud: nuevo coronavirus 2019-...   \n",
       "1220  The management of biosafety risk in clinical l...   \n",
       "1221  Lineamiento estandarizado para la Vigilancia E...   \n",
       "1222  Singapore claims first use of antibody test to...   \n",
       "1223  The network investigation on knowledge, attitu...   \n",
       "1224  Investigation on demands for antenatal care se...   \n",
       "1225  Analysis of early chest high resolution CT ima...   \n",
       "1226                            Novo Coronavirus (nCoV)   \n",
       "1227  Difficulties and strategies of public hospital...   \n",
       "1228                                           Get Real   \n",
       "1229  How Ophthalmologists Should Understand and Res...   \n",
       "1230  Experts proposal and frequently asked question...   \n",
       "1231  Surgical treatment strategy for digestive syst...   \n",
       "1232  Novel coronavirus pneumonia in the primary gen...   \n",
       "1233  A Novel Coronavirus Outbreak from Wuhan City i...   \n",
       "1234  History is repeating itself: Probable zoonotic...   \n",
       "1235  Epidemiological and Clinical Characteristics o...   \n",
       "\n",
       "                                         doi pmcid   pubmed_id      license  \\\n",
       "0                 10.1007/s00134-020-05985-9   NaN  32125455.0     cc-by-nc   \n",
       "1                  10.1038/s41421-020-0147-1   NaN         NaN        cc-by   \n",
       "2                         10.3390/jcm9020538   NaN         NaN        cc-by   \n",
       "3                         10.3390/jcm9020575   NaN  32093211.0        cc-by   \n",
       "4                 10.1007/s00134-020-05976-w   NaN  32125453.0     cc-by-nc   \n",
       "5                  10.1016/j.idm.2020.02.001   NaN         NaN  cc-by-nc-nd   \n",
       "6                  10.1016/j.idm.2020.02.002   NaN         NaN  cc-by-nc-nd   \n",
       "7              10.1016/s2214-109x(20)30065-6   NaN  32105614.0        cc-by   \n",
       "8                        10.1093/jtm/taaa030   NaN         NaN     cc-by-nc   \n",
       "9                      10.1128/mBio.02764-19   NaN         NaN        cc-by   \n",
       "10                10.1186/s41256-020-00137-4   NaN         NaN        cc-by   \n",
       "11    10.2807/1560-7917.ES.2020.25.5.2000080   NaN         NaN        cc-by   \n",
       "12            10.1093/bioinformatics/btaa145   NaN  32108862.0     cc-by-nc   \n",
       "13                  10.3346/jkms.2020.35.e79   NaN         NaN     cc-by-nc   \n",
       "14                  10.3390/pathogens9020148   NaN         NaN        cc-by   \n",
       "15              10.1016/j.eclinm.2020.100289   NaN         NaN  cc-by-nc-nd   \n",
       "16              10.1016/j.meegid.2020.104272   NaN         NaN  cc-by-nc-nd   \n",
       "17              10.1016/j.micinf.2020.01.004   NaN         NaN  cc-by-nc-nd   \n",
       "18                 10.1038/s41422-020-0282-0   NaN         NaN        cc-by   \n",
       "19                       10.1093/jtm/taaa011   NaN         NaN     cc-by-nc   \n",
       "20                       10.12809/hkmj208375   NaN         NaN  cc-by-nc-nd   \n",
       "21              10.20535/ibb.2020.4.1.195074   NaN         NaN        cc-by   \n",
       "22    10.2807/1560-7917.ES.2020.25.8.2000097   NaN         NaN        cc-by   \n",
       "23                    10.1093/pcmedi/pbaa004   NaN         NaN     cc-by-nc   \n",
       "24                  10.3389/fmicb.2019.03036   NaN         NaN        cc-by   \n",
       "25                    10.3390/ijerph17051633   NaN         NaN        cc-by   \n",
       "26                10.1007/s13238-020-00693-y   NaN  32088858.0        cc-by   \n",
       "27             10.1016/s2214-109x(20)30076-0   NaN  32113504.0        cc-by   \n",
       "28                 10.1186/s13054-020-2786-x   NaN         NaN        cc-by   \n",
       "29              10.24171/j.phrp.2020.11.1.01   NaN         NaN  cc-by-nc-nd   \n",
       "...                                      ...   ...         ...          ...   \n",
       "1206                                     NaN   NaN         NaN          NaN   \n",
       "1207                                     NaN   NaN         NaN          NaN   \n",
       "1208                                     NaN   NaN         NaN          NaN   \n",
       "1209                                     NaN   NaN         NaN          NaN   \n",
       "1210                                     NaN   NaN         NaN          NaN   \n",
       "1211                                     NaN   NaN         NaN          NaN   \n",
       "1212                                     NaN   NaN  32057210.0          NaN   \n",
       "1213                                     NaN   NaN  32078596.0          NaN   \n",
       "1214                                     NaN   NaN  32096366.0          NaN   \n",
       "1215                                     NaN   NaN  32105049.0          NaN   \n",
       "1216                                     NaN   NaN         NaN          NaN   \n",
       "1217                                     NaN   NaN         NaN          NaN   \n",
       "1218                                     NaN   NaN         NaN          NaN   \n",
       "1219                                     NaN   NaN         NaN          NaN   \n",
       "1220                                     NaN   NaN         NaN          NaN   \n",
       "1221                                     NaN   NaN         NaN          NaN   \n",
       "1222                                     NaN   NaN         NaN          NaN   \n",
       "1223                                     NaN   NaN         NaN          NaN   \n",
       "1224                                     NaN   NaN         NaN          NaN   \n",
       "1225                                     NaN   NaN         NaN          NaN   \n",
       "1226                                     NaN   NaN         NaN          NaN   \n",
       "1227                                     NaN   NaN         NaN          NaN   \n",
       "1228                                     NaN   NaN         NaN          NaN   \n",
       "1229                                     NaN   NaN         NaN          NaN   \n",
       "1230                                     NaN   NaN         NaN          NaN   \n",
       "1231                                     NaN   NaN         NaN          NaN   \n",
       "1232                                     NaN   NaN         NaN          NaN   \n",
       "1233                                     NaN   NaN         NaN          NaN   \n",
       "1234                                     NaN   NaN         NaN          NaN   \n",
       "1235                                     NaN   NaN         NaN          NaN   \n",
       "\n",
       "                                               abstract publish_time  \\\n",
       "0                                                   NaN         2020   \n",
       "1                                                   NaN         2020   \n",
       "2     The geographic spread of 2019 novel coronaviru...         2020   \n",
       "3     In December 2019, cases of unidentified pneumo...         2020   \n",
       "4                                                   NaN         2020   \n",
       "5     The basic reproduction number of an infectious...         2020   \n",
       "6     The initial cluster of severe pneumonia cases ...         2020   \n",
       "7                                                   NaN         2020   \n",
       "8     Cruise ships carry a large number of people in...         2020   \n",
       "9     Coronaviruses (CoVs) are common human and anim...         2020   \n",
       "10    Similar to outbreaks of many other infectious ...         2020   \n",
       "11    We simulated 100 2019-nCoV infected travellers...         2020   \n",
       "12    SUMMARY: Genome Detective is a web-based, user...         2020   \n",
       "13    Since mid-December of 2019, coronavirus diseas...         2020   \n",
       "14    Novel Coronavirus (2019-nCoV) is an emerging p...         2020   \n",
       "15                                                  NaN         2020   \n",
       "16    The seventh novel human infecting Betacoronavi...         2020   \n",
       "17    A zoonotic coronavirus, labeled as 2019-nCoV b...         2020   \n",
       "18                                                  NaN         2020   \n",
       "19                                                  NaN         2020   \n",
       "20                                                  NaN         2020   \n",
       "21    Information about the open-access article 'Sta...         2020   \n",
       "22                                                  NaN         2020   \n",
       "23    With the spread of novel coronavirus (2019-nCo...         2020   \n",
       "24    Porcine delta coronavirus (PDCoV) is a novel e...         2020   \n",
       "25    Within last 17 years two widespread epidemics ...         2020   \n",
       "26                                                  NaN         2020   \n",
       "27                                                  NaN         2020   \n",
       "28                                                  NaN         2020   \n",
       "29    In the current issue of Osong Public Health an...         2020   \n",
       "...                                                 ...          ...   \n",
       "1206  The March Meeting of the APS was cancelled jus...         2020   \n",
       "1207  The novel coronavirus outbreak raises question...         2020   \n",
       "1208  The pneumonia caused by the novel coronavirus ...         2020   \n",
       "1209  The silver lining of the epidemic is that scie...         2020   \n",
       "1210                                                NaN         2020   \n",
       "1211                                                NaN         2020   \n",
       "1212  During a short period of time, the outbreak of...         2020   \n",
       "1213                                                NaN         2020   \n",
       "1214  OBJECTIVE: To analyze the dynamic changes of c...         2020   \n",
       "1215  Prior to the global outbreak of SARS-CoV in 20...         2020   \n",
       "1216  Ante la situación mundial, en relación a 2019-...         2020   \n",
       "1217  Background: At the beginning of the New Year 2...         2020   \n",
       "1218  Citing the growing threat of the coronavirus, ...         2020   \n",
       "1219  Declaratoria de Emergencia en Salud Pública de...         2020   \n",
       "1220  During the outbreak of coronavirus disease-19 ...         2020   \n",
       "1221  Este documento describe la situación epidemiol...         2020   \n",
       "1222  In what appears to be a first, disease tracker...         2020   \n",
       "1223  Objective To analyze the current situation of ...         2020   \n",
       "1224  Objective To identify problems and demands for...         2020   \n",
       "1225  Objective To investigate the first chest HRCT ...         2020   \n",
       "1226  Os Coronavírus (CoV) compõem uma grande famíli...         2020   \n",
       "1227  Outbreak of the novel coronavirus pneumonia (N...         2020   \n",
       "1228  Since this is going to be a post about the cor...         2020   \n",
       "1229  The new coronavirus pneumonia that first appea...         2020   \n",
       "1230  The outbreak of novel coronavirus pneumonia (N...         2020   \n",
       "1231  The outbreak of novel coronavirus pneumonia oc...         2020   \n",
       "1232  To observe the curative effect of TCM syndrome...         2020   \n",
       "1233                                                NaN         2020   \n",
       "1234                                                NaN         2020   \n",
       "1235                                                NaN          NaN   \n",
       "\n",
       "                                                authors  \\\n",
       "0     Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...   \n",
       "1     Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...   \n",
       "2     Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...   \n",
       "3                      Deng, Sheng-Qun; Peng, Hong-Juan   \n",
       "4                                            Zhang, Wei   \n",
       "5     Tang, Biao; Bragazzi, Nicola Luigi; Li, Qian; ...   \n",
       "6     Roosa, K.; Lee, Y.; Luo, R.; Kirpich, A.; Roth...   \n",
       "7                             Zeng, Yingchun; Zhen, Yan   \n",
       "8             Rocklöv, J.; Sjödin, H.; Wilder-Smith, A.   \n",
       "9     Qing, Enya; Hantak, Michael; Perlman, Stanley;...   \n",
       "10                              Chen, Xinguang; Yu, Bin   \n",
       "11    Quilty, Billy J; Clifford, Sam; group2, CMMID ...   \n",
       "12    Cleemput, S.; Dumon, W.; Fonseca, V.; Karim, W...   \n",
       "13    Lim, Jaegyun; Jeon, Seunghyun; Shin, Hyun-Youn...   \n",
       "14    Shanmugaraj, Balamurugan; Malla, Ashwini; Phoo...   \n",
       "15    Malta, Monica; Rimoin, Anne W.; Strathdee, Ste...   \n",
       "16      Fahmi, Muhamad; Kubota, Yukihiko; Ito, Masahiro   \n",
       "17                                       Chen, Jieliang   \n",
       "18    Wang, Manli; Cao, Ruiyuan; Zhang, Leike; Yang,...   \n",
       "19    Bogoch, I. I.; Watts, A.; Thomas-Bachli, A.; H...   \n",
       "20                          Hon, K. L.; Leung, K. K. Y.   \n",
       "21                                       Nesteruk, Igor   \n",
       "22    Okada, Pilailuk; Buathong, Rome; Phuygun, Siri...   \n",
       "23    Diao, Kaiyue; Han, Peilun; Pang, Tong; Li, Yua...   \n",
       "24    Liui, Junli; Wang, Fangfang; Du, Liuyang; Li, ...   \n",
       "25    Sun, Zhong; Thilakavathy, Karuppiah; Kumar, S....   \n",
       "26                                Li, Jing; Liu, Wenjun   \n",
       "27             The Editors Of The Lancet Global, Health   \n",
       "28                 Liu, Yong; Li, Jinxiu; Feng, Yongwen   \n",
       "29                                         Cho, Hae-Wol   \n",
       "...                                                 ...   \n",
       "1206                                      Graham, Flora   \n",
       "1207                                       Makin, Simon   \n",
       "1208  XU, Hongzhen; CHEN, Shuohui; FU, Junfen; SHU, ...   \n",
       "1209                      Cohen, Kai Kupferschmidt; Jon   \n",
       "1210                       Brasil. Ministerio da, Saude   \n",
       "1211                     São Paulo Secretaria da, Saúde   \n",
       "1212                        LI, Zhi yong; MENG, Liu yan   \n",
       "1213                Murdoch, David R.; French, Nigel P.   \n",
       "1214  Wang, Jincheng; Liu, Jinpeng; Wang, Yuanyuan; ...   \n",
       "1215                                Giwa, A.; Desai, A.   \n",
       "1216                    Argentina. Ministerio de, Salud   \n",
       "1217  Keshavarz, Ahmad Tavakoli; Katayon, Vahdat; Mo...   \n",
       "1218                                        Cho, Adrian   \n",
       "1219                    Mexico. Secretaria de la, Salud   \n",
       "1220  XIAO, Yuling; LU, Xiaojun; KANG, Mei; LI, Dong...   \n",
       "1221  Mexico. Secretaria de Salud. Subsecretaria de ...   \n",
       "1222                                    Normile, Dennis   \n",
       "1223                                          CHEN, Yan   \n",
       "1224  DU, Li; GU, Yibin; CUI, Mengqing; LI, Wenxian;...   \n",
       "1225  LIU, Haifeng; ZHANG, Dongyou; YANG, Yi; LONG, ...   \n",
       "1226  Rio de Janeiro . Secretaria de Estado de Saúde...   \n",
       "1227  XU, Dong; HU, Yu; DING, Ning; XIA, Jiahong; ZH...   \n",
       "1228                                        Lowe, Derek   \n",
       "1229                                         ZHIJIE, Li   \n",
       "1230  ZHANG, Lei; CAO, Qing; WANG, Ying; LU, Quan; H...   \n",
       "1231                MA, Fuhai; HU, Haitao; TIAN, Yantao   \n",
       "1232       Liao, Rongye; Yang, Jie; Cao, Zhi; Wang, Jun   \n",
       "1233                           Alavi-Moghaddam, Mostafa   \n",
       "1234  Rodriguez-Morales, A. J.; Bonilla-Aldana, D. K...   \n",
       "1235  Chen, Nanshan and Zhou, Min and Dong, Xuan and...   \n",
       "\n",
       "                                                journal  \\\n",
       "0                                    Intensive Care Med   \n",
       "1                                        Cell Discovery   \n",
       "2                          Journal of Clinical Medicine   \n",
       "3                                            J Clin Med   \n",
       "4                                    Intensive Care Med   \n",
       "5                          Infectious Disease Modelling   \n",
       "6                          Infectious Disease Modelling   \n",
       "7                             The Lancet. Global health   \n",
       "8                            Journal of Travel Medicine   \n",
       "9                                                  mBio   \n",
       "10                    Global Health Research and Policy   \n",
       "11                                     Eurosurveillance   \n",
       "12                     Bioinformatics (Oxford, England)   \n",
       "13                    Journal of Korean Medical Science   \n",
       "14                                            Pathogens   \n",
       "15                                    EClinicalMedicine   \n",
       "16                    Infection, Genetics and Evolution   \n",
       "17                               Microbes and Infection   \n",
       "18                                        Cell Research   \n",
       "19                           Journal of travel medicine   \n",
       "20                            Hong Kong Medical Journal   \n",
       "21             Innovative Biosystems and Bioengineering   \n",
       "22                                     Eurosurveillance   \n",
       "23                          Precision Clinical Medicine   \n",
       "24                            Frontiers in Microbiology   \n",
       "25    International Journal of Environmental Researc...   \n",
       "26                                         Protein Cell   \n",
       "27                            The Lancet. Global health   \n",
       "28                                        Critical Care   \n",
       "29        Osong Public Health and Research Perspectives   \n",
       "...                                                 ...   \n",
       "1206                                             Nature   \n",
       "1207                                                NaN   \n",
       "1208         Chinese Journal of Hospital Administration   \n",
       "1209                                            Science   \n",
       "1210                                                NaN   \n",
       "1211                                                NaN   \n",
       "1212                     Chinese Journal of Stomatology   \n",
       "1213                                          N Z Med J   \n",
       "1214                 Zhejiang Da Xue Xue Bao Yi Xue Ban   \n",
       "1215                        Emergency medicine practice   \n",
       "1216                                                NaN   \n",
       "1217                      Iranian South Medical Journal   \n",
       "1218                                   Science Magazine   \n",
       "1219                                                NaN   \n",
       "1220             Chinese Journal of Laboratory Medicine   \n",
       "1221                                                NaN   \n",
       "1222                                   Science Magazine   \n",
       "1223             Chinese Journal of Preventive Medicine   \n",
       "1224       Chinese Journal of Obstetrics and Gynecology   \n",
       "1225                       Chinese Journal of Radiology   \n",
       "1226                            Nota Técnica-SVS/SES-RJ   \n",
       "1227         Chinese Journal of Hospital Administration   \n",
       "1228                                            Science   \n",
       "1229      Chinese Journal of Experimental Ophthalmology   \n",
       "1230     Chinese Journal of Applied Clinical Pediatrics   \n",
       "1231                        Chinese Journal of Oncology   \n",
       "1232  International Journal of Traditional Chinese M...   \n",
       "1233            Archives of Academic Emergency Medicine   \n",
       "1234                              Infezioni in Medicina   \n",
       "1235                                                NaN   \n",
       "\n",
       "      Microsoft Academic Paper ID WHO #Covidence has_full_text  \n",
       "0                    2.002765e+09          #3252          True  \n",
       "1                    3.003431e+09          #1861          True  \n",
       "2                    3.006065e+09          #1043          True  \n",
       "3                    1.776631e+08          #1999          True  \n",
       "4                    3.006643e+09          #3242         False  \n",
       "5                    3.006029e+09           #729          True  \n",
       "6                    3.006029e+09           #865          True  \n",
       "7                    2.627046e+09          #5386         False  \n",
       "8                    3.006304e+09          #2926          True  \n",
       "9                    3.005811e+09          #2427          True  \n",
       "10                   3.006646e+09          #5595          True  \n",
       "11                   3.005152e+09           #682          True  \n",
       "12                   3.003549e+09          #2776          True  \n",
       "13                   3.005657e+09          #3770          True  \n",
       "14                   3.003886e+09          #1502          True  \n",
       "15                   3.004913e+09          #3333          True  \n",
       "16                   2.286137e+09          #4581          True  \n",
       "17                            NaN           #236          True  \n",
       "18                   3.005213e+09           #205          True  \n",
       "19                   3.004048e+09           #102          True  \n",
       "20                   2.029408e+09          #3395          True  \n",
       "21                   3.006305e+09          #1218          True  \n",
       "22                   3.005657e+09          #2582          True  \n",
       "23                   3.006468e+09           #781          True  \n",
       "24                   3.003968e+09          #5462          True  \n",
       "25                   2.615949e+09          #3296          True  \n",
       "26                   2.766462e+09          #1815          True  \n",
       "27                   2.794879e+09          #5486         False  \n",
       "28                   2.537201e+09          #1225          True  \n",
       "29                   3.005888e+09          #2662         False  \n",
       "...                           ...            ...           ...  \n",
       "1206                          NaN          #3172           NaN  \n",
       "1207                          NaN           #294           NaN  \n",
       "1208                          NaN          #2106           NaN  \n",
       "1209                          NaN           #332           NaN  \n",
       "1210                          NaN          #1866           NaN  \n",
       "1211                          NaN           #922           NaN  \n",
       "1212                 2.377782e+09          #1228           NaN  \n",
       "1213                 2.185518e+09          #1541           NaN  \n",
       "1214                 2.349103e+09          #1922           NaN  \n",
       "1215                 3.005414e+09          #2815           NaN  \n",
       "1216                 1.604377e+09           #801           NaN  \n",
       "1217                 6.213808e+08          #4536           NaN  \n",
       "1218                 2.595675e+09          #2772           NaN  \n",
       "1219                 3.003591e+09          #1803           NaN  \n",
       "1220                 2.194450e+09          #2110           NaN  \n",
       "1221                 2.241123e+09          #1802           NaN  \n",
       "1222                 2.855668e+09          #2584           NaN  \n",
       "1223                 2.363411e+09          #2333           NaN  \n",
       "1224                 1.891210e+09          #2305           NaN  \n",
       "1225                 2.418494e+09          #2187           NaN  \n",
       "1226                 2.518955e+09           #216           NaN  \n",
       "1227                 2.348898e+09          #2108           NaN  \n",
       "1228                 2.343732e+09          #3808           NaN  \n",
       "1229                 3.005943e+09          #2051           NaN  \n",
       "1230                 2.416639e+09          #2063           NaN  \n",
       "1231                 2.366820e+09          #2176           NaN  \n",
       "1232                 2.616653e+09          #1962           NaN  \n",
       "1233                 3.006237e+09          #3509           NaN  \n",
       "1234                 3.005681e+09           #926           NaN  \n",
       "1235                 3.004581e+09            #22           NaN  \n",
       "\n",
       "[1236 rows x 14 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['source_x']=='CZI']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scrape papers from PMC\n",
    "Journals: 16593"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [],
   "source": [
    "papers_df = df_meta[(df_meta['source_x']=='PMC')&(df_meta['sha'].notnull())]\n",
    "papers_array = papers_df[['sha','doi','pmcid','publish_time','journal']].to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884\n"
     ]
    }
   ],
   "source": [
    "pmc_base_url = 'https://www.ncbi.nlm.nih.gov/pmc/articles/'\n",
    "for row in papers_array[:1]:\n",
    "    paper = {}\n",
    "    paper['id'] = row[0]\n",
    "    paper['doi'] = row[1]\n",
    "    paper['PMCID'] = row[2]\n",
    "    paper['url'] = pmc_base_url + paper['PMCID']\n",
    "    paper['date'] = row[3]\n",
    "    paper['journal'] = row[4]\n",
    "    print(paper['url'])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884'"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paper['url']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "html = session.get(paper['url'])\n",
    "tmp_soup = BeautifulSoup(html.content,'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
       "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n",
       "\n",
       "<html lang=\"en\" xml:lang=\"en\" xmlns=\"http://www.w3.org/1999/xhtml\">\n",
       "<head><meta content=\"text/html; charset=utf-8\" http-equiv=\"Content-Type\"/>\n",
       "<!-- AppResources meta begin -->\n",
       "<script type=\"text/javascript\">var ncbi_startTime = new Date();</script>\n",
       "<!-- AppResources meta end -->\n",
       "<!-- TemplateResources meta begin -->\n",
       "<meta content=\"\" name=\"paf_template\"/>\n",
       "<!-- TemplateResources meta end -->\n",
       "<!-- Logger begin -->\n",
       "<meta content=\"pmc\" name=\"ncbi_db\"/><meta content=\"article\" name=\"ncbi_pdid\"/><meta content=\"\" name=\"ncbi_acc\"/><meta content=\"plosbiol\" name=\"ncbi_domain\"/><meta content=\"record\" name=\"ncbi_report\"/><meta content=\"fulltext\" name=\"ncbi_type\"/><meta content=\"\" name=\"ncbi_objectid\"/><meta content=\"/articles/PMC1054884/\" name=\"ncbi_pcid\"/><meta content=\"pmc\" name=\"ncbi_app\"/>\n",
       "<!-- Logger end -->\n",
       "<title>Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection</title>\n",
       "<!-- AppResources external_resources begin -->\n",
       "<link href=\"/core/jig/1.14.8/css/jig.min.css\" rel=\"stylesheet\"/><script src=\"/core/jig/1.14.8/js/jig.min.js\" type=\"text/javascript\"></script>\n",
       "<!-- AppResources external_resources end -->\n",
       "<!-- Page meta begin -->\n",
       "<meta content=\"INDEX,NOFOLLOW,NOARCHIVE\" name=\"robots\"/><link href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884/\" rel=\"canonical\"/><link href=\"http://purl.org/DC/elements/1.0/\" rel=\"schema.DC\"/><meta content=\"PLoS Biology\" name=\"citation_journal_title\"/><meta content=\"Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\" name=\"citation_title\"/><meta content=\"Remy Froissart, Denis Roze, Marilyne Uzest, Lionel Galibert, Stephane Blanc, Yannis Michalakis\" name=\"citation_authors\"/><meta content=\"March 2005\" name=\"citation_date\"/><meta content=\"3\" name=\"citation_issue\"/><meta content=\"3\" name=\"citation_volume\"/><meta content=\"10.1371/journal.pbio.0030089\" name=\"citation_doi\"/><meta content=\"/pmc/articles/PMC1054884/?report=abstract\" name=\"citation_abstract_html_url\"/><meta content=\"15737066\" name=\"citation_pmid\"/><meta content=\"Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\" name=\"DC.Title\"/><meta content=\"Text\" name=\"DC.Type\"/><meta content=\"Public Library of Science\" name=\"DC.Publisher\"/><meta content=\"Remy Froissart\" name=\"DC.Contributor\"/><meta content=\"Denis Roze\" name=\"DC.Contributor\"/><meta content=\"Marilyne Uzest\" name=\"DC.Contributor\"/><meta content=\"Lionel Galibert\" name=\"DC.Contributor\"/><meta content=\"Stephane Blanc\" name=\"DC.Contributor\"/><meta content=\"Yannis Michalakis\" name=\"DC.Contributor\"/><meta content=\"2005 Mar\" name=\"DC.Date\"/><meta content=\"10.1371/journal.pbio.0030089\" name=\"DC.Identifier\"/><meta content=\"en\" name=\"DC.Language\"/><meta content=\"Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection\" property=\"og:title\"/><meta content=\"article\" property=\"og:type\"/><meta content=\"Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections ...\" property=\"og:description\"/><meta content=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1054884/\" property=\"og:url\"/><meta content=\"PubMed Central (PMC)\" property=\"og:site_name\"/><meta content=\"https://www.ncbi.nlm.nih.gov/corehtml/pmc/pmcgifs/pmc-logo-share.png\" property=\"og:image\"/><meta content=\"summary\" name=\"twitter:card\"/><meta content=\"@ncbi\" name=\"twitter:site\"/><link href=\"/corehtml/pmc/css/3.14.1/pmc.min.css\" rel=\"stylesheet\" type=\"text/css\"/><link href=\"/corehtml/pmc/css/3.14.1/pmc_extras_prnt.min.css\" media=\"print\" rel=\"stylesheet\" type=\"text/css\"/><script src=\"/corehtml/pmc/js/common.min.js\" type=\"text/javascript\">//</script><script src=\"/corehtml/pmc/js/NcbiTagServer.min.js\" type=\"text/javascript\">//</script><meta content=\"backend:'https://api.ncbi.nlm.nih.gov/lit/ctxp/v1/pmc/'\" name=\"citationexporter\"/><script src=\"https://www.ncbi.nlm.nih.gov/corehtml/pmc/ctxp/jquery.citationexporter.min.js\" type=\"text/javascript\">//</script><link href=\"https://www.ncbi.nlm.nih.gov/corehtml/pmc/ctxp/citationexporter.css\" rel=\"stylesheet\" type=\"text/css\"/><script src=\"/core/mathjax/2.6.1/MathJax.js?config=/corehtml/pmc/js/mathjax-config-classic.3.4.js\" type=\"text/javascript\"></script><script type=\"text/javascript\">window.name=\"mainwindow\";</script><style type=\"text/css\">.pmc-wm {background:transparent repeat-y top left;background-image:url(/corehtml/pmc/pmcgifs/wm-plosbiol.gif);background-size: auto, contain}</style><style type=\"text/css\">.print-view{display:block}</style><style type=\"text/css\">\n",
       "        div.pmc_para_cit li.highlight,\n",
       "        div.pmc_para_cit li.highlight .one_line_source\n",
       "        { background: #E0E0E0; }\n",
       "        a.bibr.highlight { background: #E0E0E0; } \n",
       "      </style><meta content=\"\" name=\"cited_in_systematic_reviews\"/><link href=\"/pmc/articles/PMC1054884/epub/\" rel=\"alternate\" type=\"application/epub+zip\"/><link href=\"/pmc/articles/PMC1054884/pdf/pbio.0030089.pdf\" rel=\"alternate\" type=\"application/pdf\"/>\n",
       "<!-- Page meta end -->\n",
       "<link href=\"//www.ncbi.nlm.nih.gov/favicon.ico\" rel=\"shortcut icon\"/><meta content=\"8A1B0C78E7A1C97100000000031B031B.m_8\" name=\"ncbi_phid\"/>\n",
       "<meta content=\"origin-when-cross-origin\" name=\"referrer\"/><link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/css/3852956/3985586/3808861/4121862/3974050/3917732/251717/4048120/3846471/14534/45193/4113719/3849091/3984811/3751656/4033350/3840896/3577051/3852958/3881636/3579733/4062871/12930/3964959/3855473/4047625/3854974/3854955/4076335/4128070/9685/3549676/3609192/3609193/3609213/3395586/4143404.css\" rel=\"stylesheet\" type=\"text/css\"/><link href=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/css/3411343/3882866/4157116.css\" media=\"print\" rel=\"stylesheet\" type=\"text/css\"/></head>\n",
       "<body class=\"article\">\n",
       "<div class=\"grid\">\n",
       "<div class=\"col twelve_col nomargin shadow\">\n",
       "<!-- System messages like service outage or JS required; this is handled by the TemplateResources portlet -->\n",
       "<div class=\"sysmessages\">\n",
       "<noscript>\n",
       "<p class=\"nojs\">\n",
       "<strong>Warning:</strong>\n",
       "\tThe NCBI web site requires JavaScript to function. \n",
       "\t<a href=\"/guide/browsers/#enablejs\" target=\"_blank\" title=\"Learn how to enable JavaScript\">more...</a>\n",
       "</p>\n",
       "</noscript>\n",
       "</div>\n",
       "<!--/.sysmessage-->\n",
       "<div class=\"wrap\">\n",
       "<div class=\"page\">\n",
       "<div class=\"top\">\n",
       "<div class=\"universal_header\" id=\"universal_header\"><ul class=\"inline_list jig-ncbimenu ui-ncbimenu resources_list\" id=\"navcontent\"><li class=\"ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt\"><a accesskey=\"1\" class=\"ui-ncbimenu-link-first\" href=\"/\" id=\"ncbihome\" role=\"banner\" title=\"NCBI Home\"><span class=\"offscreen_noflow\">NCBI</span><img alt=\"NCBI Logo\" class=\"ncbi_logo\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/28977\" title=\"NCBI\"/></a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"3\" href=\"#maincontent\" tabindex=\"0\" title=\"Skip to the content\">Skip to main\n",
       "                        content</a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"4\" href=\"#navcontent\" tabindex=\"0\" title=\"Skip to the navigation\">Skip to\n",
       "                        navigation</a></li><li class=\"topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset\" id=\"resource-menu\"><a class=\"ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown\" href=\"/static/header_footer_ajax/submenu/#resources\">Resources</a></li><li class=\"topmenu ui-helper-reset ui-ncbimenu-item-first\" id=\"all-howtos-menu\"><a class=\"ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown\" href=\"/static/header_footer_ajax/submenu/#howto\">How To</a></li><li class=\"offscreen_noflow ui-ncbimenu-item-skip access\"><a accesskey=\"0\" href=\"/guide/browsers/#accesskeys\" tabindex=\"0\" title=\"About My NCBI Accesskeys\">About NCBI Accesskeys</a></li></ul><div class=\"myncbi\"><span id=\"myncbiusername\" style=\"display:none\"><a href=\"/account/settings/\" id=\"mnu\" title=\"Edit account settings\"></a></span><a accesskey=\"2\" href=\"/myncbi/\" id=\"myncbi\" style=\"display:none\">My NCBI</a><a href=\"/account/\" id=\"sign_in\">Sign in to NCBI</a><a href=\"/account/signout/\" id=\"sign_out\" style=\"display:none\">Sign Out</a></div></div>\n",
       "<div class=\"header\">\n",
       "<div class=\"res_logo\">\n",
       "<h1 class=\"img_logo\"><a class=\"pmc_logo offscreen\" href=\"/pmc/\">PMC</a></h1>\n",
       "<div class=\"NLMLogo\">\n",
       "<a href=\"https://www.nlm.nih.gov/\" title=\"US National Library of Medicine\">US National Library of Medicine</a>\n",
       "<br/>\n",
       "<a href=\"https://www.nih.gov/\" title=\"National Institutes of Health\">National Institutes of Health</a>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"search\"><form action=\"/pmc/\" method=\"get\"><div class=\"search_form\"><label class=\"offscreen_noflow\" for=\"database\">Search database</label><select id=\"database\"><optgroup label=\"Recent\"><option class=\"last\" data-ac_dict=\"pmc-search-autocomplete\" selected=\"selected\" value=\"pmc\">PMC</option></optgroup><optgroup label=\"All\"><option value=\"gquery\">All Databases</option><option value=\"assembly\">Assembly</option><option value=\"biocollections\">Biocollections</option><option value=\"bioproject\">BioProject</option><option value=\"biosample\">BioSample</option><option value=\"biosystems\">BioSystems</option><option value=\"books\">Books</option><option value=\"clinvar\">ClinVar</option><option value=\"cdd\">Conserved Domains</option><option value=\"gap\">dbGaP</option><option value=\"dbvar\">dbVar</option><option value=\"gene\">Gene</option><option value=\"genome\">Genome</option><option value=\"gds\">GEO DataSets</option><option value=\"geoprofiles\">GEO Profiles</option><option value=\"gtr\">GTR</option><option value=\"homologene\">HomoloGene</option><option value=\"ipg\">Identical Protein Groups</option><option value=\"medgen\">MedGen</option><option value=\"mesh\">MeSH</option><option value=\"ncbisearch\">NCBI Web Site</option><option value=\"nlmcatalog\">NLM Catalog</option><option value=\"nuccore\">Nucleotide</option><option value=\"omim\">OMIM</option><option data-ac_dict=\"pmc-search-autocomplete\" value=\"pmc\">PMC</option><option value=\"popset\">PopSet</option><option value=\"probe\">Probe</option><option value=\"protein\">Protein</option><option value=\"proteinclusters\">Protein Clusters</option><option value=\"pcassay\">PubChem BioAssay</option><option value=\"pccompound\">PubChem Compound</option><option value=\"pcsubstance\">PubChem Substance</option><option value=\"pubmed\">PubMed</option><option value=\"snp\">SNP</option><option value=\"sparcle\">Sparcle</option><option value=\"sra\">SRA</option><option value=\"structure\">Structure</option><option value=\"taxonomy\">Taxonomy</option><option value=\"toolkit\">ToolKit</option><option value=\"toolkitall\">ToolKitAll</option><option value=\"toolkitbookgh\">ToolKitBookgh</option></optgroup></select><div class=\"nowrap\"><label accesskey=\"/\" class=\"offscreen_noflow\" for=\"term\">Search term</label><div class=\"nowrap\"><input autocomplete=\"off\" class=\"jig-ncbiclearbutton jig-ncbiautocomplete\" data-jigconfig=\"dictionary:'pmc-search-autocomplete',disableUrl:'NcbiSearchBarAutoComplCtrl'\" data-sbconfig=\"ds:'no',pjs:'no',afs:'yes'\" id=\"term\" name=\"term\" title=\"Search PMC. Use up and down arrows to choose an item from the autocomplete.\" type=\"text\" value=\"\"/></div><button class=\"button_search nowrap\" cmd=\"go\" id=\"search\" type=\"submit\">Search</button></div></div></form><ul class=\"searchlinks inline_list\"><li>\n",
       "<a href=\"/pmc/advanced/\">Advanced</a>\n",
       "</li><li>\n",
       "<a href=\"/pmc/journals/\">Journal list</a>\n",
       "</li><li class=\"help\">\n",
       "<a href=\"/books/NBK3825/\" target=\"_blank\">Help</a>\n",
       "</li></ul></div>\n",
       "</div>\n",
       "<!--<component id=\"Page\" label=\"headcontent\"/>-->\n",
       "</div>\n",
       "<div class=\"content\">\n",
       "<!-- site messages -->\n",
       "<div class=\"container\">\n",
       "<div class=\"content eight_col col\" id=\"maincontent\">\n",
       "<div class=\"navlink-box\">\n",
       "<ul class=\"page-breadcrumbs inline_list small\"><li class=\"journal-list\"><a class=\"navlink\" href=\"/pmc/journals/\">Journal List</a></li><li class=\"archive\"><a class=\"navlink\" href=\"/pmc/journals/212/\">PLoS Biol</a></li><li class=\"issue-page\"><a class=\"navlink\" href=\"/pmc/issues/115975/\">v.3(3); 2005 Mar</a></li><li class=\"accid\">PMC1054884</li></ul>\n",
       "</div>\n",
       "<!-- Journal banner -->\n",
       "<div class=\"pmc-page-banner whole_rhythm\"><div><img alt=\"Logo of plosbiol\" src=\"/corehtml/pmc/pmcgifs/logo-plosbiol.gif\" usemap=\"#logo-imagemap\"/><map id=\"logo-imagemap\" name=\"logo-imagemap\"><area alt=\"PLoS Biology\" coords=\"0,0,438,55\" href=\"http://www.plosbiology.org/\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"PLoS Biology\"/><area alt=\"Submit to PLoS\" coords=\"106,56,202,74\" href=\"http://www.plosbiology.org/static/checklist.action\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Submit to PLoS\"/><area alt=\"Get E-mail Alerts\" coords=\"204,56,302,74\" href=\"http://journals.plos.org/alerts\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Get E-mail Alerts\"/><area alt=\"Contact Us\" coords=\"304,56,374,74\" href=\"http://www.plosbiology.org/static/contact.action\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Contact Us\"/><area alt=\"Public Library of Science (PLoS)\" coords=\"440,0,500,55\" href=\"http://www.plos.org/\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A\" shape=\"rect\" target=\"pmc_ext\" title=\"Public Library of Science (PLoS)\"/><area alt=\"View this Article\" coords=\"0,56,104,74\" href=\"//dx.doi.org/10.1371/journal.pbio.0030089\" ref=\"reftype=publisher&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CBanner&amp;TO=Content%20Provider%7CArticle%7CFullText\" shape=\"rect\" target=\"pmc_ext\" title=\"View this Article\"/></map></div> </div>\n",
       "<!--component id='MainPortlet' label='search-reference'/-->\n",
       "<!-- Book content -->\n",
       "<div class=\"\">\n",
       "<div class=\"hide-overflow article lit-style content pmc-wm slang-all page-box\"><!--main-content--><div class=\"jig-ncbiinpagenav\" data-jigconfig=\"smoothScroll: false, allHeadingLevels: ['h2'], headingExclude: ':hidden'\"><div class=\"fm-sec half_rhythm no_top_margin\"><div class=\"fm-citation half_rhythm no_top_margin clearfix\"><div class=\"inline_block eight_col va_top\"><div><div><span class=\"cit\"><span id=\"pmcmata\">PLoS Biol</span>. 2005 Mar; 3(3): e89. </span></div><div><span class=\"fm-vol-iss-date\">Published online 2005 Mar 1. </span> <span class=\"doi\">doi: <a href=\"//dx.doi.org/10.1371%2Fjournal.pbio.0030089\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CCrosslink%7CDOI\" target=\"pmc_ext\">10.1371/journal.pbio.0030089</a></span></div></div></div><div class=\"inline_block four_col va_top show-overflow align_right\"><div class=\"fm-citation-ids\"><div class=\"fm-citation-pmcid\"><span class=\"fm-citation-ids-label\">PMCID: </span><span>PMC1054884</span></div><div class=\"fm-citation-pmid\">PMID: <a href=\"/pubmed/15737066\">15737066</a></div></div></div></div><h1 class=\"content-title\">Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection</h1><div class=\"half_rhythm\"><div class=\"contrib-group fm-author\"><a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550468180560\" href=\"/pubmed/?term=Froissart%20R%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Remy Froissart</a>,<sup>\n",
       "1\n",
       ",</sup><sup>¤</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550464448704\" href=\"/pubmed/?term=Roze%20D%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Denis Roze</a>,<sup>\n",
       "2\n",
       "</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550502491840\" href=\"/pubmed/?term=Uzest%20M%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Marilyne Uzest</a>,<sup>\n",
       "1\n",
       "</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550463591280\" href=\"/pubmed/?term=Galibert%20L%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Lionel Galibert</a>,<sup>\n",
       "1\n",
       "</sup> <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550504176336\" href=\"/pubmed/?term=Blanc%20S%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Stephane Blanc</a>,<sup>\n",
       "1\n",
       "</sup> and  <a class=\"affpopup\" co-class=\"co-affbox\" co-rid=\"_co_idm140550504173968\" href=\"/pubmed/?term=Michalakis%20Y%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Yannis Michalakis</a><sup><img alt=\"corresponding author\" src=\"/corehtml/pmc/pmcgifs/corrauth.gif\"/></sup><sup>\n",
       "2\n",
       "</sup></div><div class=\"contrib-group aff-tip\" style=\"display:none\"><div id=\"_co_idm140550468180560\"><h3 class=\"no_margin\">Remy Froissart</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Froissart%20R%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Remy Froissart</a></div></div><div id=\"_co_idm140550464448704\"><h3 class=\"no_margin\">Denis Roze</h3><p><strong>2</strong>Génétique et Evolution des Maladies Infectieuses, Unité Mixte de Recherche Centre National de la Recherche Scientifique (CNRS)–Institut de Recherche pour le Développement (IRD) 2724, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Roze%20D%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Denis Roze</a></div></div><div id=\"_co_idm140550502491840\"><h3 class=\"no_margin\">Marilyne Uzest</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Uzest%20M%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Marilyne Uzest</a></div></div><div id=\"_co_idm140550463591280\"><h3 class=\"no_margin\">Lionel Galibert</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Galibert%20L%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Lionel Galibert</a></div></div><div id=\"_co_idm140550504176336\"><h3 class=\"no_margin\">Stephane Blanc</h3><p><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Blanc%20S%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Stephane Blanc</a></div></div><div id=\"_co_idm140550504173968\"><h3 class=\"no_margin\">Yannis Michalakis</h3><p><strong>2</strong>Génétique et Evolution des Maladies Infectieuses, Unité Mixte de Recherche Centre National de la Recherche Scientifique (CNRS)–Institut de Recherche pour le Développement (IRD) 2724, Montpellier, France</p><div>Find articles by <a href=\"/pubmed/?term=Michalakis%20Y%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=15737066\">Yannis Michalakis</a></div></div></div></div><div class=\"contrib-group half_rhythm fm-editor\">Roger Hull, <span class=\"fm-role\">Academic Editor</span><sup></sup></div><div class=\"fm-panel half_rhythm\"><div class=\"togglers\"><a class=\"pmctoggle\" href=\"#\" rid=\"idm140550461914704_ai idm140550468180688_ai idm140550465846224_ai\">Author information</a> <a class=\"pmctoggle\" href=\"#\" rid=\"idm140550461914704_an\">Article notes</a> <a class=\"pmctoggle\" href=\"#\" rid=\"idm140550461914704_cpl\">Copyright and License information</a> <a href=\"/pmc/about/disclaimer/\">Disclaimer</a></div><div class=\"fm-authors-info fm-panel hide half_rhythm\" id=\"idm140550461914704_ai\" style=\"display:none\"><div class=\"fm-affl\" id=\"aff1\" lang=\"en\"><strong>1</strong>Biologie et Génétique des Interactions Plante-Parasite, Unité Mixte de Recherche Centre de Coopération Internationale en Recherche Agronomique pour le Développement (CIRAD)–Institut National de la Recherche Agronomique (INRA)–Ecole National Supérieure Agronomique de Montpellier (ENSAM), TA 41/K, Campus International de Baillarguet, Montpellier, France</div><div class=\"fm-affl\" id=\"aff2\" lang=\"en\"><strong>2</strong>Génétique et Evolution des Maladies Infectieuses, Unité Mixte de Recherche Centre National de la Recherche Scientifique (CNRS)–Institut de Recherche pour le Développement (IRD) 2724, Montpellier, France</div><div class=\"fm-affl\" id=\"edit1\" lang=\"en\">John Innes Center, United Kingdom</div><div><sup><img alt=\"corresponding author\" src=\"/corehtml/pmc/pmcgifs/corrauth.gif\"/></sup>Corresponding author.</div><div><span class=\"contrib-email\" id=\"idm140550504173968\">Yannis Michalakis: <a class=\"oemail\" data-email=\"rf.dri.lpm@sikalahciM.sinnaY\" href=\"mailto:dev@null\">rf.dri.lpm@sikalahciM.sinnaY</a></span> </div></div><div class=\"fm-article-notes fm-panel hide half_rhythm\" id=\"idm140550461914704_an\" style=\"display:none\"><div class=\"fm-pubdate half_rhythm\">Received 2004 Sep 15; Accepted 2005 Jan 9.</div></div><div class=\"permissions fm-panel half_rhythm hide\" id=\"idm140550461914704_cpl\" style=\"display:none\"><div class=\"fm-copyright half_rhythm\"><a href=\"/pmc/about/copyright/\">Copyright</a> : © 2005 Froissart et al.</div><div class=\"license half_rhythm\">This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.</div></div></div><div class=\"links-box whole_rhythm\" id=\"pmclinksbox\"><div class=\"fm-panel\"><div>See \"<a href=\"/pmc/articles/PMC1054890/\">Recombination as a Way of Life: Viruses Do It Every Day</a>\" , e117.</div><div style=\"border-top: 1px solid rgb(102, 102, 153); margin: 5px 10% 3px;\"></div><div>This article has been <a href=\"/pmc/articles/PMC1054884/citedby/\">cited by</a> other articles in PMC.</div></div></div></div><div class=\"sec\"></div><div class=\"tsec sec\" id=\"idm140550464314896\" lang=\"en\"><h2 class=\"head no_bottom_margin\" id=\"idm140550464314896title\">Abstract</h2><!--article-meta--><div><p class=\"p p-first-last\" id=\"__p1\">Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections occur. While the recombination rate has been recently evaluated in experimentally co-infected cell cultures for several viruses, direct quantification at the most biologically significant level, that of a host infection, is still lacking. This study fills this gap using the cauliflower mosaic virus as a model. We distributed four neutral markers along the viral genome, and co-inoculated host plants with marker-containing and wild-type viruses. The frequency of recombinant genomes was evaluated 21 d post-inoculation. On average, over 50% of viral genomes recovered after a single host infection were recombinants, clearly indicating that recombination is very frequent in this virus. Estimates of the recombination rate show that all regions of the genome are equally affected by this process. Assuming that ten viral replication cycles occurred during our experiment—based on data on the timing of coat protein detection—the per base and replication cycle recombination rate was on the order of 2 × 10<sup>−5</sup> to 4 × 10<sup>−5</sup>. This first determination of a virus recombination rate during a single multi-cellular host infection indicates that recombination is very frequent in the everyday life of this virus.</p></div></div><div class=\"tsec sec\" id=\"s1\"><h2 class=\"head no_bottom_margin\" id=\"s1title\">Introduction</h2><p class=\"p p-first\" id=\"__p3\">As increasing numbers of full-length viral sequences become available, recombinant or mosaic viruses are being recognized more frequently [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b01\" id=\"__tag_577434177\" rid=\"pbio-0030089-b01\">1</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b02\" id=\"__tag_577434170\" rid=\"pbio-0030089-b02\">2</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b03\" id=\"__tag_577434160\" rid=\"pbio-0030089-b03\">3</a>]. Recombination events have been demonstrated to be associated with viruses expanding their host range [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b04\" id=\"__tag_577434146\" rid=\"pbio-0030089-b04\">4</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b05\" id=\"__tag_577434174\" rid=\"pbio-0030089-b05\">5</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b06\" id=\"__tag_577434196\" rid=\"pbio-0030089-b06\">6</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b07\" id=\"__tag_577434152\" rid=\"pbio-0030089-b07\">7</a>] or increasing their virulence [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b08\" id=\"__tag_577434186\" rid=\"pbio-0030089-b08\">8</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b09\" id=\"__tag_577434183\" rid=\"pbio-0030089-b09\">9</a>], thus accompanying, or perhaps even being at the origin of, major changes during virus adaptation. It remains unclear, however, whether recombination events represent a highly frequent and significant phenomenon in the everyday life of these viruses.</p><p id=\"__p4\">Viruses can exchange genetic material when at least two different viral genomes co-infect the same host cell. Progeny can then become hybrid through different mechanisms, such as reassortment of segments when the parental genomes are fragmented [<a class=\"bibr popnode\" href=\"#pbio-0030089-b10\" rid=\"pbio-0030089-b10\">10</a>], intra-molecular recombination when polymerases switch templates (in RNA viruses) [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b11\" id=\"__tag_577434201\" rid=\"pbio-0030089-b11\">11</a>], or homologous or non-homologous recombination (in both RNA and DNA viruses). Quantification of viral recombination in multi-cellular organisms has been attempted under two distinct experimental approaches: in vitro (in cell cultures) [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b12\" id=\"__tag_577434150\" rid=\"pbio-0030089-b12\">12</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b13\" id=\"__tag_577434184\" rid=\"pbio-0030089-b13\">13</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b14\" id=\"__tag_577434151\" rid=\"pbio-0030089-b14\">14</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b15\" id=\"__tag_577434159\" rid=\"pbio-0030089-b15\">15</a>], and in vivo (in live hosts) [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b16\" id=\"__tag_577434172\" rid=\"pbio-0030089-b16\">16</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b17\" id=\"__tag_577434162\" rid=\"pbio-0030089-b17\">17</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b18\" id=\"__tag_577434188\" rid=\"pbio-0030089-b18\">18</a>]. The in vitro approach, which has so far been applied only to animal viruses, allows the establishment of the “intrinsic” recombination rate in experimentally co-infected cells in cell cultures [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b14\" id=\"__tag_577434195\" rid=\"pbio-0030089-b14\">14</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b15\" id=\"__tag_577434164\" rid=\"pbio-0030089-b15\">15</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b19\" id=\"__tag_577434148\" rid=\"pbio-0030089-b19\">19</a>]. However, it does not necessarily reflect the situation in entire, living hosts, where the frequency of co-infected cells is poorly known and depends on many factors such as the size of the pathogen population, the relative frequency and distribution of the different variants, and host defense mechanisms preventing secondary infection of cells. The in vivo experimental approach is closer to biological conditions and may thus be more informative of what actually happens in “the real world.”</p><p id=\"__p5\">However, as discussed below, numerous experimental constraints have so far precluded an actual quantification of the baseline rate of recombination. First, many experimental designs have used extreme positive selection, where only recombinant genomes were viable (e.g., [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b13\" id=\"__tag_577434180\" rid=\"pbio-0030089-b13\">13</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b20\" id=\"__tag_577434202\" rid=\"pbio-0030089-b20\">20</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434147\" rid=\"pbio-0030089-b21\">21</a>]). Other studies did not use complementation techniques but detected recombinants by PCR within infected hosts or tissues [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b18\" id=\"__tag_577434193\" rid=\"pbio-0030089-b18\">18</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b22\" id=\"__tag_577434178\" rid=\"pbio-0030089-b22\">22</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b23\" id=\"__tag_577434191\" rid=\"pbio-0030089-b23\">23</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b24\" id=\"__tag_577434197\" rid=\"pbio-0030089-b24\">24</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b25\" id=\"__tag_577434187\" rid=\"pbio-0030089-b25\">25</a>], which provides information on their presence but not on their frequency in the viral population. So far, no quantitative PCR or other quantitative method has been applied to evaluate the number of recombinants appearing in an experimentally infected live host. Finally, recent methods based on sequence analysis inferred the population recombination rate, rather than the individual recombination rate [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b01\" id=\"__tag_577434176\" rid=\"pbio-0030089-b01\">1</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b26\" id=\"__tag_577434169\" rid=\"pbio-0030089-b26\">26</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b27\" id=\"__tag_577434167\" rid=\"pbio-0030089-b27\">27</a>]. While results from these methods certainly take in vivo recombination into account, there are other caveats: isolates have often been collected in different hosts—sometimes in different geographical regions—and sometimes the selective neutrality of sequence variation on which these estimates are based is not clearly established. Estimates from such studies by essence address the estimation of the recombination rate at a different evolutionary scale.</p><p id=\"__p6\">Taken together, the currently available information indicates that no viral recombination rate has ever been estimated directly at time and space scales corresponding to a single multi-cellular host infection, although this level is most significant for the biology and evolution of viruses. This study intends to fill this gap by evaluating the recombination frequency of the cauliflower mosaic virus (CaMV) during a single passage in one of its host plants (the turnip <em class=\"genus-species\">Brassica rapa</em>).</p><p id=\"__p7\">CaMV is a pararetrovirus, which is a major grouping containing hepadnaviruses (e.g., hepatitis B virus), badnaviruses (e.g., banana streak virus), and caulimoviruses (e.g., CaMV). Pararetroviruses are characterized by a non-segmented double-stranded DNA genome. After entering the host cell nucleus, the viral DNA accumulates as a minichromosome [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b28\" id=\"__tag_577434200\" rid=\"pbio-0030089-b28\">28</a>] whose transcription is ensured by the host RNA polymerase II [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b29\" id=\"__tag_577434185\" rid=\"pbio-0030089-b29\">29</a>]. The CaMV genome consists in approximately 8,000 bp and encodes six viral gene products that have been detected <em>in planta</em> (<a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>) [<a class=\"bibr popnode\" href=\"#pbio-0030089-b30\" rid=\"pbio-0030089-b30\">30</a>]. Viral proteins P1 to P6 are expressed from two major transcripts, namely a 19S RNA, encoding P6, and a 35S RNA corresponding to the entire genome and serving as mRNA for proteins P1–P5 [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b31\" id=\"__tag_577434144\" rid=\"pbio-0030089-b31\">31</a>]. Using the pre-genomic 35S RNA as a matrix, the protein P5 (product of gene V) reverse-transcribes the genome into genomic DNA that is concomitantly encapsidated [<a class=\"bibr popnode\" href=\"#pbio-0030089-b30\" rid=\"pbio-0030089-b30\">30</a>].</p><!--fig ft0--><!--fig mode=article f1--><div class=\"fig iconblock whole_rhythm clearfix\" co-legend-rid=\"lgnd_pbio-0030089-g001\" id=\"pbio-0030089-g001\"><a href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><!--fig/graphic|fig/alternatives/graphic mode=\"anchored\" m1--><div class=\"figure\" data-largeobj=\"\" data-largeobj-link-rid=\"largeobj_idm140550461561296\"><a class=\"inline_block ts_canvas\" href=\"/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=1065690_pbio.0030089.g001.jpg\" target=\"tileshopwindow\"><div class=\"ts_bar small\" title=\"Click on image to zoom\"></div><img alt=\"An external file that holds a picture, illustration, etc.\n",
       "Object name is pbio.0030089.g001.jpg\" class=\"tileshop\" src=\"/pmc/articles/PMC1054884/bin/pbio.0030089.g001.jpg\" title=\"Click on image to zoom\"/></a></div><div class=\"largeobj-link align_right\" id=\"largeobj_idm140550461561296\" style=\"display: none\"><a href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/?report=objectonly\" target=\"object\">Open in a separate window</a></div></a><div class=\"icnblk_cntnt\" id=\"lgnd_pbio-0030089-g001\"><div><a class=\"figpopup\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\">Figure 1</a></div><!--caption a7--><div class=\"caption\"><!--caption a8--><strong>Genetic Map of CaMV</strong><p id=\"__p8\">The CaMV genome is a circular double-stranded DNA of 8,024 bp, represented in the figure by a double line. The thick arrows with different textures represent the organization of open reading frames I to VI, encoding proteins detected <em>in planta</em>. Markers a, b, c, and d were engineered at the positions indicated (see <a class=\"sec\" href=\"#s4\" rid=\"s4\">Materials and Methods</a> for precise positions). The inner black arrows represent monocistronic 19S RNA and polycistronic 35S RNA produced by the cellular machinery. The nucleotide position 0 (numbering according to [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b44\" id=\"__tag_577434194\" rid=\"pbio-0030089-b44\">44</a>]) indicates the origin of replication via reverse transcription, which occurs in the direction indicated by the dotted outermost circle-like arrow. Reverse transcription is accomplished by the viral reverse transcriptase, using the 35S RNA as template [<a class=\"bibr popnode\" href=\"#pbio-0030089-b49\" rid=\"pbio-0030089-b49\">49</a>].</p></div></div></div><p id=\"__p9\">The detection of CaMV recombinants in turnip hosts has been reported numerous times. Some studies have demonstrated the appearance of infectious recombinant viral genomes after inoculation (i) of a host plant with two infectious or non-infectious parental clones [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434158\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode\" href=\"#pbio-0030089-b32\" rid=\"pbio-0030089-b32\">32</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b33\" id=\"__tag_577434161\" rid=\"pbio-0030089-b33\">33</a>,<a class=\"bibr popnode\" href=\"#pbio-0030089-b34\" rid=\"pbio-0030089-b34\">34</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b35\" id=\"__tag_577434198\" rid=\"pbio-0030089-b35\">35</a>] or (ii) of a transgenic plant containing one CaMV transgene with a CaMV genome missing the corresponding genomic region [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434179\" rid=\"pbio-0030089-b36\">36</a>]. While the former revealed inter-genomic viral recombination, the latter demonstrated that CaMV can also recombine with transgenes within the host's genome. Another study based on phylogenetic analyses of various CaMV strains has clearly suggested different origins for different genomic regions and, hence, multiple recombination events during the evolution of this virus [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b37\" id=\"__tag_577434156\" rid=\"pbio-0030089-b37\">37</a>]. Indirect experimental evidence has indicated that, in some cases, CaMV recombination could occur within the host nucleus, between different viral minichromosomes, presumably through the action of the DNA repair cellular machinery [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434190\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b35\" id=\"__tag_577434182\" rid=\"pbio-0030089-b35\">35</a>]. Nevertheless, the mechanism of “template switching” during reverse transcription, predominant in all retroviruses, most certainly also applies to pararetroviruses. For this reason, and on the basis of numerous experimental data, CaMV is generally believed to recombine mostly in the cytoplasm of the host cell, by “legal” template switching between two pre-genomic RNA molecules [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434153\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b35\" id=\"__tag_577434143\" rid=\"pbio-0030089-b35\">35</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434173\" rid=\"pbio-0030089-b36\">36</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b38\" id=\"__tag_577434192\" rid=\"pbio-0030089-b38\">38</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b39\" id=\"__tag_577434171\" rid=\"pbio-0030089-b39\">39</a>], or “illegal” template switching between the 19S and the 35S RNA [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434142\" rid=\"pbio-0030089-b36\">36</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b40\" id=\"__tag_577434140\" rid=\"pbio-0030089-b40\">40</a>]. Under this hypothesis, recombination in CaMV could therefore be considered as operating on a linear template during reverse transcription, with the 5′ and 3′ extremities later ligated to circularize the genomic DNA (position 0 in <a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>). The above cited studies clearly demonstrate that CaMV is able to recombine. However, since these studies are based on complementation techniques, non-quantitative detection, or phylogenetically based inferences of recombination, they do not inform us on whether recombination is an exceptional event or an “everyday” process shaping the genetic composition of CaMV populations.</p><p class=\"p p-last\" id=\"__p10\">In the present work, we aimed at answering this question. To this end, we have constructed a CaMV genome with four genetic markers, demonstrated to be neutral in competition experiments. By co-inoculating host plants with equal amounts of wild-type and marker-containing CaMV particles, we have generated mixed populations in which impressive proportions of recombinants—distributed in several different classes corresponding to exchange of different genomic regions—have been detected and quantified. Altogether, the recombinant genomes averaged over 50% of the population. Further analysis of these data, assuming a number of viral replications during the infection period ranging from five to 20, indicates that the per nucleotide per replication cycle recombination rate of CaMV is of the same order of magnitude, i.e., on the order of a few 10<sup>−5</sup>, across the entire genome. We thereby provide the first quantification, to our knowledge, of the recombination rate in a virus population during a single passage in a single host.</p></div><div class=\"tsec sec\" id=\"s2\"><h2 class=\"head no_bottom_margin\" id=\"s2title\">Results</h2><div class=\"sec sec-first\" id=\"s2a\"><h3 id=\"s2atitle\">Recombinant Frequency in CaMV Populations from Co-Infected Plants</h3><p class=\"p p-first\" id=\"__p11\">From <a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>, and supposing that all marker-containing genomic regions can recombine, we could predict the detection and quantification of seven classes of recombinant genotypes: +bcd/a+++, a+cd/+b++, ab+d/++c+, abc+/+++d, ++cd/ab++, a++d/+bc+, and a+c+/+b+d. Indeed, all classes were detected, and their frequencies in the ten CaMV populations analyzed are summarized in <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/\" rid-figpopup=\"pbio-0030089-t001\" rid-ob=\"ob-pbio-0030089-t001\" target=\"table\"><span>Table 1</span></a>.</p><!--table ft1--><!--table-wrap mode=\"anchored\" t5--><div class=\"table-wrap table anchored whole_rhythm\" id=\"pbio-0030089-t001\"><h3>Table 1</h3><!--caption a7--><div class=\"caption\"><!--caption a8--><strong>Quantification of Recombinant Genomes in CaMV Populations</strong></div><div class=\"figure\" data-largeobj=\"\" data-largeobj-link-rid=\"largeobj_idm140550466509952\"><a class=\"inline_block ts_canvas\" href=\"/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=1065690_pbio.0030089.t001.jpg\" target=\"tileshopwindow\"><div class=\"ts_bar small\" title=\"Click on image to zoom\"></div><img alt=\"An external file that holds a picture, illustration, etc.\n",
       "Object name is pbio.0030089.t001.jpg\" class=\"tileshop\" src=\"/pmc/articles/PMC1054884/bin/pbio.0030089.t001.jpg\" title=\"Click on image to zoom\"/></a></div><div class=\"largeobj-link align_right\" id=\"largeobj_idm140550466509952\" style=\"display: none\"><a href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/?report=objectonly\" target=\"object\">Open in a separate window</a></div><div class=\"tblwrap-foot\"><div id=\"nt101\"><p class=\"p p-first-last\" id=\"__p12\">\n",
       "<sup>a</sup> Viral genomes were cloned and analyzed from ten of 24 co-infected plants</p></div><div id=\"nt102\"><p class=\"p p-first-last\" id=\"__p13\">\n",
       "<sup>b</sup> Seven possible classes of recombinants were predicted, their respective frequency in the population is expressed in percentage</p></div><div id=\"nt103\"><p class=\"p p-first-last\" id=\"__p14\">\n",
       "<sup>c</sup> The proportions of recombinants from the seven classes were added to estimate the total percentage of recombinant genomes within each tested plant</p></div></div></div><p class=\"p p-last\" id=\"__p15\">Altogether, the proportion of recombinant genomes found in the mixed viral populations was astonishingly high and very similar in the ten co-infected plants analyzed (<a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/\" rid-figpopup=\"pbio-0030089-t001\" rid-ob=\"ob-pbio-0030089-t001\" target=\"table\"><span>Table 1</span></a>, last column), ranging between 44% (plant 5) to 60% (plants 7, 12, and 20), with a mean frequency (± standard error) of 53.8% ± 2.0%. This result indicates that recombination events are very frequent during the invasion of the host plant by CaMV and represents, to our knowledge, the first direct quantification of viral recombination during the infection of a live multi-cellular host.</p></div><div class=\"sec sec-last\" id=\"s2b\"><h3 id=\"s2btitle\">Probability of Recombination between Various Pairs of Markers</h3><p class=\"p p-first\" id=\"__p16\">The inferred per generation recombination and interference rates, assuming that CaMV undergoes ten replication cycles during the 21 d between infection and sampling, are given for each of the ten plants in <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Table 2</span></a>. Recombination rates between adjacent markers are large, on the order of 0.05 to 0.1. Taking the distance in nucleotides between markers into account yields an average recombination rate per nucleotide and generation on the order of 4 × 10<sup>−5</sup>. Interestingly, this recombination rate does not vary throughout the genome (Kruskal–Wallis test, <em>p</em> = 0.16).</p><!--table ft1--><!--table-wrap mode=\"anchored\" t5--><div class=\"table-wrap table anchored whole_rhythm\" id=\"pbio-0030089-t002\"><h3>Table 2</h3><!--caption a7--><div class=\"caption\"><!--caption a8--><strong>Recombination Parameters for the Viral Populations Sampled from Ten Infected Plants</strong></div><div class=\"figure\" data-largeobj=\"\" data-largeobj-link-rid=\"largeobj_idm140550465918608\"><a class=\"inline_block ts_canvas\" href=\"/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=1065690_pbio.0030089.t002.jpg\" target=\"tileshopwindow\"><div class=\"ts_bar small\" title=\"Click on image to zoom\"></div><img alt=\"An external file that holds a picture, illustration, etc.\n",
       "Object name is pbio.0030089.t002.jpg\" class=\"tileshop\" src=\"/pmc/articles/PMC1054884/bin/pbio.0030089.t002.jpg\" title=\"Click on image to zoom\"/></a></div><div class=\"largeobj-link align_right\" id=\"largeobj_idm140550465918608\" style=\"display: none\"><a href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/?report=objectonly\" target=\"object\">Open in a separate window</a></div><div class=\"tblwrap-foot\"><div id=\"nt201\"><p class=\"p p-first-last\" id=\"__p17\">The various parameters are as follows: <em>r</em>\n",
       "<sub>1</sub>, recombination rate between markers a and b; <em>r</em>\n",
       "<sub>2</sub>, recombination rate between markers b and c; <em>r</em>\n",
       "<sub>3</sub>, recombination rate between markers c and d; <em>i</em>\n",
       "<sub>12</sub>, interference between crossovers in segments a–b and b–c; <em>i</em>\n",
       "<sub>23</sub>, interference between crossovers in segments b–c and c–d; <em>i</em>\n",
       "<sub>13</sub>, interference between crossovers in segments a–b and c–d; <em>i</em>\n",
       "<sub>123</sub>, second-order interference accounting for residual interference. The recombination rates are the maximum likelihood estimates (± 95% confidence intervals). The interference parameters were obtained numerically as explained in the <a class=\"sec\" href=\"#s4\" rid=\"s4\">Materials and Methods</a>\n",
       "</p></div></div></div><p id=\"__p18\">To relax the assumption of the number of replications during the 21 d, we calculated the recombination parameters assuming five or 20 generations. The effect of the number of generations on the estimates is linear: doubling the number of generations results in a halving of the recombination rate (detailed results not shown). For example, the average recombination rates <em>r</em>\n",
       "<sub>1</sub>, <em>r</em>\n",
       "<sub>2</sub>, and <em>r</em>\n",
       "<sub>3</sub> assuming 20 generations were equal to 0.05, 0.04, and 0.025, respectively (compare with values in <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Table 2</span></a>), yielding per nucleotide per generation recombination rates of 1.9 × 10<sup>−5</sup>, 2.2 × 10<sup>−5</sup> and 1.6 × 10<sup>−5</sup>.</p><p class=\"p p-last\" id=\"__p19\">Inspection of <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Table 2</span></a> also shows that first-order interference coefficients were in general negative, indicating that a crossing over in one genomic segment increases the probability that a crossing over will occur in another genomic segment, while the second-order coefficient parameter had an average value close to zero with a large variance. The mechanism leading to these results will be discussed in the following section.</p></div></div><div class=\"tsec sec\" id=\"s3\"><h2 class=\"head no_bottom_margin\" id=\"s3title\">Discussion</h2><p class=\"p p-first\" id=\"__p20\">One major breakthrough in the work presented here lies in the space and time scales at which the experiments were performed. Indeed, the processes occurring within the course of a single infection of one multi-cellular host are of obvious biological relevance for any disease. Previous studies on viral recombination suffered from major drawbacks in this respect, basing their conclusions on experiments relying on complementation among non-infectious viruses or between viruses with undetermined relative fitness, on phylogenetically based analyses, or on experiments in cell cultures. For reasons detailed in the Introduction, the first two methods either do not provide information on the frequency of recombination, but only its occurrence, or address the question at a different temporal, and often spatial, scale. Results from cell cultures, on the other hand, impose cell co-infection by different viral variants, potentially overestimating the frequency of recombination events. Our study circumvents these limitations by analyzing viral genotypes sampled from infected plants after the course of a single infection, and therefore the invasion and co-infection of cells in various organs and tissues is very close to natural.</p><p id=\"__p21\">More than half of the genomes (53.8% ± 2.0%; see <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t001/\" rid-figpopup=\"pbio-0030089-t001\" rid-ob=\"ob-pbio-0030089-t001\" target=\"table\"><span>Table 1</span></a>) present in a CaMV population after a single passage in its host plant were identified as recombinants, and these data allowed us to infer a per nucleotide per generation recombination rate on the order of 2 × 10<sup>−5</sup> to 4 × 10<sup>−5</sup>. The time length of one generation, i.e., the time required for a given genome to go from one replication to the next, is totally unknown in plant viruses. The only experimental data available on CaMV are based on the kinetics of gene expression in infected protoplasts, where the capsid protein is produced between 48 and 72 h [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b40\" id=\"__tag_577434145\" rid=\"pbio-0030089-b40\">40</a>]. The reverse transcription and the encapsidation of genomic DNA being two coupled phenomena [<a class=\"bibr popnode\" href=\"#pbio-0030089-b30\" rid=\"pbio-0030089-b30\">30</a>], we judged it reasonable to assume a generation time of 2 d and, thus, an average of ten generations during our experiments. In case this estimate is mistaken, we have verified a linear relationship between <em>r</em> and the number of generations, thereby allowing an immediate adjustment of <em>r</em> if the CaMV generation time is more precisely established. At this point, we must consider that all cloned genomes may not have been through all the successive replication events potentially allowed by the timing of our experiments. It was previously shown that about 95% of CaMV mature virus particles accumulate in compact inclusion bodies [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b41\" id=\"__tag_577434168\" rid=\"pbio-0030089-b41\">41</a>], where they may be sequestered for a long time, as such inclusions are very frequent in all infected cells, including those in leaves that have been invaded by the virus population for several weeks. The viral population may thus present an age structure that could bias the estimation of the recombination rate. In order to minimize this bias, the clones we analyzed were collected in one young newly formed leaf, where the chances of finding genomes from “unsequestered lines” were assumed to be higher. In any case, our data analysis is conservative, since this age structure can only lead to an underestimation of the recombination rate.</p><p id=\"__p22\">Our results show that interferences between pairs of loci are negative: a recombination event between two loci apparently increases the probability of recombination between another pair of loci. We believe that the most parsimonious explanation of these negative interferences is based on the way the infection builds up within plant hosts. Indeed, one can divide infected host cells into those infected by a single virus genotype and those infected by more than one viral genotype. In the former, analogous to clonal propagation, recombination is undetectable. In the latter, recombination is not only detectable but, as our results indicate, very frequent. Samples consisting of viruses resulting from a mixture of these two types of host cell infections will thus contain viruses with no recombination and viruses with several recombination events, thus yielding an impression of negative interference. These conceptual arguments are supported by mathematical models. It is indeed easy to show (detailed results not shown) that if a proportion <em>F</em> of the population reproduces clonally, analogous to single infections, while the remaining reproduces panmictically, negative interferences could be inferred even if they do not exist. For example, assuming a three-locus model with real recombination rates <em>r</em>\n",
       "<sub>1</sub> and <em>r</em>\n",
       "<sub>2</sub> and interference <em>i</em>\n",
       "<sub>12</sub>, the “apparent” recombination and interference parameters, would be <em>r</em>\n",
       "<sub>1</sub> = (1 − <em>F</em>)r<sub>1</sub>, <em>r</em>\n",
       "<sub>2</sub> = (1 − <em>F</em>)<em>r</em>\n",
       "<sub>2</sub>, and <em>i</em>\n",
       "<sub>12</sub> = −(<em>F</em> − <em>i</em>\n",
       "<sub>12</sub>)/(1 − <em>F</em>). Interestingly, this example also shows that our estimates of the recombination rate are conservative: that a fraction <em>F</em> of host cells are singly infected while others are multiply infected leads to an underestimation of the recombination rate.</p><p id=\"__p23\">As judged by <em>r</em>\n",
       "<sub>1</sub>, <em>r</em>\n",
       "<sub>2</sub>, and <em>r</em>\n",
       "<sub>3</sub>, calculated between markers a–b, b–c, and c–d, respectively, we found evidence for recombination through the entire CaMV genome. The values for <em>r</em>\n",
       "<sub>1,</sub>\n",
       "<em>r</em>\n",
       "<sub>2</sub>, and <em>r</em>\n",
       "<sub>3</sub> are remarkably similar, hence the recombination sites seem to be evenly distributed along the genome. We considered the template-switching model as the major way recombinants are created in CaMV. As already mentioned in the Introduction, hot spots of template switching have been predicted at the position of the 5′ extremities of the 35S and 19S RNAs [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b21\" id=\"__tag_577434166\" rid=\"pbio-0030089-b21\">21</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b36\" id=\"__tag_577434165\" rid=\"pbio-0030089-b36\">36</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b42\" id=\"__tag_577434175\" rid=\"pbio-0030089-b42\">42</a>]. If other recombination mechanisms, such as that associated with second-strand DNA synthesis or with the host cell DNA repair machinery, act significantly, hot spots would be expected at the positions of the sequence interruption Δ1, Δ2, and Δ3 [<a class=\"bibr popnode\" href=\"#pbio-0030089-b43\" rid=\"pbio-0030089-b43\">43</a>]. Due to the design of our experiment and the position of the four markers, we have no information on putative hot spots at positions corresponding to the 5′ end of the 35S RNA and to Δ1 (at nucleotide position 0). Nevertheless, the putative hot spots at the 5′ end of the 19S RNA and at Δ2 and Δ3 (nucleotide positions 4,220 and 1,635, respectively) fall between marker pairs c–d, b–c, and a–b, respectively. Our results indicate that either these hot spots are quantitatively equivalent—though predicted by different recombination mechanisms—or, more likely, that they simply do not exist. Whatever the explanation, what we observe is that the CaMV can exchange any portion of its genome, and thus any gene thereof, with an astonishingly high frequency during the course of a single host infection.</p><p id=\"__p24\">To our knowledge, the viral recombination rate has never previously been quantified experimentally for a plant virus [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b03\" id=\"__tag_577434199\" rid=\"pbio-0030089-b03\">3</a>]. In contrast, retroviruses and particularly HIV-1 have been extensively investigated in that sense. As we have already discussed for these latter cases, the quantification of the intrinsic recombination rate was carried out in artificially co-infected cell cultures. The estimated intrinsic per nucleotide per generation recombination rate in HIV-1 is on the order of 10<sup>−4</sup> [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b14\" id=\"__tag_577434155\" rid=\"pbio-0030089-b14\">14</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b15\" id=\"__tag_577434163\" rid=\"pbio-0030089-b15\">15</a>,<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b19\" id=\"__tag_577434157\" rid=\"pbio-0030089-b19\">19</a>], less than one order of magnitude higher than our estimation for CaMV. Because for various reasons detailed above we probably underestimate the within-host CaMV recombination rate, we believe that the intrinsic recombination rate in CaMV is higher and perhaps on the order of that of HIV.</p><p class=\"p p-last\" id=\"__p25\">Other pararetroviruses such as plant badnaviruses or vertebrate hepadnaviruses have a similar cycle within their host cells, including steps of nuclear minichromosome, genomic size RNA synthesis, and reverse transcription and encapsidation. Nevertheless, vertebrate hepadnaviruses (e.g., hepatitis B virus) infect hosts that are very different from plants in their biology and physiology, and this could lead to a totally different frequency of cell co-infection during the development of the virus populations. Thus, even though our results can be informative for other pararetroviruses because of the viruses' shared biological characteristics, they should not be extrapolated to vertebrate pararetroviruses without caution.</p></div><div class=\"tsec sec\" id=\"s4\"><h2 class=\"head no_bottom_margin\" id=\"s4title\">Materials and Methods</h2><div class=\"sec sec-first-last\" id=\"s4a\"><div class=\"sec sec-first\" id=\"s4a1\"><p></p><h4 class=\"inline\" id=\"s4a1title\">Viral isolates </h4><p class=\"p p-first-last\" id=\"__p26\">We used the plasmid pCa37, which is the complete genome of the CaMV isolate Cabb-S, cloned into the pBR322 plasmid at the unique SalI restriction site [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b44\" id=\"__tag_577434154\" rid=\"pbio-0030089-b44\">44</a>]. To analyze recombination in different regions of the genome, we introduced four genetic markers: a, b, c, and d, at the positions 881, 3,539, 5,365, and 6,943, respectively, thus approximately at four cardinal points of the CaMV circular double-stranded DNA of 8,024 bp (<a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>). All markers, each corresponding to a single nucleotide change, were introduced by PCR-directed mutagenesis in pCa37, and resulted in the duplication of previously unique restriction sites BsiWI, PstI, MluI, and SacI in a plasmid designated pMark-S. Because, in this study, we targeted the possible exchange of genes between viral genomes, all markers a, b, c, and d were introduced within coding regions corresponding to open reading frames I, IV, V, and VI, respectively. Another important concern was to quantify recombination in the absence of selection, i.e., to create neutral markers. Consequently all markers consist of synonymous mutations (see below).</p></div><div class=\"sec\" id=\"s4a2\"><p></p><h4 class=\"inline\" id=\"s4a2title\">Production of viral particles and co-inoculation </h4><p class=\"p p-first\" id=\"__p27\">To generate the parental virus particles, plasmids pCa37 and pMark-S were mechanically inoculated into individual plants as previously described [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b33\" id=\"__tag_577434189\" rid=\"pbio-0030089-b33\">33</a>]. All plants were turnips (<em class=\"genus-species\">B. rapa</em> cv, “Just Right”) grown under glasshouse conditions at 23 °C with a 16/8 (light/dark) photoperiod. Thirty days post-inoculation, all symptomatic leaves were harvested and viral particles were purified as described earlier [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b45\" id=\"__tag_577434149\" rid=\"pbio-0030089-b45\">45</a>].</p><p class=\"p p-last\" id=\"__p28\">The resulting preparations of parental viruses, designated Cabb-S and Mark-S, were quantified by spectrometry using the formula described by Hull et al. [<a class=\"bibr popnode\" href=\"#pbio-0030089-b46\" rid=\"pbio-0030089-b46\">46</a>]. We fixed the initial frequency of markers to a value of 0.5, and a solution containing 0.1 mg/ml of virus particles of both Cabb-S and Mark-S at a 1:1 ratio was prepared. Plantlets were co-infected by mechanical inoculation of two to three leaves with 20 μl of this virus solution, using abrasive Celite AFA (Fluka, Ronkonkoma, New York, United States). The mixed CaMV population was allowed to grow during 21 d of systemic infection.</p></div><div class=\"sec\" id=\"s4a3\"><p></p><h4 class=\"inline\" id=\"s4a3title\">Estimation of marker frequency within mixed virus populations </h4><p class=\"p p-first-last\" id=\"__p29\">We designed an experimental protocol for quantifying marker frequency within a mixed Cabb-S/Mark-S virus population after a single passage in a host plant. Twenty-four individual plants, inoculated as above with equal amounts of Cabb-S and Mark-S, were harvested 21 d post-inoculation, when symptoms were fully developed. The viral DNA was purified from 200 mg of young newly formed infected leaves according to the protocol described previously [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b47\" id=\"__tag_577434141\" rid=\"pbio-0030089-b47\">47</a>]. After the precipitation step of this protocol, the viral DNA was resuspended and further purified with the Wizard DNA clean-up kit (Promega, Fitchburg, Wisconsin, United States) in TE 1X (100 mM Tris-HCl and 10 mM EDTA [pH 8]). Aliquots of viral DNA preparations were digested by restriction enzymes corresponding either to marker a, b, c, or d and submitted to a 1% agarose gel electrophoresis, colored by ethydium bromide and exposed to UV. Each individual restriction enzyme cut once in Cabb-S DNA and twice in Mark-S, thus generating DNA fragments of different sizes attributable to one or the other in the mixed population of CaMV genomes. After scanning the agarose gels, we estimated the relative frequency of the two genotypes in each viral DNA preparation and at each marker position, by densitometry using the NIH 1.62 Image program. The statistical analyses of the frequency of the four markers are described below.</p></div><div class=\"sec\" id=\"s4a4\"><p></p><h4 class=\"inline\" id=\"s4a4title\">Isolation of individual CaMV genomes and identification of recombinants </h4><p class=\"p p-first-last\" id=\"__p30\">To identify and quantify the recombinants within the CaMV mixed populations, aliquots from ten of the 24 viral DNA preparations described above were digested by the restriction enzyme SalI, and directly cloned into pUC19 at the corresponding site. In each of the ten viral populations analyzed, 50 full-genome-length clones were digested separately by BsiWI, PstI, MluI, and SacI, to test for the presence of marker a, b, c, and d, respectively. In this experiment, with the marker representing an additional restriction site, we could easily distinguish between the Cabb-S and the Mark-S genotype at all four marker positions, upon agarose gel (1%) electrophoresis of the digested clones. Clones with none or all four markers were parental genotypes, whereas clones harboring 1, 2, or 3 markers were clearly recombinants. Due to the very high number of recombinants detected, markers eventually appearing or disappearing due to spontaneous mutations were neglected.</p></div><div class=\"sec sec-last\" id=\"s4a5\"><p></p><h4 class=\"inline\" id=\"s4a5title\">Statistical analysis </h4><p class=\"p p-first\" id=\"__p31\">Here we present the different methods we used to quantify recombination in the CaMV genome. Because all these methods assume that the different markers are neutral, we first discuss assumption.</p><p id=\"__p32\">We used two datasets to test the neutrality of markers, both resulting from plants co-infected with a 1:1 ratio of Mark-S and Cabb-S. The first consisted of viral DNA densitometry data derived from 24 plants (described above), where for each plant we have an estimate of the frequency of each marker in the genome population. The second consisted of the restriction of 50 individual full-genome-length viral clones obtained from one co-infected plant (described above), yielding an estimate of the frequency of each marker, and this was repeated on ten different plants. The frequencies of the different markers were 0.508, 0.501, 0.516, and 0.507 for markers a, b, c, and d in the first dataset and 0.521, 0.518, 0.514, and 0.524 in the second dataset. We tested whether these frequencies were significantly different from the expected value under neutrality, 0.5, using either <em>t</em>-tests, for datasets where normality could not be rejected (seven out of eight cases), or Wilcoxon signed-rank non-parametric tests otherwise (marker c in the first dataset). In all cases <em>p</em>-values were larger than 0.05.</p><p id=\"__p33\">There are several cautionary remarks regarding these analyses. First, in all cases we found an excess of markers. Unfortunately, the two datasets cannot be regarded as independent because, even though the methods through which the frequency estimates were obtained were different, the plants used in the second dataset were a subset of the plants of the first. We thus have only four independent estimates in each case, and there is minimal power to detect significant deviations from neutrality with such a small sample size. It should be noted at this stage that deviations from the expected value could also be caused either by slight deviations from the 1:1 ratio in the infecting mixed solution, or by deviations from that ratio in the frequency of the viral particles that actually get into the plants. Second, because of the relatively small sample sizes and low statistical power, the tests presented above could have detected only large deviations.</p><p id=\"__p34\">The results clearly show, however, precisely that the markers do not have large effects, if any, and that therefore recombination estimates would be affected only very slightly by any hypothetical selective effects of the markers. Because of this, along with the fact that the introduced markers provoke silent substitutions in the CaMV genome, we assumed that markers were effectively neutral in the rest of the analysis.</p><p id=\"__p35\">The dataset used to estimate the recombination frequency consisted of the 500 full-genome-length viral clones (50 from each of ten co-infected plants) individually genotyped for each of the four markers. As discussed in detail in the Results, recombination was very frequent and concerned all four markers. Indeed, approximately half of the genotyped clones exhibited a recombinant genotype. It was therefore meaningful to try to obtain quantitative estimates of recombination from our data.</p><p id=\"__p36\">Our aim was to analyze viral recombination in a live host. Consequently, we had to deal with the fact that more than one viral replication cycle occurred during the 21 d that infection lasted in our experiment (we had to wait that long for the disease to develop and to be able to recover sufficient amounts of viral DNA from each infection). Based on the kinetics of gene expression [<a class=\"bibr popnode tag_hotlink tag_tooltip\" href=\"#pbio-0030089-b40\" id=\"__tag_577434181\" rid=\"pbio-0030089-b40\">40</a>], we postulate that each replication cycle lasts between 2 and 3 d, and that therefore seven to ten cycles occurred between infection and the sampling time. In case this assumption is incorrect, we did calculations assuming five, seven, ten, or 20 replication cycles during these 21 d. As shown, the results were not affected qualitatively, and only slightly quantitatively. It is important to note that we assumed that recombination occurred through a template-switching mechanism, and that therefore, from a recombination point of view, the CaMV genome is linear. The reverse transcription starts and finishes at the position 0 in <a class=\"fig-table-link figpopup\" co-legend-rid=\"lgnd_pbio-0030089-g001\" href=\"/pmc/articles/PMC1054884/figure/pbio-0030089-g001/\" rid-figpopup=\"pbio-0030089-g001\" rid-ob=\"ob-pbio-0030089-g001\" target=\"figure\"><span>Figure 1</span></a>, which is the point of circularization of the DNA genome. This implies that changes between contiguous markers a–b, b–c, and c–d can be considered as true recombination whereas those between a and d cannot, as they may simply stem from circularization of DNA, during the synthesis of which the polymerase has switched template once anywhere between a–b, b–c, or c–d.</p><p id=\"__p37\">To estimate the recombination rate between markers, we wrote recurrence equations describing the change in frequency of each genotype over one generation, assuming random mating and no selection (i.e., the standard Wright–Fisher population genetics model). We then expressed the frequency of all possible genotypes <em>n</em> generations later as a function of their initial frequency and of the recombination parameters. Subsequently we calculated the maximum likelihood estimates of the recombination parameters and their asymptotic variances given initial frequencies (we assumed that the two “parental” genotypes, Mark-S and Cabb-S, had equal initial frequencies of 0.5 and that all other genotypes had initial frequencies of zero) and frequencies after <em>n</em> generations (the observed frequencies; as stated above we used different values of <em>n</em>). All algebraic and numerical calculations were carried out with the software Mathematica.</p><p id=\"__p38\">The recombination parameters are the recombination rates between two adjacent loci, e.g., <em>r</em>\n",
       "<sub>1</sub> for the recombination rate between markers a and b, and the interference coefficients, e.g., <em>i</em>\n",
       "<sub>12</sub> for interference between recombination events in the segments between markers a and b and b and c. To define these parameters we followed Christiansen [<a class=\"bibr popnode\" href=\"#pbio-0030089-b48\" rid=\"pbio-0030089-b48\">48</a>], and in particular the recombination distributions for two, three, and four loci (respectively, <a class=\"fig-table-link figpopup\" co-legend-rid=\"\" href=\"/pmc/articles/PMC1054884/table/pbio-0030089-t002/\" rid-figpopup=\"pbio-0030089-t002\" rid-ob=\"ob-pbio-0030089-t002\" target=\"table\"><span>Tables 2</span></a>.7, 2.8, and 2.9 of [<a class=\"bibr popnode\" href=\"#pbio-0030089-b48\" rid=\"pbio-0030089-b48\">48</a>]). It is important to realize that given the definitions of these parameters, the estimator of the recombination rate between two loci is not affected by the number of loci considered. In other words, we obtain the same estimation of the recombination rate between markers a and b whether we consider genotypic frequencies at just these two loci, or the frequencies at these two loci plus a third locus, or the complete information to which we have access, the four-marker genotypes. Information on additional loci only affects the estimates of the interference coefficients.</p><p class=\"p p-last\" id=\"__p39\">It proved impossible to carry out the calculations for four loci algebraically. Instead, we used a computer program to calculate the expected genotypic frequencies at all four loci after <em>n</em> generations, given the above stated initial frequencies and specified recombination parameters. For each combination of recombination parameters we calculated a Euclidean distance between the vector of the expected genotypic frequencies and the observed genotypic frequencies, and considered that the estimated recombination parameters were those yielding the minimal Euclidean distance. In all cases, the estimated recombination rates between pairs of loci were equal to the second decimal to those estimated algebraically from data for three or two loci.</p></div></div></div><div class=\"tsec sec\" id=\"idm140550462018496\"><h2 class=\"head no_bottom_margin\" id=\"idm140550462018496title\">Acknowledgments</h2><div class=\"sec\"><p id=\"__p40\">\n",
       "<strong>Competing interests.</strong> The authors have declared that no competing interests exist.</p></div></div><div class=\"tsec sec\" id=\"idm140550460948112\"><h2 class=\"head no_bottom_margin\" id=\"idm140550460948112title\">Abbreviation</h2><div class=\"bk-sec\"><div><table class=\"default-table glossary\"><tbody><tr><td>CaMV</td><td>cauliflower mosaic virus</td></tr></tbody></table></div></div></div><div class=\"tsec sec headless whole_rhythm\" id=\"idm140550466165536\"><!--back/fn-group--><div class=\"fm-sec half_rhythm small\"><div><p class=\"p p-first-last\" id=\"__p41\">\n",
       "<strong>Author contributions.</strong> RF, SB, and YM conceived and designed the experiments. RF, MU, LG, and SB performed the experiments. RF, DR, SB, and YM analyzed the data. RF, DR, SB, and YM wrote the paper.</p></div></div></div><div class=\"tsec sec headless whole_rhythm\" id=\"idm140550466163728\"><!--back/fn-group--><div class=\"fm-sec half_rhythm small\"><p class=\"fn sec\" id=\"n2\"><p class=\"p p-first-last\" id=\"__p42\">¤Current address: Institut National de la Santé et de la Recherche Médicale (INSERM), Equipe Ecologie et Evolution des micro-organismes E 0339, Paris, France</p></p><p class=\"fn sec\" id=\"n3\"><p class=\"p p-first-last\" id=\"__p43\">Citation: Froissart R, Roze D, Uzest M, Galibert L, Blanc S, et al. (2005) Recombination every day: Abundant recombination in a virus during a single multi-cellular host infection. PLoS Biol 3(3): e89.</p></p></div></div><div class=\"tsec sec\" id=\"idm140550502083280\"><h2 class=\"head no_bottom_margin\" id=\"idm140550502083280title\">References</h2><div class=\"ref-list-sec sec\" id=\"reference-list\"><ul class=\"back-ref-list\" style=\"list-style-type:decimal;\"><li id=\"pbio-0030089-b01\"><span class=\"element-citation\">Awadalla P. The evolutionary genomics of pathogen recombination. <span><span class=\"ref-journal\">Nat Rev Genet. </span>2003;<span class=\"ref-vol\">4</span>:50–60.</span> [<a href=\"/pubmed/12509753\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Nat+Rev+Genet&amp;title=The+evolutionary+genomics+of+pathogen+recombination&amp;author=P+Awadalla&amp;volume=4&amp;publication_year=2003&amp;pages=50-60&amp;pmid=12509753&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b02\"><span class=\"element-citation\">Hendrix RW, Hatfull GF, Smith MC. Bacteriophages with tails: Chasing their origins and evolution. <span><span class=\"ref-journal\">Res Microbiol. </span>2003;<span class=\"ref-vol\">154</span>:253–257.</span> [<a href=\"/pubmed/12798229\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Res+Microbiol&amp;title=Bacteriophages+with+tails:+Chasing+their+origins+and+evolution&amp;author=RW+Hendrix&amp;author=GF+Hatfull&amp;author=MC+Smith&amp;volume=154&amp;publication_year=2003&amp;pages=253-257&amp;pmid=12798229&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b03\"><span class=\"element-citation\">Garcia-Arenal F, Fraile A, Malpica JM. Variability and genetic structure of plant virus populations. <span><span class=\"ref-journal\">Annu Rev Phytopathol. </span>2001;<span class=\"ref-vol\">39</span>:157–186.</span> [<a href=\"/pubmed/11701863\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Annu+Rev+Phytopathol&amp;title=Variability+and+genetic+structure+of+plant+virus+populations&amp;author=F+Garcia-Arenal&amp;author=A+Fraile&amp;author=JM+Malpica&amp;volume=39&amp;publication_year=2001&amp;pages=157-186&amp;pmid=11701863&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b04\"><span class=\"element-citation\">Brown DW. Threat to humans from virus infections of non-human primates. <span><span class=\"ref-journal\">Rev Med Virol. </span>1997;<span class=\"ref-vol\">7</span>:239–246.</span> [<a href=\"/pubmed/10398488\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Rev+Med+Virol&amp;title=Threat+to+humans+from+virus+infections+of+non-human+primates&amp;author=DW+Brown&amp;volume=7&amp;publication_year=1997&amp;pages=239-246&amp;pmid=10398488&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b05\"><span class=\"element-citation\">Gibbs MJ, Weiller GF. Evidence that a plant virus switched hosts to infect a vertebrate and then recombined with a vertebrate-infecting virus. <span><span class=\"ref-journal\">Proc Natl Acad Sci U S A. </span>1999;<span class=\"ref-vol\">96</span>:8022–8027.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC22181/\">PMC free article</a>]</span> [<a href=\"/pubmed/10393941\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Proc+Natl+Acad+Sci+U+S+A&amp;title=Evidence+that+a+plant+virus+switched+hosts+to+infect+a+vertebrate+and+then+recombined+with+a+vertebrate-infecting+virus&amp;author=MJ+Gibbs&amp;author=GF+Weiller&amp;volume=96&amp;publication_year=1999&amp;pages=8022-8027&amp;pmid=10393941&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b06\"><span class=\"element-citation\">Malim MH, Emerman M. HIV-1 sequence variation: Drift, shift, and attenuation. <span><span class=\"ref-journal\">Cell. </span>2001;<span class=\"ref-vol\">104</span>:469–472.</span> [<a href=\"/pubmed/11239404\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=HIV-1+sequence+variation:+Drift,+shift,+and+attenuation&amp;author=MH+Malim&amp;author=M+Emerman&amp;volume=104&amp;publication_year=2001&amp;pages=469-472&amp;pmid=11239404&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b07\"><span class=\"element-citation\">Hu WS, Rhodes T, Dang Q, Pathak V. Retroviral recombination: Review of genetic analyses. <span><span class=\"ref-journal\">Front Biosci. </span>2003;<span class=\"ref-vol\">8</span>:D143–D155.</span> [<a href=\"/pubmed/12456341\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Front+Biosci&amp;title=Retroviral+recombination:+Review+of+genetic+analyses&amp;author=WS+Hu&amp;author=T+Rhodes&amp;author=Q+Dang&amp;author=V+Pathak&amp;volume=8&amp;publication_year=2003&amp;pages=D143-D155&amp;pmid=12456341&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b08\"><span class=\"element-citation\">Gibbs MJ, Armstrong JS, Gibbs AJ. Recombination in the hemagglutinin gene of the 1918 “Spanish flu” <span><span class=\"ref-journal\">Science. </span>2001;<span class=\"ref-vol\">293</span>:1842–1845.</span> [<a href=\"/pubmed/11546876\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Science&amp;title=Recombination+in+the+hemagglutinin+gene+of+the+1918+“Spanish+flu”&amp;author=MJ+Gibbs&amp;author=JS+Armstrong&amp;author=AJ+Gibbs&amp;volume=293&amp;publication_year=2001&amp;pages=1842-1845&amp;pmid=11546876&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b09\"><span class=\"element-citation\">Rest JS, Mindell DP. SARS associated coronavirus has a recombinant polymerase and coronaviruses have a history of host-shifting. <span><span class=\"ref-journal\">Infect Genet Evol. </span>2003;<span class=\"ref-vol\">3</span>:219–225.</span> [<a href=\"/pubmed/14522185\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Infect+Genet+Evol&amp;title=SARS+associated+coronavirus+has+a+recombinant+polymerase+and+coronaviruses+have+a+history+of+host-shifting&amp;author=JS+Rest&amp;author=DP+Mindell&amp;volume=3&amp;publication_year=2003&amp;pages=219-225&amp;pmid=14522185&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b10\"><span class=\"element-citation\">Chao L. Evolution of genetic exchange in RNA viruses. In: Morse SS, editor. The evolutionary biology of viruses. New York: Raven Press. pp. <span>1994:233–250.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar?q=Chao+L+Evolution+of+genetic+exchange+in+RNA+viruses.+In:+Morse+SS,+editor.+The+evolutionary+biology+of+viruses.+New+York:+Raven+Press.+pp+1994+233+250+\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b11\"><span class=\"element-citation\">Worobey M, Holmes EC. Evolutionary aspects of recombination in RNA viruses. <span><span class=\"ref-journal\">J Gen Virol. </span>1999;<span class=\"ref-vol\">80</span>:2535–2543.</span> [<a href=\"/pubmed/10573145\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Evolutionary+aspects+of+recombination+in+RNA+viruses&amp;author=M+Worobey&amp;author=EC+Holmes&amp;volume=80&amp;publication_year=1999&amp;pages=2535-2543&amp;pmid=10573145&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b12\"><span class=\"element-citation\">Kirkegaard K, Baltimore D. The mechanism of RNA recombination in poliovirus. <span><span class=\"ref-journal\">Cell. </span>1986;<span class=\"ref-vol\">47</span>:433–443.</span> [<a href=\"/pubmed/3021340\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=The+mechanism+of+RNA+recombination+in+poliovirus&amp;author=K+Kirkegaard&amp;author=D+Baltimore&amp;volume=47&amp;publication_year=1986&amp;pages=433-443&amp;pmid=3021340&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b13\"><span class=\"element-citation\">Clavel F, Hoggan MD, Willey RL, Strebel K, Martin MA, et al.  Genetic recombination of human immunodeficiency virus. <span><span class=\"ref-journal\">J Virol. </span>1989;<span class=\"ref-vol\">63</span>:1455–1459.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC247851/\">PMC free article</a>]</span> [<a href=\"/pubmed/2915387\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Genetic+recombination+of+human+immunodeficiency+virus&amp;author=F+Clavel&amp;author=MD+Hoggan&amp;author=RL+Willey&amp;author=K+Strebel&amp;author=MA+Martin&amp;volume=63&amp;publication_year=1989&amp;pages=1455-1459&amp;pmid=2915387&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b14\"><span class=\"element-citation\">Jetzt AE, Yu H, Klarmann GJ, Ron Y, Preston BD, et al.  High rate of recombination throughout the human immunodeficiency virus type 1 genome. <span><span class=\"ref-journal\">J Virol. </span>2000;<span class=\"ref-vol\">74</span>:1234–1240.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC111457/\">PMC free article</a>]</span> [<a href=\"/pubmed/10627533\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=High+rate+of+recombination+throughout+the+human+immunodeficiency+virus+type+1+genome&amp;author=AE+Jetzt&amp;author=H+Yu&amp;author=GJ+Klarmann&amp;author=Y+Ron&amp;author=BD+Preston&amp;volume=74&amp;publication_year=2000&amp;pages=1234-1240&amp;pmid=10627533&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b15\"><span class=\"element-citation\">Rhodes T, Wargo H, Hu WS. High rates of human immunodeficiency virus type 1 recombination: Near-random segregation of markers one kilobase apart in one round of viral replication. <span><span class=\"ref-journal\">J Virol. </span>2003;<span class=\"ref-vol\">77</span>:11193–11200.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC224990/\">PMC free article</a>]</span> [<a href=\"/pubmed/14512567\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=High+rates+of+human+immunodeficiency+virus+type+1+recombination:+Near-random+segregation+of+markers+one+kilobase+apart+in+one+round+of+viral+replication&amp;author=T+Rhodes&amp;author=H+Wargo&amp;author=WS+Hu&amp;volume=77&amp;publication_year=2003&amp;pages=11193-11200&amp;pmid=14512567&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b16\"><span class=\"element-citation\">Keck JG, Matsushima GK, Makino S, Fleming JO, Vannier DM, et al.  In vivo RNA-RNA recombination of coronavirus in mouse brain. <span><span class=\"ref-journal\">J Virol. </span>1988;<span class=\"ref-vol\">62</span>:1810–1813.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC253235/\">PMC free article</a>]</span> [<a href=\"/pubmed/2833625\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=In+vivo+RNA-RNA+recombination+of+coronavirus+in+mouse+brain&amp;author=JG+Keck&amp;author=GK+Matsushima&amp;author=S+Makino&amp;author=JO+Fleming&amp;author=DM+Vannier&amp;volume=62&amp;publication_year=1988&amp;pages=1810-1813&amp;pmid=2833625&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b17\"><span class=\"element-citation\">Fraile A, Alonso-Prados JL, Aranda MA, Bernal JJ, Malpica JM, et al.  Genetic exchange by recombination or reassortment is infrequent in natural populations of a tripartite RNA plant virus. <span><span class=\"ref-journal\">J Virol. </span>1997;<span class=\"ref-vol\">71</span>:934–940.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC191141/\">PMC free article</a>]</span> [<a href=\"/pubmed/8995610\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Genetic+exchange+by+recombination+or+reassortment+is+infrequent+in+natural+populations+of+a+tripartite+RNA+plant+virus&amp;author=A+Fraile&amp;author=JL+Alonso-Prados&amp;author=MA+Aranda&amp;author=JJ+Bernal&amp;author=JM+Malpica&amp;volume=71&amp;publication_year=1997&amp;pages=934-940&amp;pmid=8995610&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b18\"><span class=\"element-citation\">Bruyere A, Wantroba M, Flasinski S, Dzianott A, Bujarski JJ. Frequent homologous recombination events between molecules of one RNA component in a multipartite RNA virus. <span><span class=\"ref-journal\">J Virol. </span>2000;<span class=\"ref-vol\">74</span>:4214–4219.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC111936/\">PMC free article</a>]</span> [<a href=\"/pubmed/10756034\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Frequent+homologous+recombination+events+between+molecules+of+one+RNA+component+in+a+multipartite+RNA+virus&amp;author=A+Bruyere&amp;author=M+Wantroba&amp;author=S+Flasinski&amp;author=A+Dzianott&amp;author=JJ+Bujarski&amp;volume=74&amp;publication_year=2000&amp;pages=4214-4219&amp;pmid=10756034&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b19\"><span class=\"element-citation\">Zhuang J, Jetzt AE, Sun G, Yu H, Klarmann G, et al.  Human immunodeficiency virus type 1 recombination: Rate, fidelity, and putative hot spots. <span><span class=\"ref-journal\">J Virol. </span>2002;<span class=\"ref-vol\">76</span>:11273–11282.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC136766/\">PMC free article</a>]</span> [<a href=\"/pubmed/12388687\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Virol&amp;title=Human+immunodeficiency+virus+type+1+recombination:+Rate,+fidelity,+and+putative+hot+spots&amp;author=J+Zhuang&amp;author=AE+Jetzt&amp;author=G+Sun&amp;author=H+Yu&amp;author=G+Klarmann&amp;volume=76&amp;publication_year=2002&amp;pages=11273-11282&amp;pmid=12388687&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b20\"><span class=\"element-citation\">Bujarski JJ, Kaesberg P. Genetic recombination between RNA components of a multipartite plant virus. <span><span class=\"ref-journal\">Nature. </span>1986;<span class=\"ref-vol\">321</span>:528–531.</span> [<a href=\"/pubmed/2423879\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genetic+recombination+between+RNA+components+of+a+multipartite+plant+virus&amp;author=JJ+Bujarski&amp;author=P+Kaesberg&amp;volume=321&amp;publication_year=1986&amp;pages=528-531&amp;pmid=2423879&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b21\"><span class=\"element-citation\">Vaden VR, Melcher U. Recombination sites in cauliflower mosaic virus DNAs: Implications for mechanisms of recombination. <span><span class=\"ref-journal\">Virology. </span>1990;<span class=\"ref-vol\">177</span>:717–726.</span> [<a href=\"/pubmed/2371775\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Recombination+sites+in+cauliflower+mosaic+virus+DNAs:+Implications+for+mechanisms+of+recombination&amp;author=VR+Vaden&amp;author=U+Melcher&amp;volume=177&amp;publication_year=1990&amp;pages=717-726&amp;pmid=2371775&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b22\"><span class=\"element-citation\">Banner LR, Lai MM. Random nature of coronavirus RNA recombination in the absence of selection pressure. <span><span class=\"ref-journal\">Virology. </span>1991;<span class=\"ref-vol\">185</span>:441–445.</span> [<a href=\"/pubmed/1656597\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Random+nature+of+coronavirus+RNA+recombination+in+the+absence+of+selection+pressure&amp;author=LR+Banner&amp;author=MM+Lai&amp;volume=185&amp;publication_year=1991&amp;pages=441-445&amp;pmid=1656597&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b23\"><span class=\"element-citation\">Kottier SA, Cavanagh D, Britton P. Experimental evidence of recombination in coronavirus infectious bronchitis virus. <span><span class=\"ref-journal\">Virology. </span>1995;<span class=\"ref-vol\">213</span>:569–580.</span> [<a href=\"/pubmed/7491781\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Experimental+evidence+of+recombination+in+coronavirus+infectious+bronchitis+virus&amp;author=SA+Kottier&amp;author=D+Cavanagh&amp;author=P+Britton&amp;volume=213&amp;publication_year=1995&amp;pages=569-580&amp;pmid=7491781&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b24\"><span class=\"element-citation\">Revers F, Le Gall O, Candresse T, Le Romancer M, Dunez J. Frequent occurrence of recombinant potyvirus isolates. <span><span class=\"ref-journal\">J Gen Virol. </span>1996;<span class=\"ref-vol\">77</span>:1953–1965.</span> [<a href=\"/pubmed/8760448\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Frequent+occurrence+of+recombinant+potyvirus+isolates&amp;author=F+Revers&amp;author=O+Le+Gall&amp;author=T+Candresse&amp;author=M+Le+Romancer&amp;author=J+Dunez&amp;volume=77&amp;publication_year=1996&amp;pages=1953-1965&amp;pmid=8760448&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b25\"><span class=\"element-citation\">Aaziz R, Tepfer M. Recombination between genomic RNAs of two cucumoviruses under conditions of minimal selection pressure. <span><span class=\"ref-journal\">Virology. </span>1999;<span class=\"ref-vol\">263</span>:282–289.</span> [<a href=\"/pubmed/10544102\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Recombination+between+genomic+RNAs+of+two+cucumoviruses+under+conditions+of+minimal+selection+pressure&amp;author=R+Aaziz&amp;author=M+Tepfer&amp;volume=263&amp;publication_year=1999&amp;pages=282-289&amp;pmid=10544102&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b26\"><span class=\"element-citation\">McVean G, Awadalla P, Fearnhead P. A coalescent-based method for detecting and estimating recombination from gene sequences. <span><span class=\"ref-journal\">Genetics. </span>2002;<span class=\"ref-vol\">160</span>:1231–1241.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC1462015/\">PMC free article</a>]</span> [<a href=\"/pubmed/11901136\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Genetics&amp;title=A+coalescent-based+method+for+detecting+and+estimating+recombination+from+gene+sequences&amp;author=G+McVean&amp;author=P+Awadalla&amp;author=P+Fearnhead&amp;volume=160&amp;publication_year=2002&amp;pages=1231-1241&amp;pmid=11901136&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b27\"><span class=\"element-citation\">Stumpf MP, McVean GA. Estimating recombination rates from population-genetic data. <span><span class=\"ref-journal\">Nat Rev Genet. </span>2003;<span class=\"ref-vol\">4</span>:959–968.</span> [<a href=\"/pubmed/14631356\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Nat+Rev+Genet&amp;title=Estimating+recombination+rates+from+population-genetic+data&amp;author=MP+Stumpf&amp;author=GA+McVean&amp;volume=4&amp;publication_year=2003&amp;pages=959-968&amp;pmid=14631356&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b28\"><span class=\"element-citation\">Rothnie HM, Chapdelaine Y, Hohn T. Pararetroviruses and retroviruses: A comparative review of viral structure and gene expression strategies. <span><span class=\"ref-journal\">Adv Virus Res. </span>1994;<span class=\"ref-vol\">44</span>:1–67.</span> [<a href=\"/pubmed/7817872\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Adv+Virus+Res&amp;title=Pararetroviruses+and+retroviruses:+A+comparative+review+of+viral+structure+and+gene+expression+strategies&amp;author=HM+Rothnie&amp;author=Y+Chapdelaine&amp;author=T+Hohn&amp;volume=44&amp;publication_year=1994&amp;pages=1-67&amp;pmid=7817872&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b29\"><span class=\"element-citation\">Mason WS, Taylor JM, Hull R. Retroid virus genome replication. <span><span class=\"ref-journal\">Adv Virus Res. </span>1987;<span class=\"ref-vol\">32</span>:35–96.</span> [<a href=\"/pubmed/2441578\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Adv+Virus+Res&amp;title=Retroid+virus+genome+replication&amp;author=WS+Mason&amp;author=JM+Taylor&amp;author=R+Hull&amp;volume=32&amp;publication_year=1987&amp;pages=35-96&amp;pmid=2441578&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b30\"><span class=\"element-citation\">Hohn T, Fütterer J. The proteins and functions of plant pararetroviruses: Knowns and unknowns. <span><span class=\"ref-journal\">Crit Rev Plant Sci. </span>1997;<span class=\"ref-vol\">16</span>:133–167.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Crit+Rev+Plant+Sci&amp;title=The+proteins+and+functions+of+plant+pararetroviruses:+Knowns+and+unknowns&amp;author=T+Hohn&amp;author=J+Fütterer&amp;volume=16&amp;publication_year=1997&amp;pages=133-167&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b31\"><span class=\"element-citation\">Fütterer J, Hohn T. Translation in plants—Rules and exceptions. <span><span class=\"ref-journal\">Plant Mol Biol. </span>1996;<span class=\"ref-vol\">32</span>:159–189.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC7089402/\">PMC free article</a>]</span> [<a href=\"/pubmed/8980479\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Plant+Mol+Biol&amp;title=Translation+in+plants—Rules+and+exceptions&amp;author=J+Fütterer&amp;author=T+Hohn&amp;volume=32&amp;publication_year=1996&amp;pages=159-189&amp;pmid=8980479&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b32\"><span class=\"element-citation\">Melcher U, Choe IS, Lebeurier G, Richards K, Essenberg RC. Selective allele loss and interference between cauliflower mosaic virus DNAs. <span><span class=\"ref-journal\">Mol Gen Genet. </span>1986;<span class=\"ref-vol\">203</span>:230–236.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Mol+Gen+Genet&amp;title=Selective+allele+loss+and+interference+between+cauliflower+mosaic+virus+DNAs&amp;author=U+Melcher&amp;author=IS+Choe&amp;author=G+Lebeurier&amp;author=K+Richards&amp;author=RC+Essenberg&amp;volume=203&amp;publication_year=1986&amp;pages=230-236&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b33\"><span class=\"element-citation\">Melcher U, Steffens DL, Lyttle DJ, Lebeurier G, Lin H, et al.  Infectious and non-infectious mutants of cauliflower mosaic virus DNA. <span><span class=\"ref-journal\">J Gen Virol. </span>1986;<span class=\"ref-vol\">67</span>:1491–1498.</span> [<a href=\"/pubmed/3723112\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Infectious+and+non-infectious+mutants+of+cauliflower+mosaic+virus+DNA&amp;author=U+Melcher&amp;author=DL+Steffens&amp;author=DJ+Lyttle&amp;author=G+Lebeurier&amp;author=H+Lin&amp;volume=67&amp;publication_year=1986&amp;pages=1491-1498&amp;pmid=3723112&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b34\"><span class=\"element-citation\">Zhang XS, Melcher U. Competition between isolates and variants of cauliflower mosaic virus in infected turnip plants. <span><span class=\"ref-journal\">J Gen Virol. </span>1990;<span class=\"ref-vol\">70</span>:3427–3437.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Competition+between+isolates+and+variants+of+cauliflower+mosaic+virus+in+infected+turnip+plants&amp;author=XS+Zhang&amp;author=U+Melcher&amp;volume=70&amp;publication_year=1990&amp;pages=3427-3437&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b35\"><span class=\"element-citation\">Choe S, Melcher U, Richards K, Lebeurier G, Essenberg RC. Recombination between mutant CaMV DNAs. <span><span class=\"ref-journal\">Plant Mol Biol. </span>1985;<span class=\"ref-vol\">5</span>:281–289.</span> [<a href=\"/pubmed/24306919\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Plant+Mol+Biol&amp;title=Recombination+between+mutant+CaMV+DNAs&amp;author=S+Choe&amp;author=U+Melcher&amp;author=K+Richards&amp;author=G+Lebeurier&amp;author=RC+Essenberg&amp;volume=5&amp;publication_year=1985&amp;pages=281-289&amp;pmid=24306919&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b36\"><span class=\"element-citation\">Gal S, Pisan B, Hohn T, Grimsley N, Hohn B. Agroinfection of transgenic plants leads to viable cauliflower mosaic virus by intermolecular recombination. <span><span class=\"ref-journal\">Virology. </span>1992;<span class=\"ref-vol\">187</span>:525–533.</span> [<a href=\"/pubmed/1546451\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=Agroinfection+of+transgenic+plants+leads+to+viable+cauliflower+mosaic+virus+by+intermolecular+recombination&amp;author=S+Gal&amp;author=B+Pisan&amp;author=T+Hohn&amp;author=N+Grimsley&amp;author=B+Hohn&amp;volume=187&amp;publication_year=1992&amp;pages=525-533&amp;pmid=1546451&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b37\"><span class=\"element-citation\">Chenault KD, Melcher U. Phylogenetic relationships reveal recombination among isolates of cauliflower mosaic virus. <span><span class=\"ref-journal\">J Mol Evol. </span>1994;<span class=\"ref-vol\">39</span>:496–505.</span> [<a href=\"/pubmed/7807539\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Mol+Evol&amp;title=Phylogenetic+relationships+reveal+recombination+among+isolates+of+cauliflower+mosaic+virus&amp;author=KD+Chenault&amp;author=U+Melcher&amp;volume=39&amp;publication_year=1994&amp;pages=496-505&amp;pmid=7807539&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b38\"><span class=\"element-citation\">Geldreich A, Lebeurier G, Hirth L. In vivo dimerization of cauliflower mosaic virus DNA can explain recombination. <span><span class=\"ref-journal\">Gene. </span>1986;<span class=\"ref-vol\">48</span>:277–286.</span> [<a href=\"/pubmed/3557131\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Gene&amp;title=In+vivo+dimerization+of+cauliflower+mosaic+virus+DNA+can+explain+recombination&amp;author=A+Geldreich&amp;author=G+Lebeurier&amp;author=L+Hirth&amp;volume=48&amp;publication_year=1986&amp;pages=277-286&amp;pmid=3557131&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b39\"><span class=\"element-citation\">Grimsley N, Hohn T, Hohn B. Recombination in a plant virus: Template-switching in cauliflower mosaic virus. <span><span class=\"ref-journal\">EMBO J. </span>1986;<span class=\"ref-vol\">5</span>:641–646.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC1166838/\">PMC free article</a>]</span> [<a href=\"/pubmed/16453678\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=EMBO+J&amp;title=Recombination+in+a+plant+virus:+Template-switching+in+cauliflower+mosaic+virus&amp;author=N+Grimsley&amp;author=T+Hohn&amp;author=B+Hohn&amp;volume=5&amp;publication_year=1986&amp;pages=641-646&amp;pmid=16453678&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b40\"><span class=\"element-citation\">Kobayashi K, Nakayashiki H, Tsuge S, Mise K, Furusawa I. Accumulation kinetics of viral gene products in cauliflower mosaic virus-infected turnip protoplasts. <span><span class=\"ref-journal\">Microbiol Immunol. </span>1998;<span class=\"ref-vol\">42</span>:65–69.</span> [<a href=\"/pubmed/9525783\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Microbiol+Immunol&amp;title=Accumulation+kinetics+of+viral+gene+products+in+cauliflower+mosaic+virus-infected+turnip+protoplasts&amp;author=K+Kobayashi&amp;author=H+Nakayashiki&amp;author=S+Tsuge&amp;author=K+Mise&amp;author=I+Furusawa&amp;volume=42&amp;publication_year=1998&amp;pages=65-69&amp;pmid=9525783&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b41\"><span class=\"element-citation\">Drucker M, Froissart R, Hebrard E, Uzest M, Ravallec M, et al.  Intracellular distribution of viral gene products regulates a complex mechanism of cauliflower mosaic virus acquisition by its aphid vector. <span><span class=\"ref-journal\">Proc Natl Acad Sci U S A. </span>2002;<span class=\"ref-vol\">99</span>:2422–2427.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC122380/\">PMC free article</a>]</span> [<a href=\"/pubmed/11842201\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Proc+Natl+Acad+Sci+U+S+A&amp;title=Intracellular+distribution+of+viral+gene+products+regulates+a+complex+mechanism+of+cauliflower+mosaic+virus+acquisition+by+its+aphid+vector&amp;author=M+Drucker&amp;author=R+Froissart&amp;author=E+Hebrard&amp;author=M+Uzest&amp;author=M+Ravallec&amp;volume=99&amp;publication_year=2002&amp;pages=2422-2427&amp;pmid=11842201&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b42\"><span class=\"element-citation\">Schoelz JE, Wintermantel WM. Expansion of viral host range through complementation and recombination in transgenic plants. <span><span class=\"ref-journal\">Plant Cell. </span>1993;<span class=\"ref-vol\">5</span>:1669–1679.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC160395/\">PMC free article</a>]</span> [<a href=\"/pubmed/12271051\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Plant+Cell&amp;title=Expansion+of+viral+host+range+through+complementation+and+recombination+in+transgenic+plants&amp;author=JE+Schoelz&amp;author=WM+Wintermantel&amp;volume=5&amp;publication_year=1993&amp;pages=1669-1679&amp;pmid=12271051&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b43\"><span class=\"element-citation\">Hull R.  <span class=\"ref-journal\">Matthews' plant virology, 4th ed.</span> San Diego: Academic Press; 2001. 1,001 pp. <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?title=Matthews'+plant+virology,+4th+ed&amp;author=R+Hull&amp;publication_year=2001&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b44\"><span class=\"element-citation\">Franck A, Guilley H, Jonard G, Richards K, Hirth L. Nucleotide sequence of cauliflower mosaic virus DNA. <span><span class=\"ref-journal\">Cell. </span>1980;<span class=\"ref-vol\">21</span>:285–294.</span> [<a href=\"/pubmed/7407912\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=Nucleotide+sequence+of+cauliflower+mosaic+virus+DNA&amp;author=A+Franck&amp;author=H+Guilley&amp;author=G+Jonard&amp;author=K+Richards&amp;author=L+Hirth&amp;volume=21&amp;publication_year=1980&amp;pages=285-294&amp;pmid=7407912&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b45\"><span class=\"element-citation\">Leh V, Jacquot E, Geldreich A, Hermann T, Leclerc D, et al.  Aphid transmission of cauliflower mosaic virus requires the viral PIII protein. <span><span class=\"ref-journal\">EMBO J. </span>1999;<span class=\"ref-vol\">18</span>:7077–7085.</span> <span class=\"nowrap\">[<a class=\"int-reflink\" href=\"/pmc/articles/PMC1171770/\">PMC free article</a>]</span> [<a href=\"/pubmed/10601029\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=EMBO+J&amp;title=Aphid+transmission+of+cauliflower+mosaic+virus+requires+the+viral+PIII+protein&amp;author=V+Leh&amp;author=E+Jacquot&amp;author=A+Geldreich&amp;author=T+Hermann&amp;author=D+Leclerc&amp;volume=18&amp;publication_year=1999&amp;pages=7077-7085&amp;pmid=10601029&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b46\"><span class=\"element-citation\">Hull R, Shepherd RJ, Harvey JD. Cauliflower mosaic virus: An improved purification procedure and some properties of the virus particule. <span><span class=\"ref-journal\">J Gen Virol. </span>1976;<span class=\"ref-vol\">31</span>:93–100.</span> <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=J+Gen+Virol&amp;title=Cauliflower+mosaic+virus:+An+improved+purification+procedure+and+some+properties+of+the+virus+particule&amp;author=R+Hull&amp;author=RJ+Shepherd&amp;author=JD+Harvey&amp;volume=31&amp;publication_year=1976&amp;pages=93-100&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b47\"><span class=\"element-citation\">Gardner RC, Shepherd RJ. A procedure for rapid isolation and analysis of cauliflower mosaic virus DNA. <span><span class=\"ref-journal\">Virology. </span>1980;<span class=\"ref-vol\">106</span>:159–161.</span> [<a href=\"/pubmed/18631716\" ref=\"reftype=pubmed&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord\" target=\"pmc_ext\">PubMed</a>] <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?journal=Virology&amp;title=A+procedure+for+rapid+isolation+and+analysis+of+cauliflower+mosaic+virus+DNA&amp;author=RC+Gardner&amp;author=RJ+Shepherd&amp;volume=106&amp;publication_year=1980&amp;pages=159-161&amp;pmid=18631716&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b48\"><span class=\"element-citation\">Christiansen FB.  <span class=\"ref-journal\">Population genetics of multiple loci.</span> New York: John Wiley and Sons; 2000. 365 pp. <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?title=Population+genetics+of+multiple+loci&amp;author=FB+Christiansen&amp;publication_year=2000&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li><li id=\"pbio-0030089-b49\"><span class=\"element-citation\">Bonneville JM, Hohn T.  <span class=\"ref-journal\">A reverse transcriptase for cauliflower mosaic virus. The state of the art, 1992. In: Shalka N, Goff S, editors. Reverse transcriptase.</span> Plainview (New York): Cold Spring Harbor Laboratory Press; 1993. pp. 357–390. <span class=\"nowrap\">[<a href=\"https://scholar.google.com/scholar_lookup?title=A+reverse+transcriptase+for+cauliflower+mosaic+virus.+The+state+of+the+art,+1992.+In:+Shalka+N,+Goff+S,+editors.+Reverse+transcriptase&amp;author=JM+Bonneville&amp;author=T+Hohn&amp;publication_year=1993&amp;\" ref=\"reftype=other&amp;article-id=1054884&amp;issue-id=115975&amp;journal-id=212&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar\" target=\"pmc_ext\">Google Scholar</a>]</span></span></li></ul></div></div></div><!--post-content--><div class=\"courtesy-note whole_rhythm small\"><hr/>Articles from <span class=\"acknowledgment-journal-title\">PLoS Biology</span> are provided here courtesy of <strong>Public Library of Science</strong></div></div>\n",
       "</div>\n",
       "<!-- Book content -->\n",
       "</div>\n",
       "<div class=\"four_col col last\" id=\"rightcolumn\">\n",
       "<!-- Custom content above discovery portlets -->\n",
       "<div class=\"col6\">\n",
       "</div>\n",
       "<div xmlns:np=\"http://ncbi.gov/portal/XSLT/namespace\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"><div class=\"format-menu\"><h2>Formats:</h2><ul><li class=\"selected\">Article</li> | <li><a href=\"/pmc/articles/PMC1054884/?report=reader\">PubReader</a></li> | <li class=\"epub-link\"><a href=\"/pmc/articles/PMC1054884/epub/\">ePub (beta)</a></li> | <li><a href=\"/pmc/articles/PMC1054884/pdf/pbio.0030089.pdf\">PDF (129K)</a></li> | <li><a class=\"citationexporter ctxp\" data-citationid=\"PMC1054884\" href=\"#\">Citation</a></li></ul></div></div><div class=\"share-buttons\" xmlns:np=\"http://ncbi.gov/portal/XSLT/namespace\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"><h2>Share</h2><ul><li class=\"facebook\"><a href=\"https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F\"><img alt=\"Share on Facebook\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047626\"/>\n",
       "                             Facebook\n",
       "                        </a></li><li class=\"twitter\"><a href=\"https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F&amp;text=Recombination%20Every%20Day%3A%20Abundant%20Recombination%20in%20a%20Virus%20during%20a%20Single%20Multi-Cellular%20Host%20Infection\"><img alt=\"Share on Twitter\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047627\"/>\n",
       "                             Twitter\n",
       "                        </a></li><li class=\"gplus\"><a href=\"https://plus.google.com/share?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F\"><img alt=\"Share on Google Plus\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047628\"/>\n",
       "                             Google+\n",
       "                        </a></li></ul></div>\n",
       "<div data-accid=\"PMC1054884\" data-aid=\"1054884\" data-aiid=\"1065690\" data-domain=\"plosbiol\" data-domainid=\"212\" data-iid=\"115975\" data-md5=\"a9769365bbb44745ffca360502c0eeca\" data-pmid=\"15737066\" id=\"ajax-portlets\"></div>\n",
       "<!-- Custom content below discovery portlets -->\n",
       "<div class=\"col7\">\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<!-- Custom content after all -->\n",
       "<div class=\"col8\">\n",
       "</div>\n",
       "<div class=\"col9\">\n",
       "</div>\n",
       "<script src=\"/corehtml/pmc/js/jquery.scrollTo-1.4.2.js\"></script>\n",
       "<script>\n",
       "    (function($){\n",
       "        $('.skiplink').each(function(i, item){\n",
       "            var href = $($(item).attr('href'));\n",
       "            href.attr('tabindex', '-1').addClass('skiptarget'); // ensure the target can receive focus\n",
       "            $(item).on('click', function(event){\n",
       "                event.preventDefault();\n",
       "                $.scrollTo(href, 0, {\n",
       "                    onAfter: function(){\n",
       "                        href.focus();\n",
       "                    }\n",
       "                });\n",
       "            });\n",
       "        });\n",
       "    })(jQuery);\n",
       "</script>\n",
       "<div id=\"body-link-poppers\"></div>\n",
       "</div>\n",
       "<div class=\"bottom\">\n",
       "<div id=\"NCBIFooter_dynamic\">\n",
       "<a class=\"help_desk\" href=\"\" id=\"help-desk-link\" target=\"_blank\">Support Center</a>\n",
       "<a class=\"help_desk\" href=\"https://support.ncbi.nlm.nih.gov/ics/support/KBList.asp?Time=2020-03-24T11:10:12-04:00&amp;Snapshot=%2Fprojects%2FPMC%2FPMCViewer@4.46&amp;Host=ptpmc102&amp;ncbi_phid=8A1B0C78E7A1C97100000000031B031B&amp;ncbi_session=8A1AB3A7E7A22D31_0927SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1054884%2F&amp;Db=pmc&amp;folderID=132&amp;Ncbi_App=pmc&amp;Page=literature&amp;style=classic&amp;deptID=28049\" id=\"help-desk-link\" target=\"_blank\">Support Center</a>\n",
       "</div>\n",
       "<div class=\"footer\" id=\"footer\">\n",
       "<div class=\"subfooter\"> </div><script src=\"/portal/portal3rc.fcgi/static/js/preloaderWidget.js\" type=\"text/javascript\"> </script>\n",
       "<div class=\"offscreen_noflow\" id=\"external-disclaimer\">\n",
       "        External link. Please review our <a href=\"https://www.nlm.nih.gov/privacy.html\">privacy policy</a>.\n",
       "    </div>\n",
       "<div class=\"contact_info\" id=\"ncbifooter\">\n",
       "<div id=\"footer-contents-right\">\n",
       "<div id=\"nlm_thumb_logo\">\n",
       "<a href=\"https://www.nlm.nih.gov\" title=\"NLM\">NLM</a>\n",
       "</div>\n",
       "<div id=\"nih_thumb_logo\">\n",
       "<a href=\"https://www.nih.gov\" title=\"NIH\">NIH</a>\n",
       "</div>\n",
       "<div id=\"hhs_thumb_logo\">\n",
       "<a href=\"https://www.hhs.gov\" title=\"DHHS\">DHHS</a>\n",
       "</div>\n",
       "<div id=\"usagov_thumb_logo\">\n",
       "<a href=\"https://www.usa.gov\" title=\"USA.gov\">USA.gov</a>\n",
       "</div>\n",
       "</div>\n",
       "<div id=\"footer-contents-left\">\n",
       "<p class=\"address vcard\">\n",
       "<span class=\"url\">\n",
       "<a class=\"fn url newdomain\" href=\"https://www.ncbi.nlm.nih.gov\">National Center for\n",
       "                        Biotechnology Information</a>,\n",
       "                </span> <span class=\"org url newdomain\"><a href=\"https://www.nlm.nih.gov/\">U.S. National Library of Medicine</a></span>\n",
       "<span class=\"adr\">\n",
       "<span class=\"street-address\">8600 Rockville Pike</span>, <span class=\"locality\">Bethesda</span>\n",
       "<span class=\"region\">MD</span>, <span class=\"postal-code\">20894</span>\n",
       "<span class=\"country-name\">USA</span>\n",
       "</span>\n",
       "</p>\n",
       "<a href=\"/home/about/policies.shtml\">Policies and Guidelines</a> | <a href=\"/home/about/contact.shtml\">Contact</a>\n",
       "</div>\n",
       "</div>\n",
       "<script src=\"/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIConfigJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js?v=1\" type=\"text/javascript\"> </script>\n",
       "<script src=\"/portal/portal3rc.fcgi/static/js/hfjs2.js\" type=\"text/javascript\"> </script>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<!--/.page-->\n",
       "</div>\n",
       "<!--/.wrap-->\n",
       "</div><!-- /.twelve_col -->\n",
       "</div>\n",
       "<!-- /.grid -->\n",
       "<span class=\"PAFAppResources\"></span>\n",
       "<!-- BESelector tab -->\n",
       "<noscript><img alt=\"statistics\" src=\"/stat?jsdisabled=true&amp;ncbi_db=pmc&amp;ncbi_pdid=article&amp;ncbi_acc=&amp;ncbi_domain=plosbiol&amp;ncbi_report=record&amp;ncbi_type=fulltext&amp;ncbi_objectid=&amp;ncbi_pcid=/articles/PMC1054884/&amp;ncbi_app=pmc\"/></noscript>\n",
       "<!-- usually for JS scripts at page bottom -->\n",
       "<!--<component id=\"PageFixtures\" label=\"styles\"></component>-->\n",
       "<!-- 8A1AB3A7E7A22D31_0927SID /projects/PMC/PMCViewer@4.46 ptpmc102 v4.1.r585844 Mon, May 06 2019 02:53:16 -->\n",
       "<script snapshot=\"pmc\" src=\"//static.pubmed.gov/portal/portal3rc.fcgi/4160049/js/3879255/4121861/3818874/4168176/3821238/4117325/4087685/4072593/4076480/3921943/4105668/4065628.js\" type=\"text/javascript\"></script></body>\n",
       "</html>"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_soup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Recombination Every Day: Abundant Recombination in a Virus during a Single Multi-Cellular Host Infection'"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# title\n",
    "tmp_soup.find('h1',attrs={'class':'content-title'}).text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Remy Froissart', 'Denis Roze', 'Marilyne Uzest', 'Lionel Galibert', 'Stephane Blanc', 'Yannis Michalakis']\n"
     ]
    }
   ],
   "source": [
    "# authors\n",
    "authors = []\n",
    "for a in tmp_soup.find('div',attrs={'class':'contrib-group fm-author'}).find_all('a'):\n",
    "    authors.append(a.text)\n",
    "print(authors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Abstract\n",
      "Viral recombination can dramatically impact evolution and epidemiology. In viruses, the recombination rate depends on the frequency of genetic exchange between different viral genomes within an infected host cell and on the frequency at which such co-infections occur. While the recombination rate has been recently evaluated in experimentally co-infected cell cultures for several viruses, direct quantification at the most biologically significant level, that of a host infection, is still lacking. This study fills this gap using the cauliflower mosaic virus as a model. We distributed four neutral markers along the viral genome, and co-inoculated host plants with marker-containing and wild-type viruses. The frequency of recombinant genomes was evaluated 21 d post-inoculation. On average, over 50% of viral genomes recovered after a single host infection were recombinants, clearly indicating that recombination is very frequent in this virus. Estimates of the recombination rate show that all regions of the genome are equally affected by this process. Assuming that ten viral replication cycles occurred during our experiment—based on data on the timing of coat protein detection—the per base and replication cycle recombination rate was on the order of 2 × 10−5 to 4 × 10−5. This first determination of a virus recombination rate during a single multi-cellular host infection indicates that recombination is very frequent in the everyday life of this virus.\n",
      "\n",
      "Introduction\n",
      "As increasing numbers of full-length viral sequences become available, recombinant or mosaic viruses are being recognized more frequently [1,2,3]. Recombination events have been demonstrated to be associated with viruses expanding their host range [4,5,6,7] or increasing their virulence [8,9], thus accompanying, or perhaps even being at the origin of, major changes during virus adaptation. It remains unclear, however, whether recombination events represent a highly frequent and significant phenomenon in the everyday life of these viruses.\n",
      "Viruses can exchange genetic material when at least two different viral genomes co-infect the same host cell. Progeny can then become hybrid through different mechanisms, such as reassortment of segments when the parental genomes are fragmented [10], intra-molecular recombination when polymerases switch templates (in RNA viruses) [11], or homologous or non-homologous recombination (in both RNA and DNA viruses). Quantification of viral recombination in multi-cellular organisms has been attempted under two distinct experimental approaches: in vitro (in cell cultures) [12,13,14,15], and in vivo (in live hosts) [16,17,18]. The in vitro approach, which has so far been applied only to animal viruses, allows the establishment of the “intrinsic” recombination rate in experimentally co-infected cells in cell cultures [14,15,19]. However, it does not necessarily reflect the situation in entire, living hosts, where the frequency of co-infected cells is poorly known and depends on many factors such as the size of the pathogen population, the relative frequency and distribution of the different variants, and host defense mechanisms preventing secondary infection of cells. The in vivo experimental approach is closer to biological conditions and may thus be more informative of what actually happens in “the real world.”\n",
      "However, as discussed below, numerous experimental constraints have so far precluded an actual quantification of the baseline rate of recombination. First, many experimental designs have used extreme positive selection, where only recombinant genomes were viable (e.g., [13,20,21]). Other studies did not use complementation techniques but detected recombinants by PCR within infected hosts or tissues [18,22,23,24,25], which provides information on their presence but not on their frequency in the viral population. So far, no quantitative PCR or other quantitative method has been applied to evaluate the number of recombinants appearing in an experimentally infected live host. Finally, recent methods based on sequence analysis inferred the population recombination rate, rather than the individual recombination rate [1,26,27]. While results from these methods certainly take in vivo recombination into account, there are other caveats: isolates have often been collected in different hosts—sometimes in different geographical regions—and sometimes the selective neutrality of sequence variation on which these estimates are based is not clearly established. Estimates from such studies by essence address the estimation of the recombination rate at a different evolutionary scale.\n",
      "Taken together, the currently available information indicates that no viral recombination rate has ever been estimated directly at time and space scales corresponding to a single multi-cellular host infection, although this level is most significant for the biology and evolution of viruses. This study intends to fill this gap by evaluating the recombination frequency of the cauliflower mosaic virus (CaMV) during a single passage in one of its host plants (the turnip Brassica rapa).\n",
      "CaMV is a pararetrovirus, which is a major grouping containing hepadnaviruses (e.g., hepatitis B virus), badnaviruses (e.g., banana streak virus), and caulimoviruses (e.g., CaMV). Pararetroviruses are characterized by a non-segmented double-stranded DNA genome. After entering the host cell nucleus, the viral DNA accumulates as a minichromosome [28] whose transcription is ensured by the host RNA polymerase II [29]. The CaMV genome consists in approximately 8,000 bp and encodes six viral gene products that have been detected in planta (Figure 1) [30]. Viral proteins P1 to P6 are expressed from two major transcripts, namely a 19S RNA, encoding P6, and a 35S RNA corresponding to the entire genome and serving as mRNA for proteins P1–P5 [31]. Using the pre-genomic 35S RNA as a matrix, the protein P5 (product of gene V) reverse-transcribes the genome into genomic DNA that is concomitantly encapsidated [30].\n",
      "The CaMV genome is a circular double-stranded DNA of 8,024 bp, represented in the figure by a double line. The thick arrows with different textures represent the organization of open reading frames I to VI, encoding proteins detected in planta. Markers a, b, c, and d were engineered at the positions indicated (see Materials and Methods for precise positions). The inner black arrows represent monocistronic 19S RNA and polycistronic 35S RNA produced by the cellular machinery. The nucleotide position 0 (numbering according to [44]) indicates the origin of replication via reverse transcription, which occurs in the direction indicated by the dotted outermost circle-like arrow. Reverse transcription is accomplished by the viral reverse transcriptase, using the 35S RNA as template [49].\n",
      "The detection of CaMV recombinants in turnip hosts has been reported numerous times. Some studies have demonstrated the appearance of infectious recombinant viral genomes after inoculation (i) of a host plant with two infectious or non-infectious parental clones [21,32,33,34,35] or (ii) of a transgenic plant containing one CaMV transgene with a CaMV genome missing the corresponding genomic region [36]. While the former revealed inter-genomic viral recombination, the latter demonstrated that CaMV can also recombine with transgenes within the host's genome. Another study based on phylogenetic analyses of various CaMV strains has clearly suggested different origins for different genomic regions and, hence, multiple recombination events during the evolution of this virus [37]. Indirect experimental evidence has indicated that, in some cases, CaMV recombination could occur within the host nucleus, between different viral minichromosomes, presumably through the action of the DNA repair cellular machinery [21,35]. Nevertheless, the mechanism of “template switching” during reverse transcription, predominant in all retroviruses, most certainly also applies to pararetroviruses. For this reason, and on the basis of numerous experimental data, CaMV is generally believed to recombine mostly in the cytoplasm of the host cell, by “legal” template switching between two pre-genomic RNA molecules [21,35,36,38,39], or “illegal” template switching between the 19S and the 35S RNA [36,40]. Under this hypothesis, recombination in CaMV could therefore be considered as operating on a linear template during reverse transcription, with the 5′ and 3′ extremities later ligated to circularize the genomic DNA (position 0 in Figure 1). The above cited studies clearly demonstrate that CaMV is able to recombine. However, since these studies are based on complementation techniques, non-quantitative detection, or phylogenetically based inferences of recombination, they do not inform us on whether recombination is an exceptional event or an “everyday” process shaping the genetic composition of CaMV populations.\n",
      "In the present work, we aimed at answering this question. To this end, we have constructed a CaMV genome with four genetic markers, demonstrated to be neutral in competition experiments. By co-inoculating host plants with equal amounts of wild-type and marker-containing CaMV particles, we have generated mixed populations in which impressive proportions of recombinants—distributed in several different classes corresponding to exchange of different genomic regions—have been detected and quantified. Altogether, the recombinant genomes averaged over 50% of the population. Further analysis of these data, assuming a number of viral replications during the infection period ranging from five to 20, indicates that the per nucleotide per replication cycle recombination rate of CaMV is of the same order of magnitude, i.e., on the order of a few 10−5, across the entire genome. We thereby provide the first quantification, to our knowledge, of the recombination rate in a virus population during a single passage in a single host.\n",
      "\n",
      "Results\n",
      "From Figure 1, and supposing that all marker-containing genomic regions can recombine, we could predict the detection and quantification of seven classes of recombinant genotypes: +bcd/a+++, a+cd/+b++, ab+d/++c+, abc+/+++d, ++cd/ab++, a++d/+bc+, and a+c+/+b+d. Indeed, all classes were detected, and their frequencies in the ten CaMV populations analyzed are summarized in Table 1.\n",
      "\n",
      "a Viral genomes were cloned and analyzed from ten of 24 co-infected plants\n",
      "\n",
      "b Seven possible classes of recombinants were predicted, their respective frequency in the population is expressed in percentage\n",
      "\n",
      "c The proportions of recombinants from the seven classes were added to estimate the total percentage of recombinant genomes within each tested plant\n",
      "Altogether, the proportion of recombinant genomes found in the mixed viral populations was astonishingly high and very similar in the ten co-infected plants analyzed (Table 1, last column), ranging between 44% (plant 5) to 60% (plants 7, 12, and 20), with a mean frequency (± standard error) of 53.8% ± 2.0%. This result indicates that recombination events are very frequent during the invasion of the host plant by CaMV and represents, to our knowledge, the first direct quantification of viral recombination during the infection of a live multi-cellular host.\n",
      "The inferred per generation recombination and interference rates, assuming that CaMV undergoes ten replication cycles during the 21 d between infection and sampling, are given for each of the ten plants in Table 2. Recombination rates between adjacent markers are large, on the order of 0.05 to 0.1. Taking the distance in nucleotides between markers into account yields an average recombination rate per nucleotide and generation on the order of 4 × 10−5. Interestingly, this recombination rate does not vary throughout the genome (Kruskal–Wallis test, p = 0.16).\n",
      "The various parameters are as follows: r\n",
      "1, recombination rate between markers a and b; r\n",
      "2, recombination rate between markers b and c; r\n",
      "3, recombination rate between markers c and d; i\n",
      "12, interference between crossovers in segments a–b and b–c; i\n",
      "23, interference between crossovers in segments b–c and c–d; i\n",
      "13, interference between crossovers in segments a–b and c–d; i\n",
      "123, second-order interference accounting for residual interference. The recombination rates are the maximum likelihood estimates (± 95% confidence intervals). The interference parameters were obtained numerically as explained in the Materials and Methods\n",
      "\n",
      "To relax the assumption of the number of replications during the 21 d, we calculated the recombination parameters assuming five or 20 generations. The effect of the number of generations on the estimates is linear: doubling the number of generations results in a halving of the recombination rate (detailed results not shown). For example, the average recombination rates r\n",
      "1, r\n",
      "2, and r\n",
      "3 assuming 20 generations were equal to 0.05, 0.04, and 0.025, respectively (compare with values in Table 2), yielding per nucleotide per generation recombination rates of 1.9 × 10−5, 2.2 × 10−5 and 1.6 × 10−5.\n",
      "Inspection of Table 2 also shows that first-order interference coefficients were in general negative, indicating that a crossing over in one genomic segment increases the probability that a crossing over will occur in another genomic segment, while the second-order coefficient parameter had an average value close to zero with a large variance. The mechanism leading to these results will be discussed in the following section.\n",
      "\n",
      "Discussion\n",
      "One major breakthrough in the work presented here lies in the space and time scales at which the experiments were performed. Indeed, the processes occurring within the course of a single infection of one multi-cellular host are of obvious biological relevance for any disease. Previous studies on viral recombination suffered from major drawbacks in this respect, basing their conclusions on experiments relying on complementation among non-infectious viruses or between viruses with undetermined relative fitness, on phylogenetically based analyses, or on experiments in cell cultures. For reasons detailed in the Introduction, the first two methods either do not provide information on the frequency of recombination, but only its occurrence, or address the question at a different temporal, and often spatial, scale. Results from cell cultures, on the other hand, impose cell co-infection by different viral variants, potentially overestimating the frequency of recombination events. Our study circumvents these limitations by analyzing viral genotypes sampled from infected plants after the course of a single infection, and therefore the invasion and co-infection of cells in various organs and tissues is very close to natural.\n",
      "More than half of the genomes (53.8% ± 2.0%; see Table 1) present in a CaMV population after a single passage in its host plant were identified as recombinants, and these data allowed us to infer a per nucleotide per generation recombination rate on the order of 2 × 10−5 to 4 × 10−5. The time length of one generation, i.e., the time required for a given genome to go from one replication to the next, is totally unknown in plant viruses. The only experimental data available on CaMV are based on the kinetics of gene expression in infected protoplasts, where the capsid protein is produced between 48 and 72 h [40]. The reverse transcription and the encapsidation of genomic DNA being two coupled phenomena [30], we judged it reasonable to assume a generation time of 2 d and, thus, an average of ten generations during our experiments. In case this estimate is mistaken, we have verified a linear relationship between r and the number of generations, thereby allowing an immediate adjustment of r if the CaMV generation time is more precisely established. At this point, we must consider that all cloned genomes may not have been through all the successive replication events potentially allowed by the timing of our experiments. It was previously shown that about 95% of CaMV mature virus particles accumulate in compact inclusion bodies [41], where they may be sequestered for a long time, as such inclusions are very frequent in all infected cells, including those in leaves that have been invaded by the virus population for several weeks. The viral population may thus present an age structure that could bias the estimation of the recombination rate. In order to minimize this bias, the clones we analyzed were collected in one young newly formed leaf, where the chances of finding genomes from “unsequestered lines” were assumed to be higher. In any case, our data analysis is conservative, since this age structure can only lead to an underestimation of the recombination rate.\n",
      "Our results show that interferences between pairs of loci are negative: a recombination event between two loci apparently increases the probability of recombination between another pair of loci. We believe that the most parsimonious explanation of these negative interferences is based on the way the infection builds up within plant hosts. Indeed, one can divide infected host cells into those infected by a single virus genotype and those infected by more than one viral genotype. In the former, analogous to clonal propagation, recombination is undetectable. In the latter, recombination is not only detectable but, as our results indicate, very frequent. Samples consisting of viruses resulting from a mixture of these two types of host cell infections will thus contain viruses with no recombination and viruses with several recombination events, thus yielding an impression of negative interference. These conceptual arguments are supported by mathematical models. It is indeed easy to show (detailed results not shown) that if a proportion F of the population reproduces clonally, analogous to single infections, while the remaining reproduces panmictically, negative interferences could be inferred even if they do not exist. For example, assuming a three-locus model with real recombination rates r\n",
      "1 and r\n",
      "2 and interference i\n",
      "12, the “apparent” recombination and interference parameters, would be r\n",
      "1 = (1 − F)r1, r\n",
      "2 = (1 − F)r\n",
      "2, and i\n",
      "12 = −(F − i\n",
      "12)/(1 − F). Interestingly, this example also shows that our estimates of the recombination rate are conservative: that a fraction F of host cells are singly infected while others are multiply infected leads to an underestimation of the recombination rate.\n",
      "As judged by r\n",
      "1, r\n",
      "2, and r\n",
      "3, calculated between markers a–b, b–c, and c–d, respectively, we found evidence for recombination through the entire CaMV genome. The values for r\n",
      "1,\n",
      "r\n",
      "2, and r\n",
      "3 are remarkably similar, hence the recombination sites seem to be evenly distributed along the genome. We considered the template-switching model as the major way recombinants are created in CaMV. As already mentioned in the Introduction, hot spots of template switching have been predicted at the position of the 5′ extremities of the 35S and 19S RNAs [21,36,42]. If other recombination mechanisms, such as that associated with second-strand DNA synthesis or with the host cell DNA repair machinery, act significantly, hot spots would be expected at the positions of the sequence interruption Δ1, Δ2, and Δ3 [43]. Due to the design of our experiment and the position of the four markers, we have no information on putative hot spots at positions corresponding to the 5′ end of the 35S RNA and to Δ1 (at nucleotide position 0). Nevertheless, the putative hot spots at the 5′ end of the 19S RNA and at Δ2 and Δ3 (nucleotide positions 4,220 and 1,635, respectively) fall between marker pairs c–d, b–c, and a–b, respectively. Our results indicate that either these hot spots are quantitatively equivalent—though predicted by different recombination mechanisms—or, more likely, that they simply do not exist. Whatever the explanation, what we observe is that the CaMV can exchange any portion of its genome, and thus any gene thereof, with an astonishingly high frequency during the course of a single host infection.\n",
      "To our knowledge, the viral recombination rate has never previously been quantified experimentally for a plant virus [3]. In contrast, retroviruses and particularly HIV-1 have been extensively investigated in that sense. As we have already discussed for these latter cases, the quantification of the intrinsic recombination rate was carried out in artificially co-infected cell cultures. The estimated intrinsic per nucleotide per generation recombination rate in HIV-1 is on the order of 10−4 [14,15,19], less than one order of magnitude higher than our estimation for CaMV. Because for various reasons detailed above we probably underestimate the within-host CaMV recombination rate, we believe that the intrinsic recombination rate in CaMV is higher and perhaps on the order of that of HIV.\n",
      "Other pararetroviruses such as plant badnaviruses or vertebrate hepadnaviruses have a similar cycle within their host cells, including steps of nuclear minichromosome, genomic size RNA synthesis, and reverse transcription and encapsidation. Nevertheless, vertebrate hepadnaviruses (e.g., hepatitis B virus) infect hosts that are very different from plants in their biology and physiology, and this could lead to a totally different frequency of cell co-infection during the development of the virus populations. Thus, even though our results can be informative for other pararetroviruses because of the viruses' shared biological characteristics, they should not be extrapolated to vertebrate pararetroviruses without caution.\n",
      "\n",
      "Materials and Methods\n",
      "\n",
      "We used the plasmid pCa37, which is the complete genome of the CaMV isolate Cabb-S, cloned into the pBR322 plasmid at the unique SalI restriction site [44]. To analyze recombination in different regions of the genome, we introduced four genetic markers: a, b, c, and d, at the positions 881, 3,539, 5,365, and 6,943, respectively, thus approximately at four cardinal points of the CaMV circular double-stranded DNA of 8,024 bp (Figure 1). All markers, each corresponding to a single nucleotide change, were introduced by PCR-directed mutagenesis in pCa37, and resulted in the duplication of previously unique restriction sites BsiWI, PstI, MluI, and SacI in a plasmid designated pMark-S. Because, in this study, we targeted the possible exchange of genes between viral genomes, all markers a, b, c, and d were introduced within coding regions corresponding to open reading frames I, IV, V, and VI, respectively. Another important concern was to quantify recombination in the absence of selection, i.e., to create neutral markers. Consequently all markers consist of synonymous mutations (see below).\n",
      "\n",
      "To generate the parental virus particles, plasmids pCa37 and pMark-S were mechanically inoculated into individual plants as previously described [33]. All plants were turnips (B. rapa cv, “Just Right”) grown under glasshouse conditions at 23 °C with a 16/8 (light/dark) photoperiod. Thirty days post-inoculation, all symptomatic leaves were harvested and viral particles were purified as described earlier [45].\n",
      "The resulting preparations of parental viruses, designated Cabb-S and Mark-S, were quantified by spectrometry using the formula described by Hull et al. [46]. We fixed the initial frequency of markers to a value of 0.5, and a solution containing 0.1 mg/ml of virus particles of both Cabb-S and Mark-S at a 1:1 ratio was prepared. Plantlets were co-infected by mechanical inoculation of two to three leaves with 20 μl of this virus solution, using abrasive Celite AFA (Fluka, Ronkonkoma, New York, United States). The mixed CaMV population was allowed to grow during 21 d of systemic infection.\n",
      "\n",
      "We designed an experimental protocol for quantifying marker frequency within a mixed Cabb-S/Mark-S virus population after a single passage in a host plant. Twenty-four individual plants, inoculated as above with equal amounts of Cabb-S and Mark-S, were harvested 21 d post-inoculation, when symptoms were fully developed. The viral DNA was purified from 200 mg of young newly formed infected leaves according to the protocol described previously [47]. After the precipitation step of this protocol, the viral DNA was resuspended and further purified with the Wizard DNA clean-up kit (Promega, Fitchburg, Wisconsin, United States) in TE 1X (100 mM Tris-HCl and 10 mM EDTA [pH 8]). Aliquots of viral DNA preparations were digested by restriction enzymes corresponding either to marker a, b, c, or d and submitted to a 1% agarose gel electrophoresis, colored by ethydium bromide and exposed to UV. Each individual restriction enzyme cut once in Cabb-S DNA and twice in Mark-S, thus generating DNA fragments of different sizes attributable to one or the other in the mixed population of CaMV genomes. After scanning the agarose gels, we estimated the relative frequency of the two genotypes in each viral DNA preparation and at each marker position, by densitometry using the NIH 1.62 Image program. The statistical analyses of the frequency of the four markers are described below.\n",
      "\n",
      "To identify and quantify the recombinants within the CaMV mixed populations, aliquots from ten of the 24 viral DNA preparations described above were digested by the restriction enzyme SalI, and directly cloned into pUC19 at the corresponding site. In each of the ten viral populations analyzed, 50 full-genome-length clones were digested separately by BsiWI, PstI, MluI, and SacI, to test for the presence of marker a, b, c, and d, respectively. In this experiment, with the marker representing an additional restriction site, we could easily distinguish between the Cabb-S and the Mark-S genotype at all four marker positions, upon agarose gel (1%) electrophoresis of the digested clones. Clones with none or all four markers were parental genotypes, whereas clones harboring 1, 2, or 3 markers were clearly recombinants. Due to the very high number of recombinants detected, markers eventually appearing or disappearing due to spontaneous mutations were neglected.\n",
      "\n",
      "Here we present the different methods we used to quantify recombination in the CaMV genome. Because all these methods assume that the different markers are neutral, we first discuss assumption.\n",
      "We used two datasets to test the neutrality of markers, both resulting from plants co-infected with a 1:1 ratio of Mark-S and Cabb-S. The first consisted of viral DNA densitometry data derived from 24 plants (described above), where for each plant we have an estimate of the frequency of each marker in the genome population. The second consisted of the restriction of 50 individual full-genome-length viral clones obtained from one co-infected plant (described above), yielding an estimate of the frequency of each marker, and this was repeated on ten different plants. The frequencies of the different markers were 0.508, 0.501, 0.516, and 0.507 for markers a, b, c, and d in the first dataset and 0.521, 0.518, 0.514, and 0.524 in the second dataset. We tested whether these frequencies were significantly different from the expected value under neutrality, 0.5, using either t-tests, for datasets where normality could not be rejected (seven out of eight cases), or Wilcoxon signed-rank non-parametric tests otherwise (marker c in the first dataset). In all cases p-values were larger than 0.05.\n",
      "There are several cautionary remarks regarding these analyses. First, in all cases we found an excess of markers. Unfortunately, the two datasets cannot be regarded as independent because, even though the methods through which the frequency estimates were obtained were different, the plants used in the second dataset were a subset of the plants of the first. We thus have only four independent estimates in each case, and there is minimal power to detect significant deviations from neutrality with such a small sample size. It should be noted at this stage that deviations from the expected value could also be caused either by slight deviations from the 1:1 ratio in the infecting mixed solution, or by deviations from that ratio in the frequency of the viral particles that actually get into the plants. Second, because of the relatively small sample sizes and low statistical power, the tests presented above could have detected only large deviations.\n",
      "The results clearly show, however, precisely that the markers do not have large effects, if any, and that therefore recombination estimates would be affected only very slightly by any hypothetical selective effects of the markers. Because of this, along with the fact that the introduced markers provoke silent substitutions in the CaMV genome, we assumed that markers were effectively neutral in the rest of the analysis.\n",
      "The dataset used to estimate the recombination frequency consisted of the 500 full-genome-length viral clones (50 from each of ten co-infected plants) individually genotyped for each of the four markers. As discussed in detail in the Results, recombination was very frequent and concerned all four markers. Indeed, approximately half of the genotyped clones exhibited a recombinant genotype. It was therefore meaningful to try to obtain quantitative estimates of recombination from our data.\n",
      "Our aim was to analyze viral recombination in a live host. Consequently, we had to deal with the fact that more than one viral replication cycle occurred during the 21 d that infection lasted in our experiment (we had to wait that long for the disease to develop and to be able to recover sufficient amounts of viral DNA from each infection). Based on the kinetics of gene expression [40], we postulate that each replication cycle lasts between 2 and 3 d, and that therefore seven to ten cycles occurred between infection and the sampling time. In case this assumption is incorrect, we did calculations assuming five, seven, ten, or 20 replication cycles during these 21 d. As shown, the results were not affected qualitatively, and only slightly quantitatively. It is important to note that we assumed that recombination occurred through a template-switching mechanism, and that therefore, from a recombination point of view, the CaMV genome is linear. The reverse transcription starts and finishes at the position 0 in Figure 1, which is the point of circularization of the DNA genome. This implies that changes between contiguous markers a–b, b–c, and c–d can be considered as true recombination whereas those between a and d cannot, as they may simply stem from circularization of DNA, during the synthesis of which the polymerase has switched template once anywhere between a–b, b–c, or c–d.\n",
      "To estimate the recombination rate between markers, we wrote recurrence equations describing the change in frequency of each genotype over one generation, assuming random mating and no selection (i.e., the standard Wright–Fisher population genetics model). We then expressed the frequency of all possible genotypes n generations later as a function of their initial frequency and of the recombination parameters. Subsequently we calculated the maximum likelihood estimates of the recombination parameters and their asymptotic variances given initial frequencies (we assumed that the two “parental” genotypes, Mark-S and Cabb-S, had equal initial frequencies of 0.5 and that all other genotypes had initial frequencies of zero) and frequencies after n generations (the observed frequencies; as stated above we used different values of n). All algebraic and numerical calculations were carried out with the software Mathematica.\n",
      "The recombination parameters are the recombination rates between two adjacent loci, e.g., r\n",
      "1 for the recombination rate between markers a and b, and the interference coefficients, e.g., i\n",
      "12 for interference between recombination events in the segments between markers a and b and b and c. To define these parameters we followed Christiansen [48], and in particular the recombination distributions for two, three, and four loci (respectively, Tables 2.7, 2.8, and 2.9 of [48]). It is important to realize that given the definitions of these parameters, the estimator of the recombination rate between two loci is not affected by the number of loci considered. In other words, we obtain the same estimation of the recombination rate between markers a and b whether we consider genotypic frequencies at just these two loci, or the frequencies at these two loci plus a third locus, or the complete information to which we have access, the four-marker genotypes. Information on additional loci only affects the estimates of the interference coefficients.\n",
      "It proved impossible to carry out the calculations for four loci algebraically. Instead, we used a computer program to calculate the expected genotypic frequencies at all four loci after n generations, given the above stated initial frequencies and specified recombination parameters. For each combination of recombination parameters we calculated a Euclidean distance between the vector of the expected genotypic frequencies and the observed genotypic frequencies, and considered that the estimated recombination parameters were those yielding the minimal Euclidean distance. In all cases, the estimated recombination rates between pairs of loci were equal to the second decimal to those estimated algebraically from data for three or two loci.\n",
      "\n",
      "Acknowledgments\n",
      "\n",
      "Competing interests. The authors have declared that no competing interests exist.\n",
      "\n",
      "Abbreviation\n",
      "\n",
      "References\n",
      "\n",
      "Formats:\n",
      "\n",
      "Share\n",
      "\n"
     ]
    }
   ],
   "source": [
    "sections = {}\n",
    "headers = tmp_soup.find_all('h2')\n",
    "for header in headers:\n",
    "    print(header.text)\n",
    "    paragraphs = header.parent.find_all('p')\n",
    "    for paragraph in paragraphs:\n",
    "        print(paragraph.text)\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hash value: 4fee63fd9f20798a7ceea04311081650\n"
     ]
    }
   ],
   "source": [
    "import random\n",
    "\n",
    "hash = random.getrandbits(128)\n",
    "\n",
    "print(\"hash value: %032x\" % hash)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "This script only scrapes articles from biorxiv, if not already downloaded.\n",
      "581 articles to scrape\n"
     ]
    }
   ],
   "source": [
    "source = 'biorxiv'\n",
    "print(\"\\nThis script only scrapes articles from {}, if not already downloaded.\".format(source))\n",
    "\n",
    "session = requests.session()\n",
    "\n",
    "# import list of articles in data folder\n",
    "existing_shas = [name.strip('.json') for name in get_file_names('data/{}'.format(source))]\n",
    "\n",
    "# import metadata dataframe\n",
    "df_meta = pd.read_csv('metadata.csv')\n",
    "\n",
    "'''\n",
    "sources = df_meta['source_x'].unique()\n",
    "for source in sources:\n",
    "    papers_df = df_meta[df_meta['source_x']==source][['sha','doi']]\n",
    "    papers_array = papers_df.to_numpy()\n",
    "    print('There are {} papers ids from {}.'.format(len(papers_array), source))\n",
    "'''\n",
    "\n",
    "missing_articles = df_meta[(df_meta['source_x']=='biorxiv')&(~df_meta['sha'].isin(existing_shas))]\n",
    "print(\"{} articles to scrape\".format(missing_articles.shape[0]))\n",
    "\n",
    "papers_df = missing_articles[['sha','doi']]\n",
    "papers_array = papers_df.to_numpy()\n",
    "indexes = papers_df.index.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching paper data at https://biorxiv.org/content/10.1101/030742v1.full\n",
      "What’s in my pot? Real-time species identification on the MinION™ 4e5ebec665ed176a51fac9eda002eb08\n",
      "Paper ID: 4e5ebec665ed176a51fac9eda002eb08 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/060434v1.full\n",
      "The ATP synthase subunit β (ATP5B) is an entry factor for the hepatitis E virus 7b3cb4af46c7492fb2cdd23e6e3bd751\n",
      "Paper ID: 7b3cb4af46c7492fb2cdd23e6e3bd751 saved to data/biorxiv\n",
      "------------------------------------------------------------\n",
      "Fetching paper data at https://biorxiv.org/content/10.1101/073098v1.full\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'text'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-115-353969646f51>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     16\u001b[0m     \u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m'https://biorxiv.org/content/'\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mrow\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'doi.org'\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m'v1.full'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0msource\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'biorxiv'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 18\u001b[1;33m         \u001b[0mpaper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mscrape_biorxiv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     19\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     20\u001b[0m         \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Error source not biorxiv. ({})'\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msource\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-112-2926c9ef0430>\u001b[0m in \u001b[0;36mscrape_biorxiv\u001b[1;34m(paper)\u001b[0m\n\u001b[0;32m      3\u001b[0m     \u001b[0mhtml\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'url'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m     \u001b[0mtmp_soup\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhtml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcontent\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'html.parser'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m     \u001b[0mpaper_webdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbiorxiv_html_paper_data_extractor\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtmp_soup\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m     \u001b[0mpaper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mchain\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpaper\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mpaper_webdata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mpaper\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-112-2926c9ef0430>\u001b[0m in \u001b[0;36mbiorxiv_html_paper_data_extractor\u001b[1;34m(soup)\u001b[0m\n\u001b[0;32m     17\u001b[0m     \u001b[0mauthors\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     18\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mauthor\u001b[0m \u001b[1;32min\u001b[0m \u001b[0msoup\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'highwire-citation-authors'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind_all\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'highwire-citation-author'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 19\u001b[1;33m         \u001b[0mauthor\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mauthor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'nlm-given-names'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m' '\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mauthor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'span'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mattrs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'class'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;34m'nlm-surname'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     20\u001b[0m         \u001b[0mauthors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mauthor\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     21\u001b[0m     \u001b[1;31m#print(authors)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'text'"
     ]
    }
   ],
   "source": [
    "# keep track of index\n",
    "i = 0\n",
    "modified = False\n",
    "for row in papers_array:\n",
    "    paper = {}\n",
    "    paper['source'] = source\n",
    "    # if no sha => generate one that doesn't exist.\n",
    "    if str(row[0]) == 'nan':\n",
    "        paper['id'] = generate_sha(existing_shas)\n",
    "        # save new hash and add to existing_shas\n",
    "        existing_shas.append(paper['id'])\n",
    "        df_meta.loc[indexes[i], 'sha'] = paper['id']\n",
    "        modified = True\n",
    "    else:\n",
    "        paper['id'] = row[0]\n",
    "    paper['url'] = 'https://biorxiv.org/content/' + row[1].strip('doi.org') + 'v1.full'\n",
    "    if source == 'biorxiv':\n",
    "        paper = scrape_biorxiv(paper)\n",
    "    else:\n",
    "        print('Error source not biorxiv. ({})'.format(source))\n",
    "    print(paper['title'],paper['id'])\n",
    "    '''\n",
    "    with open('data/{}/{}.json'.format(source,paper['id'])):\n",
    "        json.dump(paper, fp, indent=4)\n",
    "    '''\n",
    "    print('Paper ID: {} saved to data/{}'.format(paper['id'], source))\n",
    "    print('-'*60)\n",
    "    i += 1\n",
    "\n",
    "# if dataframe was changed, save it to csv\n",
    "if modified:\n",
    "    df_meta.to_csv('new_metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_meta.to_csv('new_metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nan"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.loc[indexes[i], 'sha']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
