<html>

<head>
  <style type="text/css">
    .article {
      width: 90%;
      vertical-align: middle;
      margin-left:5%
    }

    .authors{
      text-align: center;
    }

    .section {
      width: 100%;
      margin-top:10px;
    }

    #title {
      text-align: center;
    }

    .section td {
      text-align: left;
    }

    .subtitle {
      font-weight: bold;
    }

    img {
      width: 35%;
      height: 35%;
    }

    .figure {
      width:auto;
      border: 1px solid black;
      background-color: lightgrey;
      vertical-align: center;
      margin-left: 5%;
      margin-right: 5%;
      margin-top: 10px;
      margin-bottom: 10px;
    }

    .keyword {
      background-color: yellow;
    }

    td .image{
      text-align: center;
    }
  </style>
</head>

<body>
  <table class="article">
    <tr>
      <td id="title">
        <h1><span class="keyword">SIANN</span>: Strain Identification by Alignment to Near Neighbors</h1>
      </td>
    </tr>
    <tr>
      <td class="authors">
        <span class="author">Samuel S. Minot</span>,
        <span class="author">Stephen D. Turner</span>,
        <span class="author">Krista L. Ternus</span>,
        <span class="author">Dana R. Kadavy</span>
      </td>
    </tr>
    <tr>
      <td>
        <table id="abstract" class="section">
          <tr>
            <td>
              <h2>Abstract</h2>
            </td>
          </tr>
          <tr>
            <td>
              <p>Next-generation sequencing is increasingly being used to study samples composed of mixtures of organisms, such as in clinical applications where the presence of a pathogen at very low abundance may be highly important. We present an
                analytical method (SIANN: Strain Identification by Alignment to Near Neighbors) specifically designed to rapidly detect a set of target organisms in mixed samples that achieves a high degree of species- and strain-specificity by
                aligning short sequence reads to the genomes of near neighbor organisms, as well as that of the target. Empirical benchmarking alongside the current state-of-the-art methods shows an extremely high Positive Predictive Value, even at
                very low abundances of the target organism in a mixed sample. SIANN is available as an Illumina BaseSpace app, as well as through Signature Science, LLC. SIANN results are presented in a streamlined report designed to be
                comprehensible to the non-specialist user, providing a powerful tool for rapid species detection in a mixed sample. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with
                low computational requirements while delivering highly accurate results.</p>
            </td>
          </tr>
        </table>
      </td>
    </tr>
    <tr>
      <td>
        <table id="introdution" class="section">
          <tr>
            <td>
              <h2>Introduction</h2>
            </td>
          </tr>
          <tr>
            <td>
              <p>There are many different methods that characterize the mixture of organisms present within a metagenomic dataset. Such datasets are generated when a complex environmental sample is processed by a “next-generation” high-throughput
                genome sequencing protocol, and they consist of large numbers of short nucleotide sequences. Each sequence represents a small fragment of a randomly selected genome from the very large collection of genomes present in the source
                sample. Those sequences indicate the presence of one organism or another according to their similarity to a set of known reference genomes. While a given sequence may be unique to one species, it also may be found in diverse organisms
                across the tree of life. Therefore, one analytical challenge (among many) is to take that collection of sequences (likely numbering in the millions) and accurately determine what species are present in the sample. Here we describe a
                novel method (SIANN: Strain Identification by Alignment to Near Neighbors) that is specifically designed to rapidly detect a set of targeted organisms from a metagenomic dataset by aligning reads to genomic regions that are unique at
                the strain or species level.</p>
            </td>
          </tr>
          <tr>
            <td>
              <p>The analytical question motivating a particular piece of metagenomic bioinformatic analysis may vary widely by user and sample type ([ref-11,"Segata, et al., 2013"]). For example, the function of the human gut microbiome may depend
                on the relative abundance of hundreds of species of bacteria and the types of metabolic genes they contain ([ref-15,"Wu, et al., 2011"]; [ref-10,"Schloissnig, et al., 2013"]). In contrast, the clinical treatment of a patient may
                depend on whether or not a particular virus, or a consortium of co-infecting pathogens, is/are detected in their blood. It is this second class of presence/absence questions that SIANN is designed to address. SIANN is appropriate for
                situations in which a user wants to know whether a particular organism or set of organisms is present in a sample, but isn’t interested in the functions encoded in their genomes, the relative abundance of each organism, or any other
                more in-depth analysis.</p>
            </td>
          </tr>
        </table>
      </td>
    </tr>
    <tr>
      <td>
        <table id="methods" class="section">
          <tr>
            <td>
              <h2>Methods</h2>
            </td>
          </tr>
          <tr>
            <td>
              <table id="Approach" class="subsection">
                <tr>
                  <td>
                    <h3>Approach</h3>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>Metagenomic classification methods are based on a wide variety of theoretical underpinnings. The basic varieties include alignment of reads to various nucleotide databases or exact matching to nucleotide or protein signature
                      sequences (or kmers). A representative set of recent methods are described in [T1,"Table 1"] (also see [ref-2,"Bazinet & Cummings 2012"]).</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <table id="T1" class="figure">
                      <tr>
                        <td class="image">
                            <img src="https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T1.large.jpg" alt="Table 1.">
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <b>Table 1</b>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span class="description">Summary of methods for metagenomic classification.\n</span>
                        </td>
                      </tr>
                    </table>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>Overall, these methods are designed to either classify individual reads to, and/or predict the total abundance of, clades (e.g. genus or species) across the entire tree of life. They generally require reference databases that
                      are very large and/or require a large amount of processing to generate. The gap SIANN is designed to fill is when the entire tree of life is irrelevant, and only predefined subsets of organisms need to be detected. For an
                      underlying method we chose read alignment to diagnostic genomic regions because the algorithms for read alignment are highly parallelizable and have been optimized heavily by the community at large (the current implementation of
                      SIANN uses bowtie2 [[ref-7,"Langmead & Salzberg, 2012"]] for the alignment function, but can be adapted to any alignment algorithm). This approach is distinct from using clade-specific marker genes ([ref-12,"Segata, et al.,
                      2012"]) because unique regions that are larger, smaller, or outside of genes can also be used. Furthermore, this approach supports the rapid construction of custom databases using reference genome sets that require only minimal
                      user-supplied structure.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>To understand the principle at work, consider a set of reads that have been aligned to the genomes of several strains belonging to two species. Some regions of those genomes are species-specific, some are strain-specific, and
                      some are shared ([F1,"Figure 1a"]). When a set of reads is aligned to those genomes such that each read is placed in as many locations as it has a match (at a reasonably stringent threshold), visual inspection of the
                      distribution of reads yields an intuitive understanding of the true source organism as Species I/Strain B ([F1,"Figure 1b"]). If Strain B were not present in the reference database, it would still be clear that the organism was
                      an unknown strain of Species I.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <table id="F1" class="figure">
                      <tr>
                        <td class="image">
                          <img src="https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F1.large.jpg" alt="Figure 1.">
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <b>Figure 1</b>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span class="description">A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3).
                            B) A set of reads are aligned to these genomes, and the ones that align in a species- or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the
                            organism identified.\n</span>
                        </td>
                      </tr>
                    </table>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>The unique identification of a species or strain is quantified by the proportion of the genome that is determined to be species- or strain-specific (defined as reads that are aligned to regions that are species- or
                      strain-specific). Each species and strain is then assigned a numerical measure of the proportion that is covered by these diagnostic reads, and that proportional measure is compared to the ideal case, where sequences from a
                      single organism (generated in silico) are aligned against the database in an identical manner. After that normalization factor is applied, the resulting score indicates whether the source sample contained any of the organisms in
                      the reference database. The analysis is conducted independently on both the species and the strain level, so that if the true strain is not present in the database, the species of origin will still be identified. While many
                      methods consider the complete taxonomic tree and assign reads to the least common ancestor, SIANN considers only two taxonomic levels: species and strain, throwing out anything that is not unique at one of those levels and thus
                      obviating many of the confounding factors introduced by manually curated taxonomies.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>The example shown in [F1,"Figure 1b"] indicates that species-specific reads are identified as reads that align to one species (Species I, in that case) but not the other. If Species II were not present in the example shown in
                      [F1,"Figure 1b"], a much larger number of reads would be assigned as “species-specific,” when in fact those regions are shared with other species. Therefore, the ability of this method to identify strain- and species-specific
                      sequences is a direct function of the inclusion of near neighbors in the reference database. This characteristic is shared among many classification algorithms, but it is of particular note for this method when users have an
                      opportunity to construct their own database. In order to detect a target species with a high degree of specificity (reducing false positives), it is necessary to include other related species in the reference database. Only by
                      parallel alignment to those near neighbors can the redundant sequences be separated from the species-specific ones. For example, in order to detect Bacillus anthracis in a sample, it would be necessary to include other species
                      of Bacilli in the reference database so that the presence of B. cereus or B. thuringiensis in a sample does not lead to a false call for B. anthracis.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>The nomenclature of genus, species, and strain is potentially problematic because it does not correspond to a consistent degree of evolutionary distance or genomic distinctiveness. The ability to distinguish two organisms by
                      any method using genomic sequence data is proportional to the amount of each genome that is shared or unique. One might assume that any two organisms of the same species will have a relatively predictable amount of shared
                      genomic identity. However, some pairs of organisms from the same species may have less in common than other pairs of organisms from different species or even genera. This ambiguity impacts SIANN in two ways. If two organisms
                      have very little genomic sequence to distinguish them, the sensitivity of SIANN to detect either one will diminish (the rate of false negatives will increase as the likelihood of sequencing unique regions decreases). Conversely,
                      if an organism is extremely dissimilar to the near neighbors selected for the database, the specificity with which SIANN detects that organism will decline (the rate of false positives will increase as the number of related
                      genomes available in the database decreases). For example, if a database contained only E. coli and B.anthracis, a sample containing B. cereus would be misidentified as contraining B. anthracis. In the intended use case, a
                      database targeting B. anthracis would contain B. cereus and a number of other near neighbors to prevent that kind of misidentification. It would be convenient to say that an ideal database can be made by calculating the ideal
                      genetic distance between all references and then finding an ideal set of organisms to make up that database, but the behavior of any database will be governed by the particular genomes of the organisms it encounters in the wild.
                      Because not all organisms evolve in the same manner (differences in mutation rate, horizontal gene transfer, recombination, etc), the suitability of a database and method to detect a given organism can only be determined by
                      thorough validation and benchmarking, as well as updating the reference database as needed. Users of SIANN may construct their own custom databases to include newly identified genomes or specific subsets of genomes that best
                      suit their research interests.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>Steps to construct a custom database:\n</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>The files contained within each SIANN database are a compressed genomic index and a list containing the proportion of each reference genome that was found to be strain- or species-specific during database construction.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>To run SIANN:\n</p>
                  </td>
                </tr>
              </table>
            </td>
          </tr>
          <tr>
            <td>
              <table>
                <tr>
                  <td>
                    <h2>Benchmarking</h2>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>The performance of SIANN (version 1.6) was tested in comparison to the following state-of-the-art metagenomic classification programs: LMAT (version 1.2), MetaPhlAn (version 1.7.7), and Kraken (version 0.9.1b). All of the
                      programs in
                      [T1,"Table 1"] were investigated for this effort, and three were chosen based on their ability to run on our high-performance computing cluster with an execution time and memory requirement that would be suitable to a clinical
                      lab.
                      Each program was run on a set of 600 simulated datasets generated by MetaSim ([ref-9,"Richter, et al., 2008"]). Each dataset consisted of 15,000,000 reads (100bp single-ended) with Illumina-simulated error (fourth-degree
                      polynomial)
                      ([ref-6,"Korbel, et al., 2009"]). The 600 datasets were broken into 12 sets of 50 replicates. Each of the 12 sets contained organisms at different levels of abundance as shown in [T2,"Table 2"]. Organisms were specifically
                      chosen in
                      pairs so that the ability to distinguish these near neighbors could be determined. The abundances were staggered at 4-fold intervals so that a wide range could be evaluated. All known species of near neighbors for each of the 12
                      target organisms were included in the reference database used by SIANN for this benchmarking (“Target Pathogen Database”) and are shown in [app-1,"Appendix 1"].</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <table id="T2" class="figure">
                      <tr>
                        <td class="image">
                          <img src="https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/T2.large.jpg" alt="Table 2.">
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <b>Table 2</b>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span class="description"></span>
                        </td>
                      </tr>
                    </table>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>Each program outputs a distinct measure. Kraken and LMAT both count the reads assigned to each taxon, MetaPhlAn calculates the abundance, and SIANN outputs a measure of the proportion of diagnostic genomic regions present. To
                      put
                      these measures on an even footing, we empirically calculated the false positive rate for each method over all 600 samples, at each possible measure of output. Because each dataset is made up of known organisms, any result can be
                      classified as true or false. Therefore, for any possible result (say, 513 reads classified by LMAT or 1.6% abundance assigned by MetaPhlAn), one can calculate the proportion of calls with at least the same amount of support that
                      were
                      correct (True Positives/[True Positives + False Positives]), over all of the 600 datasets. That measure is commonly given as Positive Predictive Value (PPV). For each program, the results can be translated from the raw value
                      into a
                      PPV that is based on this empirical measure of error. The key item of interest is the PPV value for the results that we know to be true positives, the defined spike organisms. Another way of describing this approach is to say
                      that the
                      results of each program have been normalized to the false positive error rate that was empirically observed. If another set of samples were generated, the PPV vs. raw value curve ([F2,"Figure 2"]) would likely fall differently,
                      but in
                      this case it gives us a means of comparing a diverse set of methods against the same ground truth. If method 1 detects an organism with a higher PPV than method 2 does, it means that method 1 has fewer false positives in the
                      range
                      that it reports true positives, which is the definition of utility in this setting.</p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <table id="F2" class="figure">
                      <tr>
                        <td class="image">
                          <img src="https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F2.large.jpg" alt="Figure 2.">
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <b>Figure 2</b>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <span class="description">Relationship of reported value for each program (horizontal axis, log scale) to
                            the
                            empirically-determined Positive Predictive Value (PPV), shown on the vertical axis. While the exact values depend on the test data used, the general values at significant cutoff values (0.8, 0.9, 0.95 PPV) remain relatively
                            constant
                            across different datasets (data not shown).\n</span>
                        </td>
                      </tr>
                    </table>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>For each method, PPV was calculated as a function of raw output value. Briefly, this was done by compiling the output for all 600 samples, labeling each result as false or true based on the sample set that it came from, and
                      then
                      calculating (at each possible value of output) what the proportion of TP/[TP + FP] was for results with at least that level of raw output. Some simplification steps were taken, such as focusing on the species-level assignments
                      (for
                      comparison with methods that do not perform strain assignment), and only taking the top hit for each species from each dataset. Custom R and BASH scripts were used for the data compilation and analysis.</p>
                  </td>
                </tr>
              </table>
            </td>
          </tr>
        </table>
      </td>
    </tr>
  <tr>
    <td>
      <table id="results" class="section">
        <tr>
          <td>
            <h2>Results</h2>
          </td>
        </tr>
        <tr>
          <td>
            <p>The relationship of raw output value to PPV is shown for each of the four methods in [F2,"Figure 2"]. The point at which PPV is very close to 1 (where 95% of results are true positives) is ∼41,000 reads for Kraken, ∼2,800 reads for
              LMAT, ∼38% abundance for MetaPhlAn, and 0.21 for SIANN. For SIANN this means that having 38% of the species-unique genome covered by reads resulted in the vast majority of calls being accurate.</p>
          </td>
        </tr>
        <tr>
          <td>
            <p>For read-assignment methods (such as LMAT and Kraken), manual inspection of the results may yield a different understanding of confidence than is presented here, or in any automated analysis. For example, while each read that is
              assigned by LMAT and Kraken fall above a certain cutoff for species-specificity, some individual reads may be much more specific than others. One could identify a read that aligns to a single species of bacteria with 100% accuracy
              over its 300bp length, with the next closest match being only 90% similar. It is extremely unlikely that a 300bp exact match would arise due to random chance, and so the user could say with confidence that the organism of interest is
              found within the sequence data (not considering contamination, horizontal gene transfer, etc). However, such an approach is not currently implemented in an automated method, and many of the steps needed to make that assertion are
              performed manually by a domain expert, including alignment to near neighbors and ensuring that the read does not fall within a transposon, plasmid, etc. Therefore, while one could say that a single read is all that is needed to state
              with high PPV that an organism is present, the amount of reads assigned in an automated manner needed to achieve that level of PPV will number in the thousands ([F2,"Fig 2"]).</p>
          </td>
        </tr>
        <tr>
          <td>
            <p>The next phase of benchmarking was to determine how many raw input reads were needed to achieve the threshold for high PPV. To demonstrate this we plotted the known abundance of each spike organism against the PPV value generated by
              each method ([F3,"Figure 3"]). Each point (an organism at a known level of abundance) is comprised of a maximum of 50 replicates, where the diameter of each point increases with an increasing number of replicates. For demonstration
              purposes we are showing two pairs of bacteria and three viruses. Recall that for each of the pairs of bacteria (and the two poxviruses) any sample containing one did not contain the other (as shown in [T1,"Table 1"]). The empty boxes
              result from the organisms not being called at any abundance. For MetaPhlAn, that is a result of no viruses being included in the version of the reference database available for this analysis. Kraken assigned no reads to Hanta virus
              because viral RNA genomes were not included in this version of the reference database (personal communication with D. Wood). This emphasizes the point that a) the ability to create custom databases targeting organisms of interest can
              be valuable, and b) the performance of any method must be benchmarked against each potential target of interest.</p>
          </td>
        </tr>
        <tr>
          <td>
            <table id="F3" class="figure">
              <tr>
                <td class="image">
                  <img src="https://www.biorxiv.org/content/biorxiv/early/2014/01/10/001727/F3.large.jpg" alt="Figure 3.">
                </td>
              </tr>
              <tr>
                <td>
                  <b>Figure 3</b>
                </td>
              </tr>
              <tr>
                <td>
                  <span class="description">The Positive Predictive Value (PPV, vertical axis) is shown for each organism (boxes
                    on right), at each level of known abundance (horizontal axis, see Table 2), and each program (boxes at top), across a maximum of 50 replicates (indicated by the size of each point). Note that the reference database for MetaPhlAn
                    does not include viruses, and the reference database for Kraken does not include RNA viruses\n</span>
                </td>
              </tr>
            </table>
          </td>
        </tr>
        <tr>
          <td>
            <p>All methods were able identify the bulk of organisms in their databases at high abundances (75% and 18%, [F3,"Figure 3"]), however performance varied considerably at lower abundances and depended on the particular organism and method
              used. SIANN detected each organism at high confidence, even at levels as low as 0.3% and 0.07% of the total.</p>
          </td>
        </tr>
      </table>
    </td>
  </tr>
  <tr>
    <td>
      <table id="discussion" class="section">
        <tr>
          <td>
            <h2>Discussion</h2>
          </td>
        </tr>
        <tr>
          <td>
            <p>The process of detecting trace amounts of a specific organism in a complex mixture of DNA is challenging enough for an expert, but that pales in comparison to the difficulty of accomplishing the same certainty of detection in an
              automated manner. The results presented here show that SIANN rapidly detects the presence of a given set of organisms with a high degree of specificity and sensitivity. For example, at the 95% confidence (PPV) cutoff of 0.2, SIANN
              reliably detects all of the organisms tested here at as low as 0.3% abundance. This strong performance is likely due to the fact that SIANN is able to use a method (read alignment to whole genomes) that would be far too
              computationally costly if it were applied to the entire collection of known genomes. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements
              while delivering highly accurate results.</p>
          </td>
        </tr>
        <tr>
          <td>
            <p>SIANN is available on Illumina’s BaseSpace ([http://www.basespace.illumina.com,"www.basespace.illumina.com"]) as a NativeApp, with the database tested here ([app-1,"Appendix 1"]), as well as a database made from the NCBI
              representative set of prokaryotic genomes ([ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/,"ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/"]) ([app-2,"Appendix 2"]) and the complete set of NCBI viral genomes
              ([ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/,"ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/"]) ([app-3,"Appendix 3"]).</p>
          </td>
        </tr>
        <tr>
          <td>
            <p>BaseSpace was chosen as an appropriate release platform because while the entire set of software and dependencies can be deployed by the user from within a graphical user interface, the actual computation takes place in a controlled
              ‘cloud’ environment. Such a distribution strategy obviates the need to satisfy the multiple software or OS dependencies that often arises with academic computational methods. Results for SIANN are compiled into a report format,
              showing both the organisms that surpass 95% confidence, as well as the closest strain match for each species. The default view masks the raw data output, so that the results are human-readable and do not present extraneous
              information. While the code for execution and database-construction on a users system is available from Signature Science, LLC, additional databases on the BaseSpace platform can be made available upon request.</p>
          </td>
        </tr>
        <tr>
          <td>
            <p>There is a neverending list of questions that one could ask of metagenomic sequencing data generated from important samples. Instead of answering them all, we demonstrate a technique with a very narrow focus that is able to report
              with a high degree of confidence whether a given set of organisms is present in a sample. These results are presented to the user in a comprehensible format, and accessible on a commonly-used web platform. The world of bioinformatics
              will continue to progress and develop more sophisticated tools for metagenomic analysis, and we hope that the utility of SIANN will convince others to package and benchmark their tools in a way that they can be used with confidence by
              the larger public, as well as the research community.</p>
          </td>
        </tr>
      </table>
    </td>
  </tr>
  <tr>
    <td>
      <table id="references" class="section">
        <tr>
          <td>
            <h2>References</h2>
          </td>
        </tr>
        <tr>
          <td><span class="reference">Scalable metagenomic taxonomy classification using a reference genome database, <span class="author">Ames SK</span>, <span class="author">Hysom DA</span>, <span class="author">Gardner SN</span>, <span
                class="author">Lloyd GS</span>, <span class="author">Gokhale MB</span>, <span class="author">Allen JE</span>, 2013, <a href="/lookup/external-ref?access_num=10.1093/bioinformatics/btt389&link_type=DOI">CrossRef</a>, <a
                href="/lookup/external-ref?access_num=23828782&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">A comparative evaluation of sequence classification programs, <span class="author">Bazinet AL</span>, <span class="author">Cummings MP</span>, 2012, <a
                href="/lookup/external-ref?access_num=10.1186/1471-2105-13-92&link_type=DOI">CrossRef</a>, <a href="/lookup/external-ref?access_num=22574964&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a></span>
          </td>
        </tr>
        <tr>
          <td><span class="reference">Rapid phylogenetic and functional classification of short genomic fragments with signature peptides, <span class="author">Berendzen J</span>, <span class="author">Bruno WJ</span>, <span class="author">Cohn
                JD</span>, <span class="author">Hengartner NW</span>, <span class="author">Kuske CR</span>, <span class="author">McMahon BH</span>, <span class="author">Wolinsky MA</span>, <span class="author">Xie G</span>, 2012, <a
                href="/lookup/external-ref?access_num=10.1186/1756-0500-5-460&link_type=DOI">CrossRef</a>, <a href="/lookup/external-ref?access_num=22925230&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a></span>
          </td>
        </tr>
        <tr>
          <td><span class="reference">Phymm and PhymmBL: metagenomic phylogenetic classification with interpolated Markov models, <span class="author">Brady A</span>, <span class="author">Salzberg SL</span>, 2009, <a
                href="/lookup/external-ref?access_num=10.1038/nmeth.1358&link_type=DOI">CrossRef</a>, <a href="/lookup/external-ref?access_num=19648916&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a>, <a
                href="/lookup/external-ref?access_num=000269382900018&link_type=ISI">Web of Science</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Integrative analysis of environmental sequences using MEGAN4, <span class="author">Huson DH</span>, <span class="author">Mitra S</span>, <span class="author">Weber N</span>, <span class="author">Ruscheweyh
                H</span>, <span class="author">Schuster SC</span>, 2011, <a
                href="/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NjoiZ2Vub21lIjtzOjU6InJlc2lkIjtzOjk6IjIxLzkvMTU1MiI7czo0OiJhdG9tIjtzOjM3OiIvYmlvcnhpdi9lYXJseS8yMDE0LzAxLzEwLzAwMTcyNy5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=">Abstract/FREE
                Full Text</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data, <span class="author">Korbel JO</span>, <span
                class="author">Abyzov A</span>, <span class="author">Mu XJ</span>, <span class="author">Carriero N</span>, <span class="author">Cayting P</span>, <span class="author">Zhang Z</span>, <span class="author">Snyder M</span>, <span
                class="author">Gerstein MB</span>, 2009, <a href="/lookup/external-ref?access_num=10.1186/gb-2009-10-2-r23&link_type=DOI">CrossRef</a>, <a
                href="/lookup/external-ref?access_num=19236709&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Fast gapped-read alignment with Bowtie 2, <span class="author">Langmead B</span>, <span class="author">Salzberg S</span>, 2012, <a
                href="/lookup/external-ref?access_num=10.1038/nmeth.1923&link_type=DOI">CrossRef</a>, <a href="/lookup/external-ref?access_num=22388286&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences, <span class="author">Liu B</span>, <span class="author">Gibbons T</span>, <span class="author">Ghodsi M</span>, <span
                class="author">Treangen T</span>, <span class="author">Pop M</span>, 2011, <a href="/lookup/external-ref?access_num=10.1186/1471-2164-12-S2-S4&link_type=DOI">CrossRef</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">MetaSim: a sequencing simulator for genomics and metagenomics, <span class="author">Richter DC</span>, <span class="author">Ott F</span>, <span class="author">Auch AF</span>, <span class="author">Schmid
                R</span>, <span class="author">Huson DH</span>, 2008, <a href="/lookup/external-ref?access_num=10.1371/journal.pone.0003373&link_type=DOI">CrossRef</a>, <a
                href="/lookup/external-ref?access_num=18841204&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Genomic variation landscape of the human gut microbiome, <span class="author">Schloissnig S</span>, <span class="author">Arumugam M</span>, <span class="author">Sunagawa S</span>, <span class="author">Mitreva
                M</span>, <span class="author">Tap J</span>, <span class="author">Zhu A</span>, <span class="author">Waller A</span>, <span class="author">Mende DR</span>, <span class="author">Kultima JR</span>, <span class="author">Martin
                J</span>, <span class="author">Kota K</span>, <span class="author">Sunyaev SR</span>, <span class="author">Weinstock GM</span>, <span class="author">Bork P</span>, 2013, <a
                href="/lookup/external-ref?access_num=10.1038/nature11711&link_type=DOI">CrossRef</a>, <a href="/lookup/external-ref?access_num=23222524&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a>, <a
                href="/lookup/external-ref?access_num=000312933800027&link_type=ISI">Web of Science</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Computational meta’omics for microbial community studies, <span class="author">Segata N</span>, <span class="author">Boernigen D</span>, <span class="author">Tickle TL</span>, <span class="author">Morgan
                XC</span>, <span class="author">Garrett WS</span>, <span class="author">Huttenhower C</span>, 2013, <a href="/lookup/external-ref?access_num=10.1038/msb.2013.22&link_type=DOI">CrossRef</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Metagenomic microbial community profiling using unique clade-specific marker genes, <span class="author">Segata N</span>, <span class="author">Waldron L</span>, <span class="author">Ballarini A</span>, <span
                class="author">Narasimhan V</span>, <span class="author">Jousson O</span>, <span class="author">Huttenhower C</span>, 2012, <a href="/lookup/external-ref?access_num=10.1038/nmeth.2066&link_type=DOI">CrossRef</a>, <a
                href="/lookup/external-ref?access_num=22688413&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a>, <a href="/lookup/external-ref?access_num=000307015700022&link_type=ISI">Web of Science</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Metagenomic species profiling using universal phylogenetic marker genes, <span class="author">Sunagawa S</span>, <span class="author">Mende DR</span>, <span class="author">Zeller G</span>, <span
                class="author">Izquierdo-Carrasco F</span>, <span class="author">Berger SA</span>, <span class="author">Kultima JR</span>, <span class="author">Coelho LP</span>, <span class="author">Arumugam M</span>, <span class="author">Tap
                J</span>, <span class="author">Nielsen HB</span>, <span class="author">Rasmussen S</span>, <span class="author">Brunak S</span>, <span class="author">Pedersen O</span>, <span class="author">Guarner F</span>, <span class="author">de
                Vos WM</span>, <span class="author">Wang J</span>, <span class="author">Li J</span>, <span class="author">Doré J</span>, <span class="author">Ehrlich SD</span>, <span class="author">Stamatakis A</span>, <span class="author">Bork
                P</span>, 2013, <a href="/lookup/external-ref?access_num=10.1038/nmeth.2693&link_type=DOI">CrossRef</a>, <a
                href="/lookup/external-ref?access_num=24141494&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2014%2F01%2F10%2F001727.atom">PubMed</a>, <a href="/lookup/external-ref?access_num=000327698100019&link_type=ISI">Web of Science</a></span></td>
        </tr>
        <tr>
          <td><span class="reference">Ultrafast metagenomic sequence classification using exact alignments, <span class="author">Wood DE</span>, <span class="author">Salzberg SL</span>, </span></td>
        </tr>
        <tr>
          <td><span class="reference">Linking long-term dietary patterns with gut microbial enterotypes, <span class="author">Wu GD</span>, <span class="author">Chen J</span>, <span class="author">Hoffmann C</span>, <span class="author">Bittinger
                K</span>, <span class="author">Chen YY</span>, <span class="author">Keilbaugh SA</span>, <span class="author">Bewtra M</span>, <span class="author">Knights D</span>, <span class="author">Walters WA</span>, <span class="author">Knight
                R</span>, <span class="author">Sinha R</span>, <span class="author">Gilroy E</span>, <span class="author">Gupta K</span>, <span class="author">Baldassano R</span>, <span class="author">Nessel L</span>, <span class="author">Li
                H</span>, <span class="author">Bushman FD</span>, <span class="author">Lewis JD</span>, 2011, <a
                href="/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEyOiIzMzQvNjA1Mi8xMDUiO3M6NDoiYXRvbSI7czozNzoiL2Jpb3J4aXYvZWFybHkvMjAxNC8wMS8xMC8wMDE3MjcuYXRvbSI7fXM6ODoiZnJhZ21lbnQiO3M6MDoiIjt9">Abstract/FREE
                Full Text</a></span></td>
        </tr>
      </table>
    </td>
  </tr>
  </table>
</body>

</html>
